<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:27 EDT 2013 -->
<TITLE>
ConfigFileParser
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="ConfigFileParser";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ConfigFileParser.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?broad/core/parser/ConfigFileParser.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ConfigFileParser.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
broad.core.parser</FONT>
<BR>
Class ConfigFileParser</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>broad.core.parser.ConfigFileParser</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>ConfigFileParser</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
This class will parse the config file provided to it.
 Helper class to broad.core.util.PipelineAutomator
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>skadri</DD>
</DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This is a helper class that maps the values of species-specific basic options to their value</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser">ConfigFileParser.Bowtie2Options</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This is a helper class is a data structure that maps a bowtie2 flag to its value</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This is a helper class is a data structure that maps a DGE flag or option to its value or path</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.MergeSamples.html" title="class in broad.core.parser">ConfigFileParser.MergeSamples</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets of samples to merge into one bam file (leaving separate bam files too)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.NovoalignOptions.html" title="class in broad.core.parser">ConfigFileParser.NovoalignOptions</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This is a helper class is a data structure that maps a novoalign flag to its value</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.RNAClass.html" title="class in broad.core.parser">ConfigFileParser.RNAClass</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This is a helper class is a data structure mapping an RNA class to its fasta file path</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser">ConfigFileParser.TophatOptions</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This is a helper class is a data structure that maps a tophat flag to its value</TD>
</TR>
</TABLE>
&nbsp;<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#basicOptions">basicOptions</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#DEFAULT_FASTQ_READ_PAIR_NUMBER_DELIMITER">DEFAULT_FASTQ_READ_PAIR_NUMBER_DELIMITER</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#dgeOptions">dgeOptions</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#fragmentSizeOptions">fragmentSizeOptions</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_ALIGN_TO_TRANSCRIPTS">OPTION_ALIGN_TO_TRANSCRIPTS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to align to a fasta file of transcript sequences</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_ANNOTATION">OPTION_ANNOTATION</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for genome annotation in gtf format for use by tophat</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_BED_FILE_FOR_WIG">OPTION_BED_FILE_FOR_WIG</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to write fragment endpoints to wig and bigwig file
 Writes counts for positions overlapping genes in this bed file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_BOWTIE2_BUILD_PATH">OPTION_BOWTIE2_BUILD_PATH</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for bowtie2-build executable file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_BOWTIE2_OPTIONS">OPTION_BOWTIE2_OPTIONS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide options to bowtie2</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_BOWTIE2_PATH">OPTION_BOWTIE2_PATH</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for bowtie2 executable file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_CHR_SIZE_FILE">OPTION_CHR_SIZE_FILE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Chromosome size file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_FASTQ_READ_PAIR_NUMBER_DELIMITER">OPTION_FASTQ_READ_PAIR_NUMBER_DELIMITER</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_FASTX_DIRECTORY">OPTION_FASTX_DIRECTORY</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Directory containing fastx binaries</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_FILTER_RRNA">OPTION_FILTER_RRNA</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide a fasta file of rRNA sequences for depletion</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_FRAGMENT_SIZE_DIST_OPTIONS">OPTION_FRAGMENT_SIZE_DIST_OPTIONS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide parameters to fragment size distribution calculation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_ASSEMBLY">OPTION_GENOME_ASSEMBLY</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for genome assembly name e.g.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_BOWTIE">OPTION_GENOME_BOWTIE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for prefix of genome bowtie index files</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_FASTA">OPTION_GENOME_FASTA</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for path to genome fasta file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_NOVOINDEX">OPTION_GENOME_NOVOINDEX</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for novoindex of genome</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.Object</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_SPACE_SIZE_FILE">OPTION_GENOME_SPACE_SIZE_FILE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for chromosome size file for genomic space stats</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_GLOBAL_STATS_JAR">OPTION_GLOBAL_STATS_JAR</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Jar file to compute alignment global stats</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_IGVTOOLS_PATH">OPTION_IGVTOOLS_PATH</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for igvtools executable</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_MERGE_SAMPLES">OPTION_MERGE_SAMPLES</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Merge a list of samples.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_NOVOALIGN_OPTIONS">OPTION_NOVOALIGN_OPTIONS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide options to novoalign e.g.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_NOVOALIGN_PATH">OPTION_NOVOALIGN_PATH</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for novoalign executable</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_COLLECT_RNASEQ_METRICS">OPTION_PICARD_COLLECT_RNASEQ_METRICS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide an alternative jar file for Picard program CollectRnaSeqMetrics</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_DIRECTORY">OPTION_PICARD_DIRECTORY</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for directory containing picard executables</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_REF_FLAT">OPTION_PICARD_REF_FLAT</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide ref_flat annotation to Picard</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_RIBOSOMAL_INTERVALS">OPTION_PICARD_RIBOSOMAL_INTERVALS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide ribosomal intervals to Picard</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_STRAND_SPECIFICITY">OPTION_PICARD_STRAND_SPECIFICITY</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide strand specificity argument to Picard</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_QUEUE_NAME">OPTION_QUEUE_NAME</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_RNA_CLASS">OPTION_RNA_CLASS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide an RNA class fasta file e.g.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_SAMTOOLS_PATH">OPTION_SAMTOOLS_PATH</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for samtools executable</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_SEQUENCING_ADAPTER_READ_1">OPTION_SEQUENCING_ADAPTER_READ_1</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sequencing adapter sequence for read 1</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_SEQUENCING_ADAPTER_READ_2">OPTION_SEQUENCING_ADAPTER_READ_2</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sequencing adapter sequence for read 2</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_TOPHAT_OPTIONS">OPTION_TOPHAT_OPTIONS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option to provide options to tophat e.g.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_TOPHAT_PATH">OPTION_TOPHAT_PATH</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for tophat executable file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.Object</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_TRANSCRIPTOME_SPACE_BED_FILE">OPTION_TRANSCRIPTOME_SPACE_BED_FILE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Config file option for bed annotation for transcriptome space stats</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_WIG_TO_BIGWIG_PATH">OPTION_WIG_TO_BIGWIG_PATH</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Path to UCSC WigToBigWig executable</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#OPTION_WIG_WRITER_JAR">OPTION_WIG_WRITER_JAR</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;WigWriter jar file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#START_OF_STRING">START_OF_STRING</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#ConfigFileParser(java.lang.String)">ConfigFileParser</A></B>(java.lang.String&nbsp;configFileName)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor will read the input config file and store the commands and parameters in the respective</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#addCommand(java.lang.String)">addCommand</A></B>(java.lang.String&nbsp;nextLine)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds the command in the current line if it is not prefixed with a "#"</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#addSpecies(java.lang.String)">addSpecies</A></B>(java.lang.String&nbsp;nextLine)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds the species</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#getAllCommands()">getAllCommands</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns all commands</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#getCommand(int)">getCommand</A></B>(int&nbsp;index)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the command at the specified index</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Set&lt;java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#getRunSpecificDGEOptions()">getRunSpecificDGEOptions</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns all run specific DGE options</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.html#hasCommandFor(java.lang.String)">hasCommandFor</A></B>(java.lang.String&nbsp;command)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if the specified command was included as runnable in the file</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="START_OF_STRING"><!-- --></A><H3>
START_OF_STRING</H3>
<PRE>
public static int <B>START_OF_STRING</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="basicOptions"><!-- --></A><H3>
basicOptions</H3>
<PRE>
public <A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A> <B>basicOptions</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="fragmentSizeOptions"><!-- --></A><H3>
fragmentSizeOptions</H3>
<PRE>
public <A HREF="../../../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A> <B>fragmentSizeOptions</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="dgeOptions"><!-- --></A><H3>
dgeOptions</H3>
<PRE>
public <A HREF="../../../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A> <B>dgeOptions</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_GENOME_FASTA"><!-- --></A><H3>
OPTION_GENOME_FASTA</H3>
<PRE>
public static java.lang.String <B>OPTION_GENOME_FASTA</B></PRE>
<DL>
<DD>Config file option for path to genome fasta file
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_QUEUE_NAME"><!-- --></A><H3>
OPTION_QUEUE_NAME</H3>
<PRE>
public static java.lang.String <B>OPTION_QUEUE_NAME</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_GENOME_BOWTIE"><!-- --></A><H3>
OPTION_GENOME_BOWTIE</H3>
<PRE>
public static java.lang.String <B>OPTION_GENOME_BOWTIE</B></PRE>
<DL>
<DD>Config file option for prefix of genome bowtie index files
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_GENOME_ASSEMBLY"><!-- --></A><H3>
OPTION_GENOME_ASSEMBLY</H3>
<PRE>
public static java.lang.String <B>OPTION_GENOME_ASSEMBLY</B></PRE>
<DL>
<DD>Config file option for genome assembly name e.g. mm9
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_ANNOTATION"><!-- --></A><H3>
OPTION_ANNOTATION</H3>
<PRE>
public static java.lang.String <B>OPTION_ANNOTATION</B></PRE>
<DL>
<DD>Config file option for genome annotation in gtf format for use by tophat
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_RNA_CLASS"><!-- --></A><H3>
OPTION_RNA_CLASS</H3>
<PRE>
public static java.lang.String <B>OPTION_RNA_CLASS</B></PRE>
<DL>
<DD>Config file option to provide an RNA class fasta file e.g. rna_classes mRNA mRNA.fa
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_FILTER_RRNA"><!-- --></A><H3>
OPTION_FILTER_RRNA</H3>
<PRE>
public static java.lang.String <B>OPTION_FILTER_RRNA</B></PRE>
<DL>
<DD>Config file option to provide a fasta file of rRNA sequences for depletion
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_ALIGN_TO_TRANSCRIPTS"><!-- --></A><H3>
OPTION_ALIGN_TO_TRANSCRIPTS</H3>
<PRE>
public static java.lang.String <B>OPTION_ALIGN_TO_TRANSCRIPTS</B></PRE>
<DL>
<DD>Config file option to align to a fasta file of transcript sequences
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_MERGE_SAMPLES"><!-- --></A><H3>
OPTION_MERGE_SAMPLES</H3>
<PRE>
public static java.lang.String <B>OPTION_MERGE_SAMPLES</B></PRE>
<DL>
<DD>Merge a list of samples. Keep separate files too.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_TOPHAT_PATH"><!-- --></A><H3>
OPTION_TOPHAT_PATH</H3>
<PRE>
public static java.lang.String <B>OPTION_TOPHAT_PATH</B></PRE>
<DL>
<DD>Config file option for tophat executable file
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_BED_FILE_FOR_WIG"><!-- --></A><H3>
OPTION_BED_FILE_FOR_WIG</H3>
<PRE>
public static java.lang.String <B>OPTION_BED_FILE_FOR_WIG</B></PRE>
<DL>
<DD>Config file option to write fragment endpoints to wig and bigwig file
 Writes counts for positions overlapping genes in this bed file
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_WIG_TO_BIGWIG_PATH"><!-- --></A><H3>
OPTION_WIG_TO_BIGWIG_PATH</H3>
<PRE>
public static java.lang.String <B>OPTION_WIG_TO_BIGWIG_PATH</B></PRE>
<DL>
<DD>Path to UCSC WigToBigWig executable
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_WIG_WRITER_JAR"><!-- --></A><H3>
OPTION_WIG_WRITER_JAR</H3>
<PRE>
public static java.lang.String <B>OPTION_WIG_WRITER_JAR</B></PRE>
<DL>
<DD>WigWriter jar file
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_CHR_SIZE_FILE"><!-- --></A><H3>
OPTION_CHR_SIZE_FILE</H3>
<PRE>
public static java.lang.String <B>OPTION_CHR_SIZE_FILE</B></PRE>
<DL>
<DD>Chromosome size file
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_GLOBAL_STATS_JAR"><!-- --></A><H3>
OPTION_GLOBAL_STATS_JAR</H3>
<PRE>
public static java.lang.String <B>OPTION_GLOBAL_STATS_JAR</B></PRE>
<DL>
<DD>Jar file to compute alignment global stats
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_TOPHAT_OPTIONS"><!-- --></A><H3>
OPTION_TOPHAT_OPTIONS</H3>
<PRE>
public static java.lang.String <B>OPTION_TOPHAT_OPTIONS</B></PRE>
<DL>
<DD>Config file option to provide options to tophat e.g. tophat_options --max-multihits 1
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_BOWTIE2_OPTIONS"><!-- --></A><H3>
OPTION_BOWTIE2_OPTIONS</H3>
<PRE>
public static java.lang.String <B>OPTION_BOWTIE2_OPTIONS</B></PRE>
<DL>
<DD>Config file option to provide options to bowtie2
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_NOVOALIGN_OPTIONS"><!-- --></A><H3>
OPTION_NOVOALIGN_OPTIONS</H3>
<PRE>
public static java.lang.String <B>OPTION_NOVOALIGN_OPTIONS</B></PRE>
<DL>
<DD>Config file option to provide options to novoalign e.g. novoalign_options -R 5
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_FRAGMENT_SIZE_DIST_OPTIONS"><!-- --></A><H3>
OPTION_FRAGMENT_SIZE_DIST_OPTIONS</H3>
<PRE>
public static java.lang.String <B>OPTION_FRAGMENT_SIZE_DIST_OPTIONS</B></PRE>
<DL>
<DD>Config file option to provide parameters to fragment size distribution calculation
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_BOWTIE2_PATH"><!-- --></A><H3>
OPTION_BOWTIE2_PATH</H3>
<PRE>
public static java.lang.String <B>OPTION_BOWTIE2_PATH</B></PRE>
<DL>
<DD>Config file option for bowtie2 executable file
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_BOWTIE2_BUILD_PATH"><!-- --></A><H3>
OPTION_BOWTIE2_BUILD_PATH</H3>
<PRE>
public static java.lang.String <B>OPTION_BOWTIE2_BUILD_PATH</B></PRE>
<DL>
<DD>Config file option for bowtie2-build executable file
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_SAMTOOLS_PATH"><!-- --></A><H3>
OPTION_SAMTOOLS_PATH</H3>
<PRE>
public static java.lang.String <B>OPTION_SAMTOOLS_PATH</B></PRE>
<DL>
<DD>Config file option for samtools executable
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_IGVTOOLS_PATH"><!-- --></A><H3>
OPTION_IGVTOOLS_PATH</H3>
<PRE>
public static java.lang.String <B>OPTION_IGVTOOLS_PATH</B></PRE>
<DL>
<DD>Config file option for igvtools executable
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_FASTQ_READ_PAIR_NUMBER_DELIMITER"><!-- --></A><H3>
OPTION_FASTQ_READ_PAIR_NUMBER_DELIMITER</H3>
<PRE>
public static java.lang.String <B>OPTION_FASTQ_READ_PAIR_NUMBER_DELIMITER</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="DEFAULT_FASTQ_READ_PAIR_NUMBER_DELIMITER"><!-- --></A><H3>
DEFAULT_FASTQ_READ_PAIR_NUMBER_DELIMITER</H3>
<PRE>
public static java.lang.String <B>DEFAULT_FASTQ_READ_PAIR_NUMBER_DELIMITER</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_PICARD_DIRECTORY"><!-- --></A><H3>
OPTION_PICARD_DIRECTORY</H3>
<PRE>
public static java.lang.String <B>OPTION_PICARD_DIRECTORY</B></PRE>
<DL>
<DD>Config file option for directory containing picard executables
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_FASTX_DIRECTORY"><!-- --></A><H3>
OPTION_FASTX_DIRECTORY</H3>
<PRE>
public static java.lang.String <B>OPTION_FASTX_DIRECTORY</B></PRE>
<DL>
<DD>Directory containing fastx binaries
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_SEQUENCING_ADAPTER_READ_1"><!-- --></A><H3>
OPTION_SEQUENCING_ADAPTER_READ_1</H3>
<PRE>
public static java.lang.String <B>OPTION_SEQUENCING_ADAPTER_READ_1</B></PRE>
<DL>
<DD>Sequencing adapter sequence for read 1
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_SEQUENCING_ADAPTER_READ_2"><!-- --></A><H3>
OPTION_SEQUENCING_ADAPTER_READ_2</H3>
<PRE>
public static java.lang.String <B>OPTION_SEQUENCING_ADAPTER_READ_2</B></PRE>
<DL>
<DD>Sequencing adapter sequence for read 2
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_NOVOALIGN_PATH"><!-- --></A><H3>
OPTION_NOVOALIGN_PATH</H3>
<PRE>
public static java.lang.String <B>OPTION_NOVOALIGN_PATH</B></PRE>
<DL>
<DD>Config file option for novoalign executable
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_GENOME_NOVOINDEX"><!-- --></A><H3>
OPTION_GENOME_NOVOINDEX</H3>
<PRE>
public static java.lang.String <B>OPTION_GENOME_NOVOINDEX</B></PRE>
<DL>
<DD>Config file option for novoindex of genome
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_PICARD_REF_FLAT"><!-- --></A><H3>
OPTION_PICARD_REF_FLAT</H3>
<PRE>
public static java.lang.String <B>OPTION_PICARD_REF_FLAT</B></PRE>
<DL>
<DD>Config file option to provide ref_flat annotation to Picard
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_PICARD_RIBOSOMAL_INTERVALS"><!-- --></A><H3>
OPTION_PICARD_RIBOSOMAL_INTERVALS</H3>
<PRE>
public static java.lang.String <B>OPTION_PICARD_RIBOSOMAL_INTERVALS</B></PRE>
<DL>
<DD>Config file option to provide ribosomal intervals to Picard
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_PICARD_STRAND_SPECIFICITY"><!-- --></A><H3>
OPTION_PICARD_STRAND_SPECIFICITY</H3>
<PRE>
public static java.lang.String <B>OPTION_PICARD_STRAND_SPECIFICITY</B></PRE>
<DL>
<DD>Config file option to provide strand specificity argument to Picard
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_PICARD_COLLECT_RNASEQ_METRICS"><!-- --></A><H3>
OPTION_PICARD_COLLECT_RNASEQ_METRICS</H3>
<PRE>
public static java.lang.String <B>OPTION_PICARD_COLLECT_RNASEQ_METRICS</B></PRE>
<DL>
<DD>Config file option to provide an alternative jar file for Picard program CollectRnaSeqMetrics
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_TRANSCRIPTOME_SPACE_BED_FILE"><!-- --></A><H3>
OPTION_TRANSCRIPTOME_SPACE_BED_FILE</H3>
<PRE>
public static final java.lang.Object <B>OPTION_TRANSCRIPTOME_SPACE_BED_FILE</B></PRE>
<DL>
<DD>Config file option for bed annotation for transcriptome space stats
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="OPTION_GENOME_SPACE_SIZE_FILE"><!-- --></A><H3>
OPTION_GENOME_SPACE_SIZE_FILE</H3>
<PRE>
public static final java.lang.Object <B>OPTION_GENOME_SPACE_SIZE_FILE</B></PRE>
<DL>
<DD>Config file option for chromosome size file for genomic space stats
<P>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="ConfigFileParser(java.lang.String)"><!-- --></A><H3>
ConfigFileParser</H3>
<PRE>
public <B>ConfigFileParser</B>(java.lang.String&nbsp;configFileName)
                 throws java.io.IOException</PRE>
<DL>
<DD>Constructor will read the input config file and store the commands and parameters in the respective
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>configFileName</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="addCommand(java.lang.String)"><!-- --></A><H3>
addCommand</H3>
<PRE>
public void <B>addCommand</B>(java.lang.String&nbsp;nextLine)</PRE>
<DL>
<DD>Adds the command in the current line if it is not prefixed with a "#"
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>nextLine</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="addSpecies(java.lang.String)"><!-- --></A><H3>
addSpecies</H3>
<PRE>
public void <B>addSpecies</B>(java.lang.String&nbsp;nextLine)</PRE>
<DL>
<DD>Adds the species
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>nextLine</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="hasCommandFor(java.lang.String)"><!-- --></A><H3>
hasCommandFor</H3>
<PRE>
public boolean <B>hasCommandFor</B>(java.lang.String&nbsp;command)</PRE>
<DL>
<DD>Returns true if the specified command was included as runnable in the file
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>command</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getCommand(int)"><!-- --></A><H3>
getCommand</H3>
<PRE>
public java.lang.String <B>getCommand</B>(int&nbsp;index)</PRE>
<DL>
<DD>Returns the command at the specified index
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>index</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getAllCommands()"><!-- --></A><H3>
getAllCommands</H3>
<PRE>
public java.util.List&lt;java.lang.String&gt; <B>getAllCommands</B>()</PRE>
<DL>
<DD>Returns all commands
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getRunSpecificDGEOptions()"><!-- --></A><H3>
getRunSpecificDGEOptions</H3>
<PRE>
public java.util.Set&lt;java.lang.String&gt; <B>getRunSpecificDGEOptions</B>()</PRE>
<DL>
<DD>Returns all run specific DGE options
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ConfigFileParser.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?broad/core/parser/ConfigFileParser.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ConfigFileParser.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
