<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:27 EDT 2013 -->
<TITLE>
ConfigFileParser.BasicOptions
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="ConfigFileParser.BasicOptions";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ConfigFileParser.BasicOptions.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?broad/core/parser/ConfigFileParser.BasicOptions.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ConfigFileParser.BasicOptions.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
broad.core.parser</FONT>
<BR>
Class ConfigFileParser.BasicOptions</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>broad.core.parser.ConfigFileParser.BasicOptions</B>
</PRE>
<DL>
<DT><B>Enclosing class:</B><DD><A HREF="../../../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A></DD>
</DL>
<HR>
<DL>
<DT><PRE>public static class <B>ConfigFileParser.BasicOptions</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
This is a helper class that maps the values of species-specific basic options to their value
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>skadri</DD>
</DL>
<HR>

<P>

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#ConfigFileParser.BasicOptions()">ConfigFileParser.BasicOptions</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#addBowtie2Option(java.lang.String, java.lang.String)">addBowtie2Option</A></B>(java.lang.String&nbsp;flag,
                 java.lang.String&nbsp;value)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Specify a bowtie2 option</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#addMergeSamples(java.lang.String)">addMergeSamples</A></B>(java.lang.String&nbsp;entireLine)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add a set of samples to merge</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#addNovoalignOption(java.lang.String, java.lang.String)">addNovoalignOption</A></B>(java.lang.String&nbsp;flag,
                   java.lang.String&nbsp;value)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Specify a novoalign option</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#addOption(java.lang.String, java.lang.String)">addOption</A></B>(java.lang.String&nbsp;flag,
          java.lang.String&nbsp;value)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#addRNAClass(java.lang.String, java.lang.String)">addRNAClass</A></B>(java.lang.String&nbsp;rnaClass,
            java.lang.String&nbsp;fileName)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add an RNA class fasta file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#addTophatOption(java.lang.String, java.lang.String)">addTophatOption</A></B>(java.lang.String&nbsp;flag,
                java.lang.String&nbsp;value)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Specify a tophat option</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getAlignmentGlobalStatsJar()">getAlignmentGlobalStatsJar</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the jar file for alignment model global stats</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getAllOptionMaps()">getAllOptionMaps</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all options in one map</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getBedFileForTranscriptomeSpaceStats()">getBedFileForTranscriptomeSpaceStats</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get bed annotation for transcriptome space stats</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getBedFileForWig()">getBedFileForWig</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get bed file that determines which genes to count when writing fragment end positions to wig and bigwig files</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getBowtie2BuildExecutablePath()">getBowtie2BuildExecutablePath</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the path to bowtie2-build executable</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getBowtie2ExecutablePath()">getBowtie2ExecutablePath</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to bowtie2 executable file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getBowtie2Options()">getBowtie2Options</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all bowtie2 flags and values</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getChrSizeFile()">getChrSizeFile</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get chromosome size file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getChrSizeFileForGenomicSpaceStats()">getChrSizeFileForGenomicSpaceStats</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get chromosome size file for genome space stats</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getFastqReadIdPairDelimiter()">getFastqReadIdPairDelimiter</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the delimiter for read pair number in fastq file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getFastxDirectory()">getFastxDirectory</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get directory containing fastx executables</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getGenomeAssembly()">getGenomeAssembly</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get name of genome assembly</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getGenomeBowtieIndex()">getGenomeBowtieIndex</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to genome bowtie index</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getGenomeFasta()">getGenomeFasta</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to genome fasta file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getGenomeNovoindex()">getGenomeNovoindex</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to genome novoindex</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getIgvtoolsPath()">getIgvtoolsPath</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to igvtools executable file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getNovoalignOptions()">getNovoalignOptions</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all novoalign flags and values</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getNovoalignPath()">getNovoalignPath</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to novoalign executable</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getOptionValueFor(java.lang.String)">getOptionValueFor</A></B>(java.lang.String&nbsp;type)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardCollectRnaSeqMetrics()">getPicardCollectRnaSeqMetrics</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to alternative jar file for CollectRnaSeqMetrics</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardDirectory()">getPicardDirectory</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get directory containing picard executables</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardRefFlat()">getPicardRefFlat</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to ref_flat annotation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardRibosomalIntervals()">getPicardRibosomalIntervals</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to ribosomal intervals for picard</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardStrandSpecificity()">getPicardStrandSpecificity</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get strand specificity argument for picard</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getQueueName()">getQueueName</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getRNAClassFile(java.lang.String)">getRNAClassFile</A></B>(java.lang.String&nbsp;rnaClass)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get fasta file for an RNA class</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getRNAClassFileNames()">getRNAClassFileNames</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getRrnaSeqsForDepletion()">getRrnaSeqsForDepletion</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get ribosomal RNA sequences to use for in silico rRNA depletion</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;java.lang.String&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getSamplesToMerge()">getSamplesToMerge</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all sets of samples to merge, by merged sample name
 Also will keep individual samples</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getSamtoolsPath()">getSamtoolsPath</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to samtools executable</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getSequencingAdapterRead1()">getSequencingAdapterRead1</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get sequencing adapter sequence for read 1</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getSequencingAdapterRead2()">getSequencingAdapterRead2</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get sequencing adapter sequence for read 2</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getTophatOptions()">getTophatOptions</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all tophat flags and values</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getTophatPath()">getTophatPath</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get path to tophat executable</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getTranscriptSeqsForAlignment()">getTranscriptSeqsForAlignment</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get fasta file of transcript sequences to align to</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getWigToBigWigExecutable()">getWigToBigWigExecutable</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get UCSC WigToBigWig executable file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#getWigWriterJar()">getWigWriterJar</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get WigWriter jar file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#hasCommandFor(java.lang.String)">hasCommandFor</A></B>(java.lang.String&nbsp;flag)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the flag has been specified</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#hasNovoalignPath()">hasNovoalignPath</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether a novoalign executable was specified in config file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#setNovoalignPath(java.lang.String)">setNovoalignPath</A></B>(java.lang.String&nbsp;path)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set path to novoalign executable</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#setQueueName(java.lang.String)">setQueueName</A></B>(java.lang.String&nbsp;name)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/core/parser/ConfigFileParser.BasicOptions.html#setTophatPath(java.lang.String)">setTophatPath</A></B>(java.lang.String&nbsp;path)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set path to tophat executable</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="ConfigFileParser.BasicOptions()"><!-- --></A><H3>
ConfigFileParser.BasicOptions</H3>
<PRE>
public <B>ConfigFileParser.BasicOptions</B>()</PRE>
<DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getRNAClassFileNames()"><!-- --></A><H3>
getRNAClassFileNames</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.lang.String&gt; <B>getRNAClassFileNames</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="addOption(java.lang.String, java.lang.String)"><!-- --></A><H3>
addOption</H3>
<PRE>
public void <B>addOption</B>(java.lang.String&nbsp;flag,
                      java.lang.String&nbsp;value)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getOptionValueFor(java.lang.String)"><!-- --></A><H3>
getOptionValueFor</H3>
<PRE>
public java.lang.String <B>getOptionValueFor</B>(java.lang.String&nbsp;type)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="hasCommandFor(java.lang.String)"><!-- --></A><H3>
hasCommandFor</H3>
<PRE>
public boolean <B>hasCommandFor</B>(java.lang.String&nbsp;flag)</PRE>
<DL>
<DD>Whether the flag has been specified
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>flag</CODE> - The flag
<DT><B>Returns:</B><DD>Whether a value has been provided</DL>
</DD>
</DL>
<HR>

<A NAME="getGenomeFasta()"><!-- --></A><H3>
getGenomeFasta</H3>
<PRE>
public java.lang.String <B>getGenomeFasta</B>()</PRE>
<DL>
<DD>Get path to genome fasta file
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The genome fasta file path specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getGenomeAssembly()"><!-- --></A><H3>
getGenomeAssembly</H3>
<PRE>
public java.lang.String <B>getGenomeAssembly</B>()</PRE>
<DL>
<DD>Get name of genome assembly
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The genome assembly name specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getFastqReadIdPairDelimiter()"><!-- --></A><H3>
getFastqReadIdPairDelimiter</H3>
<PRE>
public java.lang.String <B>getFastqReadIdPairDelimiter</B>()</PRE>
<DL>
<DD>Get the delimiter for read pair number in fastq file
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getGenomeBowtieIndex()"><!-- --></A><H3>
getGenomeBowtieIndex</H3>
<PRE>
public java.lang.String <B>getGenomeBowtieIndex</B>()</PRE>
<DL>
<DD>Get path to genome bowtie index
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to bowtie index specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getGenomeNovoindex()"><!-- --></A><H3>
getGenomeNovoindex</H3>
<PRE>
public java.lang.String <B>getGenomeNovoindex</B>()</PRE>
<DL>
<DD>Get path to genome novoindex
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to novoindex specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getPicardRefFlat()"><!-- --></A><H3>
getPicardRefFlat</H3>
<PRE>
public java.lang.String <B>getPicardRefFlat</B>()</PRE>
<DL>
<DD>Get path to ref_flat annotation
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to ref_flat annotation specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getPicardStrandSpecificity()"><!-- --></A><H3>
getPicardStrandSpecificity</H3>
<PRE>
public java.lang.String <B>getPicardStrandSpecificity</B>()</PRE>
<DL>
<DD>Get strand specificity argument for picard
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The strand specificity argument specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getPicardRibosomalIntervals()"><!-- --></A><H3>
getPicardRibosomalIntervals</H3>
<PRE>
public java.lang.String <B>getPicardRibosomalIntervals</B>()</PRE>
<DL>
<DD>Get path to ribosomal intervals for picard
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to ribosomal intervals file specified in config file, or null if not provided</DL>
</DD>
</DL>
<HR>

<A NAME="getPicardCollectRnaSeqMetrics()"><!-- --></A><H3>
getPicardCollectRnaSeqMetrics</H3>
<PRE>
public java.lang.String <B>getPicardCollectRnaSeqMetrics</B>()</PRE>
<DL>
<DD>Get path to alternative jar file for CollectRnaSeqMetrics
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to alternative jar file, or null if not provided</DL>
</DD>
</DL>
<HR>

<A NAME="getIgvtoolsPath()"><!-- --></A><H3>
getIgvtoolsPath</H3>
<PRE>
public java.lang.String <B>getIgvtoolsPath</B>()</PRE>
<DL>
<DD>Get path to igvtools executable file
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to igvtools executable specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="hasNovoalignPath()"><!-- --></A><H3>
hasNovoalignPath</H3>
<PRE>
public boolean <B>hasNovoalignPath</B>()</PRE>
<DL>
<DD>Whether a novoalign executable was specified in config file
<P>
<DD><DL>

<DT><B>Returns:</B><DD>True if novoalign executable was specified, false otherwise</DL>
</DD>
</DL>
<HR>

<A NAME="getNovoalignPath()"><!-- --></A><H3>
getNovoalignPath</H3>
<PRE>
public java.lang.String <B>getNovoalignPath</B>()</PRE>
<DL>
<DD>Get path to novoalign executable
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to novoalign executable file specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getQueueName()"><!-- --></A><H3>
getQueueName</H3>
<PRE>
public java.lang.String <B>getQueueName</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getBowtie2ExecutablePath()"><!-- --></A><H3>
getBowtie2ExecutablePath</H3>
<PRE>
public java.lang.String <B>getBowtie2ExecutablePath</B>()</PRE>
<DL>
<DD>Get path to bowtie2 executable file
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to bowtie2 executable specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getSamtoolsPath()"><!-- --></A><H3>
getSamtoolsPath</H3>
<PRE>
public java.lang.String <B>getSamtoolsPath</B>()</PRE>
<DL>
<DD>Get path to samtools executable
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to samtools executable specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getPicardDirectory()"><!-- --></A><H3>
getPicardDirectory</H3>
<PRE>
public java.lang.String <B>getPicardDirectory</B>()</PRE>
<DL>
<DD>Get directory containing picard executables
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to picard directory specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getFastxDirectory()"><!-- --></A><H3>
getFastxDirectory</H3>
<PRE>
public java.lang.String <B>getFastxDirectory</B>()</PRE>
<DL>
<DD>Get directory containing fastx executables
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to fastx directory specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="getSequencingAdapterRead1()"><!-- --></A><H3>
getSequencingAdapterRead1</H3>
<PRE>
public java.lang.String <B>getSequencingAdapterRead1</B>()</PRE>
<DL>
<DD>Get sequencing adapter sequence for read 1
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The adapter sequence</DL>
</DD>
</DL>
<HR>

<A NAME="getSequencingAdapterRead2()"><!-- --></A><H3>
getSequencingAdapterRead2</H3>
<PRE>
public java.lang.String <B>getSequencingAdapterRead2</B>()</PRE>
<DL>
<DD>Get sequencing adapter sequence for read 2
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The adapter sequence</DL>
</DD>
</DL>
<HR>

<A NAME="getBedFileForWig()"><!-- --></A><H3>
getBedFileForWig</H3>
<PRE>
public java.lang.String <B>getBedFileForWig</B>()</PRE>
<DL>
<DD>Get bed file that determines which genes to count when writing fragment end positions to wig and bigwig files
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The bed file path</DL>
</DD>
</DL>
<HR>

<A NAME="getWigToBigWigExecutable()"><!-- --></A><H3>
getWigToBigWigExecutable</H3>
<PRE>
public java.lang.String <B>getWigToBigWigExecutable</B>()</PRE>
<DL>
<DD>Get UCSC WigToBigWig executable file
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The WigToBigWig file</DL>
</DD>
</DL>
<HR>

<A NAME="getWigWriterJar()"><!-- --></A><H3>
getWigWriterJar</H3>
<PRE>
public java.lang.String <B>getWigWriterJar</B>()</PRE>
<DL>
<DD>Get WigWriter jar file
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The jar file</DL>
</DD>
</DL>
<HR>

<A NAME="getChrSizeFile()"><!-- --></A><H3>
getChrSizeFile</H3>
<PRE>
public java.lang.String <B>getChrSizeFile</B>()</PRE>
<DL>
<DD>Get chromosome size file
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The chromosome size file</DL>
</DD>
</DL>
<HR>

<A NAME="getBowtie2BuildExecutablePath()"><!-- --></A><H3>
getBowtie2BuildExecutablePath</H3>
<PRE>
public java.lang.String <B>getBowtie2BuildExecutablePath</B>()</PRE>
<DL>
<DD>Get the path to bowtie2-build executable
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path to bowtie2-build executable specified in config file</DL>
</DD>
</DL>
<HR>

<A NAME="addRNAClass(java.lang.String, java.lang.String)"><!-- --></A><H3>
addRNAClass</H3>
<PRE>
public void <B>addRNAClass</B>(java.lang.String&nbsp;rnaClass,
                        java.lang.String&nbsp;fileName)</PRE>
<DL>
<DD>Add an RNA class fasta file
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>rnaClass</CODE> - The class name<DD><CODE>fileName</CODE> - The fasta file</DL>
</DD>
</DL>
<HR>

<A NAME="getRNAClassFile(java.lang.String)"><!-- --></A><H3>
getRNAClassFile</H3>
<PRE>
public java.lang.String <B>getRNAClassFile</B>(java.lang.String&nbsp;rnaClass)</PRE>
<DL>
<DD>Get fasta file for an RNA class
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>rnaClass</CODE> - The class name
<DT><B>Returns:</B><DD>The fasta file</DL>
</DD>
</DL>
<HR>

<A NAME="addTophatOption(java.lang.String, java.lang.String)"><!-- --></A><H3>
addTophatOption</H3>
<PRE>
public void <B>addTophatOption</B>(java.lang.String&nbsp;flag,
                            java.lang.String&nbsp;value)</PRE>
<DL>
<DD>Specify a tophat option
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>flag</CODE> - The flag<DD><CODE>value</CODE> - The value</DL>
</DD>
</DL>
<HR>

<A NAME="addMergeSamples(java.lang.String)"><!-- --></A><H3>
addMergeSamples</H3>
<PRE>
public void <B>addMergeSamples</B>(java.lang.String&nbsp;entireLine)</PRE>
<DL>
<DD>Add a set of samples to merge
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>entireLine</CODE> - The entire line from config file</DL>
</DD>
</DL>
<HR>

<A NAME="addBowtie2Option(java.lang.String, java.lang.String)"><!-- --></A><H3>
addBowtie2Option</H3>
<PRE>
public void <B>addBowtie2Option</B>(java.lang.String&nbsp;flag,
                             java.lang.String&nbsp;value)</PRE>
<DL>
<DD>Specify a bowtie2 option
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>flag</CODE> - The flag<DD><CODE>value</CODE> - The value</DL>
</DD>
</DL>
<HR>

<A NAME="addNovoalignOption(java.lang.String, java.lang.String)"><!-- --></A><H3>
addNovoalignOption</H3>
<PRE>
public void <B>addNovoalignOption</B>(java.lang.String&nbsp;flag,
                               java.lang.String&nbsp;value)</PRE>
<DL>
<DD>Specify a novoalign option
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>flag</CODE> - The flag<DD><CODE>value</CODE> - The value including spaces if necessary</DL>
</DD>
</DL>
<HR>

<A NAME="setTophatPath(java.lang.String)"><!-- --></A><H3>
setTophatPath</H3>
<PRE>
public void <B>setTophatPath</B>(java.lang.String&nbsp;path)</PRE>
<DL>
<DD>Set path to tophat executable
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>path</CODE> - The path</DL>
</DD>
</DL>
<HR>

<A NAME="setNovoalignPath(java.lang.String)"><!-- --></A><H3>
setNovoalignPath</H3>
<PRE>
public void <B>setNovoalignPath</B>(java.lang.String&nbsp;path)</PRE>
<DL>
<DD>Set path to novoalign executable
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>path</CODE> - The path</DL>
</DD>
</DL>
<HR>

<A NAME="setQueueName(java.lang.String)"><!-- --></A><H3>
setQueueName</H3>
<PRE>
public void <B>setQueueName</B>(java.lang.String&nbsp;name)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getTophatPath()"><!-- --></A><H3>
getTophatPath</H3>
<PRE>
public java.lang.String <B>getTophatPath</B>()</PRE>
<DL>
<DD>Get path to tophat executable
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The path</DL>
</DD>
</DL>
<HR>

<A NAME="getTophatOptions()"><!-- --></A><H3>
getTophatOptions</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.lang.String&gt; <B>getTophatOptions</B>()</PRE>
<DL>
<DD>Get all tophat flags and values
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Map of flags to values</DL>
</DD>
</DL>
<HR>

<A NAME="getSamplesToMerge()"><!-- --></A><H3>
getSamplesToMerge</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.util.Collection&lt;java.lang.String&gt;&gt; <B>getSamplesToMerge</B>()</PRE>
<DL>
<DD>Get all sets of samples to merge, by merged sample name
 Also will keep individual samples
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Sets of samples to merge by new name of merged sample</DL>
</DD>
</DL>
<HR>

<A NAME="getBowtie2Options()"><!-- --></A><H3>
getBowtie2Options</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.lang.String&gt; <B>getBowtie2Options</B>()</PRE>
<DL>
<DD>Get all bowtie2 flags and values
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Map of flags to values</DL>
</DD>
</DL>
<HR>

<A NAME="getNovoalignOptions()"><!-- --></A><H3>
getNovoalignOptions</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.lang.String&gt; <B>getNovoalignOptions</B>()</PRE>
<DL>
<DD>Get all novoalign flags and values
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Map of flags to values</DL>
</DD>
</DL>
<HR>

<A NAME="getRrnaSeqsForDepletion()"><!-- --></A><H3>
getRrnaSeqsForDepletion</H3>
<PRE>
public java.lang.String <B>getRrnaSeqsForDepletion</B>()</PRE>
<DL>
<DD>Get ribosomal RNA sequences to use for in silico rRNA depletion
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The fasta file of sequences</DL>
</DD>
</DL>
<HR>

<A NAME="getTranscriptSeqsForAlignment()"><!-- --></A><H3>
getTranscriptSeqsForAlignment</H3>
<PRE>
public java.lang.String <B>getTranscriptSeqsForAlignment</B>()</PRE>
<DL>
<DD>Get fasta file of transcript sequences to align to
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The fasta file of sequences</DL>
</DD>
</DL>
<HR>

<A NAME="getAllOptionMaps()"><!-- --></A><H3>
getAllOptionMaps</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.lang.String&gt; <B>getAllOptionMaps</B>()</PRE>
<DL>
<DD>Get all options in one map
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Map of all flags for all programs to their values</DL>
</DD>
</DL>
<HR>

<A NAME="getAlignmentGlobalStatsJar()"><!-- --></A><H3>
getAlignmentGlobalStatsJar</H3>
<PRE>
public java.lang.String <B>getAlignmentGlobalStatsJar</B>()</PRE>
<DL>
<DD>Get the jar file for alignment model global stats
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Name of jar file</DL>
</DD>
</DL>
<HR>

<A NAME="getBedFileForTranscriptomeSpaceStats()"><!-- --></A><H3>
getBedFileForTranscriptomeSpaceStats</H3>
<PRE>
public java.lang.String <B>getBedFileForTranscriptomeSpaceStats</B>()</PRE>
<DL>
<DD>Get bed annotation for transcriptome space stats
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Name of bed file</DL>
</DD>
</DL>
<HR>

<A NAME="getChrSizeFileForGenomicSpaceStats()"><!-- --></A><H3>
getChrSizeFileForGenomicSpaceStats</H3>
<PRE>
public java.lang.String <B>getChrSizeFileForGenomicSpaceStats</B>()</PRE>
<DL>
<DD>Get chromosome size file for genome space stats
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Name of size file</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ConfigFileParser.BasicOptions.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?broad/core/parser/ConfigFileParser.BasicOptions.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ConfigFileParser.BasicOptions.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
