<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:22 EDT 2013 -->
<TITLE>
FeatureSequenceAlignment
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="FeatureSequenceAlignment";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FeatureSequenceAlignment.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../nextgen/core/alignment/ChromosomeInconsistencyException.html" title="class in nextgen.core.alignment"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?nextgen/core/alignment/FeatureSequenceAlignment.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FeatureSequenceAlignment.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
nextgen.core.alignment</FONT>
<BR>
Class FeatureSequenceAlignment</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>nextgen.core.alignment.FeatureSequenceAlignment</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>FeatureSequenceAlignment</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>prussell</DD>
</DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An unordered pair of genes</TD>
</TR>
</TABLE>
&nbsp;<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;float</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#DEFAULT_GAP_EXTEND_PENALTY">DEFAULT_GAP_EXTEND_PENALTY</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Default gap extend penalty for Smith Waterman</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;float</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#DEFAULT_GAP_OPEN_PENALTY">DEFAULT_GAP_OPEN_PENALTY</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Default gap open penalty for Smith Waterman</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;float</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#DEFAULT_MATCH_SCORE">DEFAULT_MATCH_SCORE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Default match score for Smith Waterman</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;float</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#DEFAULT_MISMATCH_SCORE">DEFAULT_MISMATCH_SCORE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Default mismatch score for Smith Waterman</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.util.Collection, java.util.Map, float, float, float, float)">FeatureSequenceAlignment</A></B>(java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                         java.util.Map&lt;java.lang.String,<A HREF="../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&gt;&nbsp;chrs,
                         float&nbsp;matchScore,
                         float&nbsp;mismatchScore,
                         float&nbsp;gapOpenPenalty,
                         float&nbsp;gapExtendPenalty)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.util.Collection, java.lang.String)">FeatureSequenceAlignment</A></B>(java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                         java.lang.String&nbsp;genomeFasta)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.util.Collection, java.lang.String, float, float, float, float)">FeatureSequenceAlignment</A></B>(java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                         java.lang.String&nbsp;genomeFasta,
                         float&nbsp;matchScore,
                         float&nbsp;mismatchScore,
                         float&nbsp;gapOpenPenalty,
                         float&nbsp;gapExtendPenalty)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.lang.String, java.lang.String)">FeatureSequenceAlignment</A></B>(java.lang.String&nbsp;featureBedFile,
                         java.lang.String&nbsp;genomeFasta)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.lang.String, java.lang.String, float, float, float, float)">FeatureSequenceAlignment</A></B>(java.lang.String&nbsp;featureBedFile,
                         java.lang.String&nbsp;genomeFasta,
                         float&nbsp;matchScore,
                         float&nbsp;mismatchScore,
                         float&nbsp;gapOpenPenalty,
                         float&nbsp;gapExtendPenalty)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>,<A HREF="../../../jaligner/Alignment.html" title="class in jaligner">Alignment</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#getAllPairwiseAntisenseAlignments(float, float)">getAllPairwiseAntisenseAlignments</A></B>(float&nbsp;minAlignLength,
                                  float&nbsp;minPctIdentity)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all antisense direction alignments passing thresholds</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>,<A HREF="../../../jaligner/Alignment.html" title="class in jaligner">Alignment</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#getAllPairwiseSenseAlignments(float, float)">getAllPairwiseSenseAlignments</A></B>(float&nbsp;minAlignLength,
                              float&nbsp;minPctIdentity)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all sense direction alignments passing thresholds</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#main(java.lang.String[])">main</A></B>(java.lang.String[]&nbsp;args)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#writeAllAntisenseAlignmentsToBed(java.lang.String, boolean, int, float)">writeAllAntisenseAlignmentsToBed</A></B>(java.lang.String&nbsp;outBedFile,
                                 boolean&nbsp;append,
                                 int&nbsp;minAlignLength,
                                 float&nbsp;minPctIdentity)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write all pairwise alignments (antisense direction only) to bed file in genome coordinates</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#writeAllAntisenseAlignmentsToBed(java.lang.String, int, float)">writeAllAntisenseAlignmentsToBed</A></B>(java.lang.String&nbsp;outBedFile,
                                 int&nbsp;minAlignLength,
                                 float&nbsp;minPctIdentity)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write all pairwise alignments (antisense direction only) to bed file in genome coordinates</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#writeAllSenseAlignmentsToBed(java.lang.String, boolean, int, float)">writeAllSenseAlignmentsToBed</A></B>(java.lang.String&nbsp;outBedFile,
                             boolean&nbsp;append,
                             int&nbsp;minAlignLength,
                             float&nbsp;minPctIdentity)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write all pairwise alignments (sense direction only) to bed file in genome coordinates</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.html#writeAllSenseAlignmentsToBed(java.lang.String, int, float)">writeAllSenseAlignmentsToBed</A></B>(java.lang.String&nbsp;outBedFile,
                             int&nbsp;minAlignLength,
                             float&nbsp;minPctIdentity)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write all pairwise alignments (sense direction only) to bed file in genome coordinates</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="DEFAULT_MATCH_SCORE"><!-- --></A><H3>
DEFAULT_MATCH_SCORE</H3>
<PRE>
public static float <B>DEFAULT_MATCH_SCORE</B></PRE>
<DL>
<DD>Default match score for Smith Waterman
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="DEFAULT_MISMATCH_SCORE"><!-- --></A><H3>
DEFAULT_MISMATCH_SCORE</H3>
<PRE>
public static float <B>DEFAULT_MISMATCH_SCORE</B></PRE>
<DL>
<DD>Default mismatch score for Smith Waterman
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="DEFAULT_GAP_OPEN_PENALTY"><!-- --></A><H3>
DEFAULT_GAP_OPEN_PENALTY</H3>
<PRE>
public static float <B>DEFAULT_GAP_OPEN_PENALTY</B></PRE>
<DL>
<DD>Default gap open penalty for Smith Waterman
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="DEFAULT_GAP_EXTEND_PENALTY"><!-- --></A><H3>
DEFAULT_GAP_EXTEND_PENALTY</H3>
<PRE>
public static float <B>DEFAULT_GAP_EXTEND_PENALTY</B></PRE>
<DL>
<DD>Default gap extend penalty for Smith Waterman
<P>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="FeatureSequenceAlignment(java.lang.String, java.lang.String)"><!-- --></A><H3>
FeatureSequenceAlignment</H3>
<PRE>
public <B>FeatureSequenceAlignment</B>(java.lang.String&nbsp;featureBedFile,
                                java.lang.String&nbsp;genomeFasta)
                         throws java.io.IOException</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>featureBedFile</CODE> - Bed file of features<DD><CODE>genomeFasta</CODE> - Genome fasta file
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="FeatureSequenceAlignment(java.util.Collection, java.lang.String)"><!-- --></A><H3>
FeatureSequenceAlignment</H3>
<PRE>
public <B>FeatureSequenceAlignment</B>(java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                                java.lang.String&nbsp;genomeFasta)
                         throws java.io.IOException</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>genes</CODE> - Set of features<DD><CODE>genomeFasta</CODE> - Genome fasta file
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="FeatureSequenceAlignment(java.lang.String, java.lang.String, float, float, float, float)"><!-- --></A><H3>
FeatureSequenceAlignment</H3>
<PRE>
public <B>FeatureSequenceAlignment</B>(java.lang.String&nbsp;featureBedFile,
                                java.lang.String&nbsp;genomeFasta,
                                float&nbsp;matchScore,
                                float&nbsp;mismatchScore,
                                float&nbsp;gapOpenPenalty,
                                float&nbsp;gapExtendPenalty)
                         throws java.io.IOException</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>featureBedFile</CODE> - Bed file of features<DD><CODE>genomeFasta</CODE> - Genome fasta file<DD><CODE>matchScore</CODE> - Match score for Smith Waterman<DD><CODE>mismatchScore</CODE> - Mismatch score for Smith Waterman<DD><CODE>gapOpenPenalty</CODE> - Gap open penalty for Smith Waterman<DD><CODE>gapExtendPenalty</CODE> - Gap extend penalty for Smith Waterman
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="FeatureSequenceAlignment(java.util.Collection, java.util.Map, float, float, float, float)"><!-- --></A><H3>
FeatureSequenceAlignment</H3>
<PRE>
public <B>FeatureSequenceAlignment</B>(java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                                java.util.Map&lt;java.lang.String,<A HREF="../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&gt;&nbsp;chrs,
                                float&nbsp;matchScore,
                                float&nbsp;mismatchScore,
                                float&nbsp;gapOpenPenalty,
                                float&nbsp;gapExtendPenalty)
                         throws java.io.IOException</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>genes</CODE> - The features<DD><CODE>chrs</CODE> - Map of chromosome names to sequences<DD><CODE>matchScore</CODE> - Match score for Smith Waterman<DD><CODE>mismatchScore</CODE> - Mismatch score for Smith Waterman<DD><CODE>gapOpenPenalty</CODE> - Gap open penalty for Smith Waterman<DD><CODE>gapExtendPenalty</CODE> - Gap extend penalty for Smith Waterman
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="FeatureSequenceAlignment(java.util.Collection, java.lang.String, float, float, float, float)"><!-- --></A><H3>
FeatureSequenceAlignment</H3>
<PRE>
public <B>FeatureSequenceAlignment</B>(java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                                java.lang.String&nbsp;genomeFasta,
                                float&nbsp;matchScore,
                                float&nbsp;mismatchScore,
                                float&nbsp;gapOpenPenalty,
                                float&nbsp;gapExtendPenalty)
                         throws java.io.IOException</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>genes</CODE> - Set of features<DD><CODE>genomeFasta</CODE> - Genome fasta file<DD><CODE>matchScore</CODE> - Match score for Smith Waterman<DD><CODE>mismatchScore</CODE> - Mismatch score for Smith Waterman<DD><CODE>gapOpenPenalty</CODE> - Gap open penalty for Smith Waterman<DD><CODE>gapExtendPenalty</CODE> - Gap extend penalty for Smith Waterman
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getAllPairwiseSenseAlignments(float, float)"><!-- --></A><H3>
getAllPairwiseSenseAlignments</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>,<A HREF="../../../jaligner/Alignment.html" title="class in jaligner">Alignment</A>&gt; <B>getAllPairwiseSenseAlignments</B>(float&nbsp;minAlignLength,
                                                                                                         float&nbsp;minPctIdentity)</PRE>
<DL>
<DD>Get all sense direction alignments passing thresholds
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>minAlignLength</CODE> - Min alignment length<DD><CODE>minPctIdentity</CODE> - Min percent identity
<DT><B>Returns:</B><DD>All sense alignments of unordered gene pair passing criteria</DL>
</DD>
</DL>
<HR>

<A NAME="getAllPairwiseAntisenseAlignments(float, float)"><!-- --></A><H3>
getAllPairwiseAntisenseAlignments</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>,<A HREF="../../../jaligner/Alignment.html" title="class in jaligner">Alignment</A>&gt; <B>getAllPairwiseAntisenseAlignments</B>(float&nbsp;minAlignLength,
                                                                                                             float&nbsp;minPctIdentity)</PRE>
<DL>
<DD>Get all antisense direction alignments passing thresholds
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>minAlignLength</CODE> - Min alignment length<DD><CODE>minPctIdentity</CODE> - Min percent identity
<DT><B>Returns:</B><DD>All antisense alignments of unordered gene pair passing criteria</DL>
</DD>
</DL>
<HR>

<A NAME="writeAllSenseAlignmentsToBed(java.lang.String, int, float)"><!-- --></A><H3>
writeAllSenseAlignmentsToBed</H3>
<PRE>
public void <B>writeAllSenseAlignmentsToBed</B>(java.lang.String&nbsp;outBedFile,
                                         int&nbsp;minAlignLength,
                                         float&nbsp;minPctIdentity)
                                  throws java.io.IOException</PRE>
<DL>
<DD>Write all pairwise alignments (sense direction only) to bed file in genome coordinates
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outBedFile</CODE> - Output bed file<DD><CODE>minAlignLength</CODE> - Min alignment length to keep<DD><CODE>minPctIdentity</CODE> - Min percent identity to keep
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeAllAntisenseAlignmentsToBed(java.lang.String, int, float)"><!-- --></A><H3>
writeAllAntisenseAlignmentsToBed</H3>
<PRE>
public void <B>writeAllAntisenseAlignmentsToBed</B>(java.lang.String&nbsp;outBedFile,
                                             int&nbsp;minAlignLength,
                                             float&nbsp;minPctIdentity)
                                      throws java.io.IOException</PRE>
<DL>
<DD>Write all pairwise alignments (antisense direction only) to bed file in genome coordinates
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outBedFile</CODE> - Output bed file<DD><CODE>minAlignLength</CODE> - Min alignment length to keep<DD><CODE>minPctIdentity</CODE> - Min percent identity to keep
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeAllSenseAlignmentsToBed(java.lang.String, boolean, int, float)"><!-- --></A><H3>
writeAllSenseAlignmentsToBed</H3>
<PRE>
public void <B>writeAllSenseAlignmentsToBed</B>(java.lang.String&nbsp;outBedFile,
                                         boolean&nbsp;append,
                                         int&nbsp;minAlignLength,
                                         float&nbsp;minPctIdentity)
                                  throws java.io.IOException</PRE>
<DL>
<DD>Write all pairwise alignments (sense direction only) to bed file in genome coordinates
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outBedFile</CODE> - Output bed file<DD><CODE>append</CODE> - Write to end of file rather than beginning<DD><CODE>minAlignLength</CODE> - Min alignment length to keep<DD><CODE>minPctIdentity</CODE> - Min percent identity to keep
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeAllAntisenseAlignmentsToBed(java.lang.String, boolean, int, float)"><!-- --></A><H3>
writeAllAntisenseAlignmentsToBed</H3>
<PRE>
public void <B>writeAllAntisenseAlignmentsToBed</B>(java.lang.String&nbsp;outBedFile,
                                             boolean&nbsp;append,
                                             int&nbsp;minAlignLength,
                                             float&nbsp;minPctIdentity)
                                      throws java.io.IOException</PRE>
<DL>
<DD>Write all pairwise alignments (antisense direction only) to bed file in genome coordinates
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outBedFile</CODE> - Output bed file<DD><CODE>append</CODE> - Write to end of file rather than beginning<DD><CODE>minAlignLength</CODE> - Min alignment length to keep<DD><CODE>minPctIdentity</CODE> - Min percent identity to keep
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="main(java.lang.String[])"><!-- --></A><H3>
main</H3>
<PRE>
public static void <B>main</B>(java.lang.String[]&nbsp;args)
                 throws java.io.IOException</PRE>
<DL>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>args</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FeatureSequenceAlignment.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../nextgen/core/alignment/ChromosomeInconsistencyException.html" title="class in nextgen.core.alignment"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?nextgen/core/alignment/FeatureSequenceAlignment.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FeatureSequenceAlignment.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
