<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:27 EDT 2013 -->
<TITLE>
TranscriptomeSpaceAlignmentModel
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="TranscriptomeSpaceAlignmentModel";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/TranscriptomeSpaceAlignmentModel.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model"><B>PREV CLASS</B></A>&nbsp;
&nbsp;NEXT CLASS</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="TranscriptomeSpaceAlignmentModel.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_classes_inherited_from_class_nextgen.core.model.AlignmentModel">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
nextgen.core.model</FONT>
<BR>
Class TranscriptomeSpaceAlignmentModel</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">nextgen.core.annotation.AbstractAnnotationCollection</A>&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;
      <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">nextgen.core.model.AlignmentModel</A>
          <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">nextgen.core.model.ScanStatisticDataAlignmentModel</A>
              <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>nextgen.core.model.TranscriptomeSpaceAlignmentModel</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD><A HREF="../../../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;</DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>TranscriptomeSpaceAlignmentModel</B><DT>extends <A HREF="../../../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A></DL>
</PRE>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>prussell
 Alignment model with special functionality for a transcriptome space</DD>
</DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
</TABLE>
&nbsp;<A NAME="nested_classes_inherited_from_class_nextgen.core.model.AlignmentModel"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Nested classes/interfaces inherited from class nextgen.core.model.<A HREF="../../../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../nextgen/core/model/AlignmentModel.AlignmentCount.html" title="class in nextgen.core.model">AlignmentModel.AlignmentCount</A></CODE></TD>
</TR>
</TABLE>
&nbsp;

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace)">TranscriptomeSpaceAlignmentModel</A></B>(java.lang.String&nbsp;bamAlignmentFile,
                                 <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)">TranscriptomeSpaceAlignmentModel</A></B>(java.lang.String&nbsp;bamAlignmentFile,
                                 <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                 <A HREF="../../../nextgen/core/alignment/AbstractPairedEndAlignment.TranscriptionRead.html" title="enum in nextgen.core.alignment">AbstractPairedEndAlignment.TranscriptionRead</A>&nbsp;transcriptionRead)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, boolean)">TranscriptomeSpaceAlignmentModel</A></B>(java.lang.String&nbsp;bamAlignmentFile,
                                 <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                 boolean&nbsp;readOrCreatePairedEndBam)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)">TranscriptomeSpaceAlignmentModel</A></B>(java.lang.String&nbsp;bamAlignmentFile,
                                 <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                 boolean&nbsp;readOrCreatePairedEndBam,
                                 <A HREF="../../../nextgen/core/alignment/AbstractPairedEndAlignment.TranscriptionRead.html" title="enum in nextgen.core.alignment">AbstractPairedEndAlignment.TranscriptionRead</A>&nbsp;transcriptionRead)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection)">TranscriptomeSpaceAlignmentModel</A></B>(java.lang.String&nbsp;bamAlignmentFile,
                                 <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                 java.util.Collection&lt;org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&gt;&nbsp;readFilterSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)">TranscriptomeSpaceAlignmentModel</A></B>(java.lang.String&nbsp;bamAlignmentFile,
                                 <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                 java.util.Collection&lt;org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&gt;&nbsp;readFilterSet,
                                 <A HREF="../../../nextgen/core/alignment/AbstractPairedEndAlignment.TranscriptionRead.html" title="enum in nextgen.core.alignment">AbstractPairedEndAlignment.TranscriptionRead</A>&nbsp;transcriptionRead)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection, boolean)">TranscriptomeSpaceAlignmentModel</A></B>(java.lang.String&nbsp;bamAlignmentFile,
                                 <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                 java.util.Collection&lt;org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&gt;&nbsp;readFilterSet,
                                 boolean&nbsp;readOrCreatePairedEndBam)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)">TranscriptomeSpaceAlignmentModel</A></B>(java.lang.String&nbsp;bamAlignmentFile,
                                 <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                 java.util.Collection&lt;org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&gt;&nbsp;readFilterSet,
                                 boolean&nbsp;readOrCreatePairedEndBam,
                                 <A HREF="../../../nextgen/core/alignment/AbstractPairedEndAlignment.TranscriptionRead.html" title="enum in nextgen.core.alignment">AbstractPairedEndAlignment.TranscriptionRead</A>&nbsp;transcriptionRead)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;java.lang.Double&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#getPositionCountList(nextgen.core.annotation.Gene)">getPositionCountList</A></B>(<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get list of position level counts in mature transcript</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#getReferenceNames()">getReferenceNames</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the names of genes in the transcriptome space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;T extends <A HREF="../../../nextgen/core/model/score/WindowScore.html" title="interface in nextgen.core.model.score">WindowScore</A>&gt; 
<BR>
<A HREF="../../../nextgen/core/model/score/WindowScoreIterator.html" title="class in nextgen.core.model.score">WindowScoreIterator</A>&lt;T&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#scan(nextgen.core.annotation.Gene, int, int, nextgen.core.model.score.WindowProcessor)">scan</A></B>(<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
     int&nbsp;windowSize,
     int&nbsp;overlap,
     <A HREF="../../../nextgen/core/model/score/WindowProcessor.html" title="interface in nextgen.core.model.score">WindowProcessor</A>&lt;T&gt;&nbsp;processor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scan windows over a gene and score
 Only scan exons of this gene</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_nextgen.core.model.ScanStatisticDataAlignmentModel"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class nextgen.core.model.<A HREF="../../../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../nextgen/core/model/ScanStatisticDataAlignmentModel.html#isExpressed(nextgen.core.annotation.Annotation, double)">isExpressed</A>, <A HREF="../../../nextgen/core/model/ScanStatisticDataAlignmentModel.html#scan(java.lang.String, int, int)">scan</A>, <A HREF="../../../nextgen/core/model/ScanStatisticDataAlignmentModel.html#scanAll(int, int)">scanAll</A>, <A HREF="../../../nextgen/core/model/ScanStatisticDataAlignmentModel.html#scoreWindow(nextgen.core.annotation.Annotation)">scoreWindow</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_nextgen.core.model.AlignmentModel"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class nextgen.core.model.<A HREF="../../../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../nextgen/core/model/AlignmentModel.html#addFilter(org.apache.commons.collections15.Predicate)">addFilter</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#addFilters(java.util.Collection)">addFilters</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#computeGlobalStats()">computeGlobalStats</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#containsReference(java.lang.String)">containsReference</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getBasesCovered(nextgen.core.annotation.Annotation)">getBasesCovered</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getBasesCovered(nextgen.core.annotation.Annotation, boolean)">getBasesCovered</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getCoordinateSpace()">getCoordinateSpace</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getCount()">getCount</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getCount(nextgen.core.annotation.Annotation)">getCount</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getCount(nextgen.core.annotation.Annotation, boolean)">getCount</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getCount(net.sf.samtools.util.CloseableIterator)">getCount</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getCountExcludingRegion(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)">getCountExcludingRegion</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getCountsPerPosition(nextgen.core.annotation.Gene)">getCountsPerPosition</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getCountStranded(nextgen.core.annotation.Annotation, boolean)">getCountStranded</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getGlobalCount()">getGlobalCount</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getGlobalLambda()">getGlobalLambda</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getGlobalLength()">getGlobalLength</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getGlobalNumReads()">getGlobalNumReads</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getGlobalPairedFragments()">getGlobalPairedFragments</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getIntronCounts(nextgen.core.annotation.Annotation)">getIntronCounts</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getOverlappingAnnotations(nextgen.core.annotation.Annotation, boolean)">getOverlappingAnnotations</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getOverlappingReadCountsStranded(nextgen.core.annotation.Annotation, boolean)">getOverlappingReadCountsStranded</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getOverlappingReadReplicateCounts(nextgen.core.annotation.Annotation, boolean)">getOverlappingReadReplicateCounts</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getOverlappingReads(nextgen.core.annotation.Annotation, boolean)">getOverlappingReads</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getOverlappingReads(java.lang.String)">getOverlappingReads</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getOverlappingSplicedReads(nextgen.core.annotation.Annotation, boolean)">getOverlappingSplicedReads</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getOverlappingSplicedReads(java.lang.String)">getOverlappingSplicedReads</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getPermutedAnnotations(nextgen.core.annotation.Annotation)">getPermutedAnnotations</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getReadIterator()">getReadIterator</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getReadIterator(nextgen.core.annotation.Annotation)">getReadIterator</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getReadSizeDistribution(nextgen.core.annotation.Annotation, nextgen.core.coordinatesystem.CoordinateSpace, int, int)">getReadSizeDistribution</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getReadSizeDistribution(nextgen.core.coordinatesystem.CoordinateSpace, int, int)">getReadSizeDistribution</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getRefSequenceLambda(java.lang.String)">getRefSequenceLambda</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getRefSequenceLength(java.lang.String)">getRefSequenceLength</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getScore(nextgen.core.annotation.Annotation, nextgen.core.model.score.WindowProcessor)">getScore</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#getSplicedCount(broad.pda.datastructures.Alignments, boolean)">getSplicedCount</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#iterator()">iterator</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#removeFilter(org.apache.commons.collections15.Predicate)">removeFilter</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#scan(nextgen.core.annotation.Annotation, int, int)">scan</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#scan(nextgen.core.annotation.Annotation, int, int, nextgen.core.model.score.WindowProcessor)">scan</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#size()">size</A>, <A HREF="../../../nextgen/core/model/AlignmentModel.html#windowIterator(int, int)">windowIterator</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_nextgen.core.annotation.AbstractAnnotationCollection"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class nextgen.core.annotation.<A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html#collapse(nextgen.core.annotation.Annotation, boolean)">collapse</A>, <A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html#getAverageCoverage(nextgen.core.annotation.Annotation)">getAverageCoverage</A>, <A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html#getCount(nextgen.core.annotation.AnnotationList)">getCount</A>, <A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html#getCount(nextgen.core.annotation.AnnotationList, boolean)">getCount</A>, <A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html#getOverlappingAnnotations(nextgen.core.annotation.Annotation)">getOverlappingAnnotations</A>, <A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html#getTotalCoverage(nextgen.core.annotation.Annotation)">getTotalCoverage</A>, <A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html#scan(int, int)">scan</A>, <A HREF="../../../nextgen/core/annotation/AbstractAnnotationCollection.html#scan(int, int, nextgen.core.model.score.WindowProcessor)">scan</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection, boolean)"><!-- --></A><H3>
TranscriptomeSpaceAlignmentModel</H3>
<PRE>
public <B>TranscriptomeSpaceAlignmentModel</B>(java.lang.String&nbsp;bamAlignmentFile,
                                        <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                        java.util.Collection&lt;org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&gt;&nbsp;readFilterSet,
                                        boolean&nbsp;readOrCreatePairedEndBam)</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamAlignmentFile</CODE> - <DD><CODE>transcriptomeSpace</CODE> - <DD><CODE>readFilterSet</CODE> - <DD><CODE>readOrCreatePairedEndBam</CODE> - </DL>
</DL>
<HR>

<A NAME="TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection)"><!-- --></A><H3>
TranscriptomeSpaceAlignmentModel</H3>
<PRE>
public <B>TranscriptomeSpaceAlignmentModel</B>(java.lang.String&nbsp;bamAlignmentFile,
                                        <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                        java.util.Collection&lt;org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&gt;&nbsp;readFilterSet)</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamAlignmentFile</CODE> - <DD><CODE>transcriptomeSpace</CODE> - <DD><CODE>readFilterSet</CODE> - </DL>
</DL>
<HR>

<A NAME="TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, boolean)"><!-- --></A><H3>
TranscriptomeSpaceAlignmentModel</H3>
<PRE>
public <B>TranscriptomeSpaceAlignmentModel</B>(java.lang.String&nbsp;bamAlignmentFile,
                                        <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                        boolean&nbsp;readOrCreatePairedEndBam)</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamAlignmentFile</CODE> - <DD><CODE>transcriptomeSpace</CODE> - <DD><CODE>readOrCreatePairedEndBam</CODE> - </DL>
</DL>
<HR>

<A NAME="TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace)"><!-- --></A><H3>
TranscriptomeSpaceAlignmentModel</H3>
<PRE>
public <B>TranscriptomeSpaceAlignmentModel</B>(java.lang.String&nbsp;bamAlignmentFile,
                                        <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace)</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamAlignmentFile</CODE> - <DD><CODE>transcriptomeSpace</CODE> - </DL>
</DL>
<HR>

<A NAME="TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><!-- --></A><H3>
TranscriptomeSpaceAlignmentModel</H3>
<PRE>
public <B>TranscriptomeSpaceAlignmentModel</B>(java.lang.String&nbsp;bamAlignmentFile,
                                        <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                        java.util.Collection&lt;org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&gt;&nbsp;readFilterSet,
                                        boolean&nbsp;readOrCreatePairedEndBam,
                                        <A HREF="../../../nextgen/core/alignment/AbstractPairedEndAlignment.TranscriptionRead.html" title="enum in nextgen.core.alignment">AbstractPairedEndAlignment.TranscriptionRead</A>&nbsp;transcriptionRead)</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamAlignmentFile</CODE> - <DD><CODE>transcriptomeSpace</CODE> - <DD><CODE>readFilterSet</CODE> - <DD><CODE>readOrCreatePairedEndBam</CODE> - <DD><CODE>transcriptionRead</CODE> - </DL>
</DL>
<HR>

<A NAME="TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><!-- --></A><H3>
TranscriptomeSpaceAlignmentModel</H3>
<PRE>
public <B>TranscriptomeSpaceAlignmentModel</B>(java.lang.String&nbsp;bamAlignmentFile,
                                        <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                        java.util.Collection&lt;org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&gt;&nbsp;readFilterSet,
                                        <A HREF="../../../nextgen/core/alignment/AbstractPairedEndAlignment.TranscriptionRead.html" title="enum in nextgen.core.alignment">AbstractPairedEndAlignment.TranscriptionRead</A>&nbsp;transcriptionRead)</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamAlignmentFile</CODE> - <DD><CODE>transcriptomeSpace</CODE> - <DD><CODE>readFilterSet</CODE> - <DD><CODE>transcriptionRead</CODE> - </DL>
</DL>
<HR>

<A NAME="TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><!-- --></A><H3>
TranscriptomeSpaceAlignmentModel</H3>
<PRE>
public <B>TranscriptomeSpaceAlignmentModel</B>(java.lang.String&nbsp;bamAlignmentFile,
                                        <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                        boolean&nbsp;readOrCreatePairedEndBam,
                                        <A HREF="../../../nextgen/core/alignment/AbstractPairedEndAlignment.TranscriptionRead.html" title="enum in nextgen.core.alignment">AbstractPairedEndAlignment.TranscriptionRead</A>&nbsp;transcriptionRead)</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamAlignmentFile</CODE> - <DD><CODE>transcriptomeSpace</CODE> - <DD><CODE>readOrCreatePairedEndBam</CODE> - <DD><CODE>transcriptionRead</CODE> - </DL>
</DL>
<HR>

<A NAME="TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><!-- --></A><H3>
TranscriptomeSpaceAlignmentModel</H3>
<PRE>
public <B>TranscriptomeSpaceAlignmentModel</B>(java.lang.String&nbsp;bamAlignmentFile,
                                        <A HREF="../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>&nbsp;transcriptomeSpace,
                                        <A HREF="../../../nextgen/core/alignment/AbstractPairedEndAlignment.TranscriptionRead.html" title="enum in nextgen.core.alignment">AbstractPairedEndAlignment.TranscriptionRead</A>&nbsp;transcriptionRead)</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamAlignmentFile</CODE> - <DD><CODE>transcriptomeSpace</CODE> - <DD><CODE>transcriptionRead</CODE> - </DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getPositionCountList(nextgen.core.annotation.Gene)"><!-- --></A><H3>
getPositionCountList</H3>
<PRE>
public java.util.List&lt;java.lang.Double&gt; <B>getPositionCountList</B>(<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)
                                                      throws java.io.IOException</PRE>
<DL>
<DD>Get list of position level counts in mature transcript
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The region
<DT><B>Returns:</B><DD>List of position counts
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scan(nextgen.core.annotation.Gene, int, int, nextgen.core.model.score.WindowProcessor)"><!-- --></A><H3>
scan</H3>
<PRE>
public &lt;T extends <A HREF="../../../nextgen/core/model/score/WindowScore.html" title="interface in nextgen.core.model.score">WindowScore</A>&gt; <A HREF="../../../nextgen/core/model/score/WindowScoreIterator.html" title="class in nextgen.core.model.score">WindowScoreIterator</A>&lt;T&gt; <B>scan</B>(<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                                           int&nbsp;windowSize,
                                                           int&nbsp;overlap,
                                                           <A HREF="../../../nextgen/core/model/score/WindowProcessor.html" title="interface in nextgen.core.model.score">WindowProcessor</A>&lt;T&gt;&nbsp;processor)</PRE>
<DL>
<DD>Scan windows over a gene and score
 Only scan exons of this gene
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene<DD><CODE>windowSize</CODE> - Window size<DD><CODE>overlap</CODE> - Overlap size<DD><CODE>processor</CODE> - The window processor
<DT><B>Returns:</B><DD>A score iterator over windows</DL>
</DD>
</DL>
<HR>

<A NAME="getReferenceNames()"><!-- --></A><H3>
getReferenceNames</H3>
<PRE>
public java.util.Collection&lt;java.lang.String&gt; <B>getReferenceNames</B>()</PRE>
<DL>
<DD>Get the names of genes in the transcriptome space
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The names of reference sequences (genes) in the transcriptome space</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/TranscriptomeSpaceAlignmentModel.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model"><B>PREV CLASS</B></A>&nbsp;
&nbsp;NEXT CLASS</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="TranscriptomeSpaceAlignmentModel.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_classes_inherited_from_class_nextgen.core.model.AlignmentModel">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
