<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:29 EDT 2013 -->
<TITLE>
Uses of Class nextgen.core.annotation.Gene
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class nextgen.core.annotation.Gene";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?nextgen/core/annotation//class-useGene.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Gene.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Class<br>nextgen.core.annotation.Gene</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.core.overlaputils"><B>broad.core.overlaputils</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.core.primer3"><B>broad.core.primer3</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.core.siphy.tools.conservation"><B>broad.core.siphy.tools.conservation</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.core.util"><B>broad.core.util</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.pda.annotation"><B>broad.pda.annotation</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.pda.gene"><B>broad.pda.gene</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.pda.rnai"><B>broad.pda.rnai</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.pda.seq.alignment"><B>broad.pda.seq.alignment</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.pda.seq.graph"><B>broad.pda.seq.graph</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.pda.seq.protection"><B>broad.pda.seq.protection</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#broad.pda.seq.segmentation"><B>broad.pda.seq.segmentation</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.alignment"><B>nextgen.core.alignment</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.annotation"><B>nextgen.core.annotation</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.coordinatesystem"><B>nextgen.core.coordinatesystem</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.debug"><B>nextgen.core.debug</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.feature"><B>nextgen.core.feature</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.model"><B>nextgen.core.model</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.normalize"><B>nextgen.core.normalize</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.scripture"><B>nextgen.core.scripture</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.structure"><B>nextgen.core.structure</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.core.writers"><B>nextgen.core.writers</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.editing"><B>nextgen.editing</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#nextgen.rnaprotein.analysis"><B>nextgen.rnaprotein.analysis</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.core.overlaputils"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/core/overlaputils/package-summary.html">broad.core.overlaputils</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/core/overlaputils/package-summary.html">broad.core.overlaputils</A> that return types with arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, boolean)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           boolean&nbsp;verbose)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map, boolean)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes,
           boolean&nbsp;verbose)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlapWithRandomizedPositions(java.util.Map, java.util.Map)">getOverlapWithRandomizedPositions</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Intersect a set of genes with another set and randomize positions within each gene in the second set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlapWithRandomizedPositions(java.util.Map, java.util.Map, java.util.Map)">getOverlapWithRandomizedPositions</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeFeatures)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Intersect a set of genes with another set and randomize positions within each gene in the second set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#mapGenesToOverlappers(java.util.Map, java.util.Map)">mapGenesToOverlappers</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                      java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;otherGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Map each gene to the collection of overlappers in another set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#mapGenesToOverlappers(java.util.Map, java.util.Map)">mapGenesToOverlappers</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                      java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;otherGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Map each gene to the collection of overlappers in another set</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../broad/core/overlaputils/package-summary.html">broad.core.overlaputils</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, boolean)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           boolean&nbsp;verbose)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, boolean)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           boolean&nbsp;verbose)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map, boolean)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes,
           boolean&nbsp;verbose)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map, boolean)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes,
           boolean&nbsp;verbose)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map, boolean)">getOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes,
           boolean&nbsp;verbose)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlapWithRandomizedPositions(java.util.Map, java.util.Map)">getOverlapWithRandomizedPositions</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Intersect a set of genes with another set and randomize positions within each gene in the second set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlapWithRandomizedPositions(java.util.Map, java.util.Map)">getOverlapWithRandomizedPositions</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Intersect a set of genes with another set and randomize positions within each gene in the second set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlapWithRandomizedPositions(java.util.Map, java.util.Map, java.util.Map)">getOverlapWithRandomizedPositions</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeFeatures)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Intersect a set of genes with another set and randomize positions within each gene in the second set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlapWithRandomizedPositions(java.util.Map, java.util.Map, java.util.Map)">getOverlapWithRandomizedPositions</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeFeatures)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Intersect a set of genes with another set and randomize positions within each gene in the second set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#getOverlapWithRandomizedPositions(java.util.Map, java.util.Map, java.util.Map)">getOverlapWithRandomizedPositions</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeFeatures)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Intersect a set of genes with another set and randomize positions within each gene in the second set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#mapGenesToOverlappers(java.util.Map, java.util.Map)">mapGenesToOverlappers</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                      java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;otherGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Map each gene to the collection of overlappers in another set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#mapGenesToOverlappers(java.util.Map, java.util.Map)">mapGenesToOverlappers</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                      java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;otherGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Map each gene to the collection of overlappers in another set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingBases(java.util.Map, java.util.Map)">numOverlappingBases</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping bases between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingBases(java.util.Map, java.util.Map)">numOverlappingBases</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping bases between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingBases(java.util.Map, java.util.Map, java.util.Map)">numOverlappingBases</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping bases between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingBases(java.util.Map, java.util.Map, java.util.Map)">numOverlappingBases</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping bases between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingBases(java.util.Map, java.util.Map, java.util.Map)">numOverlappingBases</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping bases between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingGenes(java.util.Map, java.util.Map)">numOverlappingGenes</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping genes between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingGenes(java.util.Map, java.util.Map)">numOverlappingGenes</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping genes between two gene sets</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingGenes(java.util.Map, java.util.Map, java.util.Map)">numOverlappingGenes</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping genes between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingGenes(java.util.Map, java.util.Map, java.util.Map)">numOverlappingGenes</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping genes between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#numOverlappingGenes(java.util.Map, java.util.Map, java.util.Map)">numOverlappingGenes</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
                    java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of overlapping genes between two gene sets and automatically exclude all genes in a set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#writeOverlap(java.util.Map, java.util.Map, java.util.Map, java.lang.String, boolean)">writeOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
             java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
             java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes,
             java.lang.String&nbsp;outFile,
             boolean&nbsp;randomizePositionsWithinEachGene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets, automatically excluding all genes in a set, and write to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#writeOverlap(java.util.Map, java.util.Map, java.util.Map, java.lang.String, boolean)">writeOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
             java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
             java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes,
             java.lang.String&nbsp;outFile,
             boolean&nbsp;randomizePositionsWithinEachGene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets, automatically excluding all genes in a set, and write to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#writeOverlap(java.util.Map, java.util.Map, java.util.Map, java.lang.String, boolean)">writeOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
             java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
             java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;excludeGenes,
             java.lang.String&nbsp;outFile,
             boolean&nbsp;randomizePositionsWithinEachGene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets, automatically excluding all genes in a set, and write to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#writeOverlap(java.util.Map, java.util.Map, java.lang.String, boolean)">writeOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
             java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
             java.lang.String&nbsp;outFile,
             boolean&nbsp;randomizePositionsWithinEachGene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and write to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>GeneSetIntersect.</B><B><A HREF="../../../../broad/core/overlaputils/GeneSetIntersect.html#writeOverlap(java.util.Map, java.util.Map, java.lang.String, boolean)">writeOverlap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
             java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;intersectGenes,
             java.lang.String&nbsp;outFile,
             boolean&nbsp;randomizePositionsWithinEachGene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get overlap between two gene sets and write to a file</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.core.primer3"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/core/primer3/package-summary.html">broad.core.primer3</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/core/primer3/package-summary.html">broad.core.primer3</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>qPCRPrimerDesigner.</B><B><A HREF="../../../../broad/core/primer3/qPCRPrimerDesigner.html#designIntronPrimers(broad.core.sequence.Sequence, nextgen.core.annotation.Gene, boolean, java.lang.String, java.lang.String, int, int, int)">designIntronPrimers</A></B>(<A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&nbsp;chr,
                    <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                    boolean&nbsp;repeatMask,
                    java.lang.String&nbsp;sequencePrimer,
                    java.lang.String&nbsp;sequencePrimerRevComp,
                    int&nbsp;numDesigns,
                    int&nbsp;min3,
                    int&nbsp;min5)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Set&lt;<A HREF="../../../../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>qPCRPrimerDesigner.</B><B><A HREF="../../../../broad/core/primer3/qPCRPrimerDesigner.html#designJenPrimers(broad.core.sequence.Sequence, nextgen.core.annotation.Gene, boolean, int)">designJenPrimers</A></B>(<A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&nbsp;chr,
                 <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                 boolean&nbsp;repeatMask,
                 int&nbsp;numDesigns)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>qPCRPrimerDesigner.</B><B><A HREF="../../../../broad/core/primer3/qPCRPrimerDesigner.html#designPCRPrimers(broad.core.sequence.Sequence, nextgen.core.annotation.Gene, boolean, int, boolean)">designPCRPrimers</A></B>(<A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&nbsp;chr,
                 <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                 boolean&nbsp;repeatMask,
                 int&nbsp;numDesigns,
                 boolean&nbsp;crossJunction)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>qPCRPrimerDesigner.</B><B><A HREF="../../../../broad/core/primer3/qPCRPrimerDesigner.html#designPCRPrimers(broad.core.sequence.Sequence, nextgen.core.annotation.Gene, boolean, int, broad.core.primer3.Primer3SequenceInputTags.SequenceRegionCoordinates)">designPCRPrimers</A></B>(<A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&nbsp;chr,
                 <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                 boolean&nbsp;repeatMask,
                 int&nbsp;numDesigns,
                 <A HREF="../../../../broad/core/primer3/Primer3SequenceInputTags.SequenceRegionCoordinates.html" title="class in broad.core.primer3">Primer3SequenceInputTags.SequenceRegionCoordinates</A>&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../broad/core/primer3/package-summary.html">broad.core.primer3</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/core/primer3/RNAiPCRPrimerDesigner.html#RNAiPCRPrimerDesigner(java.util.Map, java.lang.String, java.lang.String, boolean, int, int, int)">RNAiPCRPrimerDesigner</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;alignmentsByChr,
                      java.lang.String&nbsp;sequenceDirectory,
                      java.lang.String&nbsp;save,
                      boolean&nbsp;repeatMask,
                      int&nbsp;numDesigns,
                      int&nbsp;min3,
                      int&nbsp;min5)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.core.siphy.tools.conservation"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/core/siphy/tools/conservation/package-summary.html">broad.core.siphy.tools.conservation</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../broad/core/siphy/tools/conservation/package-summary.html">broad.core.siphy.tools.conservation</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/core/siphy/tools/conservation/ComputePercentageOfKmers.html#ComputePercentageOfKmers(java.io.File, java.util.Collection, java.lang.String, double)">ComputePercentageOfKmers</A></B>(java.io.File&nbsp;BEDScores,
                         java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                         java.lang.String&nbsp;save,
                         double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html#EstimateOmegaPerExon(java.util.Collection, java.io.File, java.lang.String, java.lang.String, java.lang.String)">EstimateOmegaPerExon</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                     java.io.File&nbsp;modelFile,
                     java.lang.String&nbsp;alnDir,
                     java.lang.String&nbsp;alnFileFormat,
                     java.lang.String&nbsp;save)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.core.util"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/core/util/package-summary.html">broad.core.util</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/core/util/package-summary.html">broad.core.util</A> that return types with arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>CollapseByIntersection.</B><B><A HREF="../../../../broad/core/util/CollapseByIntersection.html#makeIntervalTreeForGenes(java.util.Collection)">makeIntervalTreeForGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;alignments)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>CollapseByIntersection.</B><B><A HREF="../../../../broad/core/util/CollapseByIntersection.html#updateBoundariesWithoutCrossingIntrons(java.util.Collection)">updateBoundariesWithoutCrossingIntrons</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../broad/core/util/package-summary.html">broad.core.util</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>CollapseByIntersection.</B><B><A HREF="../../../../broad/core/util/CollapseByIntersection.html#CollapseGenesByIntersection(java.util.Collection, boolean)">CollapseGenesByIntersection</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;temp,
                            boolean&nbsp;intersection)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>CollapseByIntersection.</B><B><A HREF="../../../../broad/core/util/CollapseByIntersection.html#makeIntervalTreeForGeneExons(java.util.Collection)">makeIntervalTreeForGeneExons</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>CollapseByIntersection.</B><B><A HREF="../../../../broad/core/util/CollapseByIntersection.html#makeIntervalTreeForGenes(java.util.Collection)">makeIntervalTreeForGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;alignments)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>CollapseByIntersection.</B><B><A HREF="../../../../broad/core/util/CollapseByIntersection.html#updateBoundariesWithoutCrossingIntrons(java.util.Collection)">updateBoundariesWithoutCrossingIntrons</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.pda.annotation"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/pda/annotation/package-summary.html">broad.pda.annotation</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/annotation/package-summary.html">broad.pda.annotation</A> that return <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#getExactIsoform(broad.pda.gene.GeneWithIsoforms)">getExactIsoform</A></B>(<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>&nbsp;loci)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getExonUnion()">getExonUnion</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#getGene(java.lang.String, java.lang.String)">getGene</A></B>(java.lang.String&nbsp;geneFile,
        java.lang.String&nbsp;geneName)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getLongestIntornChainIso(java.lang.String)">getLongestIntornChainIso</A></B>(java.lang.String&nbsp;refSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getMerged()">getMerged</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/annotation/package-summary.html">broad.pda.annotation</A> that return types with arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getAllIsoforms()">getAllIsoforms</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getAllIsoforms(java.lang.String)">getAllIsoforms</A></B>(java.lang.String&nbsp;excludeSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getAllMerged()">getAllMerged</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getComaptibleBetweenAny2Sets(java.util.ArrayList, boolean)">getComaptibleBetweenAny2Sets</A></B>(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;setLst,
                             boolean&nbsp;isFully)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getComaptibleIsoforms(nextgen.core.annotation.Gene, boolean)">getComaptibleIsoforms</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;g,
                      boolean&nbsp;isFully)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#GetGenes()">GetGenes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.HashMap&lt;java.lang.String,<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#getNameGeneMap()">getNameGeneMap</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getSetIsoforms(java.lang.String)">getSetIsoforms</A></B>(java.lang.String&nbsp;setName)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getUnqComaptibleBetweenAny2Sets(java.util.ArrayList, java.util.Collection, boolean)">getUnqComaptibleBetweenAny2Sets</A></B>(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;setLst,
                                java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;unqIsoSet,
                                boolean&nbsp;isFully)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getUnqComaptibleBetweenSubsets(java.util.ArrayList, java.util.ArrayList, java.util.Collection, boolean)">getUnqComaptibleBetweenSubsets</A></B>(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;set1Lst,
                               java.util.ArrayList&lt;java.lang.String&gt;&nbsp;set2Lst,
                               java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;unqIsoSet,
                               boolean&nbsp;isFully)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#loadData(java.io.File)">loadData</A></B>(java.io.File&nbsp;file)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#loadData(java.io.File, broad.pda.datastructures.Alignments)">loadData</A></B>(java.io.File&nbsp;file,
         <A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#loadDataByChr(java.io.File)">loadDataByChr</A></B>(java.io.File&nbsp;file)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#loadDataByChr(java.lang.String)">loadDataByChr</A></B>(java.lang.String&nbsp;file)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#loadDataByChrToTree(java.io.File)">loadDataByChrToTree</A></B>(java.io.File&nbsp;file)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#loadDataByName(java.io.File)">loadDataByName</A></B>(java.io.File&nbsp;file)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#makeIntervalTreeFor3Prime(java.io.File, int)">makeIntervalTreeFor3Prime</A></B>(java.io.File&nbsp;geneFile,
                          int&nbsp;size)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.HashMap&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.ArrayList&lt;java.lang.Double&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#pctOverlapWithRef()">pctOverlapWithRef</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#SelectLongestIntronChainCandidate(java.util.Collection)">SelectLongestIntronChainCandidate</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;betweenSubsetsIsoSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#selectRandIsoSubset(java.lang.Integer)">selectRandIsoSubset</A></B>(java.lang.Integer&nbsp;maxIsoPerLoci)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#toConstituentIntroformMap()">toConstituentIntroformMap</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Builds an artificial map of isoforms where "constituent" introns are the exons.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#toConstituentIsoformMap()">toConstituentIsoformMap</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#toConstituentIsoformMap(boolean)">toConstituentIsoformMap</A></B>(boolean&nbsp;removeGenesWithoutConstituentIsoforms)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#toMap()">toMap</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/annotation/package-summary.html">broad.pda.annotation</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#addRefSeq(nextgen.core.annotation.Gene)">addRefSeq</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A></CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#getClosestDownstream(nextgen.core.annotation.Gene)">getClosestDownstream</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;annot)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A></CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#getClosestUpstream(nextgen.core.annotation.Gene)">getClosestUpstream</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;annot)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getComaptibleIsoforms(nextgen.core.annotation.Gene, boolean)">getComaptibleIsoforms</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;g,
                      boolean&nbsp;isFully)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#getOverlappers(nextgen.core.annotation.Gene)">getOverlappers</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#isFullyComaptible(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, boolean)">isFullyComaptible</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;iso1,
                  <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;iso2,
                  boolean&nbsp;stringent)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#isOverlapCompatible(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, double)">isOverlapCompatible</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;first,
                    <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;second,
                    double&nbsp;minPctOverlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../broad/pda/annotation/package-summary.html">broad.pda.annotation</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#addRefSeqSet(java.util.Collection)">addRefSeqSet</A></B>(java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;collection)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#filterByOverlap(java.util.Collection)">filterByOverlap</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getUnqComaptibleBetweenAny2Sets(java.util.ArrayList, java.util.Collection, boolean)">getUnqComaptibleBetweenAny2Sets</A></B>(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;setLst,
                                java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;unqIsoSet,
                                boolean&nbsp;isFully)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#getUnqComaptibleBetweenSubsets(java.util.ArrayList, java.util.ArrayList, java.util.Collection, boolean)">getUnqComaptibleBetweenSubsets</A></B>(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;set1Lst,
                               java.util.ArrayList&lt;java.lang.String&gt;&nbsp;set2Lst,
                               java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;unqIsoSet,
                               boolean&nbsp;isFully)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#minus(java.util.Collection, boolean)">minus</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
      boolean&nbsp;recursive)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Removes from this reader all elements that overlap the given list.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Locus.</B><B><A HREF="../../../../broad/pda/annotation/Locus.html#SelectLongestIntronChainCandidate(java.util.Collection)">SelectLongestIntronChainCandidate</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;betweenSubsetsIsoSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#writeFullBED(java.lang.String, java.util.Collection)">writeFullBED</A></B>(java.lang.String&nbsp;save,
             java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BEDFileParser.</B><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#writeFullBED(java.lang.String, java.util.Map)">writeFullBED</A></B>(java.lang.String&nbsp;save,
             java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,? extends java.lang.Object&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../broad/pda/annotation/package-summary.html">broad.pda.annotation</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#BEDFileParser(java.util.Collection)">BEDFileParser</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/annotation/BEDFileParser.html#BEDFileParser(java.util.Map)">BEDFileParser</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;transcripts)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.pda.gene"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/pda/gene/package-summary.html">broad.pda.gene</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Subclasses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/pda/gene/package-summary.html">broad.pda.gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/gene/package-summary.html">broad.pda.gene</A> that return <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#constituentIntrons()">constituentIntrons</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs an artificial gene where the constituent introns (regions that are intronic in all isoforms) are the exons of the gene.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#constituentIsoform()">constituentIsoform</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#findCompatibleGenes(broad.core.datastructures.IntervalTree, int[])">findCompatibleGenes</A></B>(<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>&gt;&nbsp;overlapTree,
                    int[]&nbsp;numIntrons)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#getMerged()">getMerged</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#selectHighScoringIsoform(nextgen.core.annotation.Gene, broad.pda.annotation.BEDFileParser)">selectHighScoringIsoform</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;transcript,
                         <A HREF="../../../../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>&nbsp;set)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/gene/package-summary.html">broad.pda.gene</A> that return types with arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#getAllIsoforms()">getAllIsoforms</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#getAllIsoforms(boolean)">getAllIsoforms</A></B>(boolean&nbsp;includeContainingIsoform)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;In cases were the containg gene is artificial, like when building gene loci and adding
 all isoforms contain in the loci.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.HashMap&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.ArrayList&lt;<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>[]&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>NeighborsNullModel.</B><B><A HREF="../../../../broad/pda/gene/NeighborsNullModel.html#getGeneRandNeighbors()">getGeneRandNeighbors</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>functionalAnnotation.</B><B><A HREF="../../../../broad/pda/gene/functionalAnnotation.html#getTermGenes(java.lang.String, broad.pda.annotation.BEDFileParser)">getTermGenes</A></B>(java.lang.String&nbsp;goTerm,
             <A HREF="../../../../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>&nbsp;ref)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeighborsNullModel.</B><B><A HREF="../../../../broad/pda/gene/NeighborsNullModel.html#getUniqueNeighbors()">getUniqueNeighbors</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#merge(java.util.Collection)">merge</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;original)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#merge(java.util.Map)">merge</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;originalMap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#mergeSets(java.util.Map, java.util.Map)">mergeSets</A></B>(java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set1,
          java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#overlap(java.util.Map, java.util.Map, java.lang.Boolean)">overlap</A></B>(java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set1,
        java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set2,
        java.lang.Boolean&nbsp;byExon)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#selectRandIsoSubset(java.lang.Integer)">selectRandIsoSubset</A></B>(java.lang.Integer&nbsp;maxIsoPerLoci)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/gene/package-summary.html">broad.pda.gene</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#addContainedIsoform(nextgen.core.annotation.Gene)">addContainedIsoform</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;isoform)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#AddIsoform(nextgen.core.annotation.Gene)">AddIsoform</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#AddIsoform(nextgen.core.annotation.Gene, boolean)">AddIsoform</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
           boolean&nbsp;updateMainIsoform)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#almostContains(nextgen.core.annotation.Gene, int)">almostContains</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other,
               int&nbsp;exonFudgeFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.ArrayList&lt;<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>[]&gt;</CODE></FONT></TD>
<TD><CODE><B>NeighborsNullModel.</B><B><A HREF="../../../../broad/pda/gene/NeighborsNullModel.html#getGeneRandNeighbors(nextgen.core.annotation.Gene)">getGeneRandNeighbors</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A></CODE></FONT></TD>
<TD><CODE><B>NeighborAnalysis.</B><B><A HREF="../../../../broad/pda/gene/NeighborAnalysis.html#getLeftNeighbor(nextgen.core.annotation.Gene, broad.pda.annotation.BEDFileParser)">getLeftNeighbor</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;g,
                <A HREF="../../../../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>&nbsp;ref)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>NeighborAnalysis.</B><B><A HREF="../../../../broad/pda/gene/NeighborAnalysis.html#getNeighborDistance(nextgen.core.annotation.Gene)">getNeighborDistance</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>[]</CODE></FONT></TD>
<TD><CODE><B>NeighborAnalysis.</B><B><A HREF="../../../../broad/pda/gene/NeighborAnalysis.html#getNeighbors(nextgen.core.annotation.Gene)">getNeighbors</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A></CODE></FONT></TD>
<TD><CODE><B>NeighborAnalysis.</B><B><A HREF="../../../../broad/pda/gene/NeighborAnalysis.html#getRightNeighbor(nextgen.core.annotation.Gene, broad.pda.annotation.BEDFileParser)">getRightNeighbor</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;g,
                 <A HREF="../../../../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>&nbsp;ref)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#IsExactOverlappingIsoform(nextgen.core.annotation.Gene)">IsExactOverlappingIsoform</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#orientationForGene(nextgen.core.annotation.Gene, broad.core.sequence.Sequence)">orientationForGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;g,
                   <A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&nbsp;cSeq)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#overlaps(nextgen.core.annotation.Gene)">overlaps</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>GeneWithIsoforms.</B><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#overlapsExon(nextgen.core.annotation.Gene)">overlapsExon</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#selectHighScoringIsoform(nextgen.core.annotation.Gene, broad.pda.annotation.BEDFileParser)">selectHighScoringIsoform</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;transcript,
                         <A HREF="../../../../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>&nbsp;set)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../broad/pda/gene/package-summary.html">broad.pda.gene</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>functionalAnnotation.</B><B><A HREF="../../../../broad/pda/gene/functionalAnnotation.html#calcAllTermsEnrichment(java.util.Collection, java.util.LinkedList)">calcAllTermsEnrichment</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;geneSet,
                       java.util.LinkedList&lt;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;randomGeneSets)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>functionalAnnotation.</B><B><A HREF="../../../../broad/pda/gene/functionalAnnotation.html#calcAllTermsEnrichment(java.util.Collection, java.util.LinkedList)">calcAllTermsEnrichment</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;geneSet,
                       java.util.LinkedList&lt;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;randomGeneSets)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A></CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#intersectGeneSets(java.util.Collection, java.util.Collection, boolean)">intersectGeneSets</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set1,
                  java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set2,
                  boolean&nbsp;takeComplement)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A></CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#intersectGeneSets(java.util.Collection, java.util.Collection, boolean)">intersectGeneSets</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set1,
                  java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set2,
                  boolean&nbsp;takeComplement)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#merge(java.util.Collection)">merge</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;original)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#merge(java.util.Map)">merge</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;originalMap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#mergeSets(java.util.Map, java.util.Map)">mergeSets</A></B>(java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set1,
          java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#mergeSets(java.util.Map, java.util.Map)">mergeSets</A></B>(java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set1,
          java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#overlap(java.util.Map, java.util.Map, java.lang.Boolean)">overlap</A></B>(java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set1,
        java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set2,
        java.lang.Boolean&nbsp;byExon)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>GeneTools.</B><B><A HREF="../../../../broad/pda/gene/GeneTools.html#overlap(java.util.Map, java.util.Map, java.lang.Boolean)">overlap</A></B>(java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set1,
        java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;set2,
        java.lang.Boolean&nbsp;byExon)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../broad/pda/gene/package-summary.html">broad.pda.gene</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(nextgen.core.annotation.Gene)">GeneWithIsoforms</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.pda.rnai"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/pda/rnai/package-summary.html">broad.pda.rnai</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/rnai/package-summary.html">broad.pda.rnai</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ExtractSequence.</B><B><A HREF="../../../../broad/pda/rnai/ExtractSequence.html#getSequenceForGene(nextgen.core.annotation.Gene, broad.core.sequence.Sequence, boolean, java.util.Map)">getSequenceForGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                   <A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&nbsp;chrom,
                   boolean&nbsp;repeatMask,
                   java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&gt;&gt;&nbsp;okRepeats)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ExtractSequence.</B><B><A HREF="../../../../broad/pda/rnai/ExtractSequence.html#getSequenceForGene(nextgen.core.annotation.Gene, broad.core.sequence.Sequence, boolean, java.util.Map, boolean)">getSequenceForGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                   <A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&nbsp;chrom,
                   boolean&nbsp;repeatMask,
                   java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&gt;&gt;&nbsp;okRepeats,
                   boolean&nbsp;stranded)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ExtractSequence.</B><B><A HREF="../../../../broad/pda/rnai/ExtractSequence.html#getSequenceForGene(nextgen.core.annotation.Gene, java.lang.String, boolean)">getSequenceForGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                   java.lang.String&nbsp;genomeDirectory,
                   boolean&nbsp;repeatMask)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../broad/pda/rnai/package-summary.html">broad.pda.rnai</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map</CODE></FONT></TD>
<TD><CODE><B>ExtractSequence.</B><B><A HREF="../../../../broad/pda/rnai/ExtractSequence.html#getSequenceForGene(java.util.Collection, java.lang.String, boolean, java.util.Map)">getSequenceForGene</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;alignments,
                   java.lang.String&nbsp;genomeDirectory,
                   boolean&nbsp;repeatMask,
                   java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&gt;&gt;&nbsp;okRepeats)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map</CODE></FONT></TD>
<TD><CODE><B>ExtractSequence.</B><B><A HREF="../../../../broad/pda/rnai/ExtractSequence.html#splitByChr(java.util.Collection)">splitByChr</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>ExtractSequence.</B><B><A HREF="../../../../broad/pda/rnai/ExtractSequence.html#writeFasta(java.lang.String, java.util.Map, java.lang.String, boolean)">writeFasta</A></B>(java.lang.String&nbsp;save,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genesByChr,
           java.lang.String&nbsp;genomeDir,
           boolean&nbsp;repeatMask)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../broad/pda/rnai/package-summary.html">broad.pda.rnai</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/rnai/RNAiGeneAnnotation.html#RNAiGeneAnnotation(nextgen.core.annotation.Gene, broad.pda.datastructures.Alignments, java.lang.String, int, java.lang.String, java.lang.String, java.lang.String)">RNAiGeneAnnotation</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                   <A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;linc,
                   java.lang.String&nbsp;geneName,
                   int&nbsp;versionNumber,
                   java.lang.String&nbsp;seq,
                   java.lang.String&nbsp;species,
                   java.lang.String&nbsp;ncbiBuild)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.pda.seq.alignment"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/pda/seq/alignment/package-summary.html">broad.pda.seq.alignment</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/seq/alignment/package-summary.html">broad.pda.seq.alignment</A> that return <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>AlignmentUtils.</B><B><A HREF="../../../../broad/pda/seq/alignment/AlignmentUtils.html#SAMFormatFullBED(net.sf.samtools.SAMRecord)">SAMFormatFullBED</A></B>(net.sf.samtools.SAMRecord&nbsp;sam)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.pda.seq.graph"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A> that return <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Path.</B><B><A HREF="../../../../broad/pda/seq/graph/Path.html#toFilledInGene()">toFilledInGene</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Path.</B><B><A HREF="../../../../broad/pda/seq/graph/Path.html#toGene()">toGene</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Path.</B><B><A HREF="../../../../broad/pda/seq/graph/Path.html#toGene(double[])">toGene</A></B>(double[]&nbsp;extraFields)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A> that return types with arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getBackwardPaths(int, int, double, java.util.Collection)">getBackwardPaths</A></B>(int&nbsp;start,
                 int&nbsp;end,
                 double&nbsp;alpha,
                 java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getGenePaths(double)">getGenePaths</A></B>(double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of paired end edges</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getGenePaths(double, double)">getGenePaths</A></B>(double&nbsp;alpha,
             double&nbsp;minSpliceFrequency)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOrphanNodes()">getOrphanNodes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns all orphan nodes in graph (Vertices without any edges orphan vertices)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOrphanNodes(broad.pda.datastructures.Alignments)">getOrphanNodes</A></B>(<A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns all orphan nodes in region (Vertices without any edges orphan vertices)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int)">getPaths</A></B>(int&nbsp;start,
         int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DO WE NEED THIS????
 Returns collection of genes between start and end</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int, boolean)">getPaths</A></B>(int&nbsp;start,
         int&nbsp;end,
         boolean&nbsp;forward)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int, double, boolean)">getPaths</A></B>(int&nbsp;start,
         int&nbsp;end,
         double&nbsp;alpha,
         boolean&nbsp;forward)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int, double, java.util.Collection)">getPaths</A></B>(int&nbsp;start,
         int&nbsp;end,
         double&nbsp;alpha,
         java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.WindowIterator.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html#next()">next</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.html#next()">next</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.ReverseWindowIterator.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ReverseWindowIterator.html#next()">next</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.WindowIterator.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html#next(int, int)">next</A></B>(int&nbsp;lowerBound,
     int&nbsp;higherBound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.html#next(int, int)">next</A></B>(int&nbsp;lowerBound,
     int&nbsp;higherBound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeWithBubblesJGraphT.ReverseWindowIterator.</B><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ReverseWindowIterator.html#next(int, int)">next</A></B>(int&nbsp;lowerBound,
     int&nbsp;higherBound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/Path.html#Path(nextgen.core.annotation.Gene, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT)">Path</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
     <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>&nbsp;graph)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.pda.seq.protection"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A> that return types with arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>TwoSampleScanSkellamAnalysis.</B><B><A HREF="../../../../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getGenes()">getGenes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the gene set</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getEnrichmentOverGene(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)">getEnrichmentOverGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                      <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get enrichment of a window over a gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getGeneAverageCoverage(nextgen.core.annotation.Gene)">getGeneAverageCoverage</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get average coverage of gene
 Get score from cache or calculate and cache score</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getGeneCount(nextgen.core.annotation.Gene)">getGeneCount</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of fragments mapping to gene
 Get score from cache or calculate and cache score</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getGeneScanPval(nextgen.core.annotation.Gene)">getGeneScanPval</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get coordinate space wide scan P value of number of fragments mapping to the gene
 Get score from cache or calculate and cache score</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>GenomeSpaceSampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/GenomeSpaceSampleData.html#getGeneScanPval(nextgen.core.annotation.Gene)">getGeneScanPval</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get genome wide scan P value of number of fragments mapping to the gene
 Get score from cache or calculate and cache score</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>TwoSampleScanSkellamAnalysis.</B><B><A HREF="../../../../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getSignificantPeaks(nextgen.core.annotation.Gene)">getSignificantPeaks</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get significant peaks for a gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>TwoSampleScanSkellamAnalysis.</B><B><A HREF="../../../../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getSignificantPeaks(nextgen.core.annotation.Gene, boolean)">getSignificantPeaks</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                    boolean&nbsp;checkBackgroundExpression)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get significant peaks for a gene providing expression P value</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>MultiSampleScanPeakCaller.</B><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#getSingleSampleScanPeaks(broad.pda.seq.protection.SampleData, nextgen.core.annotation.Gene)">getSingleSampleScanPeaks</A></B>(<A HREF="../../../../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>&nbsp;sample,
                         <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scan peaks for the sample and the gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getWindowCount(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)">getWindowCount</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
               <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of fragments mapping to gene window
 Get score from cache or calculate and cache score</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getWindowScores(nextgen.core.annotation.Gene)">getWindowScores</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scores for each window in the gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#isExpressed(nextgen.core.annotation.Gene)">isExpressed</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the gene is significantly expressed in the sample</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>MultiSampleScanPeakCaller.</B><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#isExpressed(nextgen.core.annotation.Gene)">isExpressed</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the gene is significantly expressed in the special expression sample</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>GenomeSpaceSampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/GenomeSpaceSampleData.html#isExpressed(nextgen.core.annotation.Gene)">isExpressed</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the gene is significantly expressed genome wide</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>MultiSampleScanPeakCaller.</B><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#isExpressedInAllControlSamples(nextgen.core.annotation.Gene)">isExpressedInAllControlSamples</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the gene is expressed in all control samples at the given significance level</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;net.sf.samtools.SAMRecord</CODE></FONT></TD>
<TD><CODE><B>ReadSimulator.</B><B><A HREF="../../../../broad/pda/seq/protection/ReadSimulator.html#mapSamRecord(net.sf.samtools.SAMRecord, nextgen.core.annotation.Gene, net.sf.samtools.SAMFileHeader)">mapSamRecord</A></B>(net.sf.samtools.SAMRecord&nbsp;sam,
             <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
             net.sf.samtools.SAMFileHeader&nbsp;header)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A></CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#scoreWindow(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)">scoreWindow</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
            <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scan statistic score for a specified window</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A></CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#scoreWindow(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation, double)">scoreWindow</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
            <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window,
            double&nbsp;count)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scan statistic score for a specified window</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A></CODE></FONT></TD>
<TD><CODE><B>SampleData.</B><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#scoreWindowWithFragmentLengthFilter(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)">scoreWindowWithFragmentLengthFilter</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                    <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scan statistic score for a window based on filtering fragments overlapping the window by size</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>TwoSampleScanSkellamAnalysis.</B><B><A HREF="../../../../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#signalScanPval(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)">signalScanPval</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;parentGene,
               <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scan P value in signal dataset with respect to the transcript</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/GenomeSpaceSampleData.html#GenomeSpaceSampleData(java.lang.String, java.lang.String, java.util.Map, int, int, double)">GenomeSpaceSampleData</A></B>(java.lang.String&nbsp;bamFile,
                      java.lang.String&nbsp;chrSizeFile,
                      java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                      int&nbsp;window,
                      int&nbsp;step,
                      double&nbsp;expressionCutoff)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#SampleData(java.lang.String, boolean, java.util.Map, int, int, double, boolean)">SampleData</A></B>(java.lang.String&nbsp;bamFile,
           boolean&nbsp;firstReadTranscriptionStrand,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           int&nbsp;window,
           int&nbsp;step,
           double&nbsp;expressionCutoff,
           boolean&nbsp;expByScanPval)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#TwoSampleScanSkellamAnalysis(java.lang.String, java.lang.String, java.util.Map, int, int, double, double, double)">TwoSampleScanSkellamAnalysis</A></B>(java.lang.String&nbsp;backgroundAlignmentFile,
                             java.lang.String&nbsp;signalAlignmentFile,
                             java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genesByChr,
                             int&nbsp;window,
                             int&nbsp;step,
                             double&nbsp;alphaSkellam,
                             double&nbsp;alphaScan,
                             double&nbsp;trimQuantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct object from bam files</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="broad.pda.seq.segmentation"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Subclasses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A> that return <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GenomeWithGaps.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenomeWithGaps.html#getNextWindow()">getNextWindow</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<B>Deprecated.</B>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GenomeWithGaps.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenomeWithGaps.html#getNextWindow(int)">getNextWindow</A></B>(int&nbsp;position)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<B>Deprecated.</B>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon,
        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon,
        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon,
        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GenomeWithGaps2.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenomeWithGaps2.html#getRelativeWindow(int, int)">getRelativeWindow</A></B>(int&nbsp;relativeStart,
                  int&nbsp;relativeEnd)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#trimGeneEnds(nextgen.core.annotation.Gene, double)">trimGeneEnds</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
             double&nbsp;quantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#trimMaxContiguous(nextgen.core.annotation.Gene, double)">trimMaxContiguous</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;region,
                  double&nbsp;quantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Trim a region to a contiguous subregion with max number of positions having coverage above the quantile</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#updateGeneByFirstCounts(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">updateGeneByFirstCounts</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#updateGeneByFirstCounts(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">updateGeneByFirstCounts</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A> that return types with arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#filterGenesByExpression(java.util.Map)">filterGenesByExpression</A></B>(java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.List&lt;java.lang.Double&gt;&gt;&nbsp;countsMap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This function will filter the genes in the input Map by a number of constraints on the total number of reads on the genes
 1.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#filterPaths(java.util.Map, double)">filterPaths</A></B>(java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt;&nbsp;scores,
            double&nbsp;alpha2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.List&lt;java.lang.Double&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getDataForGene(java.util.Collection)">getDataForGene</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;Annotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneExpression(java.util.Collection)">getGeneExpression</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getGeneExpression(java.util.Collection, int)">getGeneExpression</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                  int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
           java.lang.String&nbsp;chrToUse)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String, int)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
           java.lang.String&nbsp;chrToUse,
           int&nbsp;size)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String, broad.core.datastructures.IntervalTree)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
           java.lang.String&nbsp;chrToUse,
           <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#CV(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">CV</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
   <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getBasesCoveredPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getBasesCoveredPerAlignment</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                            <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                            int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#getBasesCoveredPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getBasesCoveredPerAlignment</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                            <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                            int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsOfUniqueOverlappers(nextgen.core.annotation.Annotation, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getCountsOfUniqueOverlappers</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;target,
                             <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;exclude,
                             <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                             int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsOfUniqueOverlappers(nextgen.core.annotation.Annotation, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getCountsOfUniqueOverlappers</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;target,
                             <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;exclude,
                             <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                             int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getCountsPerAlignment</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                      <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                      int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getCountsPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getCountsPerAlignment</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;window,
                      <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                      int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getCountsPerAlignment</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                      <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                      int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignmentFullyContained(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getCountsPerAlignmentFullyContained</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                    <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                                    int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAlignmentFullyContained(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getCountsPerAlignmentFullyContained</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                    <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                                    int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAnnotationtranded(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, java.lang.String)">getCountsPerAnnotationtranded</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                              <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                              int&nbsp;EF,
                              java.lang.String&nbsp;orientation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAnnotationtranded(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, java.lang.String)">getCountsPerAnnotationtranded</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                              <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                              int&nbsp;EF,
                              java.lang.String&nbsp;orientation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;java.lang.Double&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getCountsPerBp(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">getCountsPerBp</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
               <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;java.lang.Double&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getDataForGene(nextgen.core.annotation.Gene)">getDataForGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/GeneCounts.html" title="class in broad.pda.seq.segmentation">GeneCounts</A></CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getGeneCounts(nextgen.core.annotation.Gene, int)">getGeneCounts</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
              int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/GeneCounts.html" title="class in broad.pda.seq.segmentation">GeneCounts</A></CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#getGeneCounts(nextgen.core.annotation.Gene, int)">getGeneCounts</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
              int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A></CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon,
        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon,
        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon,
        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon,
        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getScorePerAlignmentFromCache(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getScorePerAlignmentFromCache</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;window,
                              <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;chunkAlignmentTree,
                              int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getScorePerAlignmentFromCache(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getScorePerAlignmentFromCache</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;window,
                              <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;chunkAlignmentTree,
                              int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#getScorePerAlignmentFromCache(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">getScorePerAlignmentFromCache</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;window,
                              <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;chunkAlignmentTree,
                              int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#passes(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, double)">passes</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
       <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
       int&nbsp;EF,
       double&nbsp;cutoff)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#passes(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, double)">passes</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
       <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
       int&nbsp;EF,
       double&nbsp;cutoff)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#passes(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, int)">passes</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;window,
       <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;chunkAlignmentTree,
       int&nbsp;i,
       int&nbsp;critVal)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Gene, int)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda,
          int&nbsp;numMarkers)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda,
          int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int, double)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda,
          int&nbsp;extensionFactor,
          double&nbsp;numMarkers)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, double)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
          double&nbsp;localLambda)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A></CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A></CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda,
          int&nbsp;numMarkers)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#trimGeneEnds(nextgen.core.annotation.Gene, double)">trimGeneEnds</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
             double&nbsp;quantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#trimMaxContiguous(nextgen.core.annotation.Gene, double)">trimMaxContiguous</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;region,
                  double&nbsp;quantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Trim a region to a contiguous subregion with max number of positions having coverage above the quantile</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>GenericAlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#updateGeneByFirstCounts(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">updateGeneByFirstCounts</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html#updateGeneByFirstCounts(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)">updateGeneByFirstCounts</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                        int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#filterGenesByExpression(java.util.Map)">filterGenesByExpression</A></B>(java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.List&lt;java.lang.Double&gt;&gt;&nbsp;countsMap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This function will filter the genes in the input Map by a number of constraints on the total number of reads on the genes
 1.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.List&lt;java.lang.Double&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getDataForGene(java.util.Collection)">getDataForGene</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;Annotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneCounts(java.util.Collection)">getGeneCounts</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getGeneCounts(java.util.Collection, int)">getGeneCounts</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
              int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneCountsWithoutDatasetTotals(java.util.Collection)">getGeneCountsWithoutDatasetTotals</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getGeneCountsWithoutDatasetTotals(java.util.Collection, int)">getGeneCountsWithoutDatasetTotals</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                                  int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get read counts mapping to exons, introns, 5'UTRs and 3'UTRs in gene set of interest</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneExpression(java.util.Collection)">getGeneExpression</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>AlignmentDataModelStats.</B><B><A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getGeneExpression(java.util.Collection, int)">getGeneExpression</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                  int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
           java.lang.String&nbsp;chrToUse)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String, int)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
           java.lang.String&nbsp;chrToUse,
           int&nbsp;size)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String, broad.core.datastructures.IntervalTree)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
           java.lang.String&nbsp;chrToUse,
           <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#writeFullBED(java.lang.String, java.util.Collection)">writeFullBED</A></B>(java.lang.String&nbsp;save,
             java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;segments)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>ContinuousDataAlignmentModel.</B><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#writeFullBED(java.lang.String, java.util.Map, double)">writeFullBED</A></B>(java.lang.String&nbsp;save,
             java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;&nbsp;map,
             double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html#GeneScore(nextgen.core.annotation.Gene)">GeneScore</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html#GeneScore(nextgen.core.annotation.Gene, double[])">GeneScore</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          double[]&nbsp;scores)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A class to represent the standard scores assigned to a window or Gene in the ContinuousDataAlignmentModel</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.alignment"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A> that return <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>FeatureSequenceAlignment.UnorderedGenePair.</B><B><A HREF="../../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html#getFirstGene()">getFirstGene</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>FeatureSequenceAlignment.UnorderedGenePair.</B><B><A HREF="../../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html#getSecondGene()">getSecondGene</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html#FeatureSequenceAlignment.UnorderedGenePair(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene)">FeatureSequenceAlignment.UnorderedGenePair</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene1,
                                           <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Determine which gene is first and which is second</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.util.Collection, java.util.Map, float, float, float, float)">FeatureSequenceAlignment</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                         java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&gt;&nbsp;chrs,
                         float&nbsp;matchScore,
                         float&nbsp;mismatchScore,
                         float&nbsp;gapOpenPenalty,
                         float&nbsp;gapExtendPenalty)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.util.Collection, java.lang.String)">FeatureSequenceAlignment</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                         java.lang.String&nbsp;genomeFasta)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.util.Collection, java.lang.String, float, float, float, float)">FeatureSequenceAlignment</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes,
                         java.lang.String&nbsp;genomeFasta,
                         float&nbsp;matchScore,
                         float&nbsp;mismatchScore,
                         float&nbsp;gapOpenPenalty,
                         float&nbsp;gapExtendPenalty)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.annotation"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A> that return <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#copy()">copy</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#extendAnnotation(int, int)">extendAnnotation</A></B>(int&nbsp;deltaStart,
                 int&nbsp;deltaEnd)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get new gene extended in either direction</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#findLongestORF()">findLongestORF</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#get3UTRGene()">get3UTRGene</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#get5UTRGene()">get5UTRGene</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#getCDS()">getCDS</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#getIntrons()">getIntrons</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get introns</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#getIntronTranscript()">getIntronTranscript</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#getOverlap(java.util.Collection)">getOverlap</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;others)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all overlaps between this gene and any gene in the collection</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#getOverlap(nextgen.core.annotation.Gene)">getOverlap</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;second)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#getStartCodon()">getStartCodon</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#takeUnion(nextgen.core.annotation.Gene)">takeUnion</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#transcriptToGenomicPosition(int, int)">transcriptToGenomicPosition</A></B>(int&nbsp;startPosOnTranscript,
                            int&nbsp;endPosOnTranscript)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convert an interval in transcriptome space to genome space</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A> that return types with arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#findAllORFs(boolean)">findAllORFs</A></B>(boolean&nbsp;trim3UTRs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all ORFs as gene objects</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#findAllORFs(broad.core.sequence.Sequence, boolean)">findAllORFs</A></B>(<A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&nbsp;chrSeq,
            boolean&nbsp;trim3UTRs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#getIsoforms()">getIsoforms</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#addIsoform(nextgen.core.annotation.Gene)">addIsoform</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#almostContains(nextgen.core.annotation.Gene, int)">almostContains</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other,
               int&nbsp;exonFudgeFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#almostEqual(nextgen.core.annotation.Gene, int)">almostEqual</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other,
            int&nbsp;exonFudgeFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#almostSameStructure(nextgen.core.annotation.Gene, int)">almostSameStructure</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other,
                    int&nbsp;intronFudgeFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#geneSpanContains(nextgen.core.annotation.Gene)">geneSpanContains</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#getOverlap(nextgen.core.annotation.Gene)">getOverlap</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;second)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#isFullyCompatible(nextgen.core.annotation.Gene)">isFullyCompatible</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;iso)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#numOfCompatibleIntrons(nextgen.core.annotation.Gene)">numOfCompatibleIntrons</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#numOverlappingExons(nextgen.core.annotation.Gene)">numOverlappingExons</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;iso)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#overlaps(nextgen.core.annotation.Gene)">overlaps</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether an exon of this gene overlaps an exon of other gene and the two genes have the same orientation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#overlaps(nextgen.core.annotation.Gene, boolean)">overlaps</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other,
         boolean&nbsp;ignoreOrientation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether an exon of this gene overlaps an exon of other gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#overlaps(nextgen.core.annotation.Gene, double)">overlaps</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other,
         double&nbsp;minPctOverlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if this and the other gene overlap but at least minPctOverlap</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#overlapsGene(nextgen.core.annotation.Gene)">overlapsGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gen)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#overlapsGeneInAnyOrientation(nextgen.core.annotation.Gene)">overlapsGeneInAnyOrientation</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;otherGene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#overlapsStranded(nextgen.core.annotation.Gene, double)">overlapsStranded</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other,
                 double&nbsp;minPctOverlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if this and the other gene overlap but at least minPctOverlap</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Double</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#percentGenomeOverlapping(nextgen.core.annotation.Gene)">percentGenomeOverlapping</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#percentOverlapping(nextgen.core.annotation.Gene)">percentOverlapping</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#takeUnion(nextgen.core.annotation.Gene)">takeUnion</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#getOverlap(java.util.Collection)">getOverlap</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;others)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get all overlaps between this gene and any gene in the collection</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Gene.</B><B><A HREF="../../../../nextgen/core/annotation/Gene.html#overlapsGene(java.util.Collection)">overlapsGene</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;others)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether this gene overlaps any gene in the collection at the exon level</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/annotation/Gene.html#Gene(nextgen.core.annotation.Gene)">Gene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.coordinatesystem"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.TreeMap&lt;java.lang.Integer,java.lang.Double&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptomeSpace.</B><B><A HREF="../../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getPositionCountMap(nextgen.core.annotation.Gene, nextgen.core.model.AlignmentModel)">getPositionCountMap</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;region,
                    <A HREF="../../../../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>&nbsp;data)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get map of position to of position level counts in mature transcript</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Iterator&lt;<A HREF="../../../../nextgen/core/feature/Window.html" title="interface in nextgen.core.feature">Window</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptomeSpace.</B><B><A HREF="../../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getWindowIterator(java.util.Collection, int, int)">getWindowIterator</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;baseGenes,
                  int&nbsp;windowSize,
                  int&nbsp;overlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Iterator&lt;<A HREF="../../../../nextgen/core/feature/Window.html" title="interface in nextgen.core.feature">Window</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>MaskedGenomicSpace.</B><B><A HREF="../../../../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getWindowIterator(java.util.Collection, int, int)">getWindowIterator</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;baseGenes,
                  int&nbsp;windowSize,
                  int&nbsp;overlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Iterator&lt;<A HREF="../../../../nextgen/core/feature/Window.html" title="interface in nextgen.core.feature">Window</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>GenomicSpace.</B><B><A HREF="../../../../nextgen/core/coordinatesystem/GenomicSpace.html#getWindowIterator(java.util.Collection, int, int)">getWindowIterator</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;baseGenes,
                  int&nbsp;windowSize,
                  int&nbsp;overlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Iterator&lt;<A HREF="../../../../nextgen/core/feature/Window.html" title="interface in nextgen.core.feature">Window</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>CoordinateSpace.</B><B><A HREF="../../../../nextgen/core/coordinatesystem/CoordinateSpace.html#getWindowIterator(java.util.Collection, int, int)">getWindowIterator</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;baseGenes,
                  int&nbsp;windowSize,
                  int&nbsp;overlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get a window iterator over a collection of GeneWindows</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;T extends <A HREF="../../../../nextgen/core/model/score/WindowScore.html" title="interface in nextgen.core.model.score">WindowScore</A>&gt; 
<BR>
<A HREF="../../../../nextgen/core/model/score/WindowScoreIterator.html" title="class in nextgen.core.model.score">WindowScoreIterator</A>&lt;T&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>TranscriptomeSpace.</B><B><A HREF="../../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html#scan(java.util.Collection, int, int, nextgen.core.model.score.WindowProcessor)">scan</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;baseGenes,
     int&nbsp;windowSize,
     int&nbsp;overlap,
     <A HREF="../../../../nextgen/core/model/score/WindowProcessor.html" title="interface in nextgen.core.model.score">WindowProcessor</A>&lt;T&gt;&nbsp;processor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scan windows over a Collection<GeneWindow></TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html#TranscriptomeSpace(java.io.File, java.util.Map, java.lang.String)">TranscriptomeSpace</A></B>(java.io.File&nbsp;bamFile,
                   java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;chrToGenesMap,
                   java.lang.String&nbsp;genomeSeq)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Builds a transcriptome space from a bam file plus a known annotation file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/coordinatesystem/TranscriptomeSpace.html#TranscriptomeSpace(java.util.Map)">TranscriptomeSpace</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;chrToGenesMap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.debug"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/debug/package-summary.html">nextgen.core.debug</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../nextgen/core/debug/package-summary.html">nextgen.core.debug</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/debug/PairedEndSam.html#PairedEndSam(java.util.Map, broad.pda.datastructures.Alignments)">PairedEndSam</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
             <A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.feature"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/feature/package-summary.html">nextgen.core.feature</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Subclasses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/feature/package-summary.html">nextgen.core.feature</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../nextgen/core/feature/package-summary.html">nextgen.core.feature</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/feature/GeneWindow.html#GeneWindow(nextgen.core.annotation.Gene)">GeneWindow</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.model"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/model/package-summary.html">nextgen.core.model</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/core/model/package-summary.html">nextgen.core.model</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B>AlignmentModel.</B><B><A HREF="../../../../nextgen/core/model/AlignmentModel.html#getCountsPerPosition(nextgen.core.annotation.Gene)">getCountsPerPosition</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This function returns an array list of counts for each position along the specified gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;java.lang.Double&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptomeSpaceAlignmentModel.</B><B><A HREF="../../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#getPositionCountList(nextgen.core.annotation.Gene)">getPositionCountList</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get list of position level counts in mature transcript</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;T extends <A HREF="../../../../nextgen/core/model/score/WindowScore.html" title="interface in nextgen.core.model.score">WindowScore</A>&gt; 
<BR>
<A HREF="../../../../nextgen/core/model/score/WindowScoreIterator.html" title="class in nextgen.core.model.score">WindowScoreIterator</A>&lt;T&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>TranscriptomeSpaceAlignmentModel.</B><B><A HREF="../../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#scan(nextgen.core.annotation.Gene, int, int, nextgen.core.model.score.WindowProcessor)">scan</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
     int&nbsp;windowSize,
     int&nbsp;overlap,
     <A HREF="../../../../nextgen/core/model/score/WindowProcessor.html" title="interface in nextgen.core.model.score">WindowProcessor</A>&lt;T&gt;&nbsp;processor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scan windows over a gene and score
 Only scan exons of this gene</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.normalize"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/normalize/package-summary.html">nextgen.core.normalize</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../nextgen/core/normalize/package-summary.html">nextgen.core.normalize</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/normalize/ComparativeCountsMatrix.html#ComparativeCountsMatrix(java.lang.String, java.util.ArrayList, java.lang.String, java.util.Map)">ComparativeCountsMatrix</A></B>(java.lang.String&nbsp;controlBamFile,
                        java.util.ArrayList&lt;java.lang.String&gt;&nbsp;sampleBamFiles,
                        java.lang.String&nbsp;chrSizeFile,
                        java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/normalize/NormalizeCountsToTranscriptAverage.html#NormalizeCountsToTranscriptAverage(java.lang.String, java.util.Map, java.lang.String, java.lang.String)">NormalizeCountsToTranscriptAverage</A></B>(java.lang.String&nbsp;bamFile,
                                   java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                                   java.lang.String&nbsp;sizes,
                                   java.lang.String&nbsp;save)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.scripture"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A> that return <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>ChromosomeTranscriptGraph.</B><B><A HREF="../../../../nextgen/core/scripture/ChromosomeTranscriptGraph.html#pathToGene(org.jgrapht.GraphPath)">pathToGene</A></B>(org.jgrapht.GraphPath&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>&gt;&nbsp;path)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B>OrientedChromosomeTranscriptGraph.</B><B><A HREF="../../../../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#pathToGene(org.jgrapht.GraphPath)">pathToGene</A></B>(org.jgrapht.GraphPath&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>&gt;&nbsp;gp)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A> that return types with arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AddEndRNASeqToScripture.IsoformMap.</B><B><A HREF="../../../../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getAllGenes()">getAllGenes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AddEndRNASeqToScripture.IsoformMap.</B><B><A HREF="../../../../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getGenesForChromosome(java.lang.String)">getGenesForChromosome</A></B>(java.lang.String&nbsp;chr)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getIsoformMap(java.util.Collection)">getIsoformMap</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getIsoformMap(java.util.Collection)">getIsoformMap</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AddEndRNASeqToScripture.IsoformMap.</B><B><A HREF="../../../../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getIsoformsForGene(nextgen.core.annotation.Gene)">getIsoformsForGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeTranscriptGraph.</B><B><A HREF="../../../../nextgen/core/scripture/ChromosomeTranscriptGraph.html#getOrphanGenes()">getOrphanGenes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getOverlappingGenes(nextgen.core.annotation.Gene, java.util.Collection, double)">getOverlappingGenes</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                    java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;allGenes,
                    double&nbsp;minPctOverlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getPaths()">getPaths</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AddEndRNASeqToScripture.IsoformMap.</B><B><A HREF="../../../../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getIsoformsForGene(nextgen.core.annotation.Gene)">getIsoformsForGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>AddEndRNASeqToScripture.IsoformMap.</B><B><A HREF="../../../../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getNumOfIsoformsForGene(nextgen.core.annotation.Gene)">getNumOfIsoformsForGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getOverlappingGenes(nextgen.core.annotation.Gene, java.util.Collection, double)">getOverlappingGenes</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                    java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;allGenes,
                    double&nbsp;minPctOverlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#isoformMapContains(java.util.Map, nextgen.core.annotation.Gene)">isoformMapContains</A></B>(java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;isoformMap,
                   <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AddEndRNASeqToScripture.IsoformMap.</B><B><A HREF="../../../../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#addChrMap(java.lang.String, java.util.Map)">addChrMap</A></B>(java.lang.String&nbsp;chr,
          java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;map)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ADDS MAPS FOR A CHROMOSOME
 If maps already exist, they will be replaced</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AddEndRNASeqToScripture.IsoformMap.</B><B><A HREF="../../../../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#addChrMap(java.lang.String, java.util.Map)">addChrMap</A></B>(java.lang.String&nbsp;chr,
          java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;map)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ADDS MAPS FOR A CHROMOSOME
 If maps already exist, they will be replaced</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture.IsoformMap</A></CODE></FONT></TD>
<TD><CODE><B>AddEndRNASeqToScripture.</B><B><A HREF="../../../../nextgen/core/scripture/AddEndRNASeqToScripture.html#buildIsoformMap(java.util.Map)">buildIsoformMap</A></B>(java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;ann)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getIsoformMap(java.util.Collection)">getIsoformMap</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getOverlappingGenes(nextgen.core.annotation.Gene, java.util.Collection, double)">getOverlappingGenes</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                    java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;allGenes,
                    double&nbsp;minPctOverlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#isoformMapContains(java.util.Map, nextgen.core.annotation.Gene)">isoformMapContains</A></B>(java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;isoformMap,
                   <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#isoformMapContains(java.util.Map, nextgen.core.annotation.Gene)">isoformMapContains</A></B>(java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.Set&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;isoformMap,
                   <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BuildScriptureCoordinateSpace.</B><B><A HREF="../../../../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#write(java.lang.String, java.util.Map)">write</A></B>(java.lang.String&nbsp;save,
      java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;rtrn)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.structure"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/structure/package-summary.html">nextgen.core.structure</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../nextgen/core/structure/package-summary.html">nextgen.core.structure</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/structure/PairwiseWindowHybridization.html#PairwiseWindowHybridization(nextgen.core.annotation.Gene, java.util.Map, int, int)">PairwiseWindowHybridization</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;transcript,
                            java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&gt;&nbsp;chrs,
                            int&nbsp;windowSize,
                            int&nbsp;stepSize)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/structure/PairwiseWindowHybridization.html#PairwiseWindowHybridization(nextgen.core.annotation.Gene, java.util.Map, int, int, float, float, float, float)">PairwiseWindowHybridization</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;transcript,
                            java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>&gt;&nbsp;chrs,
                            int&nbsp;windowSize,
                            int&nbsp;stepSize,
                            float&nbsp;matchScore,
                            float&nbsp;mismatchScore,
                            float&nbsp;gapOpenPenalty,
                            float&nbsp;gapExtendPenalty)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/structure/PairwiseWindowHybridization.html#PairwiseWindowHybridization(nextgen.core.annotation.Gene, java.lang.String, int, int)">PairwiseWindowHybridization</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;transcript,
                            java.lang.String&nbsp;genomeFasta,
                            int&nbsp;windowSize,
                            int&nbsp;stepSize)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/structure/PairwiseWindowHybridization.html#PairwiseWindowHybridization(nextgen.core.annotation.Gene, java.lang.String, int, int, float, float, float, float)">PairwiseWindowHybridization</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;transcript,
                            java.lang.String&nbsp;genomeFasta,
                            int&nbsp;windowSize,
                            int&nbsp;stepSize,
                            float&nbsp;matchScore,
                            float&nbsp;mismatchScore,
                            float&nbsp;gapOpenPenalty,
                            float&nbsp;gapExtendPenalty)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.core.writers"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/core/writers/package-summary.html">nextgen.core.writers</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../nextgen/core/writers/package-summary.html">nextgen.core.writers</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/writers/WigWriter.html#WigWriter(java.lang.String, java.util.Map, java.lang.String, boolean, boolean)">WigWriter</A></B>(java.lang.String&nbsp;bamFile,
          java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genesByChrName,
          java.lang.String&nbsp;singlePositionOnly,
          boolean&nbsp;useFragments,
          boolean&nbsp;nor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct with a transcriptome space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/core/writers/WigWriter.html#WigWriter(java.lang.String, java.util.Map, java.lang.String, boolean, boolean, java.lang.String)">WigWriter</A></B>(java.lang.String&nbsp;bamFile,
          java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genesByChrName,
          java.lang.String&nbsp;singlePositionOnly,
          boolean&nbsp;useFragments,
          boolean&nbsp;nor,
          java.lang.String&nbsp;chrToWrite)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct with a transcriptome space and only write one chromosome</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.editing"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/editing/package-summary.html">nextgen.editing</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../nextgen/editing/package-summary.html">nextgen.editing</A> with parameters of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>CRISPRDesigner.CRISPRTarget.</B><B><A HREF="../../../../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#setGene(nextgen.core.annotation.Gene)">setGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="nextgen.rnaprotein.analysis"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> in <A HREF="../../../../nextgen/rnaprotein/analysis/package-summary.html">nextgen.rnaprotein.analysis</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../nextgen/rnaprotein/analysis/package-summary.html">nextgen.rnaprotein.analysis</A> with type arguments of type <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../nextgen/rnaprotein/analysis/ComputeGeneEnrichment.html#ComputeGeneEnrichment(java.lang.String, java.lang.String, java.util.Map, java.lang.String)">ComputeGeneEnrichment</A></B>(java.lang.String&nbsp;bam1,
                      java.lang.String&nbsp;bam2,
                      java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genesByChr,
                      java.lang.String&nbsp;save)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?nextgen/core/annotation//class-useGene.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Gene.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
