<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:28 EDT 2013 -->
<TITLE>
Constant Field Values
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Constant Field Values";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="index.html?constant-values.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="constant-values.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H1>
Constant Field Values</H1>
</CENTER>
<HR SIZE="4" NOSHADE>
<B>Contents</B><UL>
<LI><A HREF="#broad.core">broad.core.*</A>
<LI><A HREF="#broad.pda">broad.pda.*</A>
<LI><A HREF="#jaligner">jaligner.*</A>
<LI><A HREF="#nextgen.core">nextgen.core.*</A>
</UL>

<A NAME="broad.core"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left"><FONT SIZE="+2">
broad.core.*</FONT></TH>
</TR>
</TABLE>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.annotation.<A HREF="broad/core/annotation/AnnotationUtils.html" title="class in broad.core.annotation">AnnotationUtils</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.annotation.AnnotationUtils.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/annotation/AnnotationUtils.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Usage: AnnotationUtils TASK=&lt;task_num&gt; &lt;task_args&gt;\n\tTasks:\n\t\t1. Take the intersection two annotation sets: -set1 &lt;File containing first set&gt;  -set2 &lt;File containing second set&gt; -minScore &lt;to filter by minimum annotation score&gt;\n\t\t\t [-set1format &lt;[BED], GFF or generic&gt; -set2format &lt;[BED], GFF or generic&gt; -outformat &lt;desired output format BED, GFF or generic, default is se1format&gt;\n\t2. Take the difference of two annotation sets (all annotations in set1 that do not overlap set2): -set1 &lt;File containing first set&gt;  -set2 &lt;File containing second set&gt;  -minScore &lt;to filter by minimum annotation score&gt;\n\t\t [-set1format &lt;[BED], GFF or generic&gt; -set2format &lt;[BED], GFF or generic&gt; -outformat &lt;desired output format BED, GFF or generic, default is se1format&gt;\n\t3. Print out all elements in one set (set1) that overlap another (set2): -set1 &lt;File containing first set&gt;  -set2 &lt;File containing second set&gt; \n\t\t [-set1format &lt;[BED], GFF or generic&gt; -set2format &lt;[BED], GFF or generic&gt; -outformat &lt;desired output format BED, GFF or generic, default is se1format&gt;\n\t4. Merge annotation file -in &lt;annotation file (dowa not support standard input input at this point)&gt; -out &lt;output file name or standard output by default&gt; [-format &lt;[BED], GFF or generic&gt;]\n\t5. Reformat annotation file  -in &lt;annotation file (dowa not support standard input input at this point)&gt; -out &lt;output file name or standard output by default&gt; [-informat &lt;[BED], GFF or generic&gt; -outformat &lt;BED,[GFF],generic]\n\t6. Shuffle scores -in &lt;Annotation file&gt; -out &lt;output file name or standard output by default&gt; [-informat &lt;[BED], GFF or generic&gt;]\n\t7. Get overlap statistics of two sets (how much each of the elements in set1 is covered by set2). \n\t\t [-set1format &lt;[BED], GFF or generic&gt; -set2format &lt;[BED], GFF or generic&gt; -outformat &lt;desired output format BED, GFF or generic, default is se1format&gt;\n\t8. Take the complement of the the first annotation set in the second set (similar to 2 but elements incompletely overlapping will be broken into the regions that don\'t overlap): -set1 &lt;File containing first set&gt;  -set2 &lt;File containing second set&gt; \n\t9. Filter any delimited file by extracting all records that overlap a given annotation file -in &lt;Delimted file&gt; -chrColumn &lt;Column containing chrmosome information (first column is 0)&gt; [-chr &lt;Alternative if there is no chromosome information in the file, it is assumed that all positions are in a given chromosome&gt;] -positionColumn &lt;Column containing position information  (first column is 0)&gt; -annotations &lt;Annotation file&gt; -annotationFormat &lt;[BED], GFF, Generic&gt; [-sepeartor &lt;Character separting columns, tab is assumed by default&gt;]\n\t10. Print out all elements in one set (set1) that overlap another (set2) and are in the same orientation as the overlapping gene: -set1 &lt;File containing first set&gt;  -set2 &lt;File containing second set&gt; \n\t\t [-set1format &lt;[BED], GFF or generic&gt; -set2format &lt;[BED], GFF or generic&gt; -outformat &lt;desired output format BED, GFF or generic, default is se1format&gt;\n\t11. Compute the probability that an interval of given size overlaps a given annotation set by chance: -set &lt;File containing first set&gt;  -sizeFile &lt;Size file with chromosome sizes&gt; \n\t12. Print distance metrics: for each element in first set, print the distance to the closest element in the second set.\n\t\t [-nonOverlapping &lt;find the closest non-overlapping element&gt; -set1format &lt;[BED], GFF or generic&gt; -set2format &lt;[BED], GFF or generic&gt; -chr &lt;limit analysis to chr&gt;\n\tSlideAndCount\tSlide a window over the annotation map and count elements that overlap each window\n\t\t -window &lt;windowSize&gt; -overlap &lt;overlapSize&gt; -chr &lt;chr&gt; -start &lt;start&gt; -end &lt;end&gt;\n\tOverlapSignifiance. Calculate the significance of the overlap between two sets of annotations by permuting across the same chromosome.\n\t\t -set1 &lt;set1&gt; -set2 &lt;set2&gt; -nperm &lt;number of permutations&gt; -sizeFile &lt;file of chromosome sizes&gt;\n"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.datastructures.<A HREF="broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.HAS_GREATER_PART"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#HAS_GREATER_PART">HAS_GREATER_PART</A></CODE></TD>
<TD ALIGN="right"><CODE>4</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.HAS_LESSER_PART"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#HAS_LESSER_PART">HAS_LESSER_PART</A></CODE></TD>
<TD ALIGN="right"><CODE>1</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.HAS_OVERLAPPING_PART"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#HAS_OVERLAPPING_PART">HAS_OVERLAPPING_PART</A></CODE></TD>
<TD ALIGN="right"><CODE>2</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.IS_ADJACENT_AND_EMPTY"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#IS_ADJACENT_AND_EMPTY">IS_ADJACENT_AND_EMPTY</A></CODE></TD>
<TD ALIGN="right"><CODE>0</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.IS_LEFT_OVERHANGING_OVERLAPPER"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#IS_LEFT_OVERHANGING_OVERLAPPER">IS_LEFT_OVERHANGING_OVERLAPPER</A></CODE></TD>
<TD ALIGN="right"><CODE>3</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.IS_RIGHT_OVERHANGING_OVERLAPPER"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#IS_RIGHT_OVERHANGING_OVERLAPPER">IS_RIGHT_OVERHANGING_OVERLAPPER</A></CODE></TD>
<TD ALIGN="right"><CODE>6</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.IS_STRICTLY_GREATER"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#IS_STRICTLY_GREATER">IS_STRICTLY_GREATER</A></CODE></TD>
<TD ALIGN="right"><CODE>4</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.IS_STRICTLY_LESS"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#IS_STRICTLY_LESS">IS_STRICTLY_LESS</A></CODE></TD>
<TD ALIGN="right"><CODE>1</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.IS_SUBSET"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#IS_SUBSET">IS_SUBSET</A></CODE></TD>
<TD ALIGN="right"><CODE>2</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.datastructures.Interval.IS_SUPERSET"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/datastructures/Interval.html#IS_SUPERSET">IS_SUPERSET</A></CODE></TD>
<TD ALIGN="right"><CODE>7</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.math.<A HREF="broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.math.EmpiricalDistribution.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/math/EmpiricalDistribution.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Usage: EmpiricDistribution TASK=&lt;task_num&gt; &lt;task_args&gt;\n\tTasks:\n\t\t1. Get empirical distribution -col &lt;column to use 0 for the first one&gt; [-in &lt;Input file standard input is default&gt; -out &lt;output file, standard output by default&gt; -separator &lt;tab (\t) is default&gt;]\t\t\t -binNum &lt;Number of bins, default is 200&gt; -maxValue &lt;Maximum value to consider&gt; -minValue &lt;Minimum value to consider&gt;\n\t2. Combine distributions -indir &lt;directory&gt; -suffix &lt;suffix of distribution files to combine&gt; -out &lt;output file name defalut is standard out&gt;\n\t3. Get quantile -probability -in &lt;Input distribution standard input is default&gt;\n"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.math.<A HREF="broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.math.Statistics.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/math/Statistics.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Usage: Statistics TASK=&lt;task_num&gt; &lt;task_args&gt;\n\tTasks:\n\t\t1. Compute Benjamini-Hochberg FDR adjusted pvalue of a given pvalue column in a file \n\t\t\t-in &lt;file (standard input is default)&gt; -fdr &lt;Maximum false discovery rate&gt; -col &lt;Data column, 0 based&gt; [-separator &lt;character used to separate columns, default is tab \'\\t&gt;\'] \n"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.motif.<A HREF="broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.motif.PositionWeightColumn.DEFAULT_ALPHABET_SIZE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/motif/PositionWeightColumn.html#DEFAULT_ALPHABET_SIZE">DEFAULT_ALPHABET_SIZE</A></CODE></TD>
<TD ALIGN="right"><CODE>4</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.motif.<A HREF="broad/core/motif/PositionWeightMatrixIO.html" title="class in broad.core.motif">PositionWeightMatrixIO</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.motif.PositionWeightMatrixIO.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/motif/PositionWeightMatrixIO.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Usage: PositionWeightMatrixIO TASK=&lt;task_num&gt; &lt;task_args&gt; -pwm &lt;PWM file in FASTA fromat&gt;\n\tTasks:\n\t\t1. Report matches of a PWMs to a sequence -in &lt;Sequence file or standard input in FASTA format&gt; -pwm &lt;File containing PWMs&gt; -minScore &lt;Minimum Score to report&gt;\n\t\t\t-start &lt;start matching at this point on the corresponding sequence include as many starts as sequenceIds&gt; -end &lt;Stop matching at this point on the corresponding sequence include as many ends as sequenceIds&gt; \n\t\t\t[-seqId &lt;Sequence Ids in the fasta file to search on specify as many as targets you want to search on, if non is specified all start/ends will be assume to relate to first sequence in file&gt;] \n\t\tic Compute information content of all motifs on the file -pwm &lt;File containing PWMs&gt;\n"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.motif.<A HREF="broad/core/motif/PWMUtils.html" title="class in broad.core.motif">PWMUtils</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.motif.PWMUtils.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/motif/PWMUtils.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Usage: ArrayDesignUtilities TASK=&lt;task_num&gt; &lt;task_args&gt;\n\tTasks:\n\t\t1. Compute log likelohood score thresholds for a given PWM set. This task will compute the distribution of scores then output thresholds for commonly used quantiles. -pwm &lt;File containing one or more PWM matrices&gt; -pA &lt;new equilibrium probability for A&gt;  -pC &lt;new equilibrium probability for C&gt;  -pG &lt;new equilibrium probability for G&gt;  -pT &lt;new equilibrium probability for T&gt; -out &lt;file name for score quantile cutoff per pwm file or standard out if none is specified&gt; -outdir &lt;output directory or current directory if this parameter is not specified&gt; [-prefix &lt;A prefix for each pwm distribution file&gt;]&gt; \n\t\tscan. Scan a set a set of genomic regions for PWM ocurrencens -pwm &lt;File containing one or more PWM matrices&gt; -regions &lt;Genomic Annotations&gt; -seqdir &lt;Sequence directory&gt; -maxOnly &lt;If only the maximal hit per region should be reported&gt; -outdir &lt;Output directory to write output file&gt; [-minScoreToWrite &lt;Filter output by minimum score, reudces output file size&gt; -prefix &lt;Prefix to use for each PWM hit file&gt; -shuffles &lt;Number of pwm shuffles to asses significance&gt;]\n\t\tpermute. Permute a PWM matrix -pwm &lt;File contining one or more PWM matrics&gt; -out &lt;Output file with permuted matrices or standard out in not specified&gt;\n\t\treverse. Reverse-compliment a PWM -pwm &lt;File contining one or more PWM matrics&gt; -out &lt;Output file with permuted matrices or standard out in not specified&gt;\n\t\tscanGenome. Scan the genome for PWM ocurrencens -pwm &lt;File containing one or more PWM matrices&gt; -seqdir &lt;Sequence directory&gt; -outdir &lt;Output directory to write output file&gt; -minScoreToWrite &lt;Filter output by minimum score, reudces output file size&gt; [-prefix &lt;Prefix to use for each PWM hit file&gt; -shuffles &lt;Number of pwm shuffles to asses significance&gt;]\n"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.multiplealignment.<A HREF="broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.multiplealignment.MultipleAlignment.A_CODE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/multiplealignment/MultipleAlignment.html#A_CODE">A_CODE</A></CODE></TD>
<TD ALIGN="right"><CODE>0</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.multiplealignment.MultipleAlignment.C_CODE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/multiplealignment/MultipleAlignment.html#C_CODE">C_CODE</A></CODE></TD>
<TD ALIGN="right"><CODE>1</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.multiplealignment.MultipleAlignment.G_CODE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/multiplealignment/MultipleAlignment.html#G_CODE">G_CODE</A></CODE></TD>
<TD ALIGN="right"><CODE>2</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.multiplealignment.MultipleAlignment.GAP_CODE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/multiplealignment/MultipleAlignment.html#GAP_CODE">GAP_CODE</A></CODE></TD>
<TD ALIGN="right"><CODE>4</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.multiplealignment.MultipleAlignment.T_CODE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/multiplealignment/MultipleAlignment.html#T_CODE">T_CODE</A></CODE></TD>
<TD ALIGN="right"><CODE>3</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.primer3.<A HREF="broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.primer3.Primer3ConfigurationFactory.HUMAN_LONG_REP"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/primer3/Primer3ConfigurationFactory.html#HUMAN_LONG_REP">HUMAN_LONG_REP</A></CODE></TD>
<TD ALIGN="right"><CODE>"/seq/mgarber/tools/RepeatMasker/Libraries/20060314/homo_sapiens/longlib"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.primer3.<A HREF="broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.primer3.qPCRPrimerDesigner.MIN_PRIMER_DES_SPACE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/primer3/qPCRPrimerDesigner.html#MIN_PRIMER_DES_SPACE">MIN_PRIMER_DES_SPACE</A></CODE></TD>
<TD ALIGN="right"><CODE>150</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.primer3.qPCRPrimerDesigner.MIN_PROD_SIZE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/primer3/qPCRPrimerDesigner.html#MIN_PROD_SIZE">MIN_PROD_SIZE</A></CODE></TD>
<TD ALIGN="right"><CODE>100</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.primer3.qPCRPrimerDesigner.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/primer3/qPCRPrimerDesigner.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Usage: PrimerDesigner TASK=&lt;task_num&gt; &lt;task_args&gt;\n\tTasks:\n\t\t1. prime gaps: OUTDIR=output directory IN=&lt;sequence AGP&gt; file MAXPRODSIZE=maximum product size BUFFER=size of buffer between primers and target\n\t\t2. Generate short amplicon set. This is useful to say desing qPCR exeperiments \n\t\t\tIN=&lt;sequence AGP&gt;\n\t\t\tOUTDIR=&lt;output directory&gt;\n\t\t\tDIST=&lt;Distance between amplicons&gt; OR NUM&lt;number of amplicons&gt;\n\t\t\tAMPSIZE=&lt;Desired amplicon size&gt;\n\t\t\tTARGETSTART=&lt;If supplied, the start of region to cover&gt;\n\t\t\tTARGETEND=&lt;If supplied, the end of region to cover&gt;\n\t\t\tREPEATOUT=&lt;RepeatMasker output file for the sequence&gt;\n\t\t3. Generate gene set amplification set -in &lt;gene list, a gene symbol per line&gt; -minsize &lt;minimum amplicon size&gt; -cdsonly &lt;include this flag if only interested in cds only amplification&gt;\n\n\t\t-maxsize &lt;max amplicon size&gt; [-numOfDesigns &lt;number of different designs to ouput (default 1)&gt; -buffer &lt;intronic bases to include (default 0)&gt;] \n\t\t4. Generate flanking primers -in &lt;Multifasta file default is standard in&gt; -outdir &lt;directory where to write output files&gt; -optimalDistFromEnds &lt;Optimal distance from sequence begining and end\n\t\t\t -maxDistFromEnds &lt;Maximum distance from ends, used when no primer could be found within optimal distance&gt; [-outprefix &lt;a prefix to prepend to output, if not specified no prefix will be prepended to file names&gt;]"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.sequence.<A HREF="broad/core/sequence/AbstractFastaParser.html" title="class in broad.core.sequence">AbstractFastaParser</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.AbstractFastaParser.BUFFER_SIZE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/AbstractFastaParser.html#BUFFER_SIZE">BUFFER_SIZE</A></CODE></TD>
<TD ALIGN="right"><CODE>8192</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.sequence.<A HREF="broad/core/sequence/FastaQualityBinaryIO.html" title="class in broad.core.sequence">FastaQualityBinaryIO</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.FastaQualityBinaryIO.BUFFER_SIZE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/FastaQualityBinaryIO.html#BUFFER_SIZE">BUFFER_SIZE</A></CODE></TD>
<TD ALIGN="right"><CODE>8192</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.FastaQualityBinaryIO.SEQ_ID_REC_LENGTH"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/FastaQualityBinaryIO.html#SEQ_ID_REC_LENGTH">SEQ_ID_REC_LENGTH</A></CODE></TD>
<TD ALIGN="right"><CODE>300</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.sequence.<A HREF="broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.FastaSequenceIO.LINE_LENGTH"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/FastaSequenceIO.html#LINE_LENGTH">LINE_LENGTH</A></CODE></TD>
<TD ALIGN="right"><CODE>60</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.sequence.<A HREF="broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_a"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_a">SHORT_ENCODED_a</A></CODE></TD>
<TD ALIGN="right"><CODE>5</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_A"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_A">SHORT_ENCODED_A</A></CODE></TD>
<TD ALIGN="right"><CODE>0</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_c"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_c">SHORT_ENCODED_c</A></CODE></TD>
<TD ALIGN="right"><CODE>6</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_C"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_C">SHORT_ENCODED_C</A></CODE></TD>
<TD ALIGN="right"><CODE>1</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_g"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_g">SHORT_ENCODED_g</A></CODE></TD>
<TD ALIGN="right"><CODE>7</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_G"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_G">SHORT_ENCODED_G</A></CODE></TD>
<TD ALIGN="right"><CODE>2</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_GAP"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_GAP">SHORT_ENCODED_GAP</A></CODE></TD>
<TD ALIGN="right"><CODE>4</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_N"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_N">SHORT_ENCODED_N</A></CODE></TD>
<TD ALIGN="right"><CODE>9</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_t"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_t">SHORT_ENCODED_t</A></CODE></TD>
<TD ALIGN="right"><CODE>8</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.Sequence.SHORT_ENCODED_T"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;short</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/Sequence.html#SHORT_ENCODED_T">SHORT_ENCODED_T</A></CODE></TD>
<TD ALIGN="right"><CODE>3</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.sequence.<A HREF="broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.sequence.SequenceRegion.INF"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/sequence/SequenceRegion.html#INF">INF</A></CODE></TD>
<TD ALIGN="right"><CODE>1000000000</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.siphy.<A HREF="broad/core/siphy/PiHMM.html" title="class in broad.core.siphy">PiHMM</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.siphy.PiHMM.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/siphy/PiHMM.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Usage: PiHMM TASK=&lt;task_num&gt; &lt;task_args&gt;\n\tTasks:\n\t\t1. Segment genome into pi-conserved and noncoserved.  -in &lt;multiple alignment file&gt;  -mod &lt;Neutral Evolutionary model consisting of aminoacid background distribution, mutation matrix and neutral phylogenetic tree&gt;\n\t\t -l &lt;The smoothness parameter, and should be the expected average conserved element length&gt; and -gamma &lt;The expected coverage parameter, how much of the genome is expected to be conserved&gt;\n\t\t -format &lt;Alignment format default is FASTA is default&gt; -ignore &lt;comma separated species to ignore&gt; -ref &lt;reference sequence id, necessary if the alignment is not in MAF format&gt;\n\t\t  [-start &lt;If MAF file, you may specify the reference start coordinate&gt; -end &lt;If MAF file, you may specify the reference end coordinate&gt;]\n\t\t2. Compute posterior probabilities of each site of being pi-conserved.  -in &lt;multiple alignment file&gt;  -mod &lt;Neutral Evolutionary model consisting of aminoacid background distribution, mutation matrix and neutral phylogenetic tree&gt;\n\t\t -l &lt;The smoothness parameter, and should be the expected minimum conserved element length&gt; and -gamma &lt;The expected coverage parameter, how much of the genome is expected to be conserved&gt;\n\t\t -format &lt;Alignment format default is FASTA is default&gt; -ignore &lt;comma separated species to ignore&gt; -ref &lt;reference sequence id, necessary if the alignment is not in MAF format&gt;\n\t\t  [-start &lt;If MAF file, you may specify the reference start coordinate&gt; -end &lt;If MAF file, you may specify the reference end coordinate&gt;]\n\t\t3. Compute log odds score for annotations in file. Scores reflect the log ratio of the probability of the path through each element being fully conserved or non conserved.  -in &lt;Annotation file default format is assumed to be BED&gt;  -mod &lt;Neutral Evolutionary model consisting of aminoacid background distribution, mutation matrix and neutral phylogenetic tree&gt;\n\t\t -l &lt;The smoothness parameter, and should be the expected minimum conserved element length&gt; and -gamma &lt;The expected coverage parameter, how much of the genome is expected to be conserved&gt;\n\t\t -format &lt;Alignment format default is FASTA is default&gt; -ignore &lt;comma separated species to ignore&gt; -ref &lt;reference sequence id, necessary if the alignment is not in MAF format&gt;\n\t\t  [-start &lt;If MAF file, you may specify the reference start coordinate&gt; -end &lt;If MAF file, you may specify the reference end coordinate&gt;]\n\t\t4. Simulate an alignment with for given HMM parameters. \n\t\tParameters:\n\t\t-mod &lt;Base m,odel to use&gt; \n\t\t-ignore &lt;optional -- comma separated species to ignore in the given model tree&gt; \n\t\t-colNum &lt;Number of columns to sample&gt; -\n\t\t-otherPI &lt;Conserved or alternative PI distribution, as a comma separated list of the A,C,G,T frequencies, add as many -otherPI as different PI states are desired. &gt; \n\t\t -l &lt;The smoothness parameter, how long in average should be the stretches of the generated alignment should be from the alternative PI model&gt; and -gamma &lt;The coverage parameter, how much of the sampled alignmnet should be made of the alternative PI model&gt; \n\t\t-out &lt;Generated alignment output file&gt;\n"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.core.siphy.<A HREF="broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.core.siphy.TreeScaler.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/core/siphy/TreeScaler.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Usage: TreeScaler TASK=&lt;task_num&gt; &lt;task_args&gt;\n\tTasks:\n\t\t1. Compute scaling of tree for each site in a multiple alignment. \n\t\tParameters:\n\t\t  -in &lt;multiple alignment file&gt; \n\t\t  -mod &lt;Neutral Evolutionary model consisting of aminoacid background distribution, mutation matrix and neutral phylogenetic tree&gt;\n\t\t  -format &lt;Alignment format default is FASTA is default&gt;\n\t\t  -outdir &lt;output directory&gt; or -out &lt;if a specific file name is desired&gt;\n\n\t  -ref &lt;reference sequence id, necessary if the alignment is not in MAF format&gt;\n\t\t  -ignore &lt;comma separated species to ignore&gt;\n\t\t  [-minTreeLength &lt;minimum tree length to actually compute omega, sites with lesser total tree length are ignored. Default is 1&gt;\n\t\t  -window &lt;Size of sliding window where tree is locally scaled, default is 1 &gt; -windowOverlap &lt;sliding window overlap, default is window length - 1&gt;\n\t\t  -neutralOmegaDist &lt;A neutral sequence generated omega distribution by tree length,\n\t\t\t you can use task 3 to generate this file if provided it will be use to compute p-values,\n\t\t\t the distribution should be generated with same window intended to run in this task&gt;]\n\t\t\t alternative, if you are sampling (see -withSampling option) you can specify: \n\t\t [-neutralOmegas &lt;a previously generated omega estimation run in neutral sequence with respect to the same model using full tree only&gt;]\n\t\t  sample evolutionary model file:\n\t\t  ALPHABET: C,T,G,A  (if the data in the model is not using the default ordering of aminoacids)\n\t\t  BACKGROUND: 0.214713 0.307324 0.248610 0.229353\n\t\t  RATE_MAT:\n\t\t  -1.113121    0.267171    0.611653    0.234297\n\t\t  0.186659   -0.888845    0.199678    0.502508 \n\t\t  0.528254    0.246835   -0.956902    0.181813 \n\t\t  0.219341    0.673340    0.197079   -1.089760 \n\t\t  TREE: ((((mm8:0.085233,rn4:0.098462):0.262242,hg18:0.128359):0.025266,canFam2:0.171487):0.308235,monDom4:0.308235);\n\t\t  [-withSampling &lt;Sample missing data from neutral model&gt; -numSamplings &lt;Number of times to run estimation to use in averaging omega. More than 10 iterations are redundant&gt;\n\t\t  [-start &lt;If MAF file, you may specify the reference start coordinate&gt; -end &lt;If MAF file, you may specify the reference end coordinate&gt;]\n\t\t3. Estimate omega distribution in neutral sequence -in &lt;Alignment file&gt; \n\t\t  -format &lt;default is FASTA, MAF is also supported&gt; \n\t\t  -mod &lt;Neutral Evolutionary model consisting of aminoacid background distribution, mutation matrix and neutral phylogenetic tree&gt;\n\t\t  -window &lt;Window size for omega estimation&gt;\n\t\t  -ref &lt;reference sequence&gt; \n\n\t\t  -ignore &lt;comma separated regions to ignore&gt;\n\t\t4. Write model -kappa &lt;HKY kappa parameter&gt; - background &lt;comma separated background nucleotide frequencies (A,C,G,T)&gt; [-mu &lt;Scaling factor default is 1&gt;]\n\t\t5. Generate sampled alignment -mod &lt;Neutral Evolutionary model consisting of nucleotide background distribution, mutation matrix and neutral phylogenetic tree&gt;\n\t\t  -format &lt;Alignment format default is FASTA &gt;\n\t\t  -out &lt;output file&gt;\n\t\t  -colNum &lt;Number of columns to generate&gt;\n\t\t6. Fill alignment gaps and missing data with neutral model generated probabilities.\n\t\t -in &lt;Alignment file&gt;\n\t\t -out &lt;Output file name&gt; \n\t\t -mod &lt;Neutral model as defined in task 1&gt;\n\t\t [-informat &lt;input alignment format, default is FASTA&gt; -ignore &lt;comma separated species to ignore&gt; -outformat &lt;output format default is FASTA]\n\t\t7. Fit base frequency pi. Parameters are the same as task 1 except for:\n\t\t one off -priorOmega &lt;a prior omega to use when fitting rhow&gt; \n\t\t\t -siteOmegas &lt;An output of the task 1 or 2 for the same region omega will then be avaraged in windows&gt; -omegaWindow &lt;windo in which to fit omega, default is 1kb&gt; \n\t\t8. Integrate omegas over windows, this assumes that the omegas provided have been computed on the same tree length (e.g. via sampling) \n\t\t\t -in &lt;A site by site omega calculation output file&gt; -out &lt;output file&gt; -shift &lt;A coordinate shift in case the original omega file is not in genomic coordinates&gt; \n\t\t\t -chr &lt;chromosome the data belongs to&gt; -window &lt;window size to integrate over&gt; -neturalOmegas &lt;A site by site omega calculation output on neutral sequence&gt;\n\t\t9. Define conserved elements based on a desired score, this works for any format provided you specify the column containing the score\n\t\t\t -in &lt;estimation file (any file generated from running tasks 1,2,7,8 or similar), default is standard input&gt; -out &lt;output file name defalut is standard out&gt; [-separator &lt;Column separator character, default is tab&gt;]\n\t\t\t -pvalcol &lt;column of site p-value (first column is 1)&gt; -positioncol &lt;column of position, position is assumed to be the start of the window(first column is 1)&gt; -minscore &lt;Minimum score used to join a site&gt; -maxscore &lt;Maximum score to join&gt; -maxgap &lt;maximum gap of sites with less than pval to paste through&gt;\n\t\t\t [-chr &lt;chromosome default is \'C\'&gt; -window &lt;window size default is 1&gt; \n\t\t10. Score regions -alignment &lt;alignment including regions, regions should be in coordinates that are consistent with the alignment&gt; -mod &lt;model file&gt; -ref &lt;reference sequence id default is the first sequence in the alignment&gt; [-window &lt;If you want to tile each region with a fixed window rather than fitting omega to the full region&gt; -overlap &lt;By default a scan of windows with window size - 1 overlap is done&gt; -minTreeLength &lt;If no omega should be computed if the minimum branch length of the kmer is below this threshold&gt;]\n\t\t\t -in &lt;Input file BED annotation file, default is standard input&gt; -out &lt;Output file, default is standard output&gt; [-shift &lt;amount to shift position&gt; -bedIsOneBased &lt;Add this flad is the positions in BED file start at 1 rather than 0 ]\n\t\t11. Integrate Stationary distribution in windows -window &lt;window size&gt; [-windowOverlap &lt;sliding window overlap, default is window length - 1&gt; -dist &lt;Neutral distribution in empirical format see??? if pvalues are desired&gt;]\n\t\t\t -in &lt;estimation file (any file generated from running tasks 1,2,7,8 or similar), default is standard input&gt; -out &lt;output file name defalut is standard out&gt;\n\t\t12. Add an empirical pvalue to computation based on the spcified column value -col &lt;column number with values to use, first column is 0&gt; -dist &lt;Neutral distribution of same values to use in empirical pValue computations&gt; \n\t\t\t-in &lt;File (standard input is default)&gt; -out &lt;new file with the extra column of &gt; [-separator &lt;character used to separate columns, default is tab \'\\t&gt;\' -rightTail &lt;add this flag if the pvalue should be 1- p(X&gt;=x) rather than p(X&lt;= x)&gt;]\n\t\t13. Integrate Omega log odds score distribution in windows -window &lt;window size&gt; [-dist &lt;Neutral distribution in empirical format see??? if pvalues are desired&gt;]\n\t\t14. Given a PWM, an alignment and a neutral model slide PWM and compute log odds likelihood (of the window being generated by the neutral or PWM models \n\t\t\t-in &lt;Alignment file (or standard input) if MAF this is required&gt; -pwm &lt;File with PWM description&gt; -mod &lt;Neutral model&gt; [-seedMinScore &lt;Minimum affinity score in order to incurr in the expense of the phylogenetic computation&gt; -outprefix &lt;If a prefix to the automatically generated output file is desired&gt;]\n\t\t\t[-ref &lt;if Alignment not in MAF format&gt; -alignmentFormat &lt;one of [FASTA],PHYLIP. Fasta is default&gt; -start &lt;If alignment is in MAF format&gt; -end &lt;If alignment is in MAF format&gt; -outdir &lt;Output directory if other than current dir&gt; ]\n\t\t15. Compute sample scores for a set of PWMs in &lt;Alignment file (or standard input) if MAF this is required NOTE: Only MAF supported at the moment&gt; -pwm &lt;File with PWM description&gt; -mod &lt;Neutral model&gt; -numOfSamples &lt;Number of samples in which to compute scores&gt; \n\t\t\t[-ref &lt;if Alignment not in MAF format&gt; -alignmentFormat &lt;one of [FASTA],PHYLIP. Fasta is default&gt; -start &lt;If alignment is in MAF format&gt; -end &lt;If alignment is in MAF format&gt; -outdir &lt;Output directory if other than current dir&gt; ]\n\t\t16. Given a PWM, an MAF alignment and a neutral model slide PWM and compute log odds likelihood (of the window being generated by the neutral or PWM models \n\t\t\t-indir &lt;Alignment directory of chromosome  MAF alignments&gt; -pwm &lt;File with PWM description&gt; -mod &lt;Neutral model&gt; [-seedMinScore &lt;Minimum affinity score in order to incurr in the expense of the phylogenetic computation&gt; -ignore &lt;comma separated species to ignore&gt;]\n\t\t\t-start &lt;start of region &gt; -end &lt;end of region &gt; -chr &lt;chromosome region&gt; -regions &lt;alternatively you can specify an annotation file &gt; -regionFormat &lt;[BED], GFF or generic&gt; \n\t\t\t[-mafSuffix &lt;A suffix for maf alignment files default is .maf&gt; -outdir &lt;Output directory if other than current dir&gt; -outprefix &lt;If a prefix to the automatically generated output file is desired&gt; -minScoreToReport &lt;Do not report scores less than this&gt; -shuffles &lt;Number of shuffles to do if suffling then a pvalue is reported&gt;&gt;]\n\t\t17. Change equilibrium distribution in model. -mod &lt;Model file to change (see task 1 for a description&gt; -pA &lt;new equilibrium probability for A&gt;  -pC &lt;new equilibrium probability for C&gt;  -pG &lt;new equilibrium probability for G&gt;  -pT &lt;new equilibrium probability for T&gt; -out &lt;Output file or standard out if none is specified&gt;\n\t\t18. Evolve sequence according to tree -ancestralSequence &lt;A nucleotide sequence to evolve&gt; -mod &lt;Model to use&gt; -ignore &lt;optional -- comma separated species to ignore in the given model tree&gt; -numColumns &lt;Number of columns to sample&gt; -bg &lt;Optional -- new background distribution, as a comma separated list of the A,C,G,T frequencies&gt; -\n\t\tbayesian. Estimate posterior P(omega | Data). Basic data (Alignment and model should be specified per in task 1) specific parameters: \n\t\t\t-out &lt;Name of output file containig P(0.25 | data) and P(1 | data) for each window, another file will also be created with the aggregated empiric distribution of P(W | Data)&gt; \n\t\t\t-printFullDistribution &lt;If set the program will print the posterior probability for the sampled values of omega for each position of the alignment. THIS GENERATES A HUGE FILE&gt;\n\t\t\t-likelihood  Computed the probability of an alignment given a model.\n\t\t -mod &lt;Neutral model as defined in task 1&gt; \n\t\t -in &lt;Alignment file&gt; \n\n\t -ref &lt;reference sequence id, necessary if the alignment is not in MAF format&gt; \n\t\t -ignore &lt;comma separated species to ignore&gt;\n\t\tmaximalPWM Given a PWM, an MAF alignment and a neutral model slide PWM and compute the maximum log odds likelihood (of the window being generated by the neutral or PWM models \n\t\t\t-indir &lt;Alignment directory of chromosome  MAF alignments&gt; -pwm &lt;File with PWM description&gt; -mod &lt;Neutral model&gt; [-seedMinScore &lt;Minimum affinity score in order to incurr in the expense of the phylogenetic computation&gt; -ignore &lt;comma separated species to ignore&gt;]\n\t\t\t-regions &lt;annotation file to score &gt; -regionFormat &lt;[BED], GFF or generic&gt; \n\t\t\t[-mafSuffix &lt;A suffix for maf alignment files default is .maf&gt; -out &lt;Output file or standard out if non is specified&gt;  -shuffles &lt;Number of shuffles to do if suffling then a pvalue is reported&gt;&gt;]\n"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>
<A NAME="broad.pda"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left"><FONT SIZE="+2">
broad.pda.*</FONT></TH>
</TR>
</TABLE>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.pda.feature.genome.<A HREF="broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.feature.genome.Chromosome.AgpEntry.CENTROMERE_TYPE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/feature/genome/Chromosome.AgpEntry.html#CENTROMERE_TYPE">CENTROMERE_TYPE</A></CODE></TD>
<TD ALIGN="right"><CODE>1</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.feature.genome.Chromosome.AgpEntry.CLONE_TYPE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/feature/genome/Chromosome.AgpEntry.html#CLONE_TYPE">CLONE_TYPE</A></CODE></TD>
<TD ALIGN="right"><CODE>0</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.feature.genome.Chromosome.AgpEntry.CONTIG_TYPE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/feature/genome/Chromosome.AgpEntry.html#CONTIG_TYPE">CONTIG_TYPE</A></CODE></TD>
<TD ALIGN="right"><CODE>4</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.feature.genome.Chromosome.AgpEntry.GAP_TYPE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/feature/genome/Chromosome.AgpEntry.html#GAP_TYPE">GAP_TYPE</A></CODE></TD>
<TD ALIGN="right"><CODE>2</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.feature.genome.Chromosome.AgpEntry.OTHER_TYPE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/feature/genome/Chromosome.AgpEntry.html#OTHER_TYPE">OTHER_TYPE</A></CODE></TD>
<TD ALIGN="right"><CODE>3</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.feature.genome.Chromosome.AgpEntry.SHORT_ARM_TYPE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/feature/genome/Chromosome.AgpEntry.html#SHORT_ARM_TYPE">SHORT_ARM_TYPE</A></CODE></TD>
<TD ALIGN="right"><CODE>5</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.pda.samtools.<A HREF="broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.samtools.SAMPairedEndFileReader.MAX_INSERT"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/samtools/SAMPairedEndFileReader.html#MAX_INSERT">MAX_INSERT</A></CODE></TD>
<TD ALIGN="right"><CODE>2000</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.pda.seq.alignment.<A HREF="broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.seq.alignment.StoredAlignmentStats.SAVED_MODEL_STATS_EXT"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/seq/alignment/StoredAlignmentStats.html#SAVED_MODEL_STATS_EXT">SAVED_MODEL_STATS_EXT</A></CODE></TD>
<TD ALIGN="right"><CODE>".srptr.aln.stats"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.pda.seq.rap.<A HREF="broad/pda/seq/rap/GCContent.html" title="class in broad.pda.seq.rap">GCContent</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.seq.rap.GCContent.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/seq/rap/GCContent.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Calculates % GC in windows across the genome and outputs in bedgraph format."</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">broad.pda.seq.rap.<A HREF="broad/pda/seq/rap/UnmarkDuplicates.html" title="class in broad.pda.seq.rap">UnmarkDuplicates</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="broad.pda.seq.rap.UnmarkDuplicates.USAGE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="broad/pda/seq/rap/UnmarkDuplicates.html#USAGE">USAGE</A></CODE></TD>
<TD ALIGN="right"><CODE>"Remove the duplicate flag from all reads in a SAM or BAM file."</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>
<A NAME="jaligner"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left"><FONT SIZE="+2">
jaligner.*</FONT></TH>
</TR>
</TABLE>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">jaligner.<A HREF="jaligner/Alignment.html" title="class in jaligner">Alignment</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Alignment.GAP"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;char</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Alignment.html#GAP">GAP</A></CODE></TD>
<TD ALIGN="right"><CODE>45</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">jaligner.<A HREF="jaligner/Directions.html" title="class in jaligner">Directions</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Directions.DIAGONAL"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;byte</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Directions.html#DIAGONAL">DIAGONAL</A></CODE></TD>
<TD ALIGN="right"><CODE>2</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Directions.LEFT"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;byte</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Directions.html#LEFT">LEFT</A></CODE></TD>
<TD ALIGN="right"><CODE>1</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Directions.STOP"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;byte</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Directions.html#STOP">STOP</A></CODE></TD>
<TD ALIGN="right"><CODE>0</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Directions.UP"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;byte</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Directions.html#UP">UP</A></CODE></TD>
<TD ALIGN="right"><CODE>3</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">jaligner.<A HREF="jaligner/Markups.html" title="class in jaligner">Markups</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Markups.GAP"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;char</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Markups.html#GAP">GAP</A></CODE></TD>
<TD ALIGN="right"><CODE>32</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Markups.IDENTITY"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;char</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Markups.html#IDENTITY">IDENTITY</A></CODE></TD>
<TD ALIGN="right"><CODE>124</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Markups.MISMATCH"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;char</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Markups.html#MISMATCH">MISMATCH</A></CODE></TD>
<TD ALIGN="right"><CODE>46</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Markups.SIMILARITY"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;char</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Markups.html#SIMILARITY">SIMILARITY</A></CODE></TD>
<TD ALIGN="right"><CODE>58</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">jaligner.<A HREF="jaligner/Sequence.html" title="class in jaligner">Sequence</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Sequence.NUCLEIC"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Sequence.html#NUCLEIC">NUCLEIC</A></CODE></TD>
<TD ALIGN="right"><CODE>0</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.Sequence.PROTEIN"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/Sequence.html#PROTEIN">PROTEIN</A></CODE></TD>
<TD ALIGN="right"><CODE>1</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>
<A NAME="jaligner.matrix"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left"><FONT SIZE="+2">
jaligner.matrix.*</FONT></TH>
</TR>
</TABLE>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">jaligner.matrix.<A HREF="jaligner/matrix/Matrix.html" title="class in jaligner.matrix">Matrix</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.matrix.Matrix.SIZE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/matrix/Matrix.html#SIZE">SIZE</A></CODE></TD>
<TD ALIGN="right"><CODE>127</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>
<A NAME="jaligner.util"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left"><FONT SIZE="+2">
jaligner.util.*</FONT></TH>
</TR>
</TABLE>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">jaligner.util.<A HREF="jaligner/util/Commons.html" title="class in jaligner.util">Commons</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="jaligner.util.Commons.TAB"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="jaligner/util/Commons.html#TAB">TAB</A></CODE></TD>
<TD ALIGN="right"><CODE>"\t"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>
<A NAME="nextgen.core"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left"><FONT SIZE="+2">
nextgen.core.*</FONT></TH>
</TR>
</TABLE>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">nextgen.core.annotation.<A HREF="nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.annotation.AbstractAnnotation.MAX_DISTANCE"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/annotation/AbstractAnnotation.html#MAX_DISTANCE">MAX_DISTANCE</A></CODE></TD>
<TD ALIGN="right"><CODE>1000000000</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">nextgen.core.coordinatesystem.<A HREF="nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.coordinatesystem.GenomicSpace.PERMUTATION_ATTEMPTS"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/coordinatesystem/GenomicSpace.html#PERMUTATION_ATTEMPTS">PERMUTATION_ATTEMPTS</A></CODE></TD>
<TD ALIGN="right"><CODE>10</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">nextgen.core.coordinatesystem.<A HREF="nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.coordinatesystem.TranscriptomeSpace.PERMUTATION_ATTEMPTS"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;int</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/coordinatesystem/TranscriptomeSpace.html#PERMUTATION_ATTEMPTS">PERMUTATION_ATTEMPTS</A></CODE></TD>
<TD ALIGN="right"><CODE>10</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">nextgen.core.model.score.<A HREF="nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.model.score.NewRatioScore.RPKM_OFFSET"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;double</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/model/score/NewRatioScore.html#RPKM_OFFSET">RPKM_OFFSET</A></CODE></TD>
<TD ALIGN="right"><CODE>0.0010</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">nextgen.core.model.score.<A HREF="nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.model.score.RatioScore.RPKM_OFFSET"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;double</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/model/score/RatioScore.html#RPKM_OFFSET">RPKM_OFFSET</A></CODE></TD>
<TD ALIGN="right"><CODE>0.0010</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">nextgen.core.model.score.<A HREF="nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.model.score.WindowAllScore.RPKM_OFFSET"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;double</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/model/score/WindowAllScore.html#RPKM_OFFSET">RPKM_OFFSET</A></CODE></TD>
<TD ALIGN="right"><CODE>0.0010</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>

<TABLE BORDER="1" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="3">nextgen.core.writers.<A HREF="nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.writers.PairedEndWriter.mateCigarFlag"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/writers/PairedEndWriter.html#mateCigarFlag">mateCigarFlag</A></CODE></TD>
<TD ALIGN="right"><CODE>"mc"</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.writers.PairedEndWriter.mateLineFlag"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/writers/PairedEndWriter.html#mateLineFlag">mateLineFlag</A></CODE></TD>
<TD ALIGN="right"><CODE>"mateLine"</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.writers.PairedEndWriter.mateSequenceFlag"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/writers/PairedEndWriter.html#mateSequenceFlag">mateSequenceFlag</A></CODE></TD>
<TD ALIGN="right"><CODE>"ms"</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.writers.PairedEndWriter.readCigarFlag"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/writers/PairedEndWriter.html#readCigarFlag">readCigarFlag</A></CODE></TD>
<TD ALIGN="right"><CODE>"rc"</CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<A NAME="nextgen.core.writers.PairedEndWriter.readStartFlag"><!-- --></A><TD ALIGN="right"><FONT SIZE="-1">
<CODE>public&nbsp;static&nbsp;final&nbsp;java.lang.String</CODE></FONT></TD>
<TD ALIGN="left"><CODE><A HREF="nextgen/core/writers/PairedEndWriter.html#readStartFlag">readStartFlag</A></CODE></TD>
<TD ALIGN="right"><CODE>"rs"</CODE></TD>
</TR>
</FONT></TD>
</TR>
</TABLE>

<P>

<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="index.html?constant-values.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="constant-values.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
