<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:24 EDT 2013 -->
<TITLE>
ChromosomeWithBubblesJGraphT
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="ChromosomeWithBubblesJGraphT";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ChromosomeWithBubblesJGraphT.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV CLASS&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ChromosomeWithBubblesJGraphT.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
broad.pda.seq.graph</FONT>
<BR>
Class ChromosomeWithBubblesJGraphT</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../resources/inherit.gif" ALT="extended by ">org.jgrapht.graph.AbstractGraph&lt;V,E&gt;
      <IMG SRC="../../../../resources/inherit.gif" ALT="extended by ">org.jgrapht.graph.AbstractBaseGraph&lt;V,E&gt;
          <IMG SRC="../../../../resources/inherit.gif" ALT="extended by ">org.jgrapht.graph.DefaultDirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;
              <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><B>broad.pda.seq.graph.ChromosomeWithBubblesJGraphT</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.io.Serializable, java.lang.Cloneable, org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;, org.jgrapht.Graph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>ChromosomeWithBubblesJGraphT</B><DT>extends org.jgrapht.graph.DefaultDirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</DL>
</PRE>

<P>
Implements jung.Graph for Dijkstra (FOR NOW ONLY)
<P>

<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../../serialized-form.html#broad.pda.seq.graph.ChromosomeWithBubblesJGraphT">Serialized Form</A></DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This class represents the Edges of the graph</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.ForwardAndBackIterator</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;FORWARD ITERATOR</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.MinReadFilter.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.MinReadFilter</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This class exercises the minimum length filter on reads</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.RegionReadFilter.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.RegionReadFilter</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Read filter overlapping region</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ReverseWindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.ReverseWindowIterator</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;REVERSE ITERATOR</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An iterator over the collection of RefSeq genes</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String)">ChromosomeWithBubblesJGraphT</A></B>(java.lang.String&nbsp;name)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CONSTRUCTORS
 TODO: A constructor that instantiates the class with a graph.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Collection, java.util.Collection)">ChromosomeWithBubblesJGraphT</A></B>(java.lang.String&nbsp;name,
                             java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;vertices,
                             java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;edges)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs a chromosome with bubbles with a given set of nodes and connections between them.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Collection, java.util.Collection, double, double, double, int)">ChromosomeWithBubblesJGraphT</A></B>(java.lang.String&nbsp;name,
                             java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;nodes,
                             java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;edges,
                             double&nbsp;lambda,
                             double&nbsp;numMarkers,
                             double&nbsp;numReads,
                             int&nbsp;minSplices)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;???
 This will always return a Null Pointer Exception</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Collection, java.util.Collection, java.util.Map, java.util.Map, double, double, double, int)">ChromosomeWithBubblesJGraphT</A></B>(java.lang.String&nbsp;name,
                             java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;vertices,
                             java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;edges,
                             java.util.Map&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;edgeCounts,
                             java.util.Map&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;nodeCounts,
                             double&nbsp;lambda,
                             double&nbsp;numMarkers,
                             double&nbsp;numReads,
                             int&nbsp;minSplices)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs a chromosome with bubbles with a given set of nodes and connections between them.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, int, int)">ChromosomeWithBubblesJGraphT</A></B>(java.lang.String&nbsp;name,
                             int&nbsp;start,
                             int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Map, java.util.Map, double, double, double, double)">ChromosomeWithBubblesJGraphT</A></B>(java.lang.String&nbsp;name,
                             java.util.Map&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;nodes,
                             java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>,java.lang.Double&gt;&nbsp;edges,
                             double&nbsp;lambda,
                             double&nbsp;numMarkers,
                             double&nbsp;numReads,
                             double&nbsp;minEdgeSupport)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs a chromosome with bubbles with a given set of nodes, edges.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#addEdge(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)">addEdge</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v1,
        <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v2,
        <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;e)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds an edge e between vertices v1 and v2</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#addEdgeAddingMissingVertices(nextgen.core.annotation.Annotation, float)">addEdgeAddingMissingVertices</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;intron,
                             float&nbsp;count)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds an edge to the graph, adding missing vertices to the graph.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#addNewEdge(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge, nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)">addNewEdge</A></B>(<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;e,
           <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v1,
           <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds a new edge between v1 and v2</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#addVertex(nextgen.core.annotation.Annotation)">addVertex</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds vertex v to the graph</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#checkOrientation(java.util.Collection)">checkOrientation</A></B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;edges)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#createEdge(broad.pda.datastructures.Alignments, nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, java.lang.Integer, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.EdgeSourceType)">createEdge</A></B>(<A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;connection,
           <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v1,
           <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v2,
           java.lang.Integer&nbsp;count,
           <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&nbsp;type)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Creates an edge of type between v1 and v2 for connection alignment with weight</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#findEdge(int, int)">findEdge</A></B>(int&nbsp;start,
         int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns edge between start and end</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#findReachableTargetVertices(nextgen.core.annotation.Annotation, java.util.Collection, java.util.Collection, java.util.Collection, int, org.jgrapht.DirectedGraph)">findReachableTargetVertices</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;source,
                            java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;alreadyTraversed,
                            java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;sinkVertices,
                            java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;reachableSinks,
                            int&nbsp;counter,
                            org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getAllPaths()">getAllPaths</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the collection of Paths through the graph.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getAllPaths(double)">getAllPaths</A></B>(double&nbsp;minSpliceFrequency)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the collection of Paths through the graph with the minimum splice frequency</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getBackwardEdges(nextgen.core.annotation.Annotation, double, java.util.Collection)">getBackwardEdges</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex,
                 double&nbsp;alpha,
                 java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getBackwardPaths(int, int, double, java.util.Collection)">getBackwardPaths</A></B>(int&nbsp;start,
                 int&nbsp;end,
                 double&nbsp;alpha,
                 java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getChromosomeLength()">getChromosomeLength</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Set&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getEdges(java.util.Collection)">getEdges</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;vertices)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns all edges connecting the specified vertices</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getEnd()">getEnd</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getForwardEdges(nextgen.core.annotation.Annotation, double, java.util.Collection)">getForwardEdges</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex,
                double&nbsp;alpha,
                java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getGenePaths(double)">getGenePaths</A></B>(double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of paired end edges</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getGenePaths(double, double)">getGenePaths</A></B>(double&nbsp;alpha,
             double&nbsp;minSpliceFrequency)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getIncidentEdges(nextgen.core.annotation.Annotation)">getIncidentEdges</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of all incoming and outgoing edges of specified vertex</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getInOrUnorientedEdges(nextgen.core.annotation.Annotation)">getInOrUnorientedEdges</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of edges that terminate at the specified vertex (if oriented) 
 or have the specified vertex as an end point</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getLambda()">getLambda</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getLocalRate()">getLocalRate</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getLocalRate(nextgen.core.annotation.Annotation)">getLocalRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;exon)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getName()">getName</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/jgrapht/util/VertexPair.html" title="class in org.jgrapht.util">VertexPair</A>&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getNodePair(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)">getNodePair</A></B>(<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;edge)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the pair of vertices for a specified edge</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getNumberOfMarkers()">getNumberOfMarkers</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getNumberOfReads()">getNumberOfReads</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOppositeVertex(nextgen.core.annotation.Annotation, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)">getOppositeVertex</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v,
                  <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;e)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the opposite vertex of the edge e from the vertex v</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOrphanNodes()">getOrphanNodes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns all orphan nodes in graph (Vertices without any edges orphan vertices)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOrphanNodes(broad.pda.datastructures.Alignments)">getOrphanNodes</A></B>(<A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns all orphan nodes in region (Vertices without any edges orphan vertices)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOrphanPaths()">getOrphanPaths</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns all orphan paths in graph (self edges for orphan vertices)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOutOrUnorientedEdges(nextgen.core.annotation.Annotation)">getOutOrUnorientedEdges</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of edges that originate at the specified vertex (if oriented) 
 or have the specified vertex as an end point</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOverlappingVertices(int, int)">getOverlappingVertices</A></B>(int&nbsp;start,
                       int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a list of all vertices overlapping region between start and end</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, double, java.util.Collection, double, org.jgrapht.DirectedGraph)">getPaths</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;startNode,
         <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;endNode,
         double&nbsp;alpha,
         java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validEdgeTypes,
         double&nbsp;minSpliceFreq,
         org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of all paths starting from startNode having the specified valid edge types</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, double, double, org.jgrapht.DirectedGraph)">getPaths</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;startNode,
         <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;endNode,
         double&nbsp;alpha,
         double&nbsp;minSpliceFreq,
         org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of all paths starting from startNode with alpha and minimum splicing frequency</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(double)">getPaths</A></B>(double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of paths with alpha</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(double, double)">getPaths</A></B>(double&nbsp;alpha,
         double&nbsp;minSpliceFrequency)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of paths with alpha and minimum spicing frequency</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int)">getPaths</A></B>(int&nbsp;start,
         int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DO WE NEED THIS????
 Returns collection of genes between start and end</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int, boolean)">getPaths</A></B>(int&nbsp;start,
         int&nbsp;end,
         boolean&nbsp;forward)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int, double, boolean)">getPaths</A></B>(int&nbsp;start,
         int&nbsp;end,
         double&nbsp;alpha,
         boolean&nbsp;forward)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int, double, java.util.Collection)">getPaths</A></B>(int&nbsp;start,
         int&nbsp;end,
         double&nbsp;alpha,
         java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getRelativeGenomeLength()">getRelativeGenomeLength</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getRPKMConstant()">getRPKMConstant</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getSinkVertices(java.util.Collection, org.jgrapht.DirectedGraph)">getSinkVertices</A></B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of all the possible target/end vertices for this graph.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getSourceVertices(java.util.Collection, org.jgrapht.DirectedGraph)">getSourceVertices</A></B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                  org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a collection of all the possible source/start vertices for the specified graph.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getSpliceWeight()">getSpliceWeight</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getStart()">getStart</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getValidOutgoingEdges(nextgen.core.annotation.Annotation, double, java.util.Collection, org.jgrapht.DirectedGraph)">getValidOutgoingEdges</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex,
                      double&nbsp;alpha,
                      java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                      org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#inOrientedDegree(nextgen.core.annotation.Annotation, java.util.Collection, org.jgrapht.DirectedGraph)">inOrientedDegree</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v,
                 java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                 org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the in degree of the given vertex for a list of allowed edge types</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#isGap(int, int)">isGap</A></B>(int&nbsp;start,
      int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if position starts in gap but goes past gap</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#isInAllGaps(int, int)">isInAllGaps</A></B>(int&nbsp;start,
            int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if the start and end positions are all found in gaps (introns??)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#isSpanningGap(int, int)">isSpanningGap</A></B>(int&nbsp;start,
              int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if this region spans an intron.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#iterator(int, int)">iterator</A></B>(int&nbsp;windowSize,
         int&nbsp;start)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#iterator(int, int, int)">iterator</A></B>(int&nbsp;windowSize,
         int&nbsp;start,
         int&nbsp;overlap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#iterator(int, int, int, boolean)">iterator</A></B>(int&nbsp;windowSize,
         int&nbsp;start,
         int&nbsp;overlap,
         boolean&nbsp;skipGapSpanningWindows)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#outOrientedDegree(nextgen.core.annotation.Annotation, java.util.Collection, org.jgrapht.DirectedGraph)">outOrientedDegree</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v,
                  java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                  org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the out degree of the given vertex for a list of allowed edge types</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#removeEdge(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)">removeEdge</A></B>(<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;e)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Removes edge from graph</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#removeSpanningGapNodes()">removeSpanningGapNodes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Removes all nodes that span a gap</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#removeVertex(nextgen.core.annotation.Annotation)">removeVertex</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Removes vertex from graph</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setChromosomeLength(int)">setChromosomeLength</A></B>(int&nbsp;chromosomeLength)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setEnd(int)">setEnd</A></B>(int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the end of this graph</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setExonScores(nextgen.core.annotation.Annotation, java.lang.Double)">setExonScores</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;node,
              java.lang.Double&nbsp;score)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setLocalRate(java.util.Collection)">setLocalRate</A></B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setLocalRate(java.util.Map)">setLocalRate</A></B>(java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;localRateMap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setRPKMConstant(double)">setRPKMConstant</A></B>(double&nbsp;num)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setSpliceWeight(double)">setSpliceWeight</A></B>(double&nbsp;spliceWeight)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the splice weight for this graph</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setStart(int)">setStart</A></B>(int&nbsp;start)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the start of this graph</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#writeGraph(java.lang.String)">writeGraph</A></B>(java.lang.String&nbsp;save)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#writeGraph(java.lang.String, broad.pda.datastructures.Alignments)">writeGraph</A></B>(java.lang.String&nbsp;save,
           <A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#writeGraph(java.lang.String, broad.pda.datastructures.Alignments, boolean)">writeGraph</A></B>(java.lang.String&nbsp;save,
           <A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region,
           boolean&nbsp;setUpShortExonLabels)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#writeGraph(java.lang.String, broad.pda.datastructures.Alignments, boolean, java.lang.String)">writeGraph</A></B>(java.lang.String&nbsp;save,
           <A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region,
           boolean&nbsp;setUpShortExonLabels,
           java.lang.String&nbsp;graphName)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Writes the graph to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#writeGraph(java.lang.String, boolean)">writeGraph</A></B>(java.lang.String&nbsp;save,
           boolean&nbsp;setUpShortExonLabels)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write the graph as a DOT format</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.jgrapht.graph.AbstractBaseGraph"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class org.jgrapht.graph.AbstractBaseGraph</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>addEdge, clone, containsEdge, containsVertex, degreeOf, edgeSet, edgesOf, getAllEdges, getEdge, getEdgeFactory, getEdgeSource, getEdgeTarget, getEdgeWeight, incomingEdgesOf, inDegreeOf, isAllowingLoops, isAllowingMultipleEdges, outDegreeOf, outgoingEdgesOf, removeEdge, setEdgeSetFactory, setEdgeWeight, vertexSet</CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.jgrapht.graph.AbstractGraph"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class org.jgrapht.graph.AbstractGraph</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>containsEdge, removeAllEdges, removeAllEdges, removeAllVertices, toString</CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.jgrapht.DirectedGraph"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from interface org.jgrapht.DirectedGraph</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>incomingEdgesOf, inDegreeOf, outDegreeOf, outgoingEdgesOf</CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.jgrapht.Graph"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from interface org.jgrapht.Graph</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>addEdge, containsEdge, containsEdge, containsVertex, edgeSet, edgesOf, getAllEdges, getEdge, getEdgeFactory, getEdgeSource, getEdgeTarget, getEdgeWeight, removeAllEdges, removeAllEdges, removeAllVertices, removeEdge, vertexSet</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="ChromosomeWithBubblesJGraphT(java.lang.String)"><!-- --></A><H3>
ChromosomeWithBubblesJGraphT</H3>
<PRE>
public <B>ChromosomeWithBubblesJGraphT</B>(java.lang.String&nbsp;name)</PRE>
<DL>
<DD>CONSTRUCTORS
 TODO: A constructor that instantiates the class with a graph. Application: subgraph
<P>
</DL>
<HR>

<A NAME="ChromosomeWithBubblesJGraphT(java.lang.String, int, int)"><!-- --></A><H3>
ChromosomeWithBubblesJGraphT</H3>
<PRE>
public <B>ChromosomeWithBubblesJGraphT</B>(java.lang.String&nbsp;name,
                                    int&nbsp;start,
                                    int&nbsp;end)</PRE>
<DL>
</DL>
<HR>

<A NAME="ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Collection, java.util.Collection, java.util.Map, java.util.Map, double, double, double, int)"><!-- --></A><H3>
ChromosomeWithBubblesJGraphT</H3>
<PRE>
public <B>ChromosomeWithBubblesJGraphT</B>(java.lang.String&nbsp;name,
                                    java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;vertices,
                                    java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;edges,
                                    java.util.Map&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;edgeCounts,
                                    java.util.Map&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;nodeCounts,
                                    double&nbsp;lambda,
                                    double&nbsp;numMarkers,
                                    double&nbsp;numReads,
                                    int&nbsp;minSplices)
                             throws java.lang.IllegalArgumentException</PRE>
<DL>
<DD>Constructs a chromosome with bubbles with a given set of nodes and connections between them. The graph will keep track
 of how many times a given edge has been added. Nodes in this case are though as exons of putative transcripts.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>name</CODE> - - Name  of the chromosome.<DD><CODE>vertices</CODE> - Collection of nodes, note than duplicate nodes will be ignored<DD><CODE>edges</CODE> - Collection of edges connecting nodes. We assume a close-open( [start, end) ) specification of the annotation, 
                              thus edge connects two nodes if the start of the edge is node1.getEnd() and the end of the edge is node2.getStart() - 1. 
                                  <b>Edges must be sorted</b>.<DD><CODE>edgeCounts</CODE> - Map of edges to their counts<DD><CODE>nodeCounts</CODE> - Map of vertices to their counts<DD><CODE>lambda</CODE> - <DD><CODE>numMarkers</CODE> - <DD><CODE>numReads</CODE> - <DD><CODE>minSplices</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.IllegalArgumentException</CODE> - if a node or and edge are on a different chromsome than the chromosomeWithBubbles instance.</DL>
</DL>
<HR>

<A NAME="ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Collection, java.util.Collection)"><!-- --></A><H3>
ChromosomeWithBubblesJGraphT</H3>
<PRE>
public <B>ChromosomeWithBubblesJGraphT</B>(java.lang.String&nbsp;name,
                                    java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;vertices,
                                    java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;edges)
                             throws java.lang.IllegalArgumentException</PRE>
<DL>
<DD>Constructs a chromosome with bubbles with a given set of nodes and connections between them. The graph will keep track
 of how many times a given edge has been added. Nodes in this case are though as exons of putative transcripts.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>name</CODE> - - Name  of the chromosome.<DD><CODE>vertices</CODE> - Collection of nodes, note than duplicate nodes will be ignored<DD><CODE>edges</CODE> - Collection of edges connecting nodes. We assume a close-open( [start, end) ) specification of the annotation, 
                              thus edge connects two nodes if the start of the edge is node1.getEnd() and the end of the edge is node2.getStart() - 1. 
                                  <b>Edges must be sorted</b>.
<DT><B>Throws:</B>
<DD><CODE>java.lang.IllegalArgumentException</CODE> - if a node or and edge are on a different chromsome than the chromosomeWithBubbles instance.</DL>
</DL>
<HR>

<A NAME="ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Map, java.util.Map, double, double, double, double)"><!-- --></A><H3>
ChromosomeWithBubblesJGraphT</H3>
<PRE>
public <B>ChromosomeWithBubblesJGraphT</B>(java.lang.String&nbsp;name,
                                    java.util.Map&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;nodes,
                                    java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>,java.lang.Double&gt;&nbsp;edges,
                                    double&nbsp;lambda,
                                    double&nbsp;numMarkers,
                                    double&nbsp;numReads,
                                    double&nbsp;minEdgeSupport)</PRE>
<DL>
<DD>Constructs a chromosome with bubbles with a given set of nodes, edges.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>name</CODE> - Name  of the chromosome.<DD><CODE>nodes</CODE> - Collection of nodes, note than duplicate nodes will be ignored<DD><CODE>edges</CODE> - Collection of edges connecting nodes. We assume a close-open( [start, end) ) specification of the annotation, 
                              thus edge connects two nodes if the start of the edge is node1.getEnd() and the end of the edge is node2.getStart() - 1. 
                                  <b>Edges must be sorted</b>.<DD><CODE>lambda</CODE> - <DD><CODE>numMarkers</CODE> - <DD><CODE>numReads</CODE> - <DD><CODE>minEdgeSupport</CODE> - </DL>
</DL>
<HR>

<A NAME="ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Collection, java.util.Collection, double, double, double, int)"><!-- --></A><H3>
ChromosomeWithBubblesJGraphT</H3>
<PRE>
public <B>ChromosomeWithBubblesJGraphT</B>(java.lang.String&nbsp;name,
                                    java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;nodes,
                                    java.util.Collection&lt;? extends <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;edges,
                                    double&nbsp;lambda,
                                    double&nbsp;numMarkers,
                                    double&nbsp;numReads,
                                    int&nbsp;minSplices)
                             throws java.lang.IllegalArgumentException</PRE>
<DL>
<DD>???
 This will always return a Null Pointer Exception
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>name</CODE> - <DD><CODE>nodes</CODE> - <DD><CODE>edges</CODE> - <DD><CODE>lambda</CODE> - <DD><CODE>numMarkers</CODE> - <DD><CODE>numReads</CODE> - <DD><CODE>minSplices</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.IllegalArgumentException</CODE></DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="addEdge(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><!-- --></A><H3>
addEdge</H3>
<PRE>
public boolean <B>addEdge</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v1,
                       <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v2,
                       <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;e)</PRE>
<DL>
<DD>Adds an edge e between vertices v1 and v2
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE>addEdge</CODE> in interface <CODE>org.jgrapht.Graph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE><DT><B>Overrides:</B><DD><CODE>addEdge</CODE> in class <CODE>org.jgrapht.graph.AbstractBaseGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>e</CODE> - <DD><CODE>v1</CODE> - <DD><CODE>v2</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="addNewEdge(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge, nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><!-- --></A><H3>
addNewEdge</H3>
<PRE>
public boolean <B>addNewEdge</B>(<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;e,
                          <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v1,
                          <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v2)</PRE>
<DL>
<DD>Adds a new edge between v1 and v2
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>e</CODE> - <DD><CODE>v1</CODE> - <DD><CODE>v2</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="addVertex(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
addVertex</H3>
<PRE>
public boolean <B>addVertex</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v)</PRE>
<DL>
<DD>Adds vertex v to the graph
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE>addVertex</CODE> in interface <CODE>org.jgrapht.Graph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE><DT><B>Overrides:</B><DD><CODE>addVertex</CODE> in class <CODE>org.jgrapht.graph.AbstractBaseGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>v</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="getGenePaths(double)"><!-- --></A><H3>
getGenePaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>getGenePaths</B>(double&nbsp;alpha)</PRE>
<DL>
<DD>Returns a collection of paired end edges
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getGenePaths(double, double)"><!-- --></A><H3>
getGenePaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>getGenePaths</B>(double&nbsp;alpha,
                                               double&nbsp;minSpliceFrequency)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getAllPaths()"><!-- --></A><H3>
getAllPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt; <B>getAllPaths</B>()</PRE>
<DL>
<DD>Returns the collection of Paths through the graph. Will call getAllPaths with 0 minimum splicing frequency.
<P>
<DD><DL>

<DT><B>Returns:</B><DD>collection of paths</DL>
</DD>
</DL>
<HR>

<A NAME="getAllPaths(double)"><!-- --></A><H3>
getAllPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt; <B>getAllPaths</B>(double&nbsp;minSpliceFrequency)</PRE>
<DL>
<DD>Returns the collection of Paths through the graph with the minimum splice frequency
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>minSpliceFrequency</CODE> - minimum splice frequency for the edges
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getOrphanNodes()"><!-- --></A><H3>
getOrphanNodes</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>getOrphanNodes</B>()</PRE>
<DL>
<DD>Returns all orphan nodes in graph (Vertices without any edges orphan vertices)
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getOrphanNodes(broad.pda.datastructures.Alignments)"><!-- --></A><H3>
getOrphanNodes</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>getOrphanNodes</B>(<A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region)</PRE>
<DL>
<DD>Returns all orphan nodes in region (Vertices without any edges orphan vertices)
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>region</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getOrphanPaths()"><!-- --></A><H3>
getOrphanPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt; <B>getOrphanPaths</B>()</PRE>
<DL>
<DD>Returns all orphan paths in graph (self edges for orphan vertices)
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getPaths(double)"><!-- --></A><H3>
getPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt; <B>getPaths</B>(double&nbsp;alpha)</PRE>
<DL>
<DD>Returns a collection of paths with alpha
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alpha</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getPaths(double, double)"><!-- --></A><H3>
getPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt; <B>getPaths</B>(double&nbsp;alpha,
                                           double&nbsp;minSpliceFrequency)</PRE>
<DL>
<DD>Returns a collection of paths with alpha and minimum spicing frequency
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alpha</CODE> - <DD><CODE>minSpliceFrequency</CODE> - minimum splicing frequency
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getEdges(java.util.Collection)"><!-- --></A><H3>
getEdges</H3>
<PRE>
public java.util.Set&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt; <B>getEdges</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;vertices)</PRE>
<DL>
<DD>Returns all edges connecting the specified vertices
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>vertices</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="findReachableTargetVertices(nextgen.core.annotation.Annotation, java.util.Collection, java.util.Collection, java.util.Collection, int, org.jgrapht.DirectedGraph)"><!-- --></A><H3>
findReachableTargetVertices</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>findReachableTargetVertices</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;source,
                                                                    java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;alreadyTraversed,
                                                                    java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;sinkVertices,
                                                                    java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;reachableSinks,
                                                                    int&nbsp;counter,
                                                                    org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getPaths(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, double, double, org.jgrapht.DirectedGraph)"><!-- --></A><H3>
getPaths</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt; <B>getPaths</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;startNode,
                                     <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;endNode,
                                     double&nbsp;alpha,
                                     double&nbsp;minSpliceFreq,
                                     org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</PRE>
<DL>
<DD>Returns a collection of all paths starting from startNode with alpha and minimum splicing frequency
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>startNode</CODE> - <DD><CODE>alpha</CODE> - <DD><CODE>minSpliceFreq</CODE> - minimum splicing frequency
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getPaths(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, double, java.util.Collection, double, org.jgrapht.DirectedGraph)"><!-- --></A><H3>
getPaths</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt; <B>getPaths</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;startNode,
                                     <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;endNode,
                                     double&nbsp;alpha,
                                     java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validEdgeTypes,
                                     double&nbsp;minSpliceFreq,
                                     org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</PRE>
<DL>
<DD>Returns a collection of all paths starting from startNode having the specified valid edge types
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>startNode</CODE> - <DD><CODE>alpha</CODE> - <DD><CODE>validEdgeTypes</CODE> - <DD><CODE>minSpliceFreq</CODE> - minimum splicing frequency
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="checkOrientation(java.util.Collection)"><!-- --></A><H3>
checkOrientation</H3>
<PRE>
public boolean <B>checkOrientation</B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;edges)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getOppositeVertex(nextgen.core.annotation.Annotation, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><!-- --></A><H3>
getOppositeVertex</H3>
<PRE>
public <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A> <B>getOppositeVertex</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v,
                                    <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;e)</PRE>
<DL>
<DD>Returns the opposite vertex of the edge e from the vertex v
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>v</CODE> - <DD><CODE>e</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getPaths(int, int)"><!-- --></A><H3>
getPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>getPaths</B>(int&nbsp;start,
                                           int&nbsp;end)</PRE>
<DL>
<DD>DO WE NEED THIS????
 Returns collection of genes between start and end
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>start</CODE> - <DD><CODE>end</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getPaths(int, int, boolean)"><!-- --></A><H3>
getPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>getPaths</B>(int&nbsp;start,
                                           int&nbsp;end,
                                           boolean&nbsp;forward)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getPaths(int, int, double, boolean)"><!-- --></A><H3>
getPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>getPaths</B>(int&nbsp;start,
                                           int&nbsp;end,
                                           double&nbsp;alpha,
                                           boolean&nbsp;forward)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getPaths(int, int, double, java.util.Collection)"><!-- --></A><H3>
getPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>getPaths</B>(int&nbsp;start,
                                           int&nbsp;end,
                                           double&nbsp;alpha,
                                           java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getBackwardPaths(int, int, double, java.util.Collection)"><!-- --></A><H3>
getBackwardPaths</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>getBackwardPaths</B>(int&nbsp;start,
                                                   int&nbsp;end,
                                                   double&nbsp;alpha,
                                                   java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getBackwardEdges(nextgen.core.annotation.Annotation, double, java.util.Collection)"><!-- --></A><H3>
getBackwardEdges</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt; <B>getBackwardEdges</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex,
                                                                                      double&nbsp;alpha,
                                                                                      java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setSpliceWeight(double)"><!-- --></A><H3>
setSpliceWeight</H3>
<PRE>
public void <B>setSpliceWeight</B>(double&nbsp;spliceWeight)</PRE>
<DL>
<DD>Sets the splice weight for this graph
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>spliceWeight</CODE> - Value for splice weight</DL>
</DD>
</DL>
<HR>

<A NAME="getIncidentEdges(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
getIncidentEdges</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt; <B>getIncidentEdges</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex)</PRE>
<DL>
<DD>Returns a collection of all incoming and outgoing edges of specified vertex
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>vertex</CODE> - 
<DT><B>Returns:</B><DD>collection of all edges for specified vertex</DL>
</DD>
</DL>
<HR>

<A NAME="getSourceVertices(java.util.Collection, org.jgrapht.DirectedGraph)"><!-- --></A><H3>
getSourceVertices</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>getSourceVertices</B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                                                          org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</PRE>
<DL>
<DD>Returns a collection of all the possible source/start vertices for the specified graph.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>validTypes</CODE> - List of allowed edge types.
<DT><B>Returns:</B><DD>All vertices having allowed edge types.</DL>
</DD>
</DL>
<HR>

<A NAME="getSinkVertices(java.util.Collection, org.jgrapht.DirectedGraph)"><!-- --></A><H3>
getSinkVertices</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>getSinkVertices</B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                                                        org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</PRE>
<DL>
<DD>Returns a collection of all the possible target/end vertices for this graph.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>validTypes</CODE> - List of allowed edge types.
<DT><B>Returns:</B><DD>All vertices having allowed edge types.</DL>
</DD>
</DL>
<HR>

<A NAME="getForwardEdges(nextgen.core.annotation.Annotation, double, java.util.Collection)"><!-- --></A><H3>
getForwardEdges</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt; <B>getForwardEdges</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex,
                                                                                     double&nbsp;alpha,
                                                                                     java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getValidOutgoingEdges(nextgen.core.annotation.Annotation, double, java.util.Collection, org.jgrapht.DirectedGraph)"><!-- --></A><H3>
getValidOutgoingEdges</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt; <B>getValidOutgoingEdges</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex,
                                                                                           double&nbsp;alpha,
                                                                                           java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                                                                                           org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="inOrientedDegree(nextgen.core.annotation.Annotation, java.util.Collection, org.jgrapht.DirectedGraph)"><!-- --></A><H3>
inOrientedDegree</H3>
<PRE>
public int <B>inOrientedDegree</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v,
                            java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                            org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</PRE>
<DL>
<DD>Returns the in degree of the given vertex for a list of allowed edge types
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>v</CODE> - specific vertex<DD><CODE>validTypes</CODE> - Collections of allowed edge types
<DT><B>Returns:</B><DD>in degree of given vertex</DL>
</DD>
</DL>
<HR>

<A NAME="outOrientedDegree(nextgen.core.annotation.Annotation, java.util.Collection, org.jgrapht.DirectedGraph)"><!-- --></A><H3>
outOrientedDegree</H3>
<PRE>
public int <B>outOrientedDegree</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v,
                             java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&gt;&nbsp;validTypes,
                             org.jgrapht.DirectedGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;&nbsp;g)</PRE>
<DL>
<DD>Returns the out degree of the given vertex for a list of allowed edge types
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>v</CODE> - specific vertex<DD><CODE>validTypes</CODE> - Collections of allowed edge types
<DT><B>Returns:</B><DD>out degree of given vertex</DL>
</DD>
</DL>
<HR>

<A NAME="isSpanningGap(int, int)"><!-- --></A><H3>
isSpanningGap</H3>
<PRE>
public boolean <B>isSpanningGap</B>(int&nbsp;start,
                             int&nbsp;end)</PRE>
<DL>
<DD>Returns true if this region spans an intron. Also, returns true if the position starts in gap but goes past gap.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>start</CODE> - <DD><CODE>end</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="isGap(int, int)"><!-- --></A><H3>
isGap</H3>
<PRE>
public boolean <B>isGap</B>(int&nbsp;start,
                     int&nbsp;end)</PRE>
<DL>
<DD>Returns true if position starts in gap but goes past gap
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>start</CODE> - <DD><CODE>end</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="isInAllGaps(int, int)"><!-- --></A><H3>
isInAllGaps</H3>
<PRE>
public boolean <B>isInAllGaps</B>(int&nbsp;start,
                           int&nbsp;end)</PRE>
<DL>
<DD>Returns true if the start and end positions are all found in gaps (introns??)
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>start</CODE> - <DD><CODE>end</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="removeSpanningGapNodes()"><!-- --></A><H3>
removeSpanningGapNodes</H3>
<PRE>
public void <B>removeSpanningGapNodes</B>()</PRE>
<DL>
<DD>Removes all nodes that span a gap
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="removeVertex(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
removeVertex</H3>
<PRE>
public boolean <B>removeVertex</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;vertex)</PRE>
<DL>
<DD>Removes vertex from graph
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE>removeVertex</CODE> in interface <CODE>org.jgrapht.Graph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE><DT><B>Overrides:</B><DD><CODE>removeVertex</CODE> in class <CODE>org.jgrapht.graph.AbstractBaseGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>vertex</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="removeEdge(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><!-- --></A><H3>
removeEdge</H3>
<PRE>
public boolean <B>removeEdge</B>(<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;e)</PRE>
<DL>
<DD>Removes edge from graph
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE>removeEdge</CODE> in interface <CODE>org.jgrapht.Graph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE><DT><B>Overrides:</B><DD><CODE>removeEdge</CODE> in class <CODE>org.jgrapht.graph.AbstractBaseGraph&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt;</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>e</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getOutOrUnorientedEdges(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
getOutOrUnorientedEdges</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt; <B>getOutOrUnorientedEdges</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v)</PRE>
<DL>
<DD>Returns a collection of edges that originate at the specified vertex (if oriented) 
 or have the specified vertex as an end point
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>v</CODE> - Specified vertex
<DT><B>Returns:</B><DD>collection of edges</DL>
</DD>
</DL>
<HR>

<A NAME="getInOrUnorientedEdges(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
getInOrUnorientedEdges</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&gt; <B>getInOrUnorientedEdges</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v)</PRE>
<DL>
<DD>Returns a collection of edges that terminate at the specified vertex (if oriented) 
 or have the specified vertex as an end point
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>v</CODE> - Specified vertex
<DT><B>Returns:</B><DD>collection of edges</DL>
</DD>
</DL>
<HR>

<A NAME="addEdgeAddingMissingVertices(nextgen.core.annotation.Annotation, float)"><!-- --></A><H3>
addEdgeAddingMissingVertices</H3>
<PRE>
public void <B>addEdgeAddingMissingVertices</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;intron,
                                         float&nbsp;count)</PRE>
<DL>
<DD>Adds an edge to the graph, adding missing vertices to the graph.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>intron</CODE> - <DD><CODE>count</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="findEdge(int, int)"><!-- --></A><H3>
findEdge</H3>
<PRE>
public <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A> <B>findEdge</B>(int&nbsp;start,
                           int&nbsp;end)</PRE>
<DL>
<DD>Returns edge between start and end
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>start</CODE> - <DD><CODE>end</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getOverlappingVertices(int, int)"><!-- --></A><H3>
getOverlappingVertices</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>getOverlappingVertices</B>(int&nbsp;start,
                                                         int&nbsp;end)</PRE>
<DL>
<DD>Returns a list of all vertices overlapping region between start and end
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>start</CODE> - <DD><CODE>end</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="createEdge(broad.pda.datastructures.Alignments, nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, java.lang.Integer, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.EdgeSourceType)"><!-- --></A><H3>
createEdge</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A> <B>createEdge</B>(<A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;connection,
                                                          <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v1,
                                                          <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;v2,
                                                          java.lang.Integer&nbsp;count,
                                                          <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.EdgeSourceType</A>&nbsp;type)</PRE>
<DL>
<DD>Creates an edge of type between v1 and v2 for connection alignment with weight
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>connection</CODE> - <DD><CODE>v1</CODE> - <DD><CODE>v2</CODE> - <DD><CODE>count</CODE> - <DD><CODE>type</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getNodePair(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><!-- --></A><H3>
getNodePair</H3>
<PRE>
public <A HREF="../../../../org/jgrapht/util/VertexPair.html" title="class in org.jgrapht.util">VertexPair</A>&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>getNodePair</B>(<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>&nbsp;edge)</PRE>
<DL>
<DD>Returns the pair of vertices for a specified edge
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>edge</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="iterator(int, int, int)"><!-- --></A><H3>
iterator</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A> <B>iterator</B>(int&nbsp;windowSize,
                                                            int&nbsp;start,
                                                            int&nbsp;overlap)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="iterator(int, int, int, boolean)"><!-- --></A><H3>
iterator</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A> <B>iterator</B>(int&nbsp;windowSize,
                                                            int&nbsp;start,
                                                            int&nbsp;overlap,
                                                            boolean&nbsp;skipGapSpanningWindows)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="iterator(int, int)"><!-- --></A><H3>
iterator</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A> <B>iterator</B>(int&nbsp;windowSize,
                                                            int&nbsp;start)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getLocalRate(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
getLocalRate</H3>
<PRE>
public double <B>getLocalRate</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;exon)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getLambda()"><!-- --></A><H3>
getLambda</H3>
<PRE>
public double <B>getLambda</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getRelativeGenomeLength()"><!-- --></A><H3>
getRelativeGenomeLength</H3>
<PRE>
public int <B>getRelativeGenomeLength</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getNumberOfMarkers()"><!-- --></A><H3>
getNumberOfMarkers</H3>
<PRE>
public double <B>getNumberOfMarkers</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getNumberOfReads()"><!-- --></A><H3>
getNumberOfReads</H3>
<PRE>
public double <B>getNumberOfReads</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getName()"><!-- --></A><H3>
getName</H3>
<PRE>
public java.lang.String <B>getName</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getRPKMConstant()"><!-- --></A><H3>
getRPKMConstant</H3>
<PRE>
public double <B>getRPKMConstant</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getLocalRate()"><!-- --></A><H3>
getLocalRate</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt; <B>getLocalRate</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getEnd()"><!-- --></A><H3>
getEnd</H3>
<PRE>
public int <B>getEnd</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getStart()"><!-- --></A><H3>
getStart</H3>
<PRE>
public int <B>getStart</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setStart(int)"><!-- --></A><H3>
setStart</H3>
<PRE>
public void <B>setStart</B>(int&nbsp;start)</PRE>
<DL>
<DD>Sets the start of this graph
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>start</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="setEnd(int)"><!-- --></A><H3>
setEnd</H3>
<PRE>
public void <B>setEnd</B>(int&nbsp;end)</PRE>
<DL>
<DD>Sets the end of this graph
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>end</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="getSpliceWeight()"><!-- --></A><H3>
getSpliceWeight</H3>
<PRE>
public double <B>getSpliceWeight</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getChromosomeLength()"><!-- --></A><H3>
getChromosomeLength</H3>
<PRE>
public int <B>getChromosomeLength</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setLocalRate(java.util.Collection)"><!-- --></A><H3>
setLocalRate</H3>
<PRE>
public void <B>setLocalRate</B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setLocalRate(java.util.Map)"><!-- --></A><H3>
setLocalRate</H3>
<PRE>
public void <B>setLocalRate</B>(java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;localRateMap)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setRPKMConstant(double)"><!-- --></A><H3>
setRPKMConstant</H3>
<PRE>
public void <B>setRPKMConstant</B>(double&nbsp;num)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setExonScores(nextgen.core.annotation.Annotation, java.lang.Double)"><!-- --></A><H3>
setExonScores</H3>
<PRE>
public void <B>setExonScores</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;node,
                          java.lang.Double&nbsp;score)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setChromosomeLength(int)"><!-- --></A><H3>
setChromosomeLength</H3>
<PRE>
public void <B>setChromosomeLength</B>(int&nbsp;chromosomeLength)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="writeGraph(java.lang.String, boolean)"><!-- --></A><H3>
writeGraph</H3>
<PRE>
public void <B>writeGraph</B>(java.lang.String&nbsp;save,
                       boolean&nbsp;setUpShortExonLabels)
                throws java.io.IOException</PRE>
<DL>
<DD>Write the graph as a DOT format
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>save</CODE> - <DD><CODE>setUpShortExonLabels</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeGraph(java.lang.String)"><!-- --></A><H3>
writeGraph</H3>
<PRE>
public void <B>writeGraph</B>(java.lang.String&nbsp;save)
                throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeGraph(java.lang.String, broad.pda.datastructures.Alignments)"><!-- --></A><H3>
writeGraph</H3>
<PRE>
public void <B>writeGraph</B>(java.lang.String&nbsp;save,
                       <A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region)
                throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeGraph(java.lang.String, broad.pda.datastructures.Alignments, boolean)"><!-- --></A><H3>
writeGraph</H3>
<PRE>
public void <B>writeGraph</B>(java.lang.String&nbsp;save,
                       <A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region,
                       boolean&nbsp;setUpShortExonLabels)
                throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeGraph(java.lang.String, broad.pda.datastructures.Alignments, boolean, java.lang.String)"><!-- --></A><H3>
writeGraph</H3>
<PRE>
public void <B>writeGraph</B>(java.lang.String&nbsp;save,
                       <A HREF="../../../../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>&nbsp;region,
                       boolean&nbsp;setUpShortExonLabels,
                       java.lang.String&nbsp;graphName)
                throws java.io.IOException</PRE>
<DL>
<DD>Writes the graph to a file
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>save</CODE> - <DD><CODE>region</CODE> - <DD><CODE>setUpShortExonLabels</CODE> - <DD><CODE>graphName</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ChromosomeWithBubblesJGraphT.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV CLASS&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ChromosomeWithBubblesJGraphT.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
