<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:30 EDT 2013 -->
<TITLE>
E-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="E-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-4.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-6.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-5.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-5.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_E_"><!-- --></A><H2>
<B>E</B></H2>
<DL>
<DT><A HREF="../broad/core/primer3/RestrictionEnzyme.html#EciI"><B>EciI</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/RestrictionEnzyme.html" title="class in broad.core.primer3">RestrictionEnzyme</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/RestrictionEnzyme.html#EcoP15I"><B>EcoP15I</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/RestrictionEnzyme.html" title="class in broad.core.primer3">RestrictionEnzyme</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#eig()"><B>eig()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Eigenvalue Decomposition
<DT><A HREF="../Jama/EigenvalueDecomposition.html" title="class in Jama"><B>EigenvalueDecomposition</B></A> - Class in <A HREF="../Jama/package-summary.html">Jama</A><DD>Eigenvalues and eigenvectors of a real matrix.<DT><A HREF="../Jama/EigenvalueDecomposition.html#EigenvalueDecomposition(Jama.Matrix)"><B>EigenvalueDecomposition(Matrix)</B></A> - 
Constructor for class Jama.<A HREF="../Jama/EigenvalueDecomposition.html" title="class in Jama">EigenvalueDecomposition</A>
<DD>Check for symmetry, then construct the eigenvalue decomposition
<DT><A HREF="../broad/core/hmm/MarkovState.html#emitObservation()"><B>emitObservation()</B></A> - 
Method in interface broad.core.hmm.<A HREF="../broad/core/hmm/MarkovState.html" title="interface in broad.core.hmm">MarkovState</A>
<DD>Uses its background distribution to emit a sampled observation.
<DT><A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html#emitObservation()"><B>emitObservation()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html" title="class in broad.core.siphy">PiHMM.ConstrainedModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.PiStateModel.html#emitObservation()"><B>emitObservation()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.PiStateModel.html" title="class in broad.core.siphy">PiHMM.PiStateModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.html#emitPath(int)"><B>emitPath(int)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm">MarkovModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math"><B>EmpiricalDistribution</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(int, double, double)"><B>EmpiricalDistribution(int, double, double)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(int, double, double, boolean)"><B>EmpiricalDistribution(int, double, double, boolean)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(java.util.Collection, int)"><B>EmpiricalDistribution(Collection&lt;Double&gt;, int)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(java.util.Collection)"><B>EmpiricalDistribution(Collection&lt;Double&gt;)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(java.util.Collection, int, boolean)"><B>EmpiricalDistribution(Collection&lt;Double&gt;, int, boolean)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(double[], int)"><B>EmpiricalDistribution(double[], int)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(java.util.List, int)"><B>EmpiricalDistribution(List&lt;Double&gt;, int)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(double[], int, double, double)"><B>EmpiricalDistribution(double[], int, double, double)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(java.io.InputStream)"><B>EmpiricalDistribution(InputStream)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(java.io.File)"><B>EmpiricalDistribution(File)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#EmpiricalDistribution(Jama.Matrix, int)"><B>EmpiricalDistribution(Matrix, int)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#encode()"><B>encode()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#encode()"><B>encode()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#encode(boolean)"><B>encode(boolean)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#encodeAsMatrix()"><B>encodeAsMatrix()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#encodeAsMatrix()"><B>encodeAsMatrix()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#encodeAsMatrix()"><B>encodeAsMatrix()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#encodeSequence()"><B>encodeSequence()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#encodeSequence(java.lang.StringBuilder)"><B>encodeSequence(StringBuilder)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#encodeSequenceAsVector()"><B>encodeSequenceAsVector()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#encodeSequenceIgnoreCase()"><B>encodeSequenceIgnoreCase()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#encodeSequenceIgnoreCase(boolean)"><B>encodeSequenceIgnoreCase(boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#encodeSequenceIgnoreCase(java.lang.StringBuilder)"><B>encodeSequenceIgnoreCase(StringBuilder)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#encodeSequenceIgnoreCase(java.lang.StringBuilder, boolean)"><B>encodeSequenceIgnoreCase(StringBuilder, boolean)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#end"><B>end</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/EndPileUpScanner.html" title="class in nextgen.core.scripture"><B>EndPileUpScanner</B></A> - Class in <A HREF="../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/EndPileUpScanner.html#EndPileUpScanner(java.io.File, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead, java.lang.String, java.lang.String, int, int, java.lang.String)"><B>EndPileUpScanner(File, AbstractPairedEndAlignment.TranscriptionRead, String, String, int, int, String)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/EndPileUpScanner.html" title="class in nextgen.core.scripture">EndPileUpScanner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3IO.html#endPrimer3Communications()"><B>endPrimer3Communications()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3IO.html" title="class in broad.core.primer3">Primer3IO</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#ENFORCE_REFERENCE"><B>ENFORCE_REFERENCE</B></A> - 
Static variable in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment"><B>EnrichmentMetric</B></A> - Class in <A HREF="../broad/pda/enrichment/package-summary.html">broad.pda.enrichment</A><DD>All enrichment metrics should inherit this class.<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#EnrichmentMetric(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String, double, double, int, int)"><B>EnrichmentMetric(String, String, String, String, String, double, double, int, int)</B></A> - 
Constructor for class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentUtils.html" title="class in broad.pda.enrichment"><B>EnrichmentUtils</B></A> - Class in <A HREF="../broad/pda/enrichment/package-summary.html">broad.pda.enrichment</A><DD>&nbsp;<DT><A HREF="../broad/pda/enrichment/EnrichmentUtils.html#EnrichmentUtils()"><B>EnrichmentUtils()</B></A> - 
Constructor for class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentUtils.html" title="class in broad.pda.enrichment">EnrichmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#ensureNoEmptyBins()"><B>ensureNoEmptyBins()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#entropy(double[])"><B>entropy(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html#enumerateAllKMers(java.lang.String, int)"><B>enumerateAllKMers(String, int)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html" title="class in broad.pda.capture.designer">ComputeOriginalScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html#enumerateAllKMers(java.lang.String, int, java.util.Map, java.util.Set, java.util.Set)"><B>enumerateAllKMers(String, int, Map&lt;String, Double&gt;, Set&lt;String&gt;, Set&lt;String&gt;)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html" title="class in broad.pda.capture.designer">ComputeOriginalScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationHandler.html#eof()"><B>eof()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationHandler.html" title="interface in broad.core.annotation">AnnotationHandler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaHandler.html#eof(broad.core.sequence.AbstractFastaParser)"><B>eof(AbstractFastaParser)</B></A> - 
Method in interface broad.core.sequence.<A HREF="../broad/core/sequence/FastaHandler.html" title="interface in broad.core.sequence">FastaHandler</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#eof()"><B>eof()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#equalizeTranscriptEnds(broad.pda.annotation.BEDFileParser)"><B>equalizeTranscriptEnds(BEDFileParser)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PrimerPickingMode.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PrimerPickingMode.html" title="class in broad.core.primer3">Primer3PrimerPickingMode</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>Equal if left and right primers sequences are the same
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#equals(nextgen.core.annotation.Annotation)"><B>equals(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#equals(nextgen.core.annotation.Annotation, boolean)"><B>equals(Annotation, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#equals(broad.pda.methylation.PadlockProbe)"><B>equals(PadlockProbe)</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Indicates whether some other object is "equal to" this one.
<DT><A HREF="../broad/pda/seq/graph/Path.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FragmentAlignment.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FragmentAlignment.html" title="class in nextgen.core.alignment">FragmentAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/PairedReadAlignment.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/PairedReadAlignment.html" title="class in nextgen.core.alignment">PairedReadAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#equals(nextgen.core.annotation.Annotation)"><B>equals(Annotation)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#equals(nextgen.core.annotation.Annotation)"><B>equals(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#equals(nextgen.core.annotation.Annotation, boolean)"><B>equals(Annotation, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#equals(nextgen.core.annotation.Annotation)"><B>equals(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#equals(nextgen.core.annotation.Annotation, boolean)"><B>equals(Annotation, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Test whether two annotations are equal regardless of whether the orientation are the same
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#equals(nextgen.core.annotation.CompoundInterval)"><B>equals(CompoundInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/SingleInterval.html#equals(nextgen.core.annotation.SingleInterval)"><B>equals(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation">SingleInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>True iff count and annotation are equal
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>True iff scan P value and annotation are equal
<DT><A HREF="../nextgen/core/readFilters/NoSpliceFilter.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/NoSpliceFilter.html" title="class in nextgen.core.readFilters">NoSpliceFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/SplicedReadFilter.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/SplicedReadFilter.html" title="class in nextgen.core.readFilters">SplicedReadFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#equals(nextgen.core.annotation.Annotation)"><B>equals(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#equals(nextgen.core.annotation.Annotation, boolean)"><B>equals(Annotation, boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../org/jgrapht/util/VertexPair.html#equals(java.lang.Object)"><B>equals(Object)</B></A> - 
Method in class org.jgrapht.util.<A HREF="../org/jgrapht/util/VertexPair.html" title="class in org.jgrapht.util">VertexPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html" title="class in broad.core.siphy.tools.conservation"><B>EstimateOmegaPerExon</B></A> - Class in <A HREF="../broad/core/siphy/tools/conservation/package-summary.html">broad.core.siphy.tools.conservation</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html#EstimateOmegaPerExon(java.io.File, java.io.File, java.lang.String, java.lang.String, java.lang.String)"><B>EstimateOmegaPerExon(File, File, String, String, String)</B></A> - 
Constructor for class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html" title="class in broad.core.siphy.tools.conservation">EstimateOmegaPerExon</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html#EstimateOmegaPerExon(java.util.Collection, java.io.File, java.lang.String, java.lang.String, java.lang.String)"><B>EstimateOmegaPerExon(Collection&lt;Gene&gt;, File, String, String, String)</B></A> - 
Constructor for class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html" title="class in broad.core.siphy.tools.conservation">EstimateOmegaPerExon</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#euclideanDistance(double[], double[])"><B>euclideanDistance(double[], double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#euclideanDistance(java.util.List, java.util.List)"><B>euclideanDistance(List&lt;Double&gt;, List&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/EuclideanDistance.html" title="class in broad.pda.geneexpression.clustering"><B>EuclideanDistance</B></A> - Class in <A HREF="../broad/pda/geneexpression/clustering/package-summary.html">broad.pda.geneexpression.clustering</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/clustering/EuclideanDistance.html#EuclideanDistance()"><B>EuclideanDistance()</B></A> - 
Constructor for class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/EuclideanDistance.html" title="class in broad.pda.geneexpression.clustering">EuclideanDistance</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/filter/FullyContainedFilter.html#evaluate(nextgen.core.annotation.Annotation)"><B>evaluate(Annotation)</B></A> - 
Method in class nextgen.core.annotation.filter.<A HREF="../nextgen/core/annotation/filter/FullyContainedFilter.html" title="class in nextgen.core.annotation.filter">FullyContainedFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/filter/OverlapFilter.html#evaluate(nextgen.core.annotation.Annotation)"><B>evaluate(Annotation)</B></A> - 
Method in class nextgen.core.annotation.filter.<A HREF="../nextgen/core/annotation/filter/OverlapFilter.html" title="class in nextgen.core.annotation.filter">OverlapFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/CanonicalSpliceFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/CanonicalSpliceFilter.html" title="class in nextgen.core.readFilters">CanonicalSpliceFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/ChimeraFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/ChimeraFilter.html" title="class in nextgen.core.readFilters">ChimeraFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/FirstOfPairFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/FirstOfPairFilter.html" title="class in nextgen.core.readFilters">FirstOfPairFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/FragmentLengthFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/FragmentLengthFilter.html" title="class in nextgen.core.readFilters">FragmentLengthFilter</A>
<DD>If the region is within a gene, return false if the fragment size is invalid for any isoform, or true otherwise
 If the region is not within a gene, filter on genomic distance
<DT><A HREF="../nextgen/core/readFilters/GenomicSpanFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/GenomicSpanFilter.html" title="class in nextgen.core.readFilters">GenomicSpanFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/IndelFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/IndelFilter.html" title="class in nextgen.core.readFilters">IndelFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/MappingQualityFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/MappingQualityFilter.html" title="class in nextgen.core.readFilters">MappingQualityFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/NoSpliceFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/NoSpliceFilter.html" title="class in nextgen.core.readFilters">NoSpliceFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/NumHitsFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/NumHitsFilter.html" title="class in nextgen.core.readFilters">NumHitsFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/PairedAndProperFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/PairedAndProperFilter.html" title="class in nextgen.core.readFilters">PairedAndProperFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/ProperPairFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/ProperPairFilter.html" title="class in nextgen.core.readFilters">ProperPairFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/SameOrientationFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/SameOrientationFilter.html" title="class in nextgen.core.readFilters">SameOrientationFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/SecondOfPairFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/SecondOfPairFilter.html" title="class in nextgen.core.readFilters">SecondOfPairFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/SplicedReadFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/SplicedReadFilter.html" title="class in nextgen.core.readFilters">SplicedReadFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/UnsplicedReadFilter.html#evaluate(nextgen.core.alignment.Alignment)"><B>evaluate(Alignment)</B></A> - 
Method in class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/UnsplicedReadFilter.html" title="class in nextgen.core.readFilters">UnsplicedReadFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy"><B>EvolutionaryModel</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>Represents an Markovian Evolutionary Model<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#EvolutionaryModel(broad.core.siphy.EvolutionaryModelParameters)"><B>EvolutionaryModel(EvolutionaryModelParameters)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.NodeLikelihoodParameters.html" title="class in broad.core.siphy"><B>EvolutionaryModel.NodeLikelihoodParameters</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/EvolutionaryModel.NodeLikelihoodParameters.html#EvolutionaryModel.NodeLikelihoodParameters(org.forester.phylogeny.PhylogenyNode)"><B>EvolutionaryModel.NodeLikelihoodParameters(PhylogenyNode)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.NodeLikelihoodParameters.html" title="class in broad.core.siphy">EvolutionaryModel.NodeLikelihoodParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy"><B>EvolutionaryModel.OmegaFit</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#EvolutionaryModel.OmegaFit()"><B>EvolutionaryModel.OmegaFit()</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#EvolutionaryModel.OmegaFit(broad.core.annotation.LightweightGenomicAnnotation, double, double, double, double)"><B>EvolutionaryModel.OmegaFit(LightweightGenomicAnnotation, double, double, double, double)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy"><B>EvolutionaryModel.PiFit</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#EvolutionaryModel.PiFit()"><B>EvolutionaryModel.PiFit()</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#EvolutionaryModel.PiFit(java.lang.String[])"><B>EvolutionaryModel.PiFit(String[])</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy"><B>EvolutionaryModelParameters</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#EvolutionaryModelParameters(org.forester.phylogeny.Phylogeny, double, double, Jama.Matrix, Jama.Matrix)"><B>EvolutionaryModelParameters(Phylogeny, double, double, Matrix, Matrix)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#EvolutionaryModelParameters(java.io.File)"><B>EvolutionaryModelParameters(File)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#EvolutionaryModelParameters(double, double[], double)"><B>EvolutionaryModelParameters(double, double[], double)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#evolve(java.lang.String)"><B>evolve(String)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>Assumes the given sequence as ancestral, the runs the markov process forward to 
 generate sequences at ach node.
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#excludeByRowNames(java.util.Collection)"><B>excludeByRowNames(Collection&lt;String&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#expand(int, int)"><B>expand(int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#expand(int, int)"><B>expand(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>Extends the annotation by delta on the unOriented Start and end.
<DT><A HREF="../nextgen/core/annotation/Annotation.html#expand(int, int)"><B>expand(int, int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Expand annotation by a set number of bases, taking into account blocks
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#expand(int, int)"><B>expand(int, int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#expandUtrs(java.lang.Integer, java.lang.Integer)"><B>expandUtrs(Integer, Integer)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#expandUtrs(java.lang.Integer, java.lang.Integer)"><B>expandUtrs(Integer, Integer)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#expanedUtrs(broad.pda.annotation.BEDFileParser, java.lang.Integer, java.lang.Integer)"><B>expanedUtrs(BEDFileParser, Integer, Integer)</B></A> - 
Static method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html#EXPRESSION_PVALUE_CUTOFF"><B>EXPRESSION_PVALUE_CUTOFF</B></A> - 
Static variable in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamPeakCaller</A>
<DD>Transcriptome wide scan P value cutoff for expression of gene
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression"><B>ExpressionExperimentInfo</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#ExpressionExperimentInfo(java.lang.String)"><B>ExpressionExperimentInfo(String)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#extendAnnotation(int, int)"><B>extendAnnotation(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Get new gene extended in either direction
<DT><A HREF="../broad/pda/samtools/PairedEndTDF.html#extendPairedEndReads(java.io.File, java.io.File)"><B>extendPairedEndReads(File, File)</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/PairedEndTDF.html" title="class in broad.pda.samtools">PairedEndTDF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/SplitFastqByIndex.html#extractBarcodeFromFastq(net.sf.picard.fastq.FastqRecord)"><B>extractBarcodeFromFastq(FastqRecord)</B></A> - 
Static method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/SplitFastqByIndex.html" title="class in broad.pda.seq.fastq">SplitFastqByIndex</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/ExtractGenes.html" title="class in broad.projection.gsa"><B>ExtractGenes</B></A> - Class in <A HREF="../broad/projection/gsa/package-summary.html">broad.projection.gsa</A><DD>&nbsp;<DT><A HREF="../broad/projection/gsa/ExtractGenes.html#ExtractGenes()"><B>ExtractGenes()</B></A> - 
Constructor for class broad.projection.gsa.<A HREF="../broad/projection/gsa/ExtractGenes.html" title="class in broad.projection.gsa">ExtractGenes</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/AlignmentExtractor.html#extractGFFRegions(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String[])"><B>extractGFFRegions(String, String, String, String, String, String[])</B></A> - 
Static method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/AlignmentExtractor.html" title="class in broad.core.multiplealignment">AlignmentExtractor</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Extractor.html" title="class in broad.core.sequence"><B>Extractor</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/Extractor.html#Extractor()"><B>Extractor()</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/Extractor.html" title="class in broad.core.sequence">Extractor</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPairReader.html#extractProductSequences(java.lang.String)"><B>extractProductSequences(String)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPairReader.html" title="class in broad.core.primer3">PrimerPairReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRecords(java.util.Collection, java.io.InputStream)"><B>extractRecords(Collection&lt;String&gt;, InputStream)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRecords(java.util.Collection, java.io.InputStream, boolean)"><B>extractRecords(Collection&lt;String&gt;, InputStream, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRecords(java.util.Collection)"><B>extractRecords(Collection&lt;String&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRecordsIntoSequenceList(java.util.List)"><B>extractRecordsIntoSequenceList(List&lt;? extends Sequence&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRecordsWithIDLike(java.lang.String, boolean)"><B>extractRecordsWithIDLike(String, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRecordsWithIDLike(java.lang.String, boolean, int)"><B>extractRecordsWithIDLike(String, boolean, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRecordsWithIDsMatching(java.util.List, boolean)"><B>extractRecordsWithIDsMatching(List&lt;String&gt;, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRecordsWithIDsMatching(java.util.List, boolean, int)"><B>extractRecordsWithIDsMatching(List&lt;String&gt;, boolean, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRegion(broad.core.sequence.SequenceRegion)"><B>extractRegion(SequenceRegion)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#extractRegion(broad.core.sequence.SequenceRegion)"><B>extractRegion(SequenceRegion)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#extractRegionBasedOnGC(float, int, int)"><B>extractRegionBasedOnGC(float, int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#extractRegionBasedOnGC(float, int, int)"><B>extractRegionBasedOnGC(float, int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRegions(java.util.List)"><B>extractRegions(List&lt;? extends SequenceRegion&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRegions(java.util.List, boolean)"><B>extractRegions(List&lt;? extends SequenceRegion&gt;, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>Extracts sequence regions from fasta file into
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#extractRegions(java.util.List, boolean, java.io.InputStream)"><B>extractRegions(List&lt;? extends SequenceRegion&gt;, boolean, InputStream)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>Extracts sequence regions from fasta file into
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#extractRegions(java.util.List)"><B>extractRegions(List&lt;? extends SequenceRegion&gt;)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/ExtractSamplesAndLabel.html" title="class in broad.pda.geneexpression.agilent"><B>ExtractSamplesAndLabel</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/ExtractSamplesAndLabel.html#ExtractSamplesAndLabel(java.lang.String, java.io.File, java.lang.String, java.lang.String)"><B>ExtractSamplesAndLabel(String, File, String, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/ExtractSamplesAndLabel.html" title="class in broad.pda.geneexpression.agilent">ExtractSamplesAndLabel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai"><B>ExtractSequence</B></A> - Class in <A HREF="../broad/pda/rnai/package-summary.html">broad.pda.rnai</A><DD>&nbsp;<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#ExtractSequence()"><B>ExtractSequence()</B></A> - 
Constructor for class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-4.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-6.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-5.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-5.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
