<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:31 EDT 2013 -->
<TITLE>
H-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="H-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-7.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-9.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-8.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-8.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_H_"><!-- --></A><H2>
<B>H</B></H2>
<DL>
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer"><B>HairpinKmer</B></A> - Class in <A HREF="../broad/pda/capture/designer/package-summary.html">broad.pda.capture.designer</A><DD>&nbsp;<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#HairpinKmer(java.lang.String, int, int)"><B>HairpinKmer(String, int, int)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#HairpinKmer(java.lang.String, int, int, java.util.Map, java.util.Set, java.util.Set)"><B>HairpinKmer(String, int, int, Map&lt;String, Double&gt;, Set&lt;String&gt;, Set&lt;String&gt;)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#hamming(double[], double[], double)"><B>hamming(double[], double[], double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>computes the hamming distance between two vectors, given a threshold.
<DT><A HREF="../broad/core/math/Statistics.html#hamming(java.util.List, java.util.List, double)"><B>hamming(List&lt;Double&gt;, List&lt;Double&gt;, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>computes the hamming distance between two vectors, given a threshold.
<DT><A HREF="../broad/core/datastructures/Interval.html#HAS_GREATER_PART"><B>HAS_GREATER_PART</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#HAS_LESSER_PART"><B>HAS_LESSER_PART</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#HAS_OVERLAPPING_PART"><B>HAS_OVERLAPPING_PART</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#hasAnnotationThatContains(nextgen.core.annotation.Annotation)"><B>hasAnnotationThatContains(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#hasAnnotationThatOverlaps(nextgen.core.annotation.Annotation)"><B>hasAnnotationThatOverlaps(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#hasBooleanFlag(java.lang.String)"><B>hasBooleanFlag(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Whether the flag is associated with a boolean argument and was specified on the command line
<DT><A HREF="../nextgen/core/annotation/Gene.html#hasCDS()"><B>hasCDS()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#hasChromosome(java.lang.String)"><B>hasChromosome(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#hasColumn(java.lang.String)"><B>hasColumn(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#hasCommandFor(java.lang.String)"><B>hasCommandFor(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Whether the flag has been specified
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#hasCommandFor(java.lang.String)"><B>hasCommandFor(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Returns true if the specified command was included as runnable in the file
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#hasDataForChromosome(java.lang.String)"><B>hasDataForChromosome(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#hasDoubleFlag(java.lang.String)"><B>hasDoubleFlag(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Whether the flag is associated with a double argument and was specified on the command line
<DT><A HREF="../broad/core/parser/CommandLineParser.html#hasFloatFlag(java.lang.String)"><B>hasFloatFlag(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Whether the flag is associated with a float argument and was specified on the command line
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#hashCode()"><B>hashCode()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#hashCode()"><B>hashCode()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>The string hash code from the string representation
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#hashCode()"><B>hashCode()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns a hash code
<DT><A HREF="../broad/pda/seq/graph/Path.html#hashCode()"><B>hashCode()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#hashCode()"><B>hashCode()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#hashCode()"><B>hashCode()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html#hashCode()"><B>hashCode()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FragmentAlignment.html#hashCode()"><B>hashCode()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FragmentAlignment.html" title="class in nextgen.core.alignment">FragmentAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/PairedReadAlignment.html#hashCode()"><B>hashCode()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/PairedReadAlignment.html" title="class in nextgen.core.alignment">PairedReadAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#hashCode()"><B>hashCode()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../org/jgrapht/util/VertexPair.html#hashCode()"><B>hashCode()</B></A> - 
Method in class org.jgrapht.util.<A HREF="../org/jgrapht/util/VertexPair.html" title="class in org.jgrapht.util">VertexPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/FilterProbesByBlatResults.html#hasHyb(java.lang.String, int)"><B>hasHyb(String, int)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/FilterProbesByBlatResults.html" title="class in broad.pda.capture.designer">FilterProbesByBlatResults</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#hasIndel()"><B>hasIndel()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#hasIndel()"><B>hasIndel()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns whether the read contains an insertion or deletion
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#hasIndel()"><B>hasIndel()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#hasIndex()"><B>hasIndex()</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#hasIndex()"><B>hasIndex()</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#hasInputFile()"><B>hasInputFile()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#hasIntFlag(java.lang.String)"><B>hasIntFlag(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Whether the flag is associated with an int argument and was specified on the command line
<DT><A HREF="../broad/core/math/CombinationGenerator.html#hasMore()"><B>hasMore()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/CombinationGenerator.html" title="class in broad.core.math">CombinationGenerator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#hasNanostringProbeClasses()"><B>hasNanostringProbeClasses()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.FwdIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.FwdIterator.html" title="class in broad.core.datastructures">IntervalTree.FwdIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.OverlapIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.OverlapIterator.html" title="class in broad.core.datastructures">IntervalTree.OverlapIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.RevIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.RevIterator.html" title="class in broad.core.datastructures">IntervalTree.RevIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.ValuesIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.ValuesIterator.html" title="class in broad.core.datastructures">IntervalTree.ValuesIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/ReversibleIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/ReversibleIterator.html" title="class in broad.core.datastructures">ReversibleIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/WindowSlider.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/WindowSlider.html" title="class in broad.core.sequence">WindowSlider</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/WrappedIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/WrappedIterator.html" title="class in broad.pda.samtools">WrappedIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.ForwardAndBackIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ReverseWindowIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ReverseWindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.ReverseWindowIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.EmptyCloseableIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.EmptyCloseableIterator.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel.EmptyCloseableIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html" title="class in broad.pda.seq.segmentation">GenomeWithGaps</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/WrappedIGVIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/WrappedIGVIterator.html" title="class in broad.pda.seq.segmentation">WrappedIGVIterator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/ShuffledIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/ShuffledIterator.html" title="class in nextgen.core.coordinatesystem">ShuffledIterator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/general/CloseableFilterIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class nextgen.core.general.<A HREF="../nextgen/core/general/CloseableFilterIterator.html" title="class in nextgen.core.general">CloseableFilterIterator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowScoreIterator.html#hasNext()"><B>hasNext()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowScoreIterator.html" title="class in nextgen.core.model.score">WindowScoreIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#hasNextExon(java.lang.String)"><B>hasNextExon(String)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#hasNextExon(java.lang.String)"><B>hasNextExon(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#hasNovoalignPath()"><B>hasNovoalignPath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Whether a novoalign executable was specified in config file
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#hasOptionMakeFragmentSizeDistribution()"><B>hasOptionMakeFragmentSizeDistribution()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#hasOrientation()"><B>hasOrientation()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#hasOrientation()"><B>hasOrientation()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#hasOrientation()"><B>hasOrientation()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#hasOrientation()"><B>hasOrientation()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#hasOverlapping3PUTR(broad.pda.gene.GeneWithIsoforms)"><B>hasOverlapping3PUTR(GeneWithIsoforms)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#hasOverlapping5PUTR(broad.pda.gene.GeneWithIsoforms)"><B>hasOverlapping5PUTR(GeneWithIsoforms)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#hasPrimers()"><B>hasPrimers()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#hasRow(java.lang.String)"><B>hasRow(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#hasStringFlag(java.lang.String)"><B>hasStringFlag(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Whether the flag is associated with a string argument and was specified on the command line
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#hasSubClusters()"><B>hasSubClusters()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Pair.html#hasValue1()"><B>hasValue1()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures">Pair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Pair.html#hasValue2()"><B>hasValue2()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures">Pair</A>
<DD>&nbsp;
<DT><A HREF="../org/jgrapht/util/VertexPair.html#hasVertex(V)"><B>hasVertex(V)</B></A> - 
Method in class org.jgrapht.util.<A HREF="../org/jgrapht/util/VertexPair.html" title="class in org.jgrapht.util">VertexPair</A>
<DD>Assess if this pair contains the vertex.
<DT><A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html" title="class in broad.pda.geneexpression.clustering"><B>HierarchicalClustering</B></A> - Class in <A HREF="../broad/pda/geneexpression/clustering/package-summary.html">broad.pda.geneexpression.clustering</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html#HierarchicalClustering(broad.core.datastructures.MatrixWithHeaders)"><B>HierarchicalClustering(MatrixWithHeaders)</B></A> - 
Constructor for class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html" title="class in broad.pda.geneexpression.clustering">HierarchicalClustering</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html#HierarchicalClustering(broad.core.datastructures.MatrixWithHeaders, java.util.Map, java.util.Map)"><B>HierarchicalClustering(MatrixWithHeaders, Map&lt;String, Collection&lt;String&gt;&gt;, Map&lt;String, Collection&lt;String&gt;&gt;)</B></A> - 
Constructor for class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html" title="class in broad.pda.geneexpression.clustering">HierarchicalClustering</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#hochbergPvalueAdjust(double[])"><B>hochbergPvalueAdjust(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>Adjust a set of P values for multiple testing with the Hochberg method
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#HUMAN_LONG_REP"><B>HUMAN_LONG_REP</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.html" title="class in broad.pda.methylation"><B>HybridCaptureDesigner</B></A> - Class in <A HREF="../broad/pda/methylation/package-summary.html">broad.pda.methylation</A><DD>&nbsp;<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.html#HybridCaptureDesigner()"><B>HybridCaptureDesigner()</B></A> - 
Constructor for class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.html" title="class in broad.pda.methylation">HybridCaptureDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.html#HybridCaptureDesigner(java.lang.String, java.lang.String, java.lang.String, int)"><B>HybridCaptureDesigner(String, String, String, int)</B></A> - 
Constructor for class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.html" title="class in broad.pda.methylation">HybridCaptureDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation"><B>HybridCaptureDesigner.Probe</B></A> - Class in <A HREF="../broad/pda/methylation/package-summary.html">broad.pda.methylation</A><DD>&nbsp;<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#HybridCaptureDesigner.Probe(java.lang.String, java.lang.String, java.lang.String, int, int, java.lang.String, java.lang.Boolean, java.lang.Boolean)"><B>HybridCaptureDesigner.Probe(String, String, String, int, int, String, Boolean, Boolean)</B></A> - 
Constructor for class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#hypergeometric(int, int, int, int)"><B>hypergeometric(int, int, int, int)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/MathUtil.html#hypot(double, double)"><B>hypot(double, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/MathUtil.html" title="class in broad.core.math">MathUtil</A>
<DD>sqrt(a^2 + b^2) without under/overflow.
<DT><A HREF="../Jama/util/Maths.html#hypot(double, double)"><B>hypot(double, double)</B></A> - 
Static method in class Jama.util.<A HREF="../Jama/util/Maths.html" title="class in Jama.util">Maths</A>
<DD>sqrt(a^2 + b^2) without under/overflow.
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-7.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-9.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-8.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-8.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
