<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:30 EDT 2013 -->
<TITLE>
D-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="D-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-3.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-5.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-4.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-4.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_D_"><!-- --></A><H2>
<B>D</B></H2>
<DL>
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#dataInverse()"><B>dataInverse()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Takes the inverse (or pseudo inverse if the underlying data is not a squared matrix)
<DT><A HREF="../broad/core/sequence/Sequence.html#decode(short[])"><B>decode(short[])</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#decodeShort(java.lang.Short)"><B>decodeShort(Short)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#DecollapseByIntronLocation(java.util.Collection, java.util.Collection, java.lang.String)"><B>DecollapseByIntronLocation(Collection&lt;Annotation&gt;, Collection&lt;Annotation&gt;, String)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>This method returns only exons that are trimmed down by intron location.
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#DecollapseByIntronLocation(java.util.Collection, java.util.Collection)"><B>DecollapseByIntronLocation(Collection&lt;Annotation&gt;, Collection&lt;Annotation&gt;)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#DecollapseByIntronLocationOld(java.util.Collection, java.util.Collection)"><B>DecollapseByIntronLocationOld(Collection&lt;Annotation&gt;, Collection&lt;Annotation&gt;)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#DecollapseByIntronLocationWorking(java.util.Collection, java.util.Collection)"><B>DecollapseByIntronLocationWorking(Collection&lt;Annotation&gt;, Collection&lt;Annotation&gt;)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#dedup()"><B>dedup()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#dedup()"><B>dedup()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#DEFAULT_ALPHABET_SIZE"><B>DEFAULT_ALPHABET_SIZE</B></A> - 
Static variable in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#DEFAULT_FASTQ_READ_PAIR_NUMBER_DELIMITER"><B>DEFAULT_FASTQ_READ_PAIR_NUMBER_DELIMITER</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#DEFAULT_GAP_EXTEND_PENALTY"><B>DEFAULT_GAP_EXTEND_PENALTY</B></A> - 
Static variable in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>Default gap extend penalty for Smith Waterman
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#DEFAULT_GAP_OPEN_PENALTY"><B>DEFAULT_GAP_OPEN_PENALTY</B></A> - 
Static variable in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>Default gap open penalty for Smith Waterman
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#DEFAULT_INS_SIZE_PVAL"><B>DEFAULT_INS_SIZE_PVAL</B></A> - 
Static variable in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#DEFAULT_INSERT_SIZE_FUDGE"><B>DEFAULT_INSERT_SIZE_FUDGE</B></A> - 
Static variable in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#DEFAULT_MAKE_GENE_OVERLAP"><B>DEFAULT_MAKE_GENE_OVERLAP</B></A> - 
Static variable in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#DEFAULT_MATCH_SCORE"><B>DEFAULT_MATCH_SCORE</B></A> - 
Static variable in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>Default match score for Smith Waterman
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#DEFAULT_MIN_MAPPING_QUALITY"><B>DEFAULT_MIN_MAPPING_QUALITY</B></A> - 
Static variable in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#DEFAULT_MISMATCH_SCORE"><B>DEFAULT_MISMATCH_SCORE</B></A> - 
Static variable in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>Default mismatch score for Smith Waterman
<DT><A HREF="../nextgen/core/model/score/CountScore.html#DEFAULT_REGION_TOTAL"><B>DEFAULT_REGION_TOTAL</B></A> - 
Static variable in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#DEFAULT_VALUE"><B>DEFAULT_VALUE</B></A> - 
Static variable in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/DefineGeneSignatures.html" title="class in broad.projection.gsa"><B>DefineGeneSignatures</B></A> - Class in <A HREF="../broad/projection/gsa/package-summary.html">broad.projection.gsa</A><DD>&nbsp;<DT><A HREF="../broad/projection/gsa/DefineGeneSignatures.html#DefineGeneSignatures(java.io.File, java.io.File, java.lang.String, double)"><B>DefineGeneSignatures(File, File, String, double)</B></A> - 
Constructor for class broad.projection.gsa.<A HREF="../broad/projection/gsa/DefineGeneSignatures.html" title="class in broad.projection.gsa">DefineGeneSignatures</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/DefineSourceAndSinkFromPairedEnds.html" title="class in broad.pda.seq.segmentation"><B>DefineSourceAndSinkFromPairedEnds</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/DefineSourceAndSinkFromPairedEnds.html#DefineSourceAndSinkFromPairedEnds()"><B>DefineSourceAndSinkFromPairedEnds()</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/DefineSourceAndSinkFromPairedEnds.html" title="class in broad.pda.seq.segmentation">DefineSourceAndSinkFromPairedEnds</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/AlignmentExtractor.html#degenerateCodonStarts"><B>degenerateCodonStarts</B></A> - 
Static variable in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/AlignmentExtractor.html" title="class in broad.core.multiplealignment">AlignmentExtractor</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#delete(broad.core.sequence.SequenceRegion)"><B>delete(SequenceRegion)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#delete(broad.core.sequence.SequenceRegion)"><B>delete(SequenceRegion)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html#DEPLETED"><B>DEPLETED</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html" title="class in broad.pda.seq.rap">RatioPermutationPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html#DEPLETED"><B>DEPLETED</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html" title="class in broad.pda.seq.rap">SkellamKLPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.html#design(broad.core.sequence.Sequence, int)"><B>design(Sequence, int)</B></A> - 
Method in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.html" title="class in nextgen.editing">CRISPRDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designCloningPrimers(broad.core.sequence.Sequence, int, broad.core.primer3.Primer3SequenceInputTags.SequenceRegionCoordinates)"><B>designCloningPrimers(Sequence, int, Primer3SequenceInputTags.SequenceRegionCoordinates)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designCloningPrimers(broad.core.sequence.Sequence, int)"><B>designCloningPrimers(Sequence, int)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designIntronPrimers(broad.core.sequence.Sequence, nextgen.core.annotation.Gene, boolean, java.lang.String, java.lang.String, int, int, int)"><B>designIntronPrimers(Sequence, Gene, boolean, String, String, int, int, int)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designJenPrimers(broad.core.sequence.Sequence, nextgen.core.annotation.Gene, boolean, int)"><B>designJenPrimers(Sequence, Gene, boolean, int)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designPCRPrimers(broad.core.sequence.Sequence, nextgen.core.annotation.Gene, boolean, int, boolean)"><B>designPCRPrimers(Sequence, Gene, boolean, int, boolean)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designPCRPrimers(broad.core.sequence.Sequence, nextgen.core.annotation.Gene, boolean, int, broad.core.primer3.Primer3SequenceInputTags.SequenceRegionCoordinates)"><B>designPCRPrimers(Sequence, Gene, boolean, int, Primer3SequenceInputTags.SequenceRegionCoordinates)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/DesignPCRTails.html" title="class in broad.pda.capture.designer"><B>DesignPCRTails</B></A> - Class in <A HREF="../broad/pda/capture/designer/package-summary.html">broad.pda.capture.designer</A><DD>&nbsp;<DT><A HREF="../broad/pda/capture/designer/DesignPCRTails.html#DesignPCRTails()"><B>DesignPCRTails()</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/DesignPCRTails.html" title="class in broad.pda.capture.designer">DesignPCRTails</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/DesignPCRTails.html#DesignPCRTails(int)"><B>DesignPCRTails(int)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/DesignPCRTails.html" title="class in broad.pda.capture.designer">DesignPCRTails</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ArrayDesigner.html#designQpcrPrimer(java.lang.String)"><B>designQpcrPrimer(String)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ArrayDesigner.html" title="class in broad.pda.capture.designer">ArrayDesigner</A>
<DD>Get a qPCR primer pair for a sequence
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designRACEPrimers(broad.core.sequence.Sequence, java.lang.String, java.lang.String, int)"><B>designRACEPrimers(Sequence, String, String, int)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designSyntheticPrimers(java.lang.String, int)"><B>designSyntheticPrimers(String, int)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designSyntheticPrimers(java.lang.String, int, int, int)"><B>designSyntheticPrimers(String, int, int, int)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#designSyntheticPrimers(java.lang.String, java.lang.String, java.lang.String, int)"><B>designSyntheticPrimers(String, String, String, int)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFIO.html#destroyFileHandle()"><B>destroyFileHandle()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFIO.html" title="class in broad.core.multiplealignment">MAFIO</A>
<DD>&nbsp;
<DT><A HREF="../Jama/LUDecomposition.html#det()"><B>det()</B></A> - 
Method in class Jama.<A HREF="../Jama/LUDecomposition.html" title="class in Jama">LUDecomposition</A>
<DD>Determinant
<DT><A HREF="../Jama/Matrix.html#det()"><B>det()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Matrix determinant
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#dgeOptions"><B>dgeOptions</B></A> - 
Variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Directions.html#DIAGONAL"><B>DIAGONAL</B></A> - 
Static variable in class jaligner.<A HREF="../jaligner/Directions.html" title="class in jaligner">Directions</A>
<DD>Traceback direction diagonal
<DT><A HREF="../broad/core/math/Statistics.html#diff(double[], double[])"><B>diff(double[], double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#differenceByGenomicRegion(broad.pda.annotation.BEDFileParser, java.lang.String)"><B>differenceByGenomicRegion(BEDFileParser, String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression"><B>DifferentialExpression</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#DifferentialExpression(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, boolean, boolean, int, double[], broad.core.datastructures.MatrixWithHeaders[], java.util.Map)"><B>DifferentialExpression(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, boolean, boolean, int, double[], MatrixWithHeaders[], Map&lt;String, FDRDistribution&gt;)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#DifferentialExpression(broad.core.datastructures.MatrixWithHeaders, java.util.Map, boolean, boolean, int, double[])"><B>DifferentialExpression(MatrixWithHeaders, Map&lt;String, Collection&lt;String&gt;&gt;, boolean, boolean, int, double[])</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#DifferentialExpression(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, boolean, boolean, int, double[], broad.core.datastructures.MatrixWithHeaders[])"><B>DifferentialExpression(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, boolean, boolean, int, double[], MatrixWithHeaders[])</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#DifferentialExpression(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, boolean, boolean, int)"><B>DifferentialExpression(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, boolean, boolean, int)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#DifferentialExpression(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, boolean, boolean)"><B>DifferentialExpression(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, boolean, boolean)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#DifferentialExpression(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection)"><B>DifferentialExpression(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#DifferentialExpression(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, int)"><B>DifferentialExpression(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, int)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialScoring.html" title="class in broad.pda.differentialExpression"><B>DifferentialScoring</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/DifferentialScoring.html#DifferentialScoring()"><B>DifferentialScoring()</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialScoring.html" title="class in broad.pda.differentialExpression">DifferentialScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/RestrictionEnzyme.html#digest(java.lang.String)"><B>digest(String)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/RestrictionEnzyme.html" title="class in broad.core.primer3">RestrictionEnzyme</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Directions.html" title="class in jaligner"><B>Directions</B></A> - Class in <A HREF="../jaligner/package-summary.html">jaligner</A><DD>Traceback directions.<DT><A HREF="../jaligner/Directions.html#Directions()"><B>Directions()</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/Directions.html" title="class in jaligner">Directions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome"><B>DirectoryInstalledGenomeAssembly</B></A> - Class in <A HREF="../broad/pda/feature/genome/package-summary.html">broad.pda.feature.genome</A><DD>Represents a Genome Assembly were each directory is in its own directory which includes the AGP file
 It is a convenient way to access genomic data when it is stored in this standard way.<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#DirectoryInstalledGenomeAssembly(java.io.File)"><B>DirectoryInstalledGenomeAssembly(File)</B></A> - 
Constructor for class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#DISCRETE_EXTENSION"><B>DISCRETE_EXTENSION</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#disect(nextgen.core.annotation.Annotation)"><B>disect(Annotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>Fragments the current annotation if it overlaps the provided one.
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#disect(java.util.List)"><B>disect(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>Fragments the current annotation if it overlaps the provided ones.
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#disect(nextgen.core.annotation.Annotation)"><B>disect(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#disect(java.util.List)"><B>disect(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#disect(nextgen.core.annotation.Annotation)"><B>disect(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Fragments the current annotation if it overlaps the provided one.
<DT><A HREF="../nextgen/core/annotation/Annotation.html#disect(java.util.List)"><B>disect(List&lt;? extends Annotation&gt;)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Fragments the current annotation if it overlaps the provided ones.
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#disect(nextgen.core.annotation.Annotation)"><B>disect(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#disect(java.util.List)"><B>disect(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#disect(nextgen.core.annotation.Annotation)"><B>disect(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#disect(java.util.List)"><B>disect(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.html#distance(double[], double[], java.lang.String)"><B>distance(double[], double[], String)</B></A> - 
Static method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.html" title="class in broad.pda.geneexpression.clustering">Kmeans</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#distanceBetween(broad.pda.datastructures.Alignments, broad.pda.datastructures.Alignments)"><B>distanceBetween(Alignments, Alignments)</B></A> - 
Static method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#distanceFrom(broad.core.motif.PositionWeightMatrix)"><B>distanceFrom(PositionWeightMatrix)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>Distance between two PWMs, uses the KL similarity
<DT><A HREF="../broad/pda/annotation/Locus.html#distanceToOppositeStrand5primeNeighbor(broad.pda.annotation.BEDFileParser)"><B>distanceToOppositeStrand5primeNeighbor(BEDFileParser)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Distribution.html" title="class in broad.core.math"><B>Distribution</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/Distribution.html#Distribution()"><B>Distribution()</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/Distribution.html" title="class in broad.core.math">Distribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#DNAConcentration"><B>DNAConcentration</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#doesOverlap(int, int)"><B>doesOverlap(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/ReadAlignmentStats.html#doneAlnAnalysis(nextgen.core.programs.ReadAlignmentStats.SAMStats)"><B>doneAlnAnalysis(ReadAlignmentStats.SAMStats)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/ReadAlignmentStats.html" title="class in nextgen.core.programs">ReadAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/ReadAlignmentStats.html#doneCounting(java.lang.String, java.lang.String, int)"><B>doneCounting(String, String, int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/ReadAlignmentStats.html" title="class in nextgen.core.programs">ReadAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/ReconstructionPostProcessor.html#doWork(java.lang.String)"><B>doWork(String)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/ReconstructionPostProcessor.html" title="class in nextgen.core.scripture">ReconstructionPostProcessor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/statistics/ConnectDisconnectedTranscripts.html#doWork(java.lang.String)"><B>doWork(String)</B></A> - 
Method in class nextgen.core.scripture.statistics.<A HREF="../nextgen/core/scripture/statistics/ConnectDisconnectedTranscripts.html" title="class in nextgen.core.scripture.statistics">ConnectDisconnectedTranscripts</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#drawAutosome()"><B>drawAutosome()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>Uses pseudo random number generation to draw a (non random) autosome with
 chance proportional to the autosome size
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#drawChromosome()"><B>drawChromosome()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>Uses pseudo random number generation to draw a (non random) chromosome with
 chance proportional to the chromosome size
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#drawRandomAutosomeRegion(int)"><B>drawRandomAutosomeRegion(int)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#drawRandomRegion(int)"><B>drawRandomRegion(int)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#drawRandomRegion(int)"><B>drawRandomRegion(int)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-3.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-5.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-4.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-4.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
