<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:30 EDT 2013 -->
<TITLE>
G-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="G-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-6.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-8.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-7.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-7.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_G_"><!-- --></A><H2>
<B>G</B></H2>
<DL>
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#G_CODE"><B>G_CODE</B></A> - 
Static variable in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#GAP"><B>GAP</B></A> - 
Static variable in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Gap character
<DT><A HREF="../jaligner/Markups.html#GAP"><B>GAP</B></A> - 
Static variable in class jaligner.<A HREF="../jaligner/Markups.html" title="class in jaligner">Markups</A>
<DD>Markup line gap character
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#GAP_CODE"><B>GAP_CODE</B></A> - 
Static variable in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#GAP_TYPE"><B>GAP_TYPE</B></A> - 
Static variable in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#gaps()"><B>gaps()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html#GC_CUTOFF"><B>GC_CUTOFF</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation">FilterSamByBaseComposition</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#gcContent()"><B>gcContent()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GCContent.html" title="class in broad.pda.seq.rap"><B>GCContent</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/GCContent.html#GCContent()"><B>GCContent()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GCContent.html" title="class in broad.pda.seq.rap">GCContent</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation"><B>Gene</B></A> - Class in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, java.lang.String, nextgen.core.annotation.Annotation.Strand, java.util.Collection, int, int)"><B>Gene(String, String, Annotation.Strand, Collection&lt;? extends Annotation&gt;, int, int)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, nextgen.core.annotation.Annotation.Strand, int, int)"><B>Gene(String, int, int, String, Annotation.Strand, int, int)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.util.Collection, java.lang.String, nextgen.core.annotation.Annotation.Strand, int, int)"><B>Gene(Collection&lt;? extends Annotation&gt;, String, Annotation.Strand, int, int)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, java.lang.String, nextgen.core.annotation.Annotation.Strand, java.util.Collection)"><B>Gene(String, String, Annotation.Strand, Collection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(nextgen.core.annotation.Annotation, java.lang.String, nextgen.core.annotation.Annotation.Strand, java.util.Collection)"><B>Gene(Annotation, String, Annotation.Strand, Collection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(nextgen.core.annotation.Annotation, java.util.Collection)"><B>Gene(Annotation, Collection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(nextgen.core.annotation.Annotation, java.util.Collection, java.lang.String, int, int)"><B>Gene(Annotation, Collection&lt;? extends Annotation&gt;, String, int, int)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(nextgen.core.annotation.Annotation, java.util.Collection, java.lang.String)"><B>Gene(Annotation, Collection&lt;? extends Annotation&gt;, String)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.util.Collection, java.lang.String, nextgen.core.annotation.Annotation.Strand)"><B>Gene(Collection&lt;? extends Annotation&gt;, String, Annotation.Strand)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, nextgen.core.annotation.Annotation.Strand)"><B>Gene(String, int, int, String, Annotation.Strand)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String)"><B>Gene(String, int, int, String)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.util.Collection, java.lang.String)"><B>Gene(Collection&lt;? extends Annotation&gt;, String)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.util.Collection)"><B>Gene(Collection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int)"><B>Gene(String, int, int)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(broad.pda.datastructures.Alignments)"><B>Gene(Alignments)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD><B>Deprecated.</B>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(nextgen.core.annotation.Annotation)"><B>Gene(Annotation)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String)"><B>Gene(String)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, boolean)"><B>Gene(String, boolean)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, double, java.lang.String, java.lang.String, java.lang.String[])"><B>Gene(String, int, int, String, double, String, String, String[])</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, double, java.lang.String, java.lang.String)"><B>Gene(String, int, int, String, double, String, String)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, double, java.lang.String, int[], int[])"><B>Gene(String, int, int, String, double, String, int[], int[])</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, java.lang.String, java.util.List, java.util.List)"><B>Gene(String, int, int, String, String, List&lt;Integer&gt;, List&lt;Integer&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, java.lang.String, java.util.List, java.util.List, int, int)"><B>Gene(String, int, int, String, String, List&lt;Integer&gt;, List&lt;Integer&gt;, int, int)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, java.lang.String, java.util.List, java.util.List, java.lang.String[])"><B>Gene(String, int, int, String, String, List&lt;Integer&gt;, List&lt;Integer&gt;, String[])</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, double, java.lang.String, int[], int[], double[], java.lang.String)"><B>Gene(String, int, int, String, double, String, int[], int[], double[], String)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, double, java.lang.String, int[], int[], double[], java.lang.String, java.lang.String[])"><B>Gene(String, int, int, String, double, String, int[], int[], double[], String, String[])</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, double, java.lang.String, java.util.List, java.util.List, int, int)"><B>Gene(String, int, int, String, double, String, List&lt;Integer&gt;, List&lt;Integer&gt;, int, int)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, java.lang.String, java.util.List, java.util.List, int, int, java.lang.String[])"><B>Gene(String, int, int, String, String, List&lt;Integer&gt;, List&lt;Integer&gt;, int, int, String[])</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(java.lang.String, int, int, java.lang.String, double, java.lang.String, java.util.List, java.util.List, int, int, java.lang.String[])"><B>Gene(String, int, int, String, double, String, List&lt;Integer&gt;, List&lt;Integer&gt;, int, int, String[])</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#Gene(nextgen.core.annotation.Gene)"><B>Gene(Gene)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneCounts.html" title="class in broad.pda.seq.segmentation"><B>GeneCounts</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/GeneCounts.html#GeneCounts(double, double, double, double)"><B>GeneCounts(double, double, double, double)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneCounts.html" title="class in broad.pda.seq.segmentation">GeneCounts</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/geneExpUtils.html" title="class in broad.pda.geneexpression"><B>geneExpUtils</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/geneExpUtils.html#geneExpUtils()"><B>geneExpUtils()</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/geneExpUtils.html" title="class in broad.pda.geneexpression">geneExpUtils</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/matrix/MatrixGenerator.html#generate(float, float)"><B>generate(float, float)</B></A> - 
Static method in class jaligner.matrix.<A HREF="../jaligner/matrix/MatrixGenerator.html" title="class in jaligner.matrix">MatrixGenerator</A>
<DD>Returns scoring matrix from a fixed match/mismatch scoring scheme.
<DT><A HREF="../broad/core/sequence/Sequence.html#generateAll5Mers()"><B>generateAll5Mers()</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#generateHKYQMatrix(double, double)"><B>generateHKYQMatrix(double, double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#generateRandomSequence(int)"><B>generateRandomSequence(int)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#generateSubregions(nextgen.core.annotation.Annotation)"><B>generateSubregions(Annotation)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation"><B>GenericAlignmentDataModel</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, boolean, double, boolean, boolean, java.lang.String, boolean)"><B>GenericAlignmentDataModel(String, String, boolean, double, boolean, boolean, String, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, boolean, double)"><B>GenericAlignmentDataModel(String, String, boolean, double)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String)"><B>GenericAlignmentDataModel(String, String)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, int)"><B>GenericAlignmentDataModel(String, String, int)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, boolean)"><B>GenericAlignmentDataModel(String, String, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, boolean, int, boolean)"><B>GenericAlignmentDataModel(String, String, boolean, int, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, int, boolean)"><B>GenericAlignmentDataModel(String, String, int, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, boolean, boolean)"><B>GenericAlignmentDataModel(String, String, boolean, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, boolean, boolean, boolean)"><B>GenericAlignmentDataModel(String, String, boolean, boolean, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, boolean, int, boolean, boolean)"><B>GenericAlignmentDataModel(String, String, boolean, int, boolean, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#GenericAlignmentDataModel(java.lang.String, java.lang.String, boolean, int, boolean, boolean, java.lang.String)"><B>GenericAlignmentDataModel(String, String, boolean, int, boolean, boolean, String)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.EmptyCloseableIterator.html" title="class in broad.pda.seq.segmentation"><B>GenericAlignmentDataModel.EmptyCloseableIterator</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>An empty iterator to avoid returning null<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.EmptyCloseableIterator.html#GenericAlignmentDataModel.EmptyCloseableIterator()"><B>GenericAlignmentDataModel.EmptyCloseableIterator()</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.EmptyCloseableIterator.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel.EmptyCloseableIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.html" title="class in broad.core.datastructures"><B>GenericRootedTree</B></A>&lt;<A HREF="../broad/core/datastructures/GenericRootedTree.html" title="type parameter in GenericRootedTree">T</A>&gt; - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/GenericRootedTree.html#GenericRootedTree(java.lang.String, T)"><B>GenericRootedTree(String, T)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.html" title="class in broad.core.datastructures">GenericRootedTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html" title="class in broad.core.datastructures"><B>GenericRootedTree.TreeNode</B></A>&lt;<A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html" title="type parameter in GenericRootedTree.TreeNode">T</A>&gt; - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html#GenericRootedTree.TreeNode(broad.core.datastructures.GenericRootedTree, T)"><B>GenericRootedTree.TreeNode(GenericRootedTree&lt;T&gt;, T)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html" title="class in broad.core.datastructures">GenericRootedTree.TreeNode</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html#GenericRootedTree.TreeNode(broad.core.datastructures.GenericRootedTree, T, broad.core.datastructures.GenericRootedTree.TreeNode)"><B>GenericRootedTree.TreeNode(GenericRootedTree&lt;T&gt;, T, GenericRootedTree.TreeNode&lt;T&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html" title="class in broad.core.datastructures">GenericRootedTree.TreeNode</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation"><B>GeneScore</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#GeneScore(nextgen.core.annotation.Gene, double[])"><B>GeneScore(Gene, double[])</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>A class to represent the standard scores assigned to a window or Gene in the ContinuousDataAlignmentModel
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#GeneScore(nextgen.core.annotation.Gene)"><B>GeneScore(Gene)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa"><B>GeneSetEnrichment</B></A> - Class in <A HREF="../broad/projection/gsa/package-summary.html">broad.projection.gsa</A><DD>&nbsp;<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#GeneSetEnrichment(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, java.util.Map, boolean, boolean)"><B>GeneSetEnrichment(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, Map&lt;String, Collection&lt;String&gt;&gt;, boolean, boolean)</B></A> - 
Constructor for class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#GeneSetEnrichment(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, java.util.Map)"><B>GeneSetEnrichment(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, Map&lt;String, Collection&lt;String&gt;&gt;)</B></A> - 
Constructor for class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#GeneSetEnrichment(broad.core.datastructures.MatrixWithHeaders, broad.core.datastructures.MatrixWithHeaders, java.util.Map, boolean, boolean)"><B>GeneSetEnrichment(MatrixWithHeaders, MatrixWithHeaders, Map&lt;String, Collection&lt;String&gt;&gt;, boolean, boolean)</B></A> - 
Constructor for class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils"><B>GeneSetIntersect</B></A> - Class in <A HREF="../broad/core/overlaputils/package-summary.html">broad.core.overlaputils</A><DD>&nbsp;<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html#GeneSetIntersect()"><B>GeneSetIntersect()</B></A> - 
Constructor for class broad.core.overlaputils.<A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils">GeneSetIntersect</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#geneSpanContains(nextgen.core.annotation.Gene)"><B>geneSpanContains(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene"><B>GeneTools</B></A> - Class in <A HREF="../broad/pda/gene/package-summary.html">broad.pda.gene</A><DD>&nbsp;<DT><A HREF="../broad/pda/gene/GeneTools.html#GeneTools()"><B>GeneTools()</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature"><B>GeneWindow</B></A> - Class in <A HREF="../nextgen/core/feature/package-summary.html">nextgen.core.feature</A><DD>&nbsp;<DT><A HREF="../nextgen/core/feature/GeneWindow.html#GeneWindow(nextgen.core.annotation.Gene)"><B>GeneWindow(Gene)</B></A> - 
Constructor for class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html#GeneWindow(nextgen.core.feature.GeneWindow)"><B>GeneWindow(GeneWindow)</B></A> - 
Constructor for class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html#GeneWindow(nextgen.core.annotation.Annotation)"><B>GeneWindow(Annotation)</B></A> - 
Constructor for class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html#GeneWindow(java.util.Collection)"><B>GeneWindow(Collection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene"><B>GeneWithIsoforms</B></A> - Class in <A HREF="../broad/pda/gene/package-summary.html">broad.pda.gene</A><DD>&nbsp;<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(java.util.Collection)"><B>GeneWithIsoforms(Collection&lt;? extends LightweightGenomicAnnotation&gt;)</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(broad.core.annotation.LightweightGenomicAnnotation)"><B>GeneWithIsoforms(LightweightGenomicAnnotation)</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(java.lang.String)"><B>GeneWithIsoforms(String)</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(java.lang.String, int, int)"><B>GeneWithIsoforms(String, int, int)</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(java.lang.String, int, int, java.lang.String, nextgen.core.annotation.Annotation.Strand, java.util.Collection)"><B>GeneWithIsoforms(String, int, int, String, Annotation.Strand, Collection&lt;Annotation&gt;)</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD><B>Deprecated.</B>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(java.lang.String, int, int, java.lang.String, java.lang.String, java.util.List, java.util.List)"><B>GeneWithIsoforms(String, int, int, String, String, List&lt;Integer&gt;, List&lt;Integer&gt;)</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD><B>Deprecated.</B>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(java.lang.String, int, int, java.lang.String, java.lang.String, java.util.List, java.util.List, int, int)"><B>GeneWithIsoforms(String, int, int, String, String, List&lt;Integer&gt;, List&lt;Integer&gt;, int, int)</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD><B>Deprecated.</B>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(java.lang.String, int, int, java.lang.String, java.lang.String, java.util.List, java.util.List, int, int, java.lang.String[])"><B>GeneWithIsoforms(String, int, int, String, String, List&lt;Integer&gt;, List&lt;Integer&gt;, int, int, String[])</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD><B>Deprecated.</B>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(java.lang.String, int, int, java.lang.String, java.lang.String, java.util.List, java.util.List, java.lang.String[])"><B>GeneWithIsoforms(String, int, int, String, String, List&lt;Integer&gt;, List&lt;Integer&gt;, String[])</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD><B>Deprecated.</B>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#GeneWithIsoforms(nextgen.core.annotation.Gene)"><B>GeneWithIsoforms(Gene)</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/PairedEndTDF.html#GENOME"><B>GENOME</B></A> - 
Variable in class broad.pda.samtools.<A HREF="../broad/pda/samtools/PairedEndTDF.html" title="class in broad.pda.samtools">PairedEndTDF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/countreads/LibraryCompositionByRnaClass.html#GENOME_CLASS_NAME"><B>GENOME_CLASS_NAME</B></A> - 
Static variable in class broad.pda.countreads.<A HREF="../broad/pda/countreads/LibraryCompositionByRnaClass.html" title="class in broad.pda.countreads">LibraryCompositionByRnaClass</A>
<DD>The name used for the genome when counting read mappings
<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap"><B>GenomeCommandLineProgram</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html#GenomeCommandLineProgram()"><B>GenomeCommandLineProgram()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap">GenomeCommandLineProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap"><B>GenomeScoringProgram</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html#GenomeScoringProgram()"><B>GenomeScoringProgram()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap">GenomeScoringProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap"><B>GenomeScoringProgramFromBed</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#GenomeScoringProgramFromBed()"><B>GenomeScoringProgramFromBed()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/GenomeSpaceSampleData.html" title="class in broad.pda.seq.protection"><B>GenomeSpaceSampleData</B></A> - Class in <A HREF="../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/protection/GenomeSpaceSampleData.html#GenomeSpaceSampleData(java.lang.String, java.lang.String, java.util.Map, int, int, double)"><B>GenomeSpaceSampleData(String, String, Map&lt;String, Collection&lt;Gene&gt;&gt;, int, int, double)</B></A> - 
Constructor for class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/GenomeSpaceSampleData.html" title="class in broad.pda.seq.protection">GenomeSpaceSampleData</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GenomeWindow.html" title="class in nextgen.core.feature"><B>GenomeWindow</B></A> - Class in <A HREF="../nextgen/core/feature/package-summary.html">nextgen.core.feature</A><DD>&nbsp;<DT><A HREF="../nextgen/core/feature/GenomeWindow.html#GenomeWindow(java.lang.String, int, int)"><B>GenomeWindow(String, int, int)</B></A> - 
Constructor for class nextgen.core.feature.<A HREF="../nextgen/core/feature/GenomeWindow.html" title="class in nextgen.core.feature">GenomeWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GenomeWindow.html#GenomeWindow(nextgen.core.annotation.BasicAnnotation)"><B>GenomeWindow(BasicAnnotation)</B></A> - 
Constructor for class nextgen.core.feature.<A HREF="../nextgen/core/feature/GenomeWindow.html" title="class in nextgen.core.feature">GenomeWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html" title="class in broad.pda.seq.segmentation"><B>GenomeWithGaps</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD><B>Deprecated.</B>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html#GenomeWithGaps(java.util.Collection, java.lang.String, int, int, int)"><B>GenomeWithGaps(Collection&lt;Alignments&gt;, String, int, int, int)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html" title="class in broad.pda.seq.segmentation">GenomeWithGaps</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps2.html" title="class in broad.pda.seq.segmentation"><B>GenomeWithGaps2</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps2.html#GenomeWithGaps2(java.util.Collection, java.lang.String, int, java.lang.String)"><B>GenomeWithGaps2(Collection&lt;Annotation&gt;, String, int, String)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenomeWithGaps2.html" title="class in broad.pda.seq.segmentation">GenomeWithGaps2</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation"><B>GenomicAnnotation</B></A> - Interface in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/GenomicAnnotationFilter.html" title="interface in broad.core.annotation"><B>GenomicAnnotationFilter</B></A>&lt;<A HREF="../broad/core/annotation/GenomicAnnotationFilter.html" title="type parameter in GenomicAnnotationFilter">T</A> extends <A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>&gt; - Interface in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem"><B>GenomicSpace</B></A> - Class in <A HREF="../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A><DD>&nbsp;<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#GenomicSpace(java.util.Map)"><B>GenomicSpace(Map&lt;String, Integer&gt;)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#GenomicSpace(java.lang.String)"><B>GenomicSpace(String)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>Chromosome name and accompanying sizes
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#GenomicSpace(java.lang.String, java.lang.String)"><B>GenomicSpace(String, String)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#GenomicSpace(java.lang.String, java.lang.String, double)"><B>GenomicSpace(String, String, double)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/GenomicSpanFilter.html" title="class in nextgen.core.readFilters"><B>GenomicSpanFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/GenomicSpanFilter.html#GenomicSpanFilter(int)"><B>GenomicSpanFilter(int)</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/GenomicSpanFilter.html" title="class in nextgen.core.readFilters">GenomicSpanFilter</A>
<DD>Instantiate the filter
<DT><A HREF="../nextgen/core/annotation/Gene.html#genomicToGenomicPositionWithOffset(int, int)"><B>genomicToGenomicPositionWithOffset(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Get the genomic coordinate of the given position plus offset along the transcript
<DT><A HREF="../nextgen/core/annotation/Gene.html#genomicToTranscriptPosition(int)"><B>genomicToTranscriptPosition(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Maps genomic coordinates to cDNA coordinates
<DT><A HREF="../broad/core/math/Statistics.html#geometricMean(double[])"><B>geometricMean(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#geometricMeanMAD(Jama.Matrix)"><B>geometricMeanMAD(Matrix)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#get(int, int)"><B>get(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#get(java.lang.String, java.lang.String)"><B>get(String, String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#get(java.lang.String, int)"><B>get(String, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#get(int, java.lang.String)"><B>get(int, String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html#get(int, int)"><B>get(int, int)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html" title="class in broad.core.hmm">MarkovModel.BackwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html#get(int, int)"><B>get(int, int)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="class in broad.core.hmm">MarkovModel.ForwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html#get(java.lang.String)"><B>get(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser">ConfigFileParser.Bowtie2Options</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.RNAClass.html#get(java.lang.String)"><B>get(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.RNAClass.html" title="class in broad.core.parser">ConfigFileParser.RNAClass</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html#get(java.lang.String)"><B>get(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser">ConfigFileParser.TophatOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#get(java.lang.String)"><B>get(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#get(java.lang.String, java.lang.String)"><B>get(String, String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#get(java.lang.String)"><B>get(String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#get(int, int)"><B>get(int, int)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Get a single element.
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#get3PRegion(int, int)"><B>get3PRegion(int, int)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#get3Prime(java.lang.String, int)"><B>get3Prime(String, int)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#get3PrimeExon(int)"><B>get3PrimeExon(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#get3PrimeExon()"><B>get3PrimeExon()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#get3PrimeGenomicDistance(broad.core.annotation.LightweightGenomicAnnotation)"><B>get3PrimeGenomicDistance(LightweightGenomicAnnotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#get3Utr()"><B>get3Utr()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Get the set of exons comprising the 3' UTR
<DT><A HREF="../nextgen/core/annotation/Gene.html#get3UTR()"><B>get3UTR()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#get3UTRGene()"><B>get3UTRGene()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#get3UtrIncludingIntrons()"><B>get3UtrIncludingIntrons()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Get the region of the gene span that lies 3' of the stop codon, including introns but not the stop codon itself
<DT><A HREF="../nextgen/core/annotation/Gene.html#get5PrimeExon()"><B>get5PrimeExon()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#get5Utr()"><B>get5Utr()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Get the set of exons comprising the 5' UTR
<DT><A HREF="../nextgen/core/annotation/Gene.html#get5UTR()"><B>get5UTR()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#get5UTRGene()"><B>get5UTRGene()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#get5UtrIncludingIntrons()"><B>get5UtrIncludingIntrons()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Get the region of the gene span that lies 5' of start codon, including introns and the start codon itself
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#getA()"><B>getA()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html#getA()"><B>getA()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation">TwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/FDRDistribution.html#getAbsFDR(double)"><B>getAbsFDR(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/FDRDistribution.html" title="class in broad.core.math">FDRDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#getAbsFDRMatrix()"><B>getAbsFDRMatrix()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getAbsoluteEnd(int)"><B>getAbsoluteEnd(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getAbsoluteStart(int)"><B>getAbsoluteStart(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#getAEnd()"><B>getAEnd()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html#getAEnd()"><B>getAEnd()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation">TwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getAlignDir()"><B>getAlignDir()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getAlignedSequence(java.lang.String, boolean)"><B>getAlignedSequence(String, boolean)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getAlignedSequence(java.lang.String)"><B>getAlignedSequence(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#getAlignedSequence(java.lang.String)"><B>getAlignedSequence(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getAlignedSequence(java.lang.String)"><B>getAlignedSequence(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getAlignedSequenceIds()"><B>getAlignedSequenceIds()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getAlignedSequenceIds()"><B>getAlignedSequenceIds()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getAlignedSequences(boolean)"><B>getAlignedSequences(boolean)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getAlignedSequences()"><B>getAlignedSequences()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getAlignedSequences()"><B>getAlignedSequences()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getAlignment()"><B>getAlignment()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/PairedEndAlignmentFactory.html#getAlignment(boolean, nextgen.core.alignment.SingleEndAlignment, nextgen.core.alignment.SingleEndAlignment, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>getAlignment(boolean, SingleEndAlignment, SingleEndAlignment, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/PairedEndAlignmentFactory.html" title="class in nextgen.core.alignment">PairedEndAlignmentFactory</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getAlignment()"><B>getAlignment()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html#getAlignmentChunkSize()"><B>getAlignmentChunkSize()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy">ConservedPWMScanner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getAlignmentDataModelStats()"><B>getAlignmentDataModelStats()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getAlignmentEnd()"><B>getAlignmentEnd()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getAlignmentEnd()"><B>getAlignmentEnd()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns the end position of the alignment
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getAlignmentEnd()"><B>getAlignmentEnd()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getAlignmentGlobalStatsJar()"><B>getAlignmentGlobalStatsJar()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get the jar file for alignment model global stats
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getAlignmentIdxSizeAtCreation()"><B>getAlignmentIdxSizeAtCreation()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getAlignmentSizeAtCreation()"><B>getAlignmentSizeAtCreation()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getAlignmentStart()"><B>getAlignmentStart()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getAlignmentStart()"><B>getAlignmentStart()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns the start position of the alignment
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getAlignmentStart()"><B>getAlignmentStart()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getAll(java.lang.String)"><B>getAll(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getAllChromosomes()"><B>getAllChromosomes()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#getAllCommands()"><B>getAllCommands()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Returns all commands
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getAllCounts()"><B>getAllCounts()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the total paired end and spliced end counts for this edge.
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getAllDataValues()"><B>getAllDataValues()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>Get list of all the data values that contributed to the bins
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getAllGenes()"><B>getAllGenes()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture.IsoformMap</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html#getAllHybridizingPairs(float, float)"><B>getAllHybridizingPairs(float, float)</B></A> - 
Method in class nextgen.core.structure.<A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html" title="class in nextgen.core.structure">PairwiseWindowHybridization</A>
<DD>Get all pairs of windows that match in antisense direction
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#getAllIsoformExonSet()"><B>getAllIsoformExonSet()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#getAllIsoformIntronSet()"><B>getAllIsoformIntronSet()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getAllIsoforms(java.lang.String)"><B>getAllIsoforms(String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getAllIsoforms()"><B>getAllIsoforms()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#getAllIsoforms()"><B>getAllIsoforms()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#getAllIsoforms(boolean)"><B>getAllIsoforms(boolean)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>In cases were the containg gene is artificial, like when building gene loci and adding
 all isoforms contain in the loci.
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getAllMandatory(java.lang.String)"><B>getAllMandatory(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getAllMerged()"><B>getAllMerged()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getAllNonRandomChromosomes()"><B>getAllNonRandomChromosomes()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getAllOptionMaps()"><B>getAllOptionMaps()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get all options in one map
<DT><A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html#getAllOptions()"><B>getAllOptions()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser">ConfigFileParser.Bowtie2Options</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html#getAllOptions()"><B>getAllOptions()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html" title="class in broad.core.parser">ConfigFileParser.NovoalignOptions</A>
<DD>Get all flags and options
<DT><A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html#getAllOptions()"><B>getAllOptions()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser">ConfigFileParser.TophatOptions</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#getAllPairwiseAntisenseAlignments(float, float)"><B>getAllPairwiseAntisenseAlignments(float, float)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>Get all antisense direction alignments passing thresholds
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#getAllPairwiseSenseAlignments(float, float)"><B>getAllPairwiseSenseAlignments(float, float)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>Get all sense direction alignments passing thresholds
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getAllPaths()"><B>getAllPaths()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns the collection of Paths through the graph.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getAllPaths(double)"><B>getAllPaths(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns the collection of Paths through the graph with the minimum splice frequency
<DT><A HREF="../broad/core/primer3/isPCRLike.html#getAllPossibleAmplicons()"><B>getAllPossibleAmplicons()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/isPCRLike.html" title="class in broad.core.primer3">isPCRLike</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html#getAllPossibleTargets(int)"><B>getAllPossibleTargets(int)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer">SmatchLike</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/WigReader.html#getAllValues()"><B>getAllValues()</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/WigReader.html" title="class in nextgen.core.readers">WigReader</A>
<DD>Get the data as a map associating chromosome with position and value
 Chromosome coordinates are zero based (unlike the original wig file which is one based)
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getAlphabetSize()"><B>getAlphabetSize()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#getAlphabetSize()"><B>getAlphabetSize()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Loess.html#getAMap()"><B>getAMap()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/Loess.html" title="class in broad.core.math">Loess</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#getAName()"><B>getAName()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html#getAName()"><B>getAName()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation">TwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#getAnnotation(java.lang.String)"><B>getAnnotation(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getAnnotation(int, java.lang.String)"><B>getAnnotation(int, String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getAnnotation(java.lang.String)"><B>getAnnotation(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowScore.AbstractWindowScore.html#getAnnotation()"><B>getAnnotation()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowScore.AbstractWindowScore.html" title="class in nextgen.core.model.score">WindowScore.AbstractWindowScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowScore.html#getAnnotation()"><B>getAnnotation()</B></A> - 
Method in interface nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowScore.html" title="interface in nextgen.core.model.score">WindowScore</A>
<DD>Returns the underlying window
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getAnnotationDirectory()"><B>getAnnotationDirectory()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getAnnotationList(int)"><B>getAnnotationList(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getAnnotationList()"><B>getAnnotationList()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#getAnnotationList()"><B>getAnnotationList()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getAnnotationOverlappingRegion(nextgen.core.annotation.Annotation)"><B>getAnnotationOverlappingRegion(Annotation)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getAnnotationOverlappingRegion(nextgen.core.annotation.Annotation)"><B>getAnnotationOverlappingRegion(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getAnnotationOverlappingRegion(nextgen.core.annotation.Annotation)"><B>getAnnotationOverlappingRegion(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getAnnotationSetMap()"><B>getAnnotationSetMap()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getAnnotationsForSequence(java.lang.String)"><B>getAnnotationsForSequence(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getAnnotationsForSequence(java.lang.String, int)"><B>getAnnotationsForSequence(String, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#getAnnotationTree(java.lang.String)"><B>getAnnotationTree(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#getAnnotationTreeMap()"><B>getAnnotationTreeMap()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getAntisense()"><B>getAntisense()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#getArgString()"><B>getArgString()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Get the argument string from the command line
<DT><A HREF="../broad/core/parser/CommandLineParser.html#getArgString(java.util.Collection)"><B>getArgString(Collection&lt;String&gt;)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Get the argument string from the command line, possibly leaving out some arguments
<DT><A HREF="../Jama/Matrix.html#getArray()"><B>getArray()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Access the internal two-dimensional array.
<DT><A HREF="../Jama/Matrix.html#getArrayCopy()"><B>getArrayCopy()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Copy the internal two-dimensional array.
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#getAsChromosomeAnnotationMap()"><B>getAsChromosomeAnnotationMap()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getAsMatrix()"><B>getAsMatrix()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#getAStart()"><B>getAStart()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html#getAStart()"><B>getAStart()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation">TwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getAttribute(java.lang.String)"><B>getAttribute(String)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns the value of the specified attribute
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getAttribute(java.lang.String)"><B>getAttribute(String)</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns the value for the specified attribute
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getAttribute(java.lang.String)"><B>getAttribute(String)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getAttribute(java.lang.String)"><B>getAttribute(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#getAttributes()"><B>getAttributes()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getAttributes()"><B>getAttributes()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getAutosomeGenomeSize()"><B>getAutosomeGenomeSize()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getAutosomeNumber()"><B>getAutosomeNumber()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getAutosomes()"><B>getAutosomes()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#getAverageCoverage(nextgen.core.annotation.Annotation)"><B>getAverageCoverage(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getAverageCoverage(nextgen.core.annotation.Annotation)"><B>getAverageCoverage(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>Returns the average base-level coverage for the given region.
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#getAverageCoverage(nextgen.core.model.AlignmentModel)"><B>getAverageCoverage(AlignmentModel)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getAverageScore(broad.core.annotation.LightweightGenomicAnnotation)"><B>getAverageScore(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#getAvergeNumberOfReads()"><B>getAvergeNumberOfReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#getB()"><B>getB()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html#getB()"><B>getB()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation">TwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getBackgroundData()"><B>getBackgroundData()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get the background data alignment model
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getBackgroundName()"><B>getBackgroundName()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get background sample name
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#getBackgroundNucleotideFreqs()"><B>getBackgroundNucleotideFreqs()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getBackwardEdges(nextgen.core.annotation.Annotation, double, java.util.Collection)"><B>getBackwardEdges(Annotation, double, Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getBackwardPaths(int, int, double, java.util.Collection)"><B>getBackwardPaths(int, int, double, Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineAutomator.DGEInput.html#getBamFileFor(java.lang.String)"><B>getBamFileFor(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.DGEInput.html" title="class in broad.core.util">PipelineAutomator.DGEInput</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html#getBaseCompositionMetrics(java.io.File, java.io.File, java.io.File, boolean)"><B>getBaseCompositionMetrics(File, File, File, boolean)</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation">FilterSamByBaseComposition</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/MethylationUtils.html#getBaseCounts(java.lang.String)"><B>getBaseCounts(String)</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/MethylationUtils.html" title="class in broad.pda.methylation">MethylationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getBaseCoverage(java.util.List)"><B>getBaseCoverage(List&lt;T&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getBaseCoverage()"><B>getBaseCoverage()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/MethylationUtils.html#getBaseFrequencies(java.lang.String)"><B>getBaseFrequencies(String)</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/MethylationUtils.html" title="class in broad.pda.methylation">MethylationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getBaseIUB()"><B>getBaseIUB()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getBasesCovered(nextgen.core.annotation.Annotation)"><B>getBasesCovered(Annotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Adds a 0 fudge fact
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getBasesCovered(nextgen.core.annotation.Annotation, boolean)"><B>getBasesCovered(Annotation, boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getBasesCovered(nextgen.core.annotation.Annotation, int)"><B>getBasesCovered(Annotation, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getBasesCovered(nextgen.core.annotation.Annotation, int, boolean)"><B>getBasesCovered(Annotation, int, boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getBasesCovered(nextgen.core.annotation.Annotation, int)"><B>getBasesCovered(Annotation, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getBasesCovered(nextgen.core.annotation.Annotation)"><B>getBasesCovered(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getBasesCovered(nextgen.core.annotation.Annotation, int)"><B>getBasesCovered(Annotation, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getBasesCovered(nextgen.core.annotation.Annotation, int)"><B>getBasesCovered(Annotation, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#getBasesCovered(nextgen.core.annotation.Annotation)"><B>getBasesCovered(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getBasesCovered(nextgen.core.annotation.Annotation, boolean)"><B>getBasesCovered(Annotation, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>This method will return the total number of bases that are covered by alignments in the collection
 If discontinuous region, it will only count blocks
 The maximum possible value is defined by region.getTranscriptSize();
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getBasesCovered(nextgen.core.annotation.Annotation)"><B>getBasesCovered(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>This method will return the total number of bases that are covered by alignments in the collection
 If discontinuous region, it will only count blocks.
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getBasesCovered(nextgen.core.annotation.Annotation, boolean)"><B>getBasesCovered(Annotation, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getBasesCovered(nextgen.core.annotation.Annotation, boolean)"><B>getBasesCovered(Annotation, boolean)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getBasesCovered(nextgen.core.annotation.Annotation)"><B>getBasesCovered(Annotation)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getBasesCoveredPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getBasesCoveredPerAlignment(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getBasesCoveredPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getBasesCoveredPerAlignment(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#getBatch()"><B>getBatch()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#getBedFile()"><B>getBedFile()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getBedFileForTranscriptomeSpaceStats()"><B>getBedFileForTranscriptomeSpaceStats()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get bed annotation for transcriptome space stats
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getBedFileForWig()"><B>getBedFileForWig()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get bed file that determines which genes to count when writing fragment end positions to wig and bigwig files
<DT><A HREF="../nextgen/core/annotation/Gene.html#getBedScore()"><B>getBedScore()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#getBEnd()"><B>getBEnd()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html#getBEnd()"><B>getBEnd()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation">TwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getBin(double)"><B>getBin(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getBinEnd(int)"><B>getBinEnd(int)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getBinMidPoint(int)"><B>getBinMidPoint(int)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getBinNumber()"><B>getBinNumber()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getBinStart(int)"><B>getBinStart(int)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getBlastHit()"><B>getBlastHit()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getBlockIterator()"><B>getBlockIterator()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#getBlocks()"><B>getBlocks()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getBlocks()"><B>getBlocks()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>Returns a list of blocks if the annotations has any (@see containsBlocks)
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getBlocks()"><B>getBlocks()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getBlocks(boolean)"><B>getBlocks(boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getBlocks()"><B>getBlocks()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getBlocks(boolean)"><B>getBlocks(boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getBlocks()"><B>getBlocks()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>Returns unoriented blocks for this gene
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getBlocks(boolean)"><B>getBlocks(boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#getBlocks()"><B>getBlocks()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getBlocks()"><B>getBlocks()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getBlocks(boolean)"><B>getBlocks(boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#getBlockSizeStarts()"><B>getBlockSizeStarts()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#getBName()"><B>getBName()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html#getBName()"><B>getBName()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation">TwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#getBooleanArg(java.lang.String)"><B>getBooleanArg(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Get value of boolean parameter specified by flag
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getBowtie2BuildExecutablePath()"><B>getBowtie2BuildExecutablePath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get the path to bowtie2-build executable
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getBowtie2ExecutablePath()"><B>getBowtie2ExecutablePath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to bowtie2 executable file
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getBowtie2Options()"><B>getBowtie2Options()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get all bowtie2 flags and values
<DT><A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html#getBowtie2Path()"><B>getBowtie2Path()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser">ConfigFileParser.Bowtie2Options</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#getBStart()"><B>getBStart()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html#getBStart()"><B>getBStart()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation">TwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#getCapturedSequence()"><B>getCapturedSequence()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html#getCDNAs()"><B>getCDNAs()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html" title="class in broad.pda.seq.segmentation">GenomeWithGaps</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getCDS()"><B>getCDS()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getCDSEnd()"><B>getCDSEnd()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getCDSRegion()"><B>getCDSRegion()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getCDSStart()"><B>getCDSStart()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html#getCentroid(java.lang.String)"><B>getCentroid(String)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html" title="class in broad.pda.geneexpression.clustering">Kmeans.KmeansInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getChIPqPCRConfiguration()"><B>getChIPqPCRConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getChr()"><B>getChr()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getChr()"><B>getChr()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getChr()"><B>getChr()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Get chromosome
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getChr()"><B>getChr()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getChr()"><B>getChr()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getChr()"><B>getChr()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getChr()"><B>getChr()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getChrLength(java.lang.String)"><B>getChrLength(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getChrLengths()"><B>getChrLengths()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getChrNum()"><B>getChrNum()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Return the chromosome number without the chr
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#getChromosome()"><B>getChromosome()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getChromosome()"><B>getChromosome()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getChromosome()"><B>getChromosome()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getChromosome(java.lang.String)"><B>getChromosome(String)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getChromosome()"><B>getChromosome()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getChromosome()"><B>getChromosome()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns the chromosome for this alignment
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getChromosome()"><B>getChromosome()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns the chromosome for this alignment
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getChromosomeAnnotationMap()"><B>getChromosomeAnnotationMap()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getChromosomeAnnotationMap(int)"><B>getChromosomeAnnotationMap(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getChromosomeBEDs(java.lang.String)"><B>getChromosomeBEDs(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getChromosomeBEDs(java.lang.String, int)"><B>getChromosomeBEDs(String, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getChromosomeIterator()"><B>getChromosomeIterator()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getChromosomeIterator()"><B>getChromosomeIterator()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getChromosomeLength()"><B>getChromosomeLength()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getChromosomeLength(java.lang.String)"><B>getChromosomeLength(String)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>Get length of a chromosome
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getChromosomeLength(java.lang.String)"><B>getChromosomeLength(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getChromosomeLength(java.lang.String)"><B>getChromosomeLength(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getChromosomeLengths()"><B>getChromosomeLengths()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getChromosomeLengths()"><B>getChromosomeLengths()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getChromosomeLengths()"><B>getChromosomeLengths()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getChromosomeLengths()"><B>getChromosomeLengths()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getChromosomeNumber()"><B>getChromosomeNumber()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getChromosomeNumber(java.lang.String)"><B>getChromosomeNumber(String)</B></A> - 
Static method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getChromosomeReadIterator(java.lang.String)"><B>getChromosomeReadIterator(String)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getChromosomeReadIterator(java.lang.String)"><B>getChromosomeReadIterator(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#getChromosomes()"><B>getChromosomes()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getChromosomes()"><B>getChromosomes()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture.IsoformMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#getChromosomeString()"><B>getChromosomeString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getChromosomeString()"><B>getChromosomeString()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Returns the usual way to write a chromosome: chrSymbol if a chromosome or the full name of the scaffold
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getChromosomeString()"><B>getChromosomeString()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getChromosomeTree(java.lang.String, int)"><B>getChromosomeTree(String, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getChromosomeTree(java.lang.String)"><B>getChromosomeTree(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getChromosomeTreeMap(int)"><B>getChromosomeTreeMap(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getChromosomeTreeMap()"><B>getChromosomeTreeMap()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getChrReadIterator(java.lang.String)"><B>getChrReadIterator(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceUtils.html#getChrSequence(java.lang.String, java.lang.String)"><B>getChrSequence(String, String)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceUtils.html" title="class in broad.core.sequence">SequenceUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#getChrSequencesFromFasta(java.lang.String)"><B>getChrSequencesFromFasta(String)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getChrSizeFile()"><B>getChrSizeFile()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get chromosome size file
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getChrSizeFileForGenomicSpaceStats()"><B>getChrSizeFileForGenomicSpaceStats()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get chromosome size file for genome space stats
<DT><A HREF="../broad/core/sequence/SequenceUtils.html#getChrSizesFile(java.lang.String)"><B>getChrSizesFile(String)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceUtils.html" title="class in broad.core.sequence">SequenceUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getChrTree(java.lang.String)"><B>getChrTree(String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getChunkEnd()"><B>getChunkEnd()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getChunkEnd()"><B>getChunkEnd()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getChunkEnd()"><B>getChunkEnd()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getChunkStart()"><B>getChunkStart()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getChunkStart()"><B>getChunkStart()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getChunkStart()"><B>getChunkStart()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getCigarString()"><B>getCigarString()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#getClosest(broad.core.annotation.LightweightGenomicAnnotation)"><B>getClosest(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getClosest(nextgen.core.annotation.Annotation)"><B>getClosest(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getClosest(nextgen.core.annotation.Annotation, boolean)"><B>getClosest(Annotation, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getClosestDownstream(nextgen.core.annotation.Gene)"><B>getClosestDownstream(Gene)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getClosestDownstream(broad.core.annotation.LightweightGenomicAnnotation)"><B>getClosestDownstream(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getClosestDownstream(nextgen.core.annotation.Annotation)"><B>getClosestDownstream(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>Returns the closest non-overlapping annotation 3-prime of the query, considering strand.
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getClosestNonOverlapping(nextgen.core.annotation.Annotation)"><B>getClosestNonOverlapping(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getClosestUpstream(nextgen.core.annotation.Gene)"><B>getClosestUpstream(Gene)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getClosestUpstream(broad.core.annotation.LightweightGenomicAnnotation)"><B>getClosestUpstream(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getClosestUpstream(nextgen.core.annotation.Annotation)"><B>getClosestUpstream(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>Returns the closest non-overlapping annotation 5-prime of the query, considering strand.
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.html#getClusterInfo()"><B>getClusterInfo()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.html" title="class in broad.pda.geneexpression.clustering">Kmeans</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html#getClusterSampleAssignment()"><B>getClusterSampleAssignment()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html" title="class in broad.pda.geneexpression.clustering">Kmeans.KmeansInfo</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Cell.html#getCol()"><B>getCol()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Cell.html" title="class in jaligner">Cell</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getColoumnName(int)"><B>getColoumnName(int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getColumn(int)"><B>getColumn(int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getColumn(java.lang.String)"><B>getColumn(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getColumn(int)"><B>getColumn(int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>Assumes that encode has been called and each aligned sequence is encoded
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getColumn(int)"><B>getColumn(int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>Assumes that encode has been called and each aligned sequence is encoded
<DT><A HREF="../Jama/Matrix.html#getColumn(int)"><B>getColumn(int)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html#getColumnDimension()"><B>getColumnDimension()</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html" title="class in broad.core.hmm">MarkovModel.BackwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html#getColumnDimension()"><B>getColumnDimension()</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="class in broad.core.hmm">MarkovModel.ForwardResult</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#getColumnDimension()"><B>getColumnDimension()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Get column dimension.
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getColumnNames()"><B>getColumnNames()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getColumnNames()"><B>getColumnNames()</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#getColumnNum()"><B>getColumnNum()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#getColumnPackedCopy()"><B>getColumnPackedCopy()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Make a one-dimensional column packed copy of the internal array.
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getColumnsAsVector(int, int)"><B>getColumnsAsVector(int, int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getColumnsAsVector(int, int)"><B>getColumnsAsVector(int, int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getComaptibleBetweenAny2Sets(java.util.ArrayList, boolean)"><B>getComaptibleBetweenAny2Sets(ArrayList&lt;String&gt;, boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getComaptibleIsoforms(nextgen.core.annotation.Gene, boolean)"><B>getComaptibleIsoforms(Gene, boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/FisherCombinedProbabilityTest.html#getCombinedPvalue()"><B>getCombinedPvalue()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/FisherCombinedProbabilityTest.html" title="class in broad.core.math">FisherCombinedProbabilityTest</A>
<DD>Get the P value of the test statistic
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#getCommand(int)"><B>getCommand(int)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Returns the command at the specified index
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getComment()"><B>getComment()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getComplement()"><B>getComplement()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getConcatenatedSubAlignment(java.lang.String, java.util.List)"><B>getConcatenatedSubAlignment(String, List&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getConfigurationUsed()"><B>getConfigurationUsed()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getConnection()"><B>getConnection()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the connection vertex for this edge
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getConsensus()"><B>getConsensus()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#getContainedValues()"><B>getContainedValues()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getContainingSequenceId()"><B>getContainingSequenceId()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#getControlClass(java.io.File)"><B>getControlClass(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#getCoordinateAtPosition(int)"><B>getCoordinateAtPosition(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#getCoordinateSpace()"><B>getCoordinateSpace()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html#getCoordinateSpace()"><B>getCoordinateSpace()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RAPPeakCaller.html#getCoordinateSpace()"><B>getCoordinateSpace()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RAPPeakCaller.html" title="class in broad.pda.seq.rap">RAPPeakCaller</A>
<DD>Returns the coordinate space for this peak caller
<DT><A HREF="../nextgen/core/analysis/PeakCaller.html#getCoordinateSpace()"><B>getCoordinateSpace()</B></A> - 
Method in interface nextgen.core.analysis.<A HREF="../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A>
<DD>Get the coordinate space
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getCoordinateSpace()"><B>getCoordinateSpace()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getCoordinateSpace()"><B>getCoordinateSpace()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getCoordinateSpace()"><B>getCoordinateSpace()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#getCoordinateSpace()"><B>getCoordinateSpace()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/AMEResultReader.AMEMatch.html#getCorrectedPval()"><B>getCorrectedPval()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/AMEResultReader.AMEMatch.html" title="class in broad.core.motif.meme">AMEResultReader.AMEMatch</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getCount(java.util.Collection)"><B>getCount(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#getCount(nextgen.core.annotation.Annotation)"><B>getCount(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#getCount(nextgen.core.annotation.AnnotationList)"><B>getCount(AnnotationList&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#getCount(nextgen.core.annotation.AnnotationList, boolean)"><B>getCount(AnnotationList&lt;? extends Annotation&gt;, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getCount(nextgen.core.annotation.Annotation, boolean)"><B>getCount(Annotation, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>This method will return the number of reads overlapping a given region
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getCount(nextgen.core.annotation.Annotation)"><B>getCount(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>This method will return the number of reads overlapping a given region (not necessarily fully contained)
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getCount(nextgen.core.annotation.AnnotationList)"><B>getCount(AnnotationList&lt;? extends Annotation&gt;)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getCount(nextgen.core.annotation.AnnotationList, boolean)"><B>getCount(AnnotationList&lt;? extends Annotation&gt;, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getCount(nextgen.core.annotation.Annotation, boolean)"><B>getCount(Annotation, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.AlignmentCount.html#getCount()"><B>getCount()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.AlignmentCount.html" title="class in nextgen.core.model">AlignmentModel.AlignmentCount</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getCount(nextgen.core.annotation.Annotation, boolean)"><B>getCount(Annotation, boolean)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Use the coordinate space to decide what is an overlapping read
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getCount()"><B>getCount()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Use the coordinate space to decide what is an overlapping read
 Gets the count over the entire genome
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getCount(net.sf.samtools.util.CloseableIterator)"><B>getCount(CloseableIterator&lt;AlignmentModel.AlignmentCount&gt;)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getCount(nextgen.core.annotation.Annotation)"><B>getCount(Annotation)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#getCount()"><B>getCount()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#getCount()"><B>getCount()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getCountByBinStart()"><B>getCountByBinStart()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>Get the distribution as a map
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getCountExcludingRegion(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>getCountExcludingRegion(Annotation, Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>Compute the reads fully contained in region that are NOT overlapping excluded
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getCountExcludingRegion(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>getCountExcludingRegion(Annotation, Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getCountExcludingRegion(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>getCountExcludingRegion(Annotation, Annotation)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCounts(java.lang.String)"><B>getCounts(String)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>Counts the number of reads for the specified chromosome
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getCounts(java.util.Collection, int)"><B>getCounts(Collection&lt;Annotation&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getCounts(java.util.Map, java.util.Collection)"><B>getCounts(Map&lt;String, IntervalTree&lt;Annotation&gt;&gt;, Collection&lt;String&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCounts(java.lang.String)"><B>getCounts(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#getCountScore()"><B>getCountScore()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>Might be bad
<DT><A HREF="../nextgen/core/annotation/Gene.html#getCountScore()"><B>getCountScore()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsOfUniqueOverlappers(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int)"><B>getCountsOfUniqueOverlappers(Annotation, Annotation, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsOfUniqueOverlappers(nextgen.core.annotation.Annotation, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getCountsOfUniqueOverlappers(Annotation, Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getCountsOfUniqueOverlappers(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int)"><B>getCountsOfUniqueOverlappers(Annotation, Annotation, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsOfUniqueOverlappers(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int)"><B>getCountsOfUniqueOverlappers(Annotation, Annotation, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsOfUniqueOverlappers(nextgen.core.annotation.Annotation, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getCountsOfUniqueOverlappers(Annotation, Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsOnBase(java.lang.String, int)"><B>getCountsOnBase(String, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsOnBase(java.lang.String, int)"><B>getCountsOnBase(String, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsOnWindow(java.lang.String, int, int)"><B>getCountsOnWindow(String, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation, int)"><B>getCountsPerAlignment(Annotation, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Annotation, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation[], broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Annotation[], IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation[], int)"><B>getCountsPerAlignment(Annotation[], int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>Similar to @see getCountsPerAlignment(Annotation[] Annotation, IntervalTree<Alignment> tree, int EF)
 but delegates the management of the alignment interval tree to the implementing class rather than the caller
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation, java.util.Map, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Annotation, Map&lt;String, IntervalTree&lt;Annotation&gt;&gt;, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getCountsPerAlignment(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Annotation, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getCountsPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getCountsPerAlignment(nextgen.core.annotation.Annotation, java.util.Map, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Annotation, Map&lt;String, IntervalTree&lt;Annotation&gt;&gt;, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation, int)"><B>getCountsPerAlignment(Annotation, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation)"><B>getCountsPerAlignment(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Annotation, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation[], broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Annotation[], IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation[], int)"><B>getCountsPerAlignment(Annotation[], int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignment(nextgen.core.annotation.Annotation, java.util.Map, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignment(Annotation, Map&lt;String, IntervalTree&lt;Annotation&gt;&gt;, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignmentForPolyA(nextgen.core.annotation.Annotation, int)"><B>getCountsPerAlignmentForPolyA(Annotation, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAlignmentFullyContained(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignmentFullyContained(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignmentFullyContained(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getCountsPerAlignmentFullyContained(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignmentUncached(java.lang.String, int, int)"><B>getCountsPerAlignmentUncached(String, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAlignmentWithSameEndpointForPolyA(nextgen.core.annotation.Annotation, int, boolean, int, int)"><B>getCountsPerAlignmentWithSameEndpointForPolyA(Annotation, int, boolean, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsPerAnnotationtranded(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, java.lang.String)"><B>getCountsPerAnnotationtranded(Gene, IntervalTree&lt;Alignment&gt;, int, String)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsPerAnnotationtranded(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, java.lang.String)"><B>getCountsPerAnnotationtranded(Gene, IntervalTree&lt;Alignment&gt;, int, String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getCountsPerBp(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree)"><B>getCountsPerBp(Annotation, IntervalTree&lt;Alignment&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getCountsPerBp(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><B>getCountsPerBp(Gene, IntervalTree&lt;Alignment&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getCountsPerPosition(nextgen.core.annotation.Gene)"><B>getCountsPerPosition(Gene)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>This function returns an array list of counts for each position along the specified gene
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getCountStranded(nextgen.core.annotation.Annotation, boolean)"><B>getCountStranded(Annotation, boolean)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Use the coordinate space to decide what is an overlapping read using strandedness
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsWithinExons(nextgen.core.annotation.Annotation, java.util.Iterator, int)"><B>getCountsWithinExons(Annotation, Iterator&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getCountsWithinExons(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int)"><B>getCountsWithinExons(Annotation, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsWithinExons(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int)"><B>getCountsWithinExons(Annotation, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getCountsWithinExons(nextgen.core.annotation.Annotation, java.util.Iterator, int)"><B>getCountsWithinExons(Annotation, Iterator&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getCoverage()"><B>getCoverage()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getCummulativeProbability(double)"><B>getCummulativeProbability(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/AbstractFastaParser.html#getCurrentBase()"><B>getCurrentBase()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/AbstractFastaParser.html" title="class in broad.core.sequence">AbstractFastaParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityParser.html#getCurrentQuality()"><B>getCurrentQuality()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityParser.html" title="class in broad.core.sequence">FastaQualityParser</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/Commons.html#getCurrentRelease()"><B>getCurrentRelease()</B></A> - 
Static method in class jaligner.util.<A HREF="../jaligner/util/Commons.html" title="class in jaligner.util">Commons</A>
<DD>Returns the current release version of JAligner
<DT><A HREF="../broad/core/sequence/AbstractFastaParser.html#getCurrentSequenceId()"><B>getCurrentSequenceId()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/AbstractFastaParser.html" title="class in broad.core.sequence">AbstractFastaParser</A>
<DD>&nbsp;
<DT><A HREF="../Jama/EigenvalueDecomposition.html#getD()"><B>getD()</B></A> - 
Method in class Jama.<A HREF="../Jama/EigenvalueDecomposition.html" title="class in Jama">EigenvalueDecomposition</A>
<DD>Return the block diagonal eigenvalue matrix
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getData()"><B>getData()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getData()"><B>getData()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get the alignment data
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getData()"><B>getData()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getData()"><B>getData()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Get the underlying AlignmentDataModelStats object
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html#getDataAccess()"><B>getDataAccess()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getDataForAlignment(nextgen.core.annotation.Annotation)"><B>getDataForAlignment(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>This function returns an array of type double with counts for each base pair across the genomic region of the gene
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getDataForGene(nextgen.core.annotation.Gene)"><B>getDataForGene(Gene)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getDataForGene(java.util.Collection)"><B>getDataForGene(Collection&lt;Gene&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceUtils.html#getDataModel(java.lang.String, java.lang.String, java.lang.String, boolean)"><B>getDataModel(String, String, String, boolean)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceUtils.html" title="class in broad.core.sequence">SequenceUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceUtils.html#getDataModel(java.lang.String, java.lang.String, boolean)"><B>getDataModel(String, String, boolean)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceUtils.html" title="class in broad.core.sequence">SequenceUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceUtils.html#getDataModel(java.lang.String, java.lang.String, int, boolean, boolean, java.lang.String, boolean)"><B>getDataModel(String, String, int, boolean, boolean, String, boolean)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceUtils.html" title="class in broad.core.sequence">SequenceUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#getDataType1()"><B>getDataType1()</B></A> - 
Method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#getDataType2()"><B>getDataType2()</B></A> - 
Method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#getDataTypes()"><B>getDataTypes()</B></A> - 
Method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#getDefaultFile(java.lang.String)"><B>getDefaultFile(String)</B></A> - 
Static method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getDenominatorCount()"><B>getDenominatorCount()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getDenominatorCount()"><B>getDenominatorCount()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getDenominatorCount()"><B>getDenominatorCount()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getDenominatorLambda()"><B>getDenominatorLambda()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getDenominatorRegionTotal()"><B>getDenominatorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html#getDenominatorRegionTotal()"><B>getDenominatorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html" title="class in nextgen.core.model.score">NewRatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getDenominatorRegionTotal()"><B>getDenominatorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.Processor.html#getDenominatorRegionTotal()"><B>getDenominatorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.Processor.html" title="class in nextgen.core.model.score">RatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getDenominatorRegionTotal()"><B>getDenominatorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html#getDenominatorRegionTotal()"><B>getDenominatorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html" title="class in nextgen.core.model.score">WindowAllScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getDenominatorRPKM()"><B>getDenominatorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getDenominatorRPKM()"><B>getDenominatorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getDenominatorRPKM()"><B>getDenominatorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.html#getDenominatorScanPVal()"><B>getDenominatorScanPVal()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.html" title="class in broad.pda.seq.rap">SkellamScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getDenominatorScanPVal()"><B>getDenominatorScanPVal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getDenominatorTotal()"><B>getDenominatorTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getDenominatorTotal()"><B>getDenominatorTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getDenominatorTotal()"><B>getDenominatorTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getDensity(int)"><B>getDensity(int)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getDepletionBinomialPValue()"><B>getDepletionBinomialPValue()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>Binomial test: combine numerator and denominator counts, then calculate cumulative probability of seeing less than this proportion
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getDepletionBinomialPValue()"><B>getDepletionBinomialPValue()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>Binomial test: combine numerator and denominator counts, then calculate cumulative probability of seeing less than this proportion
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getDepletionBinomialPValue()"><B>getDepletionBinomialPValue()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>Binomial test: combine numerator and denominator counts, then calculate cumulative probability of seeing less than this proportion
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getDepletionBinomialScore()"><B>getDepletionBinomialScore()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>Binomial test: combine numerator and denominator counts, then calculate cumulative probability of seeing less than this proportion
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#getDescription()"><B>getDescription()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#getDescription()"><B>getDescription()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Returns the sequence description
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#getdgePath()"><B>getdgePath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#getdgeType()"><B>getdgeType()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/TracebackNode.html#getDirection()"><B>getDirection()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/TracebackNode.html" title="class in jaligner">TracebackNode</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getDistancesToClosest(broad.core.annotation.AnnotationReader)"><B>getDistancesToClosest(AnnotationReader&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#getDistanceTo(broad.core.annotation.LightweightGenomicAnnotation)"><B>getDistanceTo(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>public void
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getDistanceTo(broad.core.annotation.LightweightGenomicAnnotation)"><B>getDistanceTo(LightweightGenomicAnnotation)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Calculates the distance to the another genomic annotation.
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getDistanceTo(broad.core.annotation.LightweightGenomicAnnotation)"><B>getDistanceTo(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getDistanceTo(nextgen.core.annotation.Annotation)"><B>getDistanceTo(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getDistanceTo(nextgen.core.annotation.Annotation)"><B>getDistanceTo(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getDistanceTo(nextgen.core.annotation.Annotation)"><B>getDistanceTo(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Calculates the distance to the another genomic annotation.
<DT><A HREF="../nextgen/core/annotation/SingleInterval.html#getDistanceTo(nextgen.core.annotation.SingleInterval)"><B>getDistanceTo(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation">SingleInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getDistanceTo(nextgen.core.annotation.Annotation)"><B>getDistanceTo(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getDistanceTo3primeNeighbor(broad.pda.annotation.BEDFileParser)"><B>getDistanceTo3primeNeighbor(BEDFileParser)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getDistanceTo5primeNeighbor(broad.pda.annotation.BEDFileParser)"><B>getDistanceTo5primeNeighbor(BEDFileParser)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getDistanceToClosest(broad.core.annotation.LightweightGenomicAnnotation)"><B>getDistanceToClosest(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Jesse - June 13, 2012
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getDistanceToClosest(nextgen.core.annotation.Annotation)"><B>getDistanceToClosest(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#getDistanceToTarget()"><B>getDistanceToTarget()</B></A> - 
Method in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing">CRISPRDesigner.CRISPRTarget</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getDouble(java.lang.String)"><B>getDouble(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getDouble(java.lang.String, double)"><B>getDouble(String, double)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#getDoubleArg(java.lang.String)"><B>getDoubleArg(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Get value of double parameter specified by flag
<DT><A HREF="../Jama/LUDecomposition.html#getDoublePivot()"><B>getDoublePivot()</B></A> - 
Method in class Jama.<A HREF="../Jama/LUDecomposition.html" title="class in Jama">LUDecomposition</A>
<DD>Return pivot permutation vector as a one-dimensional double array
<DT><A HREF="../broad/pda/seq/segmentation/GeneCounts.html#getdoubleronicCounts()"><B>getdoubleronicCounts()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneCounts.html" title="class in broad.pda.seq.segmentation">GeneCounts</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getDoubles(java.lang.String)"><B>getDoubles(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getEdgeList()"><B>getEdgeList()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>Returns the edges making up the path.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getEdges(java.util.Collection)"><B>getEdges(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns all edges connecting the specified vertices
<DT><A HREF="../broad/pda/seq/graph/Path.html#getEdges()"><B>getEdges()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html#getElement()"><B>getElement()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html" title="class in broad.core.datastructures">GenericRootedTree.TreeNode</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PhastconsReader.html#getElements()"><B>getElements()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PhastconsReader.html" title="class in broad.core.siphy">PhastconsReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovState.html#getEmissionLogProbability(T)"><B>getEmissionLogProbability(T)</B></A> - 
Method in interface broad.core.hmm.<A HREF="../broad/core/hmm/MarkovState.html" title="interface in broad.core.hmm">MarkovState</A>
<DD>In many cases the implementing class may be able to implement this method so that the logarithm
 is not on every call.
<DT><A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html#getEmissionLogProbability(java.util.Map)"><B>getEmissionLogProbability(Map&lt;String, Matrix&gt;)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html" title="class in broad.core.siphy">PiHMM.ConstrainedModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.PiStateModel.html#getEmissionLogProbability(java.util.Map)"><B>getEmissionLogProbability(Map&lt;String, Matrix&gt;)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.PiStateModel.html" title="class in broad.core.siphy">PiHMM.PiStateModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovState.html#getEmissionProbability(T)"><B>getEmissionProbability(T)</B></A> - 
Method in interface broad.core.hmm.<A HREF="../broad/core/hmm/MarkovState.html" title="interface in broad.core.hmm">MarkovState</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html#getEmissionProbability(java.util.Map)"><B>getEmissionProbability(Map&lt;String, Matrix&gt;)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html" title="class in broad.core.siphy">PiHMM.ConstrainedModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.PiStateModel.html#getEmissionProbability(java.util.Map)"><B>getEmissionProbability(Map&lt;String, Matrix&gt;)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.PiStateModel.html" title="class in broad.core.siphy">PiHMM.PiStateModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html#getEmptyEmpiricalDistribution()"><B>getEmptyEmpiricalDistribution()</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html" title="class in broad.pda.seq.rap">BuildRatioNullDistribution</A>
<DD>Generate an empty empirical distribution.
<DT><A HREF="../broad/core/sequence/Sequence.html#getEncodedSequence()"><B>getEncodedSequence()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getEnd()"><B>getEnd()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#getEnd()"><B>getEnd()</B></A> - 
Method in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>Returns the ending point of the interval.
<DT><A HREF="../broad/core/datastructures/Interval.Impl.html#getEnd()"><B>getEnd()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.Impl.html" title="class in broad.core.datastructures">Interval.Impl</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#getEnd()"><B>getEnd()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getEnd()"><B>getEnd()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getEnd()"><B>getEnd()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getEnd()"><B>getEnd()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getEnd()"><B>getEnd()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>TODO: This may be to slow, maybe compute end on the fly as Path is being constructed
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getEnd()"><B>getEnd()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getEnd()"><B>getEnd()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Get end of the fragment
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getEnd()"><B>getEnd()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getEnd()"><B>getEnd()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#getEnd()"><B>getEnd()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getEnd()"><B>getEnd()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getEndVertex()"><B>getEndVertex()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>Returns the end vertex in the path.
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#getEnrichment()"><B>getEnrichment()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#getEnrichment(nextgen.core.model.AlignmentModel)"><B>getEnrichment(AlignmentModel)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestEnrichmentByFeature.html#getEnrichment(nextgen.core.annotation.Annotation)"><B>getEnrichment(Annotation)</B></A> - 
Method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestEnrichmentByFeature.html" title="class in nextgen.core.tests">TestEnrichmentByFeature</A>
<DD>Get the ratio of counts normalized by the global ratio
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getEnrichmentBinomialPValue()"><B>getEnrichmentBinomialPValue()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>Binomial test: combine numerator and denominator counts, then calculate cumulative probability of seeing >= this proportion
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getEnrichmentBinomialPValue()"><B>getEnrichmentBinomialPValue()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>Binomial test: combine numerator and denominator counts, then calculate cumulative probability of seeing >= this proportion
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getEnrichmentBinomialPValue()"><B>getEnrichmentBinomialPValue()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>Binomial test: combine numerator and denominator counts, then calculate cumulative probability of seeing >= this proportion
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getEnrichmentBinomialScore()"><B>getEnrichmentBinomialScore()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>Binomial test: combine numerator and denominator counts, then calculate cumulative probability of seeing >= this proportion
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getEnrichmentOverGene(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)"><B>getEnrichmentOverGene(Gene, Annotation)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get enrichment of a window over a gene
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getEntireChromosome(java.lang.String)"><B>getEntireChromosome(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>Get an annotation spanning an entire chromosome
<DT><A HREF="../broad/pda/countreads/FastqLibraryStats.html#getEstimatedLibrarySize()"><B>getEstimatedLibrarySize()</B></A> - 
Method in class broad.pda.countreads.<A HREF="../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads">FastqLibraryStats</A>
<DD>Get the estimated library size based on number of reads and percent duplicates
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#getEstimationWindow()"><B>getEstimationWindow()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#getEval()"><B>getEval()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getExactIsoform(broad.pda.gene.GeneWithIsoforms)"><B>getExactIsoform(GeneWithIsoforms)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#getExcludedRegions()"><B>getExcludedRegions()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getExonAnnotationOverlappingRegion(nextgen.core.annotation.Annotation)"><B>getExonAnnotationOverlappingRegion(Annotation)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getExonAnnotationOverlappingRegion(nextgen.core.annotation.Annotation)"><B>getExonAnnotationOverlappingRegion(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getExonAnnotationOverlappingRegion(nextgen.core.annotation.Annotation)"><B>getExonAnnotationOverlappingRegion(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getExonAnnotationOverlappingRegion(broad.core.datastructures.IntervalTree)"><B>getExonAnnotationOverlappingRegion(IntervalTree&lt;Alignment&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneCounts.html#getExonicCounts()"><B>getExonicCounts()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneCounts.html" title="class in broad.pda.seq.segmentation">GeneCounts</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getExons()"><B>getExons()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getExonSet()"><B>getExonSet()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getExonsScores()"><B>getExonsScores()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getExonTree()"><B>getExonTree()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getExonUnion()"><B>getExonUnion()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#getExperimentBarcode()"><B>getExperimentBarcode()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#getExperimentHybDate()"><B>getExperimentHybDate()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#getExperimentName()"><B>getExperimentName()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#getExperimentPlatePosition()"><B>getExperimentPlatePosition()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getExpressionScanPvalueCutoff()"><B>getExpressionScanPvalueCutoff()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get genome wide scan P value cutoff for expression of transcript
<DT><A HREF="../jaligner/Alignment.html#getExtend()"><B>getExtend()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getExtraFields()"><B>getExtraFields()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getExtraFields(int)"><B>getExtraFields(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#getExtraScore(int)"><B>getExtraScore(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getExtraScore(int)"><B>getExtraScore(int)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Returns the ith extra score
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getExtraScore(int)"><B>getExtraScore(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#getExtraScores()"><B>getExtraScores()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getExtraScores()"><B>getExtraScores()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getExtraScores()"><B>getExtraScores()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.html#getFactory(java.lang.String)"><B>getFactory(String)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#getFastaSeqId()"><B>getFastaSeqId()</B></A> - 
Method in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing">CRISPRDesigner.CRISPRTarget</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getFastqReadIdPairDelimiter()"><B>getFastqReadIdPairDelimiter()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get the delimiter for read pair number in fastq file
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getFastxDirectory()"><B>getFastxDirectory()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get directory containing fastx executables
<DT><A HREF="../broad/core/math/FDRDistribution.html#getFDR(double)"><B>getFDR(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/FDRDistribution.html" title="class in broad.core.math">FDRDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/RunDifferentialExpression.html#getFDR()"><B>getFDR()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/RunDifferentialExpression.html" title="class in broad.pda.differentialExpression">RunDifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#getFDRDistribution()"><B>getFDRDistribution()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#getFDRMatrix()"><B>getFDRMatrix()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#getFeature()"><B>getFeature()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/FeatureScore.html#getFeature()"><B>getFeature()</B></A> - 
Method in interface nextgen.core.feature.<A HREF="../nextgen/core/feature/FeatureScore.html" title="interface in nextgen.core.feature">FeatureScore</A>
<DD>Returns the underlying feature
<DT><A HREF="../broad/core/parser/StringParser.html#getFieldCount()"><B>getFieldCount()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Get number of fields
<DT><A HREF="../jaligner/util/Commons.html#getFileSeparator()"><B>getFileSeparator()</B></A> - 
Static method in class jaligner.util.<A HREF="../jaligner/util/Commons.html" title="class in jaligner.util">Commons</A>
<DD>Returns system file separator.
<DT><A HREF="../broad/core/sequence/AbstractFastaParser.html#getFileSize()"><B>getFileSize()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/AbstractFastaParser.html" title="class in broad.core.sequence">AbstractFastaParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Loess.html#getFinalMValues()"><B>getFinalMValues()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/Loess.html" title="class in broad.core.math">Loess</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getFinishingPrimerWalkConfiguration()"><B>getFinishingPrimerWalkConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>Returns the primer3 configuration used for
 finishing primer walks
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getFinPrimerWalkUnitTestConfiguration()"><B>getFinPrimerWalkUnitTestConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>Configuration used for running unit tests.
<DT><A HREF="../org/jgrapht/util/VertexPair.html#getFirst()"><B>getFirst()</B></A> - 
Method in class org.jgrapht.util.<A HREF="../org/jgrapht/util/VertexPair.html" title="class in org.jgrapht.util">VertexPair</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getFirstExon()"><B>getFirstExon()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getFirstFragmentPositionStranded()"><B>getFirstFragmentPositionStranded()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Get the beginning position of the fragment considering strand
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getFirstFragmentPositionStranded()"><B>getFirstFragmentPositionStranded()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Get the ending position of the read considering strand
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getFirstFragmentPositionStranded()"><B>getFirstFragmentPositionStranded()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Get the beginning position of the read considering strand
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html#getFirstGene()"><B>getFirstGene()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getFirstIntersectLeftFromMedian(broad.core.math.EmpiricalDistribution)"><B>getFirstIntersectLeftFromMedian(EmpiricalDistribution)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getFirstIntersectRightFromMedian(broad.core.math.EmpiricalDistribution)"><B>getFirstIntersectRightFromMedian(EmpiricalDistribution)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getFirstMate()"><B>getFirstMate()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Get the first read
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#getFirstNBPs(int)"><B>getFirstNBPs(int)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#getFirstSequence(java.lang.String)"><B>getFirstSequence(String)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#getFirstValue(java.lang.String)"><B>getFirstValue(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/StationaryDistributionIO.html#getFits()"><B>getFits()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/StationaryDistributionIO.html" title="class in broad.core.siphy">StationaryDistributionIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getFittedLogLikelihood()"><B>getFittedLogLikelihood()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#getFittedPi()"><B>getFittedPi()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#getFivePrimeBases()"><B>getFivePrimeBases()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getFivePrimeBases()"><B>getFivePrimeBases()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getFivePrimeBases()"><B>getFivePrimeBases()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#getFlagsAndValues()"><B>getFlagsAndValues()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Get the flags and values that were specified on the command line
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getFlankingBlocks(nextgen.core.annotation.Annotation)"><B>getFlankingBlocks(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>This function returns the blocks on either ends of a given splicejunction
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getFloat(java.lang.String)"><B>getFloat(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getFloat(java.lang.String, float)"><B>getFloat(String, float)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#getFloatArg(java.lang.String)"><B>getFloatArg(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Get value of float parameter specified by flag
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getFloats(java.lang.String)"><B>getFloats(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#getFoldMatrix()"><B>getFoldMatrix()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/formats/FormatFactory.html#getFormat(java.lang.String)"><B>getFormat(String)</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/FormatFactory.html" title="class in jaligner.formats">FormatFactory</A>
<DD>Returns an instance of <A HREF="../jaligner/formats/Format.html" title="class in jaligner.formats"><CODE>Format</CODE></A>.
<DT><A HREF="../jaligner/formats/FormatFactory.html#getFormats()"><B>getFormats()</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/FormatFactory.html" title="class in jaligner.formats">FormatFactory</A>
<DD>Returns a list of registered formats
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getForwardEdges(nextgen.core.annotation.Annotation, double, java.util.Collection)"><B>getForwardEdges(Annotation, double, Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html#getForwardTargetRegions(int)"><B>getForwardTargetRegions(int)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer">SmatchLike</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html#getForwardTargets(int)"><B>getForwardTargets(int)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer">SmatchLike</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getFragment(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getFragment(CoordinateSpace)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns an object for the fragment between the read pair in the specified coordinate space
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getFragment(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getFragment(CoordinateSpace)</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>This method returns the fragment formed by the alignment in the coordinate space C, specified.
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getFragment(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getFragment(CoordinateSpace)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns an object for the fragment from start to end of alignment in the specified coordinate space
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getFragment(java.lang.String, int, int)"><B>getFragment(String, int, int)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Returns a fragment of type Window between start and end on chromosome
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getFragment(nextgen.core.annotation.Annotation)"><B>getFragment(Annotation)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Returns a fragment of type Window between start and end on chromosome
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getFragment(java.lang.String, int, int)"><B>getFragment(String, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>Returns a fragment of type Window between start and end on chromosome
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getFragment(nextgen.core.annotation.Annotation)"><B>getFragment(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getFragment(java.lang.String, int, int)"><B>getFragment(String, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>Will return the continuous fragment
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getFragment(nextgen.core.annotation.Annotation)"><B>getFragment(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getFragment(java.lang.String, int, int)"><B>getFragment(String, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>Return a fragment between start and end on chr
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getFragment(nextgen.core.annotation.Annotation)"><B>getFragment(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getFragmentEnd()"><B>getFragmentEnd()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns the end of the fragment
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getFragmentEnd()"><B>getFragmentEnd()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns the end position of the alignment in genomic space.
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getFragmentEnd()"><B>getFragmentEnd()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns the end of the fragment
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getFragmentMidpoint(nextgen.core.annotation.Annotation)"><B>getFragmentMidpoint(Annotation)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getFragmentMidpoint(nextgen.core.annotation.Annotation)"><B>getFragmentMidpoint(Annotation)</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Get midpoint of fragment with respect to coordinate space
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getFragmentMidpoint(nextgen.core.annotation.Annotation)"><B>getFragmentMidpoint(Annotation)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getFragmentSize(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getFragmentSize(CoordinateSpace)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns the size of the fragment
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getFragmentSize(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getFragmentSize(CoordinateSpace)</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns the fragment size (insert+lengths of the reads)
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getFragmentSize(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getFragmentSize(CoordinateSpace)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns the size of the fragment
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getFragmentStart()"><B>getFragmentStart()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns the start of the fragment
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getFragmentStart()"><B>getFragmentStart()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns the start position of the alignment in genomic space.
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getFragmentStart()"><B>getFragmentStart()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns the start of the fragment
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getFragmentStrand()"><B>getFragmentStrand()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns the strand for the fragment, depending on the transcription read
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getFragmentStrand()"><B>getFragmentStrand()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns the strand for the fragment
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getFragmentStrand()"><B>getFragmentStrand()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns the strand for the fragment, which is same as read strand
<DT><A HREF="../broad/core/annotation/GFF.html#getFrame()"><B>getFrame()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getFullIntervalTreeAsAnnotation(java.lang.String)"><B>getFullIntervalTreeAsAnnotation(String)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getFullIntervalTreeAsAnnotation(java.lang.String)"><B>getFullIntervalTreeAsAnnotation(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#getFullName()"><B>getFullName()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#getFullSequence()"><B>getFullSequence()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#getFullSequence(boolean)"><B>getFullSequence(boolean)</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#getFullyContainedNumberOfReads()"><B>getFullyContainedNumberOfReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#getFullyContainedNumberOfReads()"><B>getFullyContainedNumberOfReads()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/FWERDistribution.html#getFWER(double)"><B>getFWER(double)</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/FWERDistribution.html" title="class in broad.pda.differentialExpression">FWERDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#getFWERMatrix()"><B>getFWERMatrix()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#getGapAdjustedCoordinate(int)"><B>getGapAdjustedCoordinate(int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>Maps coordinates relative to start of alignment (0 based) to the gapped adjusted string position in the aligned sequence string.
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#getGappedSeqsInWindowMatrix(int, java.util.Map, int)"><B>getGappedSeqsInWindowMatrix(int, Map&lt;String, Matrix&gt;, int)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getGappedSize()"><B>getGappedSize()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html#getGaps()"><B>getGaps()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html" title="class in broad.pda.seq.segmentation">GenomeWithGaps</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getGaps()"><B>getGaps()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getGaps1()"><B>getGaps1()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Returns the number of gaps of the aligned sequence #1
<DT><A HREF="../jaligner/Alignment.html#getGaps2()"><B>getGaps2()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Returns the number of gaps of the aligned sequence #2
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#getGapSizes()"><B>getGapSizes()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getGapsSize()"><B>getGapsSize()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/MethylationUtils.html#getGcPct(java.lang.String)"><B>getGcPct(String)</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/MethylationUtils.html" title="class in broad.pda.methylation">MethylationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/MethylationUtils.html#getGcRatio(java.lang.String)"><B>getGcRatio(String)</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/MethylationUtils.html" title="class in broad.pda.methylation">MethylationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getGene(java.lang.String, java.lang.String)"><B>getGene(String, String)</B></A> - 
Static method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getGeneAverageCoverage(nextgen.core.annotation.Gene)"><B>getGeneAverageCoverage(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get average coverage of gene
 Get score from cache or calculate and cache score
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getGeneCount(nextgen.core.annotation.Gene)"><B>getGeneCount(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get number of fragments mapping to gene
 Get score from cache or calculate and cache score
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getGeneCounts(nextgen.core.annotation.Gene, int)"><B>getGeneCounts(Gene, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getGeneCounts(java.util.Collection, int)"><B>getGeneCounts(Collection&lt;Gene&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneCounts(java.util.Collection)"><B>getGeneCounts(Collection&lt;Gene&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getGeneCounts(nextgen.core.annotation.Gene, int)"><B>getGeneCounts(Gene, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getGeneCountsWithoutDatasetTotals(java.util.Collection, int)"><B>getGeneCountsWithoutDatasetTotals(Collection&lt;Gene&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>Get read counts mapping to exons, introns, 5'UTRs and 3'UTRs in gene set of interest
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneCountsWithoutDatasetTotals(java.util.Collection)"><B>getGeneCountsWithoutDatasetTotals(Collection&lt;Gene&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#getGeneDescription()"><B>getGeneDescription()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getGeneExpression(java.util.Collection, int)"><B>getGeneExpression(Collection&lt;Gene&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneExpression(java.util.Collection)"><B>getGeneExpression(Collection&lt;Gene&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/GTFFileParser.html#getGeneIDIsoMap()"><B>getGeneIDIsoMap()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/GTFFileParser.html" title="class in broad.pda.annotation">GTFFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#getGeneLength()"><B>getGeneLength()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#getGeneName()"><B>getGeneName()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html#getGeneName()"><B>getGeneName()</B></A> - 
Method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html" title="class in broad.pda.rnai">RNAiGeneAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getGenePaths(double)"><B>getGenePaths(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of paired end edges
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getGenePaths(double, double)"><B>getGenePaths(double, double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborsNullModel.html#getGeneRandNeighbors(nextgen.core.annotation.Gene)"><B>getGeneRandNeighbors(Gene)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborsNullModel.html" title="class in broad.pda.gene">NeighborsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborsNullModel.html#getGeneRandNeighbors()"><B>getGeneRandNeighbors()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborsNullModel.html" title="class in broad.pda.gene">NeighborsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#GetGenes()"><B>GetGenes()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getGenes()"><B>getGenes()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get the gene set
<DT><A HREF="../broad/pda/seq/protection/GenomeSpaceSampleData.html#getGeneScanPval(nextgen.core.annotation.Gene)"><B>getGeneScanPval(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/GenomeSpaceSampleData.html" title="class in broad.pda.seq.protection">GenomeSpaceSampleData</A>
<DD>Get genome wide scan P value of number of fragments mapping to the gene
 Get score from cache or calculate and cache score
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getGeneScanPval(nextgen.core.annotation.Gene)"><B>getGeneScanPval(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get coordinate space wide scan P value of number of fragments mapping to the gene
 Get score from cache or calculate and cache score
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getGenesForChromosome(java.lang.String)"><B>getGenesForChromosome(String)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture.IsoformMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#getGenesPassingFDR(double)"><B>getGenesPassingFDR(double)</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/RunDifferentialExpression.html#getGenesPassingFDR(double)"><B>getGenesPassingFDR(double)</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/RunDifferentialExpression.html" title="class in broad.pda.differentialExpression">RunDifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#GetGenesWithIsoforms()"><B>GetGenesWithIsoforms()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getGenomeAssembly()"><B>getGenomeAssembly()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get name of genome assembly
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getGenomeBowtieIndex()"><B>getGenomeBowtieIndex()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to genome bowtie index
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getGenomeFasta()"><B>getGenomeFasta()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to genome fasta file
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getGenomeNovoindex()"><B>getGenomeNovoindex()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to genome novoindex
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#getGenomeSequence(broad.core.annotation.GenomicAnnotation, java.lang.String, boolean)"><B>getGenomeSequence(GenomicAnnotation, String, boolean)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>Suggestion:  Use classes in net.sf.picard.reference instead, which are WAY faster because they use an indexed FASTA file
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getGenomeSize()"><B>getGenomeSize()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getGenomicLength()"><B>getGenomicLength()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getGenomicRegionOverlappers(broad.core.annotation.LightweightGenomicAnnotation)"><B>getGenomicRegionOverlappers(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getGlobalCount()"><B>getGlobalCount()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getGlobalCount()"><B>getGlobalCount()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getGlobalCount()"><B>getGlobalCount()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getGlobalLambda()"><B>getGlobalLambda()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#getGlobalLambda()"><B>getGlobalLambda()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getGlobalLength()"><B>getGlobalLength()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#getGlobalLength()"><B>getGlobalLength()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getGlobalNumReads()"><B>getGlobalNumReads()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Get total number of reads mapping to all reference sequences
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getGlobalPairedFragments()"><B>getGlobalPairedFragments()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getGlobalRPKMConstant()"><B>getGlobalRPKMConstant()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getGlobalRPKMConstant()"><B>getGlobalRPKMConstant()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getGraph()"><B>getGraph()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>Returns the graph over which this path is defined.
<DT><A HREF="../broad/core/annotation/GFF.html#getGroup()"><B>getGroup()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#getGroupIndexes(java.util.ArrayList)"><B>getGroupIndexes(ArrayList&lt;String&gt;)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#getGroupIndexes(java.io.File)"><B>getGroupIndexes(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#getGTFAttr(java.lang.String, java.lang.String)"><B>getGTFAttr(String, String)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#getGTFAttr(java.lang.String[], java.lang.String)"><B>getGTFAttr(String[], String)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/GTFFileParser.html#getGTFGeneTranscriptMap()"><B>getGTFGeneTranscriptMap()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/GTFFileParser.html" title="class in broad.pda.annotation">GTFFileParser</A>
<DD>&nbsp;
<DT><A HREF="../Jama/QRDecomposition.html#getH()"><B>getH()</B></A> - 
Method in class Jama.<A HREF="../Jama/QRDecomposition.html" title="class in Jama">QRDecomposition</A>
<DD>Return the Householder vectors
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getHeader()"><B>getHeader()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#getHeader()"><B>getHeader()</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#getHeader()"><B>getHeader()</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getHistogram(int)"><B>getHistogram(int)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>Get the total number of observations in the bin
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#getId()"><B>getId()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getId()"><B>getId()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>Genomic annotation integer identification if such exists.
<DT><A HREF="../broad/core/sequence/Sequence.html#getId()"><B>getId()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getId()"><B>getId()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#getId()"><B>getId()</B></A> - 
Method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/formats/Format.html#getId()"><B>getId()</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/Format.html" title="class in jaligner.formats">Format</A>
<DD>Returns format id
<DT><A HREF="../jaligner/matrix/Matrix.html#getId()"><B>getId()</B></A> - 
Method in class jaligner.matrix.<A HREF="../jaligner/matrix/Matrix.html" title="class in jaligner.matrix">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#getId()"><B>getId()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Returns the sequence id
<DT><A HREF="../jaligner/Alignment.html#getIdentity()"><B>getIdentity()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getIgvtoolsPath()"><B>getIgvtoolsPath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to igvtools executable file
<DT><A HREF="../Jama/EigenvalueDecomposition.html#getImagEigenvalues()"><B>getImagEigenvalues()</B></A> - 
Method in class Jama.<A HREF="../Jama/EigenvalueDecomposition.html" title="class in Jama">EigenvalueDecomposition</A>
<DD>Return the imaginary parts of the eigenvalues
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getIncidentEdges(nextgen.core.annotation.Annotation)"><B>getIncidentEdges(Annotation)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of all incoming and outgoing edges of specified vertex
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#getIncludedRegions()"><B>getIncludedRegions()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getIndecesForRowDescription(java.lang.String)"><B>getIndecesForRowDescription(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#getIndex(broad.core.datastructures.Interval)"><B>getIndex(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the rank of the specified interval.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#getIndex(int, int)"><B>getIndex(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the rank of the specified interval.
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getIndex()"><B>getIndex()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/StringParser.html#getIndexFor(java.lang.String)"><B>getIndexFor(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Get position of the string in array
<DT><A HREF="../broad/core/util/ParseGCTFile.html#getIndexGroup(java.util.ArrayList)"><B>getIndexGroup(ArrayList&lt;String&gt;)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#getIndividualGeneNames()"><B>getIndividualGeneNames()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#getInfoAsString()"><B>getInfoAsString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getInformationContent()"><B>getInformationContent()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#getInformationContent()"><B>getInformationContent()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getInitialLogLikelihood()"><B>getInitialLogLikelihood()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html#getInnerBarcodeTotalReads(java.lang.String, java.lang.String)"><B>getInnerBarcodeTotalReads(String, String)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html" title="class in broad.pda.seq.fastq">FastqSplitterByBarcode.BarcodeInformation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getInOrUnorientedEdges(nextgen.core.annotation.Annotation)"><B>getInOrUnorientedEdges(Annotation)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of edges that terminate at the specified vertex (if oriented) 
 or have the specified vertex as an end point
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getInput()"><B>getInput()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getInputDir()"><B>getInputDir()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getInputReader()"><B>getInputReader()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getInputStream()"><B>getInputStream()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/formats/FormatFactory.html#getInstance()"><B>getInstance()</B></A> - 
Static method in class jaligner.formats.<A HREF="../jaligner/formats/FormatFactory.html" title="class in jaligner.formats">FormatFactory</A>
<DD>Returns an instance for <A HREF="../jaligner/formats/FormatFactory.html" title="class in jaligner.formats"><CODE>FormatFactory</CODE></A>.
<DT><A HREF="../broad/core/parser/CommandLineParser.html#getIntArg(java.lang.String)"><B>getIntArg(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Get value of int parameter specified by flag
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getInteger(java.lang.String)"><B>getInteger(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getInteger(java.lang.String, int)"><B>getInteger(String, int)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getIntegers(java.lang.String)"><B>getIntegers(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getIntervalBetweenReads()"><B>getIntervalBetweenReads()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getIntervalBetweenReads()"><B>getIntervalBetweenReads()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Get the coordinates of the first and last position of the interval between the reads
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getIntervalBetweenReads()"><B>getIntervalBetweenReads()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getIntervalTree(java.lang.String, int, int)"><B>getIntervalTree(String, int, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getIntervalTree(broad.core.annotation.LightweightGenomicAnnotation)"><B>getIntervalTree(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getIntervalTree(java.lang.String, int, int)"><B>getIntervalTree(String, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getIntervalTree(java.lang.String, int, int)"><B>getIntervalTree(String, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>public IntervalTree<Alignment> getIntervalTree(String chr, int start, int end){
                //System.err.print("Building tree from "+chr+" "+start+"-"+end + " - " );
                int counter=0;
                IntervalTree<Alignment> tree=new IntervalTree<Alignment>();
                CloseableIterator<Alignment> iter=getAnnotationOverlappingRegion(new Alignments(chr, start, end));
                while(iter.hasNext()){
                        Alignment record=iter.next();
                        Annotation t1=new Alignments(record.getChr(), record.getAnnotationtart(), record.getAlignmentEnd());
                        Annotation t2=new Alignments(chr, start, end);
                        if(t1.overlaps(t2)){
                                //System.err.println("Getting interval overlappers "+start+"-"+end+" "+record.getChr()+":"+record.getAnnotationtart()+"-"+record.getAlignmentEnd());

                                Node<Alignment> node=tree.find(record.getAnnotationtart(), record.getAlignmentEnd()+1);
                                if(record.getMappingQuality()>1){
                                        if(node!=null){node.incrementCount();}
                                        else{tree.put(record.getAnnotationtart(), record.getAlignmentEnd()+1, record);}
                                        counter++;
                                }
                        }
                        //else{System.err.println("WARN: Doesnt overlap queried: "+chr+":"+start+"-"+end+" got "+record.getChr()+":"+record.getAnnotationtart()+"-"+record.getAlignmentEnd());}
                }
                iter.close();
                //System.err.println(tree.size()+" "+counter);
                return tree;
        }
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getIntervalTreeCached(java.lang.String, int, int)"><B>getIntervalTreeCached(String, int, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getIntervalTreeCached(java.lang.String, int, int)"><B>getIntervalTreeCached(String, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getIntervalTreeCached(java.lang.String, int, int)"><B>getIntervalTreeCached(String, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getIntervalTreeTruncated(java.lang.String, int, int)"><B>getIntervalTreeTruncated(String, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getIntervalTreeTruncatedCached(java.lang.String, int, int)"><B>getIntervalTreeTruncatedCached(String, int, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getIntervalTreeTruncatedCached(java.lang.String, int, int)"><B>getIntervalTreeTruncatedCached(String, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getIntervalTreeTruncatedCached(java.lang.String, int, int)"><B>getIntervalTreeTruncatedCached(String, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getIntervalTreeWithIsoforoms()"><B>getIntervalTreeWithIsoforoms()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getIntronCounts(nextgen.core.annotation.Annotation)"><B>getIntronCounts(Annotation)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>This will count the number of spliced reads with exactly this intron
<DT><A HREF="../nextgen/core/annotation/Gene.html#getIntrons()"><B>getIntrons()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Get introns
<DT><A HREF="../nextgen/core/annotation/Gene.html#getIntronsBlocks()"><B>getIntronsBlocks()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getIntronSet()"><B>getIntronSet()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getIntronTranscript()"><B>getIntronTranscript()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getIsComaptibleBetweenAny2Sets(java.util.ArrayList, boolean)"><B>getIsComaptibleBetweenAny2Sets(ArrayList&lt;String&gt;, boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getIsComaptibleBetweenAny2Sets(java.util.ArrayList, broad.pda.annotation.BEDFileParser)"><B>getIsComaptibleBetweenAny2Sets(ArrayList&lt;String&gt;, BEDFileParser)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getIsComaptibleBetweenSubsets(java.util.ArrayList, java.util.ArrayList, boolean)"><B>getIsComaptibleBetweenSubsets(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getIsComaptibleBetweenSubsets(java.util.ArrayList, java.util.ArrayList, broad.pda.annotation.BEDFileParser)"><B>getIsComaptibleBetweenSubsets(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, BEDFileParser)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getIsComaptibleWithRefPerSet()"><B>getIsComaptibleWithRefPerSet()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getIsFirstMate()"><B>getIsFirstMate()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getIsoformMap(java.util.Collection)"><B>getIsoformMap(Collection&lt;Gene&gt;)</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html" title="class in nextgen.core.scripture">BuildScriptureCoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getIsoforms()"><B>getIsoforms()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getIsoformsForGene(nextgen.core.annotation.Gene)"><B>getIsoformsForGene(Gene)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture.IsoformMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getIsPartiallyComaptibleWithRefPerSet()"><B>getIsPartiallyComaptibleWithRefPerSet()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/Commons.html#getJAlignerInfo()"><B>getJAlignerInfo()</B></A> - 
Static method in class jaligner.util.<A HREF="../jaligner/util/Commons.html" title="class in jaligner.util">Commons</A>
<DD>Returns information about JAligner
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getJenRTPCRConfiguration()"><B>getJenRTPCRConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineUtils.html#getJobID()"><B>getJobID()</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/PipelineUtils.html" title="class in broad.core.util">PipelineUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#getJunctions()"><B>getJunctions()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getKLDivergence()"><B>getKLDivergence()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getKmerEndPosition()"><B>getKmerEndPosition()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getKmerSequence()"><B>getKmerSequence()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getKmerStartPosition()"><B>getKmerStartPosition()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#getKSEnrichments()"><B>getKSEnrichments()</B></A> - 
Method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#getKSFDR()"><B>getKSFDR()</B></A> - 
Method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GSAScoring.html#getKSScores()"><B>getKSScores()</B></A> - 
Method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GSAScoring.html" title="class in broad.projection.gsa">GSAScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getKSStatistic(java.util.List)"><B>getKSStatistic(List&lt;Double&gt;)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>Compared a list of observed with this empirical distribution using the Kolmogorov Smirnov Test.
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getKSStatistic(double[])"><B>getKSStatistic(double[])</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#getKSStatistic(double[], double[])"><B>getKSStatistic(double[], double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../Jama/CholeskyDecomposition.html#getL()"><B>getL()</B></A> - 
Method in class Jama.<A HREF="../Jama/CholeskyDecomposition.html" title="class in Jama">CholeskyDecomposition</A>
<DD>Return triangular factor.
<DT><A HREF="../Jama/LUDecomposition.html#getL()"><B>getL()</B></A> - 
Method in class Jama.<A HREF="../Jama/LUDecomposition.html" title="class in Jama">LUDecomposition</A>
<DD>Return lower triangular factor
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getLambda()"><B>getLambda()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getLambda()"><B>getLambda()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getLambda(java.lang.String)"><B>getLambda(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getLambda(java.lang.String)"><B>getLambda(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Get an estimate of the expected read coverage per base
<DT><A HREF="../nextgen/core/utils/AnnotationUtils.html#getLargestParent(T, java.util.Map)"><B>getLargestParent(T, Map&lt;String, Collection&lt;T&gt;&gt;)</B></A> - 
Static method in class nextgen.core.utils.<A HREF="../nextgen/core/utils/AnnotationUtils.html" title="class in nextgen.core.utils">AnnotationUtils</A>
<DD>Get largest parent
<DT><A HREF="../nextgen/core/annotation/Gene.html#getLastExon()"><B>getLastExon()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getLastExons(java.util.List)"><B>getLastExons(List&lt;? extends Annotation&gt;)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getLastFragmentPositionStranded()"><B>getLastFragmentPositionStranded()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Get the ending position of the fragment considering strand
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getLastFragmentPositionStranded()"><B>getLastFragmentPositionStranded()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Get the ending position of the read considering strand
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getLastFragmentPositionStranded()"><B>getLastFragmentPositionStranded()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Get the ending position of the read considering strand
<DT><A HREF="../nextgen/core/annotation/Gene.html#getLastIntron()"><B>getLastIntron()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getLastModifiedAlignmentIdxTimeStampAtCreation()"><B>getLastModifiedAlignmentIdxTimeStampAtCreation()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getLastModifiedAlignmentTimeStampAtCreation()"><B>getLastModifiedAlignmentTimeStampAtCreation()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getLastModifiedStoredTimeStamp()"><B>getLastModifiedStoredTimeStamp()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.PrimedRegion.html#getLastPrimerPair()"><B>getLastPrimerPair()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.PrimedRegion.html" title="class in broad.core.primer3">qPCRPrimerDesigner.PrimedRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.html#getLeaves()"><B>getLeaves()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.html" title="class in broad.core.datastructures">GenericRootedTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#getLeft()"><B>getLeft()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#getLeftCapture()"><B>getLeftCapture()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getLeftHighInfoStart()"><B>getLeftHighInfoStart()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborAnalysis.html#getLeftNeighbor(nextgen.core.annotation.Gene, broad.pda.annotation.BEDFileParser)"><B>getLeftNeighbor(Gene, BEDFileParser)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborAnalysis.html" title="class in broad.pda.gene">NeighborAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getLeftNode()"><B>getLeftNode()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the from-vertex for this edge.
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getLeftNode()"><B>getLeftNode()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getLeftPrimer()"><B>getLeftPrimer()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getLeftPrimerPenalty()"><B>getLeftPrimerPenalty()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getLeftPrimerPosition()"><B>getLeftPrimerPosition()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getLeftPrimerTM()"><B>getLeftPrimerTM()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#getLeftTm()"><B>getLeftTm()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#getLength()"><B>getLength()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getLength()"><B>getLength()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD><B>Deprecated.</B>&nbsp;<I>Use length() instead</I>
<DT><A HREF="../broad/core/datastructures/Interval.html#getLength()"><B>getLength()</B></A> - 
Method in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>End - start.
<DT><A HREF="../broad/core/datastructures/Interval.Impl.html#getLength()"><B>getLength()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.Impl.html" title="class in broad.core.datastructures">Interval.Impl</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#getLength()"><B>getLength()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#getLength()"><B>getLength()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getLength()"><B>getLength()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getLength()"><B>getLength()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#getLength()"><B>getLength()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#getLength()"><B>getLength()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getLength()"><B>getLength()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Returns the length of the alignment
<DT><A HREF="../jaligner/TracebackNode.html#getLength()"><B>getLength()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/TracebackNode.html" title="class in jaligner">TracebackNode</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getLength()"><B>getLength()</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Returns the total number of positions in the coordinate space
 For non-contiguous coordinate spaces, this will return a collapsed count such that overlapping blocks are not double counted
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getLength(java.lang.String)"><B>getLength(String)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Returns the total number of positions in the coordinate space
 For non-contiguous coordinate spaces, this will return a collapsed count such that overlapping blocks are not double counted
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getLength()"><B>getLength()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getLength(java.lang.String)"><B>getLength(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getLength()"><B>getLength()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>Always returns the gapped space length
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getLength(java.lang.String)"><B>getLength(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getLength()"><B>getLength()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getLength(java.lang.String)"><B>getLength(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getLengthOnReference()"><B>getLengthOnReference()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getLengthOnReference()"><B>getLengthOnReference()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getLengthOnReference()"><B>getLengthOnReference()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getLengthOnReference()"><B>getLengthOnReference()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html#getLevel()"><B>getLevel()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html" title="class in broad.core.datastructures">GenericRootedTree.TreeNode</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/Commons.html#getLineSeparator()"><B>getLineSeparator()</B></A> - 
Static method in class jaligner.util.<A HREF="../jaligner/util/Commons.html" title="class in jaligner.util">Commons</A>
<DD>Returns system line separator.
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#getLlr()"><B>getLlr()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getLocalLambda()"><B>getLocalLambda()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#getLocalLambda()"><B>getLocalLambda()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#getLocalLambda()"><B>getLocalLambda()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getLocalRate(nextgen.core.annotation.Annotation)"><B>getLocalRate(Annotation)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getLocalRate()"><B>getLocalRate()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getLocalRpkmConstant()"><B>getLocalRpkmConstant()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#getLocationString()"><B>getLocationString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getLocationString()"><B>getLocationString()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>Returns a string of the form chr<chromosome>:start-end
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getLocationString()"><B>getLocationString()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getLog2Ratio()"><B>getLog2Ratio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getLog2Ratio()"><B>getLog2Ratio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getLog2Ratio()"><B>getLog2Ratio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getLog2RegionRatio()"><B>getLog2RegionRatio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getLog2RegionRatio()"><B>getLog2RegionRatio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getLog2RegionRatio()"><B>getLog2RegionRatio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getLogger()"><B>getLogger()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get the logger
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getLogLikelihood(char[])"><B>getLogLikelihood(char[])</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getLogLikelihood(int[])"><B>getLogLikelihood(int[])</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getLogLikelihood(short[], int)"><B>getLogLikelihood(short[], int)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#getLogLikelihood(double, org.forester.phylogeny.Phylogeny, int, int)"><B>getLogLikelihood(double, Phylogeny, int, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#getLogLikelihoodRatio()"><B>getLogLikelihoodRatio()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/Fit.html#getLogLikelihoodRatio()"><B>getLogLikelihoodRatio()</B></A> - 
Method in interface broad.core.siphy.<A HREF="../broad/core/siphy/Fit.html" title="interface in broad.core.siphy">Fit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#getLogLikelihoodRatio()"><B>getLogLikelihoodRatio()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#getLogOdds()"><B>getLogOdds()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#getLogOddsPValue()"><B>getLogOddsPValue()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getLogOddsScore()"><B>getLogOddsScore()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getLogProb(char)"><B>getLogProb(char)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getLogProb(int)"><B>getLogProb(int)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html#getLogProbability()"><B>getLogProbability()</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="class in broad.core.hmm">MarkovModel.ForwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#getLogProcessedSignal()"><B>getLogProcessedSignal()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getLogRatio(char)"><B>getLogRatio(char)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getLogRatio(char, boolean)"><B>getLogRatio(char, boolean)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getLogRatioWeightedByNegEntropy(char)"><B>getLogRatioWeightedByNegEntropy(char)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getLogRatioWeightedByNegEntropy(char, boolean)"><B>getLogRatioWeightedByNegEntropy(char, boolean)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getLongestIntornChainIso(java.lang.String)"><B>getLongestIntornChainIso(String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getLongRangePCRConfiguration()"><B>getLongRangePCRConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>Returns the primer3 configuration used for
 short range PCR
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getMandatory(java.lang.String)"><B>getMandatory(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getMannWhitneyPValue()"><B>getMannWhitneyPValue()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Loess.html#getMap()"><B>getMap()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/Loess.html" title="class in broad.core.math">Loess</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getMappingQuality()"><B>getMappingQuality()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns the sum of the mapping qualities of both mates
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getMappingQuality()"><B>getMappingQuality()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns the mapping quality
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getMappingQuality()"><B>getMappingQuality()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns the mapping quality of the alignment
<DT><A HREF="../jaligner/Alignment.html#getMarkupLine()"><B>getMarkupLine()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getMaskedRegions()"><B>getMaskedRegions()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getMaskedRegionsSize(java.lang.String)"><B>getMaskedRegionsSize(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#getMatches()"><B>getMatches()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrixIO.html#getMatrices()"><B>getMatrices()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrixIO.html" title="class in broad.core.motif">PositionWeightMatrixIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html#getMatrix()"><B>getMatrix()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html" title="class in broad.pda.geneexpression.clustering">HierarchicalClustering</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getMatrix()"><B>getMatrix()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#getMatrix(int, int, int, int)"><B>getMatrix(int, int, int, int)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Get a submatrix.
<DT><A HREF="../Jama/Matrix.html#getMatrix(int[], int[])"><B>getMatrix(int[], int[])</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Get a submatrix.
<DT><A HREF="../Jama/Matrix.html#getMatrix(int, int, int[])"><B>getMatrix(int, int, int[])</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Get a submatrix.
<DT><A HREF="../Jama/Matrix.html#getMatrix(int[], int, int)"><B>getMatrix(int[], int, int)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Get a submatrix.
<DT><A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html#getMatrix(java.lang.String)"><B>getMatrix(String)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html" title="class in nextgen.core.normalize">ComparativeCountsMatrix</A>
<DD>Get matrix of count ratios
<DT><A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html#getMatrix(java.lang.String, java.lang.String)"><B>getMatrix(String, String)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html" title="class in nextgen.core.normalize">ComparativeCountsMatrix</A>
<DD>Get matrix of count ratios for one or all chromosomes
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getMax()"><B>getMax()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getMaxExonsCovered()"><B>getMaxExonsCovered()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#getMaxFragmentSize()"><B>getMaxFragmentSize()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#getMaxGenomicSpan()"><B>getMaxGenomicSpan()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#getMaxMeanEnrichments()"><B>getMaxMeanEnrichments()</B></A> - 
Method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#getMaxMeanFDR()"><B>getMaxMeanFDR()</B></A> - 
Method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GSAScoring.html#getMaxMeanScores()"><B>getMaxMeanScores()</B></A> - 
Method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GSAScoring.html" title="class in broad.projection.gsa">GSAScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getMaxPctCovered()"><B>getMaxPctCovered()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getMaxPctGenomeCovered()"><B>getMaxPctGenomeCovered()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getMaxScoreAcrossAllIso()"><B>getMaxScoreAcrossAllIso()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getMaxScorePerSet()"><B>getMaxScorePerSet()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getMean()"><B>getMean()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#getMean()"><B>getMean()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getMeanNullDensity()"><B>getMeanNullDensity()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getMeanOverAllRows()"><B>getMeanOverAllRows()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getMeanPeakDensity()"><B>getMeanPeakDensity()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#getMedian()"><B>getMedian()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getMedianNullDensity()"><B>getMedianNullDensity()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getMedianOfAllDataValues()"><B>getMedianOfAllDataValues()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>Get median of all data values including out of range
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getMedianOverAllRows()"><B>getMedianOverAllRows()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getMedianPeakDensity()"><B>getMedianPeakDensity()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#getMembers()"><B>getMembers()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getMerged()"><B>getMerged()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#getMerged()"><B>getMerged()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getMergedAnnotationMap(double)"><B>getMergedAnnotationMap(double)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getMergedAnnotationMap(boolean)"><B>getMergedAnnotationMap(boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getMergedCopy()"><B>getMergedCopy()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/BadModelException.html#getMessage()"><B>getMessage()</B></A> - 
Method in exception broad.core.hmm.<A HREF="../broad/core/hmm/BadModelException.html" title="class in broad.core.hmm">BadModelException</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#getMethod()"><B>getMethod()</B></A> - 
Method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#getMiddle()"><B>getMiddle()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getMiddle()"><B>getMiddle()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getMiddle()"><B>getMiddle()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#getMidPoint()"><B>getMidPoint()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getMidpoint()"><B>getMidpoint()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getMidpointGenomicCoords()"><B>getMidpointGenomicCoords()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getMin()"><B>getMin()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getMinimumMappingQuality()"><B>getMinimumMappingQuality()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getMinimumMappingQuality()"><B>getMinimumMappingQuality()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getMinimumMappingQuality()"><B>getMinimumMappingQuality()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#getMinimumTreeLength()"><B>getMinimumTreeLength()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getMirScore()"><B>getMirScore()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#getModel()"><B>getModel()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getModelFilePath()"><B>getModelFilePath()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getModelFilePath()"><B>getModelFilePath()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#getModelParameters()"><B>getModelParameters()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/AMEResultReader.AMEMatch.html#getMotif()"><B>getMotif()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/AMEResultReader.AMEMatch.html" title="class in broad.core.motif.meme">AMEResultReader.AMEMatch</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SequenceMotif.html#getMotif()"><B>getMotif()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotif.html" title="class in broad.core.motif">SequenceMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/misc/ComputeConservedMotifs.html#getMotifNames(java.lang.String)"><B>getMotifNames(String)</B></A> - 
Static method in class broad.core.motif.misc.<A HREF="../broad/core/motif/misc/ComputeConservedMotifs.html" title="class in broad.core.motif.misc">ComputeConservedMotifs</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeResultReader.html#getMotifs()"><B>getMotifs()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeResultReader.html" title="class in broad.core.motif.meme">MemeResultReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#getName()"><B>getName()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/Feature.html#getName()"><B>getName()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/Feature.html" title="interface in broad.core.annotation">Feature</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getName()"><B>getName()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.html#getName()"><B>getName()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.html" title="class in broad.core.datastructures">GenericRootedTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovState.html#getName()"><B>getName()</B></A> - 
Method in interface broad.core.hmm.<A HREF="../broad/core/hmm/MarkovState.html" title="interface in broad.core.hmm">MarkovState</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#getName()"><B>getName()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getName()"><B>getName()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/RestrictionEnzyme.html#getName()"><B>getName()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/RestrictionEnzyme.html" title="class in broad.core.primer3">RestrictionEnzyme</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getName()"><B>getName()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html#getName()"><B>getName()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html" title="class in broad.core.siphy">PiHMM.ConstrainedModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.PiStateModel.html#getName()"><B>getName()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.PiStateModel.html" title="class in broad.core.siphy">PiHMM.PiStateModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getName()"><B>getName()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/AbsolutePearsonFunction.html#getName()"><B>getName()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/AbsolutePearsonFunction.html" title="class in broad.pda.geneexpression.clustering">AbsolutePearsonFunction</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/ClusterDistanceFunction.html#getName()"><B>getName()</B></A> - 
Method in interface broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/ClusterDistanceFunction.html" title="interface in broad.pda.geneexpression.clustering">ClusterDistanceFunction</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/EuclideanDistance.html#getName()"><B>getName()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/EuclideanDistance.html" title="class in broad.pda.geneexpression.clustering">EuclideanDistance</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/PearsonDistance.html#getName()"><B>getName()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/PearsonDistance.html" title="class in broad.pda.geneexpression.clustering">PearsonDistance</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#getName()"><B>getName()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getName()"><B>getName()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getName()"><B>getName()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getName()"><B>getName()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getName()"><B>getName()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getName1()"><B>getName1()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getName2()"><B>getName2()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getNameGeneMap()"><B>getNameGeneMap()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getNanostringProbeClasses()"><B>getNanostringProbeClasses()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getNanostringReplicateMap()"><B>getNanostringReplicateMap()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborAnalysis.html#getNeighborDistance(nextgen.core.annotation.Gene)"><B>getNeighborDistance(Gene)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborAnalysis.html" title="class in broad.pda.gene">NeighborAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborAnalysis.html#getNeighbors(broad.pda.annotation.BEDFileParser, java.io.BufferedWriter, boolean, double)"><B>getNeighbors(BEDFileParser, BufferedWriter, boolean, double)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborAnalysis.html" title="class in broad.pda.gene">NeighborAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborAnalysis.html#getNeighbors(nextgen.core.annotation.Gene)"><B>getNeighbors(Gene)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborAnalysis.html" title="class in broad.pda.gene">NeighborAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#getNext()"><B>getNext()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/CombinationGenerator.html#getNext()"><B>getNext()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/CombinationGenerator.html" title="class in broad.core.math">CombinationGenerator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#getNextClosest(broad.core.annotation.LightweightGenomicAnnotation)"><B>getNextClosest(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>Return
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getNextKmer(int[])"><B>getNextKmer(int[])</B></A> - 
Static method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>Gets the next logical kmer.
<DT><A HREF="../broad/core/math/CombinationGenerator.html#getNextRandom()"><B>getNextRandom()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/CombinationGenerator.html" title="class in broad.core.math">CombinationGenerator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html#getNextWindow()"><B>getNextWindow()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html" title="class in broad.pda.seq.segmentation">GenomeWithGaps</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html#getNextWindow(int)"><B>getNextWindow(int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenomeWithGaps.html" title="class in broad.pda.seq.segmentation">GenomeWithGaps</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getNodeCount(nextgen.core.annotation.Annotation)"><B>getNodeCount(Annotation)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.html#getNodeElements(int)"><B>getNodeElements(int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.html" title="class in broad.core.datastructures">GenericRootedTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#getNodeLikelihoodParameterMap()"><B>getNodeLikelihoodParameterMap()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getNodePair(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><B>getNodePair(ChromosomeWithBubblesJGraphT.BubbleEdge)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns the pair of vertices for a specified edge
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#getNodePair(nextgen.core.scripture.OrientedChromosomeTranscriptGraph.TranscriptGraphEdge)"><B>getNodePair(OrientedChromosomeTranscriptGraph.TranscriptGraphEdge)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>Returns the pair of vertices for a specified edge
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.html#getNodes(int)"><B>getNodes(int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.html" title="class in broad.core.datastructures">GenericRootedTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#getNominalPValue()"><B>getNominalPValue()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#getNominalPValueMatrix()"><B>getNominalPValueMatrix()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNonControl1PercentileSignal()"><B>getNonControl1PercentileSignal()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNonControl50PercentileSignal()"><B>getNonControl50PercentileSignal()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNonControl99PercentileSignal()"><B>getNonControl99PercentileSignal()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getNonRandomTotalSize()"><B>getNonRandomTotalSize()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNormalizationFactor()"><B>getNormalizationFactor()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/CrossSampleRawCountNormalization.html#getNormalizedCount(nextgen.core.annotation.Annotation)"><B>getNormalizedCount(Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/CrossSampleRawCountNormalization.html" title="class in nextgen.core.normalize">CrossSampleRawCountNormalization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/CrossSampleTranscriptAverageNormalization.html#getNormalizedCount(nextgen.core.annotation.Annotation)"><B>getNormalizedCount(Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/CrossSampleTranscriptAverageNormalization.html" title="class in nextgen.core.normalize">CrossSampleTranscriptAverageNormalization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/NormalizedCount.html#getNormalizedCount(nextgen.core.annotation.Annotation)"><B>getNormalizedCount(Annotation)</B></A> - 
Method in interface nextgen.core.normalize.<A HREF="../nextgen/core/normalize/NormalizedCount.html" title="interface in nextgen.core.normalize">NormalizedCount</A>
<DD>Get normalized count over a region
<DT><A HREF="../nextgen/core/normalize/RawCounts.html#getNormalizedCount(nextgen.core.annotation.Annotation)"><B>getNormalizedCount(Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/RawCounts.html" title="class in nextgen.core.normalize">RawCounts</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html#getNormalizedCount(nextgen.core.annotation.Annotation)"><B>getNormalizedCount(Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html" title="class in nextgen.core.normalize">TranscriptAverageNormalization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/CrossSampleRawCountNormalization.html#getNormalizedCountsByPosition(nextgen.core.annotation.Annotation)"><B>getNormalizedCountsByPosition(Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/CrossSampleRawCountNormalization.html" title="class in nextgen.core.normalize">CrossSampleRawCountNormalization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/CrossSampleTranscriptAverageNormalization.html#getNormalizedCountsByPosition(nextgen.core.annotation.Annotation)"><B>getNormalizedCountsByPosition(Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/CrossSampleTranscriptAverageNormalization.html" title="class in nextgen.core.normalize">CrossSampleTranscriptAverageNormalization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/NormalizedCount.html#getNormalizedCountsByPosition(nextgen.core.annotation.Annotation)"><B>getNormalizedCountsByPosition(Annotation)</B></A> - 
Method in interface nextgen.core.normalize.<A HREF="../nextgen/core/normalize/NormalizedCount.html" title="interface in nextgen.core.normalize">NormalizedCount</A>
<DD>Get position level normalized counts across a region
<DT><A HREF="../nextgen/core/normalize/RawCounts.html#getNormalizedCountsByPosition(nextgen.core.annotation.Annotation)"><B>getNormalizedCountsByPosition(Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/RawCounts.html" title="class in nextgen.core.normalize">RawCounts</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html#getNormalizedCountsByPosition(nextgen.core.annotation.Annotation)"><B>getNormalizedCountsByPosition(Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html" title="class in nextgen.core.normalize">TranscriptAverageNormalization</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#getNormalizedKSEnrichments()"><B>getNormalizedKSEnrichments()</B></A> - 
Method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#getNormalizedMaxMeanEnrichments()"><B>getNormalizedMaxMeanEnrichments()</B></A> - 
Method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html#getNormalizedRegionAverage(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>getNormalizedRegionAverage(Annotation, Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html" title="class in nextgen.core.normalize">TranscriptAverageNormalization</A>
<DD>Get the average position level read coverage over the region, normalized to the average coverage over other region
<DT><A HREF="../nextgen/core/annotation/Gene.html#getNormalizedScore()"><B>getNormalizedScore()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#getNormalizedSignal(double, double, double)"><B>getNormalizedSignal(double, double, double)</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#getNormalizedSignal(double, double)"><B>getNormalizedSignal(double, double)</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getNovoalignOptions()"><B>getNovoalignOptions()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get all novoalign flags and values
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getNovoalignPath()"><B>getNovoalignPath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to novoalign executable
<DT><A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html#getNovoalignPath()"><B>getNovoalignPath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html" title="class in broad.core.parser">ConfigFileParser.NovoalignOptions</A>
<DD>Get the path to novoalign executable
<DT><A HREF="../broad/core/math/FisherCombinedProbabilityTest.html#getNullDistDegreesOfFreedom()"><B>getNullDistDegreesOfFreedom()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/FisherCombinedProbabilityTest.html" title="class in broad.core.math">FisherCombinedProbabilityTest</A>
<DD>Get the number of degrees of freedom of the null chi square distribution
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNum3PrimeComplementarity()"><B>getNum3PrimeComplementarity()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNum3PrimerStability()"><B>getNum3PrimerStability()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#getNumber()"><B>getNumber()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getNumberColumns()"><B>getNumberColumns()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getNumberMarkers(java.lang.String)"><B>getNumberMarkers(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getNumberMarkers(java.lang.String)"><B>getNumberMarkers(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Gets the number of allowable bases per chr
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getNumberOfAligningSequences()"><B>getNumberOfAligningSequences()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>Computes the number of sequences that have aligning bases as opposed to being pure gap
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNumberOfBackgroundNonUniformFeaturesOverlimit()"><B>getNumberOfBackgroundNonUniformFeaturesOverlimit()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNumberOfBackgroundPopulationOverlimit()"><B>getNumberOfBackgroundPopulationOverlimit()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getNumberOfBPs()"><B>getNumberOfBPs()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Get total number of bases
<DT><A HREF="../broad/pda/seq/graph/Path.html#getNumberOfEdges()"><B>getNumberOfEdges()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNumberOfFoundFeatures()"><B>getNumberOfFoundFeatures()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getNumberOfIsoforms(java.lang.String)"><B>getNumberOfIsoforms(String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getNumberOfIsoforms()"><B>getNumberOfIsoforms()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getNumberOfIsoforms(broad.core.datastructures.IntervalTree)"><B>getNumberOfIsoforms(IntervalTree&lt;GeneWithIsoforms&gt;)</B></A> - 
Static method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getNumberOfIsoforms()"><B>getNumberOfIsoforms()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#getNumberOfIsoforms()"><B>getNumberOfIsoforms()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#getNumberOfIterations()"><B>getNumberOfIterations()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getNumberOfMarkers()"><B>getNumberOfMarkers()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getNumberOfMatches()"><B>getNumberOfMatches()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getNumberOfMatches2()"><B>getNumberOfMatches2()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNumberOfNonUniformFeaturesOverlimit()"><B>getNumberOfNonUniformFeaturesOverlimit()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNumberOfPopulationOverlimit()"><B>getNumberOfPopulationOverlimit()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getNumberOfReads()"><B>getNumberOfReads()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getNumberOfReads()"><B>getNumberOfReads()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getNumberOfReads()"><B>getNumberOfReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getNumberOfReads(java.lang.String)"><B>getNumberOfReads(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getNumberOfReads(java.lang.String)"><B>getNumberOfReads(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Gets the total number of reads per chr
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#getNumberOfReads()"><B>getNumberOfReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getNumberOfReadsByChr(java.lang.String)"><B>getNumberOfReadsByChr(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getNumberOfSaturatedProbes()"><B>getNumberOfSaturatedProbes()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#getNumberOfSequences()"><B>getNumberOfSequences()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getNumberOfSplicedReads(nextgen.core.annotation.Annotation)"><B>getNumberOfSplicedReads(Annotation)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getNumberOfSplicedReads(nextgen.core.annotation.Annotation)"><B>getNumberOfSplicedReads(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getNumberOfSplicedReads()"><B>getNumberOfSplicedReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getNumberOfValuesLargerThan(double)"><B>getNumberOfValuesLargerThan(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getNumberPositions(int)"><B>getNumberPositions(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getNumberRows()"><B>getNumberRows()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#getNumBins()"><B>getNumBins()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getNumCol()"><B>getNumCol()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumColdMeltingTemp()"><B>getNumColdMeltingTemp()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#getNumColumns()"><B>getNumColumns()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#getNumDegenerateBCs()"><B>getNumDegenerateBCs()</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getNumEndPosToErase(double)"><B>getNumEndPosToErase(double)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getNumeratorCount()"><B>getNumeratorCount()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getNumeratorCount()"><B>getNumeratorCount()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getNumeratorCount()"><B>getNumeratorCount()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getNumeratorLambda()"><B>getNumeratorLambda()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getNumeratorRegionTotal()"><B>getNumeratorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html#getNumeratorRegionTotal()"><B>getNumeratorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html" title="class in nextgen.core.model.score">NewRatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getNumeratorRegionTotal()"><B>getNumeratorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.Processor.html#getNumeratorRegionTotal()"><B>getNumeratorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.Processor.html" title="class in nextgen.core.model.score">RatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getNumeratorRegionTotal()"><B>getNumeratorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html#getNumeratorRegionTotal()"><B>getNumeratorRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html" title="class in nextgen.core.model.score">WindowAllScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getNumeratorRPKM()"><B>getNumeratorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getNumeratorRPKM()"><B>getNumeratorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getNumeratorRPKM()"><B>getNumeratorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.html#getNumeratorScanPVal()"><B>getNumeratorScanPVal()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.html" title="class in broad.pda.seq.rap">SkellamScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getNumeratorScanPVal()"><B>getNumeratorScanPVal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getNumeratorTotal()"><B>getNumeratorTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getNumeratorTotal()"><B>getNumeratorTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getNumeratorTotal()"><B>getNumeratorTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getNumExons()"><B>getNumExons()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumGC()"><B>getNumGC()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumGCClampFailures()"><B>getNumGCClampFailures()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumHotMeltingTemp()"><B>getNumHotMeltingTemp()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getNumIsoformsPerSet()"><B>getNumIsoformsPerSet()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/CombinationGenerator.html#getNumLeft()"><B>getNumLeft()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/CombinationGenerator.html" title="class in broad.core.math">CombinationGenerator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#getNumMarkers()"><B>getNumMarkers()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumMispriming()"><B>getNumMispriming()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumNs()"><B>getNumNs()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#getNumOfIsoformsForGene(nextgen.core.annotation.Gene)"><B>getNumOfIsoformsForGene(Gene)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture.IsoformMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getNumOfIterations()"><B>getNumOfIterations()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumOk()"><B>getNumOk()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumOligosTested()"><B>getNumOligosTested()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumOverlappingExcludedRegions()"><B>getNumOverlappingExcludedRegions()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumOverlappingTargets()"><B>getNumOverlappingTargets()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumPolyX()"><B>getNumPolyX()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.AlignmentCount.html#getNumReads()"><B>getNumReads()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.AlignmentCount.html" title="class in nextgen.core.model">AlignmentModel.AlignmentCount</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#getNumReplicates()"><B>getNumReplicates()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#getNumRows()"><B>getNumRows()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumSelfComplementarityAnywhere()"><B>getNumSelfComplementarityAnywhere()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html#getNumSeqQuality()"><B>getNumSeqQuality()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3">Primer3PickingSummary</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getNumStartPosToErase(double)"><B>getNumStartPosToErase(double)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/countreads/FastqLibraryStats.html#getNumUniqueReads()"><B>getNumUniqueReads()</B></A> - 
Method in class broad.pda.countreads.<A HREF="../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads">FastqLibraryStats</A>
<DD>Get the number of unique reads
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html#getNumWindowsDone()"><B>getNumWindowsDone()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/AbstractFastaParser.html#getOffset()"><B>getOffset()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/AbstractFastaParser.html" title="class in broad.core.sequence">AbstractFastaParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getOmega()"><B>getOmega()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#getOmega()"><B>getOmega()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#getOmegaPValue()"><B>getOmegaPValue()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/PcrTailDesigner.html#getOneInitialPrimerPair(int)"><B>getOneInitialPrimerPair(int)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/PcrTailDesigner.html" title="class in broad.pda.capture.designer">PcrTailDesigner</A>
<DD>Get one primer pair
<DT><A HREF="../jaligner/Alignment.html#getOpen()"><B>getOpen()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getOpposite(nextgen.core.annotation.Annotation)"><B>getOpposite(Annotation)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the vertex opposite another vertex across an edge.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOppositeVertex(nextgen.core.annotation.Annotation, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><B>getOppositeVertex(Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns the opposite vertex of the edge e from the vertex v
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getOptimalPCRConfiguration()"><B>getOptimalPCRConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>Returns the primer3 configuration used for
 short range PCR
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#getOption(java.lang.String)"><B>getOption(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getOptionValueFor(java.lang.String)"><B>getOptionValueFor(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#getOrCreatePairedEndFile(java.lang.String, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>getOrCreatePairedEndFile(String, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Static method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>Finds a paired end file or creates one if it doesn't exist
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#getOrdered()"><B>getOrdered()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#getOrderedListCDT(java.io.File)"><B>getOrderedListCDT(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getOrientation()"><B>getOrientation()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getOrientation()"><B>getOrientation()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getOrientation()"><B>getOrientation()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getOrientation()"><B>getOrientation()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the orientation of the edge.
<DT><A HREF="../broad/pda/seq/graph/Path.html#getOrientation()"><B>getOrientation()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getOrientation()"><B>getOrientation()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getOrientation()"><B>getOrientation()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getOrientation()"><B>getOrientation()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getOrientation()"><B>getOrientation()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#getOrientedEnd()"><B>getOrientedEnd()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getOrientedEnd()"><B>getOrientedEnd()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>Gets the end of the annotation considering its orientation
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getOrientedEnd()"><B>getOrientedEnd()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getOrientedEnd()"><B>getOrientedEnd()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getOrientedEnd()"><B>getOrientedEnd()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getOrientedEnd()"><B>getOrientedEnd()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getOrientedLastExon()"><B>getOrientedLastExon()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#getOrientedStart()"><B>getOrientedStart()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getOrientedStart()"><B>getOrientedStart()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>Gets the start of the annotation considering its orientation
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getOrientedStart()"><B>getOrientedStart()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getOrientedStart()"><B>getOrientedStart()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getOrientedStart()"><B>getOrientedStart()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getOrientedStart()"><B>getOrientedStart()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getOriginalBamFile()"><B>getOriginalBamFile()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get name of original bam file
<DT><A HREF="../broad/core/math/Loess.html#getOriginalMValues()"><B>getOriginalMValues()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/Loess.html" title="class in broad.core.math">Loess</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getOriginalScore()"><B>getOriginalScore()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getOriginalSequence1()"><B>getOriginalSequence1()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Returns original <A HREF="../jaligner/Sequence.html" title="class in jaligner"><CODE>Sequence</CODE></A> #1
<DT><A HREF="../jaligner/Alignment.html#getOriginalSequence2()"><B>getOriginalSequence2()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Returns original <A HREF="../jaligner/Sequence.html" title="class in jaligner"><CODE>Sequence</CODE></A> #2
<DT><A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html#getOrphanGenes()"><B>getOrphanGenes()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">ChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOrphanNodes()"><B>getOrphanNodes()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns all orphan nodes in graph (Vertices without any edges orphan vertices)
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOrphanNodes(broad.pda.datastructures.Alignments)"><B>getOrphanNodes(Alignments)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns all orphan nodes in region (Vertices without any edges orphan vertices)
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOrphanPaths()"><B>getOrphanPaths()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns all orphan paths in graph (self edges for orphan vertices)
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#getOrphanPaths()"><B>getOrphanPaths()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>Returns all orphan paths in graph (self edges for orphan vertices)
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#getOrphanVertices()"><B>getOrphanVertices()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>Returns a collection of all orphan vertices.
<DT><A HREF="../org/jgrapht/util/VertexPair.html#getOther(V)"><B>getOther(V)</B></A> - 
Method in class org.jgrapht.util.<A HREF="../org/jgrapht/util/VertexPair.html" title="class in org.jgrapht.util">VertexPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html#getOuterBarcodeTotalReads(java.lang.String)"><B>getOuterBarcodeTotalReads(String)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html" title="class in broad.pda.seq.fastq">FastqSplitterByBarcode.BarcodeInformation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/PcrTailDesigner.html#getOuterPrimers()"><B>getOuterPrimers()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/PcrTailDesigner.html" title="class in broad.pda.capture.designer">PcrTailDesigner</A>
<DD>Get the full set of outer primer pairs
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOutOrUnorientedEdges(nextgen.core.annotation.Annotation)"><B>getOutOrUnorientedEdges(Annotation)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of edges that originate at the specified vertex (if oriented) 
 or have the specified vertex as an end point
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#getOutput()"><B>getOutput()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getOutput()"><B>getOutput()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getOutputDir()"><B>getOutputDir()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#getOutputFile()"><B>getOutputFile()</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getOutputStream()"><B>getOutputStream()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getOutputWriter()"><B>getOutputWriter()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#getOverlap(broad.core.annotation.LightweightGenomicAnnotation)"><B>getOverlap(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getOverlap(broad.core.annotation.LightweightGenomicAnnotation)"><B>getOverlap(LightweightGenomicAnnotation)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Returns the number of bases that overlap between the two annotations.
<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map)"><B>getOverlap(Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;)</B></A> - 
Static method in class broad.core.overlaputils.<A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils">GeneSetIntersect</A>
<DD>Get overlap between two gene sets
<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, boolean)"><B>getOverlap(Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;, boolean)</B></A> - 
Static method in class broad.core.overlaputils.<A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils">GeneSetIntersect</A>
<DD>Get overlap between two gene sets
<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map)"><B>getOverlap(Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;)</B></A> - 
Static method in class broad.core.overlaputils.<A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils">GeneSetIntersect</A>
<DD>Get overlap between two gene sets and automatically exclude all genes in a set
<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html#getOverlap(java.util.Map, java.util.Map, java.util.Map, boolean)"><B>getOverlap(Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;, boolean)</B></A> - 
Static method in class broad.core.overlaputils.<A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils">GeneSetIntersect</A>
<DD>Get overlap between two gene sets and automatically exclude all genes in a set
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getOverlap(broad.core.annotation.LightweightGenomicAnnotation)"><B>getOverlap(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getOverlap(nextgen.core.annotation.Annotation)"><B>getOverlap(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getOverlap(broad.pda.capture.designer.HairpinKmer)"><B>getOverlap(HairpinKmer)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getOverlap(nextgen.core.annotation.Annotation)"><B>getOverlap(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Returns the number of bases that overlap between the two annotations.
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getOverlap(nextgen.core.annotation.Annotation)"><B>getOverlap(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getOverlap(nextgen.core.annotation.Gene)"><B>getOverlap(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getOverlap(java.util.Collection)"><B>getOverlap(Collection&lt;Gene&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Get all overlaps between this gene and any gene in the collection
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getOverlap(nextgen.core.annotation.Annotation)"><B>getOverlap(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getOverlappers(nextgen.core.annotation.Annotation)"><B>getOverlappers(Annotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getOverlappers(java.util.List)"><B>getOverlappers(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Gets a list of overlapping annotations
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getOverlappers(java.util.List, boolean)"><B>getOverlappers(List&lt;? extends Annotation&gt;, boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getOverlappers(broad.core.annotation.AnnotationReader)"><B>getOverlappers(AnnotationReader&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getOverlappers(broad.core.annotation.LightweightGenomicAnnotation)"><B>getOverlappers(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Gets a list of overlapping annotations

        public Collection<T> getOverlappers(AnnotationReader<T> other) {
                Set<T> overlappers = new TreeSet<T>();
                Iterator<String> chrIt = other.getChromosomeIterator();
                while(chrIt.hasNext()) {
                        String chr = chrIt.next();
                        Iterator<T> annotationIt = other.getChromosomeTree(chr).valueIterator();
                        IntervalTree<T> tree = getChromosomeTree(chr);
                        if(tree == null) {
                                continue;
                        }
                        while(annotationIt.hasNext()) {
                                LightweightGenomicAnnotation annot = annotationIt.next();
                                Iterator<T> overlapperIt = new IntervalTree.ValuesIterator<T>(tree.overlappers(annot.getStart(), annot.getEnd()));
                                while(overlapperIt.hasNext()) {
                                        T overlapper = overlapperIt.next();
                                        overlappers.add(overlapper);
                                }
                        }
                }
                return overlappers;
        }
<DT><A HREF="../broad/core/siphy/StationaryDistributionIO.html#getOverlappers(broad.core.annotation.LightweightGenomicAnnotation)"><B>getOverlappers(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/StationaryDistributionIO.html" title="class in broad.core.siphy">StationaryDistributionIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getOverlappers(nextgen.core.annotation.Gene)"><B>getOverlappers(Gene)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getOverlappers(broad.core.annotation.LightweightGenomicAnnotation)"><B>getOverlappers(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#getOverlappers(broad.core.annotation.LightweightGenomicAnnotation)"><B>getOverlappers(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getOverlappingAnnotationList(nextgen.core.annotation.AnnotationList)"><B>getOverlappingAnnotationList(AnnotationList&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getOverlappingAnnotationList(nextgen.core.annotation.AnnotationList, boolean)"><B>getOverlappingAnnotationList(AnnotationList&lt;? extends Annotation&gt;, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#getOverlappingAnnotations(nextgen.core.annotation.Annotation)"><B>getOverlappingAnnotations(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getOverlappingAnnotations(nextgen.core.annotation.Annotation)"><B>getOverlappingAnnotations(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>This method will return a collection of the annotations that overlap a given region (not necessarily fully contained)
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getOverlappingAnnotations(nextgen.core.annotation.Annotation, boolean)"><B>getOverlappingAnnotations(Annotation, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>This method will return a collection of the annotations that overlap a given region
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getOverlappingAnnotations(nextgen.core.annotation.Annotation, boolean)"><B>getOverlappingAnnotations(Annotation, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getOverlappingAnnotations(nextgen.core.annotation.Annotation, boolean)"><B>getOverlappingAnnotations(Annotation, boolean)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getOverlappingGenes(nextgen.core.annotation.Gene, java.util.Collection, double)"><B>getOverlappingGenes(Gene, Collection&lt;Gene&gt;, double)</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html" title="class in nextgen.core.scripture">BuildScriptureCoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getOverlappingNodes(nextgen.core.alignment.Alignment)"><B>getOverlappingNodes(Alignment)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getOverlappingReadCountsStranded(nextgen.core.annotation.Annotation, boolean)"><B>getOverlappingReadCountsStranded(Annotation, boolean)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Return the reads that overlap with this region in coordinate space
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getOverlappingReadReplicateCounts(nextgen.core.annotation.Annotation, boolean)"><B>getOverlappingReadReplicateCounts(Annotation, boolean)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Get number of replicates of each read overlapping the region
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getOverlappingReads(java.lang.String)"><B>getOverlappingReads(String)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getOverlappingReads(nextgen.core.annotation.Annotation, boolean)"><B>getOverlappingReads(Annotation, boolean)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getOverlappingRegion(java.lang.String, int, int)"><B>getOverlappingRegion(String, int, int)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Returns a collection of regions overlapping start and end on chromosome
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getOverlappingRegion(java.lang.String, int, int)"><B>getOverlappingRegion(String, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>Returns an interval tree over the genomic region
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getOverlappingRegion(java.lang.String, int, int)"><B>getOverlappingRegion(String, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>Will return the overlapping regions in gapped space
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getOverlappingRegion(java.lang.String, int, int)"><B>getOverlappingRegion(String, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>Returns an interval tree over the specified transcriptome region
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getOverlappingRegions(java.lang.String)"><B>getOverlappingRegions(String)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getOverlappingRegions(java.lang.String)"><B>getOverlappingRegions(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getOverlappingSplicedReads(nextgen.core.annotation.Annotation, boolean)"><B>getOverlappingSplicedReads(Annotation, boolean)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Return the reads that overlap with this region in coordinate space
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getOverlappingSplicedReads(java.lang.String)"><B>getOverlappingSplicedReads(String)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Return the reads that overlap with this region in coordinate space
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getOverlappingVertices(int, int)"><B>getOverlappingVertices(int, int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a list of all vertices overlapping region between start and end
<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html#getOverlapWithRandomizedPositions(java.util.Map, java.util.Map)"><B>getOverlapWithRandomizedPositions(Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;)</B></A> - 
Static method in class broad.core.overlaputils.<A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils">GeneSetIntersect</A>
<DD>Intersect a set of genes with another set and randomize positions within each gene in the second set
<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html#getOverlapWithRandomizedPositions(java.util.Map, java.util.Map, java.util.Map)"><B>getOverlapWithRandomizedPositions(Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;)</B></A> - 
Static method in class broad.core.overlaputils.<A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils">GeneSetIntersect</A>
<DD>Intersect a set of genes with another set and randomize positions within each gene in the second set
<DT><A HREF="../broad/core/math/MannWhitneyPermutations.html#getP()"><B>getP()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/MannWhitneyPermutations.html" title="class in broad.core.math">MannWhitneyPermutations</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Wilcox2Sample.html#getP()"><B>getP()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/Wilcox2Sample.html" title="class in broad.core.math">Wilcox2Sample</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getPairedData()"><B>getPairedData()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getPairedEndCounts()"><B>getPairedEndCounts()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the total paired end counts for this edge.
<DT><A HREF="../broad/pda/seq/graph/Path.html#getPairedEndEdges()"><B>getPairedEndEdges()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#getPairedEndFile(java.lang.String)"><B>getPairedEndFile(String)</B></A> - 
Static method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>Returns the file itself or the file plus the default <code>PAIRED_END_EXTENSION</code> if either
 is in paired end format.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getPairedEndSupport()"><B>getPairedEndSupport()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the collection of paired end alignments for this edge
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#getParameters()"><B>getParameters()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.html#getParameters(java.lang.String[], java.lang.String, java.lang.String)"><B>getParameters(String[], String, String)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.html" title="class in broad.core.util">CLUtil</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.html#getParameters(java.lang.String[], java.lang.String)"><B>getParameters(String[], String)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.html" title="class in broad.core.util">CLUtil</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html#getParent()"><B>getParent()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html" title="class in broad.core.datastructures">GenericRootedTree.TreeNode</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/utils/AnnotationUtils.html#getParents(T, java.util.Map)"><B>getParents(T, Map&lt;String, Collection&lt;T&gt;&gt;)</B></A> - 
Static method in class nextgen.core.utils.<A HREF="../nextgen/core/utils/AnnotationUtils.html" title="class in nextgen.core.utils">AnnotationUtils</A>
<DD>Get parents of child annotation
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#getPass()"><B>getPass()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getPATConfiguration()"><B>getPATConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(double)"><B>getPaths(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of paths with alpha
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(double, double)"><B>getPaths(double, double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of paths with alpha and minimum spicing frequency
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, double, double, org.jgrapht.DirectedGraph)"><B>getPaths(Annotation, Annotation, double, double, DirectedGraph&lt;Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of all paths starting from startNode with alpha and minimum splicing frequency
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, double, java.util.Collection, double, org.jgrapht.DirectedGraph)"><B>getPaths(Annotation, Annotation, double, Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;, double, DirectedGraph&lt;Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of all paths starting from startNode having the specified valid edge types
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int)"><B>getPaths(int, int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>DO WE NEED THIS????
 Returns collection of genes between start and end
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int, boolean)"><B>getPaths(int, int, boolean)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int, double, boolean)"><B>getPaths(int, int, double, boolean)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getPaths(int, int, double, java.util.Collection)"><B>getPaths(int, int, double, Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#getPaths()"><B>getPaths()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html" title="class in nextgen.core.scripture">BuildScriptureCoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html#getPaths()"><B>getPaths()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">ChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#getPaths()"><B>getPaths()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../org/jgrapht/alg/KShortestPaths.html#getPaths(V)"><B>getPaths(V)</B></A> - 
Method in class org.jgrapht.alg.<A HREF="../org/jgrapht/alg/KShortestPaths.html" title="class in org.jgrapht.alg">KShortestPaths</A>
<DD>Returns the k shortest simple paths in increasing order of weight.
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getPeak(Gene, Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getPeak(Gene, Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><B>getPeak(Gene, Gene, IntervalTree&lt;Alignment&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene)"><B>getPeak(Gene, Gene)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getPeak(Gene, Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getPercent3UTR()"><B>getPercent3UTR()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getPercentCDS()"><B>getPercentCDS()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getPercentConsistent(broad.core.math.EmpiricalDistribution)"><B>getPercentConsistent(EmpiricalDistribution)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/countreads/FastqLibraryStats.html#getPercentDuplicated()"><B>getPercentDuplicated()</B></A> - 
Method in class broad.pda.countreads.<A HREF="../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads">FastqLibraryStats</A>
<DD>Get the proportion of reads that are duplicates
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getPercentFeaturesWellAboveBackground()"><B>getPercentFeaturesWellAboveBackground()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#getPercentGaps()"><B>getPercentGaps()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getPercentGC()"><B>getPercentGC()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getPercentIdentity()"><B>getPercentIdentity()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Percent identity
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getPercentOfValuesBetween(double, double)"><B>getPercentOfValuesBetween(double, double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getPercentOfValuesLargerThan(double)"><B>getPercentOfValuesLargerThan(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getPercentVariantsCovered()"><B>getPercentVariantsCovered()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborsNullModel.html#getPermNum()"><B>getPermNum()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborsNullModel.html" title="class in broad.pda.gene">NeighborsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#getPermutationMatrix()"><B>getPermutationMatrix()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/CombinationGenerator.html#getPermutations()"><B>getPermutations()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/CombinationGenerator.html" title="class in nextgen.core.scripture">CombinationGenerator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getPermutedAnnotations(nextgen.core.annotation.Annotation)"><B>getPermutedAnnotations(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>Shuffle the locations of annotations within the given region and return an iterator
 over the permuted annotations.
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getPermutedAnnotations(nextgen.core.annotation.Annotation)"><B>getPermutedAnnotations(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getPermutedAnnotations(nextgen.core.annotation.Annotation)"><B>getPermutedAnnotations(Annotation)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#getPi()"><B>getPi()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#getPI()"><B>getPI()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardCollectRnaSeqMetrics()"><B>getPicardCollectRnaSeqMetrics()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to alternative jar file for CollectRnaSeqMetrics
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardDirectory()"><B>getPicardDirectory()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get directory containing picard executables
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardRefFlat()"><B>getPicardRefFlat()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to ref_flat annotation
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardRibosomalIntervals()"><B>getPicardRibosomalIntervals()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to ribosomal intervals for picard
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getPicardStrandSpecificity()"><B>getPicardStrandSpecificity()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get strand specificity argument for picard
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getPIDToName()"><B>getPIDToName()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../Jama/LUDecomposition.html#getPivot()"><B>getPivot()</B></A> - 
Method in class Jama.<A HREF="../Jama/LUDecomposition.html" title="class in Jama">LUDecomposition</A>
<DD>Return pivot permutation vector
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#getPlateName()"><B>getPlateName()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html#getPoissonLambda(int, double, int)"><B>getPoissonLambda(int, double, int)</B></A> - 
Static method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamPeakCaller</A>
<DD>Given a total number of reads, a region length, and a window size, get the parameter of the Poisson distribution modeling the number of reads in a window of this size
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#getPosition()"><B>getPosition()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/Fit.html#getPosition()"><B>getPosition()</B></A> - 
Method in interface broad.core.siphy.<A HREF="../broad/core/siphy/Fit.html" title="interface in broad.core.siphy">Fit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#getPosition()"><B>getPosition()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#getPositionAtCoordinate(int)"><B>getPositionAtCoordinate(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getPositionAtReferenceCoordinate(int)"><B>getPositionAtReferenceCoordinate(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getPositionAtReferenceCoordinate(int, boolean)"><B>getPositionAtReferenceCoordinate(int, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getPositionAtReferenceCoordinate(int)"><B>getPositionAtReferenceCoordinate(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getPositionAtReferenceCoordinate(int)"><B>getPositionAtReferenceCoordinate(int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getPositionAtReferenceCoordinate(int, boolean)"><B>getPositionAtReferenceCoordinate(int, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getPositionAtReferenceCoordinate(int, boolean)"><B>getPositionAtReferenceCoordinate(int, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getPositionAtReferenceCoordinate(int)"><B>getPositionAtReferenceCoordinate(int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getPositionAtReferenceCoordinate(int, boolean)"><B>getPositionAtReferenceCoordinate(int, boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#getPositionCountList(nextgen.core.annotation.Gene)"><B>getPositionCountList(Gene)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>Get list of position level counts in mature transcript
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getPositionCountMap(nextgen.core.annotation.Gene, nextgen.core.model.AlignmentModel)"><B>getPositionCountMap(Gene, AlignmentModel)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>Get map of position to of position level counts in mature transcript
<DT><A HREF="../nextgen/core/scripture/Assembly.html#getPossiblePremature()"><B>getPossiblePremature()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html#getPosteriorLogProbability(int, int)"><B>getPosteriorLogProbability(int, int)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html" title="class in broad.core.hmm">MarkovModel.BackwardResult</A>
<DD>Computes the posterior probability of a given observation being emitted by a state
<DT><A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html#getPosteriorProbability(int, int)"><B>getPosteriorProbability(int, int)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html" title="class in broad.core.hmm">MarkovModel.BackwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/FastaMultipleAlignmentIO.html#getPreferredFileExtension()"><B>getPreferredFileExtension()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/FastaMultipleAlignmentIO.html" title="class in broad.core.multiplealignment">FastaMultipleAlignmentIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFIO.html#getPreferredFileExtension()"><B>getPreferredFileExtension()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFIO.html" title="class in broad.core.multiplealignment">MAFIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentIO.html#getPreferredFileExtension()"><B>getPreferredFileExtension()</B></A> - 
Method in interface broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignmentIO.html" title="interface in broad.core.multiplealignment">MultipleAlignmentIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/PhylipInterleavedMultipleAlignmentIO.html#getPreferredFileExtension()"><B>getPreferredFileExtension()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/PhylipInterleavedMultipleAlignmentIO.html" title="class in broad.core.multiplealignment">PhylipInterleavedMultipleAlignmentIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/PhylipSequencialMultipleAlignmentIO.html#getPreferredFileExtension()"><B>getPreferredFileExtension()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/PhylipSequencialMultipleAlignmentIO.html" title="class in broad.core.multiplealignment">PhylipSequencialMultipleAlignmentIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#getPrev()"><B>getPrev()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getPrimerFieldsAsString()"><B>getPrimerFieldsAsString()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getPrimerFieldsAsStringForConstructor()"><B>getPrimerFieldsAsStringForConstructor()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>Get the fields in the order accepted by the constructor
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#getPrimerLeftInput()"><B>getPrimerLeftInput()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#getPrimerNumReturn()"><B>getPrimerNumReturn()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getPrimerPairId()"><B>getPrimerPairId()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getPrimerPairInformation()"><B>getPrimerPairInformation()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>Get basic information about the primer pair in a tab-delimited string
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getPrimerPairInformationFieldNames()"><B>getPrimerPairInformationFieldNames()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>The names of the fields reported by getPrimerPairInformation()
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getPrimerPairPenalty()"><B>getPrimerPairPenalty()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPairReader.html#getPrimerPairs()"><B>getPrimerPairs()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPairReader.html" title="class in broad.core.primer3">PrimerPairReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#getPrimerPickingMode()"><B>getPrimerPickingMode()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PrimerPickingResult.html#getPrimerPickingSummary()"><B>getPrimerPickingSummary()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PrimerPickingResult.html" title="interface in broad.core.primer3">Primer3PrimerPickingResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#getPrimerRightInput()"><B>getPrimerRightInput()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PrimerPickingResult.html#getPrimers()"><B>getPrimers()</B></A> - 
Method in interface broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PrimerPickingResult.html" title="interface in broad.core.primer3">Primer3PrimerPickingResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.PrimedRegion.html#getPrimers()"><B>getPrimers()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.PrimedRegion.html" title="class in broad.core.primer3">qPCRPrimerDesigner.PrimedRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/DesignPCRTails.html#getPrimers()"><B>getPrimers()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/DesignPCRTails.html" title="class in broad.pda.capture.designer">DesignPCRTails</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#getPrimerSequenceId()"><B>getPrimerSequenceId()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getPrimerSequences()"><B>getPrimerSequences()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.PrimedRegion.html#getPrimerSequences()"><B>getPrimerSequences()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.PrimedRegion.html" title="class in broad.core.primer3">qPCRPrimerDesigner.PrimedRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#getPrimerStartCodonPosition()"><B>getPrimerStartCodonPosition()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#getPriorClosest(broad.core.annotation.LightweightGenomicAnnotation)"><B>getPriorClosest(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html#getProbability()"><B>getProbability()</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="class in broad.core.hmm">MarkovModel.ForwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getProbability(double)"><B>getProbability(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getProbability(int)"><B>getProbability(int)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#getProbe()"><B>getProbe()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#getProbename()"><B>getProbename()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#getProcessedSignal()"><B>getProcessedSignal()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getProductSequence()"><B>getProductSequence()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getProductSize()"><B>getProductSize()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#getProgram()"><B>getProgram()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getPromoter(int)"><B>getPromoter(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getPromoter(int, int)"><B>getPromoter(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getPromoters()"><B>getPromoters()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#getPromoters(int, int)"><B>getPromoters(int, int)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#getProperPairsOnly()"><B>getProperPairsOnly()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/AMEResultReader.AMEMatch.html#getPval()"><B>getPval()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/AMEResultReader.AMEMatch.html" title="class in broad.core.motif.meme">AMEResultReader.AMEMatch</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#getPVal(broad.core.siphy.EvolutionaryModel.OmegaFit, boolean)"><B>getPVal(EvolutionaryModel.OmegaFit, boolean)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getPVal()"><B>getPVal()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getPValue(double, double)"><B>getPValue(double, double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/MannWhitney.html#getPvalue()"><B>getPvalue()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/MannWhitney.html" title="class in broad.core.math">MannWhitney</A>
<DD>Get P value
<DT><A HREF="../broad/core/motif/meme/MemeMotif.MatchedSequence.html#getPvalue()"><B>getPvalue()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.MatchedSequence.html" title="class in broad.core.motif.meme">MemeMotif.MatchedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#getPValue()"><B>getPValue()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#getPValue()"><B>getPValue()</B></A> - 
Method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getPvalue()"><B>getPvalue()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns P-value for this edge
<DT><A HREF="../nextgen/core/annotation/Gene.html#getPValue()"><B>getPValue()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getPValue2(double)"><B>getPValue2(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>P(x>Score)
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getPvalueCutoffScan()"><B>getPvalueCutoffScan()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get the p value cutoff for scan statistic
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getPvalueCutoffSkellam()"><B>getPvalueCutoffSkellam()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get the p value cutoff for skellam statistic
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#getPWM()"><B>getPWM()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html#getPWMiterator()"><B>getPWMiterator()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy">ConservedPWMScanner</A>
<DD>&nbsp;
<DT><A HREF="../Jama/QRDecomposition.html#getQ()"><B>getQ()</B></A> - 
Method in class Jama.<A HREF="../Jama/QRDecomposition.html" title="class in Jama">QRDecomposition</A>
<DD>Generate and return the (economy-sized) orthogonal factor
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getQpcrConfiguration()"><B>getQpcrConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>The Primer3 configuration settings used for RNA qPCR
 Added by Pam Russell May 11 2012
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getqRTPCRConfiguration()"><B>getqRTPCRConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityBinaryIO.html#getQualityBases(java.sql.Connection, java.lang.String, java.lang.String, int, int)"><B>getQualityBases(Connection, String, String, int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityBinaryIO.html" title="class in broad.core.sequence">FastaQualityBinaryIO</A>
<DD>Retrieves quality base calls for a given sequence from a binary quality assembly file.
<DT><A HREF="../broad/core/sequence/FastaQualityBinaryIO.html#getQualityBases(java.sql.Connection, java.lang.String, java.lang.String)"><B>getQualityBases(Connection, String, String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityBinaryIO.html" title="class in broad.core.sequence">FastaQualityBinaryIO</A>
<DD>Retrieves quality base calls for a given sequence from a binary quality assembly file.
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getQuantileOfBins(double)"><B>getQuantileOfBins(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getQueueName()"><B>getQueueName()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>&nbsp;
<DT><A HREF="../Jama/QRDecomposition.html#getR()"><B>getR()</B></A> - 
Method in class Jama.<A HREF="../Jama/QRDecomposition.html" title="class in Jama">QRDecomposition</A>
<DD>Return the upper triangular factor
<DT><A HREF="../broad/pda/gene/transcriptsNullModel.html#getRandomSets()"><B>getRandomSets()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/transcriptsNullModel.html" title="class in broad.pda.gene">transcriptsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GeneSetEnrichment.html#getRankedList()"><B>getRankedList()</B></A> - 
Method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GeneSetEnrichment.html" title="class in broad.projection.gsa">GeneSetEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getRAPqPCRConfiguration()"><B>getRAPqPCRConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#getRateMatrix()"><B>getRateMatrix()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getRatio()"><B>getRatio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getRatio()"><B>getRatio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getRatio()"><B>getRatio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getRawDensity(int)"><B>getRawDensity(int)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html#getRawRegionAverage(nextgen.core.annotation.Annotation)"><B>getRawRegionAverage(Annotation)</B></A> - 
Method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html" title="class in nextgen.core.normalize">TranscriptAverageNormalization</A>
<DD>Get the average position level read coverage over the region
<DT><A HREF="../nextgen/core/model/AlignmentModel.AlignmentCount.html#getRead()"><B>getRead()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.AlignmentCount.html" title="class in nextgen.core.model">AlignmentModel.AlignmentCount</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getReadAlignmentBlocks(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getReadAlignmentBlocks(CoordinateSpace)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getReadAlignmentBlocks(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getReadAlignmentBlocks(CoordinateSpace)</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>This method returns the alignment blocks that the read actually aligns to.
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getReadAlignmentBlocks(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getReadAlignmentBlocks(CoordinateSpace)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns the alignment Blocks
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getReadAlignments(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getReadAlignments(CoordinateSpace)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getReadAlignments(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getReadAlignments(CoordinateSpace)</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>This will return the actual read alignments
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getReadAlignments(nextgen.core.coordinatesystem.CoordinateSpace)"><B>getReadAlignments(CoordinateSpace)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#getReader()"><B>getReader()</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getReadIterator()"><B>getReadIterator()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getReadIterator(nextgen.core.annotation.Annotation)"><B>getReadIterator(Annotation)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getReadIterator()"><B>getReadIterator()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getReadIterator(nextgen.core.annotation.Annotation)"><B>getReadIterator(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getReadIterator()"><B>getReadIterator()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getReadIterator(nextgen.core.annotation.Annotation)"><B>getReadIterator(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getReadIterator()"><B>getReadIterator()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getReadIterator(nextgen.core.annotation.Annotation)"><B>getReadIterator(Annotation)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getReadLength()"><B>getReadLength()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns the length of the read
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getReadName()"><B>getReadName()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns the name of this read pair.
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getReadName()"><B>getReadName()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>This method returns the name of the alignment
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getReadName()"><B>getReadName()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns the name of this read pair.
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getReadSequence()"><B>getReadSequence()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getReadSequence()"><B>getReadSequence()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getReadSequence()"><B>getReadSequence()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getReadSizeDistribution(nextgen.core.coordinatesystem.CoordinateSpace, int, int)"><B>getReadSizeDistribution(CoordinateSpace, int, int)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Get the distribution of all read (or fragment) sizes with respect to a coordinate space
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getReadSizeDistribution(nextgen.core.annotation.Annotation, nextgen.core.coordinatesystem.CoordinateSpace, int, int)"><B>getReadSizeDistribution(Annotation, CoordinateSpace, int, int)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Get the distribution of read (or fragment) sizes in an annotation
<DT><A HREF="../Jama/EigenvalueDecomposition.html#getRealEigenvalues()"><B>getRealEigenvalues()</B></A> - 
Method in class Jama.<A HREF="../Jama/EigenvalueDecomposition.html" title="class in Jama">EigenvalueDecomposition</A>
<DD>Return the real parts of the eigenvalues
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getReducedRepresentation(int, java.util.List)"><B>getReducedRepresentation(int, List&lt;GenomicAnnotation&gt;)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getReference()"><B>getReference()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getReferenceAnnotation(java.lang.String)"><B>getReferenceAnnotation(String)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getReferenceAnnotation(java.lang.String)"><B>getReferenceAnnotation(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getReferenceAnnotation(java.lang.String)"><B>getReferenceAnnotation(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getReferenceAnnotation(java.lang.String)"><B>getReferenceAnnotation(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getReferenceAnnotations()"><B>getReferenceAnnotations()</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getReferenceAnnotations()"><B>getReferenceAnnotations()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getReferenceAnnotations()"><B>getReferenceAnnotations()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getReferenceAnnotations()"><B>getReferenceAnnotations()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getReferenceCoordinateAtPosition(int)"><B>getReferenceCoordinateAtPosition(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getReferenceCoordinateAtPosition(int, boolean)"><B>getReferenceCoordinateAtPosition(int, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getReferenceCoordinateAtPosition(int)"><B>getReferenceCoordinateAtPosition(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getReferenceCoordinateAtPosition(int)"><B>getReferenceCoordinateAtPosition(int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getReferenceCoordinateAtPosition(int, boolean)"><B>getReferenceCoordinateAtPosition(int, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getReferenceCoordinateAtPosition(int, boolean)"><B>getReferenceCoordinateAtPosition(int, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getReferenceCoordinateAtPosition(int)"><B>getReferenceCoordinateAtPosition(int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getReferenceCoordinateAtPosition(int, boolean)"><B>getReferenceCoordinateAtPosition(int, boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getReferenceEnd()"><B>getReferenceEnd()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getReferenceEnd()"><B>getReferenceEnd()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getReferenceId()"><B>getReferenceId()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getReferenceName()"><B>getReferenceName()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getReferenceName()"><B>getReferenceName()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getReferenceName()"><B>getReferenceName()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getReferenceName()"><B>getReferenceName()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getReferenceNames()"><B>getReferenceNames()</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getReferenceNames()"><B>getReferenceNames()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getReferenceNames()"><B>getReferenceNames()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getReferenceNames()"><B>getReferenceNames()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#getReferenceNames()"><B>getReferenceNames()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>Get the names of genes in the transcriptome space
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getReferenceStart()"><B>getReferenceStart()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getReferenceStart()"><B>getReferenceStart()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getReferenceTranscript()"><B>getReferenceTranscript()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getRefseqGeneName()"><B>getRefseqGeneName()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>Get the Refseq gene name which appears between the last pair of parentheses in fasta header
<DT><A HREF="../broad/core/sequence/Sequence.html#getRefseqRnaClassName()"><B>getRefseqRnaClassName()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getRefseqSpeciesName()"><B>getRefseqSpeciesName()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>Get the species name that appears right after the transcript ID or the PREDICTED tag
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getRefSequenceLambda(java.lang.String)"><B>getRefSequenceLambda(String)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#getRefSequenceLambda(java.lang.String)"><B>getRefSequenceLambda(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getRefSequenceLambda(java.lang.String)"><B>getRefSequenceLambda(String)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getRefSequenceLength(java.lang.String)"><B>getRefSequenceLength(String)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#getRefSequenceLengths(java.lang.String)"><B>getRefSequenceLengths(String)</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>Get the lengths of the reference sequences
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#getRefSequenceLengths()"><B>getRefSequenceLengths()</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>Get the lengths of the reference sequences
<DT><A HREF="../broad/core/sequence/Sequence.html#getRegion(int, int)"><B>getRegion(int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getRegion(broad.core.sequence.SequenceRegion)"><B>getRegion(SequenceRegion)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getRegion(broad.core.sequence.SequenceRegion, boolean)"><B>getRegion(SequenceRegion, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getRegion(broad.core.sequence.SequenceRegion, boolean, java.util.Map)"><B>getRegion(SequenceRegion, boolean, Map&lt;String, IntervalTree&lt;Alignments&gt;&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getRegion(int, int)"><B>getRegion(int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegionWindow.html#getRegion()"><B>getRegion()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegionWindow.html" title="class in broad.core.sequence">SequenceRegionWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getRegion()"><B>getRegion()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getRegion(broad.core.sequence.SequenceRegion)"><B>getRegion(SequenceRegion)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getRegion(broad.core.sequence.SequenceRegion, boolean)"><B>getRegion(SequenceRegion, boolean)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getRegion(broad.core.sequence.SequenceRegion, boolean, java.util.Map)"><B>getRegion(SequenceRegion, boolean, Map&lt;String, IntervalTree&lt;Alignments&gt;&gt;)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html#getRegion()"><B>getRegion()</B></A> - 
Method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html" title="class in broad.pda.rnai">RNAiGeneAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getRegionDenominatorRPKM()"><B>getRegionDenominatorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getRegionDenominatorRPKM()"><B>getRegionDenominatorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getRegionDenominatorRPKM()"><B>getRegionDenominatorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getRegionEnd()"><B>getRegionEnd()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#getRegionLength()"><B>getRegionLength()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getRegionMapFromParameters()"><B>getRegionMapFromParameters()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getRegionNumeratorRPKM()"><B>getRegionNumeratorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getRegionNumeratorRPKM()"><B>getRegionNumeratorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getRegionNumeratorRPKM()"><B>getRegionNumeratorRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getRegionRatio()"><B>getRegionRatio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getRegionRatio()"><B>getRegionRatio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getRegionRatio()"><B>getRegionRatio()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getRegions(java.util.List)"><B>getRegions(List&lt;? extends SequenceRegion&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getRegions(java.util.List, int)"><B>getRegions(List&lt;? extends GenomicAnnotation&gt;, int)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html#getRegions()"><B>getRegions()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap">GenomeCommandLineProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#getRegions()"><B>getRegions()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html#getRegionSet()"><B>getRegionSet()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap">GenomeCommandLineProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html#getRegionsMap()"><B>getRegionsMap()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap">GenomeCommandLineProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getRegionStart()"><B>getRegionStart()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#getRegionTotal()"><B>getRegionTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#getRelationship(broad.core.datastructures.Interval)"><B>getRelationship(Interval)</B></A> - 
Method in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>Returns a constant that describes the relationship of this interval
 to a specified interval with regard to position on the number line.
<DT><A HREF="../broad/core/datastructures/Interval.Impl.html#getRelationship(broad.core.datastructures.Interval)"><B>getRelationship(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.Impl.html" title="class in broad.core.datastructures">Interval.Impl</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#getRelationship(broad.core.datastructures.Interval)"><B>getRelationship(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getRelativeGenomeLength()"><B>getRelativeGenomeLength()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps2.html#getRelativeGenomeLength()"><B>getRelativeGenomeLength()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenomeWithGaps2.html" title="class in broad.pda.seq.segmentation">GenomeWithGaps2</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getRelativePositions(broad.pda.rnai.RNAiGeneAnnotation)"><B>getRelativePositions(RNAiGeneAnnotation)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenomeWithGaps2.html#getRelativeWindow(int, int)"><B>getRelativeWindow(int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenomeWithGaps2.html" title="class in broad.pda.seq.segmentation">GenomeWithGaps2</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotation.html#getRepeatClass()"><B>getRepeatClass()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotation.html" title="class in broad.core.annotation">RepeatMaskerAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotation.html#getRepeatFamily()"><B>getRepeatFamily()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotation.html" title="class in broad.core.annotation">RepeatMaskerAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getRepeatFile(java.lang.String)"><B>getRepeatFile(String)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotation.html#getRepeatName()"><B>getRepeatName()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotation.html" title="class in broad.core.annotation">RepeatMaskerAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ScoreDiffGenes.html#getREST(java.lang.String, java.util.Map)"><B>getREST(String, Map&lt;String, ArrayList&gt;)</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ScoreDiffGenes.html" title="class in broad.pda.geneexpression">ScoreDiffGenes</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/RunDifferentialExpression.html#getResults()"><B>getResults()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/RunDifferentialExpression.html" title="class in broad.pda.differentialExpression">RunDifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.Strand.html#getReverseStrand()"><B>getReverseStrand()</B></A> - 
Method in enum nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.Strand.html" title="enum in nextgen.core.annotation">Annotation.Strand</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html#getReverseTargets(int)"><B>getReverseTargets(int)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer">SmatchLike</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#getRgb()"><B>getRgb()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#getRight()"><B>getRight()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#getRightCapture()"><B>getRightCapture()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#getRightHighInfoStart()"><B>getRightHighInfoStart()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborAnalysis.html#getRightNeighbor(nextgen.core.annotation.Gene, broad.pda.annotation.BEDFileParser)"><B>getRightNeighbor(Gene, BEDFileParser)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborAnalysis.html" title="class in broad.pda.gene">NeighborAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getRightNode()"><B>getRightNode()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the to-vertex for this edge.
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getRightNode()"><B>getRightNode()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getRightPrimer()"><B>getRightPrimer()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getRightPrimerPenalty()"><B>getRightPrimerPenalty()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getRightPrimerPosition()"><B>getRightPrimerPosition()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getRightPrimerTM()"><B>getRightPrimerTM()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#getRightTm()"><B>getRightTm()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getRNAClassFile(java.lang.String)"><B>getRNAClassFile(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get fasta file for an RNA class
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getRNAClassFileNames()"><B>getRNAClassFileNames()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.html#getRoot()"><B>getRoot()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.html" title="class in broad.core.datastructures">GenericRootedTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getRow(java.lang.String)"><B>getRow(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getRow(int)"><B>getRow(int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getRow(java.lang.String, java.util.Collection)"><B>getRow(String, Collection&lt;String&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Cell.html#getRow()"><B>getRow()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Cell.html" title="class in jaligner">Cell</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#getRow(int)"><B>getRow(int)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#getRowAnnotations()"><B>getRowAnnotations()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getRowDescriptions()"><B>getRowDescriptions()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html#getRowDimension()"><B>getRowDimension()</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html" title="class in broad.core.hmm">MarkovModel.BackwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html#getRowDimension()"><B>getRowDimension()</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="class in broad.core.hmm">MarkovModel.ForwardResult</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#getRowDimension()"><B>getRowDimension()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Get row dimension.
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#getRowLocationMap()"><B>getRowLocationMap()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getRowName(int)"><B>getRowName(int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getRowNames()"><B>getRowNames()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#getRownum()"><B>getRownum()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#getRowPackedCopy()"><B>getRowPackedCopy()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Make a one-dimensional row packed copy of the internal array.
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#getRowPosition(java.lang.String)"><B>getRowPosition(String)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#getRowTree(java.lang.String)"><B>getRowTree(String)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#getRPKM()"><B>getRPKM()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getRPKM()"><B>getRPKM()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#getRPKM()"><B>getRPKM()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getRPKMConstant()"><B>getRPKMConstant()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getRPKMConstant(java.lang.String)"><B>getRPKMConstant(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getRrnaSeqsForDepletion()"><B>getRrnaSeqsForDepletion()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get ribosomal RNA sequences to use for in silico rRNA depletion
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getRS8Score()"><B>getRS8Score()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#getRunParameters()"><B>getRunParameters()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#getRunSpecificDGEOptions()"><B>getRunSpecificDGEOptions()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Returns all run specific DGE options
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#getrunType()"><B>getrunType()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../Jama/SingularValueDecomposition.html#getS()"><B>getS()</B></A> - 
Method in class Jama.<A HREF="../Jama/SingularValueDecomposition.html" title="class in Jama">SingularValueDecomposition</A>
<DD>Return the diagonal matrix of singular values
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getSAMEnd()"><B>getSAMEnd()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getSAMEnd()"><B>getSAMEnd()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>Returns the start position of this window into SAM coordinate space
 SAM coordinates are 1-based and inclusive whereas all of our objects are 0-based exclusive
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getSAMEnd()"><B>getSAMEnd()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Returns the start position of this window into SAM coordinate space
 SAM coordinates are 1-based and inclusive whereas all of our objects are 0-based exclusive
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getSAMEnd()"><B>getSAMEnd()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSamHeader()"><B>getSamHeader()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html#getSampleClusterAssignment()"><B>getSampleClusterAssignment()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html" title="class in broad.pda.geneexpression.clustering">Kmeans.KmeansInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getSampleName()"><B>getSampleName()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get the sample name
<DT><A HREF="../broad/core/util/PipelineAutomator.DGEInput.html#getSamples()"><B>getSamples()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.DGEInput.html" title="class in broad.core.util">PipelineAutomator.DGEInput</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getSamplesToMerge()"><B>getSamplesToMerge()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get all sets of samples to merge, by merged sample name
 Also will keep individual samples
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#getSampleType()"><B>getSampleType()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#getSAMRecordIterator()"><B>getSAMRecordIterator()</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getSAMStart()"><B>getSAMStart()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getSAMStart()"><B>getSAMStart()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>Returns the start position of this window into SAM coordinate space
 SAM coordinates are 1-based and inclusive whereas all of our objects are 0-based exclusive
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getSAMStart()"><B>getSAMStart()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Returns the start position of this annotation into SAM coordinate space
 SAM coordinates are 1-based and inclusive whereas all of our objects are 0-based exclusive
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getSAMStart()"><B>getSAMStart()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMFragmentWriter.html#getSamString(net.sf.samtools.SAMRecord)"><B>getSamString(SAMRecord)</B></A> - 
Static method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMFragmentWriter.html" title="class in broad.pda.samtools">SAMFragmentWriter</A>
<DD>Get sam line for record
<DT><A HREF="../nextgen/core/annotation/Gene.html#getSAMString()"><B>getSAMString()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getSamtoolsPath()"><B>getSamtoolsPath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to samtools executable
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getSaturationSignalValue()"><B>getSaturationSignalValue()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#getScaledWindows()"><B>getScaledWindows()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html#getScaling(int)"><B>getScaling(int)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="class in broad.core.hmm">MarkovModel.ForwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#getScanPvalue()"><B>getScanPvalue()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#getScanPvalue()"><B>getScanPvalue()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#getScore()"><B>getScore()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/Feature.html#getScore()"><B>getScore()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/Feature.html" title="interface in broad.core.annotation">Feature</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getScore()"><B>getScore()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getScore()"><B>getScore()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getScore()"><B>getScore()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegionWindow.html#getScore()"><B>getScore()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegionWindow.html" title="class in broad.core.sequence">SequenceRegionWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#getScore()"><B>getScore()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#getScore()"><B>getScore()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getScore()"><B>getScore()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.html#getScore()"><B>getScore()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.html" title="class in broad.pda.seq.rap">SkellamScore</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getScore()"><B>getScore()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Cell.html#getScore()"><B>getScore()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Cell.html" title="class in jaligner">Cell</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/matrix/Matrix.html#getScore(char, char)"><B>getScore(char, char)</B></A> - 
Method in class jaligner.matrix.<A HREF="../jaligner/matrix/Matrix.html" title="class in jaligner.matrix">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getScore()"><B>getScore()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getScore()"><B>getScore()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getScore(nextgen.core.annotation.Annotation, nextgen.core.model.score.WindowProcessor)"><B>getScore(Annotation, WindowProcessor&lt;T&gt;)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Return the score for annotation based on the scoring function passed in by WindowProcessor
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#getScore()"><B>getScore()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#getScore()"><B>getScore()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#getScore()"><B>getScore()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#getScore()"><B>getScore()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/SumScore.html#getScore()"><B>getScore()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/SumScore.html" title="class in nextgen.core.model.score">SumScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getScore()"><B>getScore()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowScore.html#getScore()"><B>getScore()</B></A> - 
Method in interface nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowScore.html" title="interface in nextgen.core.model.score">WindowScore</A>
<DD>Returns the "score" as defined by each subclass (e.g.
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getScore()"><B>getScore()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#getScoredExons()"><B>getScoredExons()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getScorePerAlignmentFromCache(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getScorePerAlignmentFromCache(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getScorePerAlignmentFromCache(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getScorePerAlignmentFromCache(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getScorePerAlignmentFromCache(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>getScorePerAlignmentFromCache(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#getScores()"><B>getScores()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/matrix/Matrix.html#getScores()"><B>getScores()</B></A> - 
Method in class jaligner.matrix.<A HREF="../jaligner/matrix/Matrix.html" title="class in jaligner.matrix">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#getScores()"><B>getScores()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getScoreSum(broad.core.annotation.LightweightGenomicAnnotation)"><B>getScoreSum(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getScoreWithNoTerminalGaps()"><B>getScoreWithNoTerminalGaps()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Calculate the score of the alignment without the terminal gaps.
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#getScoring()"><B>getScoring()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getSdNullDensity()"><B>getSdNullDensity()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getSdPeakDensity()"><B>getSdPeakDensity()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../org/jgrapht/util/VertexPair.html#getSecond()"><B>getSecond()</B></A> - 
Method in class org.jgrapht.util.<A HREF="../org/jgrapht/util/VertexPair.html" title="class in org.jgrapht.util">VertexPair</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html#getSecondGene()"><B>getSecondGene()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getSecondMate()"><B>getSecondMate()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Get the second read
<DT><A HREF="../broad/pda/capture/designer/PcrTailDesigner.html#getSecondPrimers()"><B>getSecondPrimers()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/PcrTailDesigner.html" title="class in broad.pda.capture.designer">PcrTailDesigner</A>
<DD>Get the full set of second level primer pairs
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#getSentinel()"><B>getSentinel()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Get the special sentinel value that will be used to signal novelty when putting a new interval
 into the tree, or to signal "not found" when removing an interval.
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#getSequence()"><B>getSequence()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getSequence()"><B>getSequence()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeMotif.MatchedSequence.html#getSequence()"><B>getSequence()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.MatchedSequence.html" title="class in broad.core.motif.meme">MemeMotif.MatchedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#getSequence()"><B>getSequence()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getSequence()"><B>getSequence()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#getSequence(broad.core.sequence.Sequence, boolean)"><B>getSequence(Sequence, boolean)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#getSequence(java.lang.String)"><B>getSequence(String)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getSequence()"><B>getSequence()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html#getSequence()"><B>getSequence()</B></A> - 
Method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html" title="class in broad.pda.rnai">RNAiGeneAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#getSequence()"><B>getSequence()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#getSequence()"><B>getSequence()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Returns the sequence string
<DT><A HREF="../nextgen/core/annotation/Gene.html#getSequence()"><B>getSequence()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getSequence(broad.core.sequence.Sequence)"><B>getSequence(Sequence)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getSequence(broad.core.sequence.Sequence, boolean, boolean)"><B>getSequence(Sequence, boolean, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getSequence1()"><B>getSequence1()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getSequence2()"><B>getSequence2()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getSequenceAnnotationMap()"><B>getSequenceAnnotationMap()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getSequenceAnnotationMap(int)"><B>getSequenceAnnotationMap(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getSequenceBases()"><B>getSequenceBases()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getSequenceBuilder()"><B>getSequenceBuilder()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getSequenceDir()"><B>getSequenceDir()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#getSequenceForGene(java.util.Collection, java.lang.String, boolean, java.util.Map)"><B>getSequenceForGene(Collection&lt;Gene&gt;, String, boolean, Map&lt;String, IntervalTree&lt;Alignments&gt;&gt;)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#getSequenceForGene(nextgen.core.annotation.Gene, broad.core.sequence.Sequence, boolean, java.util.Map, boolean)"><B>getSequenceForGene(Gene, Sequence, boolean, Map&lt;String, IntervalTree&lt;Alignments&gt;&gt;, boolean)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#getSequenceForGene(nextgen.core.annotation.Gene, broad.core.sequence.Sequence, boolean, java.util.Map)"><B>getSequenceForGene(Gene, Sequence, boolean, Map&lt;String, IntervalTree&lt;Alignments&gt;&gt;)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#getSequenceForGene(nextgen.core.annotation.Gene, java.lang.String, boolean)"><B>getSequenceForGene(Gene, String, boolean)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#getSequenceName()"><B>getSequenceName()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#getSequenceNames()"><B>getSequenceNames()</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#getSequenceNames()"><B>getSequenceNames()</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getSequenceObject()"><B>getSequenceObject()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#getSequencePosition(int)"><B>getSequencePosition(int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>Maps a string position to a sequence position relative to the start of the alignment.
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#getSequences()"><B>getSequences()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#getSequenceUnoriented(broad.pda.datastructures.Alignments, broad.core.sequence.Sequence, boolean, java.util.Map)"><B>getSequenceUnoriented(Alignments, Sequence, boolean, Map&lt;String, IntervalTree&lt;Alignments&gt;&gt;)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#getSequenceUnoriented(broad.pda.datastructures.Alignments, broad.core.sequence.Sequence, boolean)"><B>getSequenceUnoriented(Alignments, Sequence, boolean)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#getSequenceUnoriented(broad.pda.datastructures.Alignments, java.lang.String, boolean, java.util.Map)"><B>getSequenceUnoriented(Alignments, String, boolean, Map&lt;String, IntervalTree&lt;Alignments&gt;&gt;)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getSequencingAdapterRead1()"><B>getSequencingAdapterRead1()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get sequencing adapter sequence for read 1
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getSequencingAdapterRead2()"><B>getSequencingAdapterRead2()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get sequencing adapter sequence for read 2
<DT><A HREF="../broad/pda/annotation/Locus.html#getSetIsoforms(java.lang.String)"><B>getSetIsoforms(String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.MergeSamples.html#getSetsToMerge()"><B>getSetsToMerge()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.MergeSamples.html" title="class in broad.core.parser">ConfigFileParser.MergeSamples</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getShortArm()"><B>getShortArm()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getShortRangePCRConfiguration()"><B>getShortRangePCRConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>Returns the primer3 configuration used for
 short range PCR
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getShortRangePCRConfiguration2()"><B>getShortRangePCRConfiguration2()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>Returns a less stringent version of the short range PCR configuration
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getShortRangePCRConfiguration3()"><B>getShortRangePCRConfiguration3()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>Returns a less stringent version of the short range PCR configuration 2
 Basically it reduces the minumum product size to 50
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getShortRangeShortReadPCRConfiguration()"><B>getShortRangeShortReadPCRConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getShortRangeShortReadPCRConfiguration2()"><B>getShortRangeShortReadPCRConfiguration2()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>Returns a less stringent version of the short range PCR configuration
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getShortRangeShortReadPCRConfiguration3()"><B>getShortRangeShortReadPCRConfiguration3()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>Returns a less stringent version of the short range PCR configuration 2
 Basically it reduces the minumum product size to 50
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getSignalData()"><B>getSignalData()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get the signal data alignment model
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getSignalName()"><B>getSignalName()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get signal sample name
<DT><A HREF="../broad/pda/geneexpression/PaGEJava.html#getSignificantGeneSets(double, double)"><B>getSignificantGeneSets(double, double)</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/PaGEJava.html" title="class in broad.pda.geneexpression">PaGEJava</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getSignificantPeaks(nextgen.core.annotation.Gene)"><B>getSignificantPeaks(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get significant peaks for a gene
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getSignificantPeaks(nextgen.core.annotation.Gene, boolean)"><B>getSignificantPeaks(Gene, boolean)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get significant peaks for a gene providing expression P value
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.html#getSilhouette()"><B>getSilhouette()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.html" title="class in broad.pda.geneexpression.clustering">Kmeans</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getSimilarity()"><B>getSimilarity()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#getSingleSampleScanPeaks(broad.pda.seq.protection.SampleData, nextgen.core.annotation.Gene)"><B>getSingleSampleScanPeaks(SampleData, Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Get scan peaks for the sample and the gene
<DT><A HREF="../Jama/SingularValueDecomposition.html#getSingularValues()"><B>getSingularValues()</B></A> - 
Method in class Jama.<A HREF="../Jama/SingularValueDecomposition.html" title="class in Jama">SingularValueDecomposition</A>
<DD>Return the one-dimensional array of singular values
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getSinkVertices(java.util.Collection, org.jgrapht.DirectedGraph)"><B>getSinkVertices(Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;, DirectedGraph&lt;Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of all the possible target/end vertices for this graph.
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#getSinkVertices()"><B>getSinkVertices()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>Returns a collection of all the possible target/end vertices for this graph.
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#getSites()"><B>getSites()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getSize()"><B>getSize()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getSize()"><B>getSize()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getSize()"><B>getSize()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getSize()"><B>getSize()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getSize()"><B>getSize()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getSize(nextgen.core.annotation.Annotation)"><B>getSize(Annotation)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Get the size of the annotation with respect to the coordinate space
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getSize(nextgen.core.annotation.Annotation)"><B>getSize(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getSize(nextgen.core.annotation.Annotation)"><B>getSize(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getSize(nextgen.core.annotation.Annotation)"><B>getSize(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html#getSize()"><B>getSize()</B></A> - 
Method in class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>Returns the size of the window
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getSize()"><B>getSize()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/AlignmentUtils.html#getSizesFromSAM(java.lang.String)"><B>getSizesFromSAM(String)</B></A> - 
Static method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/AlignmentUtils.html" title="class in broad.pda.seq.alignment">AlignmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.html#getSkellamPValue()"><B>getSkellamPValue()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.html" title="class in broad.pda.seq.rap">SkellamScore</A>
<DD>Based on prussell's function in PairedSampleCoverage
 Compute Skellam P-value of read counts in a region given the two parameters
 denominatorLambda Poisson lambda for background sample
  numeratorLambda Poisson lambda for signal sample
  denominatorCount The count for background sample
  numeratorCount The count for signal sample
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#getSkellamPValue()"><B>getSkellamPValue()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>Based on prussell's function in PairedSampleCoverage
 Compute Skellam P-value of read counts in a region given the two parameters
 denominatorLambda Poisson lambda for background sample
  numeratorLambda Poisson lambda for signal sample
  denominatorCount The count for background sample
  numeratorCount The count for signal sample
<DT><A HREF="../broad/core/sequence/Sequence.html#getSlider(int, int)"><B>getSlider(int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getSlider(int, int)"><B>getSlider(int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/WindowSlider.html#getSlider(broad.core.sequence.Sequence, int, int)"><B>getSlider(Sequence, int, int)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/WindowSlider.html" title="class in broad.core.sequence">WindowSlider</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/WindowSlider.html#getSlider(broad.core.sequence.SequenceRegion, int, int)"><B>getSlider(SequenceRegion, int, int)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/WindowSlider.html" title="class in broad.core.sequence">WindowSlider</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getSoftmaskedRegions()"><B>getSoftmaskedRegions()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getSortedAndUniqueExons()"><B>getSortedAndUniqueExons()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getSortedNodes()"><B>getSortedNodes()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#getSource()"><B>getSource()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html#getSourceAnnotations()"><B>getSourceAnnotations()</B></A> - 
Method in class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GenomeWindow.html#getSourceAnnotations()"><B>getSourceAnnotations()</B></A> - 
Method in class nextgen.core.feature.<A HREF="../nextgen/core/feature/GenomeWindow.html" title="class in nextgen.core.feature">GenomeWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/Window.html#getSourceAnnotations()"><B>getSourceAnnotations()</B></A> - 
Method in interface nextgen.core.feature.<A HREF="../nextgen/core/feature/Window.html" title="interface in nextgen.core.feature">Window</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getSourceVertices(java.util.Collection, org.jgrapht.DirectedGraph)"><B>getSourceVertices(Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;, DirectedGraph&lt;Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns a collection of all the possible source/start vertices for the specified graph.
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#getSourceVertices()"><B>getSourceVertices()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>Returns a collection of all the possible source/start vertices for the specified graph.
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#getSpan()"><B>getSpan()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getSpikeInDetectionLimit()"><B>getSpikeInDetectionLimit()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getSpliceConnections()"><B>getSpliceConnections()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getSpliceConnections()"><B>getSpliceConnections()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Return the splice connections contained within this read
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getSpliceConnections()"><B>getSpliceConnections()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getSpliceConnections()"><B>getSpliceConnections()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>This will return a collection of splice junctions
 This is not the same as all gaps, it will only return splice junctions
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getSpliceConnections()"><B>getSpliceConnections()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getSpliceConnections()"><B>getSpliceConnections()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#getSplicedCount(broad.pda.datastructures.Alignments, boolean)"><B>getSplicedCount(Alignments, boolean)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getSplicedCounts()"><B>getSplicedCounts()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the total spliced end counts for this edge.
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSplicedReadExons(java.lang.String)"><B>getSplicedReadExons(String)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReadExons(java.lang.String)"><B>getSplicedReadExons(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, int, int)"><B>getSplicedReads(Annotation, int, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, java.util.Collection)"><B>getSplicedReads(Annotation, Collection&lt;Predicate&lt;Annotation&gt;&gt;)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation)"><B>getSplicedReads(Annotation)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, int, int, int, broad.core.sequence.Sequence)"><B>getSplicedReads(Annotation, int, int, int, Sequence)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, java.util.Collection, int)"><B>getSplicedReads(Annotation, Collection&lt;Predicate&lt;Annotation&gt;&gt;, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, java.util.Collection, int, broad.core.sequence.Sequence)"><B>getSplicedReads(Annotation, Collection&lt;Predicate&lt;Annotation&gt;&gt;, int, Sequence)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, org.apache.commons.collections15.Predicate)"><B>getSplicedReads(Annotation, Predicate&lt;Annotation&gt;)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSplicedReads(int, int)"><B>getSplicedReads(int, int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getSplicedReads(nextgen.core.annotation.Annotation, int, int)"><B>getSplicedReads(Annotation, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getSplicedReads(nextgen.core.annotation.Annotation, java.util.List, int)"><B>getSplicedReads(Annotation, List&lt;Predicate&lt;Annotation&gt;&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getSplicedReads(nextgen.core.annotation.Annotation, int, int, int, broad.core.sequence.Sequence)"><B>getSplicedReads(Annotation, int, int, int, Sequence)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation)"><B>getSplicedReads(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, int, int, int, broad.core.sequence.Sequence)"><B>getSplicedReads(Annotation, int, int, int, Sequence)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, int, int)"><B>getSplicedReads(Annotation, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, org.apache.commons.collections15.Predicate)"><B>getSplicedReads(Annotation, Predicate&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, java.util.Collection, int)"><B>getSplicedReads(Annotation, Collection&lt;Predicate&lt;Annotation&gt;&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, java.util.Collection, int, broad.core.sequence.Sequence)"><B>getSplicedReads(Annotation, Collection&lt;Predicate&lt;Annotation&gt;&gt;, int, Sequence)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, java.util.Collection)"><B>getSplicedReads(Annotation, Collection&lt;Predicate&lt;Annotation&gt;&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReads(nextgen.core.annotation.Annotation, java.util.Collection, broad.core.sequence.Sequence)"><B>getSplicedReads(Annotation, Collection&lt;Predicate&lt;Annotation&gt;&gt;, Sequence)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSplicedReads(int, int)"><B>getSplicedReads(int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getSpliceJunctionCoordinates()"><B>getSpliceJunctionCoordinates()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getSpliceJunctionCoordinatesTree()"><B>getSpliceJunctionCoordinatesTree()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getSpliceWeight()"><B>getSpliceWeight()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getSpliceWeight()"><B>getSpliceWeight()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getSpliceWeightFactor()"><B>getSpliceWeightFactor()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getSpliceWeightFactor()"><B>getSpliceWeightFactor()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getSpliceWeightFactor()"><B>getSpliceWeightFactor()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getStandardDeviation()"><B>getStandardDeviation()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#getStandardDeviation()"><B>getStandardDeviation()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#getStart()"><B>getStart()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#getStart()"><B>getStart()</B></A> - 
Method in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>Returns the starting point of the interval.
<DT><A HREF="../broad/core/datastructures/Interval.Impl.html#getStart()"><B>getStart()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.Impl.html" title="class in broad.core.datastructures">Interval.Impl</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#getStart()"><B>getStart()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getStart()"><B>getStart()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getStart()"><B>getStart()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getStart()"><B>getStart()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>TODO: This may be to slow, maybe compute start on the fly as Path is being constructed
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getStart()"><B>getStart()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getStart()"><B>getStart()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Get start of the fragment
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getStart()"><B>getStart()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#getStart()"><B>getStart()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>QUERY METHODS
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#getStart()"><B>getStart()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getStart()"><B>getStart()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getStart1()"><B>getStart1()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getStart2()"><B>getStart2()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getStartCodon()"><B>getStartCodon()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeMotif.MatchedSequence.html#getStartOfMatch()"><B>getStartOfMatch()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.MatchedSequence.html" title="class in broad.core.motif.meme">MemeMotif.MatchedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getStartVertex()"><B>getStartVertex()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>Returns the start vertex in the path.
<DT><A HREF="../broad/core/hmm/MarkovModel.html#getState(java.lang.String)"><B>getState(String)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm">MarkovModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#getStatistic()"><B>getStatistic()</B></A> - 
Method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getStepSize()"><B>getStepSize()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get the step size
<DT><A HREF="../broad/core/annotation/GFF.html#getStrand()"><B>getStrand()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getStrand()"><B>getStrand()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#getStrand()"><B>getStrand()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getStrand()"><B>getStrand()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getStrand(java.lang.String)"><B>getStrand(String)</B></A> - 
Static method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#getStrand(char)"><B>getStrand(char)</B></A> - 
Static method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#getStrand()"><B>getStrand()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#getStrand()"><B>getStrand()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getString()"><B>getString()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#getStringArg(java.lang.String)"><B>getStringArg(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Get value of String parameter specified by flag
<DT><A HREF="../broad/core/parser/StringParser.html#getStringArray()"><B>getStringArray()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Gets the array of tokens
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#getSubAlignment(int, int, boolean)"><B>getSubAlignment(int, int, boolean)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getSubAlignment(int, int, boolean)"><B>getSubAlignment(int, int, boolean)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getSubAlignment(java.lang.String, int, int, boolean)"><B>getSubAlignment(String, int, int, boolean)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/utils/AnnotationUtils.html#getSubAnnotationMidpointWithinAnnotation(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>getSubAnnotationMidpointWithinAnnotation(Annotation, Annotation)</B></A> - 
Static method in class nextgen.core.utils.<A HREF="../nextgen/core/utils/AnnotationUtils.html" title="class in nextgen.core.utils">AnnotationUtils</A>
<DD>Get midpoint of a sub annotation with respect to a possibly spliced parent annotation
 Ignores blocks of sub annotation and only considers blocks of parent annotation
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#getSubclusters()"><B>getSubclusters()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#getSubmethod()"><B>getSubmethod()</B></A> - 
Method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getSubSequence(broad.core.sequence.SequenceRegion, int)"><B>getSubSequence(SequenceRegion, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getSubSequence(java.lang.String, int, int, int)"><B>getSubSequence(String, int, int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>Get subsequence with extension
<DT><A HREF="../broad/core/sequence/Sequence.html#getSubSequence(java.lang.String, int, int)"><B>getSubSequence(String, int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>Get subsequence
<DT><A HREF="../broad/core/sequence/Sequence.html#getSubsequence(nextgen.core.annotation.Annotation)"><B>getSubsequence(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>Get the spliced transcribed sequence of an annotation
 Bases are reported in 5' to 3' direction
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getSum(java.lang.String)"><B>getSum(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getSum(java.lang.String)"><B>getSum(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Gets the total number of reads per chr
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#getSum()"><B>getSum()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#getSummary()"><B>getSummary()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Returns a summary for alignment
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getSymbol()"><B>getSymbol()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getSyntheticConfiguration()"><B>getSyntheticConfiguration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getSyntheticConfiguration2()"><B>getSyntheticConfiguration2()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#getTarget()"><B>getTarget()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#getTargets()"><B>getTargets()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#getTask()"><B>getTask()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/functionalAnnotation.html#getTermGenes(java.lang.String, broad.pda.annotation.BEDFileParser)"><B>getTermGenes(String, BEDFileParser)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/functionalAnnotation.html" title="class in broad.pda.gene">functionalAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getTest1Configuration()"><B>getTest1Configuration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#getTest2Configuration()"><B>getTest2Configuration()</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/FisherCombinedProbabilityTest.html#getTestStatistic()"><B>getTestStatistic()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/FisherCombinedProbabilityTest.html" title="class in broad.core.math">FisherCombinedProbabilityTest</A>
<DD>Get the Fisher test statistic X^2
<DT><A HREF="../broad/pda/differentialExpression/DifferentialExpression.html#getTestStatisticMatrix()"><B>getTestStatisticMatrix()</B></A> - 
Method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialExpression.html" title="class in broad.pda.differentialExpression">DifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#getThickEnd()"><B>getThickEnd()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#getThickStart()"><B>getThickStart()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/PcrTailDesigner.html#getThirdPrimers()"><B>getThirdPrimers()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/PcrTailDesigner.html" title="class in broad.pda.capture.designer">PcrTailDesigner</A>
<DD>Get the full set of third level primer pairs
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#getThreePrimeBases()"><B>getThreePrimeBases()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#getThreePrimeBases()"><B>getThreePrimeBases()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#getThreePrimeBases()"><B>getThreePrimeBases()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#getTimePoint()"><B>getTimePoint()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/StringParser.html#getTokens(java.lang.String)"><B>getTokens(String)</B></A> - 
Static method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Parses the string around whitespace and stores tokens
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getTophatOptions()"><B>getTophatOptions()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get all tophat flags and values
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getTophatPath()"><B>getTophatPath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get path to tophat executable
<DT><A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html#getTophatPath()"><B>getTophatPath()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser">ConfigFileParser.TophatOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/CombinationGenerator.html#getTotal()"><B>getTotal()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/CombinationGenerator.html" title="class in broad.core.math">CombinationGenerator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#getTotal()"><B>getTotal()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#getTotalCoverage(nextgen.core.annotation.Annotation)"><B>getTotalCoverage(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#getTotalCoverage(nextgen.core.annotation.Annotation)"><B>getTotalCoverage(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>Returns the total base-level coverage for the given region.
<DT><A HREF="../broad/core/siphy/TreeScaler.html#getTotalDistanceFromNode(org.forester.phylogeny.PhylogenyNode)"><B>getTotalDistanceFromNode(PhylogenyNode)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#getTotalGaps()"><B>getTotalGaps()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#getTotalInSources()"><B>getTotalInSources()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#getTotalLength()"><B>getTotalLength()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#getTotalNumberOffeatures()"><B>getTotalNumberOffeatures()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getTotalNumberOfMappedReads()"><B>getTotalNumberOfMappedReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getTotalNumberOfMappedStrandedReads()"><B>getTotalNumberOfMappedStrandedReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getTotalNumberOfReads()"><B>getTotalNumberOfReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#getTotalNumberOfStrandedReads()"><B>getTotalNumberOfStrandedReads()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getTotalNumberOfStrandedReads()"><B>getTotalNumberOfStrandedReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getTotalObservations()"><B>getTotalObservations()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/countreads/LibraryCompositionByRnaClass.html#getTotalReadCounts()"><B>getTotalReadCounts()</B></A> - 
Method in class broad.pda.countreads.<A HREF="../broad/pda/countreads/LibraryCompositionByRnaClass.html" title="class in broad.pda.countreads">LibraryCompositionByRnaClass</A>
<DD>Get the total number of reads per sample
<DT><A HREF="../broad/pda/countreads/FastqLibraryStats.html#getTotalReads()"><B>getTotalReads()</B></A> - 
Method in class broad.pda.countreads.<A HREF="../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads">FastqLibraryStats</A>
<DD>Get the total number of reads
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#getTotalReads()"><B>getTotalReads()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getTotalTime()"><B>getTotalTime()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#getTrackInfo(int)"><B>getTrackInfo(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getTranscriptLength()"><B>getTranscriptLength()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD><B>Deprecated.</B>&nbsp;
<DT><A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html#getTranscriptName()"><B>getTranscriptName()</B></A> - 
Method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html" title="class in broad.pda.rnai">RNAiGeneAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#getTranscripts()"><B>getTranscripts()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getTranscriptSeqsForAlignment()"><B>getTranscriptSeqsForAlignment()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get fasta file of transcript sequences to align to
<DT><A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html#getTranscriptSequence()"><B>getTranscriptSequence()</B></A> - 
Method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html" title="class in broad.pda.rnai">RNAiGeneAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getTransitions()"><B>getTransitions()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#getTree()"><B>getTree()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#getTree()"><B>getTree()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#getTreeBit()"><B>getTreeBit()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#getTreeLength()"><B>getTreeLength()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#getTreeLength()"><B>getTreeLength()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#getTreeLength()"><B>getTreeLength()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#getTTestPvalue()"><B>getTTestPvalue()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#getType()"><B>getType()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getType()"><B>getType()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Get the type of edge.
<DT><A HREF="../jaligner/Sequence.html#getType()"><B>getType()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Returns the sequence type (nucleic or protein)
<DT><A HREF="../Jama/LUDecomposition.html#getU()"><B>getU()</B></A> - 
Method in class Jama.<A HREF="../Jama/LUDecomposition.html" title="class in Jama">LUDecomposition</A>
<DD>Return upper triangular factor
<DT><A HREF="../Jama/SingularValueDecomposition.html#getU()"><B>getU()</B></A> - 
Method in class Jama.<A HREF="../Jama/SingularValueDecomposition.html" title="class in Jama">SingularValueDecomposition</A>
<DD>Return the left singular vectors
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getUn()"><B>getUn()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#getUngappedLength()"><B>getUngappedLength()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getUngappedReferenceIslands()"><B>getUngappedReferenceIslands()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>Finds the list of ungapped regions in the reference sequence.
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#getUngappedSequenceReferenceIslands()"><B>getUngappedSequenceReferenceIslands()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>Finds the list of ungapped sequence (as opposed to alignment gaps) regions in the reference sequence.
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#getUngappedSize()"><B>getUngappedSize()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborsNullModel.html#getUniqueNeighbors()"><B>getUniqueNeighbors()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborsNullModel.html" title="class in broad.pda.gene">NeighborsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html#getUniqueRepeatClasses()"><B>getUniqueRepeatClasses()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html" title="class in broad.core.annotation">RepeatMaskerAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html#getUniqueRepeatFamilies()"><B>getUniqueRepeatFamilies()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html" title="class in broad.core.annotation">RepeatMaskerAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html#getUniqueRepeatNames()"><B>getUniqueRepeatNames()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html" title="class in broad.core.annotation">RepeatMaskerAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getUnmaskedLength(java.lang.String)"><B>getUnmaskedLength(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getUnqComaptibleBetweenAny2Sets(java.util.ArrayList, java.util.Collection, boolean)"><B>getUnqComaptibleBetweenAny2Sets(ArrayList&lt;String&gt;, Collection&lt;? extends Gene&gt;, boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#getUnqComaptibleBetweenSubsets(java.util.ArrayList, java.util.ArrayList, java.util.Collection, boolean)"><B>getUnqComaptibleBetweenSubsets(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, Collection&lt;Gene&gt;, boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#getUsedInput()"><B>getUsedInput()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/Commons.html#getUserDirectory()"><B>getUserDirectory()</B></A> - 
Static method in class jaligner.util.<A HREF="../jaligner/util/Commons.html" title="class in jaligner.util">Commons</A>
<DD>Returns user's current directory.
<DT><A HREF="../broad/pda/seq/segmentation/GeneCounts.html#getUtr3Counts()"><B>getUtr3Counts()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneCounts.html" title="class in broad.pda.seq.segmentation">GeneCounts</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneCounts.html#getUtr5Counts()"><B>getUtr5Counts()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneCounts.html" title="class in broad.pda.seq.segmentation">GeneCounts</A>
<DD>&nbsp;
<DT><A HREF="../Jama/EigenvalueDecomposition.html#getV()"><B>getV()</B></A> - 
Method in class Jama.<A HREF="../Jama/EigenvalueDecomposition.html" title="class in Jama">EigenvalueDecomposition</A>
<DD>Return the eigenvector matrix
<DT><A HREF="../Jama/SingularValueDecomposition.html#getV()"><B>getV()</B></A> - 
Method in class Jama.<A HREF="../Jama/SingularValueDecomposition.html" title="class in Jama">SingularValueDecomposition</A>
<DD>Return the right singular vectors
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#getValidOutgoingEdges(nextgen.core.annotation.Annotation, double, java.util.Collection, org.jgrapht.DirectedGraph)"><B>getValidOutgoingEdges(Annotation, double, Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;, DirectedGraph&lt;Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#getValue()"><B>getValue()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/WigReader.html#getValue(java.lang.String, int)"><B>getValue(String, int)</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/WigReader.html" title="class in nextgen.core.readers">WigReader</A>
<DD>Get value at specified position
<DT><A HREF="../broad/core/datastructures/Pair.html#getValue1()"><B>getValue1()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures">Pair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Pair.html#getValue2()"><B>getValue2()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures">Pair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#getValues(java.lang.String)"><B>getValues(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#getValues(java.lang.String, java.util.Collection)"><B>getValues(String, Collection&lt;String&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#getVectorEncodedSequence()"><B>getVectorEncodedSequence()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#getVersion()"><B>getVersion()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#getVertexCount(nextgen.core.annotation.Annotation)"><B>getVertexCount(Annotation)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns the counts for the vertex
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#getWeight(int)"><B>getWeight(int)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#getWeight()"><B>getWeight()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>Returns the weight assigned to the path.
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#getWeight()"><B>getWeight()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#getWeight()"><B>getWeight()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#getWeight()"><B>getWeight()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>The weight to give this alignment in the scoring.
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#getWeight()"><B>getWeight()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#getWidth()"><B>getWidth()</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getWidths(java.lang.String)"><B>getWidths(String)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getWigToBigWigExecutable()"><B>getWigToBigWigExecutable()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get UCSC WigToBigWig executable file
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#getWigWriterJar()"><B>getWigWriterJar()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Get WigWriter jar file
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getWindowCount(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)"><B>getWindowCount(Gene, Annotation)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get number of fragments mapping to gene window
 Get score from cache or calculate and cache score
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getWindowIterator(int, java.lang.String, int, int, int)"><B>getWindowIterator(int, String, int, int, int)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Iterate over windows within the defined region
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getWindowIterator(java.lang.String, int, int)"><B>getWindowIterator(String, int, int)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getWindowIterator(int, int)"><B>getWindowIterator(int, int)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getWindowIterator(java.util.Collection, int, int)"><B>getWindowIterator(Collection&lt;Gene&gt;, int, int)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Get a window iterator over a collection of GeneWindows
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#getWindowIterator(nextgen.core.annotation.Annotation, int, int)"><B>getWindowIterator(Annotation, int, int)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getWindowIterator(int, java.lang.String, int, int, int)"><B>getWindowIterator(int, String, int, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>Returns an iterator on chromosome chr starting at start, ending at end, of size, windowSize with overlap between consecutive windows
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getWindowIterator(nextgen.core.annotation.Annotation, int, int)"><B>getWindowIterator(Annotation, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getWindowIterator(java.lang.String, int, int)"><B>getWindowIterator(String, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>Get a window iterator from start to finish on chromosome chr
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getWindowIterator(int, int)"><B>getWindowIterator(int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>Get a window iterator from start to finish in the entire space
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#getWindowIterator(java.util.Collection, int, int)"><B>getWindowIterator(Collection&lt;Gene&gt;, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getWindowIterator(int, java.lang.String, int, int, int)"><B>getWindowIterator(int, String, int, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>Interprets the windowSize based on how it was initialized
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getWindowIterator(java.lang.String, int, int)"><B>getWindowIterator(String, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getWindowIterator(int, int)"><B>getWindowIterator(int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getWindowIterator(nextgen.core.annotation.Annotation, int, int)"><B>getWindowIterator(Annotation, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#getWindowIterator(java.util.Collection, int, int)"><B>getWindowIterator(Collection&lt;Gene&gt;, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getWindowIterator(java.util.Collection, int, int)"><B>getWindowIterator(Collection&lt;Gene&gt;, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getWindowIterator(int, java.lang.String, int, int, int)"><B>getWindowIterator(int, String, int, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getWindowIterator(nextgen.core.annotation.Annotation, int, int)"><B>getWindowIterator(Annotation, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getWindowIterator(java.lang.String, int, int)"><B>getWindowIterator(String, int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>Get a window iterator from start to finish on chromosome
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#getWindowIterator(int, int)"><B>getWindowIterator(int, int)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html#getWindowProcessor()"><B>getWindowProcessor()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap">GenomeScoringProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html#getWindowProcessor(nextgen.core.annotation.AnnotationCollection, nextgen.core.annotation.AnnotationCollection)"><B>getWindowProcessor(AnnotationCollection&lt;? extends Annotation&gt;, AnnotationCollection&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap">GenomeScoringProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#getWindowProcessor()"><B>getWindowProcessor()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#getWindowProcessor(nextgen.core.annotation.AnnotationCollection, nextgen.core.annotation.AnnotationCollection)"><B>getWindowProcessor(AnnotationCollection&lt;? extends Annotation&gt;, AnnotationCollection&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getWindows(int)"><B>getWindows(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#getWindows(int, int, int)"><B>getWindows(int, int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html#getWindows(int, int)"><B>getWindows(int, int)</B></A> - 
Method in class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>Get collection of windows spanning the annotation
<DT><A HREF="../nextgen/core/feature/GenomeWindow.html#getWindows(int, int)"><B>getWindows(int, int)</B></A> - 
Method in class nextgen.core.feature.<A HREF="../nextgen/core/feature/GenomeWindow.html" title="class in nextgen.core.feature">GenomeWindow</A>
<DD>Get collection of windows spanning the annotation
<DT><A HREF="../nextgen/core/feature/Window.html#getWindows(int, int)"><B>getWindows(int, int)</B></A> - 
Method in interface nextgen.core.feature.<A HREF="../nextgen/core/feature/Window.html" title="interface in nextgen.core.feature">Window</A>
<DD>Get collection of windows spanning the annotation
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#getWindowScores(nextgen.core.annotation.Gene)"><B>getWindowScores(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get scores for each window in the gene
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#getWindowSize()"><B>getWindowSize()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get the window size
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html#getWinWithNoJumps()"><B>getWinWithNoJumps()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#getX()"><B>getX()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/MannWhitney.html#getZ()"><B>getZ()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/MannWhitney.html" title="class in broad.core.math">MannWhitney</A>
<DD>Get the test statistic
<DT><A HREF="../broad/core/math/Wilcox2Sample.html#getZ()"><B>getZ()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/Wilcox2Sample.html" title="class in broad.core.math">Wilcox2Sample</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#getZscore(double)"><B>getZscore(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation"><B>GFF</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/GFF.html#GFF()"><B>GFF()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#GFF(java.lang.String)"><B>GFF(String)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#GFF(broad.core.annotation.LightweightGenomicAnnotation)"><B>GFF(LightweightGenomicAnnotation)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#GFF(nextgen.core.annotation.Annotation)"><B>GFF(Annotation)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#GFF(broad.core.annotation.GFF)"><B>GFF(GFF)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#GFF(java.lang.String[])"><B>GFF(String[])</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.html#gffFactory"><B>gffFactory</B></A> - 
Static variable in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFFReader.html" title="class in broad.core.annotation"><B>GFFReader</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/GFFReader.html#GFFReader()"><B>GFFReader()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFFReader.html" title="class in broad.core.annotation">GFFReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFFReader.html#GFFReader(java.io.BufferedReader)"><B>GFFReader(BufferedReader)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFFReader.html" title="class in broad.core.annotation">GFFReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFFReader.html#GFFReader(java.lang.String)"><B>GFFReader(String)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFFReader.html" title="class in broad.core.annotation">GFFReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFFReader.html#GFFReader(java.lang.String, broad.core.annotation.GenomicAnnotationFilter)"><B>GFFReader(String, GenomicAnnotationFilter&lt;GFF&gt;)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/GFFReader.html" title="class in broad.core.annotation">GFFReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GlobalSpliceFilter.html" title="class in broad.pda.seq.segmentation"><B>GlobalSpliceFilter</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/GlobalSpliceFilter.html#GlobalSpliceFilter(broad.pda.seq.segmentation.AlignmentDataModel, java.lang.String, java.lang.String, java.lang.String)"><B>GlobalSpliceFilter(AlignmentDataModel, String, String, String)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GlobalSpliceFilter.html" title="class in broad.pda.seq.segmentation">GlobalSpliceFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/GlobalStats.html" title="class in broad.pda.geneexpression"><B>GlobalStats</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/GlobalStats.html#GlobalStats(java.io.File[], java.lang.String, java.io.File)"><B>GlobalStats(File[], String, File)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/GlobalStats.html" title="class in broad.pda.geneexpression">GlobalStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/GMTParser.html" title="class in broad.core.util"><B>GMTParser</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>&nbsp;<DT><A HREF="../broad/core/util/GMTParser.html#GMTParser()"><B>GMTParser()</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/GMTParser.html" title="class in broad.core.util">GMTParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#gotWindowScoresFromFile()"><B>gotWindowScoresFromFile()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Whether the window scores were read from an existing file
<DT><A HREF="../broad/pda/differentialExpression/GrabProbes.html" title="class in broad.pda.differentialExpression"><B>GrabProbes</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/GrabProbes.html#GrabProbes(java.io.File, java.io.File, java.lang.String)"><B>GrabProbes(File, File, String)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/GrabProbes.html" title="class in broad.pda.differentialExpression">GrabProbes</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/GrabSpecificColumns.html" title="class in broad.pda.geneexpression"><B>GrabSpecificColumns</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/GrabSpecificColumns.html#GrabSpecificColumns()"><B>GrabSpecificColumns()</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/GrabSpecificColumns.html" title="class in broad.pda.geneexpression">GrabSpecificColumns</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/GSAProjection.html" title="class in broad.pda.differentialExpression"><B>GSAProjection</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/GSAProjection.html#GSAProjection(broad.core.datastructures.MatrixWithHeaders, java.io.File, java.io.File, broad.core.datastructures.MatrixWithHeaders, java.lang.String, java.io.File)"><B>GSAProjection(MatrixWithHeaders, File, File, MatrixWithHeaders, String, File)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/GSAProjection.html" title="class in broad.pda.differentialExpression">GSAProjection</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GSAScoring.html" title="class in broad.projection.gsa"><B>GSAScoring</B></A> - Class in <A HREF="../broad/projection/gsa/package-summary.html">broad.projection.gsa</A><DD>&nbsp;<DT><A HREF="../broad/projection/gsa/GSAScoring.html#GSAScoring(java.io.File, java.io.File, java.io.File, java.lang.String)"><B>GSAScoring(File, File, File, String)</B></A> - 
Constructor for class broad.projection.gsa.<A HREF="../broad/projection/gsa/GSAScoring.html" title="class in broad.projection.gsa">GSAScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GSAScoring.html#GSAScoring(java.io.File, java.io.File, java.lang.String)"><B>GSAScoring(File, File, String)</B></A> - 
Constructor for class broad.projection.gsa.<A HREF="../broad/projection/gsa/GSAScoring.html" title="class in broad.projection.gsa">GSAScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GsaUtils.html" title="class in broad.projection.gsa"><B>GsaUtils</B></A> - Class in <A HREF="../broad/projection/gsa/package-summary.html">broad.projection.gsa</A><DD>&nbsp;<DT><A HREF="../broad/projection/gsa/GsaUtils.html#GsaUtils()"><B>GsaUtils()</B></A> - 
Constructor for class broad.projection.gsa.<A HREF="../broad/projection/gsa/GsaUtils.html" title="class in broad.projection.gsa">GsaUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/GTFFileParser.html" title="class in broad.pda.annotation"><B>GTFFileParser</B></A> - Class in <A HREF="../broad/pda/annotation/package-summary.html">broad.pda.annotation</A><DD>&nbsp;<DT><A HREF="../broad/pda/annotation/GTFFileParser.html#GTFFileParser()"><B>GTFFileParser()</B></A> - 
Constructor for class broad.pda.annotation.<A HREF="../broad/pda/annotation/GTFFileParser.html" title="class in broad.pda.annotation">GTFFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/GTFFileParser.html#GTFFileParser(java.lang.String)"><B>GTFFileParser(String)</B></A> - 
Constructor for class broad.pda.annotation.<A HREF="../broad/pda/annotation/GTFFileParser.html" title="class in broad.pda.annotation">GTFFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/GTFFileParser.html#GTFFileParser(java.lang.String, java.lang.String)"><B>GTFFileParser(String, String)</B></A> - 
Constructor for class broad.pda.annotation.<A HREF="../broad/pda/annotation/GTFFileParser.html" title="class in broad.pda.annotation">GTFFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/GTFFileParser.html#GTFWrite(java.io.BufferedWriter, java.lang.String, boolean)"><B>GTFWrite(BufferedWriter, String, boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/GTFFileParser.html" title="class in broad.pda.annotation">GTFFileParser</A>
<DD>Write a GTF using only the attributes in extraFields  (not modifying the gene_id, transcript_id as GTF2BED does)
<DT><A HREF="../broad/pda/methylation/MethylationUtils.html#guessUnconverted(java.lang.String)"><B>guessUnconverted(String)</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/MethylationUtils.html" title="class in broad.pda.methylation">MethylationUtils</A>
<DD>Returns true if it seems likely that the sequence is not bisulfite converted.
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-6.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-8.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-7.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-7.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
