<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:31 EDT 2013 -->
<TITLE>
I-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="I-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-8.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-10.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-9.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-9.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_I_"><!-- --></A><H2>
<B>I</B></H2>
<DL>
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#ic()"><B>ic()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>Computes the information content of this position weith matris
<DT><A HREF="../jaligner/Markups.html#IDENTITY"><B>IDENTITY</B></A> - 
Static variable in class jaligner.<A HREF="../jaligner/Markups.html" title="class in jaligner">Markups</A>
<DD>Markup line identity character
<DT><A HREF="../Jama/Matrix.html#identity(int, int)"><B>identity(int, int)</B></A> - 
Static method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Generate identity matrix
<DT><A HREF="../broad/core/datastructures/IllegalTreeOperationException.html" title="class in broad.core.datastructures"><B>IllegalTreeOperationException</B></A> - Exception in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/IllegalTreeOperationException.html#IllegalTreeOperationException(java.lang.String)"><B>IllegalTreeOperationException(String)</B></A> - 
Constructor for exception broad.core.datastructures.<A HREF="../broad/core/datastructures/IllegalTreeOperationException.html" title="class in broad.core.datastructures">IllegalTreeOperationException</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/PairedEndTDF.html#INCLUDE_DUPLICATES"><B>INCLUDE_DUPLICATES</B></A> - 
Variable in class broad.pda.samtools.<A HREF="../broad/pda/samtools/PairedEndTDF.html" title="class in broad.pda.samtools">PairedEndTDF</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/exception/IncompleteMethodImplementationException.html" title="class in nextgen.core.exception"><B>IncompleteMethodImplementationException</B></A> - Exception in <A HREF="../nextgen/core/exception/package-summary.html">nextgen.core.exception</A><DD>This is an exception class thrown when a given method is declared but not completely implemented<DT><A HREF="../nextgen/core/exception/IncompleteMethodImplementationException.html#IncompleteMethodImplementationException()"><B>IncompleteMethodImplementationException()</B></A> - 
Constructor for exception nextgen.core.exception.<A HREF="../nextgen/core/exception/IncompleteMethodImplementationException.html" title="class in nextgen.core.exception">IncompleteMethodImplementationException</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/exception/IncompleteMethodImplementationException.html#IncompleteMethodImplementationException(java.lang.String)"><B>IncompleteMethodImplementationException(String)</B></A> - 
Constructor for exception nextgen.core.exception.<A HREF="../nextgen/core/exception/IncompleteMethodImplementationException.html" title="class in nextgen.core.exception">IncompleteMethodImplementationException</A>
<DD>This constructor prints the specified message while throwing the exception
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#incrementCount()"><B>incrementCount()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#incrementCount()"><B>incrementCount()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Increments the count by 1
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#incrementCount(float)"><B>incrementCount(float)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Increments the count by some amount
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#IncrementScoreIfOverlap(broad.core.annotation.AnnotationReader, int, java.lang.Boolean)"><B>IncrementScoreIfOverlap(AnnotationReader&lt;? extends GenomicAnnotation&gt;, int, Boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Moran FEB23 2010
 Increments by one the score of the annotations in the current object if they overlap the input
 set.
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#IncrementScoreIfOverlap(broad.pda.annotation.BEDFileParser, int, java.lang.Boolean)"><B>IncrementScoreIfOverlap(BEDFileParser, int, Boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/IndelFilter.html" title="class in nextgen.core.readFilters"><B>IndelFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/IndelFilter.html#IndelFilter()"><B>IndelFilter()</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/IndelFilter.html" title="class in nextgen.core.readFilters">IndelFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#INF"><B>INF</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#init()"><B>init()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#init(int)"><B>init(int)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#init()"><B>init()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#init(java.util.Map, double, boolean)"><B>init(Map&lt;String, Integer&gt;, double, boolean)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#init(java.util.Map, double, boolean, java.util.Map)"><B>init(Map&lt;String, Integer&gt;, double, boolean, Map&lt;String, Double&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#initHistNumDupBC(int)"><B>initHistNumDupBC(int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#initHistNumUniqueBC(int)"><B>initHistNumUniqueBC(int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.Processor.html#initRegion(nextgen.core.annotation.Annotation)"><B>initRegion(Annotation)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.Processor.html" title="class in broad.pda.seq.rap">SkellamScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.Processor.html#initRegion(nextgen.core.annotation.Annotation)"><B>initRegion(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.Processor.html" title="class in nextgen.core.model.score">CountScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html#initRegion(nextgen.core.annotation.Annotation)"><B>initRegion(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html" title="class in nextgen.core.model.score">NewRatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.Processor.html#initRegion(nextgen.core.annotation.Annotation)"><B>initRegion(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.Processor.html" title="class in nextgen.core.model.score">RatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.Processor.html#initRegion(nextgen.core.annotation.Annotation)"><B>initRegion(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.Processor.html" title="class in nextgen.core.model.score">ScanStatisticScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html#initRegion(nextgen.core.annotation.Annotation)"><B>initRegion(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html" title="class in nextgen.core.model.score">WindowAllScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowProcessor.AbstractProcessor.html#initRegion(nextgen.core.annotation.Annotation)"><B>initRegion(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowProcessor.AbstractProcessor.html" title="class in nextgen.core.model.score">WindowProcessor.AbstractProcessor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowProcessor.html#initRegion(nextgen.core.annotation.Annotation)"><B>initRegion(Annotation)</B></A> - 
Method in interface nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowProcessor.html" title="interface in nextgen.core.model.score">WindowProcessor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#initStatsByNumBCDup(int)"><B>initStatsByNumBCDup(int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#initStatsByNumDistinctBCSeen(int)"><B>initStatsByNumDistinctBCSeen(int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#INNER_LENGTH"><B>INNER_LENGTH</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#inOrientedDegree(nextgen.core.annotation.Annotation, java.util.Collection, org.jgrapht.DirectedGraph)"><B>inOrientedDegree(Annotation, Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;, DirectedGraph&lt;Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns the in degree of the given vertex for a list of allowed edge types
<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html#INPUT"><B>INPUT</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation">FilterSamByBaseComposition</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html#INPUT"><B>INPUT</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html" title="class in broad.pda.methylation">FilterSamForMethylationAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/PairedEndTDF.html#INPUT"><B>INPUT</B></A> - 
Variable in class broad.pda.samtools.<A HREF="../broad/pda/samtools/PairedEndTDF.html" title="class in broad.pda.samtools">PairedEndTDF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#INPUT"><B>INPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/ShuffleBAM.html#INPUT"><B>INPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBAM.html" title="class in broad.pda.seq.rap">ShuffleBAM</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/ShuffleBED.html#INPUT"><B>INPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBED.html" title="class in broad.pda.seq.rap">ShuffleBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/UnmarkDuplicates.html#INPUT"><B>INPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/UnmarkDuplicates.html" title="class in broad.pda.seq.rap">UnmarkDuplicates</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html#INPUT_1"><B>INPUT_1</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html" title="class in broad.pda.methylation">FilterFastqForMethylationAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html#INPUT_2"><B>INPUT_2</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html" title="class in broad.pda.methylation">FilterFastqForMethylationAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#inReversedOrientation()"><B>inReversedOrientation()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#inReversedOrientation()"><B>inReversedOrientation()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#inReversedOrientation()"><B>inReversedOrientation()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#inReversedOrientation()"><B>inReversedOrientation()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#inReversedOrientation()"><B>inReversedOrientation()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Returns true if the connection node is on the negative strand
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#insert(broad.core.sequence.SequenceRegion, broad.core.annotation.LightweightGenomicAnnotation)"><B>insert(SequenceRegion, LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#insert(broad.core.sequence.SequenceRegion, broad.core.annotation.GenomicAnnotation)"><B>insert(SequenceRegion, GenomicAnnotation)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#INTER_VERTICAL_SPACING"><B>INTER_VERTICAL_SPACING</B></A> - 
Static variable in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#interpretBasesLiberally"><B>interpretBasesLiberally</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#intersect(broad.core.annotation.AnnotationReader)"><B>intersect(AnnotationReader&lt;T&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#intersect(java.util.List)"><B>intersect(List&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#intersect(broad.core.annotation.LightweightGenomicAnnotation)"><B>intersect(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>Change the current instance to represent its intersection
 with the provided annotation if they overlap
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#intersect(java.util.Collection)"><B>intersect(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#intersect(broad.core.annotation.LightweightGenomicAnnotation)"><B>intersect(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#intersect(java.util.List)"><B>intersect(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#intersect(nextgen.core.annotation.Annotation)"><B>intersect(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#intersect(broad.pda.annotation.BEDFileParser)"><B>intersect(BEDFileParser)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#intersect(java.util.List)"><B>intersect(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#intersect(nextgen.core.annotation.Annotation)"><B>intersect(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Returns the result of intersecting this instance with another
<DT><A HREF="../nextgen/core/annotation/Annotation.html#intersect(java.util.List)"><B>intersect(List&lt;? extends Annotation&gt;)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Returns the result of intersecting this instance with a list of annotations,
 with one (possibly blocked) intersection Annotation for each provided annotation
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#intersect(nextgen.core.annotation.AnnotationList)"><B>intersect(AnnotationList&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>Returns the intersection of this AnnotationList with another.
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#intersect(nextgen.core.annotation.Annotation)"><B>intersect(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#intersect(nextgen.core.annotation.CompoundInterval)"><B>intersect(CompoundInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/SingleInterval.html#intersect(nextgen.core.annotation.SingleInterval)"><B>intersect(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation">SingleInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#intersect(nextgen.core.annotation.Annotation)"><B>intersect(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#intersect(java.util.List)"><B>intersect(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#IntersectBlocks(broad.core.annotation.GenomicAnnotation)"><B>IntersectBlocks(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html#intersectGeneSets(java.util.Collection, java.util.Collection, boolean)"><B>intersectGeneSets(Collection&lt;Gene&gt;, Collection&lt;Gene&gt;, boolean)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures"><B>Interval</B></A> - Interface in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>Semi-open interval on the integer number line.<DT><A HREF="../broad/core/datastructures/Interval.Impl.html" title="class in broad.core.datastructures"><B>Interval.Impl</B></A> - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>A perfectly trivial implementation of the Interval interface.<DT><A HREF="../broad/core/datastructures/Interval.Impl.html#Interval.Impl(int, int)"><B>Interval.Impl(int, int)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.Impl.html" title="class in broad.core.datastructures">Interval.Impl</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures"><B>IntervalTree</B></A>&lt;<A HREF="../broad/core/datastructures/IntervalTree.html" title="type parameter in IntervalTree">V</A>&gt; - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>A Red-Black tree with intervals for keys.<DT><A HREF="../broad/core/datastructures/IntervalTree.html#IntervalTree()"><B>IntervalTree()</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.FwdIterator.html" title="class in broad.core.datastructures"><B>IntervalTree.FwdIterator</B></A> - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/IntervalTree.FwdIterator.html#IntervalTree.FwdIterator(broad.core.datastructures.IntervalTree.Node)"><B>IntervalTree.FwdIterator(IntervalTree.Node&lt;V&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.FwdIterator.html" title="class in broad.core.datastructures">IntervalTree.FwdIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures"><B>IntervalTree.Node</B></A>&lt;<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="type parameter in IntervalTree.Node">V1</A>&gt; - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#IntervalTree.Node(int, int, V1)"><B>IntervalTree.Node(int, int, V1)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.OverlapIterator.html" title="class in broad.core.datastructures"><B>IntervalTree.OverlapIterator</B></A> - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/IntervalTree.OverlapIterator.html#IntervalTree.OverlapIterator(int, int)"><B>IntervalTree.OverlapIterator(int, int)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.OverlapIterator.html" title="class in broad.core.datastructures">IntervalTree.OverlapIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.RevIterator.html" title="class in broad.core.datastructures"><B>IntervalTree.RevIterator</B></A> - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/IntervalTree.RevIterator.html#IntervalTree.RevIterator(broad.core.datastructures.IntervalTree.Node)"><B>IntervalTree.RevIterator(IntervalTree.Node&lt;V&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.RevIterator.html" title="class in broad.core.datastructures">IntervalTree.RevIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.ValuesIterator.html" title="class in broad.core.datastructures"><B>IntervalTree.ValuesIterator</B></A>&lt;<A HREF="../broad/core/datastructures/IntervalTree.ValuesIterator.html" title="type parameter in IntervalTree.ValuesIterator">V1</A>&gt; - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>This will actually iterate over the contained values such that if there is overlapping start and end we have all<DT><A HREF="../broad/core/datastructures/IntervalTree.ValuesIterator.html#IntervalTree.ValuesIterator(java.util.Iterator)"><B>IntervalTree.ValuesIterator(Iterator&lt;IntervalTree.Node&lt;V1&gt;&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.ValuesIterator.html" title="class in broad.core.datastructures">IntervalTree.ValuesIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#introduceGaps(boolean, int)"><B>introduceGaps(boolean, int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#inverse()"><B>inverse()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Matrix inverse or pseudoinverse
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#invert(broad.core.sequence.SequenceRegion)"><B>invert(SequenceRegion)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#invert(broad.core.sequence.SequenceRegion)"><B>invert(SequenceRegion)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#IS_ADJACENT_AND_EMPTY"><B>IS_ADJACENT_AND_EMPTY</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#IS_LEFT_OVERHANGING_OVERLAPPER"><B>IS_LEFT_OVERHANGING_OVERLAPPER</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#IS_RIGHT_OVERHANGING_OVERLAPPER"><B>IS_RIGHT_OVERHANGING_OVERLAPPER</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#IS_STRICTLY_GREATER"><B>IS_STRICTLY_GREATER</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#IS_STRICTLY_LESS"><B>IS_STRICTLY_LESS</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#IS_SUBSET"><B>IS_SUBSET</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#IS_SUPERSET"><B>IS_SUPERSET</B></A> - 
Static variable in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.Impl.html#isAdjacent(broad.core.datastructures.Interval)"><B>isAdjacent(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.Impl.html" title="class in broad.core.datastructures">Interval.Impl</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Interval.html#isAdjacent(broad.core.datastructures.Interval)"><B>isAdjacent(Interval)</B></A> - 
Method in interface broad.core.datastructures.<A HREF="../broad/core/datastructures/Interval.html" title="interface in broad.core.datastructures">Interval</A>
<DD>Returns true if this interval ends where the specified interval starts,
 or vice versa.
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#isAdjacent(broad.core.datastructures.Interval)"><B>isAdjacent(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#isAlignmentPaired()"><B>isAlignmentPaired()</B></A> - 
Method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#isAutosome()"><B>isAutosome()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#isAutosome(java.lang.String)"><B>isAutosome(String)</B></A> - 
Static method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#isChimera()"><B>isChimera()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#isChimera()"><B>isChimera()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>For single end, returns false.
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#isChimera()"><B>isChimera()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#isCodingGene()"><B>isCodingGene()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Whether the gene is coding or noncoding
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#isColumnNonTrivial(java.lang.String)"><B>isColumnNonTrivial(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>returns true if the column has a nonzero value
<DT><A HREF="../nextgen/core/alignment/AlignmentPair.html#isComplete()"><B>isComplete()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AlignmentPair.html" title="class in nextgen.core.alignment">AlignmentPair</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/Assembly.html#isConfident()"><B>isConfident()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/Assembly.html#isConfidentIsSet()"><B>isConfidentIsSet()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#isControl()"><B>isControl()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#isControlProbe()"><B>isControlProbe()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#isDiagonal()"><B>isDiagonal()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#isDirect()"><B>isDirect()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html#isDirect()"><B>isDirect()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation">TwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#isDuplicate()"><B>isDuplicate()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Returns true is the alignment pair is a duplicate
<DT><A HREF="../nextgen/core/alignment/Alignment.html#isDuplicate()"><B>isDuplicate()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns true if the alignment is flagged as a duplicate
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#isDuplicate()"><B>isDuplicate()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Returns true is the alignment pair is a duplicate
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#isEmpty()"><B>isEmpty()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Pair.html#isEmpty()"><B>isEmpty()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures">Pair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#isEmpty()"><B>isEmpty()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#isEmpty()"><B>isEmpty()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#isEmpty()"><B>isEmpty()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#isEmpty()"><B>isEmpty()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.trivialFilter.html#isEnough(broad.core.annotation.GenomicAnnotation)"><B>isEnough(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.trivialFilter.html" title="class in broad.core.annotation">AnnotationReader.trivialFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotationFilter.html#isEnough(T)"><B>isEnough(T)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotationFilter.html" title="interface in broad.core.annotation">GenomicAnnotationFilter</A>
<DD>Evaluates whether annotation should stop sequencial read in of further annotations
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#IsExactOverlappingIsoform(nextgen.core.annotation.Gene)"><B>IsExactOverlappingIsoform(Gene)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/GenomeSpaceSampleData.html#isExpressed(nextgen.core.annotation.Gene)"><B>isExpressed(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/GenomeSpaceSampleData.html" title="class in broad.pda.seq.protection">GenomeSpaceSampleData</A>
<DD>Whether the gene is significantly expressed genome wide
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#isExpressed(nextgen.core.annotation.Gene)"><B>isExpressed(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Whether the gene is significantly expressed in the special expression sample
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#isExpressed(nextgen.core.annotation.Gene)"><B>isExpressed(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Whether the gene is significantly expressed in the sample
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#isExpressed(nextgen.core.annotation.Annotation, double)"><B>isExpressed(Annotation, double)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>Whether the region is significantly expressed
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#isExpressedInAllControlSamples(nextgen.core.annotation.Gene)"><B>isExpressedInAllControlSamples(Gene)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Whether the gene is expressed in all control samples at the given significance level
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#isFlagged(double)"><B>isFlagged(double)</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#isFlagTrue(java.lang.String)"><B>isFlagTrue(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#isFlankedBy(broad.core.annotation.TwoSubjectAnnotation, int)"><B>isFlankedBy(TwoSubjectAnnotation, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#isFlankedBy(broad.core.annotation.TwoSubjectAnnotation, int)"><B>isFlankedBy(TwoSubjectAnnotation, int)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>Checks whether a twoSubjectAnnotation flanks the instance
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#isFlankedBy(broad.core.annotation.TwoSubjectAnnotation, int)"><B>isFlankedBy(TwoSubjectAnnotation, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#isForwardStrand()"><B>isForwardStrand()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../Jama/QRDecomposition.html#isFullRank()"><B>isFullRank()</B></A> - 
Method in class Jama.<A HREF="../Jama/QRDecomposition.html" title="class in Jama">QRDecomposition</A>
<DD>Is the matrix full rank?
<DT><A HREF="../broad/pda/annotation/Locus.html#isFullyComaptible(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, boolean)"><B>isFullyComaptible(Gene, Gene, boolean)</B></A> - 
Static method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#isFullyCompatible(nextgen.core.annotation.Gene)"><B>isFullyCompatible(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#isGap(int)"><B>isGap(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#isGap(int, int)"><B>isGap(int, int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns true if position starts in gap but goes past gap
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#isInAllGaps(int, int)"><B>isInAllGaps(int, int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns true if the start and end positions are all found in gaps (introns??)
<DT><A HREF="../nextgen/core/annotation/Gene.html#isInExon(int)"><B>isInExon(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/Commons.html#isJnlp()"><B>isJnlp()</B></A> - 
Static method in class jaligner.util.<A HREF="../jaligner/util/Commons.html" title="class in jaligner.util">Commons</A>
<DD>Returns true if jnlp is enabled
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#isMapped()"><B>isMapped()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#isMapped()"><B>isMapped()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns true if the alignment is mapped
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#isMapped()"><B>isMapped()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#isNegativeStrand()"><B>isNegativeStrand()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#isNegativeStrand()"><B>isNegativeStrand()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>For paired end, returns true if the read in direction of transcription is negative stranded
<DT><A HREF="../nextgen/core/alignment/Alignment.html#isNegativeStrand()"><B>isNegativeStrand()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>For paired end, returns true if the read in direction of transcription is negative stranded
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#isNegativeStrand()"><B>isNegativeStrand()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>Returns true if the alignment is negative stranded
<DT><A HREF="../nextgen/core/annotation/Annotation.html#isNegativeStrand()"><B>isNegativeStrand()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#isNegativeStrand()"><B>isNegativeStrand()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#isNegativeStrand()"><B>isNegativeStrand()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#isNegativeStranded()"><B>isNegativeStranded()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#isNegativeStranded()"><B>isNegativeStranded()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../Jama/LUDecomposition.html#isNonsingular()"><B>isNonsingular()</B></A> - 
Method in class Jama.<A HREF="../Jama/LUDecomposition.html" title="class in Jama">LUDecomposition</A>
<DD>Is the matrix nonsingular?
<DT><A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#isoformMapContains(java.util.Map, nextgen.core.annotation.Gene)"><B>isoformMapContains(Map&lt;Gene, Set&lt;Gene&gt;&gt;, Gene)</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html" title="class in nextgen.core.scripture">BuildScriptureCoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#isOutputSet()"><B>isOutputSet()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#isOverlapCompatible(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, double)"><B>isOverlapCompatible(Gene, Gene, double)</B></A> - 
Static method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#isPaired()"><B>isPaired()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#isPaired()"><B>isPaired()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Always returns true for objects of this class.
<DT><A HREF="../nextgen/core/alignment/Alignment.html#isPaired()"><B>isPaired()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns true if the alignment is true.
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#isPaired()"><B>isPaired()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Always returns false for objects of this class.
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#isPairedEndFormat(java.lang.String)"><B>isPairedEndFormat(String)</B></A> - 
Static method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>Returns <code>true</code> if the given SAM file is in paired end format (has the mateLine) attribute.
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#isPartialPolyA(int)"><B>isPartialPolyA(int)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/isPCRLike.html" title="class in broad.core.primer3"><B>isPCRLike</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>&nbsp;<DT><A HREF="../broad/core/primer3/isPCRLike.html#isPCRLike(broad.core.datastructures.Pair, java.util.List)"><B>isPCRLike(Pair&lt;String&gt;, List&lt;Sequence&gt;)</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/isPCRLike.html" title="class in broad.core.primer3">isPCRLike</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#isPolyA()"><B>isPolyA()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#isPositiveStranded()"><B>isPositiveStranded()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#isPositiveStranded()"><B>isPositiveStranded()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#isPresent(java.lang.String)"><B>isPresent(String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#isProbeFound()"><B>isProbeFound()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#isProbeNonUniform()"><B>isProbeNonUniform()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#isProbePopulationOutlier()"><B>isProbePopulationOutlier()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#isProbeWellAboveBackground()"><B>isProbeWellAboveBackground()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#isProductRestricted(java.lang.String)"><B>isProductRestricted(String)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#isProperPair()"><B>isProperPair()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#isProperPair()"><B>isProperPair()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Returns whether the alignment represented a proper pair defined by the aligner
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#isProperPair()"><B>isProperPair()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#isRefGapped()"><B>isRefGapped()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html#isSamLine(java.lang.String)"><B>isSamLine(String)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html" title="class in broad.core.util">PipelineAutomator.AlignmentUtils</A>
<DD>Check whether the string represents a valid sam file line
<DT><A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html#isSamLine(broad.core.parser.StringParser, java.lang.String)"><B>isSamLine(StringParser, String)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html" title="class in broad.core.util">PipelineAutomator.AlignmentUtils</A>
<DD>Check whether the string represents a valid sam file line using an exisiting StringParser object
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#isSaturated()"><B>isSaturated()</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#isSequence(java.lang.String)"><B>isSequence(String)</B></A> - 
Method in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing">CRISPRDesigner.CRISPRTarget</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#isSexChromosome()"><B>isSexChromosome()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#isSexChromosome(java.lang.String)"><B>isSexChromosome(String)</B></A> - 
Static method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#isSpanningGap(int, int)"><B>isSpanningGap(int, int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns true if this region spans an intron.
<DT><A HREF="../Jama/CholeskyDecomposition.html#isSPD()"><B>isSPD()</B></A> - 
Method in class Jama.<A HREF="../Jama/CholeskyDecomposition.html" title="class in Jama">CholeskyDecomposition</A>
<DD>Is the matrix symmetric and positive definite?
<DT><A HREF="../nextgen/core/scripture/Assembly.html#isSpurious()"><B>isSpurious()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#isStopCodon(java.lang.String)"><B>isStopCodon(String)</B></A> - 
Static method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#isStranded()"><B>isStranded()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#isStranded()"><B>isStranded()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#isSymmetric()"><B>isSymmetric()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#isTerminiCoverageBias()"><B>isTerminiCoverageBias()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#isToComplex()"><B>isToComplex()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#isUnoriented()"><B>isUnoriented()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#isUnoriented()"><B>isUnoriented()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#isUnoriented()"><B>isUnoriented()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#isUnoriented()"><B>isUnoriented()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#isValidWindow(nextgen.core.annotation.Annotation)"><B>isValidWindow(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#iterator()"><B>iterator()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Return an iterator over the entire tree.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#iterator(broad.core.datastructures.Interval)"><B>iterator(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Return an iterator over all intervals greater than or equal to the specified interval.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#iterator(int, int)"><B>iterator(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Return an iterator over all intervals greater than or equal to the specified interval.
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#iterator()"><B>iterator()</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#iterator(int, int, int)"><B>iterator(int, int, int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#iterator(int, int, int, boolean)"><B>iterator(int, int, int, boolean)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#iterator(int, int)"><B>iterator(int, int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#iterator()"><B>iterator()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#iterator()"><B>iterator()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#iterator()"><B>iterator()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#iterator()"><B>iterator()</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-8.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-10.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-9.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-9.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
