<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:31 EDT 2013 -->
<TITLE>
T-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="T-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-19.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-21.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-20.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-20.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_T_"><!-- --></A><H2>
<B>T</B></H2>
<DL>
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#T_CODE"><B>T_CODE</B></A> - 
Static variable in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/Commons.html#TAB"><B>TAB</B></A> - 
Static variable in class jaligner.util.<A HREF="../jaligner/util/Commons.html" title="class in jaligner.util">Commons</A>
<DD>Tab
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#takeComplement(java.util.List)"><B>takeComplement(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Takes the complement in the given list in this Reader.
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#takeIntersection(java.util.List)"><B>takeIntersection(List&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#takeIntersection(broad.core.annotation.LightweightGenomicAnnotation)"><B>takeIntersection(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>Change the current instance to represent its intersection
 with the provided annotation if they overlap
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#takeIntersection(broad.core.annotation.LightweightGenomicAnnotation)"><B>takeIntersection(LightweightGenomicAnnotation)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Change the current instance to represent its intersection
 with the provided annotation
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#takeIntersection(broad.core.annotation.LightweightGenomicAnnotation)"><B>takeIntersection(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#takeUnion(broad.core.annotation.LightweightGenomicAnnotation)"><B>takeUnion(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>Change the current instance to represent its union
 with the provided annotation
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#takeUnion(broad.core.annotation.LightweightGenomicAnnotation)"><B>takeUnion(LightweightGenomicAnnotation)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Change the current instance to represent its union
 with the provided annotation
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#takeUnion(broad.core.annotation.LightweightGenomicAnnotation)"><B>takeUnion(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#takeUnion(nextgen.core.annotation.Gene)"><B>takeUnion(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#tallyRevisedStats()"><B>tallyRevisedStats()</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html#TARGET"><B>TARGET</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html" title="class in broad.pda.seq.rap">BuildRatioNullDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html#TARGET"><B>TARGET</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap">GenomeScoringProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#TARGET"><B>TARGET</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html#TARGET"><B>TARGET</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html" title="class in broad.pda.seq.rap">RatioPermutationPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html#TARGET"><B>TARGET</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html" title="class in broad.pda.seq.rap">SkellamKLPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestAnnotations.html" title="class in nextgen.core.tests"><B>TestAnnotations</B></A> - Class in <A HREF="../nextgen/core/tests/package-summary.html">nextgen.core.tests</A><DD>&nbsp;<DT><A HREF="../nextgen/core/tests/TestAnnotations.html#TestAnnotations()"><B>TestAnnotations()</B></A> - 
Constructor for class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestAnnotations.html" title="class in nextgen.core.tests">TestAnnotations</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestAnnotationTrimming.html" title="class in nextgen.core.tests"><B>TestAnnotationTrimming</B></A> - Class in <A HREF="../nextgen/core/tests/package-summary.html">nextgen.core.tests</A><DD>&nbsp;<DT><A HREF="../nextgen/core/tests/TestAnnotations.html#testBasicAnnotation()"><B>testBasicAnnotation()</B></A> - 
Static method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestAnnotations.html" title="class in nextgen.core.tests">TestAnnotations</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestAnnotations.html#testCompoundInterval()"><B>testCompoundInterval()</B></A> - 
Static method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestAnnotations.html" title="class in nextgen.core.tests">TestAnnotations</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestWindow.html#testConstructors()"><B>testConstructors()</B></A> - 
Method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestWindow.html" title="class in nextgen.core.tests">TestWindow</A>
<DD>Testing All Window constructors
<DT><A HREF="../nextgen/core/tests/TestCoordinateSpace.html" title="class in nextgen.core.tests"><B>TestCoordinateSpace</B></A> - Class in <A HREF="../nextgen/core/tests/package-summary.html">nextgen.core.tests</A><DD>&nbsp;<DT><A HREF="../nextgen/core/tests/TestCoordinateSpace.html#TestCoordinateSpace()"><B>TestCoordinateSpace()</B></A> - 
Constructor for class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestCoordinateSpace.html" title="class in nextgen.core.tests">TestCoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestDynamicScoring.html" title="class in nextgen.core.tests"><B>TestDynamicScoring</B></A> - Class in <A HREF="../nextgen/core/tests/package-summary.html">nextgen.core.tests</A><DD>&nbsp;<DT><A HREF="../nextgen/core/tests/TestDynamicScoring.html#TestDynamicScoring()"><B>TestDynamicScoring()</B></A> - 
Constructor for class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestDynamicScoring.html" title="class in nextgen.core.tests">TestDynamicScoring</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestEnrichmentByFeature.html" title="class in nextgen.core.tests"><B>TestEnrichmentByFeature</B></A> - Class in <A HREF="../nextgen/core/tests/package-summary.html">nextgen.core.tests</A><DD>&nbsp;<DT><A HREF="../nextgen/core/tests/TestCoordinateSpace.html#testGenomicSpace()"><B>testGenomicSpace()</B></A> - 
Method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestCoordinateSpace.html" title="class in nextgen.core.tests">TestCoordinateSpace</A>
<DD>Testing All GenomicSpace constructors and functions
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#testGoodnessOfFit(broad.core.math.EmpiricalDistribution)"><B>testGoodnessOfFit(EmpiricalDistribution)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>This function will calculate the significance of the chi square statistic between this and other empirical distribution.
<DT><A HREF="../nextgen/core/tests/TestReadCounting.html" title="class in nextgen.core.tests"><B>TestReadCounting</B></A> - Class in <A HREF="../nextgen/core/tests/package-summary.html">nextgen.core.tests</A><DD>&nbsp;<DT><A HREF="../nextgen/core/tests/TestReadCounting.html#TestReadCounting()"><B>TestReadCounting()</B></A> - 
Constructor for class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestReadCounting.html" title="class in nextgen.core.tests">TestReadCounting</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestCoordinateSpace.html#testTranscriptomeSpace()"><B>testTranscriptomeSpace()</B></A> - 
Method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestCoordinateSpace.html" title="class in nextgen.core.tests">TestCoordinateSpace</A>
<DD>Testing All GenomicSpace constructors
<DT><A HREF="../nextgen/core/tests/TestWindow.html" title="class in nextgen.core.tests"><B>TestWindow</B></A> - Class in <A HREF="../nextgen/core/tests/package-summary.html">nextgen.core.tests</A><DD>This class tests the nextgen.core.feature.Window class<DT><A HREF="../nextgen/core/tests/TestWindow.html#TestWindow()"><B>TestWindow()</B></A> - 
Constructor for class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestWindow.html" title="class in nextgen.core.tests">TestWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/AlignmentExtractor.html#threeFoldDegenerateCodons"><B>threeFoldDegenerateCodons</B></A> - 
Static variable in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/AlignmentExtractor.html" title="class in broad.core.multiplealignment">AlignmentExtractor</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#times(broad.core.datastructures.MatrixWithHeaders)"><B>times(MatrixWithHeaders)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#times(double)"><B>times(double)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Multiply a matrix by a scalar, C = s*A
<DT><A HREF="../Jama/Matrix.html#times(Jama.Matrix)"><B>times(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Linear algebraic matrix multiplication, A * B
<DT><A HREF="../Jama/Matrix.html#timesEquals(double)"><B>timesEquals(double)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Multiply a matrix by a scalar in place, A = s*A
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#toArgString()"><B>toArgString()</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#toArray()"><B>toArray()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Returns the sequence as an array of characters.
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#toBED()"><B>toBED()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#toBED(int, int, int)"><B>toBED(int, int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#toBED()"><B>toBED()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#toBED(int, int, int)"><B>toBED(int, int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#toBED()"><B>toBED()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#toBED(int, int, int)"><B>toBED(int, int, int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toBED()"><B>toBED()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toBED(int, int, int)"><B>toBED(int, int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toBED(boolean)"><B>toBED(boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toBED(boolean, int, int, int)"><B>toBED(boolean, int, int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#toBED()"><B>toBED()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#toBED(int, int, int)"><B>toBED(int, int, int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#toBEDGraph()"><B>toBEDGraph()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#toBEDGraph()"><B>toBEDGraph()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#toBEDGraph()"><B>toBEDGraph()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#toBEDGraph()"><B>toBEDGraph()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toBEDwithBedScore()"><B>toBEDwithBedScore()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#toCigar()"><B>toCigar()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#toCigar()"><B>toCigar()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>Return the Annotation object cigar string
<DT><A HREF="../nextgen/core/annotation/Annotation.html#toCigar()"><B>toCigar()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#toCigar()"><B>toCigar()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#toCollection()"><B>toCollection()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#toConsensus()"><B>toConsensus()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#toConstituentIntroformMap()"><B>toConstituentIntroformMap()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>Builds an artificial map of isoforms where "constituent" introns are the exons.
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#toConstituentIsoformMap()"><B>toConstituentIsoformMap()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#toConstituentIsoformMap(boolean)"><B>toConstituentIsoformMap(boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toCufflinksGTF(java.lang.String, java.lang.String, java.lang.String, java.lang.String)"><B>toCufflinksGTF(String, String, String, String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#toCufflinksString(boolean)"><B>toCufflinksString(boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityBinaryIO.html#toDBFromAscii(java.io.InputStream, java.sql.Connection, java.lang.String)"><B>toDBFromAscii(InputStream, Connection, String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityBinaryIO.html" title="class in broad.core.sequence">FastaQualityBinaryIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#toDoubleArray(java.util.List)"><B>toDoubleArray(List&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#toDoubleArrayList(double[])"><B>toDoubleArrayList(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#toFasta()"><B>toFasta()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#toFasta()"><B>toFasta()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#toFasta(int)"><B>toFasta(int)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#toFastq()"><B>toFastq()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#toFileName()"><B>toFileName()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#toFilledInGene()"><B>toFilledInGene()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#toFullString()"><B>toFullString()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#toFullString(java.text.DecimalFormat)"><B>toFullString(DecimalFormat)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#toGene()"><B>toGene()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#toGene(double[])"><B>toGene(double[])</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toGTF(java.lang.String)"><B>toGTF(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toGTF(java.lang.String, java.lang.String, java.lang.String)"><B>toGTF(String, String, String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#toList()"><B>toList()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#toList()"><B>toList()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#toMap()"><B>toMap()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#toMap()"><B>toMap()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#toMidBaseLogPvalScoredGGraph(java.text.DecimalFormat)"><B>toMidBaseLogPvalScoredGGraph(DecimalFormat)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#toMultipleAlignment(boolean)"><B>toMultipleAlignment(boolean)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#toMultipleAlignment()"><B>toMultipleAlignment()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html#toNanostring()"><B>toNanostring()</B></A> - 
Method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html" title="class in broad.pda.rnai">RNAiGeneAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toRefSeq()"><B>toRefSeq()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html#toRNAi()"><B>toRNAi()</B></A> - 
Method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html" title="class in broad.pda.rnai">RNAiGeneAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toSAM()"><B>toSAM()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toSAM(java.lang.String)"><B>toSAM(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toSAM(java.lang.String, boolean)"><B>toSAM(String, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#toSAMRecord()"><B>toSAMRecord()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#toSAMRecord()"><B>toSAMRecord()</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#toSAMRecord()"><B>toSAMRecord()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#toShortBED()"><B>toShortBED()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#toShortBED()"><B>toShortBED()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#toShortBED()"><B>toShortBED()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#toShortBED()"><B>toShortBED()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#toShortString()"><B>toShortString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBED.html#toShortString()"><B>toShortString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBED.html" title="class in broad.core.annotation">ShortBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#toString(boolean)"><B>toString(boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#toString(boolean)"><B>toString(boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#toString(boolean, boolean)"><B>toString(boolean, boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotation.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotation.html" title="class in broad.core.annotation">RepeatMaskerAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBED.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBED.html" title="class in broad.core.annotation">ShortBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBED.html#toString(boolean)"><B>toString(boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBED.html" title="class in broad.core.annotation">ShortBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PrimerPickingMode.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PrimerPickingMode.html" title="class in broad.core.primer3">Primer3PrimerPickingMode</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.SequenceRegionCoordinates.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.SequenceRegionCoordinates.html" title="class in broad.core.primer3">Primer3SequenceInputTags.SequenceRegionCoordinates</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#toString(boolean)"><B>toString(boolean)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityIndex.SequenceFileLayout.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityIndex.SequenceFileLayout.html" title="class in broad.core.sequence">FastaQualityIndex.SequenceFileLayout</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PhastconsReader.PhastConsElement.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PhastconsReader.PhastConsElement.html" title="class in broad.core.siphy">PhastconsReader.PhastConsElement</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#toString()"><B>toString()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentMetric.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentMetric.html" title="class in broad.pda.enrichment">EnrichmentMetric</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#toString(java.lang.String)"><B>toString(String)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#toString(boolean)"><B>toString(boolean)</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockProbe.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation">PadlockProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/RNAiGeneAnnotation.html" title="class in broad.pda.rnai">RNAiGeneAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.html#toString()"><B>toString()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.html" title="class in broad.pda.seq.rap">SkellamScore</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#toString()"><B>toString()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#toString()"><B>toString()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Returns the sequence id and the sequence string
<DT><A HREF="../Jama/Matrix.html#toString()"><B>toString()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.Strand.html#toString()"><B>toString()</B></A> - 
Method in enum nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.Strand.html" title="enum in nextgen.core.annotation">Annotation.Strand</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/SingleInterval.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation">SingleInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/ReadAlignmentStats.SAMStats.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/ReadAlignmentStats.SAMStats.html" title="class in nextgen.core.programs">ReadAlignmentStats.SAMStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#toString()"><B>toString()</B></A> - 
Method in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing">CRISPRDesigner.CRISPRTarget</A>
<DD>&nbsp;
<DT><A HREF="../org/jgrapht/util/VertexPair.html#toString()"><B>toString()</B></A> - 
Method in class org.jgrapht.util.<A HREF="../org/jgrapht/util/VertexPair.html" title="class in org.jgrapht.util">VertexPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#toStringForWriting()"><B>toStringForWriting()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Jesse June 23, 2012
<DT><A HREF="../broad/pda/datastructures/Alignments.html#toStringNum()"><B>toStringNum()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#toStringWithScores()"><B>toStringWithScores()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html#totalReads()"><B>totalReads()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html" title="class in broad.pda.seq.fastq">FastqSplitterByBarcode.BarcodeInformation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#toUCSC()"><B>toUCSC()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#toUCSC()"><B>toUCSC()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Similar to getLocationString() @see getLocationString and provided only  for backwards 
 compatibility.
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#toUCSC()"><B>toUCSC()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#toUCSC()"><B>toUCSC()</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#toUCSC()"><B>toUCSC()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#toUCSC()"><B>toUCSC()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#toUCSC()"><B>toUCSC()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#toUCSC()"><B>toUCSC()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#toUniqueSortedList(java.util.ArrayList)"><B>toUniqueSortedList(ArrayList&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#toWIGString()"><B>toWIGString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBED.html#toWIGString()"><B>toWIGString()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBED.html" title="class in broad.core.annotation">ShortBED</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#trace()"><B>trace()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Matrix trace.
<DT><A HREF="../jaligner/TracebackNode.html" title="class in jaligner"><B>TracebackNode</B></A> - Class in <A HREF="../jaligner/package-summary.html">jaligner</A><DD>Traceback node<DT><A HREF="../jaligner/TracebackNode.html#TracebackNode()"><B>TracebackNode()</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/TracebackNode.html" title="class in jaligner">TracebackNode</A>
<DD>Constructor
<DT><A HREF="../broad/core/annotation/AnnotationHandler.html#track(java.lang.String)"><B>track(String)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationHandler.html" title="interface in broad.core.annotation">AnnotationHandler</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#track(java.lang.String)"><B>track(String)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html" title="class in nextgen.core.normalize"><B>TranscriptAverageNormalization</B></A> - Class in <A HREF="../nextgen/core/normalize/package-summary.html">nextgen.core.normalize</A><DD>&nbsp;<DT><A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html#TranscriptAverageNormalization(nextgen.core.model.AlignmentModel)"><B>TranscriptAverageNormalization(AlignmentModel)</B></A> - 
Constructor for class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html" title="class in nextgen.core.normalize">TranscriptAverageNormalization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html#TranscriptAverageNormalization(nextgen.core.model.AlignmentModel, java.lang.String, java.lang.String)"><B>TranscriptAverageNormalization(AlignmentModel, String, String)</B></A> - 
Constructor for class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html" title="class in nextgen.core.normalize">TranscriptAverageNormalization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html#TranscriptAverageNormalization(nextgen.core.model.AlignmentModel, java.util.Map)"><B>TranscriptAverageNormalization(AlignmentModel, Map&lt;Annotation, Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/TranscriptAverageNormalization.html" title="class in nextgen.core.normalize">TranscriptAverageNormalization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem"><B>TranscriptomeSpace</B></A> - Class in <A HREF="../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A><DD>&nbsp;<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#TranscriptomeSpace(java.util.Map)"><B>TranscriptomeSpace(Map&lt;String, Collection&lt;Gene&gt;&gt;)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#TranscriptomeSpace(java.io.File, java.lang.String)"><B>TranscriptomeSpace(File, String)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>Builds a transcriptome space from a bam file directly
 This will collapse all reads into an assembly of transcripts
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#TranscriptomeSpace(java.io.File, java.util.Map, java.lang.String)"><B>TranscriptomeSpace(File, Map&lt;String, Collection&lt;Gene&gt;&gt;, String)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>Builds a transcriptome space from a bam file plus a known annotation file
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model"><B>TranscriptomeSpaceAlignmentModel</B></A> - Class in <A HREF="../nextgen/core/model/package-summary.html">nextgen.core.model</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection, boolean)"><B>TranscriptomeSpaceAlignmentModel(String, TranscriptomeSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, boolean)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection)"><B>TranscriptomeSpaceAlignmentModel(String, TranscriptomeSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, boolean)"><B>TranscriptomeSpaceAlignmentModel(String, TranscriptomeSpace, boolean)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace)"><B>TranscriptomeSpaceAlignmentModel(String, TranscriptomeSpace)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>TranscriptomeSpaceAlignmentModel(String, TranscriptomeSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, boolean, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, java.util.Collection, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>TranscriptomeSpaceAlignmentModel(String, TranscriptomeSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>TranscriptomeSpaceAlignmentModel(String, TranscriptomeSpace, boolean, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#TranscriptomeSpaceAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.TranscriptomeSpace, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>TranscriptomeSpaceAlignmentModel(String, TranscriptomeSpace, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/transcriptsNullModel.html" title="class in broad.pda.gene"><B>transcriptsNullModel</B></A> - Class in <A HREF="../broad/pda/gene/package-summary.html">broad.pda.gene</A><DD>&nbsp;<DT><A HREF="../broad/pda/gene/transcriptsNullModel.html#transcriptsNullModel(broad.pda.annotation.BEDFileParser, broad.pda.annotation.BEDFileParser, java.lang.String)"><B>transcriptsNullModel(BEDFileParser, BEDFileParser, String)</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/transcriptsNullModel.html" title="class in broad.pda.gene">transcriptsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#transcriptToGenomicPosition(int)"><B>transcriptToGenomicPosition(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#transcriptToGenomicPosition(int, int)"><B>transcriptToGenomicPosition(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Convert an interval in transcriptome space to genome space
<DT><A HREF="../broad/pda/seq/graph/Path.StandardTransformer.html#transform(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><B>transform(ChromosomeWithBubblesJGraphT.BubbleEdge)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.StandardTransformer.html" title="class in broad.pda.seq.graph">Path.StandardTransformer</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html#translocate(broad.core.sequence.SequenceRegion, broad.core.sequence.SequenceRegion)"><B>translocate(SequenceRegion, SequenceRegion)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/DirectoryInstalledGenomeAssembly.html" title="class in broad.pda.feature.genome">DirectoryInstalledGenomeAssembly</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#transpose()"><B>transpose()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Matrix transpose.
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.NodeLikelihoodParameters.html#transposedTransition"><B>transposedTransition</B></A> - 
Variable in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.NodeLikelihoodParameters.html" title="class in broad.core.siphy">EvolutionaryModel.NodeLikelihoodParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy"><B>TreeScaler</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/TreeScaler.html#TreeScaler()"><B>TreeScaler()</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#TreeScaler(nextgen.core.annotation.Annotation, java.io.File, java.lang.String, java.lang.String)"><B>TreeScaler(Annotation, File, String, String)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy"><B>TreeScalerIO</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#TreeScalerIO(int)"><B>TreeScalerIO(int)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#TreeScalerIO()"><B>TreeScalerIO()</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy"><B>TreeScalerIO.ScaledWindow</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#TreeScalerIO.ScaledWindow(java.lang.String[], java.lang.String, int, int)"><B>TreeScalerIO.ScaledWindow(String[], String, int, int)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#TreeScalerIO.ScaledWindow(java.lang.String[])"><B>TreeScalerIO.ScaledWindow(String[])</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#TreeScalerIO.ScaledWindow(java.lang.String)"><B>TreeScalerIO.ScaledWindow(String)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#trim(broad.core.annotation.LightweightGenomicAnnotation)"><B>trim(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#trim(int, int)"><B>trim(int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html#TRIM"><B>TRIM</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html" title="class in broad.pda.seq.rap">RatioPermutationPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#trim(int, int)"><B>trim(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>Trims the annotation in a strand-specific manner,
 that is start will end the genomic end on a negative annotation
<DT><A HREF="../nextgen/core/annotation/Annotation.html#trim(int, int)"><B>trim(int, int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Trim annotation by a set number of bases, taking into account blocks.
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#trim(int, int)"><B>trim(int, int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#trimAbsolute(int, int)"><B>trimAbsolute(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#trimBeginning(char)"><B>trimBeginning(char)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#trimByInformationContent(double)"><B>trimByInformationContent(double)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>Trims this PWM by removing start end ending columns with information content lesser than given.
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#trimEndBases(int)"><B>trimEndBases(int)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>Trims the last numBasesToTrim bases of the fastq record and returns the nucleotides trimmed.
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#trimEnds(char)"><B>trimEnds(char)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#trimFirstNBPs(int)"><B>trimFirstNBPs(int)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#trimGene(int, int)"><B>trimGene(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Trims the gene in relative space
 Returns a GeneWindow with pointer to the original
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#trimGeneEnds(nextgen.core.annotation.Gene, double)"><B>trimGeneEnds(Gene, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#trimLinker(java.lang.String)"><B>trimLinker(String)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#trimMaxContiguous(nextgen.core.annotation.Annotation, java.util.List, double)"><B>trimMaxContiguous(Annotation, List&lt;Double&gt;, double)</B></A> - 
Static method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Trim the region to max contiguous subregion above a certain quantile
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#trimMaxContiguous(nextgen.core.annotation.Gene, double)"><B>trimMaxContiguous(Gene, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Trim a region to a contiguous subregion with max number of positions having coverage above the quantile
<DT><A HREF="../nextgen/core/scripture/Assembly.html#trimNodes()"><B>trimNodes()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#trimPeak(nextgen.core.annotation.Annotation)"><B>trimPeak(Annotation)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html#trimPeak(nextgen.core.annotation.Annotation)"><B>trimPeak(Annotation)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RAPPeakCaller.html#trimPeak(nextgen.core.annotation.Annotation)"><B>trimPeak(Annotation)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RAPPeakCaller.html" title="class in broad.pda.seq.rap">RAPPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/analysis/PeakCaller.html#trimPeak(nextgen.core.annotation.Annotation)"><B>trimPeak(Annotation)</B></A> - 
Method in interface nextgen.core.analysis.<A HREF="../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A>
<DD>Trim coarse peak to optimal coordinates
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#trimPolyA()"><B>trimPolyA()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#trimStartBases(int)"><B>trimStartBases(int)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>Trims the first numBasesToTrim bases of the fastq record and returns the nucleotides trimmed.
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#tScoresByRow(java.util.List, java.util.List, java.lang.String, double)"><B>tScoresByRow(List&lt;Integer&gt;, List&lt;Integer&gt;, String, double)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#tstat(java.util.List, java.util.List, double)"><B>tstat(List&lt;Double&gt;, List&lt;Double&gt;, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#tstat(double[], double[], double)"><B>tstat(double[], double[], double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#tstat(java.util.List, java.util.List)"><B>tstat(List&lt;Double&gt;, List&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>Two sample T statistic
<DT><A HREF="../broad/core/multiplealignment/AlignmentExtractor.html#twoFoldDegenerateCodons"><B>twoFoldDegenerateCodons</B></A> - 
Static variable in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/AlignmentExtractor.html" title="class in broad.core.multiplealignment">AlignmentExtractor</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection"><B>TwoSampleScanSkellamAnalysis</B></A> - Class in <A HREF="../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#TwoSampleScanSkellamAnalysis(java.lang.String, java.lang.String, java.lang.String, int, int, double, double, double)"><B>TwoSampleScanSkellamAnalysis(String, String, String, int, int, double, double, double)</B></A> - 
Constructor for class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Construct object from bam files
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#TwoSampleScanSkellamAnalysis(java.lang.String, java.lang.String, java.util.Map, int, int, double, double, double)"><B>TwoSampleScanSkellamAnalysis(String, String, Map&lt;String, Collection&lt;Gene&gt;&gt;, int, int, double, double, double)</B></A> - 
Constructor for class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Construct object from bam files
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection"><B>TwoSampleScanSkellamPeakCaller</B></A> - Class in <A HREF="../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html#TwoSampleScanSkellamPeakCaller()"><B>TwoSampleScanSkellamPeakCaller()</B></A> - 
Constructor for class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/TwoSubjectAnnotation.html" title="interface in broad.core.annotation"><B>TwoSubjectAnnotation</B></A> - Interface in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-19.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-21.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-20.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-20.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
