<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:30 EDT 2013 -->
<TITLE>
A-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="A-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV LETTER&nbsp;
&nbsp;<A HREF="index-2.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-1.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-1.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_A_"><!-- --></A><H2>
<B>A</B></H2>
<DL>
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#A_CODE"><B>A_CODE</B></A> - 
Static variable in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#absFold(java.util.Collection, java.util.Collection, boolean)"><B>absFold(Collection&lt;Double&gt;, Collection&lt;Double&gt;, boolean)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#absFold(double[], double[], boolean)"><B>absFold(double[], double[], boolean)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/AbsolutePearsonFunction.html" title="class in broad.pda.geneexpression.clustering"><B>AbsolutePearsonFunction</B></A> - Class in <A HREF="../broad/pda/geneexpression/clustering/package-summary.html">broad.pda.geneexpression.clustering</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/clustering/AbsolutePearsonFunction.html#AbsolutePearsonFunction()"><B>AbsolutePearsonFunction()</B></A> - 
Constructor for class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/AbsolutePearsonFunction.html" title="class in broad.pda.geneexpression.clustering">AbsolutePearsonFunction</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#absoluteToRelativePosition(int)"><B>absoluteToRelativePosition(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD><B>Deprecated.</B>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#absoluteValue(Jama.Matrix)"><B>absoluteValue(Matrix)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#absoluteValue(double[])"><B>absoluteValue(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation"><B>AbstractAnnotation</B></A> - Class in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#AbstractAnnotation()"><B>AbstractAnnotation()</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation"><B>AbstractAnnotationCollection</B></A>&lt;<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="type parameter in AbstractAnnotationCollection">T</A> extends <A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; - Class in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#AbstractAnnotationCollection()"><B>AbstractAnnotationCollection()</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/AbstractFastaParser.html" title="class in broad.core.sequence"><B>AbstractFastaParser</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/AbstractFastaParser.html#AbstractFastaParser()"><B>AbstractFastaParser()</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/AbstractFastaParser.html" title="class in broad.core.sequence">AbstractFastaParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score"><B>AbstractListScore</B></A> - Class in <A HREF="../nextgen/core/model/score/package-summary.html">nextgen.core.model.score</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#AbstractListScore(nextgen.core.annotation.Annotation)"><B>AbstractListScore(Annotation)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment"><B>AbstractPairedEndAlignment</B></A> - Class in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>&nbsp;<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#AbstractPairedEndAlignment(nextgen.core.annotation.BasicAnnotation)"><B>AbstractPairedEndAlignment(BasicAnnotation)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.TranscriptionRead.html" title="enum in nextgen.core.alignment"><B>AbstractPairedEndAlignment.TranscriptionRead</B></A> - Enum in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>This enumerator represents with of the mates in the read pair (or neither for unstranded library) is in the direction of transcription<DT><A HREF="../broad/core/annotation/AnnotationReader.trivialFilter.html#accept(broad.core.annotation.GenomicAnnotation)"><B>accept(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.trivialFilter.html" title="class in broad.core.annotation">AnnotationReader.trivialFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotationFilter.html#accept(T)"><B>accept(T)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotationFilter.html" title="interface in broad.core.annotation">GenomicAnnotationFilter</A>
<DD>Evaluates whether current annotation meets the filtering criteria
<DT><A HREF="../jaligner/Sequence.html#acidAt(int)"><B>acidAt(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Returns the acid at specific location in the sequence
<DT><A HREF="../broad/core/primer3/RestrictionEnzyme.html#AcuI"><B>AcuI</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/RestrictionEnzyme.html" title="class in broad.core.primer3">RestrictionEnzyme</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#add(broad.core.datastructures.MatrixWithHeaders)"><B>add(MatrixWithHeaders)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#add(double)"><B>add(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html#add(java.lang.String, java.lang.String)"><B>add(String, String)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html" title="class in broad.pda.seq.fastq">FastqSplitterByBarcode.BarcodeInformation</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#add(jaligner.Alignment, jaligner.Alignment)"><B>add(Alignment, Alignment)</B></A> - 
Static method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Adds this alignment to another alignment, the order is important.
<DT><A HREF="../nextgen/core/alignment/AlignmentPair.html#add(net.sf.samtools.SAMRecord)"><B>add(SAMRecord)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AlignmentPair.html" title="class in nextgen.core.alignment">AlignmentPair</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#add(T)"><B>add(T)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>Add an annotation
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#addAlignedSequence(broad.core.multiplealignment.MultipleAlignment.AlignedSequence)"><B>addAlignedSequence(MultipleAlignment.AlignedSequence)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#addAlignment(nextgen.core.alignment.Alignment)"><B>addAlignment(Alignment)</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>Write a single Alignment
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#addAll(java.util.Collection)"><B>addAll(Collection&lt;Double&gt;)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#addAll(double[])"><B>addAll(double[])</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#addAll(Jama.Matrix)"><B>addAll(Matrix)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#addAll(java.lang.Iterable)"><B>addAll(Iterable&lt;? extends T&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>Add a collection of annotations
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#addAll(nextgen.core.annotation.AnnotationList)"><B>addAll(AnnotationList&lt;? extends T&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>Add another AnnotationList.
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#AddAllIsoforms(broad.pda.gene.GeneWithIsoforms)"><B>AddAllIsoforms(GeneWithIsoforms)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#AddAllIsoforms(broad.pda.gene.GeneWithIsoforms, boolean)"><B>AddAllIsoforms(GeneWithIsoforms, boolean)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#AddAllIsoNotIncludedforms(broad.pda.gene.GeneWithIsoforms)"><B>AddAllIsoNotIncludedforms(GeneWithIsoforms)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#addAnnotation(T, int)"><B>addAnnotation(T, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#addAnnotation(T)"><B>addAnnotation(T)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#addAnnotation(T)"><B>addAnnotation(T)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html#addAnnotationToGraph(nextgen.core.annotation.Annotation)"><B>addAnnotationToGraph(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">ChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#addAnnotationToGraph(nextgen.core.annotation.Annotation)"><B>addAnnotationToGraph(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#addAnnotationToLastTrack(T)"><B>addAnnotationToLastTrack(T)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#addAttribute(java.lang.String, java.lang.String)"><B>addAttribute(String, String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#addAttribute(java.lang.String, java.lang.String)"><B>addAttribute(String, String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#addAttributeValues(java.lang.String, java.util.List)"><B>addAttributeValues(String, List&lt;String&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#addBarcode(java.lang.String)"><B>addBarcode(String)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#addBlock(java.lang.String, int, int)"><B>addBlock(String, int, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#addBlock(java.lang.String, int, int)"><B>addBlock(String, int, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#addBlock(java.lang.String, int, int)"><B>addBlock(String, int, int)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>Adds a block to the annotation if it supports blocks
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#addBlock(broad.core.multiplealignment.MAFAlignment.MAFMultipleAlignmentBlock)"><B>addBlock(MAFAlignment.MAFMultipleAlignmentBlock)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#addBlock(java.lang.String, int, int)"><B>addBlock(String, int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#addBlocks(java.util.List)"><B>addBlocks(List&lt;MAFAlignment.MAFMultipleAlignmentBlock&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#addBlocks(java.util.Iterator)"><B>addBlocks(Iterator&lt;MAFAlignment.MAFMultipleAlignmentBlock&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#addBlocks(nextgen.core.annotation.Annotation)"><B>addBlocks(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#addBlocks(java.util.Collection)"><B>addBlocks(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addBooleanArg(java.lang.String, java.lang.String, boolean)"><B>addBooleanArg(String, String, boolean)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new boolean argument to set of arguments
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addBooleanArg(java.lang.String, java.lang.String, boolean, boolean)"><B>addBooleanArg(String, String, boolean, boolean)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new boolean argument to set of arguments and stores default
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#addBowtie2Option(java.lang.String, java.lang.String)"><B>addBowtie2Option(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Specify a bowtie2 option
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#addBrowserLine(java.lang.String)"><B>addBrowserLine(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html#addChild(broad.core.datastructures.GenericRootedTree.TreeNode)"><B>addChild(GenericRootedTree.TreeNode&lt;T&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html" title="class in broad.core.datastructures">GenericRootedTree.TreeNode</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#addChrMap(java.lang.String, java.util.Map)"><B>addChrMap(String, Map&lt;Gene, Set&lt;Gene&gt;&gt;)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture.IsoformMap</A>
<DD>ADDS MAPS FOR A CHROMOSOME
 If maps already exist, they will be replaced
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#addColumn(java.lang.String)"><B>addColumn(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#addColumn(double[])"><B>addColumn(double[])</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#addCommand(java.lang.String)"><B>addCommand(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Adds the command in the current line if it is not prefixed with a "#"
<DT><A HREF="../broad/core/siphy/PiHMM.html#addConservedPi(Jama.Matrix, java.lang.String)"><B>addConservedPi(Matrix, String)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.html" title="class in broad.core.siphy">PiHMM</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#addContainedIsoform(nextgen.core.annotation.Gene)"><B>addContainedIsoform(Gene)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#addContainedValue(V1)"><B>addContainedValue(V1)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#addCountsByPosition(int, int, int)"><B>addCountsByPosition(int, int, int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#addDistribution(broad.core.math.EmpiricalDistribution)"><B>addDistribution(EmpiricalDistribution)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addDoubleArg(java.lang.String, java.lang.String, boolean)"><B>addDoubleArg(String, String, boolean)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new double argument to set of arguments
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addDoubleArg(java.lang.String, java.lang.String, boolean, double)"><B>addDoubleArg(String, String, boolean, double)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new double argument to set of arguments and stores default
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#addEdge(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><B>addEdge(Annotation, Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Adds an edge e between vertices v1 and v2
<DT><A HREF="../broad/pda/seq/graph/Path.html#addEdge(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><B>addEdge(ChromosomeWithBubblesJGraphT.BubbleEdge)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#addEdge(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>addEdge(Annotation, Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#addEdge(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, boolean)"><B>addEdge(Annotation, Annotation, boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#addEdgeAddingMissingVertices(nextgen.core.annotation.Annotation, float)"><B>addEdgeAddingMissingVertices(Annotation, float)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Adds an edge to the graph, adding missing vertices to the graph.
<DT><A HREF="../broad/pda/seq/graph/Path.html#addEdges(java.util.Collection)"><B>addEdges(Collection&lt;ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.html" title="class in nextgen.core.scripture"><B>AddEndRNASeqToScripture</B></A> - Class in <A HREF="../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.html#AddEndRNASeqToScripture(java.io.File, java.io.File, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead, java.lang.String, java.lang.String, int, java.io.File, int)"><B>AddEndRNASeqToScripture(File, File, AbstractPairedEndAlignment.TranscriptionRead, String, String, int, File, int)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html" title="class in nextgen.core.scripture"><B>AddEndRNASeqToScripture.IsoformMap</B></A> - Class in <A HREF="../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html#AddEndRNASeqToScripture.IsoformMap()"><B>AddEndRNASeqToScripture.IsoformMap()</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.IsoformMap.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture.IsoformMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#addExcludedRegion(broad.core.primer3.Primer3SequenceInputTags.SequenceRegionCoordinates)"><B>addExcludedRegion(Primer3SequenceInputTags.SequenceRegionCoordinates)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#addExcludedRegions(java.util.List)"><B>addExcludedRegions(List&lt;Primer3SequenceInputTags.SequenceRegionCoordinates&gt;)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#addExtraField(java.lang.String)"><B>addExtraField(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#addExtraScore(double)"><B>addExtraScore(double)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#addExtraScore(double)"><B>addExtraScore(double)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Adds a new extra score
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#addExtraScore(double)"><B>addExtraScore(double)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#addFilter(org.apache.commons.collections15.Predicate)"><B>addFilter(Predicate&lt;T&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#addFilter(org.apache.commons.collections15.Predicate)"><B>addFilter(Predicate&lt;T&gt;)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>Adds a filter to use to decide whether an alignment is valid
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#addFilter(org.apache.commons.collections15.Predicate)"><B>addFilter(Predicate&lt;T&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#addFilter(org.apache.commons.collections15.Predicate)"><B>addFilter(Predicate&lt;Alignment&gt;)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#addFilters(java.util.Collection)"><B>addFilters(Collection&lt;Predicate&lt;T&gt;&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#addFilters(java.util.Collection)"><B>addFilters(Collection&lt;Predicate&lt;T&gt;&gt;)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>Add a set of filters to decide whether an alignment is valid
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#addFilters(java.util.Collection)"><B>addFilters(Collection&lt;Predicate&lt;T&gt;&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#addFilters(java.util.Collection)"><B>addFilters(Collection&lt;Predicate&lt;Alignment&gt;&gt;)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addFloatArg(java.lang.String, java.lang.String, boolean)"><B>addFloatArg(String, String, boolean)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new float argument to set of arguments
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addFloatArg(java.lang.String, java.lang.String, boolean, float)"><B>addFloatArg(String, String, boolean, float)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new float argument to set of arguments and stores default
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html#addFragmentLengthFilter(java.lang.String, int)"><B>addFragmentLengthFilter(String, int)</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers">FilteredBamWriter</A>
<DD>Add a new fragment length filter with max length
<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html#addFragmentLengthFilter(java.lang.String, int, int)"><B>addFragmentLengthFilter(String, int, int)</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers">FilteredBamWriter</A>
<DD>Add a new fragment length filter with min and max lengths
<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html#addGenomicSpanFilter(int)"><B>addGenomicSpanFilter(int)</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers">FilteredBamWriter</A>
<DD>Add a new genomic span filter
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#addIncludedRegion(broad.core.primer3.Primer3SequenceInputTags.SequenceRegionCoordinates)"><B>addIncludedRegion(Primer3SequenceInputTags.SequenceRegionCoordinates)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#addIncludedRegions(java.util.List)"><B>addIncludedRegions(List&lt;Primer3SequenceInputTags.SequenceRegionCoordinates&gt;)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#addIndividualGene(java.lang.String)"><B>addIndividualGene(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineAutomator.DGEInput.html#addInput(java.lang.String[])"><B>addInput(String[])</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.DGEInput.html" title="class in broad.core.util">PipelineAutomator.DGEInput</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addIntArg(java.lang.String, java.lang.String, boolean)"><B>addIntArg(String, String, boolean)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new int argument to set of arguments
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addIntArg(java.lang.String, java.lang.String, boolean, int)"><B>addIntArg(String, String, boolean, int)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new int argument to set of arguments and stores default
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#addInterval(nextgen.core.annotation.SingleInterval)"><B>addInterval(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>Adds an interval, updating the compound interval to reflect the intersection of its
 previous state and the new interval.
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#addInterval(int, int)"><B>addInterval(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#addIsoform(broad.pda.gene.GeneWithIsoforms, java.lang.String)"><B>addIsoform(GeneWithIsoforms, String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#AddIsoform(nextgen.core.annotation.Gene)"><B>AddIsoform(Gene)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#AddIsoform(nextgen.core.annotation.Gene, boolean)"><B>AddIsoform(Gene, boolean)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#addIsoform(nextgen.core.annotation.Gene)"><B>addIsoform(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#addJunctions(java.util.Collection)"><B>addJunctions(Collection&lt;Integer&gt;)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#addListener(broad.core.siphy.ScaledWindowParserListener)"><B>addListener(ScaledWindowParserListener)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/ReadAlignmentStats.FastqReadCounter.html#addListner(nextgen.core.programs.ReadAlignmentStats)"><B>addListner(ReadAlignmentStats)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/ReadAlignmentStats.FastqReadCounter.html" title="class in nextgen.core.programs">ReadAlignmentStats.FastqReadCounter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/ReadAlignmentStats.SAMStatsRunner.html#addListner(nextgen.core.programs.ReadAlignmentStats)"><B>addListner(ReadAlignmentStats)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/ReadAlignmentStats.SAMStatsRunner.html" title="class in nextgen.core.programs">ReadAlignmentStats.SAMStatsRunner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#addMatch(jaligner.Sequence, int, double)"><B>addMatch(Sequence, int, double)</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#addMatch(net.sf.samtools.SAMRecord)"><B>addMatch(SAMRecord)</B></A> - 
Method in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing">CRISPRDesigner.CRISPRTarget</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#addMergeSamples(java.lang.String)"><B>addMergeSamples(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Add a set of samples to merge
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#addNewEdge(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge, nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>addNewEdge(ChromosomeWithBubblesJGraphT.BubbleEdge, Annotation, Annotation)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Adds a new edge between v1 and v2
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#addNovoalignOption(java.lang.String, java.lang.String)"><B>addNovoalignOption(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Specify a novoalign option
<DT><A HREF="../broad/core/sequence/FastaQualityIndex.SequenceFileLayout.html#addOffset(int, long)"><B>addOffset(int, long)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityIndex.SequenceFileLayout.html" title="class in broad.core.sequence">FastaQualityIndex.SequenceFileLayout</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#addOption(java.lang.String, java.lang.String)"><B>addOption(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html#addOption(java.lang.String, java.lang.String)"><B>addOption(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser">ConfigFileParser.Bowtie2Options</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#addOption(java.lang.String, java.lang.String)"><B>addOption(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html#addOption(java.lang.String, java.lang.String)"><B>addOption(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html" title="class in broad.core.parser">ConfigFileParser.NovoalignOptions</A>
<DD>Add a flag and value
<DT><A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html#addOption(java.lang.String, java.lang.String)"><B>addOption(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser">ConfigFileParser.TophatOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#addPairedEnd(nextgen.core.alignment.Alignment)"><B>addPairedEnd(Alignment)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#addPairedEndEdges(java.util.Collection)"><B>addPairedEndEdges(Collection&lt;Alignment&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#addPairedEndSupport(org.broad.igv.sam.Alignment)"><B>addPairedEndSupport(Alignment)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Adds a paired end alignment to the collection of paired end alignments for this edge
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.PrimedRegion.html#addPrimerPair(broad.core.primer3.PrimerPair)"><B>addPrimerPair(PrimerPair)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.PrimedRegion.html" title="class in broad.core.primer3">qPCRPrimerDesigner.PrimedRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html#addProperPairFilter()"><B>addProperPairFilter()</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers">FilteredBamWriter</A>
<DD>Add a new proper pair filter
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#addPseudoCounts()"><B>addPseudoCounts()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#addPseudoCounts()"><B>addPseudoCounts()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrixIO.html#addPseudoCounts()"><B>addPseudoCounts()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrixIO.html" title="class in broad.core.motif">PositionWeightMatrixIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#addPsuedocounts(double)"><B>addPsuedocounts(double)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#addPWMColumn(double[])"><B>addPWMColumn(double[])</B></A> - 
Method in class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html#addReadFilter(org.apache.commons.collections15.Predicate)"><B>addReadFilter(Predicate&lt;Alignment&gt;)</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers">FilteredBamWriter</A>
<DD>Add a new read filter
<DT><A HREF="../nextgen/core/writers/WigWriter.html#addReadFilter(org.apache.commons.collections15.Predicate)"><B>addReadFilter(Predicate&lt;Alignment&gt;)</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/WigWriter.html" title="class in nextgen.core.writers">WigWriter</A>
<DD>Add a read filter to alignment model
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#addRecord(net.sf.samtools.SAMRecord)"><B>addRecord(SAMRecord)</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#addRefSeq(nextgen.core.annotation.Gene)"><B>addRefSeq(Gene)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#addRefSeq(broad.pda.gene.GeneWithIsoforms)"><B>addRefSeq(GeneWithIsoforms)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#addRefSeqSet(java.util.Collection)"><B>addRefSeqSet(Collection&lt;? extends Gene&gt;)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#addRevisedCountsByPosition(int, int, int)"><B>addRevisedCountsByPosition(int, int, int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#addRNAClass(java.lang.String, java.lang.String)"><B>addRNAClass(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Add an RNA class fasta file
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#addRow(java.lang.String, java.lang.String, double[])"><B>addRow(String, String, double[])</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#addScore(double)"><B>addScore(double)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#addSequence(broad.core.multiplealignment.MultipleAlignment.AlignedSequence)"><B>addSequence(MultipleAlignment.AlignedSequence)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#addSequence(broad.core.multiplealignment.MultipleAlignment.AlignedSequence)"><B>addSequence(MultipleAlignment.AlignedSequence)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#addSequence(broad.core.sequence.Sequence)"><B>addSequence(Sequence)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#addSequenceFromRawData(java.lang.String[])"><B>addSequenceFromRawData(String[])</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#addSequences(java.util.List)"><B>addSequences(List&lt;MultipleAlignment.AlignedSequence&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#addSequences(java.util.List)"><B>addSequences(List&lt;MultipleAlignment.AlignedSequence&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.MergeSamples.html#addSet(java.lang.String, java.util.Collection)"><B>addSet(String, Collection&lt;String&gt;)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.MergeSamples.html" title="class in broad.core.parser">ConfigFileParser.MergeSamples</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#addShortEncodedColumn(java.util.Map)"><B>addShortEncodedColumn(Map&lt;String, Short&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#addShortEncodedColumn(java.util.Map)"><B>addShortEncodedColumn(Map&lt;String, Short&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#addShortEncodedColumnMatrix(java.util.Map)"><B>addShortEncodedColumnMatrix(Map&lt;String, Matrix&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#addShortEncodedRegion(java.util.Map)"><B>addShortEncodedRegion(Map&lt;String, short[]&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#addShortEncodedRegion(java.util.Map)"><B>addShortEncodedRegion(Map&lt;String, short[]&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html#addSourceAnnotation(nextgen.core.annotation.Annotation)"><B>addSourceAnnotation(Annotation)</B></A> - 
Method in class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GenomeWindow.html#addSourceAnnotation(nextgen.core.annotation.Annotation)"><B>addSourceAnnotation(Annotation)</B></A> - 
Method in class nextgen.core.feature.<A HREF="../nextgen/core/feature/GenomeWindow.html" title="class in nextgen.core.feature">GenomeWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/Window.html#addSourceAnnotation(nextgen.core.annotation.Annotation)"><B>addSourceAnnotation(Annotation)</B></A> - 
Method in interface nextgen.core.feature.<A HREF="../nextgen/core/feature/Window.html" title="interface in nextgen.core.feature">Window</A>
<DD>Add a pointer to the annotation from which this Window originated
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#addSpecies(java.lang.String)"><B>addSpecies(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Adds the species
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#addStartToEvalList(int, int)"><B>addStartToEvalList(int, int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.html#addState(broad.core.hmm.MarkovState)"><B>addState(MarkovState&lt;T&gt;)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm">MarkovModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.html#addState(broad.core.hmm.MarkovState, int)"><B>addState(MarkovState&lt;T&gt;, int)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm">MarkovModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addStringArg(java.lang.String, java.lang.String, boolean)"><B>addStringArg(String, String, boolean)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new string argument to set of arguments
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../broad/core/parser/CommandLineParser.html#addStringArg(java.lang.String, java.lang.String, boolean, java.lang.String)"><B>addStringArg(String, String, boolean, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Adds new string argument to set of arguments and stores default
 Client will crash if the same argument flag or description has already been added
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#addSuffixToName(java.lang.String)"><B>addSuffixToName(String)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#addSuffixToName(java.lang.String)"><B>addSuffixToName(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#addTarget(broad.core.primer3.Primer3SequenceInputTags.SequenceRegionCoordinates)"><B>addTarget(Primer3SequenceInputTags.SequenceRegionCoordinates)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#addTargets(java.util.List)"><B>addTargets(List&lt;Primer3SequenceInputTags.SequenceRegionCoordinates&gt;)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#addToCluster(java.lang.String)"><B>addToCluster(String)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#addToDegenerateBCs()"><B>addToDegenerateBCs()</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#addTophatOption(java.lang.String, java.lang.String)"><B>addTophatOption(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Specify a tophat option
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#addTranscript(broad.pda.rnai.RNAiGeneAnnotation)"><B>addTranscript(RNAiGeneAnnotation)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#addVariableValuePair(java.lang.String, java.lang.String)"><B>addVariableValuePair(String, String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#addVertex(nextgen.core.annotation.Annotation)"><B>addVertex(Annotation)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Adds vertex v to the graph
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#addVertex(nextgen.core.annotation.Annotation)"><B>addVertex(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#addWindow(broad.core.siphy.TreeScalerIO.ScaledWindow)"><B>addWindow(TreeScalerIO.ScaledWindow)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/OutputPadlock.html#AGILENT_COMMON_INSERT"><B>AGILENT_COMMON_INSERT</B></A> - 
Static variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/OutputPadlock.html" title="class in broad.pda.methylation">OutputPadlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/OutputPadlock.html#AGILENT_LEFT_AMP"><B>AGILENT_LEFT_AMP</B></A> - 
Static variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/OutputPadlock.html" title="class in broad.pda.methylation">OutputPadlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/OutputPadlock.html#AGILENT_RIGHT_AMP"><B>AGILENT_RIGHT_AMP</B></A> - 
Static variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/OutputPadlock.html" title="class in broad.pda.methylation">OutputPadlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent"><B>AgilentArrayStats</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html#AgilentArrayStats(java.lang.String, java.lang.String)"><B>AgilentArrayStats(String, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentArrayStats.html" title="class in broad.pda.geneexpression.agilent">AgilentArrayStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent"><B>AgilentProbe</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html#AgilentProbe(java.lang.String, java.lang.String)"><B>AgilentProbe(String, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentProbe.html" title="class in broad.pda.geneexpression.agilent">AgilentProbe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentToGCT.html" title="class in broad.pda.geneexpression.agilent"><B>AgilentToGCT</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentToGCT.html#AgilentToGCT(java.io.File[], java.lang.String, java.io.File)"><B>AgilentToGCT(File[], String, File)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentToGCT.html" title="class in broad.pda.geneexpression.agilent">AgilentToGCT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent"><B>AgilentUtils</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#AgilentUtils()"><B>AgilentUtils()</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/NeedlemanWunsch.html#align(jaligner.Sequence, jaligner.Sequence, jaligner.matrix.Matrix, float)"><B>align(Sequence, Sequence, Matrix, float)</B></A> - 
Static method in class jaligner.<A HREF="../jaligner/NeedlemanWunsch.html" title="class in jaligner">NeedlemanWunsch</A>
<DD>Aligns two sequences by Needleman-Wunsch (global)
<DT><A HREF="../jaligner/NeedlemanWunschGotoh.html#align(jaligner.Sequence, jaligner.Sequence, jaligner.matrix.Matrix, float, float)"><B>align(Sequence, Sequence, Matrix, float, float)</B></A> - 
Static method in class jaligner.<A HREF="../jaligner/NeedlemanWunschGotoh.html" title="class in jaligner">NeedlemanWunschGotoh</A>
<DD>Aligns two sequences by Needleman-Wunsch (global)
<DT><A HREF="../jaligner/SmithWatermanGotoh.html#align(jaligner.Sequence, jaligner.Sequence, jaligner.matrix.Matrix, float, float)"><B>align(Sequence, Sequence, Matrix, float, float)</B></A> - 
Static method in class jaligner.<A HREF="../jaligner/SmithWatermanGotoh.html" title="class in jaligner">SmithWatermanGotoh</A>
<DD>Aligns two sequences by Smith-Waterman (local)
<DT><A HREF="../broad/pda/countreads/LibraryCompositionByRnaClass.html#alignAndGetCounts(java.lang.String, java.lang.String, java.util.Map, java.lang.String, java.lang.String)"><B>alignAndGetCounts(String, String, Map&lt;String, String&gt;, String, String)</B></A> - 
Method in class broad.pda.countreads.<A HREF="../broad/pda/countreads/LibraryCompositionByRnaClass.html" title="class in broad.pda.countreads">LibraryCompositionByRnaClass</A>
<DD>For each sample, get counts of reads mapping to each RNA class as well as genome and unmapped
<DT><A HREF="../jaligner/Alignment.html" title="class in jaligner"><B>Alignment</B></A> - Class in <A HREF="../jaligner/package-summary.html">jaligner</A><DD>Holds the output of a pairwise sequences alignment.<DT><A HREF="../jaligner/Alignment.html#Alignment()"><B>Alignment()</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Constructor for Alignment
<DT><A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment"><B>Alignment</B></A> - Interface in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>Based on the Alignment interface in IGV.<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation"><B>AlignmentDataModel</B></A> - Interface in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation"><B>AlignmentDataModelStats</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats()"><B>AlignmentDataModelStats()</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel, java.util.Map, java.lang.String)"><B>AlignmentDataModelStats(AlignmentDataModel, Map&lt;String, Integer&gt;, String)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel, java.util.Map, java.lang.String, double, boolean)"><B>AlignmentDataModelStats(AlignmentDataModel, Map&lt;String, Integer&gt;, String, double, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel, java.util.Map)"><B>AlignmentDataModelStats(AlignmentDataModel, Map&lt;String, Integer&gt;)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel, java.util.Map, java.util.Map)"><B>AlignmentDataModelStats(AlignmentDataModel, Map&lt;String, Integer&gt;, Map&lt;String, Double&gt;)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel, java.util.Map, java.lang.String, boolean)"><B>AlignmentDataModelStats(AlignmentDataModel, Map&lt;String, Integer&gt;, String, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel, java.util.Map, double)"><B>AlignmentDataModelStats(AlignmentDataModel, Map&lt;String, Integer&gt;, double)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel, java.util.Map, double, boolean)"><B>AlignmentDataModelStats(AlignmentDataModel, Map&lt;String, Integer&gt;, double, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel, java.util.Map, java.lang.String, double)"><B>AlignmentDataModelStats(AlignmentDataModel, Map&lt;String, Integer&gt;, String, double)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel, java.lang.String)"><B>AlignmentDataModelStats(AlignmentDataModel, String)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#AlignmentDataModelStats(broad.pda.seq.segmentation.AlignmentDataModel)"><B>AlignmentDataModelStats(AlignmentDataModel)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.ReadCountNormalizerToBaselineAlignment.html" title="class in broad.pda.seq.segmentation"><B>AlignmentDataModelStats.ReadCountNormalizerToBaselineAlignment</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/AlignmentExtractor.html" title="class in broad.core.multiplealignment"><B>AlignmentExtractor</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/AlignmentExtractor.html#AlignmentExtractor()"><B>AlignmentExtractor()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/AlignmentExtractor.html" title="class in broad.core.multiplealignment">AlignmentExtractor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model"><B>AlignmentModel</B></A> - Class in <A HREF="../nextgen/core/model/package-summary.html">nextgen.core.model</A><DD>Serves the purpose of the Generic DataAlignment model and the ContinuousDataAlignmentModel
 Will populate alignments
 If a paired end file has unmapped mates, it will be represented by a single end read
 If both mates are mapped it will be represented as paired end reads
 Therefore a single model might have both, use flag X to exclude or include only one type
 This will be a bridge to the analysis modules<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection, boolean)"><B>AlignmentModel(String, CoordinateSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, boolean)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Build with a BAM file
 Populate the alignment collection
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel()"><B>AlignmentModel()</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>AlignmentModel(String, CoordinateSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, boolean, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Populate the alignment collection
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead, boolean)"><B>AlignmentModel(String, CoordinateSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, boolean, AbstractPairedEndAlignment.TranscriptionRead, boolean)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Populate the alignment collection
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection)"><B>AlignmentModel(String, CoordinateSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>AlignmentModel(String, CoordinateSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, boolean)"><B>AlignmentModel(String, CoordinateSpace, boolean)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>AlignmentModel(String, CoordinateSpace, boolean, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace)"><B>AlignmentModel(String, CoordinateSpace)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#AlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>AlignmentModel(String, CoordinateSpace, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.AlignmentCount.html" title="class in nextgen.core.model"><B>AlignmentModel.AlignmentCount</B></A> - Class in <A HREF="../nextgen/core/model/package-summary.html">nextgen.core.model</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/AlignmentModel.AlignmentCount.html#AlignmentModel.AlignmentCount(nextgen.core.alignment.Alignment)"><B>AlignmentModel.AlignmentCount(Alignment)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.AlignmentCount.html" title="class in nextgen.core.model">AlignmentModel.AlignmentCount</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/utils/AlignmentModelGlobalStats.html" title="class in nextgen.core.utils"><B>AlignmentModelGlobalStats</B></A> - Class in <A HREF="../nextgen/core/utils/package-summary.html">nextgen.core.utils</A><DD>&nbsp;<DT><A HREF="../nextgen/core/utils/AlignmentModelGlobalStats.html#AlignmentModelGlobalStats()"><B>AlignmentModelGlobalStats()</B></A> - 
Constructor for class nextgen.core.utils.<A HREF="../nextgen/core/utils/AlignmentModelGlobalStats.html" title="class in nextgen.core.utils">AlignmentModelGlobalStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AlignmentPair.html" title="class in nextgen.core.alignment"><B>AlignmentPair</B></A> - Class in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>Represents a paired alignment in all detail, with possbily multiple mappings
 through the genome.<DT><A HREF="../nextgen/core/alignment/AlignmentPair.html#AlignmentPair()"><B>AlignmentPair()</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AlignmentPair.html" title="class in nextgen.core.alignment">AlignmentPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures"><B>Alignments</B></A> - Class in <A HREF="../broad/pda/datastructures/package-summary.html">broad.pda.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(java.lang.String)"><B>Alignments(String)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(java.lang.String, int, int)"><B>Alignments(String, int, int)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(java.lang.String, int, int, nextgen.core.annotation.Annotation.Strand)"><B>Alignments(String, int, int, Annotation.Strand)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(java.lang.String, int, int, java.lang.String)"><B>Alignments(String, int, int, String)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(java.lang.String, int, int, nextgen.core.annotation.Annotation.Strand, java.lang.String)"><B>Alignments(String, int, int, Annotation.Strand, String)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(java.lang.String, java.lang.String, java.lang.String)"><B>Alignments(String, String, String)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(nextgen.core.annotation.Annotation)"><B>Alignments(Annotation)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(java.lang.String, boolean)"><B>Alignments(String, boolean)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(broad.core.annotation.LightweightGenomicAnnotation)"><B>Alignments(LightweightGenomicAnnotation)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(org.broad.igv.sam.Alignment)"><B>Alignments(Alignment)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#Alignments(nextgen.core.alignment.Alignment)"><B>Alignments(Alignment)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/AlignmentsReader.html" title="class in broad.pda.datastructures"><B>AlignmentsReader</B></A> - Class in <A HREF="../broad/pda/datastructures/package-summary.html">broad.pda.datastructures</A><DD><B>Deprecated.</B>&nbsp;<DT><A HREF="../broad/pda/datastructures/AlignmentsReader.html#AlignmentsReader()"><B>AlignmentsReader()</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/AlignmentsReader.html" title="class in broad.pda.datastructures">AlignmentsReader</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/datastructures/AlignmentsReader.html#AlignmentsReader(java.util.Collection)"><B>AlignmentsReader(Collection&lt;Alignments&gt;)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/AlignmentsReader.html" title="class in broad.pda.datastructures">AlignmentsReader</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/datastructures/AlignmentsReader.html#AlignmentsReader(java.io.BufferedReader)"><B>AlignmentsReader(BufferedReader)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/AlignmentsReader.html" title="class in broad.pda.datastructures">AlignmentsReader</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/datastructures/AlignmentsReader.html#AlignmentsReader(java.lang.String)"><B>AlignmentsReader(String)</B></A> - 
Constructor for class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/AlignmentsReader.html" title="class in broad.pda.datastructures">AlignmentsReader</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/AlignmentUtils.html" title="class in broad.pda.seq.alignment"><B>AlignmentUtils</B></A> - Class in <A HREF="../broad/pda/seq/alignment/package-summary.html">broad.pda.seq.alignment</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/alignment/AlignmentUtils.html#AlignmentUtils()"><B>AlignmentUtils()</B></A> - 
Constructor for class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/AlignmentUtils.html" title="class in broad.pda.seq.alignment">AlignmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/utils/AlignmentUtils.html" title="class in nextgen.core.utils"><B>AlignmentUtils</B></A> - Class in <A HREF="../nextgen/core/utils/package-summary.html">nextgen.core.utils</A><DD>&nbsp;<DT><A HREF="../nextgen/core/utils/AlignmentUtils.html#AlignmentUtils()"><B>AlignmentUtils()</B></A> - 
Constructor for class nextgen.core.utils.<A HREF="../nextgen/core/utils/AlignmentUtils.html" title="class in nextgen.core.utils">AlignmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#allIsoformsToBED()"><B>allIsoformsToBED()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineUtils.html#allJobsStatus(java.lang.Runtime)"><B>allJobsStatus(Runtime)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/PipelineUtils.html" title="class in broad.core.util">PipelineUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#almostContains(nextgen.core.annotation.Gene, int)"><B>almostContains(Gene, int)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#almostContains(broad.pda.gene.GeneWithIsoforms, int)"><B>almostContains(GeneWithIsoforms, int)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#almostContains(nextgen.core.annotation.Gene, int)"><B>almostContains(Gene, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#almostEqual(broad.core.annotation.LightweightGenomicAnnotation, int)"><B>almostEqual(LightweightGenomicAnnotation, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>Checks whether two annotations differ by a small (fudge) factor
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#almostEqual(broad.core.annotation.LightweightGenomicAnnotation, int)"><B>almostEqual(LightweightGenomicAnnotation, int)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Checks whether two annotations differ by a small (fudge) factor
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#almostEqual(broad.core.annotation.LightweightGenomicAnnotation, int)"><B>almostEqual(LightweightGenomicAnnotation, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>Checks whether two annotations differ by a small (fudge) factor
<DT><A HREF="../nextgen/core/annotation/Gene.html#almostEqual(nextgen.core.annotation.Gene, int)"><B>almostEqual(Gene, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#almostSameStructure(nextgen.core.annotation.Gene, int)"><B>almostSameStructure(Gene, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/general/Predicates.html#alwaysFalse()"><B>alwaysFalse()</B></A> - 
Static method in class nextgen.core.general.<A HREF="../nextgen/core/general/Predicates.html" title="class in nextgen.core.general">Predicates</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/general/Predicates.html#alwaysTrue()"><B>alwaysTrue()</B></A> - 
Static method in class nextgen.core.general.<A HREF="../nextgen/core/general/Predicates.html" title="class in nextgen.core.general">Predicates</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/AMEResultReader.html" title="class in broad.core.motif.meme"><B>AMEResultReader</B></A> - Class in <A HREF="../broad/core/motif/meme/package-summary.html">broad.core.motif.meme</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/meme/AMEResultReader.html#AMEResultReader(java.io.BufferedReader)"><B>AMEResultReader(BufferedReader)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/AMEResultReader.html" title="class in broad.core.motif.meme">AMEResultReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/AMEResultReader.html#AMEResultReader(java.io.InputStream)"><B>AMEResultReader(InputStream)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/AMEResultReader.html" title="class in broad.core.motif.meme">AMEResultReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/AMEResultReader.html#AMEResultReader(java.lang.String)"><B>AMEResultReader(String)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/AMEResultReader.html" title="class in broad.core.motif.meme">AMEResultReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/AMEResultReader.html#AMEResultReader(java.io.File)"><B>AMEResultReader(File)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/AMEResultReader.html" title="class in broad.core.motif.meme">AMEResultReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/AMEResultReader.AMEMatch.html" title="class in broad.core.motif.meme"><B>AMEResultReader.AMEMatch</B></A> - Class in <A HREF="../broad/core/motif/meme/package-summary.html">broad.core.motif.meme</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/meme/AMEResultReader.AMEMatch.html#AMEResultReader.AMEMatch(java.lang.String, double, double)"><B>AMEResultReader.AMEMatch(String, double, double)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/AMEResultReader.AMEMatch.html" title="class in broad.core.motif.meme">AMEResultReader.AMEMatch</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#ANALYSIS_NAME"><B>ANALYSIS_NAME</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/general/Predicates.html#and(java.lang.Iterable)"><B>and(Iterable&lt;? extends Predicate&lt;? super T&gt;&gt;)</B></A> - 
Static method in class nextgen.core.general.<A HREF="../nextgen/core/general/Predicates.html" title="class in nextgen.core.general">Predicates</A>
<DD>Returns a predicate that evaluates to <code>true</code> if the object being tested
 passes all component predicates.
<DT><A HREF="../nextgen/core/general/Predicates.html#and(org.apache.commons.collections15.Predicate...)"><B>and(Predicate&lt;? super T&gt;...)</B></A> - 
Static method in class nextgen.core.general.<A HREF="../nextgen/core/general/Predicates.html" title="class in nextgen.core.general">Predicates</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationHandler.html#annotation(broad.core.annotation.GenomicAnnotation)"><B>annotation(GenomicAnnotation)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationHandler.html" title="interface in broad.core.annotation">AnnotationHandler</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#annotation(broad.core.annotation.GenomicAnnotation)"><B>annotation(GenomicAnnotation)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation"><B>Annotation</B></A> - Interface in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/Annotation.Strand.html" title="enum in nextgen.core.annotation"><B>Annotation.Strand</B></A> - Enum in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#ANNOTATION_BASE"><B>ANNOTATION_BASE</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CountReads.html#ANNOTATION_FILE"><B>ANNOTATION_FILE</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CountReads.html" title="class in broad.pda.seq.rap">CountReads</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#ANNOTATION_FILE"><B>ANNOTATION_FILE</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation"><B>AnnotationCollection</B></A>&lt;<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="type parameter in AnnotationCollection">T</A> extends <A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; - Interface in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactory.html" title="interface in broad.core.annotation"><B>AnnotationFactory</B></A>&lt;<A HREF="../broad/core/annotation/AnnotationFactory.html" title="type parameter in AnnotationFactory">T</A> extends <A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>&gt; - Interface in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/AnnotationFactory.html" title="interface in nextgen.core.annotation"><B>AnnotationFactory</B></A>&lt;<A HREF="../nextgen/core/annotation/AnnotationFactory.html" title="type parameter in AnnotationFactory">T</A> extends <A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; - Interface in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.html" title="class in broad.core.annotation"><B>AnnotationFactoryFactory</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.html#AnnotationFactoryFactory()"><B>AnnotationFactoryFactory()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BasicGenomicFactory.html" title="class in broad.core.annotation"><B>AnnotationFactoryFactory.BasicGenomicFactory</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BasicGenomicFactory.html#AnnotationFactoryFactory.BasicGenomicFactory()"><B>AnnotationFactoryFactory.BasicGenomicFactory()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BasicGenomicFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BasicGenomicFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDFactory.html" title="class in broad.core.annotation"><B>AnnotationFactoryFactory.BEDFactory</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDFactory.html#AnnotationFactoryFactory.BEDFactory()"><B>AnnotationFactoryFactory.BEDFactory()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BEDFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDGraphFactory.html" title="class in broad.core.annotation"><B>AnnotationFactoryFactory.BEDGraphFactory</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDGraphFactory.html#AnnotationFactoryFactory.BEDGraphFactory()"><B>AnnotationFactoryFactory.BEDGraphFactory()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDGraphFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BEDGraphFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.GFFFactory.html" title="class in broad.core.annotation"><B>AnnotationFactoryFactory.GFFFactory</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.GFFFactory.html#AnnotationFactoryFactory.GFFFactory()"><B>AnnotationFactoryFactory.GFFFactory()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.GFFFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.GFFFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.RepeatMaskerAnnotationFactory.html" title="class in broad.core.annotation"><B>AnnotationFactoryFactory.RepeatMaskerAnnotationFactory</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.RepeatMaskerAnnotationFactory.html#AnnotationFactoryFactory.RepeatMaskerAnnotationFactory()"><B>AnnotationFactoryFactory.RepeatMaskerAnnotationFactory()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.RepeatMaskerAnnotationFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.RepeatMaskerAnnotationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.ShortBEDFactory.html" title="class in broad.core.annotation"><B>AnnotationFactoryFactory.ShortBEDFactory</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.ShortBEDFactory.html#AnnotationFactoryFactory.ShortBEDFactory()"><B>AnnotationFactoryFactory.ShortBEDFactory()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.ShortBEDFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.ShortBEDFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.WIGFactory.html" title="class in broad.core.annotation"><B>AnnotationFactoryFactory.WIGFactory</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.WIGFactory.html#AnnotationFactoryFactory.WIGFactory()"><B>AnnotationFactoryFactory.WIGFactory()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.WIGFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.WIGFactory</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationFileReader.html" title="class in nextgen.core.annotation"><B>AnnotationFileReader</B></A> - Class in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/AnnotationFileReader.html#AnnotationFileReader()"><B>AnnotationFileReader()</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationFileReader.html" title="class in nextgen.core.annotation">AnnotationFileReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationHandler.html" title="interface in broad.core.annotation"><B>AnnotationHandler</B></A> - Interface in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation"><B>AnnotationList</B></A>&lt;<A HREF="../nextgen/core/annotation/AnnotationList.html" title="type parameter in AnnotationList">T</A> extends <A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; - Class in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#AnnotationList()"><B>AnnotationList()</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#AnnotationList(nextgen.core.coordinatesystem.CoordinateSpace)"><B>AnnotationList(CoordinateSpace)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>Initializing with a coordinate space allows for some additional filtering and sanity checks
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#AnnotationList(nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection)"><B>AnnotationList(CoordinateSpace, Collection&lt;? extends T&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#AnnotationList(nextgen.core.coordinatesystem.CoordinateSpace, T...)"><B>AnnotationList(CoordinateSpace, T...)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation"><B>AnnotationReader</B></A>&lt;<A HREF="../broad/core/annotation/AnnotationReader.html" title="type parameter in AnnotationReader">T</A> extends <A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>&gt; - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationReader.html#AnnotationReader()"><B>AnnotationReader()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#AnnotationReader(java.util.Collection)"><B>AnnotationReader(Collection&lt;T&gt;)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Jesse Engreitz
 September 26, 2012
 Allow for creating a reader from a collection of annotations
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation"><B>AnnotationReader.AnnotationSet</B></A>&lt;<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="type parameter in AnnotationReader.AnnotationSet">T</A> extends <A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>&gt; - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#AnnotationReader.AnnotationSet()"><B>AnnotationReader.AnnotationSet()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.trivialFilter.html" title="class in broad.core.annotation"><B>AnnotationReader.trivialFilter</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationReader.trivialFilter.html#AnnotationReader.trivialFilter()"><B>AnnotationReader.trivialFilter()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.trivialFilter.html" title="class in broad.core.annotation">AnnotationReader.trivialFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReaderFactory.html" title="class in broad.core.annotation"><B>AnnotationReaderFactory</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationReaderFactory.html#AnnotationReaderFactory()"><B>AnnotationReaderFactory()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReaderFactory.html" title="class in broad.core.annotation">AnnotationReaderFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#ANNOTATIONS"><B>ANNOTATIONS</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationUtils.html" title="class in broad.core.annotation"><B>AnnotationUtils</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/AnnotationUtils.html#AnnotationUtils()"><B>AnnotationUtils()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationUtils.html" title="class in broad.core.annotation">AnnotationUtils</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/utils/AnnotationUtils.html" title="class in nextgen.core.utils"><B>AnnotationUtils</B></A> - Class in <A HREF="../nextgen/core/utils/package-summary.html">nextgen.core.utils</A><DD>&nbsp;<DT><A HREF="../nextgen/core/utils/AnnotationUtils.html#AnnotationUtils()"><B>AnnotationUtils()</B></A> - 
Constructor for class nextgen.core.utils.<A HREF="../nextgen/core/utils/AnnotationUtils.html" title="class in nextgen.core.utils">AnnotationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/ANOVA.html" title="class in broad.core.math"><B>ANOVA</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/ANOVA.html#ANOVA(java.util.Collection)"><B>ANOVA(Collection&lt;double[]&gt;)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/ANOVA.html" title="class in broad.core.math">ANOVA</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#anovaFStat(java.util.Collection)"><B>anovaFStat(Collection&lt;double[]&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#append(broad.core.datastructures.MatrixWithHeaders)"><B>append(MatrixWithHeaders)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#append(broad.core.multiplealignment.MAFAlignment)"><B>append(MAFAlignment)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#append(broad.core.multiplealignment.MultipleAlignment)"><B>append(MultipleAlignment)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#append(java.util.List, java.lang.String)"><B>append(List&lt;? extends Sequence&gt;, String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#append(broad.core.sequence.Sequence, java.lang.String)"><B>append(Sequence, String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#appendColumns(broad.core.datastructures.MatrixWithHeaders)"><B>appendColumns(MatrixWithHeaders)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#appendRows(broad.core.datastructures.MatrixWithHeaders)"><B>appendRows(MatrixWithHeaders)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#appendRows(broad.pda.datastructures.LocationAwareMatrix)"><B>appendRows(LocationAwareMatrix)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#appendToSequence(java.lang.String)"><B>appendToSequence(String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#appendToSequence(char)"><B>appendToSequence(char)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html#areThereEmptyClusters()"><B>areThereEmptyClusters()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html" title="class in broad.pda.geneexpression.clustering">Kmeans.KmeansInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html#areThereEmptyClusters(java.util.HashMap)"><B>areThereEmptyClusters(HashMap&lt;String, List&lt;String&gt;&gt;)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html" title="class in broad.pda.geneexpression.clustering">Kmeans.KmeansInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ArrayDesigner.html" title="class in broad.pda.capture.designer"><B>ArrayDesigner</B></A> - Class in <A HREF="../broad/pda/capture/designer/package-summary.html">broad.pda.capture.designer</A><DD>For a given set of sequences, design tiling oligos and primers that do not cross-hybridize with each other or the rest of the transcriptome
 Basic steps:
 1.<DT><A HREF="../broad/pda/capture/designer/ArrayDesignerInSilicoPcrTest.html" title="class in broad.pda.capture.designer"><B>ArrayDesignerInSilicoPcrTest</B></A> - Class in <A HREF="../broad/pda/capture/designer/package-summary.html">broad.pda.capture.designer</A><DD>&nbsp;<DT><A HREF="../broad/pda/capture/designer/ArrayDesignerInSilicoPcrTest.html#ArrayDesignerInSilicoPcrTest(java.lang.String, java.lang.String)"><B>ArrayDesignerInSilicoPcrTest(String, String)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ArrayDesignerInSilicoPcrTest.html" title="class in broad.pda.capture.designer">ArrayDesignerInSilicoPcrTest</A>
<DD>Constructor stores location of array design produced by ArrayDesigner
<DT><A HREF="../Jama/Matrix.html#arrayLeftDivide(Jama.Matrix)"><B>arrayLeftDivide(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Element-by-element left division, C = A.\B
<DT><A HREF="../Jama/Matrix.html#arrayLeftDivideEquals(Jama.Matrix)"><B>arrayLeftDivideEquals(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Element-by-element left division in place, A = A.\B
<DT><A HREF="../Jama/Matrix.html#arrayRightDivide(Jama.Matrix)"><B>arrayRightDivide(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Element-by-element right division, C = A./B
<DT><A HREF="../Jama/Matrix.html#arrayRightDivideEquals(Jama.Matrix)"><B>arrayRightDivideEquals(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Element-by-element right division in place, A = A./B
<DT><A HREF="../Jama/Matrix.html#arrayTimes(Jama.Matrix)"><B>arrayTimes(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Element-by-element multiplication, C = A.*B
<DT><A HREF="../Jama/Matrix.html#arrayTimesEquals(Jama.Matrix)"><B>arrayTimesEquals(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Element-by-element multiplication in place, A = A.*B
<DT><A HREF="../broad/core/parser/StringParser.html#asDouble(int)"><B>asDouble(int)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Gets the token at specified position and parses to a double
<DT><A HREF="../broad/core/parser/StringParser.html#asFloat(int)"><B>asFloat(int)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Gets the token at specified position and parses to a float
<DT><A HREF="../broad/core/parser/StringParser.html#asInt(int)"><B>asInt(int)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Gets the token at specified position and parses to an int
<DT><A HREF="../nextgen/core/model/score/CountScore.html#asRPKM(double, double, int)"><B>asRPKM(double, double, int)</B></A> - 
Static method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture"><B>Assembly</B></A> - Class in <A HREF="../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A><DD>This class represents an assembly<DT><A HREF="../nextgen/core/scripture/Assembly.html#Assembly(nextgen.core.annotation.Annotation)"><B>Assembly(Annotation)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/Assembly.html#Assembly(nextgen.core.annotation.Annotation, boolean)"><B>Assembly(Annotation, boolean)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/Assembly.html#Assembly(java.util.Collection, boolean)"><B>Assembly(Collection&lt;? extends Annotation&gt;, boolean)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/Assembly.html#Assembly(java.util.Collection)"><B>Assembly(Collection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/EndPileUpScanner.html#assign3pPeaksToGenes(java.lang.String)"><B>assign3pPeaksToGenes(String)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/EndPileUpScanner.html" title="class in nextgen.core.scripture">EndPileUpScanner</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/EndPileUpScanner.html#assign5pPeaksToGenes(java.lang.String)"><B>assign5pPeaksToGenes(String)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/EndPileUpScanner.html" title="class in nextgen.core.scripture">EndPileUpScanner</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/utils/AlignmentUtils.html#assignOrientationToWindow(java.lang.String, nextgen.core.annotation.Annotation, boolean, double)"><B>assignOrientationToWindow(String, Annotation, boolean, double)</B></A> - 
Static method in class nextgen.core.utils.<A HREF="../nextgen/core/utils/AlignmentUtils.html" title="class in nextgen.core.utils">AlignmentUtils</A>
<DD>Assign orientation to window based on number of reads mapping in each orientation
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html#assignRandomClusters(broad.core.datastructures.MatrixWithHeaders, int)"><B>assignRandomClusters(MatrixWithHeaders, int)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html" title="class in broad.pda.geneexpression.clustering">Kmeans.KmeansInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/StringParser.html#asString(int)"><B>asString(int)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Gets the token at specified position
<DT><A HREF="../broad/core/math/Statistics.html#average(java.util.Collection)"><B>average(Collection&lt;? extends Number&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#average(double[])"><B>average(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/AverageBranchLength.html" title="class in broad.core.siphy.tools.conservation"><B>AverageBranchLength</B></A> - Class in <A HREF="../broad/core/siphy/tools/conservation/package-summary.html">broad.core.siphy.tools.conservation</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/tools/conservation/AverageBranchLength.html#AverageBranchLength(java.io.File, java.lang.String)"><B>AverageBranchLength(File, String)</B></A> - 
Constructor for class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/AverageBranchLength.html" title="class in broad.core.siphy.tools.conservation">AverageBranchLength</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV LETTER&nbsp;
&nbsp;<A HREF="index-2.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-1.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-1.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
