<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:31 EDT 2013 -->
<TITLE>
S-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="S-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-18.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-20.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-19.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-19.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_S_"><!-- --></A><H2>
<B>S</B></H2>
<DL>
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#saltConcentration"><B>saltConcentration</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/SameOrientationFilter.html" title="class in nextgen.core.readFilters"><B>SameOrientationFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/SameOrientationFilter.html#SameOrientationFilter(nextgen.core.annotation.Annotation)"><B>SameOrientationFilter(Annotation)</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/SameOrientationFilter.html" title="class in nextgen.core.readFilters">SameOrientationFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/AlignmentUtils.html#SAMFormatFullBED(net.sf.samtools.SAMRecord)"><B>SAMFormatFullBED(SAMRecord)</B></A> - 
Static method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/AlignmentUtils.html" title="class in broad.pda.seq.alignment">AlignmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMFragmentWriter.html" title="class in broad.pda.samtools"><B>SAMFragmentWriter</B></A> - Class in <A HREF="../broad/pda/samtools/package-summary.html">broad.pda.samtools</A><DD>&nbsp;<DT><A HREF="../broad/pda/samtools/SAMFragmentWriter.html#SAMFragmentWriter(java.lang.String, java.lang.String)"><B>SAMFragmentWriter(String, String)</B></A> - 
Constructor for class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMFragmentWriter.html" title="class in broad.pda.samtools">SAMFragmentWriter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMFragmentWriter.html#SAMFragmentWriter(java.lang.String, java.lang.String, java.lang.String)"><B>SAMFragmentWriter(String, String, String)</B></A> - 
Constructor for class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMFragmentWriter.html" title="class in broad.pda.samtools">SAMFragmentWriter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools"><B>SAMPairedEndFileReader</B></A> - Class in <A HREF="../broad/pda/samtools/package-summary.html">broad.pda.samtools</A><DD>&nbsp;<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#SAMPairedEndFileReader(java.io.File)"><B>SAMPairedEndFileReader(File)</B></A> - 
Constructor for class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#sample()"><B>sample()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#sample(org.forester.phylogeny.PhylogenyNode, short)"><B>sample(PhylogenyNode, short)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#sampleColumns(int, int)"><B>sampleColumns(int, int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#sampleColumns(int, int)"><B>sampleColumns(int, int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection"><B>SampleData</B></A> - Class in <A HREF="../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/protection/SampleData.html#SampleData(java.lang.String, boolean, java.util.Map, int, int, double, boolean)"><B>SampleData(String, boolean, Map&lt;String, Collection&lt;Gene&gt;&gt;, int, int, double, boolean)</B></A> - 
Constructor for class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#sampleNameFromBamName(java.lang.String)"><B>sampleNameFromBamName(String)</B></A> - 
Static method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get sample name from bam file name
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#SAVED_MODEL_STATS_EXT"><B>SAVED_MODEL_STATS_EXT</B></A> - 
Static variable in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ScaledWindowParserListener.html" title="interface in broad.core.siphy"><B>ScaledWindowParserListener</B></A> - Interface in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#scaleNorm()"><B>scaleNorm()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#scaleRegion(java.util.List, broad.core.annotation.GenomicAnnotation)"><B>scaleRegion(List&lt;String&gt;, GenomicAnnotation)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#scaleRegion(nextgen.core.annotation.Annotation)"><B>scaleRegion(Annotation)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#scaleTree(int, java.io.BufferedWriter, java.util.List, int)"><B>scaleTree(int, BufferedWriter, List&lt;String&gt;, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#scaleTree(int, java.util.List, int)"><B>scaleTree(int, List&lt;String&gt;, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#scaleTreeWithSampling(int, java.io.BufferedWriter, java.util.List, int)"><B>scaleTreeWithSampling(int, BufferedWriter, List&lt;String&gt;, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html#scan(broad.core.multiplealignment.MAFIO, java.util.List, int, float)"><B>scan(MAFIO, List&lt;BED&gt;, int, float)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy">ConservedPWMScanner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scan(int[], double)"><B>scan(int[], double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scan(int[], double, java.lang.String)"><B>scan(int[], double, String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#scan(int, int)"><B>scan(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#scan(int, int, nextgen.core.model.score.WindowProcessor)"><B>scan(int, int, WindowProcessor&lt;W&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>Iterate through the whole coordinate space in windows and score them using the provided window processor
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#scan(nextgen.core.annotation.Annotation, int, int, nextgen.core.model.score.WindowProcessor)"><B>scan(Annotation, int, int, WindowProcessor&lt;W&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#scan(nextgen.core.annotation.Annotation, int, int)"><B>scan(Annotation, int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#scan(int, int)"><B>scan(int, int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#scan(int, int, nextgen.core.model.score.WindowProcessor)"><B>scan(int, int, WindowProcessor&lt;W&gt;)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#scan(nextgen.core.annotation.Annotation, int, int, nextgen.core.model.score.WindowProcessor)"><B>scan(Annotation, int, int, WindowProcessor&lt;W&gt;)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#scan(nextgen.core.annotation.Annotation, int, int)"><B>scan(Annotation, int, int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#scan(java.util.Collection, int, int, nextgen.core.model.score.WindowProcessor)"><B>scan(Collection&lt;Gene&gt;, int, int, WindowProcessor&lt;T&gt;)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>Scan windows over a Collection<GeneWindow>
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#scan(nextgen.core.annotation.Annotation, int, int, nextgen.core.model.score.WindowProcessor)"><B>scan(Annotation, int, int, WindowProcessor&lt;T&gt;)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Scan windows over a Collection<GeneWindow> and score
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#scan(nextgen.core.annotation.Annotation, int, int)"><B>scan(Annotation, int, int)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#scan(java.lang.String, int, int)"><B>scan(String, int, int)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>Scan across windows and return basic level statistics
<DT><A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html#scan(nextgen.core.annotation.Gene, int, int, nextgen.core.model.score.WindowProcessor)"><B>scan(Gene, int, int, WindowProcessor&lt;T&gt;)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A>
<DD>Scan windows over a gene and score
 Only scan exons of this gene
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#scanAll(int, int)"><B>scanAll(int, int)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>Iterate through the whole coordinate space in windows and score them using the count function
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanGenome(broad.pda.seq.segmentation.ContinuousDataAlignmentModel, int, java.lang.String, java.io.FileWriter, boolean)"><B>scanGenome(ContinuousDataAlignmentModel, int, String, FileWriter, boolean)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Gene, int)"><B>scanPRate(Gene, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(broad.core.annotation.LightweightGenomicAnnotation, int)"><B>scanPRate(LightweightGenomicAnnotation, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int)"><B>scanPRate(Gene, IntervalTree&lt;Alignment&gt;, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, double, int)"><B>scanPRate(Annotation, IntervalTree&lt;Alignment&gt;, double, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Annotation, double, int)"><B>scanPRate(Annotation, double, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Annotation, double, double, int)"><B>scanPRate(Annotation, double, double, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)"><B>scanPRate(Gene, IntervalTree&lt;Alignment&gt;, double, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int, double)"><B>scanPRate(Gene, IntervalTree&lt;Alignment&gt;, double, int, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene)"><B>scanPRate(Gene)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><B>scanPRate(Gene, IntervalTree&lt;Alignment&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double)"><B>scanPRate(Gene, IntervalTree&lt;Alignment&gt;, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)"><B>scanPRate(Gene, IntervalTree&lt;Alignment&gt;, double, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PWMUtils.html#scanRegion(int, java.lang.String, double, boolean, java.lang.String, java.util.List, java.lang.String, nextgen.core.annotation.Annotation, double[], short[])"><B>scanRegion(int, String, double, boolean, String, List&lt;PositionWeightMatrix&gt;, String, Annotation, double[], short[])</B></A> - 
Static method in class broad.core.motif.<A HREF="../broad/core/motif/PWMUtils.html" title="class in broad.core.motif">PWMUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#scanRegion(int, broad.core.annotation.LightweightGenomicAnnotation)"><B>scanRegion(int, LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanRegion(int, broad.core.annotation.LightweightGenomicAnnotation)"><B>scanRegion(int, LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model"><B>ScanStatisticDataAlignmentModel</B></A> - Class in <A HREF="../nextgen/core/model/package-summary.html">nextgen.core.model</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#ScanStatisticDataAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection)"><B>ScanStatisticDataAlignmentModel(String, CoordinateSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#ScanStatisticDataAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection, boolean)"><B>ScanStatisticDataAlignmentModel(String, CoordinateSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, boolean)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#ScanStatisticDataAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace)"><B>ScanStatisticDataAlignmentModel(String, CoordinateSpace)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#ScanStatisticDataAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, boolean)"><B>ScanStatisticDataAlignmentModel(String, CoordinateSpace, boolean)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#ScanStatisticDataAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>ScanStatisticDataAlignmentModel(String, CoordinateSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#ScanStatisticDataAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, java.util.Collection, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>ScanStatisticDataAlignmentModel(String, CoordinateSpace, Collection&lt;Predicate&lt;Alignment&gt;&gt;, boolean, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#ScanStatisticDataAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>ScanStatisticDataAlignmentModel(String, CoordinateSpace, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#ScanStatisticDataAlignmentModel(java.lang.String, nextgen.core.coordinatesystem.CoordinateSpace, boolean, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>ScanStatisticDataAlignmentModel(String, CoordinateSpace, boolean, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score"><B>ScanStatisticScore</B></A> - Class in <A HREF="../nextgen/core/model/score/package-summary.html">nextgen.core.model.score</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#ScanStatisticScore(nextgen.core.model.AlignmentModel, nextgen.core.annotation.Annotation)"><B>ScanStatisticScore(AlignmentModel, Annotation)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#ScanStatisticScore(nextgen.core.model.AlignmentModel, nextgen.core.annotation.Annotation, nextgen.core.model.score.ScanStatisticScore, double)"><B>ScanStatisticScore(AlignmentModel, Annotation, ScanStatisticScore, double)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#ScanStatisticScore(nextgen.core.model.AlignmentModel, nextgen.core.annotation.Annotation, double, double)"><B>ScanStatisticScore(AlignmentModel, Annotation, double, double)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#ScanStatisticScore(nextgen.core.model.AlignmentModel, nextgen.core.annotation.Annotation, double, double, double, double)"><B>ScanStatisticScore(AlignmentModel, Annotation, double, double, double, double)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>Control region length and total, global length and total
 Scan P value depends on global stats
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.Processor.html" title="class in nextgen.core.model.score"><B>ScanStatisticScore.Processor</B></A> - Class in <A HREF="../nextgen/core/model/score/package-summary.html">nextgen.core.model.score</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.Processor.html#ScanStatisticScore.Processor(nextgen.core.model.AlignmentModel)"><B>ScanStatisticScore.Processor(AlignmentModel)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.Processor.html" title="class in nextgen.core.model.score">ScanStatisticScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PositionWeightMatrixModel.html#score(java.util.Map)"><B>score(Map&lt;String, Matrix&gt;)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PositionWeightMatrixModel.html" title="class in broad.core.siphy">PositionWeightMatrixModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html#SCORE"><B>SCORE</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap">GenomeScoringProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#SCORE"><B>SCORE</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/AlignmentUtils.html#scoreAnnotations(broad.core.annotation.BEDReader, broad.pda.seq.segmentation.ContinuousDataAlignmentModel)"><B>scoreAnnotations(BEDReader, ContinuousDataAlignmentModel)</B></A> - 
Static method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/AlignmentUtils.html" title="class in broad.pda.seq.alignment">AlignmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ScoreDiffGenes.html" title="class in broad.pda.geneexpression"><B>ScoreDiffGenes</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/ScoreDiffGenes.html#ScoreDiffGenes()"><B>ScoreDiffGenes()</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ScoreDiffGenes.html" title="class in broad.pda.geneexpression">ScoreDiffGenes</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene)"><B>scoreGene(Gene)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, double)"><B>scoreGene(Gene, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double)"><B>scoreGene(Gene, IntervalTree&lt;Alignment&gt;, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)"><B>scoreGene(Gene, IntervalTree&lt;Alignment&gt;, double, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><B>scoreGene(Gene, IntervalTree&lt;Alignment&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String, broad.core.datastructures.IntervalTree)"><B>scoreGenes(Collection&lt;Gene&gt;, String, IntervalTree&lt;Alignment&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection)"><B>scoreGenes(Collection&lt;Gene&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String)"><B>scoreGenes(Collection&lt;Gene&gt;, String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String, int)"><B>scoreGenes(Collection&lt;Gene&gt;, String, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(broad.pda.annotation.BEDFileParser)"><B>scoreGenes(BEDFileParser)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ScoreDiffGenes.html#scoreListByTStatistic(broad.core.datastructures.MatrixWithHeaders, java.util.Map, java.lang.String, double)"><B>scoreListByTStatistic(MatrixWithHeaders, Map&lt;String, ArrayList&gt;, String, double)</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ScoreDiffGenes.html" title="class in broad.pda.geneexpression">ScoreDiffGenes</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/misc/ScoreMotifInstances.html" title="class in broad.core.motif.misc"><B>ScoreMotifInstances</B></A> - Class in <A HREF="../broad/core/motif/misc/package-summary.html">broad.core.motif.misc</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/misc/ScoreMotifInstances.html#ScoreMotifInstances(java.lang.String, java.lang.String, broad.pda.datastructures.Alignments, java.lang.String, java.lang.String)"><B>ScoreMotifInstances(String, String, Alignments, String, String)</B></A> - 
Constructor for class broad.core.motif.misc.<A HREF="../broad/core/motif/misc/ScoreMotifInstances.html" title="class in broad.core.motif.misc">ScoreMotifInstances</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/misc/ComputeConservedMotifs.html#scoreMotifs(broad.pda.datastructures.Alignments, broad.core.siphy.TreeScaler, java.lang.String, java.lang.String, java.lang.String, boolean)"><B>scoreMotifs(Alignments, TreeScaler, String, String, String, boolean)</B></A> - 
Static method in class broad.core.motif.misc.<A HREF="../broad/core/motif/misc/ComputeConservedMotifs.html" title="class in broad.core.motif.misc">ComputeConservedMotifs</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scorePaths(java.util.Collection, double)"><B>scorePaths(Collection&lt;Path&gt;, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scorePaths(java.util.Collection, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT, double)"><B>scorePaths(Collection&lt;Path&gt;, ChromosomeWithBubblesJGraphT, double)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scorePaths(java.util.Collection, double, double, double, double, java.util.Map, double)"><B>scorePaths(Collection&lt;Path&gt;, double, double, double, double, Map&lt;Annotation, Double&gt;, double)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scorePathsForOnlyRPKM(java.util.Collection)"><B>scorePathsForOnlyRPKM(Collection&lt;Path&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreSegment(nextgen.core.annotation.Annotation)"><B>scoreSegment(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreSegments(java.util.Collection)"><B>scoreSegments(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreSegments(java.util.Collection, java.lang.String)"><B>scoreSegments(Collection&lt;Annotation&gt;, String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreSegments(java.util.Collection, java.lang.String, boolean)"><B>scoreSegments(Collection&lt;Annotation&gt;, String, boolean)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#scoreWindow(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)"><B>scoreWindow(Gene, Annotation)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get scan statistic score for a specified window
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#scoreWindow(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation, double)"><B>scoreWindow(Gene, Annotation, double)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get scan statistic score for a specified window
<DT><A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html#scoreWindow(nextgen.core.annotation.Annotation)"><B>scoreWindow(Annotation)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/ScanStatisticDataAlignmentModel.html" title="class in nextgen.core.model">ScanStatisticDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#scoreWindows(java.util.Collection)"><B>scoreWindows(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html#scoreWindows(java.util.Collection)"><B>scoreWindows(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RAPPeakCaller.html#scoreWindows(java.util.Collection)"><B>scoreWindows(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RAPPeakCaller.html" title="class in broad.pda.seq.rap">RAPPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/analysis/PeakCaller.html#scoreWindows(java.util.Collection)"><B>scoreWindows(Collection&lt;Annotation&gt;)</B></A> - 
Method in interface nextgen.core.analysis.<A HREF="../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A>
<DD>Score windows
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#scoreWindowWithFragmentLengthFilter(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)"><B>scoreWindowWithFragmentLengthFilter(Gene, Annotation)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Get scan statistic score for a window based on filtering fragments overlapping the window by size
<DT><A HREF="../broad/core/sequence/Sequence.html#scramble()"><B>scramble()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SearchException.html" title="class in broad.core.motif"><B>SearchException</B></A> - Exception in <A HREF="../broad/core/motif/package-summary.html">broad.core.motif</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/SearchException.html#SearchException()"><B>SearchException()</B></A> - 
Constructor for exception broad.core.motif.<A HREF="../broad/core/motif/SearchException.html" title="class in broad.core.motif">SearchException</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SearchException.html#SearchException(java.lang.String)"><B>SearchException(String)</B></A> - 
Constructor for exception broad.core.motif.<A HREF="../broad/core/motif/SearchException.html" title="class in broad.core.motif">SearchException</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SearchException.html#SearchException(java.lang.String, java.lang.Throwable)"><B>SearchException(String, Throwable)</B></A> - 
Constructor for exception broad.core.motif.<A HREF="../broad/core/motif/SearchException.html" title="class in broad.core.motif">SearchException</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SearchException.html#SearchException(java.lang.Throwable)"><B>SearchException(Throwable)</B></A> - 
Constructor for exception broad.core.motif.<A HREF="../broad/core/motif/SearchException.html" title="class in broad.core.motif">SearchException</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/SecondOfPairFilter.html" title="class in nextgen.core.readFilters"><B>SecondOfPairFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/SecondOfPairFilter.html#SecondOfPairFilter()"><B>SecondOfPairFilter()</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/SecondOfPairFilter.html" title="class in nextgen.core.readFilters">SecondOfPairFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html#selectHighScoringIsoform(nextgen.core.annotation.Gene, broad.pda.annotation.BEDFileParser)"><B>selectHighScoringIsoform(Gene, BEDFileParser)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#SelectLongestIntronChainCandidate(java.util.Collection)"><B>SelectLongestIntronChainCandidate(Collection&lt;Gene&gt;)</B></A> - 
Static method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#selectRandIsoSubset(java.lang.Integer)"><B>selectRandIsoSubset(Integer)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#selectRandIsoSubset(java.lang.Integer)"><B>selectRandIsoSubset(Integer)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#selfAnyAlignScore"><B>selfAnyAlignScore</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#selfEndAlignScore"><B>selfEndAlignScore</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityBinaryIO.html#SEQ_ID_REC_LENGTH"><B>SEQ_ID_REC_LENGTH</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityBinaryIO.html" title="class in broad.core.sequence">FastaQualityBinaryIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence"><B>Sequence</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/Sequence.html#Sequence(java.lang.String)"><B>Sequence(String)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#Sequence(java.lang.String, boolean)"><B>Sequence(String, boolean)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#Sequence(java.lang.String, int)"><B>Sequence(String, int)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#sequence"><B>sequence</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html" title="class in jaligner"><B>Sequence</B></A> - Class in <A HREF="../jaligner/package-summary.html">jaligner</A><DD>A basic (nucleic or protein) sequence.<DT><A HREF="../jaligner/Sequence.html#Sequence()"><B>Sequence()</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Constructor
<DT><A HREF="../jaligner/Sequence.html#Sequence(java.lang.String)"><B>Sequence(String)</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Constructor
<DT><A HREF="../jaligner/Sequence.html#Sequence(java.lang.String, char[])"><B>Sequence(String, char[])</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Constructor
<DT><A HREF="../jaligner/Sequence.html#Sequence(java.lang.String, java.lang.String)"><B>Sequence(String, String)</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Constructor
<DT><A HREF="../jaligner/Sequence.html#Sequence(java.lang.String, java.lang.String, java.lang.String, int)"><B>Sequence(String, String, String, int)</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Constructor
<DT><A HREF="../broad/core/motif/SequenceMotif.html" title="class in broad.core.motif"><B>SequenceMotif</B></A> - Class in <A HREF="../broad/core/motif/package-summary.html">broad.core.motif</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/SequenceMotif.html#SequenceMotif(java.util.regex.Pattern)"><B>SequenceMotif(Pattern)</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotif.html" title="class in broad.core.motif">SequenceMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SequenceMotif.html#SequenceMotif(java.lang.String, int)"><B>SequenceMotif(String, int)</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotif.html" title="class in broad.core.motif">SequenceMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SequenceMotifCluster.html" title="class in broad.core.motif"><B>SequenceMotifCluster</B></A> - Class in <A HREF="../broad/core/motif/package-summary.html">broad.core.motif</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/SequenceMotifCluster.html#SequenceMotifCluster(java.lang.String, int, int)"><B>SequenceMotifCluster(String, int, int)</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotifCluster.html" title="class in broad.core.motif">SequenceMotifCluster</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/SequenceParser.html" title="class in jaligner.util"><B>SequenceParser</B></A> - Class in <A HREF="../jaligner/util/package-summary.html">jaligner.util</A><DD>SequenceParser to sequences from different formats.<DT><A HREF="../jaligner/util/SequenceParser.html#SequenceParser()"><B>SequenceParser()</B></A> - 
Constructor for class jaligner.util.<A HREF="../jaligner/util/SequenceParser.html" title="class in jaligner.util">SequenceParser</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/SequenceParserException.html" title="class in jaligner.util"><B>SequenceParserException</B></A> - Exception in <A HREF="../jaligner/util/package-summary.html">jaligner.util</A><DD>&nbsp;<DT><A HREF="../jaligner/util/SequenceParserException.html#SequenceParserException(java.lang.String)"><B>SequenceParserException(String)</B></A> - 
Constructor for exception jaligner.util.<A HREF="../jaligner/util/SequenceParserException.html" title="class in jaligner.util">SequenceParserException</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence"><B>SequenceRegion</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/SequenceRegion.html#SequenceRegion(java.lang.String)"><B>SequenceRegion(String)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#SequenceRegion(java.lang.String, java.lang.String, int, int)"><B>SequenceRegion(String, String, int, int)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#SequenceRegion(java.lang.String, nextgen.core.annotation.Annotation)"><B>SequenceRegion(String, Annotation)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegionWindow.html" title="class in broad.core.sequence"><B>SequenceRegionWindow</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/SequenceRegionWindow.html#SequenceRegionWindow(broad.core.sequence.SequenceRegion)"><B>SequenceRegionWindow(SequenceRegion)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegionWindow.html" title="class in broad.core.sequence">SequenceRegionWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceUtils.html" title="class in broad.core.sequence"><B>SequenceUtils</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/SequenceUtils.html#SequenceUtils()"><B>SequenceUtils()</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceUtils.html" title="class in broad.core.sequence">SequenceUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#set(int, int, double)"><B>set(int, int, double)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#set(java.lang.String, java.lang.String, double)"><B>set(String, String, double)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#set(java.lang.String, int, double)"><B>set(String, int, double)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#set(int, java.lang.String, double)"><B>set(int, String, double)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html#set(int, int, double)"><B>set(int, int, double)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html" title="class in broad.core.hmm">MarkovModel.BackwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html#set(int, int, double)"><B>set(int, int, double)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="class in broad.core.hmm">MarkovModel.ForwardResult</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Cell.html#set(int, int, float)"><B>set(int, int, float)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Cell.html" title="class in jaligner">Cell</A>
<DD>Sets the row, column and score of the cell.
<DT><A HREF="../Jama/Matrix.html#set(int, int, double)"><B>set(int, int, double)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Set a single element.
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#setA(broad.core.annotation.LightweightGenomicAnnotation)"><B>setA(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#setAlignment(java.lang.String, java.lang.String)"><B>setAlignment(String, String)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#setAlignment(broad.core.multiplealignment.MultipleAlignment)"><B>setAlignment(MultipleAlignment)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html#setAlignmentChunkSize(int)"><B>setAlignmentChunkSize(int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy">ConservedPWMScanner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#setAlignmentDataModel(broad.pda.seq.segmentation.AlignmentDataModel)"><B>setAlignmentDataModel(AlignmentDataModel)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#setAlignmentInfoFromRawData(java.lang.String[])"><B>setAlignmentInfoFromRawData(String[])</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#setAlignPos(int)"><B>setAlignPos(int)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setAlpha(double)"><B>setAlpha(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#setAlphabetSize(int)"><B>setAlphabetSize(int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#setAnnotationTreeMap(java.util.Map)"><B>setAnnotationTreeMap(Map&lt;String, IntervalTree&lt;T&gt;&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#setAttribute(java.lang.String, java.lang.String)"><B>setAttribute(String, String)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Sets the specified attribute value
<DT><A HREF="../broad/core/annotation/GFF.html#setAttributes(java.lang.String)"><B>setAttributes(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#setAvergeNumberOfReads(double)"><B>setAvergeNumberOfReads(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html#setB(broad.core.annotation.LightweightGenomicAnnotation)"><B>setB(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicTwoSubjectAnnotation.html" title="class in broad.core.annotation">BasicTwoSubjectAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html#setBackgroundModel(broad.core.annotation.AnnotationReader, java.util.Map)"><B>setBackgroundModel(AnnotationReader&lt;? extends GenomicAnnotation&gt;, Map&lt;String, Sequence&gt;)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy">ConservedPWMScanner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#setBackgroundNucleotideFreqs(Jama.Matrix)"><B>setBackgroundNucleotideFreqs(Matrix)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#setBedFile(java.lang.String)"><B>setBedFile(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#setBedScore(double)"><B>setBedScore(double)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#setBedScore(double)"><B>setBedScore(double)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#setBlastHit(boolean)"><B>setBlastHit(boolean)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#setBlocks(java.util.List)"><B>setBlocks(List&lt;MAFAlignment.MAFMultipleAlignmentBlock&gt;)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#setBlockSizeStarts(int[][])"><B>setBlockSizeStarts(int[][])</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#setBoundariesFromAnnoations(java.util.List)"><B>setBoundariesFromAnnoations(List&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html#setBowtie2Path(java.lang.String)"><B>setBowtie2Path(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser">ConfigFileParser.Bowtie2Options</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#setCapacity(int)"><B>setCapacity(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#setCDS(nextgen.core.annotation.Annotation)"><B>setCDS(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#setCDSEnd(int)"><B>setCDSEnd(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#setCDSRegion(int, int)"><B>setCDSRegion(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Set coding region coordinates (includes start codon but not stop codon)
<DT><A HREF="../nextgen/core/annotation/Gene.html#setCDSStart(int)"><B>setCDSStart(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#setCharAt(int, char)"><B>setCharAt(int, char)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#setChromosome(java.lang.String)"><B>setChromosome(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#setChromosome(java.lang.String)"><B>setChromosome(String)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setChromosome(java.lang.String)"><B>setChromosome(String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setChromosomeLength(int)"><B>setChromosomeLength(int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#setChunkSize(int)"><B>setChunkSize(int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#setChunkSize(int)"><B>setChunkSize(int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setChunkSize(int)"><B>setChunkSize(int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html#setClusterDistanceFunction(broad.pda.geneexpression.clustering.ClusterDistanceFunction)"><B>setClusterDistanceFunction(ClusterDistanceFunction)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html" title="class in broad.pda.geneexpression.clustering">HierarchicalClustering</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html#setClusterDistanceFunction(java.lang.String)"><B>setClusterDistanceFunction(String)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html" title="class in broad.pda.geneexpression.clustering">HierarchicalClustering</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Cell.html#setCol(int)"><B>setCol(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Cell.html" title="class in jaligner">Cell</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#setColumn(double[], java.lang.String)"><B>setColumn(double[], String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#setColumn(double[], int)"><B>setColumn(double[], int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#setColumn(int, double[])"><B>setColumn(int, double[])</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#setColumns(java.util.List)"><B>setColumns(List&lt;String&gt;)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/Assembly.html#setConfident(boolean)"><B>setConfident(boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#setConfigurationUsed(broad.core.primer3.Primer3Configuration)"><B>setConfigurationUsed(Primer3Configuration)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setCoordinateSpace(nextgen.core.coordinatesystem.CoordinateSpace)"><B>setCoordinateSpace(CoordinateSpace)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html#setCoordinateSpace(nextgen.core.coordinatesystem.CoordinateSpace)"><B>setCoordinateSpace(CoordinateSpace)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RAPPeakCaller.html#setCoordinateSpace(nextgen.core.coordinatesystem.CoordinateSpace)"><B>setCoordinateSpace(CoordinateSpace)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RAPPeakCaller.html" title="class in broad.pda.seq.rap">RAPPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/analysis/PeakCaller.html#setCoordinateSpace(nextgen.core.coordinatesystem.CoordinateSpace)"><B>setCoordinateSpace(CoordinateSpace)</B></A> - 
Method in interface nextgen.core.analysis.<A HREF="../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A>
<DD>Set the coordinate space
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#setCount(double)"><B>setCount(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Sets the counts for this edge.
<DT><A HREF="../nextgen/core/model/score/CountScore.html#setCount(double)"><B>setCount(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#setCountNormalizer(broad.pda.seq.segmentation.ReadCountNormalizer)"><B>setCountNormalizer(ReadCountNormalizer)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#setCountScore(double)"><B>setCountScore(double)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#setCountScore(double)"><B>setCountScore(double)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#setDataDimensions(int, int)"><B>setDataDimensions(int, int)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#setDenominatorCount(double)"><B>setDenominatorCount(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#setDenominatorCount(double)"><B>setDenominatorCount(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#setDenominatorCount(double)"><B>setDenominatorCount(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#setDenominatorRegionTotal(double)"><B>setDenominatorRegionTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#setDenominatorRegionTotal(double)"><B>setDenominatorRegionTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#setDenominatorRegionTotal(double)"><B>setDenominatorRegionTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#setDenominatorScanPVal(double)"><B>setDenominatorScanPVal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#setDenominatorTotal(double)"><B>setDenominatorTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#setDenominatorTotal(double)"><B>setDenominatorTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#setDenominatorTotal(double)"><B>setDenominatorTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Sets the sequence description
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#setdgePath(java.lang.String)"><B>setdgePath(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#setdgeType(java.lang.String)"><B>setdgeType(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/TracebackNode.html#setDiagonal()"><B>setDiagonal()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/TracebackNode.html" title="class in jaligner">TracebackNode</A>
<DD>Sets the direction to diagonal
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#setDuplicateFlag(boolean)"><B>setDuplicateFlag(boolean)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Sets the duplicate flag
<DT><A HREF="../nextgen/core/alignment/Alignment.html#setDuplicateFlag(boolean)"><B>setDuplicateFlag(boolean)</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Sets the isDuplicate Flag
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#setDuplicateFlag(boolean)"><B>setDuplicateFlag(boolean)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Sets the duplicate flag
<DT><A HREF="../broad/core/siphy/PiHMM.PiStateModel.html#setEmissionProbOfUnalignRegion(double)"><B>setEmissionProbOfUnalignRegion(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.PiStateModel.html" title="class in broad.core.siphy">PiHMM.PiStateModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#setEnd(int)"><B>setEnd(int)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setEnd(int)"><B>setEnd(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setEnd(int)"><B>setEnd(int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Sets the end of this graph
<DT><A HREF="../nextgen/core/annotation/Annotation.html#setEnd(int)"><B>setEnd(int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Sets a new end position.
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#setEnd(int)"><B>setEnd(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#setEnd(int)"><B>setEnd(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>Extends or trims the end coordinate.
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#setEnd(int)"><B>setEnd(int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.html#setEndStateTransitionProbability(int, double)"><B>setEndStateTransitionProbability(int, double)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm">MarkovModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#setEnrichment(double)"><B>setEnrichment(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#setEstimationWindow(int)"><B>setEstimationWindow(int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setExonScores(nextgen.core.annotation.Annotation, java.lang.Double)"><B>setExonScores(Annotation, Double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setExpressionScanPvalueCutoff(double)"><B>setExpressionScanPvalueCutoff(double)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Set genome wide scan P value cutoff for expression of transcript
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#setExpressionScanPvalueCutoff(double)"><B>setExpressionScanPvalueCutoff(double)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Set genome wide scan P value cutoff for expression of transcript
<DT><A HREF="../jaligner/Alignment.html#setExtend(float)"><B>setExtend(float)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#setExtensionFactor(int)"><B>setExtensionFactor(int)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#setExtensionFactor(int)"><B>setExtensionFactor(int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setExtensionFactor(int)"><B>setExtensionFactor(int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setExtensionFactor(int)"><B>setExtensionFactor(int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#setExtraFields(double[])"><B>setExtraFields(double[])</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#setExtraFields(double[])"><B>setExtraFields(double[])</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#setExtraFields(double, int)"><B>setExtraFields(double, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#setFastaSeqId(java.lang.String)"><B>setFastaSeqId(String)</B></A> - 
Method in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing">CRISPRDesigner.CRISPRTarget</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#setFeature(java.lang.String)"><B>setFeature(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#setFields(java.lang.String, java.lang.String, java.lang.String)"><B>setFields(String, String, String)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setFilterByStrand(boolean)"><B>setFilterByStrand(boolean)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Set whether to use strand information in reads to filter peaks
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setFindMaxContiguous(boolean)"><B>setFindMaxContiguous(boolean)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#setFirstRead()"><B>setFirstRead()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setFirstRead()"><B>setFirstRead()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setFirstReadTranscriptionStrand(boolean)"><B>setFirstReadTranscriptionStrand(boolean)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Set whether read 1 is transcription strand
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#setFivePrimeBuffer(int)"><B>setFivePrimeBuffer(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#setFragmentFlag(boolean)"><B>setFragmentFlag(boolean)</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#setFragmentStrand(nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>setFragmentStrand(AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>Sets the strand for the fragment to the strand of the transcription read
<DT><A HREF="../nextgen/core/alignment/Alignment.html#setFragmentStrand(nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>setFragmentStrand(AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#setFragmentStrand(nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>setFragmentStrand(AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Sets the strand for the fragment to the strand passed as argument
<DT><A HREF="../broad/core/annotation/GFF.html#setFrame(int)"><B>setFrame(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#setFullyContainedNumberOfReads(double)"><B>setFullyContainedNumberOfReads(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#setFullyContainedNumberOfReads(double)"><B>setFullyContainedNumberOfReads(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setGaps(int)"><B>setGaps(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#setGene(nextgen.core.annotation.Gene)"><B>setGene(Gene)</B></A> - 
Method in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing">CRISPRDesigner.CRISPRTarget</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#setGeneLength(double)"><B>setGeneLength(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#setGlobalLength(double)"><B>setGlobalLength(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#setGroup(java.lang.String)"><B>setGroup(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#setId(java.lang.String)"><B>setId(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#setId(java.lang.String)"><B>setId(String)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#setId(java.lang.String)"><B>setId(String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setId(java.lang.String)"><B>setId(String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/formats/Format.html#setId(java.lang.String)"><B>setId(String)</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/Format.html" title="class in jaligner.formats">Format</A>
<DD>Sets format id
<DT><A HREF="../jaligner/Sequence.html#setId(java.lang.String)"><B>setId(String)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Sets the sequence id
<DT><A HREF="../jaligner/Alignment.html#setIdentity(int)"><B>setIdentity(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html#setIgnoreList(java.util.List)"><B>setIgnoreList(List&lt;String&gt;)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy">ConservedPWMScanner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#setInfoFromRawData(java.lang.String)"><B>setInfoFromRawData(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.html#setInitialStateTransitionProbability(int, double)"><B>setInitialStateTransitionProbability(int, double)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm">MarkovModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#setIOHelper(broad.core.multiplealignment.MultipleAlignmentIO)"><B>setIOHelper(MultipleAlignmentIO)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#setIsFirstMate(boolean)"><B>setIsFirstMate(boolean)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>Will set the value of the isFirstMate flag
<DT><A HREF="../broad/pda/seq/graph/Path.html#setIsToComplex(boolean)"><B>setIsToComplex(boolean)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/Commons.html#setJnlp(boolean)"><B>setJnlp(boolean)</B></A> - 
Static method in class jaligner.util.<A HREF="../jaligner/util/Commons.html" title="class in jaligner.util">Commons</A>
<DD>Sets the JNLP flag to true of false
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#setLambda(java.lang.String, double)"><B>setLambda(String, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/TracebackNode.html#setLeft(int)"><B>setLeft(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/TracebackNode.html" title="class in jaligner">TracebackNode</A>
<DD>Sets the direction to left
<DT><A HREF="../broad/core/primer3/PrimerPair.html#setLeftPrimerPosition(int)"><B>setLeftPrimerPosition(int)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityIndex.SequenceFileLayout.html#setLength(int)"><B>setLength(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityIndex.SequenceFileLayout.html" title="class in broad.core.sequence">FastaQualityIndex.SequenceFileLayout</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#setLine(java.lang.String)"><B>setLine(String)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html#setLinkage(java.lang.String)"><B>setLinkage(String)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html" title="class in broad.pda.geneexpression.clustering">HierarchicalClustering</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#setLocalLambda(double)"><B>setLocalLambda(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#setLocalLambda(double)"><B>setLocalLambda(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setLocalRate(java.util.Collection)"><B>setLocalRate(Collection&lt;Path&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setLocalRate(java.util.Map)"><B>setLocalRate(Map&lt;Annotation, Double&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#setLogOdds(double)"><B>setLogOdds(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#setLogOddsPValue(double)"><B>setLogOddsPValue(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setMarkupLine(char[])"><B>setMarkupLine(char[])</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setMaskFileData(java.util.Map)"><B>setMaskFileData(Map&lt;String, Integer&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#setMatrix(java.util.List)"><B>setMatrix(List&lt;double[]&gt;)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setMatrix(jaligner.matrix.Matrix)"><B>setMatrix(Matrix)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#setMatrix(int, int, int, int, Jama.Matrix)"><B>setMatrix(int, int, int, int, Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Set a submatrix.
<DT><A HREF="../Jama/Matrix.html#setMatrix(int[], int[], Jama.Matrix)"><B>setMatrix(int[], int[], Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Set a submatrix.
<DT><A HREF="../Jama/Matrix.html#setMatrix(int[], int, int, Jama.Matrix)"><B>setMatrix(int[], int, int, Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Set a submatrix.
<DT><A HREF="../Jama/Matrix.html#setMatrix(int, int, int[], Jama.Matrix)"><B>setMatrix(int, int, int[], Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Set a submatrix.
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#setMatrixFromRawData(java.util.List)"><B>setMatrixFromRawData(List&lt;String[]&gt;)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#setMaxAllowableInsert(int)"><B>setMaxAllowableInsert(int)</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#setMaxFragmentSize(int)"><B>setMaxFragmentSize(int)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#setMaxGenomicSpan(int)"><B>setMaxGenomicSpan(int)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setMinContguousSegmentSize(int)"><B>setMinContguousSegmentSize(int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#setMinimumMappingQuality(double)"><B>setMinimumMappingQuality(double)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#setMinimumMappingQuality(double)"><B>setMinimumMappingQuality(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setMinimumMappingQuality(double)"><B>setMinimumMappingQuality(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setMinimumMappingQuality(double)"><B>setMinimumMappingQuality(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#setMinimumTreeLength(double)"><B>setMinimumTreeLength(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setMinNumberOfSplices(int)"><B>setMinNumberOfSplices(int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#setMirScore(double)"><B>setMirScore(double)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html#setModel(broad.core.siphy.EvolutionaryModel)"><B>setModel(EvolutionaryModel)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy">ConservedPWMScanner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#setModel(broad.core.siphy.EvolutionaryModel)"><B>setModel(EvolutionaryModel)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#setModelParameters(broad.core.siphy.EvolutionaryModelParameters)"><B>setModelParameters(EvolutionaryModelParameters)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setName1(java.lang.String)"><B>setName1(String)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setName2(java.lang.String)"><B>setName2(String)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#setNegativeStranded()"><B>setNegativeStranded()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setNegativeStranded()"><B>setNegativeStranded()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#setNeutralModel(java.io.File)"><B>setNeutralModel(File)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#setNeutralOmegaDistribution(java.lang.String)"><B>setNeutralOmegaDistribution(String)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#setNominalPValue(double)"><B>setNominalPValue(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#setNormalizationStrategy(broad.pda.seq.segmentation.ReadCountNormalizer)"><B>setNormalizationStrategy(ReadCountNormalizer)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setNormalizationStrategy(broad.pda.seq.segmentation.ReadCountNormalizer)"><B>setNormalizationStrategy(ReadCountNormalizer)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#setNovoalignPath(java.lang.String)"><B>setNovoalignPath(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Set path to novoalign executable
<DT><A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html#setNovoalignPath(java.lang.String)"><B>setNovoalignPath(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html" title="class in broad.core.parser">ConfigFileParser.NovoalignOptions</A>
<DD>Set path to novoalign executable
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#setNum(int[])"><B>setNum(int[])</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#setNum(int, int, int, int)"><B>setNum(int, int, int, int)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#setNumber(int)"><B>setNumber(int)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#setNumberFormatter(java.text.DecimalFormat)"><B>setNumberFormatter(DecimalFormat)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#setNumberOfMarkers(java.lang.String, double)"><B>setNumberOfMarkers(String, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#setNumberOfReads(java.lang.String, double)"><B>setNumberOfReads(String, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#setNumberOfReads(double)"><B>setNumberOfReads(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#setNumBins(int)"><B>setNumBins(int)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#setNumeratorCount(double)"><B>setNumeratorCount(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#setNumeratorCount(double)"><B>setNumeratorCount(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#setNumeratorCount(double)"><B>setNumeratorCount(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#setNumeratorRegionTotal(double)"><B>setNumeratorRegionTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#setNumeratorRegionTotal(double)"><B>setNumeratorRegionTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#setNumeratorRegionTotal(double)"><B>setNumeratorRegionTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#setNumeratorScanPVal(double)"><B>setNumeratorScanPVal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.html#setNumeratorTotal(double)"><B>setNumeratorTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.html" title="class in nextgen.core.model.score">NewRatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.html#setNumeratorTotal(double)"><B>setNumeratorTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.html" title="class in nextgen.core.model.score">RatioScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.html#setNumeratorTotal(double)"><B>setNumeratorTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.html" title="class in nextgen.core.model.score">WindowAllScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#setOmega(double)"><B>setOmega(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#setOmega(double)"><B>setOmega(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#setOmegaDistByTreeLength(java.lang.String)"><B>setOmegaDistByTreeLength(String)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#setOmegaPValue(double)"><B>setOmegaPValue(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setOpen(float)"><B>setOpen(float)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#setOrientation(boolean)"><B>setOrientation(boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#setOrientation(java.lang.String)"><B>setOrientation(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#setOrientation(java.lang.String)"><B>setOrientation(String)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#setOrientation(java.lang.String)"><B>setOrientation(String)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setOrientation(java.lang.String)"><B>setOrientation(String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setOrientation(char)"><B>setOrientation(char)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setOrientation(nextgen.core.annotation.Annotation.Strand)"><B>setOrientation(Annotation.Strand)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#setOrientation(char)"><B>setOrientation(char)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#setOrientation(char)"><B>setOrientation(char)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#setOrientation(nextgen.core.annotation.Annotation.Strand)"><B>setOrientation(Annotation.Strand)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#setOrientation(nextgen.core.annotation.Annotation.Strand)"><B>setOrientation(Annotation.Strand)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#setOrientation(java.lang.String)"><B>setOrientation(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#setOrientation(char)"><B>setOrientation(char)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#setOrientation(nextgen.core.annotation.Annotation.Strand)"><B>setOrientation(Annotation.Strand)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setOrientedEnd(int)"><B>setOrientedEnd(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#setOrientedEnd(int)"><B>setOrientedEnd(int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#setOrientedEnd(int)"><B>setOrientedEnd(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#setOrientedEnd(int)"><B>setOrientedEnd(int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setOrientedStart(int)"><B>setOrientedStart(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#setOrientedStart(int)"><B>setOrientedStart(int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#setOrientedStart(int)"><B>setOrientedStart(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#setOrientedStart(int)"><B>setOrientedStart(int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#setOriginalScore(double)"><B>setOriginalScore(double)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setOriginalSequence1(jaligner.Sequence)"><B>setOriginalSequence1(Sequence)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setOriginalSequence2(jaligner.Sequence)"><B>setOriginalSequence2(Sequence)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#setOutputInDirectory()"><B>setOutputInDirectory()</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#setPairedEndCounts(double)"><B>setPairedEndCounts(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Sets the total spliced end counts for this edge.
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#setParameters(broad.core.siphy.EvolutionaryModelParameters)"><B>setParameters(EvolutionaryModelParameters)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html#setParent(broad.core.datastructures.GenericRootedTree.TreeNode)"><B>setParent(GenericRootedTree.TreeNode&lt;T&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/GenericRootedTree.TreeNode.html" title="class in broad.core.datastructures">GenericRootedTree.TreeNode</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#setParent(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT)"><B>setParent(ChromosomeWithBubblesJGraphT)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Sets the parent graph of the edge
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#setPass(int)"><B>setPass(int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setPeakCutoffMostCommonReplicate(double)"><B>setPeakCutoffMostCommonReplicate(double)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Set cutoff for the percentage of fragments overlapping a peak that come from the most common replicate fragment
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setPeakTrimQuantile(double)"><B>setPeakTrimQuantile(double)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Set quantile for trim max contiguous algorithm for peak calling
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setPeakWindowCountCutoff(double)"><B>setPeakWindowCountCutoff(double)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Set minimum number of fragments overlapping a window to be considered for possible peak
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setPeakWindowScanPvalCutoff(double)"><B>setPeakWindowScanPvalCutoff(double)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Set scan P value cutoff for window within transcript to be considered for possible peak
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#setPercentMaskedAllowed(double)"><B>setPercentMaskedAllowed(double)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#setPercentVariantsCovered(double)"><B>setPercentVariantsCovered(double)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#setPi(Jama.Matrix)"><B>setPi(Matrix)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#setPIDToName(java.util.Map)"><B>setPIDToName(Map&lt;String, String&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#setPosition(int)"><B>setPosition(int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#setPositiveStranded()"><B>setPositiveStranded()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setPositiveStranded()"><B>setPositiveStranded()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/Assembly.html#setPossiblePremature(boolean)"><B>setPossiblePremature(boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#setPrimerLeftInput(java.lang.String)"><B>setPrimerLeftInput(String)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#setPrimerMin3PrimeOverlapOfJunction(int)"><B>setPrimerMin3PrimeOverlapOfJunction(int)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#setPrimerMin5PrimeOverlapOfJunction(int)"><B>setPrimerMin5PrimeOverlapOfJunction(int)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#setPrimerNumReturn(int)"><B>setPrimerNumReturn(int)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#setPrimerPairId(java.lang.String)"><B>setPrimerPairId(String)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#setPrimerPairPenalty(float)"><B>setPrimerPairPenalty(float)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#setPrimerPickingMode(broad.core.primer3.Primer3PrimerPickingMode)"><B>setPrimerPickingMode(Primer3PrimerPickingMode)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>Sets the primer3 primer picking mode
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#setPrimerRightInput(java.lang.String)"><B>setPrimerRightInput(String)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#setPrimers(broad.core.primer3.PrimerPair)"><B>setPrimers(PrimerPair)</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#setPrimerSequenceId(java.lang.String)"><B>setPrimerSequenceId(String)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#setPrimerStartCodonPosition(int)"><B>setPrimerStartCodonPosition(int)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#setProgram(java.lang.String)"><B>setProgram(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#setProgramDescription(java.lang.String)"><B>setProgramDescription(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Sets program description to be printed as part of help menu
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#setProperPairFlag(boolean)"><B>setProperPairFlag(boolean)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/Alignment.html#setProperPairFlag(boolean)"><B>setProperPairFlag(boolean)</B></A> - 
Method in interface nextgen.core.alignment.<A HREF="../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>
<DD>Set whether the alignment represented a proper pair defined by the aligner
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#setProperPairFlag(boolean)"><B>setProperPairFlag(boolean)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#setProperPairsOnly(boolean)"><B>setProperPairsOnly(boolean)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#setPvalue(double)"><B>setPvalue(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Sets the p-value for this edge
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#setQueueName(java.lang.String)"><B>setQueueName(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#setRateMatrix(Jama.Matrix)"><B>setRateMatrix(Matrix)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#setReferenceId(java.lang.String)"><B>setReferenceId(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setReferenceName(java.lang.String)"><B>setReferenceName(String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#setReferenceName(java.lang.String)"><B>setReferenceName(String)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#setReferenceName(java.lang.String)"><B>setReferenceName(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#setReferenceName(java.lang.String)"><B>setReferenceName(String)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#setRefGapped(boolean)"><B>setRefGapped(boolean)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#setRegion(nextgen.core.annotation.Annotation)"><B>setRegion(Annotation)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setRegionEnd(int)"><B>setRegionEnd(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#setRegionLength(double)"><B>setRegionLength(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setRegionStart(int)"><B>setRegionStart(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#setRegionTotal(double)"><B>setRegionTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setRemoveDuplicatesFlag(boolean)"><B>setRemoveDuplicatesFlag(boolean)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#setReversedOrientation(boolean)"><B>setReversedOrientation(boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#setReversedOrientation(boolean)"><B>setReversedOrientation(boolean)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setReversedOrientation(boolean)"><B>setReversedOrientation(boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#setRgb(java.lang.String)"><B>setRgb(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#setRightPrimerPosition(int)"><B>setRightPrimerPosition(int)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#setRow(java.lang.String, double[])"><B>setRow(String, double[])</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Cell.html#setRow(int)"><B>setRow(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Cell.html" title="class in jaligner">Cell</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#setRow(int, double[])"><B>setRow(int, double[])</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#setRow(int, int[])"><B>setRow(int, int[])</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#setRowDescription(int, java.lang.String)"><B>setRowDescription(int, String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#setRowDescription(java.lang.String, java.lang.String)"><B>setRowDescription(String, String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#setRPKM(double)"><B>setRPKM(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setRPKMConstant(double)"><B>setRPKMConstant(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#setRS8Score(double)"><B>setRS8Score(double)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#setRunParameters(java.lang.String)"><B>setRunParameters(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#setRunType(java.lang.String)"><B>setRunType(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#setSampleName(java.lang.String)"><B>setSampleName(String)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Set the sample name
<DT><A HREF="../nextgen/core/annotation/Gene.html#setSAMString(java.lang.String)"><B>setSAMString(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html#setScaling(int, double)"><B>setScaling(int, double)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="class in broad.core.hmm">MarkovModel.ForwardResult</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GeneScore.html#setScanPvalue(double)"><B>setScanPvalue(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.html#setScanPvalue(double)"><B>setScanPvalue(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#setScore(double)"><B>setScore(double)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#setScore(double)"><B>setScore(double)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#setScore(float)"><B>setScore(float)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setScore(double)"><B>setScore(double)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#setScore(double)"><B>setScore(double)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setScore(float)"><B>setScore(float)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Cell.html#setScore(float)"><B>setScore(float)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Cell.html" title="class in jaligner">Cell</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#setScore(double)"><B>setScore(double)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#setScore(double)"><B>setScore(double)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#setScore(double)"><B>setScore(double)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#setScores(java.util.List)"><B>setScores(List&lt;Double&gt;)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/AbstractListScore.html#setScores(java.util.List)"><B>setScores(List&lt;Double&gt;)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/AbstractListScore.html" title="class in nextgen.core.model.score">AbstractListScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#setScoring(java.lang.String)"><B>setScoring(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#setSecondRead()"><B>setSecondRead()</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setSecondRead()"><B>setSecondRead()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#setSeed(long)"><B>setSeed(long)</B></A> - 
Static method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>Set the seed for the random generator (used for permuting regions).
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#setSentinel(V)"><B>setSentinel(V)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Set the special sentinel value that will be used to signal novelty when putting a new interval
 into the tree, or to signal "not found" when removing an interval.
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#setSequence(broad.core.sequence.Sequence)"><B>setSequence(Sequence)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#setSequence(broad.core.sequence.Sequence)"><B>setSequence(Sequence)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setSequence(broad.core.sequence.Sequence)"><B>setSequence(Sequence)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#setSequence(java.lang.String)"><B>setSequence(String)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Sets the sequence string
<DT><A HREF="../nextgen/core/annotation/Gene.html#setSequence(java.lang.String)"><B>setSequence(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setSequence1(char[])"><B>setSequence1(char[])</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setSequence2(char[])"><B>setSequence2(char[])</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#setSequenceBases(java.lang.String)"><B>setSequenceBases(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#setSequenceBases(java.lang.String)"><B>setSequenceBases(String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#setSequenceFile(java.io.File)"><B>setSequenceFile(File)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#setSequenceFromChromosome(broad.core.sequence.Sequence)"><B>setSequenceFromChromosome(Sequence)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setSimilarity(int)"><B>setSimilarity(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#setSource(java.lang.String)"><B>setSource(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#setSource(java.io.File)"><B>setSource(File)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/PhylipInterleavedMultipleAlignmentIO.html#setSpacing(int)"><B>setSpacing(int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/PhylipInterleavedMultipleAlignmentIO.html" title="class in broad.core.multiplealignment">PhylipInterleavedMultipleAlignmentIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#setSpliceEdgeCount(double)"><B>setSpliceEdgeCount(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Sets the spliced edge count for this edge.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setSpliceWeight(double)"><B>setSpliceWeight(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Sets the splice weight for this graph
<DT><A HREF="../broad/pda/seq/graph/Path.html#setSpliceWeight(double)"><B>setSpliceWeight(double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/Assembly.html#setSpurious(boolean)"><B>setSpurious(boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/Assembly.html" title="class in nextgen.core.scripture">Assembly</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#setStart(int)"><B>setStart(int)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#setStart(int)"><B>setStart(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#setStart(int)"><B>setStart(int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Sets the start of this graph
<DT><A HREF="../nextgen/core/annotation/Annotation.html#setStart(int)"><B>setStart(int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Sets a new start position.
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#setStart(int)"><B>setStart(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>SETTING METHODS
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#setStart(int)"><B>setStart(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>Extends or trims the start coordinate.
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#setStart(int)"><B>setStart(int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setStart1(int)"><B>setStart1(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#setStart2(int)"><B>setStart2(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.html#setStateTransitionProbability(int, int, double)"><B>setStateTransitionProbability(int, int, double)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm">MarkovModel</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/TracebackNode.html#setStop()"><B>setStop()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/TracebackNode.html" title="class in jaligner">TracebackNode</A>
<DD>Sets the direction to stop
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#setStoredStats(broad.pda.seq.alignment.StoredAlignmentStats)"><B>setStoredStats(StoredAlignmentStats)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#setStrand(java.lang.String)"><B>setStrand(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#setStrand(java.lang.String)"><B>setStrand(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#setSubcluster(broad.pda.geneexpression.clustering.Cluster, broad.pda.geneexpression.clustering.Cluster)"><B>setSubcluster(Cluster, Cluster)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#setThickEnd(int)"><B>setThickEnd(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#setThickStart(int)"><B>setThickStart(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#setThreePrimeBuffer(int)"><B>setThreePrimeBuffer(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#setThreshold1AndMeanNumDup(int)"><B>setThreshold1AndMeanNumDup(int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#setThreshold2(int, int)"><B>setThreshold2(int, int)</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#setTophatPath(java.lang.String)"><B>setTophatPath(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>Set path to tophat executable
<DT><A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html#setTophatPath(java.lang.String)"><B>setTophatPath(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser">ConfigFileParser.TophatOptions</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#setTotal(double)"><B>setTotal(double)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#setTotalLength(int)"><B>setTotalLength(int)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#setTree(org.forester.phylogeny.Phylogeny)"><B>setTree(Phylogeny)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#setTreeBit(int)"><B>setTreeBit(int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html#setTreeLength(double)"><B>setTreeLength(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.OmegaFit.html" title="class in broad.core.siphy">EvolutionaryModel.OmegaFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html#setTreeLength(double)"><B>setTreeLength(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.PiFit.html" title="class in broad.core.siphy">EvolutionaryModel.PiFit</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#setTreeLength(double)"><B>setTreeLength(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setTrimEnds(boolean)"><B>setTrimEnds(boolean)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setTrimQuantile(double)"><B>setTrimQuantile(double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#setType(int)"><B>setType(int)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#setType(int)"><B>setType(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Sets the sequence type (nucleic or protein)
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#setUninformativeNodes(java.util.Map, java.util.List)"><B>setUninformativeNodes(Map&lt;String, Matrix&gt;, List&lt;String&gt;)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#setUninformativeNodes(java.util.Map, java.util.List, int)"><B>setUninformativeNodes(Map&lt;String, Matrix&gt;, List&lt;String&gt;, int)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#setUnparsedRow(int, java.lang.String[])"><B>setUnparsedRow(int, String[])</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/TracebackNode.html#setUp(int)"><B>setUp(int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/TracebackNode.html" title="class in jaligner">TracebackNode</A>
<DD>Sets the direction to up
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#setUpAlignment(broad.core.util.CLUtil.ArgumentMap, java.lang.String, java.lang.String, java.util.List, broad.core.siphy.EvolutionaryModel)"><B>setUpAlignment(CLUtil.ArgumentMap, String, String, List&lt;String&gt;, EvolutionaryModel)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#setUpMAF(java.lang.String, java.util.List, broad.core.siphy.EvolutionaryModel, int, int)"><B>setUpMAF(String, List&lt;String&gt;, EvolutionaryModel, int, int)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#setUpMAF(broad.core.multiplealignment.MAFIO, java.util.List, broad.core.siphy.EvolutionaryModel, int, int)"><B>setUpMAF(MAFIO, List&lt;String&gt;, EvolutionaryModel, int, int)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#setUpMAF(broad.core.multiplealignment.MAFIO, java.util.List, org.forester.phylogeny.Phylogeny, int, int)"><B>setUpMAF(MAFIO, List&lt;String&gt;, Phylogeny, int, int)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setUpWeightSplices(boolean)"><B>setUpWeightSplices(boolean)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#setUsedInput(broad.core.primer3.Primer3SequenceInputTags)"><B>setUsedInput(Primer3SequenceInputTags)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.Node.html#setValue(V1)"><B>setValue(V1)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.Node.html" title="class in broad.core.datastructures">IntervalTree.Node</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Pair.html#setValue1(T1)"><B>setValue1(T1)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures">Pair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Pair.html#setValue2(T1)"><B>setValue2(T1)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures">Pair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#setVersion(java.lang.String)"><B>setVersion(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#setVertexCount(nextgen.core.annotation.Annotation, double)"><B>setVertexCount(Annotation, double)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Sets the counts for the vertex
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#setWeighReadCountsFlag(boolean)"><B>setWeighReadCountsFlag(boolean)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#shift(int)"><B>shift(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#shift(int)"><B>shift(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/StationaryDistributionIO.html#shift(int)"><B>shift(int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/StationaryDistributionIO.html" title="class in broad.core.siphy">StationaryDistributionIO</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#shift(int)"><B>shift(int)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#shift(int)"><B>shift(int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Shift the positioning of the annotation by a given number of base pairs.
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#shift(int)"><B>shift(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#shift(int)"><B>shift(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>Shift all blocks in the compound interval by a given delta
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#shift(int)"><B>shift(int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#SHORT_ARM_TYPE"><B>SHORT_ARM_TYPE</B></A> - 
Static variable in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_A"><B>SHORT_ENCODED_A</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_a"><B>SHORT_ENCODED_a</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_C"><B>SHORT_ENCODED_C</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_c"><B>SHORT_ENCODED_c</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_G"><B>SHORT_ENCODED_G</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_g"><B>SHORT_ENCODED_g</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_GAP"><B>SHORT_ENCODED_GAP</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_N"><B>SHORT_ENCODED_N</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_T"><B>SHORT_ENCODED_T</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_ENCODED_t"><B>SHORT_ENCODED_t</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#SHORT_READ_FLOW"><B>SHORT_READ_FLOW</B></A> - 
Static variable in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBED.html" title="class in broad.core.annotation"><B>ShortBED</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/ShortBED.html#ShortBED(java.lang.String)"><B>ShortBED(String)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBED.html" title="class in broad.core.annotation">ShortBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBED.html#ShortBED(broad.core.annotation.LightweightGenomicAnnotation)"><B>ShortBED(LightweightGenomicAnnotation)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBED.html" title="class in broad.core.annotation">ShortBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBED.html#ShortBED(java.lang.String, java.lang.String, int, int)"><B>ShortBED(String, String, int, int)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBED.html" title="class in broad.core.annotation">ShortBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBED.html#ShortBED(java.lang.String[])"><B>ShortBED(String[])</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBED.html" title="class in broad.core.annotation">ShortBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.html#shortBEDFactory"><B>shortBEDFactory</B></A> - 
Static variable in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBEDReader.html" title="class in broad.core.annotation"><B>ShortBEDReader</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/ShortBEDReader.html#ShortBEDReader()"><B>ShortBEDReader()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBEDReader.html" title="class in broad.core.annotation">ShortBEDReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBEDReader.html#ShortBEDReader(java.io.BufferedReader)"><B>ShortBEDReader(BufferedReader)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBEDReader.html" title="class in broad.core.annotation">ShortBEDReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBEDReader.html#ShortBEDReader(java.lang.String)"><B>ShortBEDReader(String)</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBEDReader.html" title="class in broad.core.annotation">ShortBEDReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/MathUtil.html#shuffle(double[])"><B>shuffle(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/MathUtil.html" title="class in broad.core.math">MathUtil</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/MathUtil.html#shuffle(java.util.List)"><B>shuffle(List&lt;T&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/MathUtil.html" title="class in broad.core.math">MathUtil</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#shuffle()"><B>shuffle()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#shuffle(broad.core.annotation.GenomicAnnotation, java.util.List)"><B>shuffle(GenomicAnnotation, List&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#shuffle(java.util.List, java.util.List)"><B>shuffle(List&lt;? extends GenomicAnnotation&gt;, List&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>Randomizes (using a quasi uniform distribution) a list of genomic annotations
<DT><A HREF="../broad/pda/seq/rap/ShuffleBED.html#shuffleAndWriteAnnotations(nextgen.core.annotation.AnnotationList, nextgen.core.coordinatesystem.GenomicSpace, nextgen.core.annotation.AnnotationList, java.io.File, int)"><B>shuffleAndWriteAnnotations(AnnotationList&lt;Annotation&gt;, GenomicSpace, AnnotationList&lt;? extends Annotation&gt;, File, int)</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBED.html" title="class in broad.pda.seq.rap">ShuffleBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/ShuffleBAM.html" title="class in broad.pda.seq.rap"><B>ShuffleBAM</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/ShuffleBAM.html#ShuffleBAM()"><B>ShuffleBAM()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBAM.html" title="class in broad.pda.seq.rap">ShuffleBAM</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/ShuffleBED.html" title="class in broad.pda.seq.rap"><B>ShuffleBED</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/ShuffleBED.html#ShuffleBED()"><B>ShuffleBED()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBED.html" title="class in broad.pda.seq.rap">ShuffleBED</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/ShuffledIterator.html" title="class in nextgen.core.coordinatesystem"><B>ShuffledIterator</B></A>&lt;<A HREF="../nextgen/core/coordinatesystem/ShuffledIterator.html" title="type parameter in ShuffledIterator">T</A> extends <A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; - Class in <A HREF="../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A><DD>&nbsp;<DT><A HREF="../nextgen/core/coordinatesystem/ShuffledIterator.html#ShuffledIterator(net.sf.samtools.util.CloseableIterator, nextgen.core.coordinatesystem.CoordinateSpace, nextgen.core.annotation.Annotation)"><B>ShuffledIterator(CloseableIterator&lt;T&gt;, CoordinateSpace, Annotation)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/ShuffledIterator.html" title="class in nextgen.core.coordinatesystem">ShuffledIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#shuffleScores(java.lang.String, int)"><B>shuffleScores(String, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#shuffleScores(java.lang.String)"><B>shuffleScores(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#shuffleWithinValid(nextgen.core.annotation.Annotation, java.util.List)"><B>shuffleWithinValid(Annotation, List&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#shuffleWithinValid(java.util.List, java.util.List)"><B>shuffleWithinValid(List&lt;? extends Annotation&gt;, List&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#signalScanPval(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)"><B>signalScanPval(Gene, Annotation)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get scan P value in signal dataset with respect to the transcript
<DT><A HREF="../jaligner/Markups.html#SIMILARITY"><B>SIMILARITY</B></A> - 
Static variable in class jaligner.<A HREF="../jaligner/Markups.html" title="class in jaligner">Markups</A>
<DD>Markup line similarity character
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment"><B>SingleEndAlignment</B></A> - Class in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>&nbsp;<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#SingleEndAlignment(net.sf.samtools.SAMRecord)"><B>SingleEndAlignment(SAMRecord)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#SingleEndAlignment(org.broad.igv.sam.Alignment)"><B>SingleEndAlignment(Alignment)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>This is to populate our Alignment from the old legacy IGV alignments
<DT><A HREF="../nextgen/core/alignment/SingleEndAlignment.html#SingleEndAlignment(net.sf.samtools.SAMRecord, boolean)"><B>SingleEndAlignment(SAMRecord, boolean)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/SingleEndAlignment.html" title="class in nextgen.core.alignment">SingleEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation"><B>SingleInterval</B></A> - Class in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/SingleInterval.html#SingleInterval(int, int)"><B>SingleInterval(int, int)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation">SingleInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/SingleInterval.html#SingleInterval(broad.core.datastructures.Interval)"><B>SingleInterval(Interval)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation">SingleInterval</A>
<DD>&nbsp;
<DT><A HREF="../Jama/SingularValueDecomposition.html" title="class in Jama"><B>SingularValueDecomposition</B></A> - Class in <A HREF="../Jama/package-summary.html">Jama</A><DD>Singular Value Decomposition.<DT><A HREF="../Jama/SingularValueDecomposition.html#SingularValueDecomposition(Jama.Matrix)"><B>SingularValueDecomposition(Matrix)</B></A> - 
Constructor for class Jama.<A HREF="../Jama/SingularValueDecomposition.html" title="class in Jama">SingularValueDecomposition</A>
<DD>Construct the singular value decomposition
<DT><A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html#size()"><B>size()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.AnnotationSet.html" title="class in broad.core.annotation">AnnotationReader.AnnotationSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#size()"><B>size()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Jesse June 23, 2012
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#size()"><B>size()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Return the number of intervals in the tree.
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#size()"><B>size()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/matrix/Matrix.html#SIZE"><B>SIZE</B></A> - 
Static variable in class jaligner.matrix.<A HREF="../jaligner/matrix/Matrix.html" title="class in jaligner.matrix">Matrix</A>
<DD>The size of the scoring matrix.
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#size()"><B>size()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#size()"><B>size()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#size()"><B>size()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#size()"><B>size()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#size()"><B>size()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#size()"><B>size()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html#SIZES"><B>SIZES</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap">GenomeCommandLineProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#SIZES"><B>SIZES</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Distribution.html#skellamCdf(int, double, double)"><B>skellamCdf(int, double, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Distribution.html" title="class in broad.core.math">Distribution</A>
<DD>Compute the cdf of a Skellam distribution, the distribution of the difference of two Poisson random variables
<DT><A HREF="../broad/core/math/Distribution.html#skellamDensity(int, double, double)"><B>skellamDensity(int, double, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Distribution.html" title="class in broad.core.math">Distribution</A>
<DD>Compute the pdf of a Skellam distribution, the distribution of the difference of two Poisson random variables
<DT><A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html" title="class in broad.pda.seq.rap"><B>SkellamKLPeakCaller</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html#SkellamKLPeakCaller()"><B>SkellamKLPeakCaller()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html" title="class in broad.pda.seq.rap">SkellamKLPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Distribution.html#skellamLeftTail(int, double, double)"><B>skellamLeftTail(int, double, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Distribution.html" title="class in broad.core.math">Distribution</A>
<DD>Compute the left tail of a Skellam distribution, the distribution of the difference of two Poisson random variables
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#skellamPval(nextgen.core.annotation.Annotation, double, double)"><B>skellamPval(Annotation, double, double)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>Get skellam P value of difference of read counts in region
<DT><A HREF="../broad/core/math/Distribution.html#skellamPvalue(int, double, double)"><B>skellamPvalue(int, double, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Distribution.html" title="class in broad.core.math">Distribution</A>
<DD>Compute the one-tailed P-value of a Skellam distribution, the distribution of the difference of two Poisson random variables
<DT><A HREF="../broad/core/math/Distribution.html#skellamRightTail(int, double, double)"><B>skellamRightTail(int, double, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Distribution.html" title="class in broad.core.math">Distribution</A>
<DD>Compute the right tail of a Skellam distribution, the distribution of the difference of two Poisson random variables
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.html" title="class in broad.pda.seq.rap"><B>SkellamScore</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/SkellamScore.html#SkellamScore(nextgen.core.annotation.Annotation)"><B>SkellamScore(Annotation)</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.html" title="class in broad.pda.seq.rap">SkellamScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.html#SkellamScore(nextgen.core.model.AlignmentModel, nextgen.core.model.AlignmentModel, nextgen.core.annotation.Annotation, double, double)"><B>SkellamScore(AlignmentModel, AlignmentModel, Annotation, double, double)</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.html" title="class in broad.pda.seq.rap">SkellamScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.Processor.html" title="class in broad.pda.seq.rap"><B>SkellamScore.Processor</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/SkellamScore.Processor.html#SkellamScore.Processor(nextgen.core.model.AlignmentModel, nextgen.core.model.AlignmentModel, double, double)"><B>SkellamScore.Processor(AlignmentModel, AlignmentModel, double, double)</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.Processor.html" title="class in broad.pda.seq.rap">SkellamScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html#skipFirstReads()"><B>skipFirstReads()</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers">FilteredBamWriter</A>
<DD>Don't write any first reads
<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html#skipSecondReads()"><B>skipSecondReads()</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers">FilteredBamWriter</A>
<DD>Don't write any second reads
<DT><A HREF="../broad/pda/seq/rap/SlideAndCalculate.html" title="class in broad.pda.seq.rap"><B>SlideAndCalculate</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/SlideAndCalculate.html#SlideAndCalculate()"><B>SlideAndCalculate()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCalculate.html" title="class in broad.pda.seq.rap">SlideAndCalculate</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SlideAndCount.html" title="class in broad.pda.seq.rap"><B>SlideAndCount</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/SlideAndCount.html#SlideAndCount()"><B>SlideAndCount()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCount.html" title="class in broad.pda.seq.rap">SlideAndCount</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#slidePWM(broad.core.motif.PositionWeightMatrix, double, short[], java.util.List, int)"><B>slidePWM(PositionWeightMatrix, double, short[], List&lt;int[]&gt;, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#slidePWM(broad.core.motif.PositionWeightMatrix, double, int)"><B>slidePWM(PositionWeightMatrix, double, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html#slideWindowComputeOmega(nextgen.core.annotation.Annotation, java.io.File, java.lang.String, java.lang.String, int, int)"><B>slideWindowComputeOmega(Annotation, File, String, String, int, int)</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html" title="class in broad.core.siphy.tools.conservation">EstimateOmegaPerExon</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html#slideWindowComputeOmega(broad.pda.datastructures.Alignments, broad.core.siphy.TreeScaler, int, int)"><B>slideWindowComputeOmega(Alignments, TreeScaler, int, int)</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html" title="class in broad.core.siphy.tools.conservation">EstimateOmegaPerExon</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer"><B>SmatchLike</B></A> - Class in <A HREF="../broad/pda/capture/designer/package-summary.html">broad.pda.capture.designer</A><DD>&nbsp;<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html#SmatchLike(java.lang.String, java.util.List, int)"><B>SmatchLike(String, List&lt;Sequence&gt;, int)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer">SmatchLike</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html#SmatchLike(java.lang.String, java.lang.String, int)"><B>SmatchLike(String, String, int)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer">SmatchLike</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html#SmatchLike(java.lang.String, java.util.Collection, broad.core.datastructures.Pair)"><B>SmatchLike(String, Collection&lt;Pair&lt;String&gt;&gt;, Pair&lt;String&gt;)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer">SmatchLike</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html#SmatchLike(java.lang.String, java.util.Collection, broad.core.primer3.PrimerPair)"><B>SmatchLike(String, Collection&lt;Pair&lt;String&gt;&gt;, PrimerPair)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer">SmatchLike</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/SmithWatermanGotoh.html" title="class in jaligner"><B>SmithWatermanGotoh</B></A> - Class in <A HREF="../jaligner/package-summary.html">jaligner</A><DD>An implementation of the Smith-Waterman algorithm with Gotoh's improvement
 for biological local pairwise sequence alignment.<DT><A HREF="../Jama/CholeskyDecomposition.html#solve(Jama.Matrix)"><B>solve(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/CholeskyDecomposition.html" title="class in Jama">CholeskyDecomposition</A>
<DD>Solve A*X = B
<DT><A HREF="../Jama/LUDecomposition.html#solve(Jama.Matrix)"><B>solve(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/LUDecomposition.html" title="class in Jama">LUDecomposition</A>
<DD>Solve A*X = B
<DT><A HREF="../Jama/Matrix.html#solve(Jama.Matrix)"><B>solve(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Solve A*X = B
<DT><A HREF="../Jama/QRDecomposition.html#solve(Jama.Matrix)"><B>solve(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/QRDecomposition.html" title="class in Jama">QRDecomposition</A>
<DD>Least squares solution of A*X = B
<DT><A HREF="../Jama/Matrix.html#solveTranspose(Jama.Matrix)"><B>solveTranspose(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Solve X*A = B, which is also A'*X' = B'
<DT><A HREF="../Jama/Matrix.html#sortColumns()"><B>sortColumns()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#sortColumnsKeepingIndex(Jama.Matrix)"><B>sortColumnsKeepingIndex(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#sortList(int)"><B>sortList(int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#spearmanCorrelation(java.util.List, java.util.List)"><B>spearmanCorrelation(List&lt;Double&gt;, List&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#spearmanCorrelation(double[], double[])"><B>spearmanCorrelation(double[], double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/SplicedReadFilter.html" title="class in nextgen.core.readFilters"><B>SplicedReadFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/SplicedReadFilter.html#SplicedReadFilter()"><B>SplicedReadFilter()</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/SplicedReadFilter.html" title="class in nextgen.core.readFilters">SplicedReadFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/SplitAndTrimIndex.html" title="class in broad.pda.seq.fastq"><B>SplitAndTrimIndex</B></A> - Class in <A HREF="../broad/pda/seq/fastq/package-summary.html">broad.pda.seq.fastq</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/fastq/SplitAndTrimIndex.html#SplitAndTrimIndex(java.io.File, java.io.File, java.io.File, java.lang.String)"><B>SplitAndTrimIndex(File, File, File, String)</B></A> - 
Constructor for class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/SplitAndTrimIndex.html" title="class in broad.pda.seq.fastq">SplitAndTrimIndex</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#splitByChr(java.util.Collection)"><B>splitByChr(Collection&lt;Gene&gt;)</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/SplitByName.html" title="class in broad.pda.differentialExpression"><B>SplitByName</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/SplitByName.html#SplitByName(java.io.File, java.io.File, java.io.File, java.lang.String, boolean, boolean)"><B>SplitByName(File, File, File, String, boolean, boolean)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/SplitByName.html" title="class in broad.pda.differentialExpression">SplitByName</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/SplitFastqByIndex.html" title="class in broad.pda.seq.fastq"><B>SplitFastqByIndex</B></A> - Class in <A HREF="../broad/pda/seq/fastq/package-summary.html">broad.pda.seq.fastq</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/fastq/SplitFastqByIndex.html#SplitFastqByIndex(java.lang.String, java.lang.String, java.lang.String, java.lang.String, boolean)"><B>SplitFastqByIndex(String, String, String, String, boolean)</B></A> - 
Constructor for class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/SplitFastqByIndex.html" title="class in broad.pda.seq.fastq">SplitFastqByIndex</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/SplitFile.html" title="class in broad.core.siphy.tools.conservation"><B>SplitFile</B></A> - Class in <A HREF="../broad/core/siphy/tools/conservation/package-summary.html">broad.core.siphy.tools.conservation</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/tools/conservation/SplitFile.html#SplitFile(java.io.File, java.lang.String, int)"><B>SplitFile(File, String, int)</B></A> - 
Constructor for class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/SplitFile.html" title="class in broad.core.siphy.tools.conservation">SplitFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#standardizeOreintation()"><B>standardizeOreintation()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#start"><B>start</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#start(java.io.File)"><B>start(File)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#start(java.io.BufferedReader)"><B>start(BufferedReader)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#START_CODON_PATRN"><B>START_CODON_PATRN</B></A> - 
Static variable in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#START_OF_STRING"><B>START_OF_STRING</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3IO.html#startPrimer3Communications()"><B>startPrimer3Communications()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3IO.html" title="class in broad.core.primer3">Primer3IO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#startTrack(java.lang.String)"><B>startTrack(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#startTrack(java.lang.String, int)"><B>startTrack(String, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/StationaryDistributionIO.html" title="class in broad.core.siphy"><B>StationaryDistributionIO</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/StationaryDistributionIO.html#StationaryDistributionIO()"><B>StationaryDistributionIO()</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/StationaryDistributionIO.html" title="class in broad.core.siphy">StationaryDistributionIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math"><B>Statistics</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/Statistics.html#Statistics()"><B>Statistics()</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#stdev(java.util.Collection)"><B>stdev(Collection&lt;? extends Number&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#stdev(double[])"><B>stdev(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#stitch(int, int)"><B>stitch(int, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#stitchList(java.util.Collection, int)"><B>stitchList(Collection&lt;? extends Annotation&gt;, int)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#stitchTo(broad.core.annotation.LightweightGenomicAnnotation)"><B>stitchTo(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#stitchTo(broad.core.annotation.LightweightGenomicAnnotation)"><B>stitchTo(LightweightGenomicAnnotation)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>Enlarges this annotation by
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#stitchTo(broad.core.annotation.LightweightGenomicAnnotation)"><B>stitchTo(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#stitchTo(nextgen.core.annotation.Annotation)"><B>stitchTo(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#stitchTo(nextgen.core.annotation.Annotation)"><B>stitchTo(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#stitchTo(nextgen.core.annotation.Annotation)"><B>stitchTo(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#stitchTo(nextgen.core.annotation.Annotation)"><B>stitchTo(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html#stitchToList(java.util.List)"><B>stitchToList(List&lt;TreeScalerIO.ScaledWindow&gt;)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.ScaledWindow.html" title="class in broad.core.siphy">TreeScalerIO.ScaledWindow</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Directions.html#STOP"><B>STOP</B></A> - 
Static variable in class jaligner.<A HREF="../jaligner/Directions.html" title="class in jaligner">Directions</A>
<DD>Traceback direction stop
<DT><A HREF="../nextgen/core/annotation/Gene.html#STOP_CODONS"><B>STOP_CODONS</B></A> - 
Static variable in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#store(broad.pda.seq.segmentation.AlignmentDataModelStats)"><B>store(AlignmentDataModelStats)</B></A> - 
Static method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment"><B>StoredAlignmentStats</B></A> - Class in <A HREF="../broad/pda/seq/alignment/package-summary.html">broad.pda.seq.alignment</A><DD>Class that manages the storage of an AlignmentDataModelStats to permanent storage
 Format of the file is simple, the first line stores the timestamp of file creation
 the next two columns are tab separated lines containing information about the alignment and index file when the stats were stored:
 Size, last modified date
 The next line stores the global RPKM constant
 The next line should be a P if the alignment is paired and anything else, usually U if it is not.<DT><A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html#StoredAlignmentStats()"><B>StoredAlignmentStats()</B></A> - 
Constructor for class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/StoredAlignmentStats.html" title="class in broad.pda.seq.alignment">StoredAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#strand"><B>strand</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser"><B>StringParser</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>&nbsp;<DT><A HREF="../broad/core/parser/StringParser.html#StringParser()"><B>StringParser()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#submatrixByColumnIndex(java.util.Collection)"><B>submatrixByColumnIndex(Collection&lt;Integer&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#submatrixByColumnNames(java.util.Collection)"><B>submatrixByColumnNames(Collection&lt;String&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#submatrixByColumnNames(java.lang.String[])"><B>submatrixByColumnNames(String[])</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#submatrixByColumns(java.util.Collection)"><B>submatrixByColumns(Collection&lt;String&gt;)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/LocationAwareMatrix.html#submatrixByColumns(java.lang.String[])"><B>submatrixByColumns(String[])</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/LocationAwareMatrix.html" title="class in broad.pda.datastructures">LocationAwareMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#submatrixByRowDescriptions(java.util.Collection, java.util.List)"><B>submatrixByRowDescriptions(Collection&lt;String&gt;, List&lt;String&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#submatrixByRowNames(java.util.Collection)"><B>submatrixByRowNames(Collection&lt;String&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#submatrixByRowNames(java.lang.String)"><B>submatrixByRowNames(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#submatrixByRowNames(java.lang.String[])"><B>submatrixByRowNames(String[])</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#subsequence(int, int)"><B>subsequence(int, int)</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Returns a subsequence
<DT><A HREF="../broad/core/math/Statistics.html#sum(java.util.List)"><B>sum(List&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#sum(double[])"><B>sum(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/SumScore.html" title="class in nextgen.core.model.score"><B>SumScore</B></A> - Class in <A HREF="../nextgen/core/model/score/package-summary.html">nextgen.core.model.score</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/score/SumScore.html#SumScore(nextgen.core.annotation.Annotation)"><B>SumScore(Annotation)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/SumScore.html" title="class in nextgen.core.model.score">SumScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/SumScore.html#SumScore(nextgen.core.annotation.AnnotationCollection, nextgen.core.annotation.Annotation)"><B>SumScore(AnnotationCollection&lt;? extends Annotation&gt;, Annotation)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/SumScore.html" title="class in nextgen.core.model.score">SumScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/SumScore.Processor.html" title="class in nextgen.core.model.score"><B>SumScore.Processor</B></A> - Class in <A HREF="../nextgen/core/model/score/package-summary.html">nextgen.core.model.score</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/score/SumScore.Processor.html#SumScore.Processor(nextgen.core.annotation.AnnotationCollection)"><B>SumScore.Processor(AnnotationCollection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/SumScore.Processor.html" title="class in nextgen.core.model.score">SumScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#svd()"><B>svd()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Singular Value Decomposition
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#SYMMETRIC"><B>SYMMETRIC</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.html#systemTest(java.lang.String)"><B>systemTest(String)</B></A> - 
Method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.html" title="class in broad.pda.methylation">HybridCaptureDesigner</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-18.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-20.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-19.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-19.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
