<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:31 EDT 2013 -->
<TITLE>
M-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="M-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-12.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-14.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-13.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-13.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_M_"><!-- --></A><H2>
<B>M</B></H2>
<DL>
<DT><A HREF="../broad/core/math/Statistics.html#MAD(double[])"><B>MAD(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment"><B>MAFAlignment</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#MAFAlignment()"><B>MAFAlignment()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#MAFAlignment(java.lang.String)"><B>MAFAlignment(String)</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment"><B>MAFAlignment.MAFHeader</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html#MAFAlignment.MAFHeader()"><B>MAFAlignment.MAFHeader()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFHeader.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFHeader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment"><B>MAFAlignment.MAFMultipleAlignmentBlock</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#MAFAlignment.MAFMultipleAlignmentBlock()"><B>MAFAlignment.MAFMultipleAlignmentBlock()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#MAFAlignment.MAFMultipleAlignmentBlock(broad.core.multiplealignment.MultipleAlignment)"><B>MAFAlignment.MAFMultipleAlignmentBlock(MultipleAlignment)</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFIO.html" title="class in broad.core.multiplealignment"><B>MAFIO</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/MAFIO.html#MAFIO(java.lang.String, boolean)"><B>MAFIO(String, boolean)</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFIO.html" title="class in broad.core.multiplealignment">MAFIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFIO.html#MAFIO()"><B>MAFIO()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFIO.html" title="class in broad.core.multiplealignment">MAFIO</A>
<DD>&nbsp;
<DT><A HREF="../Jama/examples/MagicSquareExample.html#magic(int)"><B>magic(int)</B></A> - 
Static method in class Jama.examples.<A HREF="../Jama/examples/MagicSquareExample.html" title="class in Jama.examples">MagicSquareExample</A>
<DD>Generate magic square test matrix.
<DT><A HREF="../Jama/examples/MagicSquareExample.html" title="class in Jama.examples"><B>MagicSquareExample</B></A> - Class in <A HREF="../Jama/examples/package-summary.html">Jama.examples</A><DD>Example of use of Matrix Class, featuring magic squares.<DT><A HREF="../Jama/examples/MagicSquareExample.html#MagicSquareExample()"><B>MagicSquareExample()</B></A> - 
Constructor for class Jama.examples.<A HREF="../Jama/examples/MagicSquareExample.html" title="class in Jama.examples">MagicSquareExample</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationUtils.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationUtils.html" title="class in broad.core.annotation">AnnotationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html" title="class in broad.core.annotation">MaximumContiguousSubsequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaderUtils.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaderUtils.html" title="class in broad.core.datastructures">MatrixWithHeaderUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/ANOVA.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/ANOVA.html" title="class in broad.core.math">ANOVA</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/CombinationGenerator.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/CombinationGenerator.html" title="class in broad.core.math">CombinationGenerator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/FDRDistribution.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/FDRDistribution.html" title="class in broad.core.math">FDRDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Loess.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Loess.html" title="class in broad.core.math">Loess</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/misc/ComputeConservedMotifs.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.motif.misc.<A HREF="../broad/core/motif/misc/ComputeConservedMotifs.html" title="class in broad.core.motif.misc">ComputeConservedMotifs</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/misc/ScoreMotifInstances.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.motif.misc.<A HREF="../broad/core/motif/misc/ScoreMotifInstances.html" title="class in broad.core.motif.misc">ScoreMotifInstances</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrixIO.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrixIO.html" title="class in broad.core.motif">PositionWeightMatrixIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PWMUtils.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.motif.<A HREF="../broad/core/motif/PWMUtils.html" title="class in broad.core.motif">PWMUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SequenceMotif.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotif.html" title="class in broad.core.motif">SequenceMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SequenceMotifCluster.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotifCluster.html" title="class in broad.core.motif">SequenceMotifCluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/AlignmentExtractor.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/AlignmentExtractor.html" title="class in broad.core.multiplealignment">AlignmentExtractor</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html" title="class in broad.core.multiplealignment">MultipleAlignmentFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.overlaputils.<A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils">GeneSetIntersect</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/isPCRLike.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/isPCRLike.html" title="class in broad.core.primer3">isPCRLike</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPairReader.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPairReader.html" title="class in broad.core.primer3">PrimerPairReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerUtils.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerUtils.html" title="class in broad.core.primer3">PrimerUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/RNAiPCRPrimerDesigner.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/RNAiPCRPrimerDesigner.html" title="class in broad.core.primer3">RNAiPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Extractor.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Extractor.html" title="class in broad.core.sequence">Extractor</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityIndex.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityIndex.html" title="class in broad.core.sequence">FastaQualityIndex</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.html" title="class in broad.core.siphy">PiHMM</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/AverageBranchLength.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/AverageBranchLength.html" title="class in broad.core.siphy.tools.conservation">AverageBranchLength</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/BatchSubmitEstimateExon.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/BatchSubmitEstimateExon.html" title="class in broad.core.siphy.tools.conservation">BatchSubmitEstimateExon</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/ComputeKmersForRegion.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/ComputeKmersForRegion.html" title="class in broad.core.siphy.tools.conservation">ComputeKmersForRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/ComputeMaxPercentileOfElements.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/ComputeMaxPercentileOfElements.html" title="class in broad.core.siphy.tools.conservation">ComputeMaxPercentileOfElements</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/ComputePercentageOfKmers.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/ComputePercentageOfKmers.html" title="class in broad.core.siphy.tools.conservation">ComputePercentageOfKmers</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html" title="class in broad.core.siphy.tools.conservation">EstimateOmegaPerExon</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/SplitFile.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/SplitFile.html" title="class in broad.core.siphy.tools.conservation">SplitFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScaler.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineAutomator.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.html" title="class in broad.core.util">PipelineAutomator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ArrayDesigner.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ArrayDesigner.html" title="class in broad.pda.capture.designer">ArrayDesigner</A>
<DD>Main method designs array(s) and outputs file of oligos and primers.
<DT><A HREF="../broad/pda/capture/designer/ArrayDesignerInSilicoPcrTest.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ArrayDesignerInSilicoPcrTest.html" title="class in broad.pda.capture.designer">ArrayDesignerInSilicoPcrTest</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeMIRScore.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeMIRScore.html" title="class in broad.pda.capture.designer">ComputeMIRScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/DesignPCRTails.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/DesignPCRTails.html" title="class in broad.pda.capture.designer">DesignPCRTails</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/FilterProbesByBlatResults.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/FilterProbesByBlatResults.html" title="class in broad.pda.capture.designer">FilterProbesByBlatResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/PcrTailDesigner.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/PcrTailDesigner.html" title="class in broad.pda.capture.designer">PcrTailDesigner</A>
<DD>Just write subprimers to a file
<DT><A HREF="../broad/pda/capture/designer/SmatchLike.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/SmatchLike.html" title="class in broad.pda.capture.designer">SmatchLike</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/countreads/FastqLibraryStats.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.countreads.<A HREF="../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads">FastqLibraryStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/BatchNorm.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/BatchNorm.html" title="class in broad.pda.differentialExpression">BatchNorm</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/CisVsTrans.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/CisVsTrans.html" title="class in broad.pda.differentialExpression">CisVsTrans</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/CollapseProjections.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/CollapseProjections.html" title="class in broad.pda.differentialExpression">CollapseProjections</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/ConvertToDiscreteMatrix.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/ConvertToDiscreteMatrix.html" title="class in broad.pda.differentialExpression">ConvertToDiscreteMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/FixedControlDifferentialExpression.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/FixedControlDifferentialExpression.html" title="class in broad.pda.differentialExpression">FixedControlDifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/FoldSignificance.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/FoldSignificance.html" title="class in broad.pda.differentialExpression">FoldSignificance</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/GrabProbes.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/GrabProbes.html" title="class in broad.pda.differentialExpression">GrabProbes</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/GSAProjection.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/GSAProjection.html" title="class in broad.pda.differentialExpression">GSAProjection</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/MakePositionalGeneSets.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/MakePositionalGeneSets.html" title="class in broad.pda.differentialExpression">MakePositionalGeneSets</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/MergeGSEAOutputIntoMatrix.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/MergeGSEAOutputIntoMatrix.html" title="class in broad.pda.differentialExpression">MergeGSEAOutputIntoMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/RunAllGSEABatch.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/RunAllGSEABatch.html" title="class in broad.pda.differentialExpression">RunAllGSEABatch</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/RunDifferentialExpression.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/RunDifferentialExpression.html" title="class in broad.pda.differentialExpression">RunDifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/RunGSEA.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/RunGSEA.html" title="class in broad.pda.differentialExpression">RunGSEA</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/RunTimeSeries.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/RunTimeSeries.html" title="class in broad.pda.differentialExpression">RunTimeSeries</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/SplitByName.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/SplitByName.html" title="class in broad.pda.differentialExpression">SplitByName</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/functionalAnnotation.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/functionalAnnotation.html" title="class in broad.pda.gene">functionalAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborAnalysis.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborAnalysis.html" title="class in broad.pda.gene">NeighborAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentToGCT.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentToGCT.html" title="class in broad.pda.geneexpression.agilent">AgilentToGCT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/analysis/ProcessDataset.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.analysis.<A HREF="../broad/pda/geneexpression/agilent/analysis/ProcessDataset.html" title="class in broad.pda.geneexpression.agilent.analysis">ProcessDataset</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/ClassifyAgilentLincProbes.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/ClassifyAgilentLincProbes.html" title="class in broad.pda.geneexpression.agilent">ClassifyAgilentLincProbes</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/CollapseProbesByMinPValue.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/CollapseProbesByMinPValue.html" title="class in broad.pda.geneexpression.agilent">CollapseProbesByMinPValue</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/CorrelationAgilentNanostring.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/CorrelationAgilentNanostring.html" title="class in broad.pda.geneexpression.agilent">CorrelationAgilentNanostring</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/CorrelationBetweenReplicates.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/CorrelationBetweenReplicates.html" title="class in broad.pda.geneexpression.agilent">CorrelationBetweenReplicates</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/ExtractSamplesAndLabel.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/ExtractSamplesAndLabel.html" title="class in broad.pda.geneexpression.agilent">ExtractSamplesAndLabel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/FilterMatrixByMinCutoff.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/FilterMatrixByMinCutoff.html" title="class in broad.pda.geneexpression.agilent">FilterMatrixByMinCutoff</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/FindProbesForGeneSet.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/FindProbesForGeneSet.html" title="class in broad.pda.geneexpression.agilent">FindProbesForGeneSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/MakeCLSFiles.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/MakeCLSFiles.html" title="class in broad.pda.geneexpression.agilent">MakeCLSFiles</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/MakeCLSFromDescription.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/MakeCLSFromDescription.html" title="class in broad.pda.geneexpression.agilent">MakeCLSFromDescription</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/QCAgilentResults.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/QCAgilentResults.html" title="class in broad.pda.geneexpression.agilent">QCAgilentResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/ComputeCorrelationOfReplicates.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/ComputeCorrelationOfReplicates.html" title="class in broad.pda.geneexpression.clustering">ComputeCorrelationOfReplicates</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html" title="class in broad.pda.geneexpression.clustering">HierarchicalClustering</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/RankOrder.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/RankOrder.html" title="class in broad.pda.geneexpression.clustering">RankOrder</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ComputeHypergeomtricEnrichment.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ComputeHypergeomtricEnrichment.html" title="class in broad.pda.geneexpression">ComputeHypergeomtricEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/CountNums.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/CountNums.html" title="class in broad.pda.geneexpression">CountNums</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/FilterGCTFile.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/FilterGCTFile.html" title="class in broad.pda.geneexpression">FilterGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/geneExpUtils.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/geneExpUtils.html" title="class in broad.pda.geneexpression">geneExpUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/GlobalStats.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/GlobalStats.html" title="class in broad.pda.geneexpression">GlobalStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/GrabSpecificColumns.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/GrabSpecificColumns.html" title="class in broad.pda.geneexpression">GrabSpecificColumns</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/PaGEJava.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/PaGEJava.html" title="class in broad.pda.geneexpression">PaGEJava</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/PreprocessDataSet.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/PreprocessDataSet.html" title="class in broad.pda.geneexpression">PreprocessDataSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html" title="class in broad.pda.methylation">FilterFastqForMethylationAlignment</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation">FilterSamByBaseComposition</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html" title="class in broad.pda.methylation">FilterSamForMethylationAlignment</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.html" title="class in broad.pda.methylation">HybridCaptureDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/OutputPadlock.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/OutputPadlock.html" title="class in broad.pda.methylation">OutputPadlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/PadlockAnalysis.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/PadlockAnalysis.html" title="class in broad.pda.methylation">PadlockAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/WriteCpGBED.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/WriteCpGBED.html" title="class in broad.pda.methylation">WriteCpGBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rap/misc/FindEnrichedDepletedRegions.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.rap.misc.<A HREF="../broad/pda/rap/misc/FindEnrichedDepletedRegions.html" title="class in broad.pda.rap.misc">FindEnrichedDepletedRegions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/rnai/ExtractSequence.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.rnai.<A HREF="../broad/pda/rnai/ExtractSequence.html" title="class in broad.pda.rnai">ExtractSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/PairedEndTDF.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/PairedEndTDF.html" title="class in broad.pda.samtools">PairedEndTDF</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/samtools/SAMFragmentWriter.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMFragmentWriter.html" title="class in broad.pda.samtools">SAMFragmentWriter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/alignment/AlignmentUtils.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.alignment.<A HREF="../broad/pda/seq/alignment/AlignmentUtils.html" title="class in broad.pda.seq.alignment">AlignmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/chip/ComputeEnrichments.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.chip.<A HREF="../broad/pda/seq/chip/ComputeEnrichments.html" title="class in broad.pda.seq.chip">ComputeEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/chip/FindDiffrentialEnrichedRegions.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.chip.<A HREF="../broad/pda/seq/chip/FindDiffrentialEnrichedRegions.html" title="class in broad.pda.seq.chip">FindDiffrentialEnrichedRegions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/chip/PeakFilter.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.chip.<A HREF="../broad/pda/seq/chip/PeakFilter.html" title="class in broad.pda.seq.chip">PeakFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.html" title="class in broad.pda.seq.fastq">FastqSplitterByBarcode</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/SplitAndTrimIndex.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/SplitAndTrimIndex.html" title="class in broad.pda.seq.fastq">SplitAndTrimIndex</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/SplitFastqByIndex.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/SplitFastqByIndex.html" title="class in broad.pda.seq.fastq">SplitFastqByIndex</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/BatchedMultiSampleScanPeakCaller.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/BatchedMultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">BatchedMultiSampleScanPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanSkellamAnalysis.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">MultiSampleScanSkellamAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/ReadSimulator.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/ReadSimulator.html" title="class in broad.pda.seq.protection">ReadSimulator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamAnalysis</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html" title="class in broad.pda.seq.rap">BuildRatioNullDistribution</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/CountReads.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CountReads.html" title="class in broad.pda.seq.rap">CountReads</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/GCContent.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GCContent.html" title="class in broad.pda.seq.rap">GCContent</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/RAPPeakCaller.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RAPPeakCaller.html" title="class in broad.pda.seq.rap">RAPPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html" title="class in broad.pda.seq.rap">RatioPermutationPeakCaller</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/ShuffleBAM.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBAM.html" title="class in broad.pda.seq.rap">ShuffleBAM</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/ShuffleBED.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBED.html" title="class in broad.pda.seq.rap">ShuffleBED</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html" title="class in broad.pda.seq.rap">SkellamKLPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SlideAndCalculate.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCalculate.html" title="class in broad.pda.seq.rap">SlideAndCalculate</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/SlideAndCount.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCount.html" title="class in broad.pda.seq.rap">SlideAndCount</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/rap/UnmarkDuplicates.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/UnmarkDuplicates.html" title="class in broad.pda.seq.rap">UnmarkDuplicates</A>
<DD>Stock main method.
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GlobalSpliceFilter.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GlobalSpliceFilter.html" title="class in broad.pda.seq.segmentation">GlobalSpliceFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/PrecomputeDataAlignmentStats.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/PrecomputeDataAlignmentStats.html" title="class in broad.pda.seq.segmentation">PrecomputeDataAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/utils/ReconstructionComparer.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.seq.segmentation.utils.<A HREF="../broad/pda/seq/segmentation/utils/ReconstructionComparer.html" title="class in broad.pda.seq.segmentation.utils">ReconstructionComparer</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/tools/PSLXToSAM.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.pda.tools.<A HREF="../broad/pda/tools/PSLXToSAM.html" title="class in broad.pda.tools">PSLXToSAM</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/DefineGeneSignatures.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/DefineGeneSignatures.html" title="class in broad.projection.gsa">DefineGeneSignatures</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/ExtractGenes.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/ExtractGenes.html" title="class in broad.projection.gsa">ExtractGenes</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GSAScoring.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GSAScoring.html" title="class in broad.projection.gsa">GSAScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/projection/gsa/GsaUtils.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class broad.projection.gsa.<A HREF="../broad/projection/gsa/GsaUtils.html" title="class in broad.projection.gsa">GsaUtils</A>
<DD>&nbsp;
<DT><A HREF="../Jama/examples/MagicSquareExample.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class Jama.examples.<A HREF="../Jama/examples/MagicSquareExample.html" title="class in Jama.examples">MagicSquareExample</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/debug/PairedEndSam.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.debug.<A HREF="../nextgen/core/debug/PairedEndSam.html" title="class in nextgen.core.debug">PairedEndSam</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/BAMToCoverage.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/BAMToCoverage.html" title="class in nextgen.core.normalize">BAMToCoverage</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html" title="class in nextgen.core.normalize">ComparativeCountsMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/NormalizeCountsToTranscriptAverage.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/NormalizeCountsToTranscriptAverage.html" title="class in nextgen.core.normalize">NormalizeCountsToTranscriptAverage</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/ConvertBamToPairedEnd.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/ConvertBamToPairedEnd.html" title="class in nextgen.core.programs">ConvertBamToPairedEnd</A>
<DD>Stock main method.
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/ReadAlignmentStats.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/ReadAlignmentStats.html" title="class in nextgen.core.programs">ReadAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/WigReader.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/WigReader.html" title="class in nextgen.core.readers">WigReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html" title="class in nextgen.core.scripture">BuildScriptureCoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/CombinationGenerator.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/CombinationGenerator.html" title="class in nextgen.core.scripture">CombinationGenerator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/EndPileUpScanner.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/EndPileUpScanner.html" title="class in nextgen.core.scripture">EndPileUpScanner</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/ReconstructionPostProcessor.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/ReconstructionPostProcessor.html" title="class in nextgen.core.scripture">ReconstructionPostProcessor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/statistics/ClassifyReconstructions.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.scripture.statistics.<A HREF="../nextgen/core/scripture/statistics/ClassifyReconstructions.html" title="class in nextgen.core.scripture.statistics">ClassifyReconstructions</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/statistics/ConnectDisconnectedTranscripts.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.scripture.statistics.<A HREF="../nextgen/core/scripture/statistics/ConnectDisconnectedTranscripts.html" title="class in nextgen.core.scripture.statistics">ConnectDisconnectedTranscripts</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/statistics/OldReconstructionCompare.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.scripture.statistics.<A HREF="../nextgen/core/scripture/statistics/OldReconstructionCompare.html" title="class in nextgen.core.scripture.statistics">OldReconstructionCompare</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.structure.<A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html" title="class in nextgen.core.structure">PairwiseWindowHybridization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestAnnotations.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestAnnotations.html" title="class in nextgen.core.tests">TestAnnotations</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestAnnotationTrimming.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestAnnotationTrimming.html" title="class in nextgen.core.tests">TestAnnotationTrimming</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestDynamicScoring.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestDynamicScoring.html" title="class in nextgen.core.tests">TestDynamicScoring</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestEnrichmentByFeature.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestEnrichmentByFeature.html" title="class in nextgen.core.tests">TestEnrichmentByFeature</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestReadCounting.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestReadCounting.html" title="class in nextgen.core.tests">TestReadCounting</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/utils/AlignmentModelGlobalStats.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.utils.<A HREF="../nextgen/core/utils/AlignmentModelGlobalStats.html" title="class in nextgen.core.utils">AlignmentModelGlobalStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers">FilteredBamWriter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/WigWriter.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/WigWriter.html" title="class in nextgen.core.writers">WigWriter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.html" title="class in nextgen.editing">CRISPRDesigner</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/rnaprotein/analysis/ComputeGeneEnrichment.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class nextgen.rnaprotein.analysis.<A HREF="../nextgen/rnaprotein/analysis/ComputeGeneEnrichment.html" title="class in nextgen.rnaprotein.analysis">ComputeGeneEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborsNullModel.html#makeBinnedIntervalModel(boolean)"><B>makeBinnedIntervalModel(boolean)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborsNullModel.html" title="class in broad.pda.gene">NeighborsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html#makeBowtie2Index(java.lang.String, java.lang.String, java.lang.String, java.lang.String)"><B>makeBowtie2Index(String, String, String, String)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html" title="class in broad.core.util">PipelineAutomator.AlignmentUtils</A>
<DD>Make bowtie2 index of fasta file
<DT><A HREF="../broad/pda/geneexpression/agilent/MakeCLSFiles.html" title="class in broad.pda.geneexpression.agilent"><B>MakeCLSFiles</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/MakeCLSFiles.html#MakeCLSFiles(java.lang.String, java.io.File, java.lang.String)"><B>MakeCLSFiles(String, File, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/MakeCLSFiles.html" title="class in broad.pda.geneexpression.agilent">MakeCLSFiles</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/MakeCLSFromDescription.html" title="class in broad.pda.geneexpression.agilent"><B>MakeCLSFromDescription</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/MakeCLSFromDescription.html#MakeCLSFromDescription(java.io.File, java.io.File, java.lang.String)"><B>MakeCLSFromDescription(File, File, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/MakeCLSFromDescription.html" title="class in broad.pda.geneexpression.agilent">MakeCLSFromDescription</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborsNullModel.html#makeDartThrowingModel()"><B>makeDartThrowingModel()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborsNullModel.html" title="class in broad.pda.gene">NeighborsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#makeDataMatrix(java.io.File, java.util.Collection)"><B>makeDataMatrix(File, Collection&lt;String&gt;)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#makeDataMatrix(java.io.File[])"><B>makeDataMatrix(File[])</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#makeDataMatrix(java.io.File[], java.util.Map)"><B>makeDataMatrix(File[], Map&lt;String, ExpressionExperimentInfo&gt;)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#makeGenes()"><B>makeGenes()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#makeGenes(double)"><B>makeGenes(double)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#makeIntervalTree(java.util.Collection)"><B>makeIntervalTree(Collection&lt;? extends Annotation&gt;)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#makeIntervalTreeFor3Prime(java.io.File, int)"><B>makeIntervalTreeFor3Prime(File, int)</B></A> - 
Static method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#makeIntervalTreeForGeneExons(java.util.Collection)"><B>makeIntervalTreeForGeneExons(Collection&lt;Gene&gt;)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#makeIntervalTreeForGenes(java.util.Collection)"><B>makeIntervalTreeForGenes(Collection&lt;Gene&gt;)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborsNullModel.html#makeNaiveRandModel(java.lang.String)"><B>makeNaiveRandModel(String)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborsNullModel.html" title="class in broad.pda.gene">NeighborsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AlignmentPair.html#makePairs()"><B>makePairs()</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AlignmentPair.html" title="class in nextgen.core.alignment">AlignmentPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/MakePositionalGeneSets.html" title="class in broad.pda.differentialExpression"><B>MakePositionalGeneSets</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/MakePositionalGeneSets.html#MakePositionalGeneSets(java.io.File, java.lang.String, int)"><B>MakePositionalGeneSets(File, String, int)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/MakePositionalGeneSets.html" title="class in broad.pda.differentialExpression">MakePositionalGeneSets</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/transcriptsNullModel.html#makeRandTranscriptList(int, broad.pda.annotation.BEDFileParser)"><B>makeRandTranscriptList(int, BEDFileParser)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/transcriptsNullModel.html" title="class in broad.pda.gene">transcriptsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#manhattanDistance(double[], double[])"><B>manhattanDistance(double[], double[])</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/MannWhitney.html" title="class in broad.core.math"><B>MannWhitney</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/MannWhitney.html#MannWhitney(double[], double[])"><B>MannWhitney(double[], double[])</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/MannWhitney.html" title="class in broad.core.math">MannWhitney</A>
<DD>Instantiate with arrays of measurements
<DT><A HREF="../broad/core/math/MannWhitney.html#MannWhitney(java.util.ArrayList, java.util.ArrayList)"><B>MannWhitney(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/MannWhitney.html" title="class in broad.core.math">MannWhitney</A>
<DD>Instantiate with lists of measurements
<DT><A HREF="../broad/core/math/MannWhitneyPermutations.html" title="class in broad.core.math"><B>MannWhitneyPermutations</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/MannWhitneyPermutations.html#MannWhitneyPermutations(java.util.List, java.util.List, int)"><B>MannWhitneyPermutations(List&lt;Double&gt;, List&lt;Double&gt;, int)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/MannWhitneyPermutations.html" title="class in broad.core.math">MannWhitneyPermutations</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/utils/AnnotationUtils.html#mapChildToLargestParent(java.util.Map, java.util.Map)"><B>mapChildToLargestParent(Map&lt;String, Collection&lt;T&gt;&gt;, Map&lt;String, Collection&lt;T&gt;&gt;)</B></A> - 
Static method in class nextgen.core.utils.<A HREF="../nextgen/core/utils/AnnotationUtils.html" title="class in nextgen.core.utils">AnnotationUtils</A>
<DD>Map each child annotation to its largest parent defined as annotations overlapping and containing child
<DT><A HREF="../nextgen/core/utils/AnnotationUtils.html#mapChildToParents(java.util.Map, java.util.Map)"><B>mapChildToParents(Map&lt;String, Collection&lt;T&gt;&gt;, Map&lt;String, Collection&lt;T&gt;&gt;)</B></A> - 
Static method in class nextgen.core.utils.<A HREF="../nextgen/core/utils/AnnotationUtils.html" title="class in nextgen.core.utils">AnnotationUtils</A>
<DD>Map each child annotation to its set of parents defined as annotations overlapping and containing child
<DT><A HREF="../broad/core/overlaputils/GeneSetIntersect.html#mapGenesToOverlappers(java.util.Map, java.util.Map)"><B>mapGenesToOverlappers(Map&lt;String, Collection&lt;Gene&gt;&gt;, Map&lt;String, Collection&lt;Gene&gt;&gt;)</B></A> - 
Static method in class broad.core.overlaputils.<A HREF="../broad/core/overlaputils/GeneSetIntersect.html" title="class in broad.core.overlaputils">GeneSetIntersect</A>
<DD>Map each gene to the collection of overlappers in another set
<DT><A HREF="../nextgen/core/readFilters/MappingQualityFilter.html" title="class in nextgen.core.readFilters"><B>MappingQualityFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/MappingQualityFilter.html#MappingQualityFilter(double, double)"><B>MappingQualityFilter(double, double)</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/MappingQualityFilter.html" title="class in nextgen.core.readFilters">MappingQualityFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/MappingQualityFilter.html#MappingQualityFilter(double)"><B>MappingQualityFilter(double)</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/MappingQualityFilter.html" title="class in nextgen.core.readFilters">MappingQualityFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/ReadSimulator.html#mapSamRecord(net.sf.samtools.SAMRecord, nextgen.core.annotation.Gene, net.sf.samtools.SAMFileHeader)"><B>mapSamRecord(SAMRecord, Gene, SAMFileHeader)</B></A> - 
Static method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/ReadSimulator.html" title="class in broad.pda.seq.protection">ReadSimulator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.html#markDuplicates(int, net.sf.samtools.SAMFileReader, net.sf.samtools.BAMFileWriter, boolean, int, int, double, int)"><B>markDuplicates(int, SAMFileReader, BAMFileWriter, boolean, int, int, double, int)</B></A> - 
Static method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.html" title="class in nextgen.core.programs"><B>MarkDuplicatesFromNameUMI</B></A> - Class in <A HREF="../nextgen/core/programs/package-summary.html">nextgen.core.programs</A><DD>Utility to go through a sorted alignment file and mark
 likely PCR duplicates based on both alignment location 
 and a molecular barcode added to the read name.<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.html#MarkDuplicatesFromNameUMI()"><B>MarkDuplicatesFromNameUMI()</B></A> - 
Constructor for class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs"><B>MarkDuplicatesFromNameUMI.MarkDuplicatesResults</B></A> - Class in <A HREF="../nextgen/core/programs/package-summary.html">nextgen.core.programs</A><DD>&nbsp;<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#MarkDuplicatesFromNameUMI.MarkDuplicatesResults(int, int, double)"><B>MarkDuplicatesFromNameUMI.MarkDuplicatesResults(int, int, double)</B></A> - 
Constructor for class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm"><B>MarkovModel</B></A>&lt;<A HREF="../broad/core/hmm/MarkovModel.html" title="type parameter in MarkovModel">T</A>&gt; - Class in <A HREF="../broad/core/hmm/package-summary.html">broad.core.hmm</A><DD>&nbsp;<DT><A HREF="../broad/core/hmm/MarkovModel.html#MarkovModel(int)"><B>MarkovModel(int)</B></A> - 
Constructor for class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm">MarkovModel</A>
<DD>Initialize the Model by specifying the number of states.
<DT><A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html" title="class in broad.core.hmm"><B>MarkovModel.BackwardResult</B></A>&lt;<A HREF="../broad/core/hmm/MarkovModel.BackwardResult.html" title="type parameter in MarkovModel.BackwardResult">T</A>&gt; - Class in <A HREF="../broad/core/hmm/package-summary.html">broad.core.hmm</A><DD>&nbsp;<DT><A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="class in broad.core.hmm"><B>MarkovModel.ForwardResult</B></A>&lt;<A HREF="../broad/core/hmm/MarkovModel.ForwardResult.html" title="type parameter in MarkovModel.ForwardResult">T</A>&gt; - Class in <A HREF="../broad/core/hmm/package-summary.html">broad.core.hmm</A><DD>&nbsp;<DT><A HREF="../broad/core/hmm/MarkovState.html" title="interface in broad.core.hmm"><B>MarkovState</B></A>&lt;<A HREF="../broad/core/hmm/MarkovState.html" title="type parameter in MarkovState">T</A>&gt; - Interface in <A HREF="../broad/core/hmm/package-summary.html">broad.core.hmm</A><DD>&nbsp;<DT><A HREF="../jaligner/Markups.html" title="class in jaligner"><B>Markups</B></A> - Class in <A HREF="../jaligner/package-summary.html">jaligner</A><DD>Markups line characters.<DT><A HREF="../jaligner/Markups.html#Markups()"><B>Markups()</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/Markups.html" title="class in jaligner">Markups</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html#MASK_FILE"><B>MASK_FILE</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap">GenomeCommandLineProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#MASK_FILE"><B>MASK_FILE</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem"><B>MaskedGenomicSpace</B></A> - Class in <A HREF="../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A><DD>&nbsp;<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#MaskedGenomicSpace(java.util.Map, java.util.Map, int)"><B>MaskedGenomicSpace(Map&lt;String, Integer&gt;, Map&lt;String, Collection&lt;Annotation&gt;&gt;, int)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>This is meant to represent a coordinate space that excludes masked regions
 This way we wont count reads in the global stats in these regions
 Wont use them in estimating parameters or length calculations
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#MaskedGenomicSpace(java.util.Map, java.util.Map, int, boolean)"><B>MaskedGenomicSpace(Map&lt;String, Integer&gt;, Map&lt;String, Collection&lt;Annotation&gt;&gt;, int, boolean)</B></A> - 
Constructor for class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>This is meant to represent a coordinate space that excludes masked regions
 This way we wont count reads in the global stats in these regions
 Wont use them in estimating parameters or length calculations
<DT><A HREF="../broad/core/sequence/Sequence.html#maskSoftmaskedRegions()"><B>maskSoftmaskedRegions()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#match(broad.core.sequence.SequenceRegion, double[], float)"><B>match(SequenceRegion, double[], float)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SequenceMotif.html#match(broad.core.sequence.Sequence)"><B>match(Sequence)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotif.html" title="class in broad.core.motif">SequenceMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SequenceMotif.html#match(broad.core.sequence.Sequence, java.util.List, int)"><B>match(Sequence, List&lt;SequenceRegion&gt;, int)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotif.html" title="class in broad.core.motif">SequenceMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SequenceMotifCluster.html#match(broad.core.sequence.Sequence)"><B>match(Sequence)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotifCluster.html" title="class in broad.core.motif">SequenceMotifCluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/SequenceMotifCluster.html#match(broad.core.sequence.Sequence, boolean)"><B>match(Sequence, boolean)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/SequenceMotifCluster.html" title="class in broad.core.motif">SequenceMotifCluster</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#mateCigarFlag"><B>mateCigarFlag</B></A> - 
Static variable in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#mateLineFlag"><B>mateLineFlag</B></A> - 
Static variable in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#mateSequenceFlag"><B>mateSequenceFlag</B></A> - 
Static variable in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>Writes a file to represent a BAM file where the paired end reads are concatenated
 We will represent the basic alignment start and end for the whole fragment
 The attributes will have information for the component parts
<DT><A HREF="../Jama/util/Maths.html" title="class in Jama.util"><B>Maths</B></A> - Class in <A HREF="../Jama/util/package-summary.html">Jama.util</A><DD>&nbsp;<DT><A HREF="../Jama/util/Maths.html#Maths()"><B>Maths()</B></A> - 
Constructor for class Jama.util.<A HREF="../Jama/util/Maths.html" title="class in Jama.util">Maths</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/MathUtil.html" title="class in broad.core.math"><B>MathUtil</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/MathUtil.html#MathUtil()"><B>MathUtil()</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/MathUtil.html" title="class in broad.core.math">MathUtil</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/matrix/MatricesCompartor.html" title="class in jaligner.matrix"><B>MatricesCompartor</B></A> - Class in <A HREF="../jaligner/matrix/package-summary.html">jaligner.matrix</A><DD>Comparator to sort the scoring matrices by their names.<DT><A HREF="../jaligner/matrix/MatricesCompartor.html#MatricesCompartor()"><B>MatricesCompartor()</B></A> - 
Constructor for class jaligner.matrix.<A HREF="../jaligner/matrix/MatricesCompartor.html" title="class in jaligner.matrix">MatricesCompartor</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/matrix/Matrix.html" title="class in jaligner.matrix"><B>Matrix</B></A> - Class in <A HREF="../jaligner/matrix/package-summary.html">jaligner.matrix</A><DD>Scoring matrix.<DT><A HREF="../jaligner/matrix/Matrix.html#Matrix(java.lang.String, float[][])"><B>Matrix(String, float[][])</B></A> - 
Constructor for class jaligner.matrix.<A HREF="../jaligner/matrix/Matrix.html" title="class in jaligner.matrix">Matrix</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html" title="class in Jama"><B>Matrix</B></A> - Class in <A HREF="../Jama/package-summary.html">Jama</A><DD>Jama = Java Matrix class.<DT><A HREF="../Jama/Matrix.html#Matrix(int, int)"><B>Matrix(int, int)</B></A> - 
Constructor for class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Construct an m-by-n matrix of zeros.
<DT><A HREF="../Jama/Matrix.html#Matrix(int, int, double)"><B>Matrix(int, int, double)</B></A> - 
Constructor for class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Construct an m-by-n constant matrix.
<DT><A HREF="../Jama/Matrix.html#Matrix(double[][])"><B>Matrix(double[][])</B></A> - 
Constructor for class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Construct a matrix from a 2-D array.
<DT><A HREF="../Jama/Matrix.html#Matrix(double[][], int, int)"><B>Matrix(double[][], int, int)</B></A> - 
Constructor for class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Construct a matrix quickly without checking arguments.
<DT><A HREF="../Jama/Matrix.html#Matrix(double[], int)"><B>Matrix(double[], int)</B></A> - 
Constructor for class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Construct a matrix from a one-dimensional packed array
<DT><A HREF="../jaligner/matrix/MatrixGenerator.html" title="class in jaligner.matrix"><B>MatrixGenerator</B></A> - Class in <A HREF="../jaligner/matrix/package-summary.html">jaligner.matrix</A><DD>A utility class to generate a scoring matrix from a fixed match and mismatch
 scoring scheme.<DT><A HREF="../jaligner/matrix/MatrixGenerator.html#MatrixGenerator()"><B>MatrixGenerator()</B></A> - 
Constructor for class jaligner.matrix.<A HREF="../jaligner/matrix/MatrixGenerator.html" title="class in jaligner.matrix">MatrixGenerator</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/matrix/MatrixLoader.html" title="class in jaligner.matrix"><B>MatrixLoader</B></A> - Class in <A HREF="../jaligner/matrix/package-summary.html">jaligner.matrix</A><DD>Scoring matrices loader from a jar file or a file system.<DT><A HREF="../jaligner/matrix/MatrixLoader.html#MatrixLoader()"><B>MatrixLoader()</B></A> - 
Constructor for class jaligner.matrix.<A HREF="../jaligner/matrix/MatrixLoader.html" title="class in jaligner.matrix">MatrixLoader</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/matrix/MatrixLoaderException.html" title="class in jaligner.matrix"><B>MatrixLoaderException</B></A> - Exception in <A HREF="../jaligner/matrix/package-summary.html">jaligner.matrix</A><DD>&nbsp;<DT><A HREF="../jaligner/matrix/MatrixLoaderException.html#MatrixLoaderException(java.lang.String)"><B>MatrixLoaderException(String)</B></A> - 
Constructor for exception jaligner.matrix.<A HREF="../jaligner/matrix/MatrixLoaderException.html" title="class in jaligner.matrix">MatrixLoaderException</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures"><B>MatrixWithHeaders</B></A> - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(java.lang.String)"><B>MatrixWithHeaders(String)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(java.io.BufferedReader)"><B>MatrixWithHeaders(BufferedReader)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(Jama.Matrix, java.util.List, java.util.List)"><B>MatrixWithHeaders(Matrix, List&lt;String&gt;, List&lt;String&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(Jama.Matrix, java.util.List, java.util.List, java.util.List)"><B>MatrixWithHeaders(Matrix, List&lt;String&gt;, List&lt;String&gt;, List&lt;String&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(java.util.List, java.util.List, java.util.Map)"><B>MatrixWithHeaders(List&lt;String&gt;, List&lt;String&gt;, Map&lt;String, String&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(broad.core.datastructures.MatrixWithHeaders, java.util.List, java.util.List)"><B>MatrixWithHeaders(MatrixWithHeaders, List&lt;String&gt;, List&lt;String&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Create a new matrix that has a subset of columns and rows of the original matrix
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(java.util.List, java.util.List)"><B>MatrixWithHeaders(List&lt;String&gt;, List&lt;String&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(java.util.List, java.util.List, java.util.List)"><B>MatrixWithHeaders(List&lt;String&gt;, List&lt;String&gt;, List&lt;String&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(java.lang.String, java.lang.String)"><B>MatrixWithHeaders(String, String)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(java.util.List, java.util.ArrayList, java.util.List)"><B>MatrixWithHeaders(List&lt;List&lt;Double&gt;&gt;, ArrayList&lt;String&gt;, List&lt;String&gt;)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#MatrixWithHeaders(java.util.ArrayList, java.util.ArrayList, int)"><B>MatrixWithHeaders(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, int)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaderUtils.html" title="class in broad.core.datastructures"><B>MatrixWithHeaderUtils</B></A> - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/MatrixWithHeaderUtils.html#MatrixWithHeaderUtils()"><B>MatrixWithHeaderUtils()</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaderUtils.html" title="class in broad.core.datastructures">MatrixWithHeaderUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#max()"><B>max()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the greatest interval in the tree.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#max(broad.core.datastructures.Interval)"><B>max(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the latest interval in the tree less than or equal to the specified interval.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#max(int, int)"><B>max(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the latest interval in the tree less than or equal to the specified interval.
<DT><A HREF="../broad/core/math/MathUtil.html#max(java.util.Collection)"><B>max(Collection&lt;? extends Number&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/MathUtil.html" title="class in broad.core.math">MathUtil</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#max(java.util.Collection)"><B>max(Collection&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#max(double[])"><B>max(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#max(int[])"><B>max(int[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#MAX_DISTANCE"><B>MAX_DISTANCE</B></A> - 
Static variable in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html#MAX_FRAGMENT_LENGTH"><B>MAX_FRAGMENT_LENGTH</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap">GenomeCommandLineProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/PairedEndTDF.html#MAX_INSERT"><B>MAX_INSERT</B></A> - 
Variable in class broad.pda.samtools.<A HREF="../broad/pda/samtools/PairedEndTDF.html" title="class in broad.pda.samtools">PairedEndTDF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#MAX_INSERT"><B>MAX_INSERT</B></A> - 
Static variable in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/FWERDistribution.html#maxDist"><B>maxDist</B></A> - 
Variable in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/FWERDistribution.html" title="class in broad.pda.differentialExpression">FWERDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#maxGCContent"><B>maxGCContent</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html" title="class in broad.core.annotation"><B>MaximumContiguousSubsequence</B></A> - Class in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html#MaximumContiguousSubsequence()"><B>MaximumContiguousSubsequence()</B></A> - 
Constructor for class broad.core.annotation.<A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html" title="class in broad.core.annotation">MaximumContiguousSubsequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#maxInt(java.util.Collection)"><B>maxInt(Collection&lt;Integer&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#maxMAD(Jama.Matrix)"><B>maxMAD(Matrix)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#maxmean(double[])"><B>maxmean(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#maxMeltingTemp"><B>maxMeltingTemp</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#maxMeltingTempDiff"><B>maxMeltingTempDiff</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#maxMisspriming"><B>maxMisspriming</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#maxNumPrimersToReturn"><B>maxNumPrimersToReturn</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#maxPolyX"><B>maxPolyX</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#maxPrimerSize"><B>maxPrimerSize</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#maxProductSize"><B>maxProductSize</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#maxSelfComplementarity()"><B>maxSelfComplementarity()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#maxStdev(Jama.Matrix)"><B>maxStdev(Matrix)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html#maxSubSum3(int[])"><B>maxSubSum3(int[])</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html" title="class in broad.core.annotation">MaximumContiguousSubsequence</A>
<DD>Linear-time maximum contiguous subsequence sum algorithm.
<DT><A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html#maxSubSum3(double[])"><B>maxSubSum3(double[])</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html" title="class in broad.core.annotation">MaximumContiguousSubsequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#mayHaveBlocks()"><B>mayHaveBlocks()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BED.html#mayHaveBlocks()"><B>mayHaveBlocks()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BED.html" title="class in broad.core.annotation">BED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GenomicAnnotation.html#mayHaveBlocks()"><B>mayHaveBlocks()</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/GenomicAnnotation.html" title="interface in broad.core.annotation">GenomicAnnotation</A>
<DD>Returns true if the annotation (like a full BED or a RefSeq) has sub annotations like exons or blocks
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#mayHaveBlocks()"><B>mayHaveBlocks()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#mean(double[])"><B>mean(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#mean(Jama.Matrix)"><B>mean(Matrix)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#mean(java.util.Collection)"><B>mean(Collection&lt;? extends Number&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#meanNormalizeColumns()"><B>meanNormalizeColumns()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Substitutes each value in a column with the ratio of the value and the mean of column value
 Only applicable to positive value matrices.
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#meanNormalizeRows()"><B>meanNormalizeRows()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Substitutes each value in a row with the ratio of the value and the mean of row value
 Only applicable to positive value matrices.
<DT><A HREF="../broad/pda/geneexpression/clustering/AbsolutePearsonFunction.html#measure(double[], double[])"><B>measure(double[], double[])</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/AbsolutePearsonFunction.html" title="class in broad.pda.geneexpression.clustering">AbsolutePearsonFunction</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/ClusterDistanceFunction.html#measure(double[], double[])"><B>measure(double[], double[])</B></A> - 
Method in interface broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/ClusterDistanceFunction.html" title="interface in broad.pda.geneexpression.clustering">ClusterDistanceFunction</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/EuclideanDistance.html#measure(double[], double[])"><B>measure(double[], double[])</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/EuclideanDistance.html" title="class in broad.pda.geneexpression.clustering">EuclideanDistance</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/PearsonDistance.html#measure(double[], double[])"><B>measure(double[], double[])</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/PearsonDistance.html" title="class in broad.pda.geneexpression.clustering">PearsonDistance</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#median(java.util.Collection)"><B>median(Collection&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#median(double[])"><B>median(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#median(int[])"><B>median(int[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/BatchNorm.html#median(double[])"><B>median(double[])</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/BatchNorm.html" title="class in broad.pda.differentialExpression">BatchNorm</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#medianCenterColumns()"><B>medianCenterColumns()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#medianCenterColumnsThis()"><B>medianCenterColumnsThis()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Mean centers the columns of the current matrix rather than creating a new one.
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#medianCenterRows()"><B>medianCenterRows()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#medianCenterRowsThis()"><B>medianCenterRowsThis()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Mean centers the rows of the current matrix rather than creating a new one.
<DT><A HREF="../broad/core/math/Statistics.html#medianCollection(java.util.Collection)"><B>medianCollection(Collection&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#medianNormalizeColumns()"><B>medianNormalizeColumns()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Substitutes each value in a column with the ratio of the value and the mean of column value
 Only applicable to positive value matrices.
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#medianNormalizeRows()"><B>medianNormalizeRows()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Substitutes each value in a row with the ratio of the value and the mean of row value
 Only applicable to positive value matrices.
<DT><A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme"><B>MemeMotif</B></A> - Class in <A HREF="../broad/core/motif/meme/package-summary.html">broad.core.motif.meme</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/meme/MemeMotif.html#MemeMotif(java.lang.String)"><B>MemeMotif(String)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeMotif.html" title="class in broad.core.motif.meme">MemeMotif</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeMotif.MatchedSequence.html" title="class in broad.core.motif.meme"><B>MemeMotif.MatchedSequence</B></A> - Class in <A HREF="../broad/core/motif/meme/package-summary.html">broad.core.motif.meme</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/meme/MemeResultReader.html" title="class in broad.core.motif.meme"><B>MemeResultReader</B></A> - Class in <A HREF="../broad/core/motif/meme/package-summary.html">broad.core.motif.meme</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/meme/MemeResultReader.html#MemeResultReader(java.io.BufferedReader)"><B>MemeResultReader(BufferedReader)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeResultReader.html" title="class in broad.core.motif.meme">MemeResultReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeResultReader.html#MemeResultReader(java.io.InputStream)"><B>MemeResultReader(InputStream)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeResultReader.html" title="class in broad.core.motif.meme">MemeResultReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeResultReader.html#MemeResultReader(java.lang.String)"><B>MemeResultReader(String)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeResultReader.html" title="class in broad.core.motif.meme">MemeResultReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/meme/MemeResultReader.html#MemeResultReader(java.io.File)"><B>MemeResultReader(File)</B></A> - 
Constructor for class broad.core.motif.meme.<A HREF="../broad/core/motif/meme/MemeResultReader.html" title="class in broad.core.motif.meme">MemeResultReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#memoryStats()"><B>memoryStats()</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html" title="class in nextgen.core.scripture">BuildScriptureCoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#merge()"><B>merge()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Merge list of annotations by combining overlapping annotations into one.
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#merge(int)"><B>merge(int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Merge list of annotations by combining overlapping annotations into one.
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#merge()"><B>merge()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#merge(broad.pda.datastructures.Alignments, broad.pda.datastructures.Alignments)"><B>merge(Alignments, Alignments)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#merge(broad.pda.datastructures.Alignments)"><B>merge(Alignments)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html#merge(java.util.Map)"><B>merge(Map&lt;String, Collection&lt;Gene&gt;&gt;)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html#merge(java.util.Collection)"><B>merge(Collection&lt;Gene&gt;)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#mergeAllOverlappers(java.util.TreeSet)"><B>mergeAllOverlappers(TreeSet&lt;BasicLightweightAnnotation&gt;)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>Merge overlapping regions into a single region and leave singletons alone
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#mergeByGtfGeneId()"><B>mergeByGtfGeneId()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html#mergeByGtfGeneId(java.lang.String, java.lang.String, java.lang.String, java.lang.String, boolean)"><B>mergeByGtfGeneId(String, String, String, String, boolean)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/MergeGSEAOutputIntoMatrix.html" title="class in broad.pda.differentialExpression"><B>MergeGSEAOutputIntoMatrix</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/MergeGSEAOutputIntoMatrix.html#MergeGSEAOutputIntoMatrix(java.io.File[], java.lang.String)"><B>MergeGSEAOutputIntoMatrix(File[], String)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/MergeGSEAOutputIntoMatrix.html" title="class in broad.pda.differentialExpression">MergeGSEAOutputIntoMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/utils/AnnotationUtils.html#mergeOverlappingBlocks(java.util.TreeSet)"><B>mergeOverlappingBlocks(TreeSet&lt;Annotation&gt;)</B></A> - 
Static method in class nextgen.core.utils.<A HREF="../nextgen/core/utils/AnnotationUtils.html" title="class in nextgen.core.utils">AnnotationUtils</A>
<DD>Merge annotations that overlap in the same orientation as each other, and leave singleton windows the same
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#mergePeaks(java.util.Collection)"><B>mergePeaks(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html#mergePeaks(java.util.Collection)"><B>mergePeaks(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RAPPeakCaller.html#mergePeaks(java.util.Collection)"><B>mergePeaks(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RAPPeakCaller.html" title="class in broad.pda.seq.rap">RAPPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/analysis/PeakCaller.html#mergePeaks(java.util.Collection)"><B>mergePeaks(Collection&lt;Annotation&gt;)</B></A> - 
Method in interface nextgen.core.analysis.<A HREF="../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A>
<DD>Merge a collection of peaks
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#mergeRecords(broad.core.sequence.Sequence, java.lang.String)"><B>mergeRecords(Sequence, String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html#mergeSets(java.util.Map, java.util.Map)"><B>mergeSets(Map&lt;String, IntervalTree&lt;Gene&gt;&gt;, Map&lt;String, IntervalTree&lt;Gene&gt;&gt;)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#metaAvg(java.util.Collection)"><B>metaAvg(Collection&lt;double[]&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#methylated"><B>methylated</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/MethylationUtils.html" title="class in broad.pda.methylation"><B>MethylationUtils</B></A> - Class in <A HREF="../broad/pda/methylation/package-summary.html">broad.pda.methylation</A><DD>&nbsp;<DT><A HREF="../broad/pda/methylation/MethylationUtils.html#MethylationUtils()"><B>MethylationUtils()</B></A> - 
Constructor for class broad.pda.methylation.<A HREF="../broad/pda/methylation/MethylationUtils.html" title="class in broad.pda.methylation">MethylationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html#METRICS"><B>METRICS</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation">FilterSamByBaseComposition</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#MIDDLE_LENGTH"><B>MIDDLE_LENGTH</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/WigWriter.html#MIDPOINT_POSITION_DESCRIPTION"><B>MIDPOINT_POSITION_DESCRIPTION</B></A> - 
Static variable in class nextgen.core.writers.<A HREF="../nextgen/core/writers/WigWriter.html" title="class in nextgen.core.writers">WigWriter</A>
<DD>Description of fragment midpoint position calculation
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#min()"><B>min()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the least interval in the tree.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#min(broad.core.datastructures.Interval)"><B>min(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the earliest interval in the tree greater than or equal to the specified interval.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#min(int, int)"><B>min(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the earliest interval in the tree greater than or equal to the specified interval.
<DT><A HREF="../broad/core/math/Statistics.html#min(java.util.Collection)"><B>min(Collection&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#min(double[])"><B>min(double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#min(int[])"><B>min(int[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html#MIN_MAPPING_QUALITY"><B>MIN_MAPPING_QUALITY</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap">GenomeCommandLineProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#MIN_PRIMER_DES_SPACE"><B>MIN_PRIMER_DES_SPACE</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/qPCRPrimerDesigner.html#MIN_PROD_SIZE"><B>MIN_PROD_SIZE</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/qPCRPrimerDesigner.html" title="class in broad.core.primer3">qPCRPrimerDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/FWERDistribution.html#minDist"><B>minDist</B></A> - 
Variable in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/FWERDistribution.html" title="class in broad.pda.differentialExpression">FWERDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#minGCContent"><B>minGCContent</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#minInt(java.util.Collection)"><B>minInt(Collection&lt;Integer&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#minMeltingTemp"><B>minMeltingTemp</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#minOverlapper(broad.core.datastructures.Interval)"><B>minOverlapper(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the earliest interval in the tree that overlaps the specified interval.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#minOverlapper(int, int)"><B>minOverlapper(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the earliest interval in the tree that overlaps the specified interval.
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#minPrimerSize"><B>minPrimerSize</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#minProductSize"><B>minProductSize</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#minQualityScore"><B>minQualityScore</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#minus(java.util.List)"><B>minus(List&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Removes from this reader all elements that overlap the given list.
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#minus(broad.core.datastructures.MatrixWithHeaders)"><B>minus(MatrixWithHeaders)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#minus(nextgen.core.annotation.Annotation)"><B>minus(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#minus(java.util.List)"><B>minus(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#minus(java.util.Collection)"><B>minus(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#minus(java.util.Collection, boolean)"><B>minus(Collection&lt;Gene&gt;, boolean)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>Removes from this reader all elements that overlap the given list.
<DT><A HREF="../Jama/Matrix.html#minus(Jama.Matrix)"><B>minus(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>C = A - B
<DT><A HREF="../nextgen/core/annotation/Annotation.html#minus(nextgen.core.annotation.Annotation)"><B>minus(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Returns the difference (all regions in this genomic annotation not in the given other annotation)
 between this genomic annotation and the one
<DT><A HREF="../nextgen/core/annotation/Annotation.html#minus(java.util.Collection)"><B>minus(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Returns the difference (all regions in this genomic annotation not in the given list)
 between this genomic annotation and the given list
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#minus(nextgen.core.annotation.AnnotationList)"><B>minus(AnnotationList&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>Returns a new AnnotationList that does not contain any annotations that overlap annotations in other
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#minus(nextgen.core.annotation.Annotation)"><B>minus(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#minus(java.util.Collection)"><B>minus(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#minus(nextgen.core.annotation.CompoundInterval)"><B>minus(CompoundInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#minus(nextgen.core.annotation.Annotation)"><B>minus(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#minus(java.util.Collection)"><B>minus(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#minusEquals(Jama.Matrix)"><B>minusEquals(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>A = A - B
<DT><A HREF="../jaligner/Markups.html#MISMATCH"><B>MISMATCH</B></A> - 
Static variable in class jaligner.<A HREF="../jaligner/Markups.html" title="class in jaligner">Markups</A>
<DD>Markup line mismatch character
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#missprimingLibraryFile"><B>missprimingLibraryFile</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/RestrictionEnzyme.html#MmeI"><B>MmeI</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/RestrictionEnzyme.html" title="class in broad.core.primer3">RestrictionEnzyme</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#moveAnnotation(nextgen.core.annotation.Annotation, int, nextgen.core.annotation.Annotation)"><B>moveAnnotation(Annotation, int, Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>Moves the annotation to a new start location using the coordinate space to decide on gaps
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#moveToCoordinate(int)"><B>moveToCoordinate(int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html#moveToCoordinate(int)"><B>moveToCoordinate(int)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/AbstractPairedEndAlignment.html" title="class in nextgen.core.alignment">AbstractPairedEndAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#moveToCoordinate(int)"><B>moveToCoordinate(int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Move an annotation, preserving the relationships between its blocks, to a new coordinate.
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#moveToCoordinate(int)"><B>moveToCoordinate(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#moveToCoordinate(int)"><B>moveToCoordinate(int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>Move an annotation, preserving the relationships between its blocks, to a new coordinate.
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#moveToCoordinate(int)"><B>moveToCoordinate(int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment"><B>MultipleAlignment</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#MultipleAlignment()"><B>MultipleAlignment()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#MultipleAlignment(java.util.List, java.lang.String, java.lang.String[])"><B>MultipleAlignment(List&lt;? extends MultipleAlignment&gt;, String, String[])</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>Constructs a multiple alignment object out of piecemeal multiple alignment blocks with, possible,
 different number of aligned sequences.
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment"><B>MultipleAlignment.AlignedSequence</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#MultipleAlignment.AlignedSequence(broad.core.sequence.Sequence)"><B>MultipleAlignment.AlignedSequence(Sequence)</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#MultipleAlignment.AlignedSequence(java.lang.String)"><B>MultipleAlignment.AlignedSequence(String)</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html" title="class in broad.core.multiplealignment"><B>MultipleAlignmentFactory</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html#MultipleAlignmentFactory()"><B>MultipleAlignmentFactory()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html" title="class in broad.core.multiplealignment">MultipleAlignmentFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentIO.html" title="interface in broad.core.multiplealignment"><B>MultipleAlignmentIO</B></A> - Interface in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentIOFactory.html" title="class in broad.core.multiplealignment"><B>MultipleAlignmentIOFactory</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentIOFactory.html#MultipleAlignmentIOFactory()"><B>MultipleAlignmentIOFactory()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignmentIOFactory.html" title="class in broad.core.multiplealignment">MultipleAlignmentIOFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#multiplyColumnsWithConstants(double[])"><B>multiplyColumnsWithConstants(double[])</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection"><B>MultiSampleScanPeakCaller</B></A> - Class in <A HREF="../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#MultiSampleScanPeakCaller(java.lang.String, java.lang.String, java.lang.String, int, int)"><B>MultiSampleScanPeakCaller(String, String, String, int, int)</B></A> - 
Constructor for class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Instantiate using file of sample information
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#MultiSampleScanPeakCaller(java.lang.String, java.lang.String, java.lang.String, java.lang.String)"><B>MultiSampleScanPeakCaller(String, String, String, String)</B></A> - 
Constructor for class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Instantiate with a single sample and an expression sample and default paramters
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#MultiSampleScanPeakCaller(java.lang.String, java.lang.String, java.lang.String, java.lang.String, int, int)"><B>MultiSampleScanPeakCaller(String, String, String, String, int, int)</B></A> - 
Constructor for class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>Instantiate with a single signal sample and an expression sample
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection"><B>MultiSampleScanSkellamAnalysis</B></A> - Class in <A HREF="../broad/pda/seq/protection/package-summary.html">broad.pda.seq.protection</A><DD>&nbsp;<DT><A HREF="../broad/pda/countreads/LibraryCompositionByRnaClass.html#MYSTERY_READS_CLASS_NAME"><B>MYSTERY_READS_CLASS_NAME</B></A> - 
Static variable in class broad.pda.countreads.<A HREF="../broad/pda/countreads/LibraryCompositionByRnaClass.html" title="class in broad.pda.countreads">LibraryCompositionByRnaClass</A>
<DD>The name used for unmapped reads when counting read mappings
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-12.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-14.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-13.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-13.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
