<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:30 EDT 2013 -->
<TITLE>
F-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="F-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-5.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-7.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-6.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-6.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_F_"><!-- --></A><H2>
<B>F</B></H2>
<DL>
<DT><A HREF="../broad/core/math/ANOVA.html#F()"><B>F()</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/ANOVA.html" title="class in broad.core.math">ANOVA</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/MathUtil.html#factorial(int)"><B>factorial(int)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/MathUtil.html" title="class in broad.core.math">MathUtil</A>
<DD>Factorial function
<DT><A HREF="../jaligner/formats/FASTA.html" title="class in jaligner.formats"><B>FASTA</B></A> - Class in <A HREF="../jaligner/formats/package-summary.html">jaligner.formats</A><DD><a href="http://www.ncbi.nlm.nih.gov/BLAST/fasta.html">FASTA</a> format.<DT><A HREF="../jaligner/formats/FASTA.html#FASTA()"><B>FASTA()</B></A> - 
Constructor for class jaligner.formats.<A HREF="../jaligner/formats/FASTA.html" title="class in jaligner.formats">FASTA</A>
<DD>Constructor for FASTA.
<DT><A HREF="../broad/core/sequence/FastaHandler.html" title="interface in broad.core.sequence"><B>FastaHandler</B></A> - Interface in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/FastaMultipleAlignmentIO.html" title="class in broad.core.multiplealignment"><B>FastaMultipleAlignmentIO</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/FastaMultipleAlignmentIO.html#FastaMultipleAlignmentIO()"><B>FastaMultipleAlignmentIO()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/FastaMultipleAlignmentIO.html" title="class in broad.core.multiplealignment">FastaMultipleAlignmentIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaParser.html" title="class in broad.core.sequence"><B>FastaParser</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/FastaParser.html#FastaParser()"><B>FastaParser()</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/FastaParser.html" title="class in broad.core.sequence">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityBinaryIO.html" title="class in broad.core.sequence"><B>FastaQualityBinaryIO</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/FastaQualityBinaryIO.html#FastaQualityBinaryIO()"><B>FastaQualityBinaryIO()</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityBinaryIO.html" title="class in broad.core.sequence">FastaQualityBinaryIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityHandler.html" title="interface in broad.core.sequence"><B>FastaQualityHandler</B></A> - Interface in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/FastaQualityIndex.html" title="class in broad.core.sequence"><B>FastaQualityIndex</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/FastaQualityIndex.html#FastaQualityIndex()"><B>FastaQualityIndex()</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityIndex.html" title="class in broad.core.sequence">FastaQualityIndex</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityIndex.SequenceFileLayout.html" title="class in broad.core.sequence"><B>FastaQualityIndex.SequenceFileLayout</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/FastaQualityIndex.SequenceFileLayout.html#FastaQualityIndex.SequenceFileLayout(java.lang.String)"><B>FastaQualityIndex.SequenceFileLayout(String)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityIndex.SequenceFileLayout.html" title="class in broad.core.sequence">FastaQualityIndex.SequenceFileLayout</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityParser.html" title="class in broad.core.sequence"><B>FastaQualityParser</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>&nbsp;<DT><A HREF="../broad/core/sequence/FastaQualityParser.html#FastaQualityParser()"><B>FastaQualityParser()</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityParser.html" title="class in broad.core.sequence">FastaQualityParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#fastaSeqId"><B>fastaSeqId</B></A> - 
Variable in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing">CRISPRDesigner.CRISPRTarget</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence"><B>FastaSequenceIO</B></A> - Class in <A HREF="../broad/core/sequence/package-summary.html">broad.core.sequence</A><DD>Suppose to be a very generic and smart Fasta manipulation
 Class.<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#FastaSequenceIO(java.lang.String)"><B>FastaSequenceIO(String)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#FastaSequenceIO(java.io.File)"><B>FastaSequenceIO(File)</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaSequenceIO.html#FastaSequenceIO()"><B>FastaSequenceIO()</B></A> - 
Constructor for class broad.core.sequence.<A HREF="../broad/core/sequence/FastaSequenceIO.html" title="class in broad.core.sequence">FastaSequenceIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads"><B>FastqLibraryStats</B></A> - Class in <A HREF="../broad/pda/countreads/package-summary.html">broad.pda.countreads</A><DD>&nbsp;<DT><A HREF="../broad/pda/countreads/FastqLibraryStats.html#FastqLibraryStats(java.lang.String)"><B>FastqLibraryStats(String)</B></A> - 
Constructor for class broad.pda.countreads.<A HREF="../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads">FastqLibraryStats</A>
<DD>Constructor for single end reads
<DT><A HREF="../broad/pda/countreads/FastqLibraryStats.html#FastqLibraryStats(java.lang.String, java.lang.String)"><B>FastqLibraryStats(String, String)</B></A> - 
Constructor for class broad.pda.countreads.<A HREF="../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads">FastqLibraryStats</A>
<DD>Constructor for paired end reads
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq"><B>FastqParser</B></A> - Class in <A HREF="../broad/pda/seq/fastq/package-summary.html">broad.pda.seq.fastq</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#FastqParser(java.io.File)"><B>FastqParser(File)</B></A> - 
Constructor for class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD><B>Deprecated.</B>&nbsp;<I>This constructure is highly discouraged as it opens a reader. Use the empty constructor instead and set the 
 file to read, then call start.</I>
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#FastqParser()"><B>FastqParser()</B></A> - 
Constructor for class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq"><B>FastqSequence</B></A> - Class in <A HREF="../broad/pda/seq/fastq/package-summary.html">broad.pda.seq.fastq</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/fastq/FastqSequence.html#FastqSequence(java.lang.String, java.lang.String, java.lang.String, java.lang.String)"><B>FastqSequence(String, String, String, String)</B></A> - 
Constructor for class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSequence.html" title="class in broad.pda.seq.fastq">FastqSequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.html" title="class in broad.pda.seq.fastq"><B>FastqSplitterByBarcode</B></A> - Class in <A HREF="../broad/pda/seq/fastq/package-summary.html">broad.pda.seq.fastq</A><DD>Reads through a fastq file and writes records to different files depending on 
 the barcode found within the start or end of the read.<DT><A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.html#FastqSplitterByBarcode()"><B>FastqSplitterByBarcode()</B></A> - 
Constructor for class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.html" title="class in broad.pda.seq.fastq">FastqSplitterByBarcode</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html" title="class in broad.pda.seq.fastq"><B>FastqSplitterByBarcode.BarcodeInformation</B></A> - Class in <A HREF="../broad/pda/seq/fastq/package-summary.html">broad.pda.seq.fastq</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html#FastqSplitterByBarcode.BarcodeInformation(int)"><B>FastqSplitterByBarcode.BarcodeInformation(int)</B></A> - 
Constructor for class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqSplitterByBarcode.BarcodeInformation.html" title="class in broad.pda.seq.fastq">FastqSplitterByBarcode.BarcodeInformation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/ComputeFDR.html#FDR(broad.core.math.EmpiricalDistribution, broad.core.math.EmpiricalDistribution[], double)"><B>FDR(EmpiricalDistribution, EmpiricalDistribution[], double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/ComputeFDR.html" title="class in broad.core.math">ComputeFDR</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/ComputeFDR.html#FDR(java.util.Map, broad.core.math.EmpiricalDistribution[], double)"><B>FDR(Map&lt;String, Double&gt;, EmpiricalDistribution[], double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/ComputeFDR.html" title="class in broad.core.math">ComputeFDR</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#FDR(broad.core.math.EmpiricalDistribution, broad.core.math.EmpiricalDistribution[], double)"><B>FDR(EmpiricalDistribution, EmpiricalDistribution[], double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#FDRCorrect(double[], double)"><B>FDRCorrect(double[], double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#FDRCorrect(java.util.List, double)"><B>FDRCorrect(List&lt;Double&gt;, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/FDRDistribution.html" title="class in broad.core.math"><B>FDRDistribution</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/FDRDistribution.html#FDRDistribution(double[], Jama.Matrix, double)"><B>FDRDistribution(double[], Matrix, double)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/FDRDistribution.html" title="class in broad.core.math">FDRDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/FDRDistribution.html#FDRDistribution(double[], broad.core.math.FDRDistribution)"><B>FDRDistribution(double[], FDRDistribution)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/FDRDistribution.html" title="class in broad.core.math">FDRDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/FDRDistribution.html#FDRDistribution(double[], Jama.Matrix, double, boolean)"><B>FDRDistribution(double[], Matrix, double, boolean)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/FDRDistribution.html" title="class in broad.core.math">FDRDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/ComputeFDR.html#FDRExact(java.util.Map, java.util.Map[], double)"><B>FDRExact(Map, Map[], double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/ComputeFDR.html" title="class in broad.core.math">ComputeFDR</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/ComputeFDR.html#FDRExactPrint(java.util.Map, java.util.Map[], double)"><B>FDRExactPrint(Map, Map[], double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/ComputeFDR.html" title="class in broad.core.math">ComputeFDR</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/Feature.html" title="interface in broad.core.annotation"><B>Feature</B></A> - Interface in <A HREF="../broad/core/annotation/package-summary.html">broad.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/feature/FeatureScore.html" title="interface in nextgen.core.feature"><B>FeatureScore</B></A>&lt;<A HREF="../nextgen/core/feature/FeatureScore.html" title="type parameter in FeatureScore">T</A>&gt; - Interface in <A HREF="../nextgen/core/feature/package-summary.html">nextgen.core.feature</A><DD>Represents a generic scoring object<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment"><B>FeatureSequenceAlignment</B></A> - Class in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>&nbsp;<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.lang.String, java.lang.String)"><B>FeatureSequenceAlignment(String, String)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.util.Collection, java.lang.String)"><B>FeatureSequenceAlignment(Collection&lt;Gene&gt;, String)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.lang.String, java.lang.String, float, float, float, float)"><B>FeatureSequenceAlignment(String, String, float, float, float, float)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.util.Collection, java.util.Map, float, float, float, float)"><B>FeatureSequenceAlignment(Collection&lt;Gene&gt;, Map&lt;String, Sequence&gt;, float, float, float, float)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html#FeatureSequenceAlignment(java.util.Collection, java.lang.String, float, float, float, float)"><B>FeatureSequenceAlignment(Collection&lt;Gene&gt;, String, float, float, float, float)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment"><B>FeatureSequenceAlignment.UnorderedGenePair</B></A> - Class in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>An unordered pair of genes<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html#FeatureSequenceAlignment.UnorderedGenePair(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene)"><B>FeatureSequenceAlignment.UnorderedGenePair(Gene, Gene)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>
<DD>Determine which gene is first and which is second
<DT><A HREF="../nextgen/core/utils/FileUtil.html#fileLinesAsList(java.lang.String)"><B>fileLinesAsList(String)</B></A> - 
Static method in class nextgen.core.utils.<A HREF="../nextgen/core/utils/FileUtil.html" title="class in nextgen.core.utils">FileUtil</A>
<DD>Get the lines of a file as a list
<DT><A HREF="../nextgen/core/utils/FileUtil.html" title="class in nextgen.core.utils"><B>FileUtil</B></A> - Class in <A HREF="../nextgen/core/utils/package-summary.html">nextgen.core.utils</A><DD>&nbsp;<DT><A HREF="../nextgen/core/utils/FileUtil.html#FileUtil()"><B>FileUtil()</B></A> - 
Constructor for class nextgen.core.utils.<A HREF="../nextgen/core/utils/FileUtil.html" title="class in nextgen.core.utils">FileUtil</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/MethylationUtils.html#fillArray(java.util.List, java.util.List, int)"><B>fillArray(List&lt;String&gt;, List&lt;String&gt;, int)</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/MethylationUtils.html" title="class in broad.pda.methylation">MethylationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#filter(java.util.Collection, java.lang.String)"><B>filter(Collection&lt;Annotation&gt;, String)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html#FILTER"><B>FILTER</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation">FilterSamByBaseComposition</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#filterByOverlap(java.util.List)"><B>filterByOverlap(List&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Filters this reader by removing all elements that do not overlap the given list of annotations
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#filterByOverlap(java.util.List, int)"><B>filterByOverlap(List&lt;? extends Annotation&gt;, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#filterByOverlap(java.util.Collection)"><B>filterByOverlap(Collection&lt;Gene&gt;)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#filterByOverlapAndOrientation(java.util.List)"><B>filterByOverlapAndOrientation(List&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>ADDED BY MORAN Feb 8th 2010
 Filters this reader by removing all elements that do not overlap AND HAVE THE SAME ORIENTATION as the given list of annotations
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#filterByOverlapOfLastExon(java.util.List, int)"><B>filterByOverlapOfLastExon(List&lt;? extends GenomicAnnotation&gt;, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#filterByOverlapOfLastExonOrAfterLastExon(java.util.List, int)"><B>filterByOverlapOfLastExonOrAfterLastExon(List&lt;? extends GenomicAnnotation&gt;, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#filterByScore(double)"><B>filterByScore(double)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html#FILTERED_READS"><B>FILTERED_READS</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html" title="class in broad.pda.methylation">FilterFastqForMethylationAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html#FILTERED_READS"><B>FILTERED_READS</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation">FilterSamByBaseComposition</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html#FILTERED_READS"><B>FILTERED_READS</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html" title="class in broad.pda.methylation">FilterSamForMethylationAlignment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers"><B>FilteredBamWriter</B></A> - Class in <A HREF="../nextgen/core/writers/package-summary.html">nextgen.core.writers</A><DD>&nbsp;<DT><A HREF="../nextgen/core/writers/FilteredBamWriter.html#FilteredBamWriter(java.lang.String)"><B>FilteredBamWriter(String)</B></A> - 
Constructor for class nextgen.core.writers.<A HREF="../nextgen/core/writers/FilteredBamWriter.html" title="class in nextgen.core.writers">FilteredBamWriter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html" title="class in broad.pda.methylation"><B>FilterFastqForMethylationAlignment</B></A> - Class in <A HREF="../broad/pda/methylation/package-summary.html">broad.pda.methylation</A><DD>&nbsp;<DT><A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html#FilterFastqForMethylationAlignment()"><B>FilterFastqForMethylationAlignment()</B></A> - 
Constructor for class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html" title="class in broad.pda.methylation">FilterFastqForMethylationAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/FilterGCT.html" title="class in broad.pda.geneexpression"><B>FilterGCT</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/FilterGCT.html#FilterGCT()"><B>FilterGCT()</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/FilterGCT.html" title="class in broad.pda.geneexpression">FilterGCT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/FilterGCTFile.html" title="class in broad.pda.geneexpression"><B>FilterGCTFile</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/FilterGCTFile.html#FilterGCTFile(java.lang.String, java.lang.String, double)"><B>FilterGCTFile(String, String, double)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/FilterGCTFile.html" title="class in broad.pda.geneexpression">FilterGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#filterGenesByExpression(java.util.Map)"><B>filterGenesByExpression(Map&lt;Gene, List&lt;Double&gt;&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>This function will filter the genes in the input Map by a number of constraints on the total number of reads on the genes
 1.
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#filterInvariantGenes(broad.core.datastructures.MatrixWithHeaders, double)"><B>filterInvariantGenes(MatrixWithHeaders, double)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/FilterMatrixByMinCutoff.html" title="class in broad.pda.geneexpression.agilent"><B>FilterMatrixByMinCutoff</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/FilterMatrixByMinCutoff.html#FilterMatrixByMinCutoff(broad.core.datastructures.MatrixWithHeaders, java.lang.String, java.lang.String, double)"><B>FilterMatrixByMinCutoff(MatrixWithHeaders, String, String, double)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/FilterMatrixByMinCutoff.html" title="class in broad.pda.geneexpression.agilent">FilterMatrixByMinCutoff</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#filterPaths(java.util.Map, double)"><B>filterPaths(Map&lt;Path, double[]&gt;, double)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/FilterProbesByBlatResults.html" title="class in broad.pda.capture.designer"><B>FilterProbesByBlatResults</B></A> - Class in <A HREF="../broad/pda/capture/designer/package-summary.html">broad.pda.capture.designer</A><DD>&nbsp;<DT><A HREF="../broad/pda/capture/designer/FilterProbesByBlatResults.html#FilterProbesByBlatResults()"><B>FilterProbesByBlatResults()</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/FilterProbesByBlatResults.html" title="class in broad.pda.capture.designer">FilterProbesByBlatResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation"><B>FilterSamByBaseComposition</B></A> - Class in <A HREF="../broad/pda/methylation/package-summary.html">broad.pda.methylation</A><DD>&nbsp;<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html#FilterSamByBaseComposition()"><B>FilterSamByBaseComposition()</B></A> - 
Constructor for class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation">FilterSamByBaseComposition</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html" title="class in broad.pda.methylation"><B>FilterSamForMethylationAlignment</B></A> - Class in <A HREF="../broad/pda/methylation/package-summary.html">broad.pda.methylation</A><DD>&nbsp;<DT><A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html#FilterSamForMethylationAlignment()"><B>FilterSamForMethylationAlignment()</B></A> - 
Constructor for class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html" title="class in broad.pda.methylation">FilterSamForMethylationAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#filterValuesLargerThanUsingColumn(int, double)"><B>filterValuesLargerThanUsingColumn(int, double)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#filterWindows(java.util.Collection)"><B>filterWindows(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/MultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">MultiSampleScanPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html#filterWindows(java.util.Collection)"><B>filterWindows(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/TwoSampleScanSkellamPeakCaller.html" title="class in broad.pda.seq.protection">TwoSampleScanSkellamPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RAPPeakCaller.html#filterWindows(java.util.Collection)"><B>filterWindows(Collection&lt;Annotation&gt;)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RAPPeakCaller.html" title="class in broad.pda.seq.rap">RAPPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/analysis/PeakCaller.html#filterWindows(java.util.Collection)"><B>filterWindows(Collection&lt;Annotation&gt;)</B></A> - 
Method in interface nextgen.core.analysis.<A HREF="../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A>
<DD>Filter windows that do not contain peaks
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#find(java.lang.String, int, int)"><B>find(String, int, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#find(broad.core.datastructures.Interval)"><B>find(Interval)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find an interval.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#find(int, int)"><B>find(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find an interval.
<DT><A HREF="../nextgen/core/annotation/Gene.html#findAllORFs(boolean)"><B>findAllORFs(boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Get all ORFs as gene objects
<DT><A HREF="../nextgen/core/annotation/Gene.html#findAllORFs(java.lang.String)"><B>findAllORFs(String)</B></A> - 
Static method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#findAllORFs(broad.core.sequence.Sequence, boolean)"><B>findAllORFs(Sequence, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#findAllStartCodons()"><B>findAllStartCodons()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Find start codons
<DT><A HREF="../broad/pda/enrichment/EnrichmentUtils.html#findAnnotationFiles(java.io.File, java.lang.String[])"><B>findAnnotationFiles(File, String[])</B></A> - 
Static method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentUtils.html" title="class in broad.pda.enrichment">EnrichmentUtils</A>
<DD>Recursively visits directories to find genomic annotation files.
<DT><A HREF="../broad/pda/enrichment/EnrichmentUtils.html#findAnnotationFiles(java.io.File, java.lang.String[], java.lang.String)"><B>findAnnotationFiles(File, String[], String)</B></A> - 
Static method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentUtils.html" title="class in broad.pda.enrichment">EnrichmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/enrichment/EnrichmentUtils.html#findAnnotationFiles(java.io.File, java.lang.String[], java.lang.String, java.util.SortedMap)"><B>findAnnotationFiles(File, String[], String, SortedMap&lt;String, File&gt;)</B></A> - 
Static method in class broad.pda.enrichment.<A HREF="../broad/pda/enrichment/EnrichmentUtils.html" title="class in broad.pda.enrichment">EnrichmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#findByIndex(int)"><B>findByIndex(int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Find the nth interval in the tree.
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#findClosest(broad.core.annotation.LightweightGenomicAnnotation)"><B>findClosest(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#findClosest(broad.core.annotation.LightweightGenomicAnnotation, boolean)"><B>findClosest(LightweightGenomicAnnotation, boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Added "nonOverlapping" option June 13, 2012
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#findClosestForAll(broad.core.annotation.AnnotationReader, boolean)"><B>findClosestForAll(AnnotationReader&lt;? extends GenomicAnnotation&gt;, boolean)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Jesse Engreitz
 June 24, 2012
 Return a map of elements in this reader to the closest elements in another reader
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#findClosestForAll(broad.core.annotation.AnnotationReader)"><B>findClosestForAll(AnnotationReader&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#findClosestNonOverlapping(broad.core.annotation.LightweightGenomicAnnotation)"><B>findClosestNonOverlapping(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#findClosestNonoverlappingForAll(broad.core.annotation.AnnotationReader)"><B>findClosestNonoverlappingForAll(AnnotationReader&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#findCompatibleGenes(broad.core.datastructures.IntervalTree, int[])"><B>findCompatibleGenes(IntervalTree&lt;GeneWithIsoforms&gt;, int[])</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.html#findCompleteTranscripts(java.lang.String)"><B>findCompleteTranscripts(String)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/AddEndRNASeqToScripture.html" title="class in nextgen.core.scripture">AddEndRNASeqToScripture</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/MethylationUtils.html#findCpGs(java.lang.String)"><B>findCpGs(String)</B></A> - 
Static method in class broad.pda.methylation.<A HREF="../broad/pda/methylation/MethylationUtils.html" title="class in broad.pda.methylation">MethylationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/chip/FindDiffrentialEnrichedRegions.html" title="class in broad.pda.seq.chip"><B>FindDiffrentialEnrichedRegions</B></A> - Class in <A HREF="../broad/pda/seq/chip/package-summary.html">broad.pda.seq.chip</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/chip/FindDiffrentialEnrichedRegions.html#FindDiffrentialEnrichedRegions(java.util.Collection, broad.pda.seq.segmentation.AlignmentDataModel, broad.pda.seq.segmentation.AlignmentDataModel, java.lang.String)"><B>FindDiffrentialEnrichedRegions(Collection&lt;Annotation&gt;, AlignmentDataModel, AlignmentDataModel, String)</B></A> - 
Constructor for class broad.pda.seq.chip.<A HREF="../broad/pda/seq/chip/FindDiffrentialEnrichedRegions.html" title="class in broad.pda.seq.chip">FindDiffrentialEnrichedRegions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#findEdge(int, int)"><B>findEdge(int, int)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns edge between start and end
<DT><A HREF="../broad/pda/rap/misc/FindEnrichedDepletedRegions.html" title="class in broad.pda.rap.misc"><B>FindEnrichedDepletedRegions</B></A> - Class in <A HREF="../broad/pda/rap/misc/package-summary.html">broad.pda.rap.misc</A><DD>&nbsp;<DT><A HREF="../broad/pda/rap/misc/FindEnrichedDepletedRegions.html#FindEnrichedDepletedRegions(nextgen.core.model.ScanStatisticDataAlignmentModel, int, java.lang.String, java.lang.String)"><B>FindEnrichedDepletedRegions(ScanStatisticDataAlignmentModel, int, String, String)</B></A> - 
Constructor for class broad.pda.rap.misc.<A HREF="../broad/pda/rap/misc/FindEnrichedDepletedRegions.html" title="class in broad.pda.rap.misc">FindEnrichedDepletedRegions</A>
<DD>Try and find large regions of enrichment and depletion that are unexpected given a  randomized model on a given chromosome
<DT><A HREF="../nextgen/core/annotation/Gene.html#findLongestORF()"><B>findLongestORF()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#findLongestORF(java.lang.String)"><B>findLongestORF(String)</B></A> - 
Static method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3IO.html#findPrimerPair(broad.core.primer3.Primer3SequenceInputTags, broad.core.primer3.Primer3Configuration)"><B>findPrimerPair(Primer3SequenceInputTags, Primer3Configuration)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3IO.html" title="class in broad.core.primer3">Primer3IO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/FindProbesForGeneSet.html" title="class in broad.pda.geneexpression.agilent"><B>FindProbesForGeneSet</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/FindProbesForGeneSet.html#FindProbesForGeneSet(java.util.Map, java.util.Map, java.lang.String)"><B>FindProbesForGeneSet(Map&lt;String, String&gt;, Map&lt;String, String&gt;, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/FindProbesForGeneSet.html" title="class in broad.pda.geneexpression.agilent">FindProbesForGeneSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#findReachableTargetVertices(nextgen.core.annotation.Annotation, java.util.Collection, java.util.Collection, java.util.Collection, int, org.jgrapht.DirectedGraph)"><B>findReachableTargetVertices(Annotation, Collection&lt;Annotation&gt;, Collection&lt;Annotation&gt;, Collection&lt;Annotation&gt;, int, DirectedGraph&lt;Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html#findUngappedChunks()"><B>findUngappedChunks()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.AlignedSequence.html" title="class in broad.core.multiplealignment">MultipleAlignment.AlignedSequence</A>
<DD>Finds the list of ungapped regions in the sequence.
<DT><A HREF="../broad/core/sequence/Sequence.html#findUngappedSequenceChunks()"><B>findUngappedSequenceChunks()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>Finds the list of ungapped regions in the sequence.
<DT><A HREF="../nextgen/core/model/score/WindowProcessor.AbstractProcessor.html#finishedRegion()"><B>finishedRegion()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowProcessor.AbstractProcessor.html" title="class in nextgen.core.model.score">WindowProcessor.AbstractProcessor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowProcessor.html#finishedRegion()"><B>finishedRegion()</B></A> - 
Method in interface nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowProcessor.html" title="interface in nextgen.core.model.score">WindowProcessor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/FirstOfPairFilter.html" title="class in nextgen.core.readFilters"><B>FirstOfPairFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/FirstOfPairFilter.html#FirstOfPairFilter()"><B>FirstOfPairFilter()</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/FirstOfPairFilter.html" title="class in nextgen.core.readFilters">FirstOfPairFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/protection/SampleData.html#firstReadTranscriptionStrand()"><B>firstReadTranscriptionStrand()</B></A> - 
Method in class broad.pda.seq.protection.<A HREF="../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>
<DD>Whether read 1 is in direction of transcription
<DT><A HREF="../broad/core/math/FisherCombinedProbabilityTest.html" title="class in broad.core.math"><B>FisherCombinedProbabilityTest</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/FisherCombinedProbabilityTest.html#FisherCombinedProbabilityTest(double[])"><B>FisherCombinedProbabilityTest(double[])</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/FisherCombinedProbabilityTest.html" title="class in broad.core.math">FisherCombinedProbabilityTest</A>
<DD>Instantiate with array of P values
<DT><A HREF="../broad/core/math/FisherCombinedProbabilityTest.html#FisherCombinedProbabilityTest(java.util.Collection)"><B>FisherCombinedProbabilityTest(Collection&lt;Double&gt;)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/FisherCombinedProbabilityTest.html" title="class in broad.core.math">FisherCombinedProbabilityTest</A>
<DD>Instantiate with collection of P values
<DT><A HREF="../broad/core/siphy/Fit.html" title="interface in broad.core.siphy"><B>Fit</B></A> - Interface in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>Simple interface that represents a model fit<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#fitOmega(java.util.Map, org.forester.phylogeny.Phylogeny, int)"><B>fitOmega(Map&lt;String, Matrix&gt;, Phylogeny, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>Implementation of Felsenstein's prunning algorithm to compute the 
 likelihood given this model.
<DT><A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html#fitOmega(nextgen.core.annotation.Annotation, java.io.File, java.lang.String, java.lang.String)"><B>fitOmega(Annotation, File, String, String)</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html" title="class in broad.core.siphy.tools.conservation">EstimateOmegaPerExon</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html#fitOmega(broad.pda.datastructures.Alignments, broad.core.siphy.TreeScaler)"><B>fitOmega(Alignments, TreeScaler)</B></A> - 
Static method in class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/EstimateOmegaPerExon.html" title="class in broad.core.siphy.tools.conservation">EstimateOmegaPerExon</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#fitPI(java.util.Map, org.forester.phylogeny.Phylogeny)"><B>fitPI(Map&lt;String, Matrix&gt;, Phylogeny)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>Implementation of the EM algorithm to fit the base distribution rho at a site
 assuming omega is constant.
<DT><A HREF="../broad/core/siphy/TreeScaler.html#fitPI(org.forester.phylogeny.Phylogeny, int)"><B>fitPI(Phylogeny, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScaler.html" title="class in broad.core.siphy">TreeScaler</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/FixedControlDifferentialExpression.html" title="class in broad.pda.differentialExpression"><B>FixedControlDifferentialExpression</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>For a set of several experiments and a fixed set of control experiments this class wrapps
 differential expression tools to perform analysis of different subsets of experiments against
 the controls.<DT><A HREF="../broad/pda/differentialExpression/FixedControlDifferentialExpression.html#FixedControlDifferentialExpression()"><B>FixedControlDifferentialExpression()</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/FixedControlDifferentialExpression.html" title="class in broad.pda.differentialExpression">FixedControlDifferentialExpression</A>
<DD>&nbsp;
<DT><A HREF="../Jama/examples/MagicSquareExample.html#fixedWidthDoubletoString(double, int, int)"><B>fixedWidthDoubletoString(double, int, int)</B></A> - 
Static method in class Jama.examples.<A HREF="../Jama/examples/MagicSquareExample.html" title="class in Jama.examples">MagicSquareExample</A>
<DD>Format double with Fw.d.
<DT><A HREF="../Jama/examples/MagicSquareExample.html#fixedWidthIntegertoString(int, int)"><B>fixedWidthIntegertoString(int, int)</B></A> - 
Static method in class Jama.examples.<A HREF="../Jama/examples/MagicSquareExample.html" title="class in Jama.examples">MagicSquareExample</A>
<DD>Format integer with Iw.
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#flatWeight(double)"><B>flatWeight(double)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#flatWeightMatrix(double)"><B>flatWeightMatrix(double)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#floor(double)"><B>floor(double)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Resets all values below floor to floor.
<DT><A HREF="../broad/core/primer3/RestrictionEnzyme.html#FokI"><B>FokI</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/RestrictionEnzyme.html" title="class in broad.core.primer3">RestrictionEnzyme</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#fold(java.util.Collection, java.util.Collection, boolean)"><B>fold(Collection&lt;Double&gt;, Collection&lt;Double&gt;, boolean)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#fold(double[], double[], boolean)"><B>fold(double[], double[], boolean)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#fold(double[], boolean)"><B>fold(double[], boolean)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/FoldSignificance.html" title="class in broad.pda.differentialExpression"><B>FoldSignificance</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/FoldSignificance.html#FoldSignificance(broad.core.datastructures.MatrixWithHeaders, broad.core.datastructures.MatrixWithHeaders, java.util.Map, java.lang.String, double)"><B>FoldSignificance(MatrixWithHeaders, MatrixWithHeaders, Map&lt;String, Collection&lt;String&gt;&gt;, String, double)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/FoldSignificance.html" title="class in broad.pda.differentialExpression">FoldSignificance</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/formats/CLUSTAL.html#format(java.lang.String[], java.lang.String[])"><B>format(String[], String[])</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/CLUSTAL.html" title="class in jaligner.formats">CLUSTAL</A>
<DD>Returns CLUSTAL format
<DT><A HREF="../jaligner/formats/CLUSTAL.html#format(jaligner.Alignment)"><B>format(Alignment)</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/CLUSTAL.html" title="class in jaligner.formats">CLUSTAL</A>
<DD>Returns CLUSTAL format of the alignment
<DT><A HREF="../jaligner/formats/FASTA.html#format(jaligner.Sequence)"><B>format(Sequence)</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/FASTA.html" title="class in jaligner.formats">FASTA</A>
<DD>Returns the name, description and sequence combined in one string.
<DT><A HREF="../jaligner/formats/FASTA.html#format(jaligner.Alignment)"><B>format(Alignment)</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/FASTA.html" title="class in jaligner.formats">FASTA</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/formats/Format.html" title="class in jaligner.formats"><B>Format</B></A> - Class in <A HREF="../jaligner/formats/package-summary.html">jaligner.formats</A><DD>Abstract format<DT><A HREF="../jaligner/formats/Format.html#Format()"><B>Format()</B></A> - 
Constructor for class jaligner.formats.<A HREF="../jaligner/formats/Format.html" title="class in jaligner.formats">Format</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/formats/Format.html#format(jaligner.Alignment)"><B>format(Alignment)</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/Format.html" title="class in jaligner.formats">Format</A>
<DD>Formats alignment
<DT><A HREF="../jaligner/formats/Pair.html#format(jaligner.Alignment)"><B>format(Alignment)</B></A> - 
Method in class jaligner.formats.<A HREF="../jaligner/formats/Pair.html" title="class in jaligner.formats">Pair</A>
<DD>Formats an alignment object to the Pair_FORMAT format
<DT><A HREF="../jaligner/formats/FormatFactory.html" title="class in jaligner.formats"><B>FormatFactory</B></A> - Class in <A HREF="../jaligner/formats/package-summary.html">jaligner.formats</A><DD>Formats factory.<DT><A HREF="../broad/core/primer3/Primer3PrimerPickingMode.html#FORWARD_ONLY"><B>FORWARD_ONLY</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PrimerPickingMode.html" title="class in broad.core.primer3">Primer3PrimerPickingMode</A>
<DD>Only picks forward primers
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#Fp(int, double)"><B>Fp(int, double)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#FpWithBreaking(int, double, double, double)"><B>FpWithBreaking(int, double, double, double)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FragmentAlignment.html" title="class in nextgen.core.alignment"><B>FragmentAlignment</B></A> - Class in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>&nbsp;<DT><A HREF="../nextgen/core/alignment/FragmentAlignment.html#FragmentAlignment(nextgen.core.alignment.SingleEndAlignment, nextgen.core.alignment.SingleEndAlignment)"><B>FragmentAlignment(SingleEndAlignment, SingleEndAlignment)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FragmentAlignment.html" title="class in nextgen.core.alignment">FragmentAlignment</A>
<DD>Constructs a paired end alignment object from two alignments.
<DT><A HREF="../nextgen/core/alignment/FragmentAlignment.html#FragmentAlignment(nextgen.core.alignment.SingleEndAlignment, nextgen.core.alignment.SingleEndAlignment, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>FragmentAlignment(SingleEndAlignment, SingleEndAlignment, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FragmentAlignment.html" title="class in nextgen.core.alignment">FragmentAlignment</A>
<DD>Constructs a paired end alignment object from two alignments and provides the read that is in the direction of transcription
<DT><A HREF="../nextgen/core/readFilters/FragmentLengthFilter.html" title="class in nextgen.core.readFilters"><B>FragmentLengthFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/FragmentLengthFilter.html#FragmentLengthFilter(nextgen.core.coordinatesystem.CoordinateSpace, int)"><B>FragmentLengthFilter(CoordinateSpace, int)</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/FragmentLengthFilter.html" title="class in nextgen.core.readFilters">FragmentLengthFilter</A>
<DD>Instantiate the filter with a maximum length and no minimum length
<DT><A HREF="../nextgen/core/readFilters/FragmentLengthFilter.html#FragmentLengthFilter(nextgen.core.coordinatesystem.CoordinateSpace, int, int)"><B>FragmentLengthFilter(CoordinateSpace, int, int)</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/FragmentLengthFilter.html" title="class in nextgen.core.readFilters">FragmentLengthFilter</A>
<DD>Instantiate the filter with minimum and maximum lengths
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#fragmentSizeOptions"><B>fragmentSizeOptions</B></A> - 
Variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/filter/FullyContainedFilter.html" title="class in nextgen.core.annotation.filter"><B>FullyContainedFilter</B></A> - Class in <A HREF="../nextgen/core/annotation/filter/package-summary.html">nextgen.core.annotation.filter</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/filter/FullyContainedFilter.html#FullyContainedFilter(nextgen.core.annotation.Annotation)"><B>FullyContainedFilter(Annotation)</B></A> - 
Constructor for class nextgen.core.annotation.filter.<A HREF="../nextgen/core/annotation/filter/FullyContainedFilter.html" title="class in nextgen.core.annotation.filter">FullyContainedFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/filter/FullyContainedFilter.html#FullyContainedFilter(java.util.List)"><B>FullyContainedFilter(List&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.filter.<A HREF="../nextgen/core/annotation/filter/FullyContainedFilter.html" title="class in nextgen.core.annotation.filter">FullyContainedFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/filter/FullyContainedFilter.html#FullyContainedFilter(nextgen.core.annotation.AnnotationList)"><B>FullyContainedFilter(AnnotationList&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.filter.<A HREF="../nextgen/core/annotation/filter/FullyContainedFilter.html" title="class in nextgen.core.annotation.filter">FullyContainedFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#fullyContains(nextgen.core.annotation.Annotation)"><B>fullyContains(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#fullyContains(nextgen.core.annotation.Annotation)"><B>fullyContains(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#fullyContains(nextgen.core.annotation.Annotation)"><B>fullyContains(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Test whether the annotation is fully contained in this
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#fullyContains(nextgen.core.annotation.Annotation)"><B>fullyContains(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/functionalAnnotation.html" title="class in broad.pda.gene"><B>functionalAnnotation</B></A> - Class in <A HREF="../broad/pda/gene/package-summary.html">broad.pda.gene</A><DD>&nbsp;<DT><A HREF="../broad/pda/gene/functionalAnnotation.html#functionalAnnotation()"><B>functionalAnnotation()</B></A> - 
Constructor for class broad.pda.gene.<A HREF="../broad/pda/gene/functionalAnnotation.html" title="class in broad.pda.gene">functionalAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/FWERDistribution.html" title="class in broad.pda.differentialExpression"><B>FWERDistribution</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/FWERDistribution.html#FWERDistribution(double[], Jama.Matrix)"><B>FWERDistribution(double[], Matrix)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/FWERDistribution.html" title="class in broad.pda.differentialExpression">FWERDistribution</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-5.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-7.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-6.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-6.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
