<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:30 EDT 2013 -->
<TITLE>
C-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="C-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-2.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-4.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-3.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-3.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_C_"><!-- --></A><H2>
<B>C</B></H2>
<DL>
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#C_CODE"><B>C_CODE</B></A> - 
Static variable in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/functionalAnnotation.html#calcAllTermsEnrichment(java.util.Collection, java.util.LinkedList)"><B>calcAllTermsEnrichment(Collection&lt;Gene&gt;, LinkedList&lt;Collection&lt;Gene&gt;&gt;)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/functionalAnnotation.html" title="class in broad.pda.gene">functionalAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.html#calcLightSilhouette()"><B>calcLightSilhouette()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.html" title="class in broad.pda.geneexpression.clustering">Kmeans</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#calcMeanNumDup()"><B>calcMeanNumDup()</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.html#calcSilhouette()"><B>calcSilhouette()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.html" title="class in broad.pda.geneexpression.clustering">Kmeans</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/functionalAnnotation.html#calcTermEnrichment(java.lang.String, java.util.HashMap, java.util.LinkedList, java.lang.Double[], java.lang.Double[], java.lang.Integer[], java.util.LinkedList, java.util.LinkedList)"><B>calcTermEnrichment(String, HashMap&lt;String, String&gt;, LinkedList&lt;HashMap&lt;String, String&gt;&gt;, Double[], Double[], Integer[], LinkedList&lt;Double&gt;, LinkedList&lt;Integer&gt;)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/functionalAnnotation.html" title="class in broad.pda.gene">functionalAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#calculateApproximatePVal(int, double, double, double, double)"><B>calculateApproximatePVal(int, double, double, double, double)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/tests/TestEnrichmentByFeature.html#calculateEnrichment(double, double)"><B>calculateEnrichment(double, double)</B></A> - 
Method in class nextgen.core.tests.<A HREF="../nextgen/core/tests/TestEnrichmentByFeature.html" title="class in nextgen.core.tests">TestEnrichmentByFeature</A>
<DD>Calculate enrichment off of counts
<DT><A HREF="../broad/pda/seq/rap/GCContent.html#calculateGc(byte[])"><B>calculateGc(byte[])</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GCContent.html" title="class in broad.pda.seq.rap">GCContent</A>
<DD>Returns -1 if no bases are ATCG (e.g.
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#calculatePVal(int, double, double, double)"><B>calculatePVal(int, double, double, double)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>Params written by Jesse Aug 20, 2012 ...
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#calculatePVal(int, double, double, double)"><B>calculatePVal(int, double, double, double)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#calculateScore()"><B>calculateScore()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Calculate the score of the alignment, not using the score field (the
 function only uses sequence1, sequence2, matrix and gap penalties).
<DT><A HREF="../broad/pda/seq/rap/RAPPeakCaller.html#callKLPeaks(java.lang.String, double)"><B>callKLPeaks(String, double)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RAPPeakCaller.html" title="class in broad.pda.seq.rap">RAPPeakCaller</A>
<DD>For each window, calculate the poisson lambda for the given window over the entire chromosome
<DT><A HREF="../broad/pda/seq/rap/RAPPeakCaller.html#callSkellamPeaks(java.lang.String, double, double)"><B>callSkellamPeaks(String, double, double)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RAPPeakCaller.html" title="class in broad.pda.seq.rap">RAPPeakCaller</A>
<DD>Call peaks using skellam
<DT><A HREF="../nextgen/core/readFilters/CanonicalSpliceFilter.html" title="class in nextgen.core.readFilters"><B>CanonicalSpliceFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>This class will filter reads by whether they have canonical splice junctions
 Canonical splice junctions are defined as junctions spanning an AG/GT pair<DT><A HREF="../nextgen/core/readFilters/CanonicalSpliceFilter.html#CanonicalSpliceFilter(java.lang.String)"><B>CanonicalSpliceFilter(String)</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/CanonicalSpliceFilter.html" title="class in nextgen.core.readFilters">CanonicalSpliceFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/CanonicalSpliceFilter.html#CanonicalSpliceFilter()"><B>CanonicalSpliceFilter()</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/CanonicalSpliceFilter.html" title="class in nextgen.core.readFilters">CanonicalSpliceFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#canViolateConstraints"><B>canViolateConstraints</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Cell.html" title="class in jaligner"><B>Cell</B></A> - Class in <A HREF="../jaligner/package-summary.html">jaligner</A><DD>A cell in a similarity matrix, to hold row, column and score.<DT><A HREF="../jaligner/Cell.html#Cell()"><B>Cell()</B></A> - 
Constructor for class jaligner.<A HREF="../jaligner/Cell.html" title="class in jaligner">Cell</A>
<DD>Constructor
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#centroidOf(java.util.Collection)"><B>centroidOf(Collection&lt;PositionWeightMatrix&gt;)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>Computes the euclidean centroid, it may not be the formal centroid for different metrics but intuitively the
 average counts should provide a good cluster representative which is what this method intends to return
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#CENTROMERE_TYPE"><B>CENTROMERE_TYPE</B></A> - 
Static variable in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#changeBackground(double, double, double, double)"><B>changeBackground(double, double, double, double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#changeBackground(Jama.Matrix)"><B>changeBackground(Matrix)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineUtils.html#checkForLSFFailures(java.lang.String, java.lang.Runtime)"><B>checkForLSFFailures(String, Runtime)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/PipelineUtils.html" title="class in broad.core.util">PipelineUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#checkOrientation(java.util.Collection)"><B>checkOrientation(Collection&lt;ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#checkScore()"><B>checkScore()</B></A> - 
Method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Check if the calculated score matches the field score.
<DT><A HREF="../nextgen/core/readFilters/ChimeraFilter.html" title="class in nextgen.core.readFilters"><B>ChimeraFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/ChimeraFilter.html#ChimeraFilter()"><B>ChimeraFilter()</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/ChimeraFilter.html" title="class in nextgen.core.readFilters">ChimeraFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/EmpiricalDistribution.html#chiSquareDistance(broad.core.math.EmpiricalDistribution)"><B>chiSquareDistance(EmpiricalDistribution)</B></A> - 
Method in class broad.core.math.<A HREF="../broad/core/math/EmpiricalDistribution.html" title="class in broad.core.math">EmpiricalDistribution</A>
<DD>This function will calculate the chi square distance between this and other empirical distribution.
<DT><A HREF="../Jama/Matrix.html#chol()"><B>chol()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Cholesky Decomposition
<DT><A HREF="../Jama/CholeskyDecomposition.html" title="class in Jama"><B>CholeskyDecomposition</B></A> - Class in <A HREF="../Jama/package-summary.html">Jama</A><DD>Cholesky Decomposition.<DT><A HREF="../Jama/CholeskyDecomposition.html#CholeskyDecomposition(Jama.Matrix)"><B>CholeskyDecomposition(Matrix)</B></A> - 
Constructor for class Jama.<A HREF="../Jama/CholeskyDecomposition.html" title="class in Jama">CholeskyDecomposition</A>
<DD>Cholesky algorithm for symmetric and positive definite matrix.
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#chr"><B>chr</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome"><B>Chromosome</B></A> - Class in <A HREF="../broad/pda/feature/genome/package-summary.html">broad.pda.feature.genome</A><DD>Represents a chromosome in a genome<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#Chromosome(java.lang.String)"><B>Chromosome(String)</B></A> - 
Constructor for class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#Chromosome(java.lang.String, java.lang.String)"><B>Chromosome(String, String)</B></A> - 
Constructor for class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome"><B>Chromosome.AgpEntry</B></A> - Class in <A HREF="../broad/pda/feature/genome/package-summary.html">broad.pda.feature.genome</A><DD>&nbsp;<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#Chromosome.AgpEntry(java.lang.String)"><B>Chromosome.AgpEntry(String)</B></A> - 
Constructor for class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntryFactory.html" title="class in broad.pda.feature.genome"><B>Chromosome.AgpEntryFactory</B></A> - Class in <A HREF="../broad/pda/feature/genome/package-summary.html">broad.pda.feature.genome</A><DD>&nbsp;<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntryFactory.html#Chromosome.AgpEntryFactory()"><B>Chromosome.AgpEntryFactory()</B></A> - 
Constructor for class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntryFactory.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntryFactory</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/ChromosomeInconsistencyException.html" title="class in nextgen.core.alignment"><B>ChromosomeInconsistencyException</B></A> - Exception in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>This is an exception class for the PairedEndAlignment classes.<DT><A HREF="../nextgen/core/alignment/ChromosomeInconsistencyException.html#ChromosomeInconsistencyException()"><B>ChromosomeInconsistencyException()</B></A> - 
Constructor for exception nextgen.core.alignment.<A HREF="../nextgen/core/alignment/ChromosomeInconsistencyException.html" title="class in nextgen.core.alignment">ChromosomeInconsistencyException</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/ChromosomeInconsistencyException.html#ChromosomeInconsistencyException(java.lang.String)"><B>ChromosomeInconsistencyException(String)</B></A> - 
Constructor for exception nextgen.core.alignment.<A HREF="../nextgen/core/alignment/ChromosomeInconsistencyException.html" title="class in nextgen.core.alignment">ChromosomeInconsistencyException</A>
<DD>This constructor prints the specified message while throwing the exception
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#chromosomeLength(java.lang.String)"><B>chromosomeLength(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ChromosomeStringComparator.html" title="class in broad.core.util"><B>ChromosomeStringComparator</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>&nbsp;<DT><A HREF="../broad/core/util/ChromosomeStringComparator.html#ChromosomeStringComparator()"><B>ChromosomeStringComparator()</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/ChromosomeStringComparator.html" title="class in broad.core.util">ChromosomeStringComparator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture"><B>ChromosomeTranscriptGraph</B></A> - Class in <A HREF="../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html#ChromosomeTranscriptGraph(java.lang.String)"><B>ChromosomeTranscriptGraph(String)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">ChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph"><B>ChromosomeWithBubblesJGraphT</B></A> - Class in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>Implements jung.Graph for Dijkstra (FOR NOW ONLY)<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String)"><B>ChromosomeWithBubblesJGraphT(String)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>CONSTRUCTORS
 TODO: A constructor that instantiates the class with a graph.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, int, int)"><B>ChromosomeWithBubblesJGraphT(String, int, int)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Collection, java.util.Collection, java.util.Map, java.util.Map, double, double, double, int)"><B>ChromosomeWithBubblesJGraphT(String, Collection&lt;? extends Annotation&gt;, Collection&lt;? extends Annotation&gt;, Map&lt;? extends Annotation, Double&gt;, Map&lt;? extends Annotation, Double&gt;, double, double, double, int)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Constructs a chromosome with bubbles with a given set of nodes and connections between them.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Collection, java.util.Collection)"><B>ChromosomeWithBubblesJGraphT(String, Collection&lt;? extends Annotation&gt;, Collection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Constructs a chromosome with bubbles with a given set of nodes and connections between them.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Map, java.util.Map, double, double, double, double)"><B>ChromosomeWithBubblesJGraphT(String, Map&lt;? extends Annotation, Double&gt;, Map&lt;ChromosomeWithBubblesJGraphT.BubbleEdge, Double&gt;, double, double, double, double)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Constructs a chromosome with bubbles with a given set of nodes, edges.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#ChromosomeWithBubblesJGraphT(java.lang.String, java.util.Collection, java.util.Collection, double, double, double, int)"><B>ChromosomeWithBubblesJGraphT(String, Collection&lt;? extends Annotation&gt;, Collection&lt;? extends Annotation&gt;, double, double, double, int)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>???
 This will always return a Null Pointer Exception
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph"><B>ChromosomeWithBubblesJGraphT.BubbleEdge</B></A> - Class in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>This class represents the Edges of the graph<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#ChromosomeWithBubblesJGraphT.BubbleEdge(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, double, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.EdgeSourceType)"><B>ChromosomeWithBubblesJGraphT.BubbleEdge(Annotation, Annotation, Annotation, double, ChromosomeWithBubblesJGraphT.EdgeSourceType)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>This constructor creates an instance of an edge of "type" from v1 to v2.
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#ChromosomeWithBubblesJGraphT.BubbleEdge(nextgen.core.annotation.Annotation)"><B>ChromosomeWithBubblesJGraphT.BubbleEdge(Annotation)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>This constructor creates a self-edge from startVertex to startVertex with weight 0
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.EdgeSourceType.html" title="enum in broad.pda.seq.graph"><B>ChromosomeWithBubblesJGraphT.EdgeSourceType</B></A> - Enum in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.html" title="class in broad.pda.seq.graph"><B>ChromosomeWithBubblesJGraphT.ForwardAndBackIterator</B></A> - Class in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>FORWARD ITERATOR<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.html#ChromosomeWithBubblesJGraphT.ForwardAndBackIterator(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT, int, int, boolean)"><B>ChromosomeWithBubblesJGraphT.ForwardAndBackIterator(ChromosomeWithBubblesJGraphT, int, int, boolean)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ForwardAndBackIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.ForwardAndBackIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.MinReadFilter.html" title="class in broad.pda.seq.graph"><B>ChromosomeWithBubblesJGraphT.MinReadFilter</B></A> - Class in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>This class exercises the minimum length filter on reads<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.MinReadFilter.html#ChromosomeWithBubblesJGraphT.MinReadFilter(int)"><B>ChromosomeWithBubblesJGraphT.MinReadFilter(int)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.MinReadFilter.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.MinReadFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.RegionReadFilter.html" title="class in broad.pda.seq.graph"><B>ChromosomeWithBubblesJGraphT.RegionReadFilter</B></A> - Class in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>Read filter overlapping region<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.RegionReadFilter.html#ChromosomeWithBubblesJGraphT.RegionReadFilter(int, int)"><B>ChromosomeWithBubblesJGraphT.RegionReadFilter(int, int)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.RegionReadFilter.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.RegionReadFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ReverseWindowIterator.html" title="class in broad.pda.seq.graph"><B>ChromosomeWithBubblesJGraphT.ReverseWindowIterator</B></A> - Class in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>REVERSE ITERATOR<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ReverseWindowIterator.html#ChromosomeWithBubblesJGraphT.ReverseWindowIterator(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT, int, int, int, boolean)"><B>ChromosomeWithBubblesJGraphT.ReverseWindowIterator(ChromosomeWithBubblesJGraphT, int, int, int, boolean)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.ReverseWindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.ReverseWindowIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph"><B>ChromosomeWithBubblesJGraphT.WindowIterator</B></A> - Class in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>An iterator over the collection of RefSeq genes<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html#ChromosomeWithBubblesJGraphT.WindowIterator(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT, int, int, int)"><B>ChromosomeWithBubblesJGraphT.WindowIterator(ChromosomeWithBubblesJGraphT, int, int, int)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html#ChromosomeWithBubblesJGraphT.WindowIterator(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT, int, int, int, boolean)"><B>ChromosomeWithBubblesJGraphT.WindowIterator(ChromosomeWithBubblesJGraphT, int, int, int, boolean)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.WindowIterator.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.WindowIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#chunk(int, int)"><B>chunk(int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/CisVsTrans.html" title="class in broad.pda.differentialExpression"><B>CisVsTrans</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/CisVsTrans.html#CisVsTrans(broad.core.datastructures.MatrixWithHeaders, java.io.File, java.io.File, java.lang.String, java.io.File, java.io.File)"><B>CisVsTrans(MatrixWithHeaders, File, File, String, File, File)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/CisVsTrans.html" title="class in broad.pda.differentialExpression">CisVsTrans</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/ClassifyAgilentLincProbes.html" title="class in broad.pda.geneexpression.agilent"><B>ClassifyAgilentLincProbes</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/ClassifyAgilentLincProbes.html#ClassifyAgilentLincProbes(java.io.File, java.io.File, java.lang.String)"><B>ClassifyAgilentLincProbes(File, File, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/ClassifyAgilentLincProbes.html" title="class in broad.pda.geneexpression.agilent">ClassifyAgilentLincProbes</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/statistics/ClassifyReconstructions.html" title="class in nextgen.core.scripture.statistics"><B>ClassifyReconstructions</B></A> - Class in <A HREF="../nextgen/core/scripture/statistics/package-summary.html">nextgen.core.scripture.statistics</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/statistics/ClassifyReconstructions.html#ClassifyReconstructions(java.lang.String, java.lang.String, java.lang.String)"><B>ClassifyReconstructions(String, String, String)</B></A> - 
Constructor for class nextgen.core.scripture.statistics.<A HREF="../nextgen/core/scripture/statistics/ClassifyReconstructions.html" title="class in nextgen.core.scripture.statistics">ClassifyReconstructions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#cleanIsoforms()"><B>cleanIsoforms()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#clear()"><B>clear()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#clear()"><B>clear()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Remove all entries.
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#clear()"><B>clear()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#clear()"><B>clear()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#clear()"><B>clear()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFF.html#clearAttribute(java.lang.String)"><B>clearAttribute(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFF.html" title="class in broad.core.annotation">GFF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#clearFullIntervalTreeAsAlignmentsCached(java.lang.String)"><B>clearFullIntervalTreeAsAlignmentsCached(String)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#clearFullIntervalTreeAsAlignmentsCached(java.lang.String)"><B>clearFullIntervalTreeAsAlignmentsCached(String)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#clone()"><B>clone()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Clone the Matrix object.
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#CLONE_TYPE"><B>CLONE_TYPE</B></A> - 
Static variable in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#cloneRegions()"><B>cloneRegions()</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#close()"><B>close()</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/WrappedIterator.html#close()"><B>close()</B></A> - 
Method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/WrappedIterator.html" title="class in broad.pda.samtools">WrappedIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#close()"><B>close()</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.EmptyCloseableIterator.html#close()"><B>close()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.EmptyCloseableIterator.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel.EmptyCloseableIterator</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/WrappedIGVIterator.html#close()"><B>close()</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/WrappedIGVIterator.html" title="class in broad.pda.seq.segmentation">WrappedIGVIterator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/ShuffledIterator.html#close()"><B>close()</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/ShuffledIterator.html" title="class in nextgen.core.coordinatesystem">ShuffledIterator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/general/CloseableFilterIterator.html#close()"><B>close()</B></A> - 
Method in class nextgen.core.general.<A HREF="../nextgen/core/general/CloseableFilterIterator.html" title="class in nextgen.core.general">CloseableFilterIterator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowScoreIterator.html#close()"><B>close()</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowScoreIterator.html" title="class in nextgen.core.model.score">WindowScoreIterator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#close()"><B>close()</B></A> - 
Method in class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#close()"><B>close()</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/general/CloseableFilterIterator.html" title="class in nextgen.core.general"><B>CloseableFilterIterator</B></A>&lt;<A HREF="../nextgen/core/general/CloseableFilterIterator.html" title="type parameter in CloseableFilterIterator">T</A>&gt; - Class in <A HREF="../nextgen/core/general/package-summary.html">nextgen.core.general</A><DD>&nbsp;<DT><A HREF="../nextgen/core/general/CloseableFilterIterator.html#CloseableFilterIterator(net.sf.samtools.util.CloseableIterator, org.apache.commons.collections15.Predicate)"><B>CloseableFilterIterator(CloseableIterator&lt;T&gt;, Predicate&lt;? super T&gt;)</B></A> - 
Constructor for class nextgen.core.general.<A HREF="../nextgen/core/general/CloseableFilterIterator.html" title="class in nextgen.core.general">CloseableFilterIterator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/general/CloseableFilterIterator.html#CloseableFilterIterator(java.util.Iterator, org.apache.commons.collections15.Predicate)"><B>CloseableFilterIterator(Iterator&lt;T&gt;, Predicate&lt;? super T&gt;)</B></A> - 
Constructor for class nextgen.core.general.<A HREF="../nextgen/core/general/CloseableFilterIterator.html" title="class in nextgen.core.general">CloseableFilterIterator</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/formats/CLUSTAL.html" title="class in jaligner.formats"><B>CLUSTAL</B></A> - Class in <A HREF="../jaligner/formats/package-summary.html">jaligner.formats</A><DD>CLUSTAL format.<DT><A HREF="../jaligner/formats/CLUSTAL.html#CLUSTAL()"><B>CLUSTAL()</B></A> - 
Constructor for class jaligner.formats.<A HREF="../jaligner/formats/CLUSTAL.html" title="class in jaligner.formats">CLUSTAL</A>
<DD>Constructor
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering"><B>Cluster</B></A> - Class in <A HREF="../broad/pda/geneexpression/clustering/package-summary.html">broad.pda.geneexpression.clustering</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#Cluster(java.lang.String)"><B>Cluster(String)</B></A> - 
Constructor for class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/Cluster.html#Cluster(java.util.Collection, java.lang.String)"><B>Cluster(Collection&lt;String&gt;, String)</B></A> - 
Constructor for class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Cluster.html" title="class in broad.pda.geneexpression.clustering">Cluster</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html#cluster(boolean, boolean)"><B>cluster(boolean, boolean)</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/HierarchicalClustering.html" title="class in broad.pda.geneexpression.clustering">HierarchicalClustering</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/ClusterDistanceFunction.html" title="interface in broad.pda.geneexpression.clustering"><B>ClusterDistanceFunction</B></A> - Interface in <A HREF="../broad/pda/geneexpression/clustering/package-summary.html">broad.pda.geneexpression.clustering</A><DD>&nbsp;<DT><A HREF="../broad/core/util/CLUtil.html" title="class in broad.core.util"><B>CLUtil</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>&nbsp;<DT><A HREF="../broad/core/util/CLUtil.html#CLUtil()"><B>CLUtil()</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/CLUtil.html" title="class in broad.core.util">CLUtil</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util"><B>CLUtil.ArgumentMap</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>&nbsp;<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#CLUtil.ArgumentMap(int, java.lang.String, java.lang.String)"><B>CLUtil.ArgumentMap(int, String, String)</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/AlignmentExtractor.html#codonToAminoacid"><B>codonToAminoacid</B></A> - 
Static variable in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/AlignmentExtractor.html" title="class in broad.core.multiplealignment">AlignmentExtractor</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#colapse(int)"><B>colapse(int)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html#collapse(nextgen.core.annotation.Annotation, boolean)"><B>collapse(Annotation, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotationCollection.html" title="class in nextgen.core.annotation">AbstractAnnotationCollection</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationCollection.html#collapse(nextgen.core.annotation.Annotation, boolean)"><B>collapse(Annotation, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationCollection.html" title="interface in nextgen.core.annotation">AnnotationCollection</A>
<DD>Collapse all reads within the defined region into the overlapping set
<DT><A HREF="../nextgen/core/annotation/AnnotationList.html#collapse()"><B>collapse()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>
<DD>Collapse overlapping annotations.
<DT><A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util"><B>CollapseByIntersection</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>&nbsp;<DT><A HREF="../broad/core/util/CollapseByIntersection.html#CollapseByIntersection()"><B>CollapseByIntersection()</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#collapseByIntersection(java.util.Collection, boolean)"><B>collapseByIntersection(Collection&lt;? extends Annotation&gt;, boolean)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/CollapseByIntersection.html#CollapseGenesByIntersection(java.util.Collection, boolean)"><B>CollapseGenesByIntersection(Collection&lt;Gene&gt;, boolean)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/CollapseByIntersection.html" title="class in broad.core.util">CollapseByIntersection</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/CollapseProbesByMinPValue.html" title="class in broad.pda.geneexpression.agilent"><B>CollapseProbesByMinPValue</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/CollapseProbesByMinPValue.html#CollapseProbesByMinPValue(java.io.File, java.lang.String)"><B>CollapseProbesByMinPValue(File, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/CollapseProbesByMinPValue.html" title="class in broad.pda.geneexpression.agilent">CollapseProbesByMinPValue</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/CollapseProjections.html" title="class in broad.pda.differentialExpression"><B>CollapseProjections</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/CollapseProjections.html#CollapseProjections(broad.core.datastructures.MatrixWithHeaders, java.lang.String)"><B>CollapseProjections(MatrixWithHeaders, String)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/CollapseProjections.html" title="class in broad.pda.differentialExpression">CollapseProjections</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#collapseSingleExonTranscripts()"><B>collapseSingleExonTranscripts()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>Merges all single exon transcripts
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap"><B>CollectAnnotationEnrichments</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#CollectAnnotationEnrichments()"><B>CollectAnnotationEnrichments()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap"><B>CollectAnnotationEnrichments.OverlapEnrichmentScore</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html#CollectAnnotationEnrichments.OverlapEnrichmentScore(java.lang.String, java.lang.String, java.util.List, java.util.List)"><B>CollectAnnotationEnrichments.OverlapEnrichmentScore(String, String, List&lt;CountScore&gt;, List&lt;CountScore&gt;)</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.OverlapEnrichmentScore.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments.OverlapEnrichmentScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#columnDensityNormalization()"><B>columnDensityNormalization()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Normalizes each column so it adds up to 1
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#columnDimension()"><B>columnDimension()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/CombinationGenerator.html" title="class in broad.core.math"><B>CombinationGenerator</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/CombinationGenerator.html#CombinationGenerator(int, int)"><B>CombinationGenerator(int, int)</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/CombinationGenerator.html" title="class in broad.core.math">CombinationGenerator</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/CombinationGenerator.html" title="class in nextgen.core.scripture"><B>CombinationGenerator</B></A> - Class in <A HREF="../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/CombinationGenerator.html#CombinationGenerator(int[])"><B>CombinationGenerator(int[])</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/CombinationGenerator.html" title="class in nextgen.core.scripture">CombinationGenerator</A>
<DD>Choose a value from each of the possible values in the array until all permutations are exhausted
<DT><A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser"><B>CommandLineParser</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>&nbsp;<DT><A HREF="../broad/core/parser/CommandLineParser.html#CommandLineParser()"><B>CommandLineParser()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#commaSeparatedStringToList(java.lang.String)"><B>commaSeparatedStringToList(String)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/OutputPadlock.html#COMMON_INSERT"><B>COMMON_INSERT</B></A> - 
Static variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/OutputPadlock.html" title="class in broad.pda.methylation">OutputPadlock</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/Commons.html" title="class in jaligner.util"><B>Commons</B></A> - Class in <A HREF="../jaligner/util/package-summary.html">jaligner.util</A><DD>Global constants/variables/settings<DT><A HREF="../jaligner/util/Commons.html#Commons()"><B>Commons()</B></A> - 
Constructor for class jaligner.util.<A HREF="../jaligner/util/Commons.html" title="class in jaligner.util">Commons</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html" title="class in nextgen.core.normalize"><B>ComparativeCountsMatrix</B></A> - Class in <A HREF="../nextgen/core/normalize/package-summary.html">nextgen.core.normalize</A><DD>&nbsp;<DT><A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html#ComparativeCountsMatrix(java.lang.String, java.lang.String, java.lang.String, java.lang.String)"><B>ComparativeCountsMatrix(String, String, String, String)</B></A> - 
Constructor for class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html" title="class in nextgen.core.normalize">ComparativeCountsMatrix</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html#ComparativeCountsMatrix(java.lang.String, java.util.ArrayList, java.lang.String, java.util.Map)"><B>ComparativeCountsMatrix(String, ArrayList&lt;String&gt;, String, Map&lt;String, Collection&lt;Gene&gt;&gt;)</B></A> - 
Constructor for class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/ComparativeCountsMatrix.html" title="class in nextgen.core.normalize">ComparativeCountsMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ChromosomeStringComparator.html#compare(java.lang.String, java.lang.String)"><B>compare(String, String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/ChromosomeStringComparator.html" title="class in broad.core.util">ChromosomeStringComparator</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/matrix/MatricesCompartor.html#compare(java.lang.Object, java.lang.Object)"><B>compare(Object, Object)</B></A> - 
Method in class jaligner.matrix.<A HREF="../jaligner/matrix/MatricesCompartor.html" title="class in jaligner.matrix">MatricesCompartor</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#compareNames(java.lang.String, java.lang.String)"><B>compareNames(String, String)</B></A> - 
Static method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotation.html#compareTo(broad.core.annotation.GenomicAnnotation)"><B>compareTo(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotation.html" title="class in broad.core.annotation">BasicGenomicAnnotation</A>
<DD>Default compareTo method: If annotations are in the same chromosome
 their start/end locations are compared otherwise their chromosomes are.
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#compareTo(broad.core.annotation.LightweightGenomicAnnotation)"><B>compareTo(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#compareTo(java.lang.Object)"><B>compareTo(Object)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#compareTo(broad.core.annotation.GenomicAnnotation)"><B>compareTo(GenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#compareTo(broad.core.annotation.LightweightGenomicAnnotation)"><B>compareTo(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#compareTo(nextgen.core.annotation.Annotation)"><B>compareTo(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/HairpinKmer.html#compareTo(broad.pda.capture.designer.HairpinKmer)"><B>compareTo(HairpinKmer)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/HairpinKmer.html" title="class in broad.pda.capture.designer">HairpinKmer</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.html#compareTo(java.lang.Object)"><B>compareTo(Object)</B></A> - 
Method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.html" title="class in broad.pda.feature.genome">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html#compareTo(broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.BubbleEdge)"><B>compareTo(ChromosomeWithBubblesJGraphT.BubbleEdge)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.BubbleEdge.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.BubbleEdge</A>
<DD>Compares this edge to another edge.
<DT><A HREF="../broad/pda/seq/graph/Path.html#compareTo(broad.pda.seq.graph.Path)"><B>compareTo(Path)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html#compareTo(nextgen.core.alignment.FeatureSequenceAlignment.UnorderedGenePair)"><B>compareTo(FeatureSequenceAlignment.UnorderedGenePair)</B></A> - 
Method in class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/FeatureSequenceAlignment.UnorderedGenePair.html" title="class in nextgen.core.alignment">FeatureSequenceAlignment.UnorderedGenePair</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#compareTo(nextgen.core.annotation.Annotation)"><B>compareTo(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#compareTo(nextgen.core.annotation.CompoundInterval)"><B>compareTo(CompoundInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#compareTo(nextgen.core.annotation.Annotation)"><B>compareTo(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/SingleInterval.html#compareTo(nextgen.core.annotation.SingleInterval)"><B>compareTo(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation">SingleInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/feature/GeneWindow.html#compareTo(nextgen.core.annotation.Annotation)"><B>compareTo(Annotation)</B></A> - 
Method in class nextgen.core.feature.<A HREF="../nextgen/core/feature/GeneWindow.html" title="class in nextgen.core.feature">GeneWindow</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#compareTo(nextgen.core.model.score.CountScore)"><B>compareTo(CountScore)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>First compare counts
 Then compare annotations
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#compareTo(nextgen.core.annotation.Annotation)"><B>compareTo(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#compareToAnnotation(nextgen.core.annotation.Annotation)"><B>compareToAnnotation(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#compareToAnnotation(nextgen.core.annotation.Annotation)"><B>compareToAnnotation(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#compareToAnnotation(nextgen.core.annotation.Annotation)"><B>compareToAnnotation(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>Should behave like compareTo for a basic annotation
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#compareToAnnotation(nextgen.core.annotation.Annotation)"><B>compareToAnnotation(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html#compatible(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>compatible(Annotation, Annotation)</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/BuildScriptureCoordinateSpace.html" title="class in nextgen.core.scripture">BuildScriptureCoordinateSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#complement(java.lang.String)"><B>complement(String)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#complement()"><B>complement()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeMIRScore.html#complement(java.lang.String)"><B>complement(String)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeMIRScore.html" title="class in broad.pda.capture.designer">ComputeMIRScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#complement()"><B>complement()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#complement()"><B>complement()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#complement()"><B>complement()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#complement(int, int)"><B>complement(int, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#complement()"><B>complement()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/DesignPCRTails.html#complementary(java.lang.String, java.lang.String)"><B>complementary(String, String)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/DesignPCRTails.html" title="class in broad.pda.capture.designer">DesignPCRTails</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation"><B>CompoundInterval</B></A> - Class in <A HREF="../nextgen/core/annotation/package-summary.html">nextgen.core.annotation</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#CompoundInterval()"><B>CompoundInterval()</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#CompoundInterval(int, int)"><B>CompoundInterval(int, int)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#CompoundInterval(nextgen.core.annotation.CompoundInterval)"><B>CompoundInterval(CompoundInterval)</B></A> - 
Constructor for class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#compress()"><B>compress()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#compress()"><B>compress()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>Compresses alignment by removing columns containing reference gaps
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#compressMatrixByRowDescriptions()"><B>compressMatrixByRowDescriptions()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>Replaces each set of rows with same description with a row with column values the mean of the original row column values
<DT><A HREF="../broad/core/sequence/Sequence.html#compute454Flow(char[])"><B>compute454Flow(char[])</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#compute454Flow(char[], int, int)"><B>compute454Flow(char[], int, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialScoring.html#computeAbsFoldMatrix(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection)"><B>computeAbsFoldMatrix(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;)</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialScoring.html" title="class in broad.pda.differentialExpression">DifferentialScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/misc/ComputeConservedMotifs.html" title="class in broad.core.motif.misc"><B>ComputeConservedMotifs</B></A> - Class in <A HREF="../broad/core/motif/misc/package-summary.html">broad.core.motif.misc</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/misc/ComputeConservedMotifs.html#ComputeConservedMotifs(broad.pda.datastructures.Alignments, java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.io.File)"><B>ComputeConservedMotifs(Alignments, String, String, String, String, File)</B></A> - 
Constructor for class broad.core.motif.misc.<A HREF="../broad/core/motif/misc/ComputeConservedMotifs.html" title="class in broad.core.motif.misc">ComputeConservedMotifs</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/ComputeCorrelationOfReplicates.html" title="class in broad.pda.geneexpression.clustering"><B>ComputeCorrelationOfReplicates</B></A> - Class in <A HREF="../broad/pda/geneexpression/clustering/package-summary.html">broad.pda.geneexpression.clustering</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/clustering/ComputeCorrelationOfReplicates.html#ComputeCorrelationOfReplicates(java.io.File, java.lang.String)"><B>ComputeCorrelationOfReplicates(File, String)</B></A> - 
Constructor for class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/ComputeCorrelationOfReplicates.html" title="class in broad.pda.geneexpression.clustering">ComputeCorrelationOfReplicates</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/chip/ComputeEnrichments.html" title="class in broad.pda.seq.chip"><B>ComputeEnrichments</B></A> - Class in <A HREF="../broad/pda/seq/chip/package-summary.html">broad.pda.seq.chip</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/chip/ComputeEnrichments.html#ComputeEnrichments(java.io.File[], java.util.Collection, java.lang.String, java.lang.String)"><B>ComputeEnrichments(File[], Collection&lt;Annotation&gt;, String, String)</B></A> - 
Constructor for class broad.pda.seq.chip.<A HREF="../broad/pda/seq/chip/ComputeEnrichments.html" title="class in broad.pda.seq.chip">ComputeEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/ComputeFDR.html" title="class in broad.core.math"><B>ComputeFDR</B></A> - Class in <A HREF="../broad/core/math/package-summary.html">broad.core.math</A><DD>&nbsp;<DT><A HREF="../broad/core/math/ComputeFDR.html#ComputeFDR()"><B>ComputeFDR()</B></A> - 
Constructor for class broad.core.math.<A HREF="../broad/core/math/ComputeFDR.html" title="class in broad.core.math">ComputeFDR</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialScoring.html#computeFoldMatrix(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection)"><B>computeFoldMatrix(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;)</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialScoring.html" title="class in broad.pda.differentialExpression">DifferentialScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#computeGCContent(java.lang.String)"><B>computeGCContent(String)</B></A> - 
Static method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html#computeGCPercent(java.lang.String)"><B>computeGCPercent(String)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html" title="class in broad.pda.capture.designer">ComputeOriginalScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/rnaprotein/analysis/ComputeGeneEnrichment.html" title="class in nextgen.rnaprotein.analysis"><B>ComputeGeneEnrichment</B></A> - Class in <A HREF="../nextgen/rnaprotein/analysis/package-summary.html">nextgen.rnaprotein.analysis</A><DD>&nbsp;<DT><A HREF="../nextgen/rnaprotein/analysis/ComputeGeneEnrichment.html#ComputeGeneEnrichment(java.lang.String, java.lang.String, java.util.Map, java.lang.String)"><B>ComputeGeneEnrichment(String, String, Map&lt;String, Collection&lt;Gene&gt;&gt;, String)</B></A> - 
Constructor for class nextgen.rnaprotein.analysis.<A HREF="../nextgen/rnaprotein/analysis/ComputeGeneEnrichment.html" title="class in nextgen.rnaprotein.analysis">ComputeGeneEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#computeGlobalStats()"><B>computeGlobalStats()</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>Compute global stats and cache to a file for future use, or read from file if already cached
<DT><A HREF="../broad/pda/geneexpression/ComputeHypergeomtricEnrichment.html" title="class in broad.pda.geneexpression"><B>ComputeHypergeomtricEnrichment</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/ComputeHypergeomtricEnrichment.html#ComputeHypergeomtricEnrichment(broad.core.datastructures.MatrixWithHeaders, java.util.Map, java.util.Map, java.lang.String, double, int)"><B>ComputeHypergeomtricEnrichment(MatrixWithHeaders, Map&lt;String, Collection&lt;String&gt;&gt;, Map&lt;String, String&gt;, String, double, int)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ComputeHypergeomtricEnrichment.html" title="class in broad.pda.geneexpression">ComputeHypergeomtricEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/ComputeHypergeomtricEnrichment.html#ComputeHypergeomtricEnrichment(java.util.Collection, java.util.Collection, java.util.Collection)"><B>ComputeHypergeomtricEnrichment(Collection&lt;String&gt;, Collection&lt;String&gt;, Collection&lt;String&gt;)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ComputeHypergeomtricEnrichment.html" title="class in broad.pda.geneexpression">ComputeHypergeomtricEnrichment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#computeJMatrix(org.forester.phylogeny.PhylogenyNode)"><B>computeJMatrix(PhylogenyNode)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/ComputeKmersForRegion.html" title="class in broad.core.siphy.tools.conservation"><B>ComputeKmersForRegion</B></A> - Class in <A HREF="../broad/core/siphy/tools/conservation/package-summary.html">broad.core.siphy.tools.conservation</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/tools/conservation/ComputeKmersForRegion.html#ComputeKmersForRegion(java.io.File, java.io.File, java.lang.String, java.lang.String, java.lang.String, int)"><B>ComputeKmersForRegion(File, File, String, String, String, int)</B></A> - 
Constructor for class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/ComputeKmersForRegion.html" title="class in broad.core.siphy.tools.conservation">ComputeKmersForRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#computeLikelihood(java.util.Map, org.forester.phylogeny.PhylogenyNode, int)"><B>computeLikelihood(Map&lt;String, Matrix&gt;, PhylogenyNode, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/ComputeMaxPercentileOfElements.html" title="class in broad.core.siphy.tools.conservation"><B>ComputeMaxPercentileOfElements</B></A> - Class in <A HREF="../broad/core/siphy/tools/conservation/package-summary.html">broad.core.siphy.tools.conservation</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/tools/conservation/ComputeMaxPercentileOfElements.html#ComputeMaxPercentileOfElements(java.io.File, java.lang.String, double)"><B>ComputeMaxPercentileOfElements(File, String, double)</B></A> - 
Constructor for class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/ComputeMaxPercentileOfElements.html" title="class in broad.core.siphy.tools.conservation">ComputeMaxPercentileOfElements</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/ComputeMaxPercentileOfElements.html#ComputeMaxPercentileOfElements(java.io.File, java.io.File, java.lang.String, java.lang.String, java.lang.String, int, double)"><B>ComputeMaxPercentileOfElements(File, File, String, String, String, int, double)</B></A> - 
Constructor for class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/ComputeMaxPercentileOfElements.html" title="class in broad.core.siphy.tools.conservation">ComputeMaxPercentileOfElements</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerUtils.html#computeMaxTM(jaligner.Alignment)"><B>computeMaxTM(Alignment)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerUtils.html" title="class in broad.core.primer3">PrimerUtils</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html#computeMeanNumDup()"><B>computeMeanNumDup()</B></A> - 
Method in class nextgen.core.programs.<A HREF="../nextgen/core/programs/MarkDuplicatesFromNameUMI.MarkDuplicatesResults.html" title="class in nextgen.core.programs">MarkDuplicatesFromNameUMI.MarkDuplicatesResults</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeMIRScore.html" title="class in broad.pda.capture.designer"><B>ComputeMIRScore</B></A> - Class in <A HREF="../broad/pda/capture/designer/package-summary.html">broad.pda.capture.designer</A><DD>&nbsp;<DT><A HREF="../broad/pda/capture/designer/ComputeMIRScore.html#ComputeMIRScore()"><B>ComputeMIRScore()</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeMIRScore.html" title="class in broad.pda.capture.designer">ComputeMIRScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeMIRScore.html#computeMIRScore(java.lang.String, java.io.File, java.io.File)"><B>computeMIRScore(String, File, File)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeMIRScore.html" title="class in broad.pda.capture.designer">ComputeMIRScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeMIRScore.html#computeMIRScore(java.lang.String, java.util.Map, java.util.Set, java.util.Set)"><B>computeMIRScore(String, Map&lt;String, Double&gt;, Set&lt;String&gt;, Set&lt;String&gt;)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeMIRScore.html" title="class in broad.pda.capture.designer">ComputeMIRScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeMIRScore.html#computeMIRScore(java.lang.String)"><B>computeMIRScore(String)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeMIRScore.html" title="class in broad.pda.capture.designer">ComputeMIRScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html" title="class in broad.pda.capture.designer"><B>ComputeOriginalScore</B></A> - Class in <A HREF="../broad/pda/capture/designer/package-summary.html">broad.pda.capture.designer</A><DD>&nbsp;<DT><A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html#ComputeOriginalScore()"><B>ComputeOriginalScore()</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html" title="class in broad.pda.capture.designer">ComputeOriginalScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html#computeOriginalScore(java.lang.String, int)"><B>computeOriginalScore(String, int)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html" title="class in broad.pda.capture.designer">ComputeOriginalScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html#computeOriginalScorePerKMer(java.lang.String)"><B>computeOriginalScorePerKMer(String)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeOriginalScore.html" title="class in broad.pda.capture.designer">ComputeOriginalScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/hmm/MarkovModel.html#computePathLogLikelihood(java.util.List, short[], int, int)"><B>computePathLogLikelihood(List&lt;T&gt;, short[], int, int)</B></A> - 
Method in class broad.core.hmm.<A HREF="../broad/core/hmm/MarkovModel.html" title="class in broad.core.hmm">MarkovModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.html#computePathLogLikelihood(broad.core.multiplealignment.MultipleAlignment, short[], int, int)"><B>computePathLogLikelihood(MultipleAlignment, short[], int, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.html" title="class in broad.core.siphy">PiHMM</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/ComputePercentageOfKmers.html" title="class in broad.core.siphy.tools.conservation"><B>ComputePercentageOfKmers</B></A> - Class in <A HREF="../broad/core/siphy/tools/conservation/package-summary.html">broad.core.siphy.tools.conservation</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/tools/conservation/ComputePercentageOfKmers.html#ComputePercentageOfKmers(java.io.File, java.lang.String, double)"><B>ComputePercentageOfKmers(File, String, double)</B></A> - 
Constructor for class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/ComputePercentageOfKmers.html" title="class in broad.core.siphy.tools.conservation">ComputePercentageOfKmers</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/tools/conservation/ComputePercentageOfKmers.html#ComputePercentageOfKmers(java.io.File, java.util.Collection, java.lang.String, double)"><B>ComputePercentageOfKmers(File, Collection&lt;Gene&gt;, String, double)</B></A> - 
Constructor for class broad.core.siphy.tools.conservation.<A HREF="../broad/core/siphy/tools/conservation/ComputePercentageOfKmers.html" title="class in broad.core.siphy.tools.conservation">ComputePercentageOfKmers</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#computePermutedZScores(double, double[], double)"><B>computePermutedZScores(double, double[], double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#computePermutedZScores(double, double[])"><B>computePermutedZScores(double, double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#computeProb()"><B>computeProb()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.NodeLikelihoodParameters.html#computeProbabilityOfLetterAtNode()"><B>computeProbabilityOfLetterAtNode()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.NodeLikelihoodParameters.html" title="class in broad.core.siphy">EvolutionaryModel.NodeLikelihoodParameters</A>
<DD>Estimates base probabilities at the node.
<DT><A HREF="../broad/core/sequence/SequenceRegionWindow.html#computeScore()"><B>computeScore()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegionWindow.html" title="class in broad.core.sequence">SequenceRegionWindow</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#computeScoreDistribution(float, float, float, float, int)"><B>computeScoreDistribution(float, float, float, float, int)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>Compute Distribution of scores for motif.
<DT><A HREF="../broad/core/motif/PWMUtils.html#computeSequenceBGSequences(java.util.Map, broad.core.annotation.AnnotationReader)"><B>computeSequenceBGSequences(Map&lt;String, Sequence&gt;, AnnotationReader&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Static method in class broad.core.motif.<A HREF="../broad/core/motif/PWMUtils.html" title="class in broad.core.motif">PWMUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialScoring.html#computeTestStatistics(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, double[], boolean, boolean)"><B>computeTestStatistics(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, double[], boolean, boolean)</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialScoring.html" title="class in broad.pda.differentialExpression">DifferentialScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/DifferentialScoring.html#computeTestStatistics(broad.core.datastructures.MatrixWithHeaders, java.util.Map, double[], boolean, boolean)"><B>computeTestStatistics(MatrixWithHeaders, Map&lt;String, Collection&lt;String&gt;&gt;, double[], boolean, boolean)</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/DifferentialScoring.html" title="class in broad.pda.differentialExpression">DifferentialScoring</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerUtils.html#computeTM(java.lang.String)"><B>computeTM(String)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerUtils.html" title="class in broad.core.primer3">PrimerUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerUtils.html#computeTM(jaligner.Alignment)"><B>computeTM(Alignment)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerUtils.html" title="class in broad.core.primer3">PrimerUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerUtils.html#computeTM2(jaligner.Alignment)"><B>computeTM2(Alignment)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerUtils.html" title="class in broad.core.primer3">PrimerUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#computeTransitions(double)"><B>computeTransitions(double)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>Computes the vector of transition probabilities for the given branch length and value of the end state.
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#concatAnnotationSet(int, java.util.List)"><B>concatAnnotationSet(int, List&lt;? extends GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Moran, Feb22 2010
 concat  an annotation set to an exciting one (without merging)
 WARNING: assumes that same types are concatenated- uses casting to current 
 objets annotation type
<DT><A HREF="../Jama/Matrix.html#cond()"><B>cond()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Matrix condition (2 norm)
<DT><A HREF="../Jama/SingularValueDecomposition.html#cond()"><B>cond()</B></A> - 
Method in class Jama.<A HREF="../Jama/SingularValueDecomposition.html" title="class in Jama">SingularValueDecomposition</A>
<DD>Two norm condition number
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#confidenceByColumn(java.util.List)"><B>confidenceByColumn(List&lt;String&gt;)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser"><B>ConfigFileParser</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>This class will parse the config file provided to it.<DT><A HREF="../broad/core/parser/ConfigFileParser.html#ConfigFileParser(java.lang.String)"><B>ConfigFileParser(String)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Constructor will read the input config file and store the commands and parameters in the respective
<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser"><B>ConfigFileParser.BasicOptions</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>This is a helper class that maps the values of species-specific basic options to their value<DT><A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html#ConfigFileParser.BasicOptions()"><B>ConfigFileParser.BasicOptions()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.BasicOptions.html" title="class in broad.core.parser">ConfigFileParser.BasicOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser"><B>ConfigFileParser.Bowtie2Options</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>This is a helper class is a data structure that maps a bowtie2 flag to its value<DT><A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html#ConfigFileParser.Bowtie2Options()"><B>ConfigFileParser.Bowtie2Options()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser">ConfigFileParser.Bowtie2Options</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html#ConfigFileParser.Bowtie2Options(java.util.HashMap)"><B>ConfigFileParser.Bowtie2Options(HashMap&lt;String, String&gt;)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser">ConfigFileParser.Bowtie2Options</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html#ConfigFileParser.Bowtie2Options(java.lang.String, java.lang.String)"><B>ConfigFileParser.Bowtie2Options(String, String)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.Bowtie2Options.html" title="class in broad.core.parser">ConfigFileParser.Bowtie2Options</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser"><B>ConfigFileParser.DGEOptions</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>This is a helper class is a data structure that maps a DGE flag or option to its value or path<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#ConfigFileParser.DGEOptions(java.util.HashMap)"><B>ConfigFileParser.DGEOptions(HashMap&lt;String, String&gt;)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#ConfigFileParser.DGEOptions()"><B>ConfigFileParser.DGEOptions()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html#ConfigFileParser.DGEOptions(java.lang.String, java.lang.String)"><B>ConfigFileParser.DGEOptions(String, String)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.DGEOptions.html" title="class in broad.core.parser">ConfigFileParser.DGEOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser"><B>ConfigFileParser.FragmentSizeDistributionOptions</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>&nbsp;<DT><A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html#ConfigFileParser.FragmentSizeDistributionOptions()"><B>ConfigFileParser.FragmentSizeDistributionOptions()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.FragmentSizeDistributionOptions.html" title="class in broad.core.parser">ConfigFileParser.FragmentSizeDistributionOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.MergeSamples.html" title="class in broad.core.parser"><B>ConfigFileParser.MergeSamples</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>Sets of samples to merge into one bam file (leaving separate bam files too)<DT><A HREF="../broad/core/parser/ConfigFileParser.MergeSamples.html#ConfigFileParser.MergeSamples()"><B>ConfigFileParser.MergeSamples()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.MergeSamples.html" title="class in broad.core.parser">ConfigFileParser.MergeSamples</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html" title="class in broad.core.parser"><B>ConfigFileParser.NovoalignOptions</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>This is a helper class is a data structure that maps a novoalign flag to its value<DT><A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html#ConfigFileParser.NovoalignOptions()"><B>ConfigFileParser.NovoalignOptions()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html" title="class in broad.core.parser">ConfigFileParser.NovoalignOptions</A>
<DD>Instantiate object
<DT><A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html#ConfigFileParser.NovoalignOptions(java.util.HashMap)"><B>ConfigFileParser.NovoalignOptions(HashMap&lt;String, String&gt;)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html" title="class in broad.core.parser">ConfigFileParser.NovoalignOptions</A>
<DD>Instantiate object with existing set of options
<DT><A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html#ConfigFileParser.NovoalignOptions(java.lang.String, java.lang.String)"><B>ConfigFileParser.NovoalignOptions(String, String)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.NovoalignOptions.html" title="class in broad.core.parser">ConfigFileParser.NovoalignOptions</A>
<DD>Instantiate object with one existing option
<DT><A HREF="../broad/core/parser/ConfigFileParser.RNAClass.html" title="class in broad.core.parser"><B>ConfigFileParser.RNAClass</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>This is a helper class is a data structure mapping an RNA class to its fasta file path<DT><A HREF="../broad/core/parser/ConfigFileParser.RNAClass.html#ConfigFileParser.RNAClass()"><B>ConfigFileParser.RNAClass()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.RNAClass.html" title="class in broad.core.parser">ConfigFileParser.RNAClass</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.RNAClass.html#ConfigFileParser.RNAClass(java.util.HashMap)"><B>ConfigFileParser.RNAClass(HashMap&lt;String, String&gt;)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.RNAClass.html" title="class in broad.core.parser">ConfigFileParser.RNAClass</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.RNAClass.html#ConfigFileParser.RNAClass(java.lang.String, java.lang.String)"><B>ConfigFileParser.RNAClass(String, String)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.RNAClass.html" title="class in broad.core.parser">ConfigFileParser.RNAClass</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser"><B>ConfigFileParser.TophatOptions</B></A> - Class in <A HREF="../broad/core/parser/package-summary.html">broad.core.parser</A><DD>This is a helper class is a data structure that maps a tophat flag to its value<DT><A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html#ConfigFileParser.TophatOptions()"><B>ConfigFileParser.TophatOptions()</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser">ConfigFileParser.TophatOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html#ConfigFileParser.TophatOptions(java.util.HashMap)"><B>ConfigFileParser.TophatOptions(HashMap&lt;String, String&gt;)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser">ConfigFileParser.TophatOptions</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html#ConfigFileParser.TophatOptions(java.lang.String, java.lang.String)"><B>ConfigFileParser.TophatOptions(String, String)</B></A> - 
Constructor for class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.TophatOptions.html" title="class in broad.core.parser">ConfigFileParser.TophatOptions</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/statistics/ConnectDisconnectedTranscripts.html" title="class in nextgen.core.scripture.statistics"><B>ConnectDisconnectedTranscripts</B></A> - Class in <A HREF="../nextgen/core/scripture/statistics/package-summary.html">nextgen.core.scripture.statistics</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/statistics/ConnectDisconnectedTranscripts.html#ConnectDisconnectedTranscripts(java.lang.String, java.lang.String, java.io.File, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead, int)"><B>ConnectDisconnectedTranscripts(String, String, File, AbstractPairedEndAlignment.TranscriptionRead, int)</B></A> - 
Constructor for class nextgen.core.scripture.statistics.<A HREF="../nextgen/core/scripture/statistics/ConnectDisconnectedTranscripts.html" title="class in nextgen.core.scripture.statistics">ConnectDisconnectedTranscripts</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html#connectVertexToGraph(nextgen.core.annotation.Annotation)"><B>connectVertexToGraph(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">ChromosomeTranscriptGraph</A>
<DD>Adds the annotation as a vertex to the positive/negative graph depending on the 
 orientation of the vertex
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#connectVertexToGraph(nextgen.core.annotation.Annotation)"><B>connectVertexToGraph(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>Adds an exon to a graph
<DT><A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy"><B>ConservationUtils</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/ConservationUtils.html#ConservationUtils()"><B>ConservationUtils()</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy"><B>ConservedPWMScanner</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html#ConservedPWMScanner(java.lang.String)"><B>ConservedPWMScanner(String)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy">ConservedPWMScanner</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservedPWMScanner.html#ConservedPWMScanner(java.util.List)"><B>ConservedPWMScanner(List&lt;PositionWeightMatrix&gt;)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/ConservedPWMScanner.html" title="class in broad.core.siphy">ConservedPWMScanner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#constituentIntrons()"><B>constituentIntrons()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>Constructs an artificial gene where the constituent introns (regions that are intronic in all isoforms) are the exons of the gene.
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#constituentIsoform()"><B>constituentIsoform()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#constructWithCopy(double[][])"><B>constructWithCopy(double[][])</B></A> - 
Static method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Construct a matrix from a copy of a 2-D array.
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#containChr(java.lang.String)"><B>containChr(String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#containedWithin(broad.pda.datastructures.Alignments)"><B>containedWithin(Alignments)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#containedWithin(broad.pda.datastructures.Alignments, broad.pda.datastructures.Alignments)"><B>containedWithin(Alignments, Alignments)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#contains(broad.core.annotation.LightweightGenomicAnnotation)"><B>contains(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html#contains(broad.core.annotation.LightweightGenomicAnnotation)"><B>contains(LightweightGenomicAnnotation)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#contains(broad.core.annotation.LightweightGenomicAnnotation)"><B>contains(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#contains(java.lang.String)"><B>contains(String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#contains(broad.core.annotation.LightweightGenomicAnnotation)"><B>contains(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#contains(nextgen.core.annotation.Annotation)"><B>contains(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#contains(java.lang.String)"><B>contains(String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#contains(nextgen.core.annotation.Annotation)"><B>contains(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#contains(nextgen.core.annotation.Annotation)"><B>contains(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#contains(nextgen.core.annotation.CompoundInterval)"><B>contains(CompoundInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/SingleInterval.html#contains(nextgen.core.annotation.SingleInterval)"><B>contains(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation">SingleInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#contains(nextgen.core.annotation.Annotation)"><B>contains(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#containsAnnotation(int, java.lang.String)"><B>containsAnnotation(int, String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#containsAnnotation(java.lang.String)"><B>containsAnnotation(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#containsAttribute(java.lang.String)"><B>containsAttribute(String)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#containsColumn(java.lang.String)"><B>containsColumn(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#containsDataFor(java.lang.String)"><B>containsDataFor(String)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>Returns true if this coordinate space contains data for the specified chromosome
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#containsExactInterval(nextgen.core.annotation.SingleInterval)"><B>containsExactInterval(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#containsInterval(nextgen.core.annotation.SingleInterval)"><B>containsInterval(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/AlignmentModel.html#containsReference(java.lang.String)"><B>containsReference(String)</B></A> - 
Method in class nextgen.core.model.<A HREF="../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#containsRow(java.lang.String)"><B>containsRow(String)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#CONTIG_TYPE"><B>CONTIG_TYPE</B></A> - 
Static variable in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html#contiguousEndSubSequenceOverMin(double[], double)"><B>contiguousEndSubSequenceOverMin(double[], double)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html" title="class in broad.core.annotation">MaximumContiguousSubsequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html#contiguousStartSubSequenceOverMin(double[], double)"><B>contiguousStartSubSequenceOverMin(double[], double)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/MaximumContiguousSubsequence.html" title="class in broad.core.annotation">MaximumContiguousSubsequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#CONTINUOUS_EXTENSION"><B>CONTINUOUS_EXTENSION</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation"><B>ContinuousDataAlignmentModel</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModel)"><B>ContinuousDataAlignmentModel(AlignmentDataModel)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, int, int)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats, int, int)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats, File[], int)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats, File[], int, int, AlignmentDataModelStats, AlignmentDataModelStats)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats, boolean)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats, File[], int, int, AlignmentDataModelStats, AlignmentDataModelStats, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.util.Map, int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats, Map&lt;String, Integer&gt;, int, int, AlignmentDataModelStats, AlignmentDataModelStats)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.util.Map, int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats, boolean)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats, Map&lt;String, Integer&gt;, int, int, AlignmentDataModelStats, AlignmentDataModelStats, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.util.Map, int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats, int, boolean)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats, Map&lt;String, Integer&gt;, int, int, AlignmentDataModelStats, AlignmentDataModelStats, int, boolean)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int, broad.pda.seq.segmentation.AlignmentDataModelStats)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats, File[], int, int, AlignmentDataModelStats)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int)"><B>ContinuousDataAlignmentModel(AlignmentDataModelStats, File[], int, int)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>Constructor to build the data alignment model
<DT><A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html#CONTROL"><B>CONTROL</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html" title="class in broad.pda.seq.rap">BuildRatioNullDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html#CONTROL"><B>CONTROL</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap">GenomeScoringProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#CONTROL"><B>CONTROL</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html#CONTROL"><B>CONTROL</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html" title="class in broad.pda.seq.rap">RatioPermutationPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/programs/ConvertBamToPairedEnd.html" title="class in nextgen.core.programs"><B>ConvertBamToPairedEnd</B></A> - Class in <A HREF="../nextgen/core/programs/package-summary.html">nextgen.core.programs</A><DD>&nbsp;<DT><A HREF="../nextgen/core/programs/ConvertBamToPairedEnd.html#ConvertBamToPairedEnd()"><B>ConvertBamToPairedEnd()</B></A> - 
Constructor for class nextgen.core.programs.<A HREF="../nextgen/core/programs/ConvertBamToPairedEnd.html" title="class in nextgen.core.programs">ConvertBamToPairedEnd</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#converted"><B>converted</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#convertInputToPairedEnd()"><B>convertInputToPairedEnd()</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>Convert the bamFile provided in the constructor to paired end format.
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#convertInputToPairedEnd(nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>convertInputToPairedEnd(AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>Convert the bamFile provided in the constructor to paired end format.
<DT><A HREF="../broad/pda/differentialExpression/ConvertToDiscreteMatrix.html" title="class in broad.pda.differentialExpression"><B>ConvertToDiscreteMatrix</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/ConvertToDiscreteMatrix.html#ConvertToDiscreteMatrix(broad.core.datastructures.MatrixWithHeaders, java.lang.String, double)"><B>ConvertToDiscreteMatrix(MatrixWithHeaders, String, double)</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/ConvertToDiscreteMatrix.html" title="class in broad.pda.differentialExpression">ConvertToDiscreteMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#convertToFasta(java.lang.String)"><B>convertToFasta(String)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#convertToNumberedFasta(java.lang.String)"><B>convertToNumberedFasta(String)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html#convertToPairedEndFragment(net.sf.samtools.SAMRecord)"><B>convertToPairedEndFragment(SAMRecord)</B></A> - 
Static method in class broad.pda.samtools.<A HREF="../broad/pda/samtools/SAMPairedEndFileReader.html" title="class in broad.pda.samtools">SAMPairedEndFileReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem"><B>CoordinateSpace</B></A> - Interface in <A HREF="../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A><DD>This interface represents the coordinate space in which we analyze the models<DT><A HREF="../nextgen/core/writers/WigWriter.html#coordinateToWigPosition(int)"><B>coordinateToWigPosition(int)</B></A> - 
Static method in class nextgen.core.writers.<A HREF="../nextgen/core/writers/WigWriter.html" title="class in nextgen.core.writers">WigWriter</A>
<DD>Get the one based wig format position for a zero based coordinate
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#copy()"><B>copy()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#copy()"><B>copy()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#copy()"><B>copy()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#copy()"><B>copy()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#copy()"><B>copy()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModelParameters.html#copy()"><B>copy()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModelParameters.html" title="class in broad.core.siphy">EvolutionaryModelParameters</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#copy()"><B>copy()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#copy()"><B>copy()</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Alignment.html#copy(jaligner.Alignment)"><B>copy(Alignment)</B></A> - 
Static method in class jaligner.<A HREF="../jaligner/Alignment.html" title="class in jaligner">Alignment</A>
<DD>Copies an alignment to another alignment.
<DT><A HREF="../Jama/Matrix.html#copy()"><B>copy()</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Make a deep copy of a matrix
<DT><A HREF="../nextgen/core/annotation/Annotation.html#copy()"><B>copy()</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#copy()"><B>copy()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#copy()"><B>copy()</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#copy()"><B>copy()</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/CorrelationAgilentNanostring.html" title="class in broad.pda.geneexpression.agilent"><B>CorrelationAgilentNanostring</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/CorrelationAgilentNanostring.html#CorrelationAgilentNanostring(broad.core.datastructures.MatrixWithHeaders, broad.core.datastructures.MatrixWithHeaders, java.io.File, java.io.File, java.lang.String, java.io.File)"><B>CorrelationAgilentNanostring(MatrixWithHeaders, MatrixWithHeaders, File, File, String, File)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/CorrelationAgilentNanostring.html" title="class in broad.pda.geneexpression.agilent">CorrelationAgilentNanostring</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/CorrelationBetweenReplicates.html" title="class in broad.pda.geneexpression.agilent"><B>CorrelationBetweenReplicates</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/package-summary.html">broad.pda.geneexpression.agilent</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/CorrelationBetweenReplicates.html#CorrelationBetweenReplicates(broad.core.datastructures.MatrixWithHeaders, java.io.File, java.lang.String)"><B>CorrelationBetweenReplicates(MatrixWithHeaders, File, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/CorrelationBetweenReplicates.html" title="class in broad.pda.geneexpression.agilent">CorrelationBetweenReplicates</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/CorrelationMatrix.html" title="class in broad.pda.geneexpression.clustering"><B>CorrelationMatrix</B></A> - Class in <A HREF="../broad/pda/geneexpression/clustering/package-summary.html">broad.pda.geneexpression.clustering</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/clustering/CorrelationMatrix.html#CorrelationMatrix(broad.core.datastructures.MatrixWithHeaders, java.lang.String)"><B>CorrelationMatrix(MatrixWithHeaders, String)</B></A> - 
Constructor for class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/CorrelationMatrix.html" title="class in broad.pda.geneexpression.clustering">CorrelationMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#count(nextgen.core.annotation.Annotation, int)"><B>count(Annotation, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#count(nextgen.core.annotation.Annotation)"><B>count(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html#countAlignments(java.lang.String, java.lang.String, java.lang.String, boolean)"><B>countAlignments(String, String, String, boolean)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html" title="class in broad.core.util">PipelineAutomator.AlignmentUtils</A>
<DD>Count total number of mapped reads in a sam or bam file
<DT><A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html#countAlignments(java.lang.String, java.lang.String, java.lang.String, boolean, boolean)"><B>countAlignments(String, String, String, boolean, boolean)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html" title="class in broad.core.util">PipelineAutomator.AlignmentUtils</A>
<DD>Count total number of mapped or unmapped reads in a sam file
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.html#countLessThanLambda()"><B>countLessThanLambda()</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.html" title="class in broad.pda.seq.rap">SkellamScore</A>
<DD>Returns true if the signal count is less than the chromosome lambda
<DT><A HREF="../broad/pda/geneexpression/CountNums.html" title="class in broad.pda.geneexpression"><B>CountNums</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/CountNums.html#CountNums(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.lang.String, double)"><B>CountNums(MatrixWithHeaders, Collection&lt;String&gt;, String, double)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/CountNums.html" title="class in broad.pda.geneexpression">CountNums</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CountReads.html" title="class in broad.pda.seq.rap"><B>CountReads</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/CountReads.html#CountReads()"><B>CountReads()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CountReads.html" title="class in broad.pda.seq.rap">CountReads</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#countReads(nextgen.core.alignment.Alignment)"><B>countReads(Alignment)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score"><B>CountScore</B></A> - Class in <A HREF="../nextgen/core/model/score/package-summary.html">nextgen.core.model.score</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/score/CountScore.html#CountScore(nextgen.core.annotation.Annotation)"><B>CountScore(Annotation)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#CountScore(nextgen.core.model.score.CountScore)"><B>CountScore(CountScore)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#CountScore(nextgen.core.model.score.CountScore, nextgen.core.annotation.Annotation, double)"><B>CountScore(CountScore, Annotation, double)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#CountScore(nextgen.core.annotation.AnnotationCollection, nextgen.core.annotation.Annotation)"><B>CountScore(AnnotationCollection&lt;? extends Annotation&gt;, Annotation)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#CountScore(nextgen.core.annotation.AnnotationCollection, nextgen.core.annotation.Annotation, double)"><B>CountScore(AnnotationCollection&lt;? extends Annotation&gt;, Annotation, double)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.html#CountScore(nextgen.core.annotation.Annotation, double, double, double)"><B>CountScore(Annotation, double, double, double)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.html" title="class in nextgen.core.model.score">CountScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.Processor.html" title="class in nextgen.core.model.score"><B>CountScore.Processor</B></A> - Class in <A HREF="../nextgen/core/model/score/package-summary.html">nextgen.core.model.score</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/score/CountScore.Processor.html#CountScore.Processor(nextgen.core.annotation.AnnotationCollection)"><B>CountScore.Processor(AnnotationCollection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.Processor.html" title="class in nextgen.core.model.score">CountScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.Processor.html#CountScore.Processor(nextgen.core.annotation.AnnotationCollection, boolean)"><B>CountScore.Processor(AnnotationCollection&lt;? extends Annotation&gt;, boolean)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.Processor.html" title="class in nextgen.core.model.score">CountScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/Sequence.html#countSoftMaskedBases()"><B>countSoftMaskedBases()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/Sequence.html" title="class in broad.core.sequence">Sequence</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#countWithinExon(nextgen.core.annotation.Annotation, int)"><B>countWithinExon(Annotation, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#covariance(java.util.List, java.util.List, double, double)"><B>covariance(List&lt;? extends Number&gt;, List&lt;? extends Number&gt;, double, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#covariance(java.util.List, java.util.List)"><B>covariance(List&lt;? extends Number&gt;, List&lt;? extends Number&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CoverageScore.html" title="class in nextgen.core.model.score"><B>CoverageScore</B></A> - Class in <A HREF="../nextgen/core/model/score/package-summary.html">nextgen.core.model.score</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/score/CoverageScore.html#CoverageScore(nextgen.core.annotation.Annotation)"><B>CoverageScore(Annotation)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CoverageScore.html" title="class in nextgen.core.model.score">CoverageScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CoverageScore.html#CoverageScore(nextgen.core.annotation.AnnotationCollection, nextgen.core.annotation.Annotation)"><B>CoverageScore(AnnotationCollection&lt;? extends Annotation&gt;, Annotation)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CoverageScore.html" title="class in nextgen.core.model.score">CoverageScore</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CoverageScore.Processor.html" title="class in nextgen.core.model.score"><B>CoverageScore.Processor</B></A> - Class in <A HREF="../nextgen/core/model/score/package-summary.html">nextgen.core.model.score</A><DD>&nbsp;<DT><A HREF="../nextgen/core/model/score/CoverageScore.Processor.html#CoverageScore.Processor(nextgen.core.annotation.AnnotationCollection)"><B>CoverageScore.Processor(AnnotationCollection&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CoverageScore.Processor.html" title="class in nextgen.core.model.score">CoverageScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactory.html" title="interface in broad.core.annotation">AnnotationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactory.html#create(broad.core.annotation.GenomicAnnotation)"><B>create(GenomicAnnotation)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactory.html" title="interface in broad.core.annotation">AnnotationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Method in interface broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactory.html" title="interface in broad.core.annotation">AnnotationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BasicGenomicFactory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BasicGenomicFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BasicGenomicFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BasicGenomicFactory.html#create(broad.core.annotation.GenomicAnnotation)"><B>create(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BasicGenomicFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BasicGenomicFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BasicGenomicFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BasicGenomicFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BasicGenomicFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDFactory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BEDFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDFactory.html#create(broad.core.annotation.GenomicAnnotation)"><B>create(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BEDFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BEDFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDGraphFactory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDGraphFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BEDGraphFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDGraphFactory.html#create(broad.core.annotation.GenomicAnnotation)"><B>create(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDGraphFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BEDGraphFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDGraphFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.BEDGraphFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.BEDGraphFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.GFFFactory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.GFFFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.GFFFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.GFFFactory.html#create(broad.core.annotation.GenomicAnnotation)"><B>create(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.GFFFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.GFFFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.GFFFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.GFFFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.GFFFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.RepeatMaskerAnnotationFactory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.RepeatMaskerAnnotationFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.RepeatMaskerAnnotationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.RepeatMaskerAnnotationFactory.html#create(broad.core.annotation.GenomicAnnotation)"><B>create(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.RepeatMaskerAnnotationFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.RepeatMaskerAnnotationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.RepeatMaskerAnnotationFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.RepeatMaskerAnnotationFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.RepeatMaskerAnnotationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.ShortBEDFactory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.ShortBEDFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.ShortBEDFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.ShortBEDFactory.html#create(broad.core.annotation.GenomicAnnotation)"><B>create(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.ShortBEDFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.ShortBEDFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.ShortBEDFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.ShortBEDFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.ShortBEDFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.WIGFactory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.WIGFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.WIGFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.WIGFactory.html#create(broad.core.annotation.GenomicAnnotation)"><B>create(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.WIGFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.WIGFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationFactoryFactory.WIGFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationFactoryFactory.WIGFactory.html" title="class in broad.core.annotation">AnnotationFactoryFactory.WIGFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReaderFactory.html#create(java.lang.String, java.lang.String)"><B>create(String, String)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReaderFactory.html" title="class in broad.core.annotation">AnnotationReaderFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReaderFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReaderFactory.html" title="class in broad.core.annotation">AnnotationReaderFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReaderFactory.html#create(java.lang.String, java.lang.String, double)"><B>create(String, String, double)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReaderFactory.html" title="class in broad.core.annotation">AnnotationReaderFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReaderFactory.html#create(java.lang.String, java.lang.String, java.lang.String)"><B>create(String, String, String)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReaderFactory.html" title="class in broad.core.annotation">AnnotationReaderFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html#create(java.lang.String, java.lang.String)"><B>create(String, String)</B></A> - 
Static method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html" title="class in broad.core.multiplealignment">MultipleAlignmentFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html#create(java.io.InputStream, java.lang.String)"><B>create(InputStream, String)</B></A> - 
Static method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html" title="class in broad.core.multiplealignment">MultipleAlignmentFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Static method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignmentFactory.html" title="class in broad.core.multiplealignment">MultipleAlignmentFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignmentIOFactory.html#create(java.lang.String)"><B>create(String)</B></A> - 
Static method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignmentIOFactory.html" title="class in broad.core.multiplealignment">MultipleAlignmentIOFactory</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AnnotationFactory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AnnotationFactory.html" title="interface in nextgen.core.annotation">AnnotationFactory</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.Factory.html#create(java.lang.String[])"><B>create(String[])</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.Factory.html" title="class in nextgen.core.annotation">BasicAnnotation.Factory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#createAnnotation(broad.core.annotation.GenomicAnnotation)"><B>createAnnotation(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicAnnotationReader.html#createAnnotation(broad.core.annotation.GenomicAnnotation)"><B>createAnnotation(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicAnnotationReader.html" title="class in broad.core.annotation">BasicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicAnnotationReader.html#createAnnotation(nextgen.core.annotation.Annotation)"><B>createAnnotation(Annotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicAnnotationReader.html" title="class in broad.core.annotation">BasicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#createAnnotation(broad.core.annotation.GenomicAnnotation)"><B>createAnnotation(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BEDGraphReader.html#createAnnotation(broad.core.annotation.GenomicAnnotation)"><B>createAnnotation(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BEDGraphReader.html" title="class in broad.core.annotation">BEDGraphReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BEDReader.html#createAnnotation(broad.core.annotation.GenomicAnnotation)"><B>createAnnotation(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BEDReader.html" title="class in broad.core.annotation">BEDReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFFReader.html#createAnnotation(broad.core.annotation.GenomicAnnotation)"><B>createAnnotation(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFFReader.html" title="class in broad.core.annotation">GFFReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html#createAnnotation(broad.core.annotation.GenomicAnnotation)"><B>createAnnotation(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html" title="class in broad.core.annotation">RepeatMaskerAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBEDReader.html#createAnnotation(broad.core.annotation.GenomicAnnotation)"><B>createAnnotation(GenomicAnnotation)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBEDReader.html" title="class in broad.core.annotation">ShortBEDReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/AlignmentsReader.html#createAnnotation(broad.core.annotation.GenomicAnnotation)"><B>createAnnotation(GenomicAnnotation)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/AlignmentsReader.html" title="class in broad.pda.datastructures">AlignmentsReader</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityBinaryIO.html#createAsciiFromBinary(java.lang.String, int)"><B>createAsciiFromBinary(String, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityBinaryIO.html" title="class in broad.core.sequence">FastaQualityBinaryIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityBinaryIO.html#createBinaryFromAscii(java.io.InputStream, java.lang.String)"><B>createBinaryFromAscii(InputStream, String)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityBinaryIO.html" title="class in broad.core.sequence">FastaQualityBinaryIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.html#createDefaultTwoStateChain(double, double, broad.core.siphy.EvolutionaryModel)"><B>createDefaultTwoStateChain(double, double, EvolutionaryModel)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.html" title="class in broad.core.siphy">PiHMM</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#createEdge(broad.pda.datastructures.Alignments, nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, java.lang.Integer, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT.EdgeSourceType)"><B>createEdge(Alignments, Annotation, Annotation, Integer, ChromosomeWithBubblesJGraphT.EdgeSourceType)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Creates an edge of type between v1 and v2 for connection alignment with weight
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory.html#createEdge(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>createEdge(Annotation, Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntryFactory.html#createEntry(java.lang.String[])"><B>createEntry(String[])</B></A> - 
Static method in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntryFactory.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntryFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#createFromUCSC(java.lang.String)"><B>createFromUCSC(String)</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#createFromUCSC(java.lang.String)"><B>createFromUCSC(String)</B></A> - 
Static method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#createIndex(java.lang.String)"><B>createIndex(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityIndex.html#createIndexFromQualityFile(java.lang.String, long)"><B>createIndexFromQualityFile(String, long)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityIndex.html" title="class in broad.core.sequence">FastaQualityIndex</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#createIsoPWM(double[], java.lang.String)"><B>createIsoPWM(double[], String)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>Creates a PWM of similar dimension to this PWM but with all columns set to the given vector,
 usually a neutral mutation vector.
<DT><A HREF="../broad/pda/capture/designer/PcrTailDesigner.html#createPrimers()"><B>createPrimers()</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/PcrTailDesigner.html" title="class in broad.pda.capture.designer">PcrTailDesigner</A>
<DD>Create the final full primer set
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html#createSequenceFromRawData(java.lang.String[])"><B>createSequenceFromRawData(String[])</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.MAFMultipleAlignmentBlock.html" title="class in broad.core.multiplealignment">MAFAlignment.MAFMultipleAlignmentBlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.html#createTwoStateChain(double, double, broad.core.siphy.EvolutionaryModel, Jama.Matrix)"><B>createTwoStateChain(double, double, EvolutionaryModel, Matrix)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.html" title="class in broad.core.siphy">PiHMM</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFIO.html#createUnloadedAlignment(java.lang.String)"><B>createUnloadedAlignment(String)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFIO.html" title="class in broad.core.multiplealignment">MAFIO</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.html#CRISPR_TARGET_MOTIF"><B>CRISPR_TARGET_MOTIF</B></A> - 
Static variable in class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.html" title="class in nextgen.editing">CRISPRDesigner</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.html" title="class in nextgen.editing"><B>CRISPRDesigner</B></A> - Class in <A HREF="../nextgen/editing/package-summary.html">nextgen.editing</A><DD>&nbsp;<DT><A HREF="../nextgen/editing/CRISPRDesigner.html#CRISPRDesigner()"><B>CRISPRDesigner()</B></A> - 
Constructor for class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.html" title="class in nextgen.editing">CRISPRDesigner</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing"><B>CRISPRDesigner.CRISPRTarget</B></A> - Class in <A HREF="../nextgen/editing/package-summary.html">nextgen.editing</A><DD>&nbsp;<DT><A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html#CRISPRDesigner.CRISPRTarget(broad.core.sequence.SequenceRegion)"><B>CRISPRDesigner.CRISPRTarget(SequenceRegion)</B></A> - 
Constructor for class nextgen.editing.<A HREF="../nextgen/editing/CRISPRDesigner.CRISPRTarget.html" title="class in nextgen.editing">CRISPRDesigner.CRISPRTarget</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/CrossJunctionPCRDesigner.html" title="class in broad.core.primer3"><B>CrossJunctionPCRDesigner</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>&nbsp;<DT><A HREF="../broad/core/primer3/CrossJunctionPCRDesigner.html#CrossJunctionPCRDesigner()"><B>CrossJunctionPCRDesigner()</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/CrossJunctionPCRDesigner.html" title="class in broad.core.primer3">CrossJunctionPCRDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/DesignPCRTails.html#crossPrime(broad.core.primer3.PrimerPair, broad.core.primer3.PrimerPair)"><B>crossPrime(PrimerPair, PrimerPair)</B></A> - 
Method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/DesignPCRTails.html" title="class in broad.pda.capture.designer">DesignPCRTails</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/transcriptsNullModel.html#CrossRandModelsWithBed(java.lang.String, broad.pda.annotation.BEDFileParser, java.lang.String)"><B>CrossRandModelsWithBed(String, BEDFileParser, String)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/transcriptsNullModel.html" title="class in broad.pda.gene">transcriptsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/CrossSampleRawCountNormalization.html" title="class in nextgen.core.normalize"><B>CrossSampleRawCountNormalization</B></A> - Class in <A HREF="../nextgen/core/normalize/package-summary.html">nextgen.core.normalize</A><DD>&nbsp;<DT><A HREF="../nextgen/core/normalize/CrossSampleRawCountNormalization.html#CrossSampleRawCountNormalization(nextgen.core.model.AlignmentModel, nextgen.core.model.AlignmentModel)"><B>CrossSampleRawCountNormalization(AlignmentModel, AlignmentModel)</B></A> - 
Constructor for class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/CrossSampleRawCountNormalization.html" title="class in nextgen.core.normalize">CrossSampleRawCountNormalization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/normalize/CrossSampleTranscriptAverageNormalization.html" title="class in nextgen.core.normalize"><B>CrossSampleTranscriptAverageNormalization</B></A> - Class in <A HREF="../nextgen/core/normalize/package-summary.html">nextgen.core.normalize</A><DD>&nbsp;<DT><A HREF="../nextgen/core/normalize/CrossSampleTranscriptAverageNormalization.html#CrossSampleTranscriptAverageNormalization(nextgen.core.normalize.TranscriptAverageNormalization, nextgen.core.normalize.TranscriptAverageNormalization)"><B>CrossSampleTranscriptAverageNormalization(TranscriptAverageNormalization, TranscriptAverageNormalization)</B></A> - 
Constructor for class nextgen.core.normalize.<A HREF="../nextgen/core/normalize/CrossSampleTranscriptAverageNormalization.html" title="class in nextgen.core.normalize">CrossSampleTranscriptAverageNormalization</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html#CUTOFF"><B>CUTOFF</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html" title="class in broad.pda.seq.rap">RatioPermutationPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html#CUTOFF"><B>CUTOFF</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html" title="class in broad.pda.seq.rap">SkellamKLPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#CV(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><B>CV(Gene, IntervalTree&lt;Alignment&gt;)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-2.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-4.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-3.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-3.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
