<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:31 EDT 2013 -->
<TITLE>
P-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="P-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-15.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-17.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-16.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-16.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_P_"><!-- --></A><H2>
<B>P</B></H2>
<DL>
<DT><A HREF="../broad/pda/methylation/PadlockAnalysis.html" title="class in broad.pda.methylation"><B>PadlockAnalysis</B></A> - Class in <A HREF="../broad/pda/methylation/package-summary.html">broad.pda.methylation</A><DD>&nbsp;<DT><A HREF="../broad/pda/methylation/PadlockProbe.html" title="class in broad.pda.methylation"><B>PadlockProbe</B></A> - Class in <A HREF="../broad/pda/methylation/package-summary.html">broad.pda.methylation</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/PaGEJava.html" title="class in broad.pda.geneexpression"><B>PaGEJava</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/PaGEJava.html#PaGEJava(broad.core.datastructures.MatrixWithHeaders, java.util.Map, java.lang.String)"><B>PaGEJava(MatrixWithHeaders, Map&lt;String, ArrayList&gt;, String)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/PaGEJava.html" title="class in broad.pda.geneexpression">PaGEJava</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/PaGEJava.html#PaGEJava(java.io.File, java.io.File)"><B>PaGEJava(File, File)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/PaGEJava.html" title="class in broad.pda.geneexpression">PaGEJava</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures"><B>Pair</B></A>&lt;<A HREF="../broad/core/datastructures/Pair.html" title="type parameter in Pair">T1</A>&gt; - Class in <A HREF="../broad/core/datastructures/package-summary.html">broad.core.datastructures</A><DD>&nbsp;<DT><A HREF="../broad/core/datastructures/Pair.html#Pair(T1, T1)"><B>Pair(T1, T1)</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures">Pair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/Pair.html#Pair()"><B>Pair()</B></A> - 
Constructor for class broad.core.datastructures.<A HREF="../broad/core/datastructures/Pair.html" title="class in broad.core.datastructures">Pair</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/formats/Pair.html" title="class in jaligner.formats"><B>Pair</B></A> - Class in <A HREF="../jaligner/formats/package-summary.html">jaligner.formats</A><DD><a href="http://www.hgmp.mrc.ac.uk/Software/EMBOSS/Themes/AlignExamples/align.pair">Pair</a> format.<DT><A HREF="../jaligner/formats/Pair.html#Pair()"><B>Pair()</B></A> - 
Constructor for class jaligner.formats.<A HREF="../jaligner/formats/Pair.html" title="class in jaligner.formats">Pair</A>
<DD>Constructor
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#PAIRED_END"><B>PAIRED_END</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CountReads.html#PAIRED_END"><B>PAIRED_END</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CountReads.html" title="class in broad.pda.seq.rap">CountReads</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SlideAndCalculate.html#PAIRED_END"><B>PAIRED_END</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCalculate.html" title="class in broad.pda.seq.rap">SlideAndCalculate</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SlideAndCount.html#PAIRED_END"><B>PAIRED_END</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCount.html" title="class in broad.pda.seq.rap">SlideAndCount</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#PAIRED_END_EXTENSION"><B>PAIRED_END_EXTENSION</B></A> - 
Static variable in class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/PairedAndProperFilter.html" title="class in nextgen.core.readFilters"><B>PairedAndProperFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/PairedAndProperFilter.html#PairedAndProperFilter()"><B>PairedAndProperFilter()</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/PairedAndProperFilter.html" title="class in nextgen.core.readFilters">PairedAndProperFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/PairedEndAlignmentFactory.html" title="class in nextgen.core.alignment"><B>PairedEndAlignmentFactory</B></A> - Class in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>&nbsp;<DT><A HREF="../nextgen/core/alignment/PairedEndAlignmentFactory.html#PairedEndAlignmentFactory()"><B>PairedEndAlignmentFactory()</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/PairedEndAlignmentFactory.html" title="class in nextgen.core.alignment">PairedEndAlignmentFactory</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers"><B>PairedEndReader</B></A> - Class in <A HREF="../nextgen/core/readers/package-summary.html">nextgen.core.readers</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#PairedEndReader(java.io.File)"><B>PairedEndReader(File)</B></A> - 
Constructor for class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#PairedEndReader(java.io.File, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>PairedEndReader(File, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.html#PairedEndReader(java.io.File, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead, boolean)"><B>PairedEndReader(File, AbstractPairedEndAlignment.TranscriptionRead, boolean)</B></A> - 
Constructor for class nextgen.core.readers.<A HREF="../nextgen/core/readers/PairedEndReader.html" title="class in nextgen.core.readers">PairedEndReader</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readers/PairedEndReader.AlignmentType.html" title="enum in nextgen.core.readers"><B>PairedEndReader.AlignmentType</B></A> - Enum in <A HREF="../nextgen/core/readers/package-summary.html">nextgen.core.readers</A><DD>&nbsp;<DT><A HREF="../nextgen/core/debug/PairedEndSam.html" title="class in nextgen.core.debug"><B>PairedEndSam</B></A> - Class in <A HREF="../nextgen/core/debug/package-summary.html">nextgen.core.debug</A><DD>&nbsp;<DT><A HREF="../nextgen/core/debug/PairedEndSam.html#PairedEndSam(java.lang.String, broad.pda.datastructures.Alignments, nextgen.core.coordinatesystem.CoordinateSpace)"><B>PairedEndSam(String, Alignments, CoordinateSpace)</B></A> - 
Constructor for class nextgen.core.debug.<A HREF="../nextgen/core/debug/PairedEndSam.html" title="class in nextgen.core.debug">PairedEndSam</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/debug/PairedEndSam.html#PairedEndSam(java.util.Map, broad.pda.datastructures.Alignments)"><B>PairedEndSam(Map&lt;String, Collection&lt;Gene&gt;&gt;, Alignments)</B></A> - 
Constructor for class nextgen.core.debug.<A HREF="../nextgen/core/debug/PairedEndSam.html" title="class in nextgen.core.debug">PairedEndSam</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/PairedEndTDF.html" title="class in broad.pda.samtools"><B>PairedEndTDF</B></A> - Class in <A HREF="../broad/pda/samtools/package-summary.html">broad.pda.samtools</A><DD>&nbsp;<DT><A HREF="../broad/pda/samtools/PairedEndTDF.html#PairedEndTDF()"><B>PairedEndTDF()</B></A> - 
Constructor for class broad.pda.samtools.<A HREF="../broad/pda/samtools/PairedEndTDF.html" title="class in broad.pda.samtools">PairedEndTDF</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers"><B>PairedEndWriter</B></A> - Class in <A HREF="../nextgen/core/writers/package-summary.html">nextgen.core.writers</A><DD>&nbsp;<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#PairedEndWriter(java.io.File)"><B>PairedEndWriter(File)</B></A> - 
Constructor for class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/writers/PairedEndWriter.html#PairedEndWriter(java.io.File, java.lang.String)"><B>PairedEndWriter(File, String)</B></A> - 
Constructor for class nextgen.core.writers.<A HREF="../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers">PairedEndWriter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/alignment/PairedReadAlignment.html" title="class in nextgen.core.alignment"><B>PairedReadAlignment</B></A> - Class in <A HREF="../nextgen/core/alignment/package-summary.html">nextgen.core.alignment</A><DD>&nbsp;<DT><A HREF="../nextgen/core/alignment/PairedReadAlignment.html#PairedReadAlignment(nextgen.core.alignment.SingleEndAlignment, nextgen.core.alignment.SingleEndAlignment)"><B>PairedReadAlignment(SingleEndAlignment, SingleEndAlignment)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/PairedReadAlignment.html" title="class in nextgen.core.alignment">PairedReadAlignment</A>
<DD>Constructs a paired end alignment object from two alignments.
<DT><A HREF="../nextgen/core/alignment/PairedReadAlignment.html#PairedReadAlignment(nextgen.core.alignment.SingleEndAlignment, nextgen.core.alignment.SingleEndAlignment, nextgen.core.alignment.AbstractPairedEndAlignment.TranscriptionRead)"><B>PairedReadAlignment(SingleEndAlignment, SingleEndAlignment, AbstractPairedEndAlignment.TranscriptionRead)</B></A> - 
Constructor for class nextgen.core.alignment.<A HREF="../nextgen/core/alignment/PairedReadAlignment.html" title="class in nextgen.core.alignment">PairedReadAlignment</A>
<DD>Constructs a paired end alignment object from two alignments and provides the read that is in the direction of transcription
<DT><A HREF="../broad/core/math/Statistics.html#pairedTStat(java.util.List, java.util.List)"><B>pairedTStat(List&lt;Double&gt;, List&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#pairedTStat(double[], double[])"><B>pairedTStat(double[], double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html" title="class in nextgen.core.structure"><B>PairwiseWindowHybridization</B></A> - Class in <A HREF="../nextgen/core/structure/package-summary.html">nextgen.core.structure</A><DD>&nbsp;<DT><A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html#PairwiseWindowHybridization(nextgen.core.annotation.Gene, java.util.Map, int, int)"><B>PairwiseWindowHybridization(Gene, Map&lt;String, Sequence&gt;, int, int)</B></A> - 
Constructor for class nextgen.core.structure.<A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html" title="class in nextgen.core.structure">PairwiseWindowHybridization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html#PairwiseWindowHybridization(nextgen.core.annotation.Gene, java.lang.String, int, int)"><B>PairwiseWindowHybridization(Gene, String, int, int)</B></A> - 
Constructor for class nextgen.core.structure.<A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html" title="class in nextgen.core.structure">PairwiseWindowHybridization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html#PairwiseWindowHybridization(nextgen.core.annotation.Gene, java.lang.String, int, int, float, float, float, float)"><B>PairwiseWindowHybridization(Gene, String, int, int, float, float, float, float)</B></A> - 
Constructor for class nextgen.core.structure.<A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html" title="class in nextgen.core.structure">PairwiseWindowHybridization</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html#PairwiseWindowHybridization(nextgen.core.annotation.Gene, java.util.Map, int, int, float, float, float, float)"><B>PairwiseWindowHybridization(Gene, Map&lt;String, Sequence&gt;, int, int, float, float, float, float)</B></A> - 
Constructor for class nextgen.core.structure.<A HREF="../nextgen/core/structure/PairwiseWindowHybridization.html" title="class in nextgen.core.structure">PairwiseWindowHybridization</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#parse(java.lang.String, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(String, GenomicAnnotationFilter&lt;T&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#parse(java.io.BufferedReader, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(BufferedReader, GenomicAnnotationFilter&lt;T&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicAnnotationReader.html#parse(java.io.BufferedReader, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(BufferedReader, GenomicAnnotationFilter&lt;BasicGenomicAnnotation&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicAnnotationReader.html" title="class in broad.core.annotation">BasicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicAnnotationReader.html#parse(java.lang.String, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(String, GenomicAnnotationFilter&lt;BasicGenomicAnnotation&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicAnnotationReader.html" title="class in broad.core.annotation">BasicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#parse(java.lang.String, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(String, GenomicAnnotationFilter&lt;GenomicAnnotation&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#parse(java.io.BufferedReader, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(BufferedReader, GenomicAnnotationFilter&lt;GenomicAnnotation&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BEDGraphReader.html#parse(java.io.BufferedReader, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(BufferedReader, GenomicAnnotationFilter&lt;BEDGraph&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BEDGraphReader.html" title="class in broad.core.annotation">BEDGraphReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BEDGraphReader.html#parse(java.lang.String, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(String, GenomicAnnotationFilter&lt;BEDGraph&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BEDGraphReader.html" title="class in broad.core.annotation">BEDGraphReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BEDReader.html#parse(java.io.BufferedReader, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(BufferedReader, GenomicAnnotationFilter&lt;BED&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BEDReader.html" title="class in broad.core.annotation">BEDReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BEDReader.html#parse(java.lang.String, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(String, GenomicAnnotationFilter&lt;BED&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BEDReader.html" title="class in broad.core.annotation">BEDReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFFReader.html#parse(java.io.BufferedReader, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(BufferedReader, GenomicAnnotationFilter&lt;GFF&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFFReader.html" title="class in broad.core.annotation">GFFReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/GFFReader.html#parse(java.lang.String, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(String, GenomicAnnotationFilter&lt;GFF&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/GFFReader.html" title="class in broad.core.annotation">GFFReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html#parse(java.io.BufferedReader, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(BufferedReader, GenomicAnnotationFilter&lt;RepeatMaskerAnnotation&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html" title="class in broad.core.annotation">RepeatMaskerAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html#parse(java.lang.String, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(String, GenomicAnnotationFilter&lt;RepeatMaskerAnnotation&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/RepeatMaskerAnnotationReader.html" title="class in broad.core.annotation">RepeatMaskerAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBEDReader.html#parse(java.io.BufferedReader, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(BufferedReader, GenomicAnnotationFilter&lt;ShortBED&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBEDReader.html" title="class in broad.core.annotation">ShortBEDReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/ShortBEDReader.html#parse(java.lang.String, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(String, GenomicAnnotationFilter&lt;ShortBED&gt;, AnnotationHandler)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/ShortBEDReader.html" title="class in broad.core.annotation">ShortBEDReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/CommandLineParser.html#parse(java.lang.String[])"><B>parse(String[])</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/CommandLineParser.html" title="class in broad.core.parser">CommandLineParser</A>
<DD>Parse command arguments
 If command line is not in proper form, prints help menu and exits
 If a required argument is missing, prints help menu and exits
<DT><A HREF="../broad/core/parser/StringParser.html#parse(java.lang.String)"><B>parse(String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Parses the string around whitespace and stores tokens
<DT><A HREF="../broad/core/parser/StringParser.html#parse(java.lang.String, java.lang.String)"><B>parse(String, String)</B></A> - 
Method in class broad.core.parser.<A HREF="../broad/core/parser/StringParser.html" title="class in broad.core.parser">StringParser</A>
<DD>Parses the string around the specified delimiter and stores tokens
<DT><A HREF="../broad/core/sequence/FastaParser.html#parse(java.io.InputStream, broad.core.sequence.FastaHandler)"><B>parse(InputStream, FastaHandler)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaParser.html" title="class in broad.core.sequence">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/FastaQualityParser.html#parse(java.io.InputStream, broad.core.sequence.FastaQualityHandler)"><B>parse(InputStream, FastaQualityHandler)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/FastaQualityParser.html" title="class in broad.core.sequence">FastaQualityParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#parse(java.io.File, int, java.lang.String)"><B>parse(File, int, String)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/capture/designer/ComputeMIRScore.html#parse(java.io.File, boolean)"><B>parse(File, boolean)</B></A> - 
Static method in class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/ComputeMIRScore.html" title="class in broad.pda.capture.designer">ComputeMIRScore</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/AlignmentsReader.html#parse(java.io.BufferedReader, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(BufferedReader, GenomicAnnotationFilter&lt;Alignments&gt;, AnnotationHandler)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/AlignmentsReader.html" title="class in broad.pda.datastructures">AlignmentsReader</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/datastructures/AlignmentsReader.html#parse(java.lang.String, broad.core.annotation.GenomicAnnotationFilter, broad.core.annotation.AnnotationHandler)"><B>parse(String, GenomicAnnotationFilter&lt;Alignments&gt;, AnnotationHandler)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/AlignmentsReader.html" title="class in broad.pda.datastructures">AlignmentsReader</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentToGCT.html#parse(java.io.File)"><B>parse(File)</B></A> - 
Method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentToGCT.html" title="class in broad.pda.geneexpression.agilent">AgilentToGCT</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/fastq/FastqParser.html#parse(java.io.File)"><B>parse(File)</B></A> - 
Method in class broad.pda.seq.fastq.<A HREF="../broad/pda/seq/fastq/FastqParser.html" title="class in broad.pda.seq.fastq">FastqParser</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/util/SequenceParser.html#parse(java.lang.String)"><B>parse(String)</B></A> - 
Static method in class jaligner.util.<A HREF="../jaligner/util/SequenceParser.html" title="class in jaligner.util">SequenceParser</A>
<DD>Returns a parsed Sequence from a string.
<DT><A HREF="../jaligner/util/SequenceParser.html#parse(java.io.File)"><B>parse(File)</B></A> - 
Static method in class jaligner.util.<A HREF="../jaligner/util/SequenceParser.html" title="class in jaligner.util">SequenceParser</A>
<DD>Returns a Sequence parsed and loaded from a file
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseAgilentExperiment(java.io.File)"><B>parseAgilentExperiment(File)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseAgilentExperimentGlobalStats(java.io.File)"><B>parseAgilentExperimentGlobalStats(File)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/tools/PSLXToSAM.html#parseAndWrite(java.io.InputStream, java.io.Writer)"><B>parseAndWrite(InputStream, Writer)</B></A> - 
Method in class broad.pda.tools.<A HREF="../broad/pda/tools/PSLXToSAM.html" title="class in broad.pda.tools">PSLXToSAM</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/tools/PSLXToSAM.html#parseAndWrite(java.io.File, java.lang.String)"><B>parseAndWrite(File, String)</B></A> - 
Method in class broad.pda.tools.<A HREF="../broad/pda/tools/PSLXToSAM.html" title="class in broad.pda.tools">PSLXToSAM</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#parseCDT(java.io.File)"><B>parseCDT(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/GMTParser.html#parseCHIPFile(java.io.File)"><B>parseCHIPFile(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/GMTParser.html" title="class in broad.core.util">GMTParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#parseChipFile(java.io.File)"><B>parseChipFile(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/GMTParser.html#parseCHIPFileByName(java.io.File)"><B>parseCHIPFileByName(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/GMTParser.html" title="class in broad.core.util">GMTParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#parseCLS(java.io.File)"><B>parseCLS(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#parseCLSGroups(java.io.File)"><B>parseCLSGroups(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseControlSamples(java.lang.String)"><B>parseControlSamples(String)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/error/ParseException.html" title="class in broad.core.error"><B>ParseException</B></A> - Exception in <A HREF="../broad/core/error/package-summary.html">broad.core.error</A><DD>&nbsp;<DT><A HREF="../broad/core/error/ParseException.html#ParseException()"><B>ParseException()</B></A> - 
Constructor for exception broad.core.error.<A HREF="../broad/core/error/ParseException.html" title="class in broad.core.error">ParseException</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/error/ParseException.html#ParseException(java.lang.String)"><B>ParseException(String)</B></A> - 
Constructor for exception broad.core.error.<A HREF="../broad/core/error/ParseException.html" title="class in broad.core.error">ParseException</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/error/ParseException.html#ParseException(java.lang.String, java.lang.Throwable)"><B>ParseException(String, Throwable)</B></A> - 
Constructor for exception broad.core.error.<A HREF="../broad/core/error/ParseException.html" title="class in broad.core.error">ParseException</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/error/ParseException.html#ParseException(java.lang.Throwable)"><B>ParseException(Throwable)</B></A> - 
Constructor for exception broad.core.error.<A HREF="../broad/core/error/ParseException.html" title="class in broad.core.error">ParseException</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseExperimentInfoFile(java.io.File)"><B>parseExperimentInfoFile(File)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseExperimentInfoFileToGroup(java.io.File)"><B>parseExperimentInfoFileToGroup(File)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseExperimentInfoFileToGroups(java.io.File)"><B>parseExperimentInfoFileToGroups(File)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseExperimentInfoFileToName(java.io.File)"><B>parseExperimentInfoFileToName(File)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseExperimentInfoFileToName(java.io.File, boolean)"><B>parseExperimentInfoFileToName(File, boolean)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseExperimentInfoGroups(java.lang.String)"><B>parseExperimentInfoGroups(String)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html#parseExperimentInfoNonControls(java.lang.String)"><B>parseExperimentInfoNonControls(String)</B></A> - 
Static method in class broad.pda.geneexpression.agilent.<A HREF="../broad/pda/geneexpression/agilent/AgilentUtils.html" title="class in broad.pda.geneexpression.agilent">AgilentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util"><B>ParseGCTFile</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>&nbsp;<DT><A HREF="../broad/core/util/ParseGCTFile.html#ParseGCTFile()"><B>ParseGCTFile()</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#parseGeneList(java.io.File)"><B>parseGeneList(File)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/GMTParser.html#ParseGeneSetFile(java.io.File, int)"><B>ParseGeneSetFile(File, int)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/GMTParser.html" title="class in broad.core.util">GMTParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/GMTParser.html#ParseGMTFile(java.io.File, int)"><B>ParseGMTFile(File, int)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/GMTParser.html" title="class in broad.core.util">GMTParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/GMTParser.html#parseGMXFile(java.io.File, int)"><B>parseGMXFile(File, int)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/GMTParser.html" title="class in broad.core.util">GMTParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/functionalAnnotation.html#parseGoRawFiles(java.lang.String, java.lang.String, java.lang.String, java.lang.String, int, int, java.lang.String)"><B>parseGoRawFiles(String, String, String, String, int, int, String)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/functionalAnnotation.html" title="class in broad.pda.gene">functionalAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#parseMaskFiles(java.io.File[])"><B>parseMaskFiles(File[])</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/ParseGCTFile.html#parseNumericCls(java.io.File, java.util.List)"><B>parseNumericCls(File, List&lt;String&gt;)</B></A> - 
Static method in class broad.core.util.<A HREF="../broad/core/util/ParseGCTFile.html" title="class in broad.core.util">ParseGCTFile</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#parseRawPointRegions(java.lang.String[])"><B>parseRawPointRegions(String[])</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#parseRawRegions(java.lang.String[])"><B>parseRawRegions(String[])</B></A> - 
Static method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>Parses an array of cannonical region descriptors of the form chrN:start-end
<DT><A HREF="../broad/core/primer3/Primer3IO.html#parseReply(java.io.BufferedReader)"><B>parseReply(BufferedReader)</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3IO.html" title="class in broad.core.primer3">Primer3IO</A>
<DD>Updated for Primer3 upgrade version 2.0
<DT><A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html#passedQC()"><B>passedQC()</B></A> - 
Method in class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/ExpressionExperimentInfo.html" title="class in broad.pda.geneexpression">ExpressionExperimentInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.MinReadFilter.html#passes(broad.pda.datastructures.Alignments)"><B>passes(Alignments)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.MinReadFilter.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.MinReadFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.MinReadFilter.html#passes(org.broad.igv.sam.Alignment)"><B>passes(Alignment)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.MinReadFilter.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.MinReadFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.RegionReadFilter.html#passes(broad.pda.datastructures.Alignments)"><B>passes(Alignments)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.RegionReadFilter.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT.RegionReadFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#passes(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, double)"><B>passes(Gene, IntervalTree&lt;Alignment&gt;, int, double)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#passes(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, int)"><B>passes(Gene, IntervalTree&lt;Alignment&gt;, int, int)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#passes(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, int, double)"><B>passes(Gene, IntervalTree&lt;Alignment&gt;, int, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ReadFilter.html#passes(T)"><B>passes(T)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ReadFilter.html" title="interface in broad.pda.seq.segmentation">ReadFilter</A>
<DD>Evaluates whether or not a given read alignment passes the filtering criteria.
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html#passesCutoff(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int, double)"><B>passesCutoff(Annotation, IntervalTree&lt;Alignment&gt;, int, double)</B></A> - 
Method in interface broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html#passesCutoff(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree, int, double)"><B>passesCutoff(Annotation, IntervalTree&lt;Alignment&gt;, int, double)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/GenericAlignmentDataModel.html" title="class in broad.pda.seq.segmentation">GenericAlignmentDataModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph"><B>Path</B></A> - Class in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/graph/Path.html#Path(java.util.Collection, double)"><B>Path(Collection&lt;ChromosomeWithBubblesJGraphT.BubbleEdge&gt;, double)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#Path(java.util.Collection, org.apache.commons.collections15.Transformer, double)"><B>Path(Collection&lt;ChromosomeWithBubblesJGraphT.BubbleEdge&gt;, Transformer&lt;ChromosomeWithBubblesJGraphT.BubbleEdge, Number&gt;, double)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#Path(double)"><B>Path(double)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#Path()"><B>Path()</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#Path(nextgen.core.annotation.Gene, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT)"><B>Path(Gene, ChromosomeWithBubblesJGraphT)</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.StandardTransformer.html" title="class in broad.pda.seq.graph"><B>Path.StandardTransformer</B></A> - Class in <A HREF="../broad/pda/seq/graph/package-summary.html">broad.pda.seq.graph</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/graph/Path.StandardTransformer.html#Path.StandardTransformer()"><B>Path.StandardTransformer()</B></A> - 
Constructor for class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.StandardTransformer.html" title="class in broad.pda.seq.graph">Path.StandardTransformer</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html#pathToGene(org.jgrapht.GraphPath)"><B>pathToGene(GraphPath&lt;Annotation, OrientedChromosomeTranscriptGraph.TranscriptGraphEdge&gt;)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/ChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">ChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#pathToGene(org.jgrapht.GraphPath)"><B>pathToGene(GraphPath&lt;Annotation, OrientedChromosomeTranscriptGraph.TranscriptGraphEdge&gt;)</B></A> - 
Static method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PrimerPickingMode.html#PCR"><B>PCR</B></A> - 
Static variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3PrimerPickingMode.html" title="class in broad.core.primer3">Primer3PrimerPickingMode</A>
<DD>Picks paired primers for PCR.
<DT><A HREF="../broad/pda/capture/designer/PcrTailDesigner.html" title="class in broad.pda.capture.designer"><B>PcrTailDesigner</B></A> - Class in <A HREF="../broad/pda/capture/designer/package-summary.html">broad.pda.capture.designer</A><DD>&nbsp;<DT><A HREF="../broad/pda/capture/designer/PcrTailDesigner.html#PcrTailDesigner(int, int, int, int, int, int, int, java.lang.String)"><B>PcrTailDesigner(int, int, int, int, int, int, int, String)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/PcrTailDesigner.html" title="class in broad.pda.capture.designer">PcrTailDesigner</A>
<DD>Constructor when there are two overlapping primers
<DT><A HREF="../broad/pda/capture/designer/PcrTailDesigner.html#PcrTailDesigner(int, int, int, int, int, int, int, int, int, int, int, int, java.lang.String)"><B>PcrTailDesigner(int, int, int, int, int, int, int, int, int, int, int, int, String)</B></A> - 
Constructor for class broad.pda.capture.designer.<A HREF="../broad/pda/capture/designer/PcrTailDesigner.html" title="class in broad.pda.capture.designer">PcrTailDesigner</A>
<DD>Constructor when there are three overlapping primers
<DT><A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html#PCT_MASKED_ALLOWED"><B>PCT_MASKED_ALLOWED</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeCommandLineProgram.html" title="class in broad.pda.seq.rap">GenomeCommandLineProgram</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html#PCT_MASKED_ALLOWED"><B>PCT_MASKED_ALLOWED</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GenomeScoringProgramFromBed.html" title="class in broad.pda.seq.rap">GenomeScoringProgramFromBed</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/Locus.html#pctOverlapWithRef()"><B>pctOverlapWithRef()</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/Locus.html" title="class in broad.pda.annotation">Locus</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis"><B>PeakCaller</B></A> - Interface in <A HREF="../nextgen/core/analysis/package-summary.html">nextgen.core.analysis</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/chip/PeakFilter.html" title="class in broad.pda.seq.chip"><B>PeakFilter</B></A> - Class in <A HREF="../broad/pda/seq/chip/package-summary.html">broad.pda.seq.chip</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/chip/PeakFilter.html#PeakFilter()"><B>PeakFilter()</B></A> - 
Constructor for class broad.pda.seq.chip.<A HREF="../broad/pda/seq/chip/PeakFilter.html" title="class in broad.pda.seq.chip">PeakFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#pearsonCorrelation(java.util.List, java.util.List)"><B>pearsonCorrelation(List&lt;? extends Number&gt;, List&lt;? extends Number&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#pearsonDistance(double[], double[])"><B>pearsonDistance(double[], double[])</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#pearsonDistance(java.util.List, java.util.List)"><B>pearsonDistance(List&lt;Double&gt;, List&lt;Double&gt;)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/clustering/PearsonDistance.html" title="class in broad.pda.geneexpression.clustering"><B>PearsonDistance</B></A> - Class in <A HREF="../broad/pda/geneexpression/clustering/package-summary.html">broad.pda.geneexpression.clustering</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/clustering/PearsonDistance.html#PearsonDistance()"><B>PearsonDistance()</B></A> - 
Constructor for class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/PearsonDistance.html" title="class in broad.pda.geneexpression.clustering">PearsonDistance</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerUtils.html#percentGC(java.lang.String)"><B>percentGC(String)</B></A> - 
Static method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerUtils.html" title="class in broad.core.primer3">PrimerUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#percentGC()"><B>percentGC()</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#percentGenomeOverlapping(nextgen.core.annotation.Gene)"><B>percentGenomeOverlapping(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#percentOverlapping(nextgen.core.annotation.Gene)"><B>percentOverlapping(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#PERMUTATION_ATTEMPTS"><B>PERMUTATION_ATTEMPTS</B></A> - 
Static variable in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#PERMUTATION_ATTEMPTS"><B>PERMUTATION_ATTEMPTS</B></A> - 
Static variable in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/PermutationNotFoundException.html" title="class in nextgen.core.coordinatesystem"><B>PermutationNotFoundException</B></A> - Exception in <A HREF="../nextgen/core/coordinatesystem/package-summary.html">nextgen.core.coordinatesystem</A><DD>&nbsp;<DT><A HREF="../nextgen/core/coordinatesystem/PermutationNotFoundException.html#PermutationNotFoundException(int)"><B>PermutationNotFoundException(int)</B></A> - 
Constructor for exception nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/PermutationNotFoundException.html" title="class in nextgen.core.coordinatesystem">PermutationNotFoundException</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/Permutations.html" title="class in broad.pda.differentialExpression"><B>Permutations</B></A> - Class in <A HREF="../broad/pda/differentialExpression/package-summary.html">broad.pda.differentialExpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/differentialExpression/Permutations.html#Permutations()"><B>Permutations()</B></A> - 
Constructor for class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/Permutations.html" title="class in broad.pda.differentialExpression">Permutations</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html#PERMUTATIONS"><B>PERMUTATIONS</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html" title="class in broad.pda.seq.rap">BuildRatioNullDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#PERMUTATIONS"><B>PERMUTATIONS</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/ShuffleBAM.html#PERMUTATIONS"><B>PERMUTATIONS</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBAM.html" title="class in broad.pda.seq.rap">ShuffleBAM</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/ShuffleBED.html#PERMUTATIONS"><B>PERMUTATIONS</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBED.html" title="class in broad.pda.seq.rap">ShuffleBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/TreeScalerIO.html#permute()"><B>permute()</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/TreeScalerIO.html" title="class in broad.core.siphy">TreeScalerIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#PERMUTE"><B>PERMUTE</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/Permutations.html#permuteAndAssignFDR(broad.core.datastructures.MatrixWithHeaders, broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, int, double[], boolean, boolean, double)"><B>permuteAndAssignFDR(MatrixWithHeaders, MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, int, double[], boolean, boolean, double)</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/Permutations.html" title="class in broad.pda.differentialExpression">Permutations</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#permuteAnnotation(nextgen.core.annotation.Annotation)"><B>permuteAnnotation(Annotation)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Permutes an annotation to a new location on the same reference in the given coordinate space, preserving the 
 length of the annotation and maintaining a consistent block structure.
<DT><A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html#permuteAnnotation(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>permuteAnnotation(Annotation, Annotation)</B></A> - 
Method in interface nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>
<DD>Permutes an annotation to a new location within the given bounds, preserving the 
 length of the annotation and maintaining a consistent block structure.
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#permuteAnnotation(nextgen.core.annotation.Annotation)"><B>permuteAnnotation(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html#permuteAnnotation(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>permuteAnnotation(Annotation, Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/GenomicSpace.html" title="class in nextgen.core.coordinatesystem">GenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#permuteAnnotation(nextgen.core.annotation.Annotation)"><B>permuteAnnotation(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html#permuteAnnotation(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>permuteAnnotation(Annotation, Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/MaskedGenomicSpace.html" title="class in nextgen.core.coordinatesystem">MaskedGenomicSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#permuteAnnotation(nextgen.core.annotation.Annotation)"><B>permuteAnnotation(Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html#permuteAnnotation(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>permuteAnnotation(Annotation, Annotation)</B></A> - 
Method in class nextgen.core.coordinatesystem.<A HREF="../nextgen/core/coordinatesystem/TranscriptomeSpace.html" title="class in nextgen.core.coordinatesystem">TranscriptomeSpace</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/AnnotationReader.html#permuteAnnotationsOnChromosome(java.util.Map)"><B>permuteAnnotationsOnChromosome(Map&lt;String, Integer&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/AnnotationReader.html" title="class in broad.core.annotation">AnnotationReader</A>
<DD>Jesse Engreitz
 June 24, 2012
 Takes the annotations in the reader and permutes them on their respective chromosomes.
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#permuteColumns(boolean)"><B>permuteColumns(boolean)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#permuteColumns()"><B>permuteColumns()</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/Permutations.html#permuteData(broad.core.datastructures.MatrixWithHeaders, java.util.Collection, java.util.Collection, int, double[], boolean, boolean)"><B>permuteData(MatrixWithHeaders, Collection&lt;String&gt;, Collection&lt;String&gt;, int, double[], boolean, boolean)</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/Permutations.html" title="class in broad.pda.differentialExpression">Permutations</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/Permutations.html#permuteData(broad.core.datastructures.MatrixWithHeaders, java.util.Map, int, double[], boolean, boolean)"><B>permuteData(MatrixWithHeaders, Map&lt;String, Collection&lt;String&gt;&gt;, int, double[], boolean, boolean)</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/Permutations.html" title="class in broad.pda.differentialExpression">Permutations</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/differentialExpression/Permutations.html#permuteDataByGroupSize(broad.core.datastructures.MatrixWithHeaders, int, java.util.Map, int, double[], boolean)"><B>permuteDataByGroupSize(MatrixWithHeaders, int, Map&lt;String, Collection&lt;String&gt;&gt;, int, double[], boolean)</B></A> - 
Static method in class broad.pda.differentialExpression.<A HREF="../broad/pda/differentialExpression/Permutations.html" title="class in broad.pda.differentialExpression">Permutations</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#permuteRegions(nextgen.core.annotation.Annotation)"><B>permuteRegions(Annotation)</B></A> - 
Method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PhastconsReader.html" title="class in broad.core.siphy"><B>PhastconsReader</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/PhastconsReader.html#PhastconsReader(java.lang.String)"><B>PhastconsReader(String)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/PhastconsReader.html" title="class in broad.core.siphy">PhastconsReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PhastconsReader.PhastConsElement.html" title="class in broad.core.siphy"><B>PhastconsReader.PhastConsElement</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/PhastconsReader.PhastConsElement.html#PhastconsReader.PhastConsElement(java.lang.String[])"><B>PhastconsReader.PhastConsElement(String[])</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/PhastconsReader.PhastConsElement.html" title="class in broad.core.siphy">PhastconsReader.PhastConsElement</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/PhylipInterleavedMultipleAlignmentIO.html" title="class in broad.core.multiplealignment"><B>PhylipInterleavedMultipleAlignmentIO</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/PhylipInterleavedMultipleAlignmentIO.html#PhylipInterleavedMultipleAlignmentIO()"><B>PhylipInterleavedMultipleAlignmentIO()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/PhylipInterleavedMultipleAlignmentIO.html" title="class in broad.core.multiplealignment">PhylipInterleavedMultipleAlignmentIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/PhylipSequencialMultipleAlignmentIO.html" title="class in broad.core.multiplealignment"><B>PhylipSequencialMultipleAlignmentIO</B></A> - Class in <A HREF="../broad/core/multiplealignment/package-summary.html">broad.core.multiplealignment</A><DD>&nbsp;<DT><A HREF="../broad/core/multiplealignment/PhylipSequencialMultipleAlignmentIO.html#PhylipSequencialMultipleAlignmentIO()"><B>PhylipSequencialMultipleAlignmentIO()</B></A> - 
Constructor for class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/PhylipSequencialMultipleAlignmentIO.html" title="class in broad.core.multiplealignment">PhylipSequencialMultipleAlignmentIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#piEMIteration(java.util.Map, org.forester.phylogeny.PhylogenyNode, Jama.Matrix)"><B>piEMIteration(Map&lt;String, Matrix&gt;, PhylogenyNode, Matrix)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.html" title="class in broad.core.siphy"><B>PiHMM</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/PiHMM.html#PiHMM(broad.core.siphy.EvolutionaryModel, int)"><B>PiHMM(EvolutionaryModel, int)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.html" title="class in broad.core.siphy">PiHMM</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.html#PiHMM(broad.core.siphy.EvolutionaryModel, double, double)"><B>PiHMM(EvolutionaryModel, double, double)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.html" title="class in broad.core.siphy">PiHMM</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html" title="class in broad.core.siphy"><B>PiHMM.ConstrainedModel</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html#PiHMM.ConstrainedModel(broad.core.siphy.EvolutionaryModel, java.lang.String)"><B>PiHMM.ConstrainedModel(EvolutionaryModel, String)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.ConstrainedModel.html" title="class in broad.core.siphy">PiHMM.ConstrainedModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.PiStateModel.html" title="class in broad.core.siphy"><B>PiHMM.PiStateModel</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/PiHMM.PiStateModel.html#PiHMM.PiStateModel(broad.core.siphy.EvolutionaryModel, java.lang.String)"><B>PiHMM.PiStateModel(EvolutionaryModel, String)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.PiStateModel.html" title="class in broad.core.siphy">PiHMM.PiStateModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PiHMM.PiStateModel.html#PiHMM.PiStateModel(broad.core.siphy.EvolutionaryModel, Jama.Matrix, java.lang.String)"><B>PiHMM.PiStateModel(EvolutionaryModel, Matrix, String)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/PiHMM.PiStateModel.html" title="class in broad.core.siphy">PiHMM.PiStateModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineAutomator.html" title="class in broad.core.util"><B>PipelineAutomator</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>This class will use an input list of files and a config file to run an automated pipeline to align and process sequencing data<DT><A HREF="../broad/core/util/PipelineAutomator.html#PipelineAutomator(java.lang.String, java.lang.String)"><B>PipelineAutomator(String, String)</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.html" title="class in broad.core.util">PipelineAutomator</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html" title="class in broad.core.util"><B>PipelineAutomator.AlignmentUtils</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>Bowtie command lines needed for this pipeline<DT><A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html#PipelineAutomator.AlignmentUtils()"><B>PipelineAutomator.AlignmentUtils()</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.AlignmentUtils.html" title="class in broad.core.util">PipelineAutomator.AlignmentUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineAutomator.DGEInput.html" title="class in broad.core.util"><B>PipelineAutomator.DGEInput</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>&nbsp;<DT><A HREF="../broad/core/util/PipelineAutomator.DGEInput.html#PipelineAutomator.DGEInput()"><B>PipelineAutomator.DGEInput()</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/PipelineAutomator.DGEInput.html" title="class in broad.core.util">PipelineAutomator.DGEInput</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/util/PipelineUtils.html" title="class in broad.core.util"><B>PipelineUtils</B></A> - Class in <A HREF="../broad/core/util/package-summary.html">broad.core.util</A><DD>&nbsp;<DT><A HREF="../broad/core/util/PipelineUtils.html#PipelineUtils()"><B>PipelineUtils()</B></A> - 
Constructor for class broad.core.util.<A HREF="../broad/core/util/PipelineUtils.html" title="class in broad.core.util">PipelineUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#placeInTree(int, int, V)"><B>placeInTree(int, int, V)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Put a new interval into the tree (or update the value associated with an existing interval).
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#plot(java.awt.Graphics2D, int, int, java.util.List)"><B>plot(Graphics2D, int, int, List&lt;GenomicAnnotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap"><B>PlotAggregateRegions</B></A> - Class in <A HREF="../broad/pda/seq/rap/package-summary.html">broad.pda.seq.rap</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#PlotAggregateRegions()"><B>PlotAggregateRegions()</B></A> - 
Constructor for class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>&nbsp;
<DT><A HREF="../Jama/Matrix.html#plus(Jama.Matrix)"><B>plus(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>C = A + B
<DT><A HREF="../Jama/Matrix.html#plusEquals(Jama.Matrix)"><B>plusEquals(Matrix)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>A = A + B
<DT><A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html#POINT_PATTERN"><B>POINT_PATTERN</B></A> - 
Static variable in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicGenomicAnnotationReader.html" title="class in broad.core.annotation">BasicGenomicAnnotationReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html#poisson(int, double)"><B>poisson(int, double)</B></A> - 
Static method in class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif"><B>PositionWeightColumn</B></A> - Class in <A HREF="../broad/core/motif/package-summary.html">broad.core.motif</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#PositionWeightColumn()"><B>PositionWeightColumn()</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#PositionWeightColumn(java.lang.String[])"><B>PositionWeightColumn(String[])</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#PositionWeightColumn(int[])"><B>PositionWeightColumn(int[])</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#PositionWeightColumn(int, int, int, int)"><B>PositionWeightColumn(int, int, int, int)</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#PositionWeightColumn(double[])"><B>PositionWeightColumn(double[])</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif"><B>PositionWeightMatrix</B></A> - Class in <A HREF="../broad/core/motif/package-summary.html">broad.core.motif</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#PositionWeightMatrix(java.lang.String)"><B>PositionWeightMatrix(String)</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrixIO.html" title="class in broad.core.motif"><B>PositionWeightMatrixIO</B></A> - Class in <A HREF="../broad/core/motif/package-summary.html">broad.core.motif</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/PositionWeightMatrixIO.html#PositionWeightMatrixIO()"><B>PositionWeightMatrixIO()</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrixIO.html" title="class in broad.core.motif">PositionWeightMatrixIO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/PositionWeightMatrixModel.html" title="class in broad.core.siphy"><B>PositionWeightMatrixModel</B></A> - Class in <A HREF="../broad/core/siphy/package-summary.html">broad.core.siphy</A><DD>&nbsp;<DT><A HREF="../broad/core/siphy/PositionWeightMatrixModel.html#PositionWeightMatrixModel(broad.core.motif.PositionWeightMatrix, broad.core.siphy.EvolutionaryModel)"><B>PositionWeightMatrixModel(PositionWeightMatrix, EvolutionaryModel)</B></A> - 
Constructor for class broad.core.siphy.<A HREF="../broad/core/siphy/PositionWeightMatrixModel.html" title="class in broad.core.siphy">PositionWeightMatrixModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/MatrixWithHeaders.html#pow()"><B>pow()</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/MatrixWithHeaders.html" title="class in broad.core.datastructures">MatrixWithHeaders</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html#pp"><B>pp</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.Probe.html" title="class in broad.pda.methylation">HybridCaptureDesigner.Probe</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/segmentation/PrecomputeDataAlignmentStats.html" title="class in broad.pda.seq.segmentation"><B>PrecomputeDataAlignmentStats</B></A> - Class in <A HREF="../broad/pda/seq/segmentation/package-summary.html">broad.pda.seq.segmentation</A><DD>&nbsp;<DT><A HREF="../broad/pda/seq/segmentation/PrecomputeDataAlignmentStats.html#PrecomputeDataAlignmentStats(java.io.File, java.lang.String, java.lang.String)"><B>PrecomputeDataAlignmentStats(File, String, String)</B></A> - 
Constructor for class broad.pda.seq.segmentation.<A HREF="../broad/pda/seq/segmentation/PrecomputeDataAlignmentStats.html" title="class in broad.pda.seq.segmentation">PrecomputeDataAlignmentStats</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/general/Predicates.html" title="class in nextgen.core.general"><B>Predicates</B></A> - Class in <A HREF="../nextgen/core/general/package-summary.html">nextgen.core.general</A><DD>&nbsp;<DT><A HREF="../nextgen/core/general/Predicates.html#Predicates()"><B>Predicates()</B></A> - 
Constructor for class nextgen.core.general.<A HREF="../nextgen/core/general/Predicates.html" title="class in nextgen.core.general">Predicates</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/PreprocessDataSet.html" title="class in broad.pda.geneexpression"><B>PreprocessDataSet</B></A> - Class in <A HREF="../broad/pda/geneexpression/package-summary.html">broad.pda.geneexpression</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/PreprocessDataSet.html#PreprocessDataSet(java.io.File, java.io.File, java.lang.String)"><B>PreprocessDataSet(File, File, String)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/PreprocessDataSet.html" title="class in broad.pda.geneexpression">PreprocessDataSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/PreprocessDataSet.html#PreprocessDataSet(java.io.File[], java.io.File, java.lang.String, double, java.io.File)"><B>PreprocessDataSet(File[], File, String, double, File)</B></A> - 
Constructor for class broad.pda.geneexpression.<A HREF="../broad/pda/geneexpression/PreprocessDataSet.html" title="class in broad.pda.geneexpression">PreprocessDataSet</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3"><B>Primer3Configuration</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>A dumb data object that represents configuration information
 used to drive primer3.<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#Primer3Configuration()"><B>Primer3Configuration()</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3"><B>Primer3ConfigurationFactory</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>A factory that produces configurations for primer
 picking.<DT><A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html#Primer3ConfigurationFactory()"><B>Primer3ConfigurationFactory()</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3ConfigurationFactory.html" title="class in broad.core.primer3">Primer3ConfigurationFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Exception.html" title="class in broad.core.primer3"><B>Primer3Exception</B></A> - Exception in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>Exception thrown when there is a problem generating
 primers using primer3.<DT><A HREF="../broad/core/primer3/Primer3Exception.html#Primer3Exception(java.lang.String)"><B>Primer3Exception(String)</B></A> - 
Constructor for exception broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Exception.html" title="class in broad.core.primer3">Primer3Exception</A>
<DD>Creates a new Primer3Exception
 with the the given message.
<DT><A HREF="../broad/core/primer3/Primer3Exception.html#Primer3Exception(java.lang.Throwable)"><B>Primer3Exception(Throwable)</B></A> - 
Constructor for exception broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Exception.html" title="class in broad.core.primer3">Primer3Exception</A>
<DD>Creates a new Primer3Exception
 with a Throwable that was the
 underlying cause of the problem.
<DT><A HREF="../broad/core/primer3/Primer3IO.html" title="class in broad.core.primer3"><B>Primer3IO</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>&nbsp;<DT><A HREF="../broad/core/primer3/Primer3IO.html#Primer3IO()"><B>Primer3IO()</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3IO.html" title="class in broad.core.primer3">Primer3IO</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3PickingSummary.html" title="interface in broad.core.primer3"><B>Primer3PickingSummary</B></A> - Interface in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>A Primer3PickingSummary is a summary of how many
 primers primer3 picked in a primer picking session.<DT><A HREF="../broad/core/primer3/Primer3PrimerPickingMode.html" title="class in broad.core.primer3"><B>Primer3PrimerPickingMode</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>A type-safe enum of the different "modes"
 of operation of primer3.<DT><A HREF="../broad/core/primer3/Primer3PrimerPickingResult.html" title="interface in broad.core.primer3"><B>Primer3PrimerPickingResult</B></A> - Interface in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>TODO: document me!<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3"><B>Primer3SequenceInputTags</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>&nbsp;<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.html#Primer3SequenceInputTags(broad.core.sequence.Sequence)"><B>Primer3SequenceInputTags(Sequence)</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.html" title="class in broad.core.primer3">Primer3SequenceInputTags</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.SequenceRegionCoordinates.html" title="class in broad.core.primer3"><B>Primer3SequenceInputTags.SequenceRegionCoordinates</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>&nbsp;<DT><A HREF="../broad/core/primer3/Primer3SequenceInputTags.SequenceRegionCoordinates.html#Primer3SequenceInputTags.SequenceRegionCoordinates(int, int)"><B>Primer3SequenceInputTags.SequenceRegionCoordinates(int, int)</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3SequenceInputTags.SequenceRegionCoordinates.html" title="class in broad.core.primer3">Primer3SequenceInputTags.SequenceRegionCoordinates</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#primerMin3PrimeOverlapOfJunction"><B>primerMin3PrimeOverlapOfJunction</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#primerMin5PrimerOverlapOfJunction"><B>primerMin5PrimerOverlapOfJunction</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#primerNumReturn"><B>primerNumReturn</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3"><B>PrimerPair</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>&nbsp;<DT><A HREF="../broad/core/primer3/PrimerPair.html#PrimerPair(java.lang.String[])"><B>PrimerPair(String[])</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#PrimerPair()"><B>PrimerPair()</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPairReader.html" title="class in broad.core.primer3"><B>PrimerPairReader</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>&nbsp;<DT><A HREF="../broad/core/primer3/PrimerPairReader.html#PrimerPairReader(java.lang.String)"><B>PrimerPairReader(String)</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPairReader.html" title="class in broad.core.primer3">PrimerPairReader</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerUtils.html" title="class in broad.core.primer3"><B>PrimerUtils</B></A> - Class in <A HREF="../broad/core/primer3/package-summary.html">broad.core.primer3</A><DD>&nbsp;<DT><A HREF="../broad/core/primer3/PrimerUtils.html#PrimerUtils()"><B>PrimerUtils()</B></A> - 
Constructor for class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerUtils.html" title="class in broad.core.primer3">PrimerUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#primerWindowSize"><B>primerWindowSize</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#print()"><B>print()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/PrimerPair.html#print()"><B>print()</B></A> - 
Method in class broad.core.primer3.<A HREF="../broad/core/primer3/PrimerPair.html" title="class in broad.core.primer3">PrimerPair</A>
<DD>Write the whitespace separated data line in the order that the constructor uses
<DT><A HREF="../Jama/Matrix.html#print(int, int)"><B>print(int, int)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Print the matrix to stdout.
<DT><A HREF="../Jama/Matrix.html#print(java.io.PrintWriter, int, int)"><B>print(PrintWriter, int, int)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Print the matrix to the output stream.
<DT><A HREF="../Jama/Matrix.html#print(java.text.NumberFormat, int)"><B>print(NumberFormat, int)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Print the matrix to stdout.
<DT><A HREF="../Jama/Matrix.html#print(java.io.PrintWriter, java.text.NumberFormat, int)"><B>print(PrintWriter, NumberFormat, int)</B></A> - 
Method in class Jama.<A HREF="../Jama/Matrix.html" title="class in Jama">Matrix</A>
<DD>Print the matrix to the output stream.
<DT><A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html#printClusterSizes()"><B>printClusterSizes()</B></A> - 
Method in class broad.pda.geneexpression.clustering.<A HREF="../broad/pda/geneexpression/clustering/Kmeans.KmeansInfo.html" title="class in broad.pda.geneexpression.clustering">Kmeans.KmeansInfo</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#printKmer(int[])"><B>printKmer(int[])</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightColumn.html#printProb()"><B>printProb()</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightColumn.html" title="class in broad.core.motif">PositionWeightColumn</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/transcriptsNullModel.html#printRandModels(java.lang.String)"><B>printRandModels(String)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/transcriptsNullModel.html" title="class in broad.pda.gene">transcriptsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/NeighborsNullModel.html#printRandNeighborsInfo()"><B>printRandNeighborsInfo()</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/NeighborsNullModel.html" title="class in broad.pda.gene">NeighborsNullModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/motif/PositionWeightMatrix.html#printRefinedWeightMatrix(double)"><B>printRefinedWeightMatrix(double)</B></A> - 
Method in class broad.core.motif.<A HREF="../broad/core/motif/PositionWeightMatrix.html" title="class in broad.core.motif">PositionWeightMatrix</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.html#PROBE_LENGTH"><B>PROBE_LENGTH</B></A> - 
Static variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.html" title="class in broad.pda.methylation">HybridCaptureDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/geneexpression/agilent/analysis/ProcessDataset.html" title="class in broad.pda.geneexpression.agilent.analysis"><B>ProcessDataset</B></A> - Class in <A HREF="../broad/pda/geneexpression/agilent/analysis/package-summary.html">broad.pda.geneexpression.agilent.analysis</A><DD>&nbsp;<DT><A HREF="../broad/pda/geneexpression/agilent/analysis/ProcessDataset.html#ProcessDataset(broad.core.datastructures.MatrixWithHeaders, java.util.Map, java.util.Map, java.lang.String)"><B>ProcessDataset(MatrixWithHeaders, Map&lt;String, Collection&lt;String&gt;&gt;, Map&lt;String, String&gt;, String)</B></A> - 
Constructor for class broad.pda.geneexpression.agilent.analysis.<A HREF="../broad/pda/geneexpression/agilent/analysis/ProcessDataset.html" title="class in broad.pda.geneexpression.agilent.analysis">ProcessDataset</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.Processor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.Processor.html" title="class in broad.pda.seq.rap">SkellamScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamScore.Processor.html#processWindow(nextgen.core.annotation.Annotation, broad.pda.seq.rap.SkellamScore)"><B>processWindow(Annotation, SkellamScore)</B></A> - 
Method in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamScore.Processor.html" title="class in broad.pda.seq.rap">SkellamScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.Processor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.Processor.html" title="class in nextgen.core.model.score">CountScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CountScore.Processor.html#processWindow(nextgen.core.annotation.Annotation, nextgen.core.model.score.CountScore)"><B>processWindow(Annotation, CountScore)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CountScore.Processor.html" title="class in nextgen.core.model.score">CountScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CoverageScore.Processor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CoverageScore.Processor.html" title="class in nextgen.core.model.score">CoverageScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/CoverageScore.Processor.html#processWindow(nextgen.core.annotation.Annotation, nextgen.core.model.score.CoverageScore)"><B>processWindow(Annotation, CoverageScore)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/CoverageScore.Processor.html" title="class in nextgen.core.model.score">CoverageScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/LengthScore.Processor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/LengthScore.Processor.html" title="class in nextgen.core.model.score">LengthScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/LengthScore.Processor.html#processWindow(nextgen.core.annotation.Annotation, nextgen.core.model.score.LengthScore)"><B>processWindow(Annotation, LengthScore)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/LengthScore.Processor.html" title="class in nextgen.core.model.score">LengthScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html" title="class in nextgen.core.model.score">NewRatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html#processWindow(nextgen.core.annotation.Annotation, nextgen.core.model.score.NewRatioScore)"><B>processWindow(Annotation, NewRatioScore)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/NewRatioScore.Processor.html" title="class in nextgen.core.model.score">NewRatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.Processor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.Processor.html" title="class in nextgen.core.model.score">RatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/RatioScore.Processor.html#processWindow(nextgen.core.annotation.Annotation, nextgen.core.model.score.RatioScore)"><B>processWindow(Annotation, RatioScore)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/RatioScore.Processor.html" title="class in nextgen.core.model.score">RatioScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.Processor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.Processor.html" title="class in nextgen.core.model.score">ScanStatisticScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/ScanStatisticScore.Processor.html#processWindow(nextgen.core.annotation.Annotation, nextgen.core.model.score.ScanStatisticScore)"><B>processWindow(Annotation, ScanStatisticScore)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/ScanStatisticScore.Processor.html" title="class in nextgen.core.model.score">ScanStatisticScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/SumScore.Processor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/SumScore.Processor.html" title="class in nextgen.core.model.score">SumScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/SumScore.Processor.html#processWindow(nextgen.core.annotation.Annotation, nextgen.core.model.score.SumScore)"><B>processWindow(Annotation, SumScore)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/SumScore.Processor.html" title="class in nextgen.core.model.score">SumScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html" title="class in nextgen.core.model.score">WindowAllScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html#processWindow(nextgen.core.annotation.Annotation, nextgen.core.model.score.WindowAllScore)"><B>processWindow(Annotation, WindowAllScore)</B></A> - 
Method in class nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowAllScore.Processor.html" title="class in nextgen.core.model.score">WindowAllScore.Processor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowProcessor.html#processWindow(nextgen.core.annotation.Annotation)"><B>processWindow(Annotation)</B></A> - 
Method in interface nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowProcessor.html" title="interface in nextgen.core.model.score">WindowProcessor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/model/score/WindowProcessor.html#processWindow(nextgen.core.annotation.Annotation, S)"><B>processWindow(Annotation, S)</B></A> - 
Method in interface nextgen.core.model.score.<A HREF="../nextgen/core/model/score/WindowProcessor.html" title="interface in nextgen.core.model.score">WindowProcessor</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/readFilters/ProperPairFilter.html" title="class in nextgen.core.readFilters"><B>ProperPairFilter</B></A> - Class in <A HREF="../nextgen/core/readFilters/package-summary.html">nextgen.core.readFilters</A><DD>&nbsp;<DT><A HREF="../nextgen/core/readFilters/ProperPairFilter.html#ProperPairFilter()"><B>ProperPairFilter()</B></A> - 
Constructor for class nextgen.core.readFilters.<A HREF="../nextgen/core/readFilters/ProperPairFilter.html" title="class in nextgen.core.readFilters">ProperPairFilter</A>
<DD>&nbsp;
<DT><A HREF="../jaligner/Sequence.html#PROTEIN"><B>PROTEIN</B></A> - 
Static variable in class jaligner.<A HREF="../jaligner/Sequence.html" title="class in jaligner">Sequence</A>
<DD>Sequence type protein.
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#pruneAndPeel(java.util.Map, org.forester.phylogeny.PhylogenyNode, int)"><B>pruneAndPeel(Map&lt;String, Matrix&gt;, PhylogenyNode, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#pruneTree(java.util.List, org.forester.phylogeny.Phylogeny)"><B>pruneTree(List&lt;String&gt;, Phylogeny)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/ConservationUtils.html#pruneTree(java.lang.String, org.forester.phylogeny.Phylogeny)"><B>pruneTree(String, Phylogeny)</B></A> - 
Static method in class broad.core.siphy.<A HREF="../broad/core/siphy/ConservationUtils.html" title="class in broad.core.siphy">ConservationUtils</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#pruneTree(java.util.List)"><B>pruneTree(List&lt;String&gt;)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/tools/PSLXToSAM.html" title="class in broad.pda.tools"><B>PSLXToSAM</B></A> - Class in <A HREF="../broad/pda/tools/package-summary.html">broad.pda.tools</A><DD>&nbsp;<DT><A HREF="../broad/pda/tools/PSLXToSAM.html#PSLXToSAM()"><B>PSLXToSAM()</B></A> - 
Constructor for class broad.pda.tools.<A HREF="../broad/pda/tools/PSLXToSAM.html" title="class in broad.pda.tools">PSLXToSAM</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#put(broad.core.datastructures.Interval, V)"><B>put(Interval, V)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Put a new interval into the tree (or update the value associated with an existing interval).
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#put(int, int, V)"><B>put(int, int, V)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Rewrote put to check that there is an already existing node
<DT><A HREF="../broad/core/util/CLUtil.ArgumentMap.html#put(java.lang.String, java.lang.String)"><B>put(String, String)</B></A> - 
Method in class broad.core.util.<A HREF="../broad/core/util/CLUtil.ArgumentMap.html" title="class in broad.core.util">CLUtil.ArgumentMap</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/math/Statistics.html#pvalue(java.util.List, double, boolean)"><B>pvalue(List&lt;Double&gt;, double, boolean)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>Computes what is P(X > value) using list of values
<DT><A HREF="../broad/core/math/Statistics.html#pvalue(double[], double, boolean)"><B>pvalue(double[], double, boolean)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>Computes what is P(X > value) using list of values
<DT><A HREF="../broad/core/math/Statistics.html#pvalue(java.util.List, double)"><B>pvalue(List&lt;Double&gt;, double)</B></A> - 
Static method in class broad.core.math.<A HREF="../broad/core/math/Statistics.html" title="class in broad.core.math">Statistics</A>
<DD>Computes what is P(X > value) using list of values
<DT><A HREF="../broad/core/motif/PWMUtils.html" title="class in broad.core.motif"><B>PWMUtils</B></A> - Class in <A HREF="../broad/core/motif/package-summary.html">broad.core.motif</A><DD>&nbsp;<DT><A HREF="../broad/core/motif/PWMUtils.html#PWMUtils()"><B>PWMUtils()</B></A> - 
Constructor for class broad.core.motif.<A HREF="../broad/core/motif/PWMUtils.html" title="class in broad.core.motif">PWMUtils</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-15.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-17.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-16.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-16.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
