<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:31 EDT 2013 -->
<TITLE>
O-Index
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="O-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-14.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-16.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-15.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-15.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>
<A NAME="_O_"><!-- --></A><H2>
<B>O</B></H2>
<DL>
<DT><A HREF="../nextgen/core/scripture/statistics/OldReconstructionCompare.html" title="class in nextgen.core.scripture.statistics"><B>OldReconstructionCompare</B></A> - Class in <A HREF="../nextgen/core/scripture/statistics/package-summary.html">nextgen.core.scripture.statistics</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/statistics/OldReconstructionCompare.html#OldReconstructionCompare(java.io.File, java.io.File, java.lang.String, double)"><B>OldReconstructionCompare(File, File, String, double)</B></A> - 
Constructor for class nextgen.core.scripture.statistics.<A HREF="../nextgen/core/scripture/statistics/OldReconstructionCompare.html" title="class in nextgen.core.scripture.statistics">OldReconstructionCompare</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/siphy/EvolutionaryModel.html#omegaEMIteration(java.util.Map, org.forester.phylogeny.PhylogenyNode, double, int)"><B>omegaEMIteration(Map&lt;String, Matrix&gt;, PhylogenyNode, double, int)</B></A> - 
Method in class broad.core.siphy.<A HREF="../broad/core/siphy/EvolutionaryModel.html" title="class in broad.core.siphy">EvolutionaryModel</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#optimalMeltingTemp"><B>optimalMeltingTemp</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#optimalPrimerSize"><B>optimalPrimerSize</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#optimalProductSize"><B>optimalProductSize</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_ALIGN_TO_TRANSCRIPTS"><B>OPTION_ALIGN_TO_TRANSCRIPTS</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to align to a fasta file of transcript sequences
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_ANNOTATION"><B>OPTION_ANNOTATION</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for genome annotation in gtf format for use by tophat
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_BED_FILE_FOR_WIG"><B>OPTION_BED_FILE_FOR_WIG</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to write fragment endpoints to wig and bigwig file
 Writes counts for positions overlapping genes in this bed file
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_BOWTIE2_BUILD_PATH"><B>OPTION_BOWTIE2_BUILD_PATH</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for bowtie2-build executable file
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_BOWTIE2_OPTIONS"><B>OPTION_BOWTIE2_OPTIONS</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide options to bowtie2
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_BOWTIE2_PATH"><B>OPTION_BOWTIE2_PATH</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for bowtie2 executable file
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_CHR_SIZE_FILE"><B>OPTION_CHR_SIZE_FILE</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Chromosome size file
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_FASTQ_READ_PAIR_NUMBER_DELIMITER"><B>OPTION_FASTQ_READ_PAIR_NUMBER_DELIMITER</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_FASTX_DIRECTORY"><B>OPTION_FASTX_DIRECTORY</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Directory containing fastx binaries
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_FILTER_RRNA"><B>OPTION_FILTER_RRNA</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide a fasta file of rRNA sequences for depletion
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_FRAGMENT_SIZE_DIST_OPTIONS"><B>OPTION_FRAGMENT_SIZE_DIST_OPTIONS</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide parameters to fragment size distribution calculation
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_ASSEMBLY"><B>OPTION_GENOME_ASSEMBLY</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for genome assembly name e.g.
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_BOWTIE"><B>OPTION_GENOME_BOWTIE</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for prefix of genome bowtie index files
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_FASTA"><B>OPTION_GENOME_FASTA</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for path to genome fasta file
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_NOVOINDEX"><B>OPTION_GENOME_NOVOINDEX</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for novoindex of genome
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_GENOME_SPACE_SIZE_FILE"><B>OPTION_GENOME_SPACE_SIZE_FILE</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for chromosome size file for genomic space stats
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_GLOBAL_STATS_JAR"><B>OPTION_GLOBAL_STATS_JAR</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Jar file to compute alignment global stats
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_IGVTOOLS_PATH"><B>OPTION_IGVTOOLS_PATH</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for igvtools executable
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_MERGE_SAMPLES"><B>OPTION_MERGE_SAMPLES</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Merge a list of samples.
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_NOVOALIGN_OPTIONS"><B>OPTION_NOVOALIGN_OPTIONS</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide options to novoalign e.g.
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_NOVOALIGN_PATH"><B>OPTION_NOVOALIGN_PATH</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for novoalign executable
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_COLLECT_RNASEQ_METRICS"><B>OPTION_PICARD_COLLECT_RNASEQ_METRICS</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide an alternative jar file for Picard program CollectRnaSeqMetrics
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_DIRECTORY"><B>OPTION_PICARD_DIRECTORY</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for directory containing picard executables
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_REF_FLAT"><B>OPTION_PICARD_REF_FLAT</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide ref_flat annotation to Picard
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_RIBOSOMAL_INTERVALS"><B>OPTION_PICARD_RIBOSOMAL_INTERVALS</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide ribosomal intervals to Picard
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_PICARD_STRAND_SPECIFICITY"><B>OPTION_PICARD_STRAND_SPECIFICITY</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide strand specificity argument to Picard
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_QUEUE_NAME"><B>OPTION_QUEUE_NAME</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_RNA_CLASS"><B>OPTION_RNA_CLASS</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide an RNA class fasta file e.g.
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_SAMTOOLS_PATH"><B>OPTION_SAMTOOLS_PATH</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for samtools executable
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_SEQUENCING_ADAPTER_READ_1"><B>OPTION_SEQUENCING_ADAPTER_READ_1</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Sequencing adapter sequence for read 1
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_SEQUENCING_ADAPTER_READ_2"><B>OPTION_SEQUENCING_ADAPTER_READ_2</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Sequencing adapter sequence for read 2
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_TOPHAT_OPTIONS"><B>OPTION_TOPHAT_OPTIONS</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option to provide options to tophat e.g.
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_TOPHAT_PATH"><B>OPTION_TOPHAT_PATH</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for tophat executable file
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_TRANSCRIPTOME_SPACE_BED_FILE"><B>OPTION_TRANSCRIPTOME_SPACE_BED_FILE</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Config file option for bed annotation for transcriptome space stats
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_WIG_TO_BIGWIG_PATH"><B>OPTION_WIG_TO_BIGWIG_PATH</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>Path to UCSC WigToBigWig executable
<DT><A HREF="../broad/core/parser/ConfigFileParser.html#OPTION_WIG_WRITER_JAR"><B>OPTION_WIG_WRITER_JAR</B></A> - 
Static variable in class broad.core.parser.<A HREF="../broad/core/parser/ConfigFileParser.html" title="class in broad.core.parser">ConfigFileParser</A>
<DD>WigWriter jar file
<DT><A HREF="../nextgen/core/general/Predicates.html#or(java.lang.Iterable)"><B>or(Iterable&lt;? extends Predicate&lt;? super T&gt;&gt;)</B></A> - 
Static method in class nextgen.core.general.<A HREF="../nextgen/core/general/Predicates.html" title="class in nextgen.core.general">Predicates</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/general/Predicates.html#or(org.apache.commons.collections15.Predicate...)"><B>or(Predicate&lt;? super T&gt;...)</B></A> - 
Static method in class nextgen.core.general.<A HREF="../nextgen/core/general/Predicates.html" title="class in nextgen.core.general">Predicates</A>
<DD>&nbsp;
<DT><A HREF="../org/jgrapht/alg/package-summary.html"><B>org.jgrapht.alg</B></A> - package org.jgrapht.alg<DD>&nbsp;<DT><A HREF="../org/jgrapht/util/package-summary.html"><B>org.jgrapht.util</B></A> - package org.jgrapht.util<DD>&nbsp;<DT><A HREF="../broad/pda/gene/GeneTools.html#orientationForGene(nextgen.core.annotation.Gene, broad.core.sequence.Sequence)"><B>orientationForGene(Gene, Sequence)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html#orientationFromSpliceSites(nextgen.core.annotation.Annotation, broad.core.sequence.Sequence, boolean)"><B>orientationFromSpliceSites(Annotation, Sequence, boolean)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html#orientationFromSpliceSites(nextgen.core.annotation.Annotation, broad.core.sequence.Sequence)"><B>orientationFromSpliceSites(Annotation, Sequence)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture"><B>OrientedChromosomeTranscriptGraph</B></A> - Class in <A HREF="../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html#OrientedChromosomeTranscriptGraph(java.lang.String, java.lang.String)"><B>OrientedChromosomeTranscriptGraph(String, String)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture"><B>OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</B></A> - Class in <A HREF="../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#OrientedChromosomeTranscriptGraph.TranscriptGraphEdge(nextgen.core.annotation.Annotation)"><B>OrientedChromosomeTranscriptGraph.TranscriptGraphEdge(Annotation)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#OrientedChromosomeTranscriptGraph.TranscriptGraphEdge(nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation, nextgen.core.annotation.Annotation)"><B>OrientedChromosomeTranscriptGraph.TranscriptGraphEdge(Annotation, Annotation, Annotation)</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory.html" title="class in nextgen.core.scripture"><B>OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory</B></A> - Class in <A HREF="../nextgen/core/scripture/package-summary.html">nextgen.core.scripture</A><DD>&nbsp;<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory.html#OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory()"><B>OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory()</B></A> - 
Constructor for class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdgeFactory</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html#OTHER_TYPE"><B>OTHER_TYPE</B></A> - 
Static variable in class broad.pda.feature.genome.<A HREF="../broad/pda/feature/genome/Chromosome.AgpEntry.html" title="class in broad.pda.feature.genome">Chromosome.AgpEntry</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#OUTER_LENGTH"><B>OUTER_LENGTH</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html#outOrientedDegree(nextgen.core.annotation.Annotation, java.util.Collection, org.jgrapht.DirectedGraph)"><B>outOrientedDegree(Annotation, Collection&lt;ChromosomeWithBubblesJGraphT.EdgeSourceType&gt;, DirectedGraph&lt;Annotation, ChromosomeWithBubblesJGraphT.BubbleEdge&gt;)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>
<DD>Returns the out degree of the given vertex for a list of allowed edge types
<DT><A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterFastqForMethylationAlignment.html" title="class in broad.pda.methylation">FilterFastqForMethylationAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamByBaseComposition.html" title="class in broad.pda.methylation">FilterSamByBaseComposition</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/FilterSamForMethylationAlignment.html" title="class in broad.pda.methylation">FilterSamForMethylationAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/samtools/PairedEndTDF.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.samtools.<A HREF="../broad/pda/samtools/PairedEndTDF.html" title="class in broad.pda.samtools">PairedEndTDF</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html" title="class in broad.pda.seq.rap">BuildRatioNullDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/CountReads.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/CountReads.html" title="class in broad.pda.seq.rap">CountReads</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/GCContent.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/GCContent.html" title="class in broad.pda.seq.rap">GCContent</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html" title="class in broad.pda.seq.rap">RatioPermutationPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/ShuffleBAM.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBAM.html" title="class in broad.pda.seq.rap">ShuffleBAM</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/ShuffleBED.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/ShuffleBED.html" title="class in broad.pda.seq.rap">ShuffleBED</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html" title="class in broad.pda.seq.rap">SkellamKLPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SlideAndCalculate.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCalculate.html" title="class in broad.pda.seq.rap">SlideAndCalculate</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SlideAndCount.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCount.html" title="class in broad.pda.seq.rap">SlideAndCount</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/UnmarkDuplicates.html#OUTPUT"><B>OUTPUT</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/UnmarkDuplicates.html" title="class in broad.pda.seq.rap">UnmarkDuplicates</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/OutputPadlock.html" title="class in broad.pda.methylation"><B>OutputPadlock</B></A> - Class in <A HREF="../broad/pda/methylation/package-summary.html">broad.pda.methylation</A><DD>&nbsp;<DT><A HREF="../broad/pda/methylation/OutputPadlock.html#OutputPadlock(java.lang.String, int, boolean)"><B>OutputPadlock(String, int, boolean)</B></A> - 
Constructor for class broad.pda.methylation.<A HREF="../broad/pda/methylation/OutputPadlock.html" title="class in broad.pda.methylation">OutputPadlock</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#overGCContentPenaltyWeight"><B>overGCContentPenaltyWeight</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.html#OVERHANG"><B>OVERHANG</B></A> - 
Static variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.html" title="class in broad.pda.methylation">HybridCaptureDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneTools.html#overlap(java.util.Map, java.util.Map, java.lang.Boolean)"><B>overlap(Map&lt;String, IntervalTree&lt;Gene&gt;&gt;, Map&lt;String, IntervalTree&lt;Gene&gt;&gt;, Boolean)</B></A> - 
Static method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneTools.html" title="class in broad.pda.gene">GeneTools</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/methylation/HybridCaptureDesigner.html#OVERLAP"><B>OVERLAP</B></A> - 
Static variable in class broad.pda.methylation.<A HREF="../broad/pda/methylation/HybridCaptureDesigner.html" title="class in broad.pda.methylation">HybridCaptureDesigner</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html#OVERLAP"><B>OVERLAP</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/BuildRatioNullDistribution.html" title="class in broad.pda.seq.rap">BuildRatioNullDistribution</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html#OVERLAP"><B>OVERLAP</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/PlotAggregateRegions.html" title="class in broad.pda.seq.rap">PlotAggregateRegions</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html#OVERLAP"><B>OVERLAP</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/RatioPermutationPeakCaller.html" title="class in broad.pda.seq.rap">RatioPermutationPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html#OVERLAP"><B>OVERLAP</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SkellamKLPeakCaller.html" title="class in broad.pda.seq.rap">SkellamKLPeakCaller</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SlideAndCalculate.html#OVERLAP"><B>OVERLAP</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCalculate.html" title="class in broad.pda.seq.rap">SlideAndCalculate</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/rap/SlideAndCount.html#OVERLAP"><B>OVERLAP</B></A> - 
Variable in class broad.pda.seq.rap.<A HREF="../broad/pda/seq/rap/SlideAndCount.html" title="class in broad.pda.seq.rap">SlideAndCount</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#overlap_GenomicLevel(broad.pda.annotation.BEDFileParser, java.lang.String, java.lang.String)"><B>overlap_GenomicLevel(BEDFileParser, String, String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/annotation/BEDFileParser.html#overlapByGenomicRegion(broad.pda.annotation.BEDFileParser, java.lang.String)"><B>overlapByGenomicRegion(BEDFileParser, String)</B></A> - 
Method in class broad.pda.annotation.<A HREF="../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/filter/OverlapFilter.html" title="class in nextgen.core.annotation.filter"><B>OverlapFilter</B></A> - Class in <A HREF="../nextgen/core/annotation/filter/package-summary.html">nextgen.core.annotation.filter</A><DD>&nbsp;<DT><A HREF="../nextgen/core/annotation/filter/OverlapFilter.html#OverlapFilter(nextgen.core.annotation.Annotation)"><B>OverlapFilter(Annotation)</B></A> - 
Constructor for class nextgen.core.annotation.filter.<A HREF="../nextgen/core/annotation/filter/OverlapFilter.html" title="class in nextgen.core.annotation.filter">OverlapFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/filter/OverlapFilter.html#OverlapFilter(java.util.List)"><B>OverlapFilter(List&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.filter.<A HREF="../nextgen/core/annotation/filter/OverlapFilter.html" title="class in nextgen.core.annotation.filter">OverlapFilter</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/filter/OverlapFilter.html#OverlapFilter(nextgen.core.annotation.AnnotationList)"><B>OverlapFilter(AnnotationList&lt;? extends Annotation&gt;)</B></A> - 
Constructor for class nextgen.core.annotation.filter.<A HREF="../nextgen/core/annotation/filter/OverlapFilter.html" title="class in nextgen.core.annotation.filter">OverlapFilter</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#overlappers(int, int)"><B>overlappers(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>Return an iterator over all intervals overlapping the specified range.
<DT><A HREF="../broad/core/datastructures/IntervalTree.html#overlappingValueIterator(int, int)"><B>overlappingValueIterator(int, int)</B></A> - 
Method in class broad.core.datastructures.<A HREF="../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#overlaps(java.util.Collection, int)"><B>overlaps(Collection&lt;? extends Annotation&gt;, int)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/annotation/BasicLightweightAnnotation.html#overlaps(java.util.Collection)"><B>overlaps(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.annotation.<A HREF="../broad/core/annotation/BasicLightweightAnnotation.html" title="class in broad.core.annotation">BasicLightweightAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MAFAlignment.html#overlaps(broad.core.annotation.LightweightGenomicAnnotation)"><B>overlaps(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MAFAlignment.html" title="class in broad.core.multiplealignment">MAFAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/multiplealignment/MultipleAlignment.html#overlaps(broad.core.sequence.SequenceRegion)"><B>overlaps(SequenceRegion)</B></A> - 
Method in class broad.core.multiplealignment.<A HREF="../broad/core/multiplealignment/MultipleAlignment.html" title="class in broad.core.multiplealignment">MultipleAlignment</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#overlaps(broad.core.annotation.LightweightGenomicAnnotation, int)"><B>overlaps(LightweightGenomicAnnotation, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#overlaps(broad.core.annotation.LightweightGenomicAnnotation)"><B>overlaps(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#overlaps(java.util.Collection, int)"><B>overlaps(Collection&lt;? extends Annotation&gt;, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#overlaps(java.util.Collection)"><B>overlaps(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#overlaps(nextgen.core.annotation.Annotation, int)"><B>overlaps(Annotation, int)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#overlaps(nextgen.core.annotation.Annotation)"><B>overlaps(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#overlaps(nextgen.core.annotation.Annotation, int, boolean)"><B>overlaps(Annotation, int, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#overlaps(nextgen.core.annotation.Annotation, boolean)"><B>overlaps(Annotation, boolean)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#overlaps(nextgen.core.annotation.Gene)"><B>overlaps(Gene)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#overlaps(broad.pda.seq.graph.Path)"><B>overlaps(Path)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#overlaps(nextgen.core.annotation.Annotation)"><B>overlaps(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#overlaps(nextgen.core.annotation.Annotation, int)"><B>overlaps(Annotation, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#overlaps(nextgen.core.annotation.Annotation, boolean)"><B>overlaps(Annotation, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#overlaps(nextgen.core.annotation.Annotation, int, boolean)"><B>overlaps(Annotation, int, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#overlaps(java.util.Collection)"><B>overlaps(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#overlaps(java.util.Collection, int)"><B>overlaps(Collection&lt;? extends Annotation&gt;, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#overlaps(nextgen.core.annotation.Annotation, int)"><B>overlaps(Annotation, int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#overlaps(java.util.Collection, int)"><B>overlaps(Collection&lt;? extends Annotation&gt;, int)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#overlaps(nextgen.core.annotation.Annotation)"><B>overlaps(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#overlaps(java.util.Collection)"><B>overlaps(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#overlaps(nextgen.core.annotation.Annotation, int, boolean)"><B>overlaps(Annotation, int, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#overlaps(nextgen.core.annotation.Annotation, boolean)"><B>overlaps(Annotation, boolean)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#overlaps(nextgen.core.annotation.Annotation, int, boolean)"><B>overlaps(Annotation, int, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>COMPARISON METHODS
<DT><A HREF="../nextgen/core/annotation/BasicAnnotation.html#overlaps(java.util.List, int)"><B>overlaps(List&lt;? extends Annotation&gt;, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/BasicAnnotation.html" title="class in nextgen.core.annotation">BasicAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#overlaps(nextgen.core.annotation.CompoundInterval)"><B>overlaps(CompoundInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#overlaps(nextgen.core.annotation.CompoundInterval, int, boolean)"><B>overlaps(CompoundInterval, int, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#overlaps(nextgen.core.annotation.CompoundInterval, boolean)"><B>overlaps(CompoundInterval, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#overlaps(nextgen.core.annotation.CompoundInterval, int)"><B>overlaps(CompoundInterval, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#overlaps(nextgen.core.annotation.SingleInterval)"><B>overlaps(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/CompoundInterval.html#overlaps(nextgen.core.annotation.SingleInterval, int)"><B>overlaps(SingleInterval, int)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/CompoundInterval.html" title="class in nextgen.core.annotation">CompoundInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlaps(nextgen.core.annotation.Gene)"><B>overlaps(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Whether an exon of this gene overlaps an exon of other gene and the two genes have the same orientation
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlaps(nextgen.core.annotation.Gene, boolean)"><B>overlaps(Gene, boolean)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Whether an exon of this gene overlaps an exon of other gene
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlaps(nextgen.core.annotation.Gene, double)"><B>overlaps(Gene, double)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Returns true if this and the other gene overlap but at least minPctOverlap
<DT><A HREF="../nextgen/core/annotation/SingleInterval.html#overlaps(nextgen.core.annotation.SingleInterval)"><B>overlaps(SingleInterval)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/SingleInterval.html" title="class in nextgen.core.annotation">SingleInterval</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#overlaps(nextgen.core.annotation.Annotation, int)"><B>overlaps(Annotation, int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#overlaps(java.util.Collection, int)"><B>overlaps(Collection&lt;? extends Annotation&gt;, int)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#overlaps(nextgen.core.annotation.Annotation)"><B>overlaps(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#overlaps(java.util.Collection)"><B>overlaps(Collection&lt;? extends Annotation&gt;)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#overlaps(nextgen.core.annotation.Annotation, int, boolean)"><B>overlaps(Annotation, int, boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#overlaps(nextgen.core.annotation.Annotation, boolean)"><B>overlaps(Annotation, boolean)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlaps2(broad.core.annotation.LightweightGenomicAnnotation)"><B>overlaps2(LightweightGenomicAnnotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#overlapsAtAll(broad.pda.datastructures.Alignments)"><B>overlapsAtAll(Alignments)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#overlapsAtAll(java.util.Set)"><B>overlapsAtAll(Set&lt;Alignments&gt;)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#overlapsByGenomicRegion(broad.core.datastructures.IntervalTree, boolean)"><B>overlapsByGenomicRegion(IntervalTree&lt;GeneWithIsoforms&gt;, boolean)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/datastructures/Alignments.html#overlapsCollection(java.util.Collection)"><B>overlapsCollection(Collection&lt;Alignments&gt;)</B></A> - 
Method in class broad.pda.datastructures.<A HREF="../broad/pda/datastructures/Alignments.html" title="class in broad.pda.datastructures">Alignments</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#overlapsExon(broad.core.annotation.LightweightGenomicAnnotation)"><B>overlapsExon(LightweightGenomicAnnotation)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/gene/GeneWithIsoforms.html#overlapsExon(nextgen.core.annotation.Gene)"><B>overlapsExon(Gene)</B></A> - 
Method in class broad.pda.gene.<A HREF="../broad/pda/gene/GeneWithIsoforms.html" title="class in broad.pda.gene">GeneWithIsoforms</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlapsExon(nextgen.core.annotation.Annotation)"><B>overlapsExon(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlapsExon(broad.core.annotation.LightweightGenomicAnnotation)"><B>overlapsExon(LightweightGenomicAnnotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlapsGene(java.util.Collection)"><B>overlapsGene(Collection&lt;Gene&gt;)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Whether this gene overlaps any gene in the collection at the exon level
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlapsGene(nextgen.core.annotation.Gene)"><B>overlapsGene(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlapsGeneInAnyOrientation(nextgen.core.annotation.Gene)"><B>overlapsGeneInAnyOrientation(Gene)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#overlapsLeft(nextgen.core.alignment.Alignment)"><B>overlapsLeft(Alignment)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/pda/seq/graph/Path.html#overlapsRight(nextgen.core.alignment.Alignment)"><B>overlapsRight(Alignment)</B></A> - 
Method in class broad.pda.seq.graph.<A HREF="../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/sequence/SequenceRegion.html#overlapsStranded(nextgen.core.annotation.Annotation)"><B>overlapsStranded(Annotation)</B></A> - 
Method in class broad.core.sequence.<A HREF="../broad/core/sequence/SequenceRegion.html" title="class in broad.core.sequence">SequenceRegion</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/AbstractAnnotation.html#overlapsStranded(nextgen.core.annotation.Annotation)"><B>overlapsStranded(Annotation)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/AbstractAnnotation.html" title="class in nextgen.core.annotation">AbstractAnnotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Annotation.html#overlapsStranded(nextgen.core.annotation.Annotation)"><B>overlapsStranded(Annotation)</B></A> - 
Method in interface nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>
<DD>&nbsp;
<DT><A HREF="../nextgen/core/annotation/Gene.html#overlapsStranded(nextgen.core.annotation.Gene, double)"><B>overlapsStranded(Gene, double)</B></A> - 
Method in class nextgen.core.annotation.<A HREF="../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>
<DD>Returns true if this and the other gene overlap but at least minPctOverlap
<DT><A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html#overlapsStranded(nextgen.core.annotation.Annotation)"><B>overlapsStranded(Annotation)</B></A> - 
Method in class nextgen.core.scripture.<A HREF="../nextgen/core/scripture/OrientedChromosomeTranscriptGraph.TranscriptGraphEdge.html" title="class in nextgen.core.scripture">OrientedChromosomeTranscriptGraph.TranscriptGraphEdge</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#overLengthPenaltyWeight"><B>overLengthPenaltyWeight</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#overMeltingTempPenaltyWeight"><B>overMeltingTempPenaltyWeight</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
<DT><A HREF="../broad/core/primer3/Primer3Configuration.html#overProductSizePenaltyWeight"><B>overProductSizePenaltyWeight</B></A> - 
Variable in class broad.core.primer3.<A HREF="../broad/core/primer3/Primer3Configuration.html" title="class in broad.core.primer3">Primer3Configuration</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-14.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-16.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-15.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-15.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">Z</A> <HR>

</BODY>
</HTML>
