<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:23 EDT 2013 -->
<TITLE>
MultiSampleScanPeakCaller
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="MultiSampleScanPeakCaller";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/MultiSampleScanPeakCaller.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../broad/pda/seq/protection/GenomeSpaceSampleData.html" title="class in broad.pda.seq.protection"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/protection/MultiSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/protection/MultiSampleScanPeakCaller.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="MultiSampleScanPeakCaller.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
broad.pda.seq.protection</FONT>
<BR>
Class MultiSampleScanPeakCaller</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><B>broad.pda.seq.protection.MultiSampleScanPeakCaller</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD><A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></DD>
</DL>
<DL>
<DT><B>Direct Known Subclasses:</B> <DD><A HREF="../../../../broad/pda/seq/protection/BatchedMultiSampleScanPeakCaller.html" title="class in broad.pda.seq.protection">BatchedMultiSampleScanPeakCaller</A></DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>MultiSampleScanPeakCaller</B><DT>extends java.lang.Object<DT>implements <A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></DL>
</PRE>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>prussell</DD>
</DL>
<HR>

<P>

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#MultiSampleScanPeakCaller(java.lang.String, java.lang.String, java.lang.String, int, int)">MultiSampleScanPeakCaller</A></B>(java.lang.String&nbsp;sampleListFile,
                          java.lang.String&nbsp;bedFile,
                          java.lang.String&nbsp;chrSizeFile,
                          int&nbsp;window,
                          int&nbsp;step)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Instantiate using file of sample information</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#MultiSampleScanPeakCaller(java.lang.String, java.lang.String, java.lang.String, java.lang.String)">MultiSampleScanPeakCaller</A></B>(java.lang.String&nbsp;expressionBamFile,
                          java.lang.String&nbsp;signalBamFile,
                          java.lang.String&nbsp;bedFile,
                          java.lang.String&nbsp;chrSizeFile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Instantiate with a single sample and an expression sample and default paramters</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#MultiSampleScanPeakCaller(java.lang.String, java.lang.String, java.lang.String, java.lang.String, int, int)">MultiSampleScanPeakCaller</A></B>(java.lang.String&nbsp;expressionBamFile,
                          java.lang.String&nbsp;signalBamFile,
                          java.lang.String&nbsp;bedFile,
                          java.lang.String&nbsp;chrSizeFile,
                          int&nbsp;window,
                          int&nbsp;step)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Instantiate with a single signal sample and an expression sample</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#filterWindows(java.util.Collection)">filterWindows</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;windows)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Filter windows that do not contain peaks</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#getCoordinateSpace()">getCoordinateSpace</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the coordinate space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#getSingleSampleScanPeaks(broad.pda.seq.protection.SampleData, nextgen.core.annotation.Gene)">getSingleSampleScanPeaks</A></B>(<A HREF="../../../../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>&nbsp;sample,
                         <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scan peaks for the sample and the gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#isExpressed(nextgen.core.annotation.Gene)">isExpressed</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the gene is significantly expressed in the special expression sample</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#isExpressedInAllControlSamples(nextgen.core.annotation.Gene)">isExpressedInAllControlSamples</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the gene is expressed in all control samples at the given significance level</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#main(java.lang.String[])">main</A></B>(java.lang.String[]&nbsp;args)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#mergePeaks(java.util.Collection)">mergePeaks</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;peaks)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Merge a collection of peaks</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#scoreWindows(java.util.Collection)">scoreWindows</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;windows)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Score windows</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setCoordinateSpace(nextgen.core.coordinatesystem.CoordinateSpace)">setCoordinateSpace</A></B>(<A HREF="../../../../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>&nbsp;space)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set the coordinate space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setExpressionScanPvalueCutoff(double)">setExpressionScanPvalueCutoff</A></B>(double&nbsp;expressionScanPvalCutoff)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set genome wide scan P value cutoff for expression of transcript</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setFilterByStrand(boolean)">setFilterByStrand</A></B>(boolean&nbsp;useStrandFilter)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set whether to use strand information in reads to filter peaks</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setFirstReadTranscriptionStrand(boolean)">setFirstReadTranscriptionStrand</A></B>(boolean&nbsp;firstReadIsTranscriptionStrand)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set whether read 1 is transcription strand</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setPeakCutoffMostCommonReplicate(double)">setPeakCutoffMostCommonReplicate</A></B>(double&nbsp;maxPct)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set cutoff for the percentage of fragments overlapping a peak that come from the most common replicate fragment</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setPeakTrimQuantile(double)">setPeakTrimQuantile</A></B>(double&nbsp;trimPeakQuantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set quantile for trim max contiguous algorithm for peak calling</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setPeakWindowCountCutoff(double)">setPeakWindowCountCutoff</A></B>(double&nbsp;peakCountCutoff)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set minimum number of fragments overlapping a window to be considered for possible peak</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#setPeakWindowScanPvalCutoff(double)">setPeakWindowScanPvalCutoff</A></B>(double&nbsp;peakScanPvalCutoff)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set scan P value cutoff for window within transcript to be considered for possible peak</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#trimPeak(nextgen.core.annotation.Annotation)">trimPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;peak)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Trim coarse peak to optimal coordinates</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#writeResult(java.util.Collection, java.io.FileWriter)">writeResult</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;windows,
            java.io.FileWriter&nbsp;writer)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write one peak to a file stream</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/MultiSampleScanPeakCaller.html#writeResults(java.util.Collection, java.lang.String)">writeResults</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;windows,
             java.lang.String&nbsp;out)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a set of peaks to a file</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="MultiSampleScanPeakCaller(java.lang.String, java.lang.String, java.lang.String, int, int)"><!-- --></A><H3>
MultiSampleScanPeakCaller</H3>
<PRE>
public <B>MultiSampleScanPeakCaller</B>(java.lang.String&nbsp;sampleListFile,
                                 java.lang.String&nbsp;bedFile,
                                 java.lang.String&nbsp;chrSizeFile,
                                 int&nbsp;window,
                                 int&nbsp;step)
                          throws java.io.IOException</PRE>
<DL>
<DD>Instantiate using file of sample information
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>sampleListFile</CODE> - File containing sample list<DD><CODE>bedFile</CODE> - Bed gene annotation<DD><CODE>chrSizeFile</CODE> - Chromosome size file<DD><CODE>window</CODE> - Window size<DD><CODE>step</CODE> - Step size
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="MultiSampleScanPeakCaller(java.lang.String, java.lang.String, java.lang.String, java.lang.String)"><!-- --></A><H3>
MultiSampleScanPeakCaller</H3>
<PRE>
public <B>MultiSampleScanPeakCaller</B>(java.lang.String&nbsp;expressionBamFile,
                                 java.lang.String&nbsp;signalBamFile,
                                 java.lang.String&nbsp;bedFile,
                                 java.lang.String&nbsp;chrSizeFile)
                          throws java.io.IOException</PRE>
<DL>
<DD>Instantiate with a single sample and an expression sample and default paramters
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>expressionBamFile</CODE> - Bam file for expression sample<DD><CODE>signalBamFile</CODE> - Bam file for signal sample<DD><CODE>bedFile</CODE> - Bed gene annotation<DD><CODE>chrSizeFile</CODE> - Chromosome size file
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="MultiSampleScanPeakCaller(java.lang.String, java.lang.String, java.lang.String, java.lang.String, int, int)"><!-- --></A><H3>
MultiSampleScanPeakCaller</H3>
<PRE>
public <B>MultiSampleScanPeakCaller</B>(java.lang.String&nbsp;expressionBamFile,
                                 java.lang.String&nbsp;signalBamFile,
                                 java.lang.String&nbsp;bedFile,
                                 java.lang.String&nbsp;chrSizeFile,
                                 int&nbsp;window,
                                 int&nbsp;step)
                          throws java.io.IOException</PRE>
<DL>
<DD>Instantiate with a single signal sample and an expression sample
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>expressionBamFile</CODE> - Bam file for expression sample<DD><CODE>signalBamFile</CODE> - Bam file for signal sample<DD><CODE>bedFile</CODE> - Bed gene annotation<DD><CODE>chrSizeFile</CODE> - Chromosome size file<DD><CODE>window</CODE> - Window size<DD><CODE>step</CODE> - Step size
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="setFilterByStrand(boolean)"><!-- --></A><H3>
setFilterByStrand</H3>
<PRE>
public void <B>setFilterByStrand</B>(boolean&nbsp;useStrandFilter)</PRE>
<DL>
<DD>Set whether to use strand information in reads to filter peaks
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>useStrandFilter</CODE> - Whether to apply strand filter</DL>
</DD>
</DL>
<HR>

<A NAME="setPeakCutoffMostCommonReplicate(double)"><!-- --></A><H3>
setPeakCutoffMostCommonReplicate</H3>
<PRE>
public void <B>setPeakCutoffMostCommonReplicate</B>(double&nbsp;maxPct)</PRE>
<DL>
<DD>Set cutoff for the percentage of fragments overlapping a peak that come from the most common replicate fragment
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>maxPct</CODE> - The max percentage</DL>
</DD>
</DL>
<HR>

<A NAME="setPeakTrimQuantile(double)"><!-- --></A><H3>
setPeakTrimQuantile</H3>
<PRE>
public void <B>setPeakTrimQuantile</B>(double&nbsp;trimPeakQuantile)</PRE>
<DL>
<DD>Set quantile for trim max contiguous algorithm for peak calling
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>trimPeakQuantile</CODE> - The quantile of data values</DL>
</DD>
</DL>
<HR>

<A NAME="setPeakWindowCountCutoff(double)"><!-- --></A><H3>
setPeakWindowCountCutoff</H3>
<PRE>
public void <B>setPeakWindowCountCutoff</B>(double&nbsp;peakCountCutoff)</PRE>
<DL>
<DD>Set minimum number of fragments overlapping a window to be considered for possible peak
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>peakCountCutoff</CODE> - Minimum fragment count for each window</DL>
</DD>
</DL>
<HR>

<A NAME="setPeakWindowScanPvalCutoff(double)"><!-- --></A><H3>
setPeakWindowScanPvalCutoff</H3>
<PRE>
public void <B>setPeakWindowScanPvalCutoff</B>(double&nbsp;peakScanPvalCutoff)</PRE>
<DL>
<DD>Set scan P value cutoff for window within transcript to be considered for possible peak
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>peakScanPvalCutoff</CODE> - Max scan P value</DL>
</DD>
</DL>
<HR>

<A NAME="setFirstReadTranscriptionStrand(boolean)"><!-- --></A><H3>
setFirstReadTranscriptionStrand</H3>
<PRE>
public void <B>setFirstReadTranscriptionStrand</B>(boolean&nbsp;firstReadIsTranscriptionStrand)</PRE>
<DL>
<DD>Set whether read 1 is transcription strand
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>firstReadIsTranscriptionStrand</CODE> - True if read 1 is transcription strand</DL>
</DD>
</DL>
<HR>

<A NAME="setExpressionScanPvalueCutoff(double)"><!-- --></A><H3>
setExpressionScanPvalueCutoff</H3>
<PRE>
public void <B>setExpressionScanPvalueCutoff</B>(double&nbsp;expressionScanPvalCutoff)</PRE>
<DL>
<DD>Set genome wide scan P value cutoff for expression of transcript
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>expressionScanPvalCutoff</CODE> - P value cutoff for transcript expression against genomic background</DL>
</DD>
</DL>
<HR>

<A NAME="isExpressedInAllControlSamples(nextgen.core.annotation.Gene)"><!-- --></A><H3>
isExpressedInAllControlSamples</H3>
<PRE>
public boolean <B>isExpressedInAllControlSamples</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</PRE>
<DL>
<DD>Whether the gene is expressed in all control samples at the given significance level
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene
<DT><B>Returns:</B><DD>Whether the gene is expressed by these criteria</DL>
</DD>
</DL>
<HR>

<A NAME="isExpressed(nextgen.core.annotation.Gene)"><!-- --></A><H3>
isExpressed</H3>
<PRE>
public boolean <B>isExpressed</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</PRE>
<DL>
<DD>Whether the gene is significantly expressed in the special expression sample
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene
<DT><B>Returns:</B><DD>True iff the gene is expressed in the expression sample</DL>
</DD>
</DL>
<HR>

<A NAME="getSingleSampleScanPeaks(broad.pda.seq.protection.SampleData, nextgen.core.annotation.Gene)"><!-- --></A><H3>
getSingleSampleScanPeaks</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>getSingleSampleScanPeaks</B>(<A HREF="../../../../broad/pda/seq/protection/SampleData.html" title="class in broad.pda.seq.protection">SampleData</A>&nbsp;sample,
                                                                 <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)
                                                          throws java.io.IOException</PRE>
<DL>
<DD>Get scan peaks for the sample and the gene
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>sample</CODE> - The sample<DD><CODE>gene</CODE> - The gene
<DT><B>Returns:</B><DD>Significant scan peaks
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreWindows(java.util.Collection)"><!-- --></A><H3>
scoreWindows</H3>
<PRE>
public void <B>scoreWindows</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;windows)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#scoreWindows(java.util.Collection)">PeakCaller</A></CODE></B></DD>
<DD>Score windows
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#scoreWindows(java.util.Collection)">scoreWindows</A></CODE> in interface <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>windows</CODE> - Windows to score</DL>
</DD>
</DL>
<HR>

<A NAME="filterWindows(java.util.Collection)"><!-- --></A><H3>
filterWindows</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>filterWindows</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;windows)
                                               throws java.io.IOException</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#filterWindows(java.util.Collection)">PeakCaller</A></CODE></B></DD>
<DD>Filter windows that do not contain peaks
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#filterWindows(java.util.Collection)">filterWindows</A></CODE> in interface <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>windows</CODE> - Windows to filter
<DT><B>Returns:</B><DD>Filtered collection of windows
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="mergePeaks(java.util.Collection)"><!-- --></A><H3>
mergePeaks</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>mergePeaks</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;peaks)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#mergePeaks(java.util.Collection)">PeakCaller</A></CODE></B></DD>
<DD>Merge a collection of peaks
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#mergePeaks(java.util.Collection)">mergePeaks</A></CODE> in interface <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>peaks</CODE> - Peaks to merge
<DT><B>Returns:</B><DD>Collection of merged peaks</DL>
</DD>
</DL>
<HR>

<A NAME="trimPeak(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
trimPeak</H3>
<PRE>
public <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A> <B>trimPeak</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;peak)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#trimPeak(nextgen.core.annotation.Annotation)">PeakCaller</A></CODE></B></DD>
<DD>Trim coarse peak to optimal coordinates
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#trimPeak(nextgen.core.annotation.Annotation)">trimPeak</A></CODE> in interface <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>peak</CODE> - The peak
<DT><B>Returns:</B><DD>The trimmed peak</DL>
</DD>
</DL>
<HR>

<A NAME="writeResults(java.util.Collection, java.lang.String)"><!-- --></A><H3>
writeResults</H3>
<PRE>
public void <B>writeResults</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;windows,
                         java.lang.String&nbsp;out)
                  throws java.io.IOException</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#writeResults(java.util.Collection, java.lang.String)">PeakCaller</A></CODE></B></DD>
<DD>Write a set of peaks to a file
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#writeResults(java.util.Collection, java.lang.String)">writeResults</A></CODE> in interface <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>windows</CODE> - Regions to write<DD><CODE>out</CODE> - Output file name
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeResult(java.util.Collection, java.io.FileWriter)"><!-- --></A><H3>
writeResult</H3>
<PRE>
public void <B>writeResult</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;windows,
                        java.io.FileWriter&nbsp;writer)
                 throws java.io.IOException</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#writeResult(java.util.Collection, java.io.FileWriter)">PeakCaller</A></CODE></B></DD>
<DD>Write one peak to a file stream
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#writeResult(java.util.Collection, java.io.FileWriter)">writeResult</A></CODE> in interface <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>windows</CODE> - Region to write<DD><CODE>writer</CODE> - File writer
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="setCoordinateSpace(nextgen.core.coordinatesystem.CoordinateSpace)"><!-- --></A><H3>
setCoordinateSpace</H3>
<PRE>
public void <B>setCoordinateSpace</B>(<A HREF="../../../../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A>&nbsp;space)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#setCoordinateSpace(nextgen.core.coordinatesystem.CoordinateSpace)">PeakCaller</A></CODE></B></DD>
<DD>Set the coordinate space
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#setCoordinateSpace(nextgen.core.coordinatesystem.CoordinateSpace)">setCoordinateSpace</A></CODE> in interface <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>space</CODE> - The coordinate space to use</DL>
</DD>
</DL>
<HR>

<A NAME="getCoordinateSpace()"><!-- --></A><H3>
getCoordinateSpace</H3>
<PRE>
public <A HREF="../../../../nextgen/core/coordinatesystem/CoordinateSpace.html" title="interface in nextgen.core.coordinatesystem">CoordinateSpace</A> <B>getCoordinateSpace</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#getCoordinateSpace()">PeakCaller</A></CODE></B></DD>
<DD>Get the coordinate space
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html#getCoordinateSpace()">getCoordinateSpace</A></CODE> in interface <CODE><A HREF="../../../../nextgen/core/analysis/PeakCaller.html" title="interface in nextgen.core.analysis">PeakCaller</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The coordinate space</DL>
</DD>
</DL>
<HR>

<A NAME="main(java.lang.String[])"><!-- --></A><H3>
main</H3>
<PRE>
public static void <B>main</B>(java.lang.String[]&nbsp;args)
                 throws java.io.IOException,
                        java.lang.InterruptedException</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>args</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE>
<DD><CODE>java.lang.InterruptedException</CODE></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/MultiSampleScanPeakCaller.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../broad/pda/seq/protection/GenomeSpaceSampleData.html" title="class in broad.pda.seq.protection"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/protection/MultiSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/protection/MultiSampleScanPeakCaller.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="MultiSampleScanPeakCaller.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
