<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:23 EDT 2013 -->
<TITLE>
SampleData
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SampleData";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/SampleData.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../broad/pda/seq/protection/ReadSimulator.html" title="class in broad.pda.seq.protection"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/protection/SampleData.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="SampleData.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
broad.pda.seq.protection</FONT>
<BR>
Class SampleData</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><B>broad.pda.seq.protection.SampleData</B>
</PRE>
<DL>
<DT><B>Direct Known Subclasses:</B> <DD><A HREF="../../../../broad/pda/seq/protection/GenomeSpaceSampleData.html" title="class in broad.pda.seq.protection">GenomeSpaceSampleData</A></DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>SampleData</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>prussell</DD>
</DL>
<HR>

<P>

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#SampleData(java.lang.String, boolean, java.util.Map, int, int, double, boolean)">SampleData</A></B>(java.lang.String&nbsp;bamFile,
           boolean&nbsp;firstReadTranscriptionStrand,
           java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
           int&nbsp;window,
           int&nbsp;step,
           double&nbsp;expressionCutoff,
           boolean&nbsp;expByScanPval)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#firstReadTranscriptionStrand()">firstReadTranscriptionStrand</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether read 1 is in direction of transcription</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getData()">getData</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the alignment data</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getEnrichmentOverGene(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)">getEnrichmentOverGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                      <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get enrichment of a window over a gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getExpressionScanPvalueCutoff()">getExpressionScanPvalueCutoff</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get genome wide scan P value cutoff for expression of transcript</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getGeneAverageCoverage(nextgen.core.annotation.Gene)">getGeneAverageCoverage</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get average coverage of gene
 Get score from cache or calculate and cache score</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getGeneCount(nextgen.core.annotation.Gene)">getGeneCount</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of fragments mapping to gene
 Get score from cache or calculate and cache score</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getGeneScanPval(nextgen.core.annotation.Gene)">getGeneScanPval</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get coordinate space wide scan P value of number of fragments mapping to the gene
 Get score from cache or calculate and cache score</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;org.apache.log4j.Logger</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getLogger()">getLogger</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the logger</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getOriginalBamFile()">getOriginalBamFile</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get name of original bam file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getSampleName()">getSampleName</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the sample name</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getWindowCount(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)">getWindowCount</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
               <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get number of fragments mapping to gene window
 Get score from cache or calculate and cache score</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#getWindowScores(nextgen.core.annotation.Gene)">getWindowScores</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scores for each window in the gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#gotWindowScoresFromFile()">gotWindowScoresFromFile</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the window scores were read from an existing file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#hashCode()">hashCode</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#isExpressed(nextgen.core.annotation.Gene)">isExpressed</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the gene is significantly expressed in the sample</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#scoreWindow(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)">scoreWindow</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
            <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scan statistic score for a specified window</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#scoreWindow(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation, double)">scoreWindow</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
            <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window,
            double&nbsp;count)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scan statistic score for a specified window</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#scoreWindowWithFragmentLengthFilter(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)">scoreWindowWithFragmentLengthFilter</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                    <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get scan statistic score for a window based on filtering fragments overlapping the window by size</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#setExpressionScanPvalueCutoff(double)">setExpressionScanPvalueCutoff</A></B>(double&nbsp;expressionScanPvalCutoff)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set genome wide scan P value cutoff for expression of transcript</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#setSampleName(java.lang.String)">setSampleName</A></B>(java.lang.String&nbsp;name)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set the sample name</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/protection/SampleData.html#trimMaxContiguous(nextgen.core.annotation.Annotation, java.util.List, double)">trimMaxContiguous</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window,
                  java.util.List&lt;java.lang.Double&gt;&nbsp;data,
                  double&nbsp;quantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Trim the region to max contiguous subregion above a certain quantile</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="SampleData(java.lang.String, boolean, java.util.Map, int, int, double, boolean)"><!-- --></A><H3>
SampleData</H3>
<PRE>
public <B>SampleData</B>(java.lang.String&nbsp;bamFile,
                  boolean&nbsp;firstReadTranscriptionStrand,
                  java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genes,
                  int&nbsp;window,
                  int&nbsp;step,
                  double&nbsp;expressionCutoff,
                  boolean&nbsp;expByScanPval)
           throws java.io.IOException</PRE>
<DL>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamFile</CODE> - Bam file<DD><CODE>firstReadTranscriptionStrand</CODE> - Whether read 1 is in direction of transcription<DD><CODE>genes</CODE> - Genes by chromosome<DD><CODE>window</CODE> - Window size<DD><CODE>step</CODE> - Step size<DD><CODE>expressionCutoff</CODE> - P value cutoff for scan test of gene expression<DD><CODE>expByScanPval</CODE> - Expression is assessed by scan P value. If false, uses average read depth
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="hashCode()"><!-- --></A><H3>
hashCode</H3>
<PRE>
public int <B>hashCode</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>hashCode</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setExpressionScanPvalueCutoff(double)"><!-- --></A><H3>
setExpressionScanPvalueCutoff</H3>
<PRE>
public void <B>setExpressionScanPvalueCutoff</B>(double&nbsp;expressionScanPvalCutoff)</PRE>
<DL>
<DD>Set genome wide scan P value cutoff for expression of transcript
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>expressionScanPvalCutoff</CODE> - P value cutoff for transcript expression against genomic background</DL>
</DD>
</DL>
<HR>

<A NAME="getExpressionScanPvalueCutoff()"><!-- --></A><H3>
getExpressionScanPvalueCutoff</H3>
<PRE>
public double <B>getExpressionScanPvalueCutoff</B>()</PRE>
<DL>
<DD>Get genome wide scan P value cutoff for expression of transcript
<P>
<DD><DL>

<DT><B>Returns:</B><DD>P value cutoff for transcript expression against genomic background</DL>
</DD>
</DL>
<HR>

<A NAME="getGeneCount(nextgen.core.annotation.Gene)"><!-- --></A><H3>
getGeneCount</H3>
<PRE>
public double <B>getGeneCount</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</PRE>
<DL>
<DD>Get number of fragments mapping to gene
 Get score from cache or calculate and cache score
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene
<DT><B>Returns:</B><DD>The number of fragments mapping to the gene</DL>
</DD>
</DL>
<HR>

<A NAME="getGeneScanPval(nextgen.core.annotation.Gene)"><!-- --></A><H3>
getGeneScanPval</H3>
<PRE>
public double <B>getGeneScanPval</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</PRE>
<DL>
<DD>Get coordinate space wide scan P value of number of fragments mapping to the gene
 Get score from cache or calculate and cache score
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene
<DT><B>Returns:</B><DD>The scan P value of the number of fragments mapping to the gene with respect to teh coordinate space</DL>
</DD>
</DL>
<HR>

<A NAME="getGeneAverageCoverage(nextgen.core.annotation.Gene)"><!-- --></A><H3>
getGeneAverageCoverage</H3>
<PRE>
public double <B>getGeneAverageCoverage</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</PRE>
<DL>
<DD>Get average coverage of gene
 Get score from cache or calculate and cache score
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene
<DT><B>Returns:</B><DD>The average coverage of the gene</DL>
</DD>
</DL>
<HR>

<A NAME="isExpressed(nextgen.core.annotation.Gene)"><!-- --></A><H3>
isExpressed</H3>
<PRE>
public boolean <B>isExpressed</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</PRE>
<DL>
<DD>Whether the gene is significantly expressed in the sample
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene
<DT><B>Returns:</B><DD>Whether the gene is expressed at the given significance level</DL>
</DD>
</DL>
<HR>

<A NAME="getWindowCount(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)"><!-- --></A><H3>
getWindowCount</H3>
<PRE>
public double <B>getWindowCount</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                             <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</PRE>
<DL>
<DD>Get number of fragments mapping to gene window
 Get score from cache or calculate and cache score
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene<DD><CODE>window</CODE> - The window
<DT><B>Returns:</B><DD>The number of fragments mapping to the gene</DL>
</DD>
</DL>
<HR>

<A NAME="firstReadTranscriptionStrand()"><!-- --></A><H3>
firstReadTranscriptionStrand</H3>
<PRE>
public boolean <B>firstReadTranscriptionStrand</B>()</PRE>
<DL>
<DD>Whether read 1 is in direction of transcription
<P>
<DD><DL>

<DT><B>Returns:</B><DD>True iff first read transcription strand is true</DL>
</DD>
</DL>
<HR>

<A NAME="getWindowScores(nextgen.core.annotation.Gene)"><!-- --></A><H3>
getWindowScores</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,<A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A>&gt; <B>getWindowScores</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</PRE>
<DL>
<DD>Get scores for each window in the gene
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene
<DT><B>Returns:</B><DD>The scores by window</DL>
</DD>
</DL>
<HR>

<A NAME="gotWindowScoresFromFile()"><!-- --></A><H3>
gotWindowScoresFromFile</H3>
<PRE>
public boolean <B>gotWindowScoresFromFile</B>()</PRE>
<DL>
<DD>Whether the window scores were read from an existing file
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Whether the window scores were read from an existing file</DL>
</DD>
</DL>
<HR>

<A NAME="getEnrichmentOverGene(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)"><!-- --></A><H3>
getEnrichmentOverGene</H3>
<PRE>
public double <B>getEnrichmentOverGene</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                    <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</PRE>
<DL>
<DD>Get enrichment of a window over a gene
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene<DD><CODE>window</CODE> - Window contained in the gene
<DT><B>Returns:</B><DD>Enrichment of window over gene background</DL>
</DD>
</DL>
<HR>

<A NAME="scoreWindowWithFragmentLengthFilter(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)"><!-- --></A><H3>
scoreWindowWithFragmentLengthFilter</H3>
<PRE>
public <A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A> <B>scoreWindowWithFragmentLengthFilter</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                                              <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</PRE>
<DL>
<DD>Get scan statistic score for a window based on filtering fragments overlapping the window by size
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - The gene<DD><CODE>window</CODE> - The window
<DT><B>Returns:</B><DD>Scan statistic score for window with global stats referring to parent transcript</DL>
</DD>
</DL>
<HR>

<A NAME="scoreWindow(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation)"><!-- --></A><H3>
scoreWindow</H3>
<PRE>
public <A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A> <B>scoreWindow</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                      <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window)</PRE>
<DL>
<DD>Get scan statistic score for a specified window
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - Parent gene<DD><CODE>window</CODE> - The window
<DT><B>Returns:</B><DD>Scan statistic score for window with global stats referring to parent transcript</DL>
</DD>
</DL>
<HR>

<A NAME="scoreWindow(nextgen.core.annotation.Gene, nextgen.core.annotation.Annotation, double)"><!-- --></A><H3>
scoreWindow</H3>
<PRE>
public <A HREF="../../../../nextgen/core/model/score/ScanStatisticScore.html" title="class in nextgen.core.model.score">ScanStatisticScore</A> <B>scoreWindow</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                      <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window,
                                      double&nbsp;count)</PRE>
<DL>
<DD>Get scan statistic score for a specified window
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene</CODE> - Parent gene<DD><CODE>window</CODE> - The window<DD><CODE>count</CODE> - The number to use as the region count in score
<DT><B>Returns:</B><DD>Scan statistic score for window with global stats referring to parent transcript</DL>
</DD>
</DL>
<HR>

<A NAME="getLogger()"><!-- --></A><H3>
getLogger</H3>
<PRE>
public org.apache.log4j.Logger <B>getLogger</B>()</PRE>
<DL>
<DD>Get the logger
<P>
<DD><DL>

<DT><B>Returns:</B><DD>The logger</DL>
</DD>
</DL>
<HR>

<A NAME="getSampleName()"><!-- --></A><H3>
getSampleName</H3>
<PRE>
public java.lang.String <B>getSampleName</B>()</PRE>
<DL>
<DD>Get the sample name
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Sample name</DL>
</DD>
</DL>
<HR>

<A NAME="getOriginalBamFile()"><!-- --></A><H3>
getOriginalBamFile</H3>
<PRE>
public java.lang.String <B>getOriginalBamFile</B>()</PRE>
<DL>
<DD>Get name of original bam file
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Bam file name</DL>
</DD>
</DL>
<HR>

<A NAME="setSampleName(java.lang.String)"><!-- --></A><H3>
setSampleName</H3>
<PRE>
public void <B>setSampleName</B>(java.lang.String&nbsp;name)</PRE>
<DL>
<DD>Set the sample name
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>name</CODE> - Sample name</DL>
</DD>
</DL>
<HR>

<A NAME="getData()"><!-- --></A><H3>
getData</H3>
<PRE>
public <A HREF="../../../../nextgen/core/model/TranscriptomeSpaceAlignmentModel.html" title="class in nextgen.core.model">TranscriptomeSpaceAlignmentModel</A> <B>getData</B>()</PRE>
<DL>
<DD>Get the alignment data
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Alignment model</DL>
</DD>
</DL>
<HR>

<A NAME="trimMaxContiguous(nextgen.core.annotation.Annotation, java.util.List, double)"><!-- --></A><H3>
trimMaxContiguous</H3>
<PRE>
public static <A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A> <B>trimMaxContiguous</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;window,
                                           java.util.List&lt;java.lang.Double&gt;&nbsp;data,
                                           double&nbsp;quantile)</PRE>
<DL>
<DD>Trim the region to max contiguous subregion above a certain quantile
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>window</CODE> - The region<DD><CODE>data</CODE> - Position level list of counts within the region<DD><CODE>quantile</CODE> - Quantile for trim max contiguous
<DT><B>Returns:</B><DD>Trimmed region</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/SampleData.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../broad/pda/seq/protection/ReadSimulator.html" title="class in broad.pda.seq.protection"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/protection/TwoSampleScanSkellamAnalysis.html" title="class in broad.pda.seq.protection"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/protection/SampleData.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="SampleData.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
