<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:23 EDT 2013 -->
<TITLE>
GenomeCommandLineProgram
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="GenomeCommandLineProgram";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/GenomeCommandLineProgram.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../broad/pda/seq/rap/GCContent.html" title="class in broad.pda.seq.rap"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/rap/GenomeCommandLineProgram.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="GenomeCommandLineProgram.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
broad.pda.seq.rap</FONT>
<BR>
Class GenomeCommandLineProgram</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../resources/inherit.gif" ALT="extended by ">net.sf.picard.cmdline.CommandLineProgram
      <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><B>broad.pda.seq.rap.GenomeCommandLineProgram</B>
</PRE>
<DL>
<DT><B>Direct Known Subclasses:</B> <DD><A HREF="../../../../broad/pda/seq/rap/BuildRatioNullDistribution.html" title="class in broad.pda.seq.rap">BuildRatioNullDistribution</A>, <A HREF="../../../../broad/pda/seq/rap/CollectAnnotationEnrichments.html" title="class in broad.pda.seq.rap">CollectAnnotationEnrichments</A>, <A HREF="../../../../broad/pda/seq/rap/GCContent.html" title="class in broad.pda.seq.rap">GCContent</A>, <A HREF="../../../../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap">GenomeScoringProgram</A>, <A HREF="../../../../broad/pda/seq/rap/RatioPermutationPeakCaller.html" title="class in broad.pda.seq.rap">RatioPermutationPeakCaller</A>, <A HREF="../../../../broad/pda/seq/rap/ShuffleBAM.html" title="class in broad.pda.seq.rap">ShuffleBAM</A>, <A HREF="../../../../broad/pda/seq/rap/ShuffleBED.html" title="class in broad.pda.seq.rap">ShuffleBED</A>, <A HREF="../../../../broad/pda/seq/rap/SkellamKLPeakCaller.html" title="class in broad.pda.seq.rap">SkellamKLPeakCaller</A></DD>
</DL>
<HR>
<DL>
<DT><PRE>public abstract class <B>GenomeCommandLineProgram</B><DT>extends net.sf.picard.cmdline.CommandLineProgram</DL>
</PRE>

<P>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#MASK_FILE">MASK_FILE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Integer</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#MAX_FRAGMENT_LENGTH">MAX_FRAGMENT_LENGTH</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Integer</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#MIN_MAPPING_QUALITY">MIN_MAPPING_QUALITY</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#PCT_MASKED_ALLOWED">PCT_MASKED_ALLOWED</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#REGION">REGION</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#SIZES">SIZES</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="fields_inherited_from_class_net.sf.picard.cmdline.CommandLineProgram"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Fields inherited from class net.sf.picard.cmdline.CommandLineProgram</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>COMPRESSION_LEVEL, CREATE_INDEX, CREATE_MD5_FILE, MAX_RECORDS_IN_RAM, QUIET, TMP_DIR, VALIDATION_STRINGENCY, VERBOSITY</CODE></TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#GenomeCommandLineProgram()">GenomeCommandLineProgram</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#getRegions()">getRegions</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#getRegionSet()">getRegionSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#getRegionsMap()">getRegionsMap</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#loadAlignmentModel(java.io.File)">loadAlignmentModel</A></B>(java.io.File&nbsp;bamFile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/rap/GenomeCommandLineProgram.html#loadAlignmentModel(java.io.File, boolean)">loadAlignmentModel</A></B>(java.io.File&nbsp;bamFile,
                   boolean&nbsp;pairedEnd)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_net.sf.picard.cmdline.CommandLineProgram"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class net.sf.picard.cmdline.CommandLineProgram</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>getCommandLine, getCommandLineParser, getDefaultHeaders, getProgramVersion, getStandardUsagePreamble, instanceMain, instanceMainWithExit, setDefaultHeaders</CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="SIZES"><!-- --></A><H3>
SIZES</H3>
<PRE>
<FONT SIZE="-1">@Option(doc="File specifying chromosome sizes.")
</FONT>public java.lang.String <B>SIZES</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="MASK_FILE"><!-- --></A><H3>
MASK_FILE</H3>
<PRE>
<FONT SIZE="-1">@Option(doc="File containing masked regions.",
        optional=true)
</FONT>public java.lang.String <B>MASK_FILE</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="PCT_MASKED_ALLOWED"><!-- --></A><H3>
PCT_MASKED_ALLOWED</H3>
<PRE>
<FONT SIZE="-1">@Option(doc="Percent masked allowable per sliding window",
        optional=true)
</FONT>public double <B>PCT_MASKED_ALLOWED</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="REGION"><!-- --></A><H3>
REGION</H3>
<PRE>
<FONT SIZE="-1">@Option(doc="Region to process (e.g. chr1, chr1:5000-50230).  Default (null) processes the entire genome",
        optional=true)
</FONT>public java.lang.String <B>REGION</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="MAX_FRAGMENT_LENGTH"><!-- --></A><H3>
MAX_FRAGMENT_LENGTH</H3>
<PRE>
<FONT SIZE="-1">@Option(doc="Maximum paired-end read fragment length to consider.",
        optional=true)
</FONT>public java.lang.Integer <B>MAX_FRAGMENT_LENGTH</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="MIN_MAPPING_QUALITY"><!-- --></A><H3>
MIN_MAPPING_QUALITY</H3>
<PRE>
<FONT SIZE="-1">@Option(doc="Minimum mapping quality for reads")
</FONT>public java.lang.Integer <B>MIN_MAPPING_QUALITY</B></PRE>
<DL>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="GenomeCommandLineProgram()"><!-- --></A><H3>
GenomeCommandLineProgram</H3>
<PRE>
public <B>GenomeCommandLineProgram</B>()</PRE>
<DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getRegions()"><!-- --></A><H3>
getRegions</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>getRegions</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getRegionsMap()"><!-- --></A><H3>
getRegionsMap</H3>
<PRE>
public java.util.Map&lt;java.lang.String,<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>getRegionsMap</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getRegionSet()"><!-- --></A><H3>
getRegionSet</H3>
<PRE>
public <A HREF="../../../../nextgen/core/annotation/AnnotationList.html" title="class in nextgen.core.annotation">AnnotationList</A>&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>getRegionSet</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="loadAlignmentModel(java.io.File)"><!-- --></A><H3>
loadAlignmentModel</H3>
<PRE>
public <A HREF="../../../../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A> <B>loadAlignmentModel</B>(java.io.File&nbsp;bamFile)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="loadAlignmentModel(java.io.File, boolean)"><!-- --></A><H3>
loadAlignmentModel</H3>
<PRE>
public <A HREF="../../../../nextgen/core/model/AlignmentModel.html" title="class in nextgen.core.model">AlignmentModel</A> <B>loadAlignmentModel</B>(java.io.File&nbsp;bamFile,
                                         boolean&nbsp;pairedEnd)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/GenomeCommandLineProgram.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../broad/pda/seq/rap/GCContent.html" title="class in broad.pda.seq.rap"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/rap/GenomeScoringProgram.html" title="class in broad.pda.seq.rap"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/rap/GenomeCommandLineProgram.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="GenomeCommandLineProgram.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
