<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:27 EDT 2013 -->
<TITLE>
LibraryCompositionByRnaClass
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="LibraryCompositionByRnaClass";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/LibraryCompositionByRnaClass.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads"><B>PREV CLASS</B></A>&nbsp;
&nbsp;NEXT CLASS</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?broad/pda/countreads/LibraryCompositionByRnaClass.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="LibraryCompositionByRnaClass.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
broad.pda.countreads</FONT>
<BR>
Class LibraryCompositionByRnaClass</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>broad.pda.countreads.LibraryCompositionByRnaClass</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>LibraryCompositionByRnaClass</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>prussell</DD>
</DL>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/pda/countreads/LibraryCompositionByRnaClass.html#GENOME_CLASS_NAME">GENOME_CLASS_NAME</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The name used for the genome when counting read mappings</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/pda/countreads/LibraryCompositionByRnaClass.html#MYSTERY_READS_CLASS_NAME">MYSTERY_READS_CLASS_NAME</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The name used for unmapped reads when counting read mappings</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../broad/pda/countreads/LibraryCompositionByRnaClass.html#LibraryCompositionByRnaClass(java.lang.String, java.util.Map, java.util.Map, java.util.Map, org.apache.log4j.Logger)">LibraryCompositionByRnaClass</A></B>(java.lang.String&nbsp;genomeBowtieIndexBase,
                             java.util.Map&lt;java.lang.String,java.lang.String&gt;&nbsp;rnaClasses,
                             java.util.Map&lt;java.lang.String,java.lang.String&gt;&nbsp;read1fastqs,
                             java.util.Map&lt;java.lang.String,java.lang.String&gt;&nbsp;read2fastqs,
                             org.apache.log4j.Logger&nbsp;log)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor establishes locations of files and counts total reads per sample</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/pda/countreads/LibraryCompositionByRnaClass.html#alignAndGetCounts(java.lang.String, java.lang.String, java.util.Map, java.lang.String, java.lang.String)">alignAndGetCounts</A></B>(java.lang.String&nbsp;samtoolsExecutable,
                  java.lang.String&nbsp;bowtie2Executable,
                  java.util.Map&lt;java.lang.String,java.lang.String&gt;&nbsp;bowtie2options,
                  java.lang.String&nbsp;bowtie2BuildExecutable,
                  java.lang.String&nbsp;logDir)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For each sample, get counts of reads mapping to each RNA class as well as genome and unmapped</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.lang.Integer&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/pda/countreads/LibraryCompositionByRnaClass.html#getTotalReadCounts()">getTotalReadCounts</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the total number of reads per sample</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../broad/pda/countreads/LibraryCompositionByRnaClass.html#readsPaired()">readsPaired</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Whether the reads are paired</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="GENOME_CLASS_NAME"><!-- --></A><H3>
GENOME_CLASS_NAME</H3>
<PRE>
public static java.lang.String <B>GENOME_CLASS_NAME</B></PRE>
<DL>
<DD>The name used for the genome when counting read mappings
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="MYSTERY_READS_CLASS_NAME"><!-- --></A><H3>
MYSTERY_READS_CLASS_NAME</H3>
<PRE>
public static java.lang.String <B>MYSTERY_READS_CLASS_NAME</B></PRE>
<DL>
<DD>The name used for unmapped reads when counting read mappings
<P>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="LibraryCompositionByRnaClass(java.lang.String, java.util.Map, java.util.Map, java.util.Map, org.apache.log4j.Logger)"><!-- --></A><H3>
LibraryCompositionByRnaClass</H3>
<PRE>
public <B>LibraryCompositionByRnaClass</B>(java.lang.String&nbsp;genomeBowtieIndexBase,
                                    java.util.Map&lt;java.lang.String,java.lang.String&gt;&nbsp;rnaClasses,
                                    java.util.Map&lt;java.lang.String,java.lang.String&gt;&nbsp;read1fastqs,
                                    java.util.Map&lt;java.lang.String,java.lang.String&gt;&nbsp;read2fastqs,
                                    org.apache.log4j.Logger&nbsp;log)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor establishes locations of files and counts total reads per sample
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>genomeBowtieIndexBase</CODE> - Genome fasta file<DD><CODE>rnaClasses</CODE> - Map associating RNA class name with fasta file of sequences<DD><CODE>read1fastqs</CODE> - Fastq file containing read1 if paired or single end reads if unpaired<DD><CODE>read2fastqs</CODE> - Fastq file containing read2 if paired or null if unpaired<DD><CODE>log</CODE> - Logger object
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getTotalReadCounts()"><!-- --></A><H3>
getTotalReadCounts</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.lang.Integer&gt; <B>getTotalReadCounts</B>()</PRE>
<DL>
<DD>Get the total number of reads per sample
<P>
<DD><DL>

<DT><B>Returns:</B><DD>Map associating sample name with read count; regardless of whether reads are paired, the count is total single reads</DL>
</DD>
</DL>
<HR>

<A NAME="readsPaired()"><!-- --></A><H3>
readsPaired</H3>
<PRE>
public boolean <B>readsPaired</B>()</PRE>
<DL>
<DD>Whether the reads are paired
<P>
<DD><DL>

<DT><B>Returns:</B><DD>True if paired, false if unpaired</DL>
</DD>
</DL>
<HR>

<A NAME="alignAndGetCounts(java.lang.String, java.lang.String, java.util.Map, java.lang.String, java.lang.String)"><!-- --></A><H3>
alignAndGetCounts</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&gt; <B>alignAndGetCounts</B>(java.lang.String&nbsp;samtoolsExecutable,
                                                                                                           java.lang.String&nbsp;bowtie2Executable,
                                                                                                           java.util.Map&lt;java.lang.String,java.lang.String&gt;&nbsp;bowtie2options,
                                                                                                           java.lang.String&nbsp;bowtie2BuildExecutable,
                                                                                                           java.lang.String&nbsp;logDir)
                                                                                                    throws java.io.IOException,
                                                                                                           java.lang.InterruptedException</PRE>
<DL>
<DD>For each sample, get counts of reads mapping to each RNA class as well as genome and unmapped
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>samtoolsExecutable</CODE> - Samtools executable file<DD><CODE>bowtie2Executable</CODE> - Bowtie2 executable file<DD><CODE>bowtie2options</CODE> - <DD><CODE>bowtie2BuildExecutable</CODE> - Bowtie2-build executable file<DD><CODE>logDir</CODE> - Output directory for logs and alignments
<DT><B>Returns:</B><DD>Map from sample name to class name to count
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE>
<DD><CODE>java.lang.InterruptedException</CODE></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/LibraryCompositionByRnaClass.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../broad/pda/countreads/FastqLibraryStats.html" title="class in broad.pda.countreads"><B>PREV CLASS</B></A>&nbsp;
&nbsp;NEXT CLASS</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?broad/pda/countreads/LibraryCompositionByRnaClass.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="LibraryCompositionByRnaClass.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
