<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:27 EDT 2013 -->
<TITLE>
WigWriter
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="WigWriter";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/WigWriter.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers"><B>PREV CLASS</B></A>&nbsp;
&nbsp;NEXT CLASS</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?nextgen/core/writers/WigWriter.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="WigWriter.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
nextgen.core.writers</FONT>
<BR>
Class WigWriter</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>nextgen.core.writers.WigWriter</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>WigWriter</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>prussell</DD>
</DL>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#BEGINNING_POSITION_DESCRIPTION">BEGINNING_POSITION_DESCRIPTION</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Description of fragment beginning position calculation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#MIDPOINT_POSITION_DESCRIPTION">MIDPOINT_POSITION_DESCRIPTION</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Description of fragment midpoint position calculation</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#WigWriter(java.lang.String, java.util.Map, java.lang.String, boolean, boolean)">WigWriter</A></B>(java.lang.String&nbsp;bamFile,
          java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genesByChrName,
          java.lang.String&nbsp;singlePositionOnly,
          boolean&nbsp;useFragments,
          boolean&nbsp;nor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct with a transcriptome space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#WigWriter(java.lang.String, java.util.Map, java.lang.String, boolean, boolean, java.lang.String)">WigWriter</A></B>(java.lang.String&nbsp;bamFile,
          java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genesByChrName,
          java.lang.String&nbsp;singlePositionOnly,
          boolean&nbsp;useFragments,
          boolean&nbsp;nor,
          java.lang.String&nbsp;chrToWrite)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct with a transcriptome space and only write one chromosome</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#WigWriter(java.lang.String, java.lang.String, java.lang.String, boolean, boolean)">WigWriter</A></B>(java.lang.String&nbsp;bamFile,
          java.lang.String&nbsp;chrSizeFile,
          java.lang.String&nbsp;singlePositionOnly,
          boolean&nbsp;useFragments,
          boolean&nbsp;nor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct with a genome space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#WigWriter(java.lang.String, java.lang.String, java.lang.String, boolean, boolean, java.lang.String)">WigWriter</A></B>(java.lang.String&nbsp;bamFile,
          java.lang.String&nbsp;chrSizeFile,
          java.lang.String&nbsp;singlePositionOnly,
          boolean&nbsp;useFragments,
          boolean&nbsp;nor,
          java.lang.String&nbsp;chrToWrite)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct with a genome space and only write one chromosome</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#WigWriter(java.lang.String, java.lang.String, java.lang.String, java.lang.String, boolean, boolean)">WigWriter</A></B>(java.lang.String&nbsp;bamFile,
          java.lang.String&nbsp;geneBedFile,
          java.lang.String&nbsp;chrSizeFile,
          java.lang.String&nbsp;singlePositionOnly,
          boolean&nbsp;useFragments,
          boolean&nbsp;nor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct with either a genomic space or a transcriptome space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#WigWriter(java.lang.String, java.lang.String, java.lang.String, java.lang.String, boolean, boolean, java.lang.String)">WigWriter</A></B>(java.lang.String&nbsp;bamFile,
          java.lang.String&nbsp;geneBedFile,
          java.lang.String&nbsp;chrSizeFile,
          java.lang.String&nbsp;singlePositionOnly,
          boolean&nbsp;useFragments,
          boolean&nbsp;nor,
          java.lang.String&nbsp;chrToWrite)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct with either a genomic space or a transcriptome space and only write one chromosome</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#addReadFilter(org.apache.commons.collections15.Predicate)">addReadFilter</A></B>(org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;filter)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add a read filter to alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#coordinateToWigPosition(int)">coordinateToWigPosition</A></B>(int&nbsp;zeroBasedCoordinate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the one based wig format position for a zero based coordinate</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#main(java.lang.String[])">main</A></B>(java.lang.String[]&nbsp;args)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#wigPositionToCoordinate(int)">wigPositionToCoordinate</A></B>(int&nbsp;wigPosition)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the zero based coordinate for a wig position</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#write(java.io.FileWriter, java.lang.String, java.util.Map)">write</A></B>(java.io.FileWriter&nbsp;w,
      java.lang.String&nbsp;chr,
      java.util.Map&lt;java.lang.Integer,java.lang.Double&gt;&nbsp;counts)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a set of counts to an existing file writer</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../nextgen/core/writers/WigWriter.html#writeFullWig(java.lang.String)">writeFullWig</A></B>(java.lang.String&nbsp;outFilePrefix)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write all counts to a wig file</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="BEGINNING_POSITION_DESCRIPTION"><!-- --></A><H3>
BEGINNING_POSITION_DESCRIPTION</H3>
<PRE>
public static java.lang.String <B>BEGINNING_POSITION_DESCRIPTION</B></PRE>
<DL>
<DD>Description of fragment beginning position calculation
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="MIDPOINT_POSITION_DESCRIPTION"><!-- --></A><H3>
MIDPOINT_POSITION_DESCRIPTION</H3>
<PRE>
public static java.lang.String <B>MIDPOINT_POSITION_DESCRIPTION</B></PRE>
<DL>
<DD>Description of fragment midpoint position calculation
<P>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="WigWriter(java.lang.String, java.util.Map, java.lang.String, boolean, boolean)"><!-- --></A><H3>
WigWriter</H3>
<PRE>
public <B>WigWriter</B>(java.lang.String&nbsp;bamFile,
                 java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genesByChrName,
                 java.lang.String&nbsp;singlePositionOnly,
                 boolean&nbsp;useFragments,
                 boolean&nbsp;nor)</PRE>
<DL>
<DD>Construct with a transcriptome space
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamFile</CODE> - Bam alignments<DD><CODE>genesByChrName</CODE> - Genes by reference sequence name<DD><CODE>singlePositionOnly</CODE> - Certain position of each read to count<DD><CODE>useFragments</CODE> - Whether to convert reads to fragments<DD><CODE>nor</CODE> - Normalize by global amounts</DL>
</DL>
<HR>

<A NAME="WigWriter(java.lang.String, java.util.Map, java.lang.String, boolean, boolean, java.lang.String)"><!-- --></A><H3>
WigWriter</H3>
<PRE>
public <B>WigWriter</B>(java.lang.String&nbsp;bamFile,
                 java.util.Map&lt;java.lang.String,java.util.Collection&lt;<A HREF="../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&gt;&nbsp;genesByChrName,
                 java.lang.String&nbsp;singlePositionOnly,
                 boolean&nbsp;useFragments,
                 boolean&nbsp;nor,
                 java.lang.String&nbsp;chrToWrite)</PRE>
<DL>
<DD>Construct with a transcriptome space and only write one chromosome
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamFile</CODE> - Bam alignments<DD><CODE>genesByChrName</CODE> - Genes by reference sequence name<DD><CODE>singlePositionOnly</CODE> - Certain position of each read to count<DD><CODE>useFragments</CODE> - Whether to convert reads to fragments<DD><CODE>nor</CODE> - Normalize by global amounts<DD><CODE>chrToWrite</CODE> - Only write this chromosome</DL>
</DL>
<HR>

<A NAME="WigWriter(java.lang.String, java.lang.String, java.lang.String, boolean, boolean)"><!-- --></A><H3>
WigWriter</H3>
<PRE>
public <B>WigWriter</B>(java.lang.String&nbsp;bamFile,
                 java.lang.String&nbsp;chrSizeFile,
                 java.lang.String&nbsp;singlePositionOnly,
                 boolean&nbsp;useFragments,
                 boolean&nbsp;nor)</PRE>
<DL>
<DD>Construct with a genome space
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamFile</CODE> - Bam alignments<DD><CODE>chrSizeFile</CODE> - Chromosome size file<DD><CODE>singlePositionOnly</CODE> - Certain position of each read to count<DD><CODE>useFragments</CODE> - Whether to convert reads to fragments<DD><CODE>nor</CODE> - Normalize by global amounts</DL>
</DL>
<HR>

<A NAME="WigWriter(java.lang.String, java.lang.String, java.lang.String, boolean, boolean, java.lang.String)"><!-- --></A><H3>
WigWriter</H3>
<PRE>
public <B>WigWriter</B>(java.lang.String&nbsp;bamFile,
                 java.lang.String&nbsp;chrSizeFile,
                 java.lang.String&nbsp;singlePositionOnly,
                 boolean&nbsp;useFragments,
                 boolean&nbsp;nor,
                 java.lang.String&nbsp;chrToWrite)</PRE>
<DL>
<DD>Construct with a genome space and only write one chromosome
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamFile</CODE> - Bam alignments<DD><CODE>chrSizeFile</CODE> - Chromosome size file<DD><CODE>singlePositionOnly</CODE> - Certain position of each read to count<DD><CODE>useFragments</CODE> - Whether to convert reads to fragments<DD><CODE>nor</CODE> - Normalize by global amounts<DD><CODE>chrToWrite</CODE> - Only write this chromosome</DL>
</DL>
<HR>

<A NAME="WigWriter(java.lang.String, java.lang.String, java.lang.String, java.lang.String, boolean, boolean)"><!-- --></A><H3>
WigWriter</H3>
<PRE>
public <B>WigWriter</B>(java.lang.String&nbsp;bamFile,
                 java.lang.String&nbsp;geneBedFile,
                 java.lang.String&nbsp;chrSizeFile,
                 java.lang.String&nbsp;singlePositionOnly,
                 boolean&nbsp;useFragments,
                 boolean&nbsp;nor)
          throws java.io.IOException</PRE>
<DL>
<DD>Construct with either a genomic space or a transcriptome space
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamFile</CODE> - Bam alignments<DD><CODE>geneBedFile</CODE> - Bed file of annotations for transcriptome space<DD><CODE>chrSizeFile</CODE> - File of chromosome sizes for genomic space<DD><CODE>singlePositionOnly</CODE> - Certain position of each read to count<DD><CODE>useFragments</CODE> - Whether to convert reads to fragments<DD><CODE>nor</CODE> - Normalize by global amounts
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="WigWriter(java.lang.String, java.lang.String, java.lang.String, java.lang.String, boolean, boolean, java.lang.String)"><!-- --></A><H3>
WigWriter</H3>
<PRE>
public <B>WigWriter</B>(java.lang.String&nbsp;bamFile,
                 java.lang.String&nbsp;geneBedFile,
                 java.lang.String&nbsp;chrSizeFile,
                 java.lang.String&nbsp;singlePositionOnly,
                 boolean&nbsp;useFragments,
                 boolean&nbsp;nor,
                 java.lang.String&nbsp;chrToWrite)
          throws java.io.IOException</PRE>
<DL>
<DD>Construct with either a genomic space or a transcriptome space and only write one chromosome
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>bamFile</CODE> - Bam alignments<DD><CODE>geneBedFile</CODE> - Bed file of annotations for transcriptome space<DD><CODE>chrSizeFile</CODE> - File of chromosome sizes for genomic space<DD><CODE>singlePositionOnly</CODE> - Certain position of each read to count<DD><CODE>useFragments</CODE> - Whether to convert reads to fragments<DD><CODE>nor</CODE> - Normalize by global amounts<DD><CODE>chrToWrite</CODE> - Only write this chromosome
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="addReadFilter(org.apache.commons.collections15.Predicate)"><!-- --></A><H3>
addReadFilter</H3>
<PRE>
public void <B>addReadFilter</B>(org.apache.commons.collections15.Predicate&lt;<A HREF="../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;filter)</PRE>
<DL>
<DD>Add a read filter to alignment model
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>filter</CODE> - The filter</DL>
</DD>
</DL>
<HR>

<A NAME="coordinateToWigPosition(int)"><!-- --></A><H3>
coordinateToWigPosition</H3>
<PRE>
public static int <B>coordinateToWigPosition</B>(int&nbsp;zeroBasedCoordinate)</PRE>
<DL>
<DD>Get the one based wig format position for a zero based coordinate
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>zeroBasedCoordinate</CODE> - The zero based coordinate
<DT><B>Returns:</B><DD>The wig position</DL>
</DD>
</DL>
<HR>

<A NAME="wigPositionToCoordinate(int)"><!-- --></A><H3>
wigPositionToCoordinate</H3>
<PRE>
public static int <B>wigPositionToCoordinate</B>(int&nbsp;wigPosition)</PRE>
<DL>
<DD>Get the zero based coordinate for a wig position
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>wigPosition</CODE> - The wig position
<DT><B>Returns:</B><DD>The zero based coordinate</DL>
</DD>
</DL>
<HR>

<A NAME="write(java.io.FileWriter, java.lang.String, java.util.Map)"><!-- --></A><H3>
write</H3>
<PRE>
public static void <B>write</B>(java.io.FileWriter&nbsp;w,
                         java.lang.String&nbsp;chr,
                         java.util.Map&lt;java.lang.Integer,java.lang.Double&gt;&nbsp;counts)
                  throws java.io.IOException</PRE>
<DL>
<DD>Write a set of counts to an existing file writer
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>w</CODE> - File writer<DD><CODE>chr</CODE> - Chromosome name<DD><CODE>counts</CODE> - Counts by position
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeFullWig(java.lang.String)"><!-- --></A><H3>
writeFullWig</H3>
<PRE>
public void <B>writeFullWig</B>(java.lang.String&nbsp;outFilePrefix)
                  throws java.io.IOException</PRE>
<DL>
<DD>Write all counts to a wig file
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outFilePrefix</CODE> - Output file prefix for wig
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="main(java.lang.String[])"><!-- --></A><H3>
main</H3>
<PRE>
public static void <B>main</B>(java.lang.String[]&nbsp;args)
                 throws java.io.IOException</PRE>
<DL>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>args</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/WigWriter.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../nextgen/core/writers/PairedEndWriter.html" title="class in nextgen.core.writers"><B>PREV CLASS</B></A>&nbsp;
&nbsp;NEXT CLASS</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?nextgen/core/writers/WigWriter.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="WigWriter.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
