<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_37) on Mon Jun 03 10:04:23 EDT 2013 -->
<TITLE>
ContinuousDataAlignmentModel
</TITLE>

<META NAME="date" CONTENT="2013-06-03">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="ContinuousDataAlignmentModel";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ContinuousDataAlignmentModel.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.ReadCountNormalizerToBaselineAlignment.html" title="class in broad.pda.seq.segmentation"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/DefineSourceAndSinkFromPairedEnds.html" title="class in broad.pda.seq.segmentation"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ContinuousDataAlignmentModel.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
broad.pda.seq.segmentation</FONT>
<BR>
Class ContinuousDataAlignmentModel</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><B>broad.pda.seq.segmentation.ContinuousDataAlignmentModel</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>ContinuousDataAlignmentModel</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#DEFAULT_INS_SIZE_PVAL">DEFAULT_INS_SIZE_PVAL</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#DEFAULT_INSERT_SIZE_FUDGE">DEFAULT_INSERT_SIZE_FUDGE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#DEFAULT_MAKE_GENE_OVERLAP">DEFAULT_MAKE_GENE_OVERLAP</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#DEFAULT_MIN_MAPPING_QUALITY">DEFAULT_MIN_MAPPING_QUALITY</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModel)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>&nbsp;data)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                             java.io.File[]&nbsp;maskFiles,
                             int&nbsp;minSpliceSupport)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                             java.io.File[]&nbsp;maskFiles,
                             int&nbsp;EF,
                             int&nbsp;minSpliceSupport)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int, broad.pda.seq.segmentation.AlignmentDataModelStats)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                             java.io.File[]&nbsp;maskFiles,
                             int&nbsp;EF,
                             int&nbsp;minSpliceSupport,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                             java.io.File[]&nbsp;maskFiles,
                             int&nbsp;EF,
                             int&nbsp;minSpliceSupport,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats, boolean)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                             java.io.File[]&nbsp;maskFiles,
                             int&nbsp;EF,
                             int&nbsp;minSpliceSupport,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads,
                             boolean&nbsp;upweight)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, int, int)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                             int&nbsp;EF,
                             int&nbsp;minSpliceSupport)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.util.Map, int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                             java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&nbsp;maskFileData,
                             int&nbsp;EF,
                             int&nbsp;minSpliceSupport,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.util.Map, int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats, boolean)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                             java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&nbsp;maskFileData,
                             int&nbsp;EF,
                             int&nbsp;minSpliceSupport,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads,
                             boolean&nbsp;upWeight)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.util.Map, int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats, int, boolean)">ContinuousDataAlignmentModel</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                             java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&nbsp;maskFileData,
                             int&nbsp;EF,
                             int&nbsp;minSpliceSupport,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                             <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads,
                             int&nbsp;chunkSize,
                             boolean&nbsp;upWeightSplices)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor to build the data alignment model</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#calculatePVal(int, double, double, double)">calculatePVal</A></B>(int&nbsp;k,
              double&nbsp;lambda,
              double&nbsp;w,
              double&nbsp;T)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#count(nextgen.core.annotation.Annotation)">count</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#CV(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">CV</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
   <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#filterGenesByExpression(java.util.Map)">filterGenesByExpression</A></B>(java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.List&lt;java.lang.Double&gt;&gt;&nbsp;countsMap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This function will filter the genes in the input Map by a number of constraints on the total number of reads on the genes
 1.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#filterPaths(java.util.Map, double)">filterPaths</A></B>(java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt;&nbsp;scores,
            double&nbsp;alpha2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getAlignmentDataModelStats()">getAlignmentDataModelStats</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getBasesCovered(nextgen.core.annotation.Annotation)">getBasesCovered</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;region)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getBasesCovered(nextgen.core.annotation.Annotation, int)">getBasesCovered</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;region,
                int&nbsp;EF)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getChromosomeLength(java.lang.String)">getChromosomeLength</A></B>(java.lang.String&nbsp;chr)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.lang.Integer&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getChromosomeLengths()">getChromosomeLengths</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getCount(java.util.Collection)">getCount</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;Annotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getCounts(java.util.Map, java.util.Collection)">getCounts</A></B>(java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&gt;&nbsp;geneTree,
          java.util.Collection&lt;java.lang.String&gt;&nbsp;multimappers)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getCountsPerBp(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree)">getCountsPerBp</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align,
               <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;java.lang.Double&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getCountsPerBp(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">getCountsPerBp</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
               <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getData()">getData</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the underlying AlignmentDataModelStats object</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getDataForAlignment(nextgen.core.annotation.Annotation)">getDataForAlignment</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This function returns an array of type double with counts for each base pair across the genomic region of the gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.List&lt;java.lang.Double&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getDataForGene(java.util.Collection)">getDataForGene</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;Annotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;java.lang.Double&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getDataForGene(nextgen.core.annotation.Gene)">getDataForGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneCounts(java.util.Collection)">getGeneCounts</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneCountsWithoutDatasetTotals(java.util.Collection)">getGeneCountsWithoutDatasetTotals</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getGeneExpression(java.util.Collection)">getGeneExpression</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getLambda(java.lang.String)">getLambda</A></B>(java.lang.String&nbsp;chr)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get an estimate of the expected read coverage per base</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getNumberMarkers(java.lang.String)">getNumberMarkers</A></B>(java.lang.String&nbsp;chr)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Gets the number of allowable bases per chr</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getNumberOfBPs()">getNumberOfBPs</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get total number of bases</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getNumberOfReads(java.lang.String)">getNumberOfReads</A></B>(java.lang.String&nbsp;chr)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Gets the total number of reads per chr</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getPairedData()">getPairedData</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">getPeak</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
        <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon,
        <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getSum(java.lang.String)">getSum</A></B>(java.lang.String&nbsp;chr)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Gets the total number of reads per chr</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#getWidths(java.lang.String)">getWidths</A></B>(java.lang.String&nbsp;str)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#hasDataForChromosome(java.lang.String)">hasDataForChromosome</A></B>(java.lang.String&nbsp;chr)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#main(java.lang.String[])">main</A></B>(java.lang.String[]&nbsp;args)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;java.lang.String,java.lang.Integer&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#parseMaskFiles(java.io.File[])">parseMaskFiles</A></B>(java.io.File[]&nbsp;files)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#permuteRegions(nextgen.core.annotation.Annotation)">permuteRegions</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#rawRegions(nextgen.core.annotation.Annotation)">rawRegions</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#resetTreeCache()">resetTreeCache</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scan(int[], double)">scan</A></B>(int[]&nbsp;windowSizes,
     double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scan(int[], double, java.lang.String)">scan</A></B>(int[]&nbsp;windowSizes,
     double&nbsp;alpha,
     java.lang.String&nbsp;chr)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanGenome(broad.pda.seq.segmentation.ContinuousDataAlignmentModel, int, java.lang.String, java.io.FileWriter, boolean)">scanGenome</A></B>(<A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>&nbsp;data2,
           int&nbsp;windowSize,
           java.lang.String&nbsp;chr,
           java.io.FileWriter&nbsp;writer,
           boolean&nbsp;filterSignificance)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)">scanPRate</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda,
          int&nbsp;numMarkers)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scanRegion(int, broad.core.annotation.LightweightGenomicAnnotation)">scanRegion</A></B>(int&nbsp;fixedWidth,
           <A HREF="../../../../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>&nbsp;region)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, double)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
          double&nbsp;localLambda)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)">scoreGene</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
          double&nbsp;localLambda,
          int&nbsp;numMarkers)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(broad.pda.annotation.BEDFileParser)">scoreGenes</A></B>(<A HREF="../../../../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
           java.lang.String&nbsp;chrToUse)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String, int)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
           java.lang.String&nbsp;chrToUse,
           int&nbsp;size)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreGenes(java.util.Collection, java.lang.String, broad.core.datastructures.IntervalTree)">scoreGenes</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
           java.lang.String&nbsp;chrToUse,
           <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scorePaths(java.util.Collection, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT, double)">scorePaths</A></B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths,
           <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>&nbsp;graph,
           double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scorePaths(java.util.Collection, double)">scorePaths</A></B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths,
           double&nbsp;lambda)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scorePaths(java.util.Collection, double, double, double, double, java.util.Map, double)">scorePaths</A></B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths,
           double&nbsp;lambda,
           double&nbsp;numMarkers,
           double&nbsp;numReads,
           double&nbsp;RPKMContant,
           java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;localRates,
           double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scorePathsForOnlyRPKM(java.util.Collection)">scorePathsForOnlyRPKM</A></B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreSegment(nextgen.core.annotation.Annotation)">scoreSegment</A></B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreSegments(java.util.Collection)">scoreSegments</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;set)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreSegments(java.util.Collection, java.lang.String)">scoreSegments</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;set,
              java.lang.String&nbsp;chrToUse)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,double[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#scoreSegments(java.util.Collection, java.lang.String, boolean)">scoreSegments</A></B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;set,
              java.lang.String&nbsp;chrToUse,
              boolean&nbsp;ignoreAlpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setAlpha(double)">setAlpha</A></B>(double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setExtensionFactor(int)">setExtensionFactor</A></B>(int&nbsp;extensionFactor)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setFindMaxContiguous(boolean)">setFindMaxContiguous</A></B>(boolean&nbsp;findMaxContiguous)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setMaskFileData(java.util.Map)">setMaskFileData</A></B>(java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&nbsp;maskedRegionMap)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setMinContguousSegmentSize(int)">setMinContguousSegmentSize</A></B>(int&nbsp;minSize)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setMinimumMappingQuality(double)">setMinimumMappingQuality</A></B>(double&nbsp;minMapQual)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setMinNumberOfSplices(int)">setMinNumberOfSplices</A></B>(int&nbsp;minSpliceSupport)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setTrimEnds(boolean)">setTrimEnds</A></B>(boolean&nbsp;trimEnds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setTrimQuantile(double)">setTrimQuantile</A></B>(double&nbsp;quantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#setUpWeightSplices(boolean)">setUpWeightSplices</A></B>(boolean&nbsp;setFlag)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#trimGeneEnds(nextgen.core.annotation.Gene, double)">trimGeneEnds</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
             double&nbsp;quantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#trimMaxContiguous(nextgen.core.annotation.Gene, double)">trimMaxContiguous</A></B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;region,
                  double&nbsp;quantile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Trim a region to a contiguous subregion with max number of positions having coverage above the quantile</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#writeConnections(java.lang.String, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT)">writeConnections</A></B>(java.lang.String&nbsp;save,
                 <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>&nbsp;graph)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#writeFullBED(java.lang.String, java.util.Collection)">writeFullBED</A></B>(java.lang.String&nbsp;save,
             java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;segments)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html#writeFullBED(java.lang.String, java.util.Map, double)">writeFullBED</A></B>(java.lang.String&nbsp;save,
             java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;&nbsp;map,
             double&nbsp;alpha)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="DEFAULT_MIN_MAPPING_QUALITY"><!-- --></A><H3>
DEFAULT_MIN_MAPPING_QUALITY</H3>
<PRE>
public static int <B>DEFAULT_MIN_MAPPING_QUALITY</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="DEFAULT_INSERT_SIZE_FUDGE"><!-- --></A><H3>
DEFAULT_INSERT_SIZE_FUDGE</H3>
<PRE>
public static int <B>DEFAULT_INSERT_SIZE_FUDGE</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="DEFAULT_INS_SIZE_PVAL"><!-- --></A><H3>
DEFAULT_INS_SIZE_PVAL</H3>
<PRE>
public static double <B>DEFAULT_INS_SIZE_PVAL</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="DEFAULT_MAKE_GENE_OVERLAP"><!-- --></A><H3>
DEFAULT_MAKE_GENE_OVERLAP</H3>
<PRE>
public static double <B>DEFAULT_MAKE_GENE_OVERLAP</B></PRE>
<DL>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModel)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModel.html" title="interface in broad.pda.seq.segmentation">AlignmentDataModel</A>&nbsp;data)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that stores the read data
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, int, int)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                                    int&nbsp;EF,
                                    int&nbsp;minSpliceSupport)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data<DD><CODE>EF</CODE> - an extension factor used to extend reads<DD><CODE>minSpliceSupport</CODE> - the minimum number of spliced reads needed to add a connection in the graph
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                                    java.io.File[]&nbsp;maskFiles,
                                    int&nbsp;minSpliceSupport)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data<DD><CODE>maskFiles</CODE> - the mask files<DD><CODE>minSpliceSupport</CODE> - the minimum number of spliced reads needed to add a connection in the graph
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                                    java.io.File[]&nbsp;maskFiles,
                                    int&nbsp;EF,
                                    int&nbsp;minSpliceSupport,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data<DD><CODE>maskFiles</CODE> - the mask files<DD><CODE>EF</CODE> - an extension factor used to extend reads<DD><CODE>minSpliceSupport</CODE> - the minimum number of spliced reads needed to add a connection in the graph<DD><CODE>pairedData</CODE> - the alignment data model that stores the stats for paired end reads and the paired end read data<DD><CODE>strandSpecificReads</CODE> - an alignment data model for the strand specific reads
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats, boolean)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                                    java.io.File[]&nbsp;maskFiles,
                                    int&nbsp;EF,
                                    int&nbsp;minSpliceSupport,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads,
                                    boolean&nbsp;upweight)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data<DD><CODE>maskFiles</CODE> - the mask files<DD><CODE>EF</CODE> - an extension factor used to extend reads<DD><CODE>minSpliceSupport</CODE> - the minimum number of spliced reads needed to add a connection in the graph<DD><CODE>pairedData</CODE> - the alignment data model that stores the stats for paired end reads and the paired end read data<DD><CODE>strandSpecificReads</CODE> - an alignment data model for the strand specific reads<DD><CODE>chunkSize</CODE> - the size of the cache chunk used
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.util.Map, int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                                    java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&nbsp;maskFileData,
                                    int&nbsp;EF,
                                    int&nbsp;minSpliceSupport,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data<DD><CODE>maskFileData</CODE> - a map of all unalignable regions of the genome<DD><CODE>EF</CODE> - an extension factor used to extend reads<DD><CODE>minSpliceSupport</CODE> - the minimum number of spliced reads needed to add a connection in the graph<DD><CODE>pairedData</CODE> - the alignment data model that stores the stats for paired end reads and the paired end read data<DD><CODE>strandSpecificReads</CODE> - an alignment data model for the strand specific reads
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.util.Map, int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats, boolean)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                                    java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&nbsp;maskFileData,
                                    int&nbsp;EF,
                                    int&nbsp;minSpliceSupport,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads,
                                    boolean&nbsp;upWeight)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data<DD><CODE>maskFileData</CODE> - a map of all unalignable regions of the genome<DD><CODE>EF</CODE> - an extension factor used to extend reads<DD><CODE>minSpliceSupport</CODE> - the minimum number of spliced reads needed to add a connection in the graph<DD><CODE>pairedData</CODE> - the alignment data model that stores the stats for paired end reads and the paired end read data<DD><CODE>strandSpecificReads</CODE> - an alignment data model for the strand specific reads<DD><CODE>upWeightSplices</CODE> - whether to weight spliced reads based on their proportioncoverage scores
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.util.Map, int, int, broad.pda.seq.segmentation.AlignmentDataModelStats, broad.pda.seq.segmentation.AlignmentDataModelStats, int, boolean)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                                    java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&nbsp;maskFileData,
                                    int&nbsp;EF,
                                    int&nbsp;minSpliceSupport,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;strandSpecificReads,
                                    int&nbsp;chunkSize,
                                    boolean&nbsp;upWeightSplices)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data<DD><CODE>maskFileData</CODE> - a map of all unalignable regions of the genome<DD><CODE>EF</CODE> - an extension factor used to extend reads<DD><CODE>minSpliceSupport</CODE> - the minimum number of spliced reads needed to add a connection in the graph<DD><CODE>pairedData</CODE> - the alignment data model that stores the stats for paired end reads and the paired end read data<DD><CODE>strandSpecificReads</CODE> - an alignment data model for the strand specific reads<DD><CODE>chunkSize</CODE> - the size of the cache chunk used<DD><CODE>upWeightSplices</CODE> - whether to weight spliced reads based on their proportion when computing coverage scores
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int, broad.pda.seq.segmentation.AlignmentDataModelStats)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                                    java.io.File[]&nbsp;maskFiles,
                                    int&nbsp;EF,
                                    int&nbsp;minSpliceSupport,
                                    <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;pairedData)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data<DD><CODE>maskFiles</CODE> - the mask files<DD><CODE>EF</CODE> - an extension factor used to extend reads<DD><CODE>minSpliceSupport</CODE> - the minimum number of spliced reads needed to add a connection in the graph<DD><CODE>pairedData</CODE> - the alignment data model that stores the stats for paired end reads and the paired end read data
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>
<HR>

<A NAME="ContinuousDataAlignmentModel(broad.pda.seq.segmentation.AlignmentDataModelStats, java.io.File[], int, int)"><!-- --></A><H3>
ContinuousDataAlignmentModel</H3>
<PRE>
public <B>ContinuousDataAlignmentModel</B>(<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A>&nbsp;data,
                                    java.io.File[]&nbsp;maskFiles,
                                    int&nbsp;EF,
                                    int&nbsp;minSpliceSupport)
                             throws java.io.IOException</PRE>
<DL>
<DD>Constructor to build the data alignment model
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>data</CODE> - the alignment data model that computes the statistics and stores the read data<DD><CODE>maskFiles</CODE> - the mask files<DD><CODE>EF</CODE> - an extension factor used to extend reads<DD><CODE>minSpliceSupport</CODE> - the minimum number of spliced reads needed to add a connection in the graph
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="setFindMaxContiguous(boolean)"><!-- --></A><H3>
setFindMaxContiguous</H3>
<PRE>
public void <B>setFindMaxContiguous</B>(boolean&nbsp;findMaxContiguous)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setTrimEnds(boolean)"><!-- --></A><H3>
setTrimEnds</H3>
<PRE>
public void <B>setTrimEnds</B>(boolean&nbsp;trimEnds)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setMinContguousSegmentSize(int)"><!-- --></A><H3>
setMinContguousSegmentSize</H3>
<PRE>
public void <B>setMinContguousSegmentSize</B>(int&nbsp;minSize)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setAlpha(double)"><!-- --></A><H3>
setAlpha</H3>
<PRE>
public void <B>setAlpha</B>(double&nbsp;alpha)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setUpWeightSplices(boolean)"><!-- --></A><H3>
setUpWeightSplices</H3>
<PRE>
public void <B>setUpWeightSplices</B>(boolean&nbsp;setFlag)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setMinNumberOfSplices(int)"><!-- --></A><H3>
setMinNumberOfSplices</H3>
<PRE>
public void <B>setMinNumberOfSplices</B>(int&nbsp;minSpliceSupport)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setExtensionFactor(int)"><!-- --></A><H3>
setExtensionFactor</H3>
<PRE>
public void <B>setExtensionFactor</B>(int&nbsp;extensionFactor)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setMaskFileData(java.util.Map)"><!-- --></A><H3>
setMaskFileData</H3>
<PRE>
public void <B>setMaskFileData</B>(java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&nbsp;maskedRegionMap)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setMinimumMappingQuality(double)"><!-- --></A><H3>
setMinimumMappingQuality</H3>
<PRE>
public void <B>setMinimumMappingQuality</B>(double&nbsp;minMapQual)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setTrimQuantile(double)"><!-- --></A><H3>
setTrimQuantile</H3>
<PRE>
public void <B>setTrimQuantile</B>(double&nbsp;quantile)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getData()"><!-- --></A><H3>
getData</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A> <B>getData</B>()</PRE>
<DL>
<DD>Get the underlying AlignmentDataModelStats object
<P>
<DD><DL>

<DT><B>Returns:</B><DD>the AlignmentDataModelStats object</DL>
</DD>
</DL>
<HR>

<A NAME="getLambda(java.lang.String)"><!-- --></A><H3>
getLambda</H3>
<PRE>
public double <B>getLambda</B>(java.lang.String&nbsp;chr)
                 throws java.io.IOException</PRE>
<DL>
<DD>Get an estimate of the expected read coverage per base
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>chr</CODE> - chromosome to retrieve
<DT><B>Returns:</B><DD>lambda
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getSum(java.lang.String)"><!-- --></A><H3>
getSum</H3>
<PRE>
public double <B>getSum</B>(java.lang.String&nbsp;chr)
              throws java.io.IOException</PRE>
<DL>
<DD>Gets the total number of reads per chr
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>chr</CODE> - chromosome to retrieve
<DT><B>Returns:</B><DD>total number of reads
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getNumberOfReads(java.lang.String)"><!-- --></A><H3>
getNumberOfReads</H3>
<PRE>
public double <B>getNumberOfReads</B>(java.lang.String&nbsp;chr)
                        throws java.io.IOException</PRE>
<DL>
<DD>Gets the total number of reads per chr
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>chr</CODE> - chromosome to retrieve
<DT><B>Returns:</B><DD>total number of reads
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getNumberMarkers(java.lang.String)"><!-- --></A><H3>
getNumberMarkers</H3>
<PRE>
public double <B>getNumberMarkers</B>(java.lang.String&nbsp;chr)
                        throws java.io.IOException</PRE>
<DL>
<DD>Gets the number of allowable bases per chr
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>chr</CODE> - chromosome to retrieve
<DT><B>Returns:</B><DD>total number of allowable bases
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getNumberOfBPs()"><!-- --></A><H3>
getNumberOfBPs</H3>
<PRE>
public double <B>getNumberOfBPs</B>()</PRE>
<DL>
<DD>Get total number of bases
<P>
<DD><DL>

<DT><B>Returns:</B><DD>total number of bases</DL>
</DD>
</DL>
<HR>

<A NAME="getCount(java.util.Collection)"><!-- --></A><H3>
getCount</H3>
<PRE>
public int <B>getCount</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;Annotation)
             throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getGeneCounts(java.util.Collection)"><!-- --></A><H3>
getGeneCounts</H3>
<PRE>
public int[] <B>getGeneCounts</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)
                    throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getGeneCountsWithoutDatasetTotals(java.util.Collection)"><!-- --></A><H3>
getGeneCountsWithoutDatasetTotals</H3>
<PRE>
public int[] <B>getGeneCountsWithoutDatasetTotals</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)
                                        throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getCountsPerBp(nextgen.core.annotation.Annotation, broad.core.datastructures.IntervalTree)"><!-- --></A><H3>
getCountsPerBp</H3>
<PRE>
public double[] <B>getCountsPerBp</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align,
                               <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getCountsPerBp(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><!-- --></A><H3>
getCountsPerBp</H3>
<PRE>
public java.util.List&lt;java.lang.Double&gt; <B>getCountsPerBp</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                                                       <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getGeneExpression(java.util.Collection)"><!-- --></A><H3>
getGeneExpression</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt; <B>getGeneExpression</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)
                                               throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getCounts(java.util.Map, java.util.Collection)"><!-- --></A><H3>
getCounts</H3>
<PRE>
public int <B>getCounts</B>(java.util.Map&lt;java.lang.String,<A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&gt;&nbsp;geneTree,
                     java.util.Collection&lt;java.lang.String&gt;&nbsp;multimappers)
              throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getChromosomeLength(java.lang.String)"><!-- --></A><H3>
getChromosomeLength</H3>
<PRE>
public int <B>getChromosomeLength</B>(java.lang.String&nbsp;chr)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getChromosomeLengths()"><!-- --></A><H3>
getChromosomeLengths</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.lang.Integer&gt; <B>getChromosomeLengths</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getWidths(java.lang.String)"><!-- --></A><H3>
getWidths</H3>
<PRE>
public static int[] <B>getWidths</B>(java.lang.String&nbsp;str)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getDataForAlignment(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
getDataForAlignment</H3>
<PRE>
public double[] <B>getDataForAlignment</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)
                             throws java.io.IOException</PRE>
<DL>
<DD>This function returns an array of type double with counts for each base pair across the genomic region of the gene
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>align</CODE> - 
<DT><B>Returns:</B><DD>
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getDataForGene(nextgen.core.annotation.Gene)"><!-- --></A><H3>
getDataForGene</H3>
<PRE>
public java.util.List&lt;java.lang.Double&gt; <B>getDataForGene</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)
                                                throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getDataForGene(java.util.Collection)"><!-- --></A><H3>
getDataForGene</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.List&lt;java.lang.Double&gt;&gt; <B>getDataForGene</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;Annotation)
                                                                    throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getPairedData()"><!-- --></A><H3>
getPairedData</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A> <B>getPairedData</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="scan(int[], double)"><!-- --></A><H3>
scan</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>scan</B>(int[]&nbsp;windowSizes,
                                             double&nbsp;alpha)
                                      throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scan(int[], double, java.lang.String)"><!-- --></A><H3>
scan</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>scan</B>(int[]&nbsp;windowSizes,
                                             double&nbsp;alpha,
                                             java.lang.String&nbsp;chr)
                                      throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scanGenome(broad.pda.seq.segmentation.ContinuousDataAlignmentModel, int, java.lang.String, java.io.FileWriter, boolean)"><!-- --></A><H3>
scanGenome</H3>
<PRE>
public void <B>scanGenome</B>(<A HREF="../../../../broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" title="class in broad.pda.seq.segmentation">ContinuousDataAlignmentModel</A>&nbsp;data2,
                       int&nbsp;windowSize,
                       java.lang.String&nbsp;chr,
                       java.io.FileWriter&nbsp;writer,
                       boolean&nbsp;filterSignificance)
                throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scanPRate(nextgen.core.annotation.Gene)"><!-- --></A><H3>
scanPRate</H3>
<PRE>
public double[] <B>scanPRate</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene)
                   throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><!-- --></A><H3>
scanPRate</H3>
<PRE>
public double[] <B>scanPRate</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)
                   throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double)"><!-- --></A><H3>
scanPRate</H3>
<PRE>
public double[] <B>scanPRate</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                          double&nbsp;localLambda)
                   throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scanPRate(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)"><!-- --></A><H3>
scanPRate</H3>
<PRE>
public double[] <B>scanPRate</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                          double&nbsp;localLambda,
                          int&nbsp;numMarkers)
                   throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scanRegion(int, broad.core.annotation.LightweightGenomicAnnotation)"><!-- --></A><H3>
scanRegion</H3>
<PRE>
public double[] <B>scanRegion</B>(int&nbsp;fixedWidth,
                           <A HREF="../../../../broad/core/annotation/LightweightGenomicAnnotation.html" title="interface in broad.core.annotation">LightweightGenomicAnnotation</A>&nbsp;region)
                    throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreSegment(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
scoreSegment</H3>
<PRE>
public double[] <B>scoreSegment</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)
                      throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreSegments(java.util.Collection)"><!-- --></A><H3>
scoreSegments</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,double[]&gt; <B>scoreSegments</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;set)
                                                 throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreSegments(java.util.Collection, java.lang.String)"><!-- --></A><H3>
scoreSegments</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,double[]&gt; <B>scoreSegments</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;set,
                                                        java.lang.String&nbsp;chrToUse)
                                                 throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreSegments(java.util.Collection, java.lang.String, boolean)"><!-- --></A><H3>
scoreSegments</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,double[]&gt; <B>scoreSegments</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt;&nbsp;set,
                                                        java.lang.String&nbsp;chrToUse,
                                                        boolean&nbsp;ignoreAlpha)
                                                 throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreGene(nextgen.core.annotation.Gene)"><!-- --></A><H3>
scoreGene</H3>
<PRE>
public double[] <B>scoreGene</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align)
                   throws java.io.IOException</PRE>
<DL>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>align</CODE> - 
<DT><B>Returns:</B><DD>{calculatePVal(new Double(sum).intValue(), localLambda, geneLength, getNumberMarkers(chr)), enrich, sum, avgCoverage, rpkm, localLambda, geneLength,nominalP, fullyContained}
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreGene(nextgen.core.annotation.Gene, double)"><!-- --></A><H3>
scoreGene</H3>
<PRE>
public double[] <B>scoreGene</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
                          double&nbsp;localLambda)
                   throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double)"><!-- --></A><H3>
scoreGene</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A> <B>scoreGene</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
                           <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                           double&nbsp;localLambda)
                    throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree, double, int)"><!-- --></A><H3>
scoreGene</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A> <B>scoreGene</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
                           <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree,
                           double&nbsp;localLambda,
                           int&nbsp;numMarkers)
                    throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreGenes(java.util.Collection, java.lang.String, broad.core.datastructures.IntervalTree)"><!-- --></A><H3>
scoreGenes</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt; <B>scoreGenes</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
                                               java.lang.String&nbsp;chrToUse,
                                               <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)
                                        throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><!-- --></A><H3>
getPeak</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/segmentation/GeneScore.html" title="class in broad.pda.seq.segmentation">GeneScore</A> <B>getPeak</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                         <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon,
                         <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)
                  throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getPeak(nextgen.core.annotation.Gene, nextgen.core.annotation.Gene)"><!-- --></A><H3>
getPeak</H3>
<PRE>
public <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> <B>getPeak</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                    <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;startCodon)
             throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getAlignmentDataModelStats()"><!-- --></A><H3>
getAlignmentDataModelStats</H3>
<PRE>
public <A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.html" title="class in broad.pda.seq.segmentation">AlignmentDataModelStats</A> <B>getAlignmentDataModelStats</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="scoreGene(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><!-- --></A><H3>
scoreGene</H3>
<PRE>
public double[] <B>scoreGene</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;align,
                          <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)
                   throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreGenes(java.util.Collection)"><!-- --></A><H3>
scoreGenes</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt; <B>scoreGenes</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;genes)
                                        throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreGenes(java.util.Collection, java.lang.String)"><!-- --></A><H3>
scoreGenes</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt; <B>scoreGenes</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
                                               java.lang.String&nbsp;chrToUse)
                                        throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreGenes(java.util.Collection, java.lang.String, int)"><!-- --></A><H3>
scoreGenes</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt; <B>scoreGenes</B>(java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;set,
                                               java.lang.String&nbsp;chrToUse,
                                               int&nbsp;size)
                                        throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scoreGenes(broad.pda.annotation.BEDFileParser)"><!-- --></A><H3>
scoreGenes</H3>
<PRE>
public void <B>scoreGenes</B>(<A HREF="../../../../broad/pda/annotation/BEDFileParser.html" title="class in broad.pda.annotation">BEDFileParser</A>&nbsp;genes)
                throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scorePathsForOnlyRPKM(java.util.Collection)"><!-- --></A><H3>
scorePathsForOnlyRPKM</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt; <B>scorePathsForOnlyRPKM</B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="scorePaths(java.util.Collection, double)"><!-- --></A><H3>
scorePaths</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt; <B>scorePaths</B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths,
                                               double&nbsp;lambda)
                                        throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="scorePaths(java.util.Collection, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT, double)"><!-- --></A><H3>
scorePaths</H3>
<PRE>
public static java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt; <B>scorePaths</B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths,
                                                      <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>&nbsp;graph,
                                                      double&nbsp;alpha)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="scorePaths(java.util.Collection, double, double, double, double, java.util.Map, double)"><!-- --></A><H3>
scorePaths</H3>
<PRE>
public static java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt; <B>scorePaths</B>(java.util.Collection&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>&gt;&nbsp;paths,
                                                      double&nbsp;lambda,
                                                      double&nbsp;numMarkers,
                                                      double&nbsp;numReads,
                                                      double&nbsp;RPKMContant,
                                                      java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>,java.lang.Double&gt;&nbsp;localRates,
                                                      double&nbsp;alpha)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="count(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
count</H3>
<PRE>
public double <B>count</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)
             throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="trimMaxContiguous(nextgen.core.annotation.Gene, double)"><!-- --></A><H3>
trimMaxContiguous</H3>
<PRE>
public <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> <B>trimMaxContiguous</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;region,
                              double&nbsp;quantile)
                       throws java.io.IOException</PRE>
<DL>
<DD>Trim a region to a contiguous subregion with max number of positions having coverage above the quantile
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>region</CODE> - The original region<DD><CODE>quantile</CODE> - The coverage quantile of the original region
<DT><B>Returns:</B><DD>The contiguous subregion with maximal number of positions having coverage above the quantile of the original region or null if can't trim
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="calculatePVal(int, double, double, double)"><!-- --></A><H3>
calculatePVal</H3>
<PRE>
public static double <B>calculatePVal</B>(int&nbsp;k,
                                   double&nbsp;lambda,
                                   double&nbsp;w,
                                   double&nbsp;T)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="filterGenesByExpression(java.util.Map)"><!-- --></A><H3>
filterGenesByExpression</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>filterGenesByExpression</B>(java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,java.util.List&lt;java.lang.Double&gt;&gt;&nbsp;countsMap)</PRE>
<DL>
<DD>This function will filter the genes in the input Map by a number of constraints on the total number of reads on the genes
 1. All genes with expression < 5.0 read counts will be removed
 2. All genes with expression in the 75th quantile of the remaining expressed genes will be returned
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>countsMap:</CODE> - A map of RefSeqGene to a List of counts along the length of the gene
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="trimGeneEnds(nextgen.core.annotation.Gene, double)"><!-- --></A><H3>
trimGeneEnds</H3>
<PRE>
public <A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A> <B>trimGeneEnds</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                         double&nbsp;quantile)
                  throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeFullBED(java.lang.String, java.util.Map, double)"><!-- --></A><H3>
writeFullBED</H3>
<PRE>
public static void <B>writeFullBED</B>(java.lang.String&nbsp;save,
                                java.util.Map&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>,double[]&gt;&nbsp;map,
                                double&nbsp;alpha)
                         throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeFullBED(java.lang.String, java.util.Collection)"><!-- --></A><H3>
writeFullBED</H3>
<PRE>
public static void <B>writeFullBED</B>(java.lang.String&nbsp;save,
                                java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt;&nbsp;segments)
                         throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeConnections(java.lang.String, broad.pda.seq.graph.ChromosomeWithBubblesJGraphT)"><!-- --></A><H3>
writeConnections</H3>
<PRE>
public void <B>writeConnections</B>(java.lang.String&nbsp;save,
                             <A HREF="../../../../broad/pda/seq/graph/ChromosomeWithBubblesJGraphT.html" title="class in broad.pda.seq.graph">ChromosomeWithBubblesJGraphT</A>&nbsp;graph)
                      throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="hasDataForChromosome(java.lang.String)"><!-- --></A><H3>
hasDataForChromosome</H3>
<PRE>
public boolean <B>hasDataForChromosome</B>(java.lang.String&nbsp;chr)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="main(java.lang.String[])"><!-- --></A><H3>
main</H3>
<PRE>
public static void <B>main</B>(java.lang.String[]&nbsp;args)
                 throws java.lang.Exception</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="resetTreeCache()"><!-- --></A><H3>
resetTreeCache</H3>
<PRE>
public void <B>resetTreeCache</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="CV(nextgen.core.annotation.Gene, broad.core.datastructures.IntervalTree)"><!-- --></A><H3>
CV</H3>
<PRE>
public double <B>CV</B>(<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&nbsp;gene,
                 <A HREF="../../../../broad/core/datastructures/IntervalTree.html" title="class in broad.core.datastructures">IntervalTree</A>&lt;<A HREF="../../../../nextgen/core/alignment/Alignment.html" title="interface in nextgen.core.alignment">Alignment</A>&gt;&nbsp;tree)
          throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="permuteRegions(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
permuteRegions</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>permuteRegions</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)
                                                throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="rawRegions(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
rawRegions</H3>
<PRE>
public java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&gt; <B>rawRegions</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;align)
                                            throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="filterPaths(java.util.Map, double)"><!-- --></A><H3>
filterPaths</H3>
<PRE>
public static java.util.Collection&lt;<A HREF="../../../../nextgen/core/annotation/Gene.html" title="class in nextgen.core.annotation">Gene</A>&gt; <B>filterPaths</B>(java.util.Map&lt;<A HREF="../../../../broad/pda/seq/graph/Path.html" title="class in broad.pda.seq.graph">Path</A>,double[]&gt;&nbsp;scores,
                                                     double&nbsp;alpha2)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getBasesCovered(nextgen.core.annotation.Annotation)"><!-- --></A><H3>
getBasesCovered</H3>
<PRE>
public int <B>getBasesCovered</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;region)
                    throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getBasesCovered(nextgen.core.annotation.Annotation, int)"><!-- --></A><H3>
getBasesCovered</H3>
<PRE>
public int <B>getBasesCovered</B>(<A HREF="../../../../nextgen/core/annotation/Annotation.html" title="interface in nextgen.core.annotation">Annotation</A>&nbsp;region,
                           int&nbsp;EF)
                    throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="parseMaskFiles(java.io.File[])"><!-- --></A><H3>
parseMaskFiles</H3>
<PRE>
public static java.util.Map&lt;java.lang.String,java.lang.Integer&gt; <B>parseMaskFiles</B>(java.io.File[]&nbsp;files)
                                                                        throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ContinuousDataAlignmentModel.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/AlignmentDataModelStats.ReadCountNormalizerToBaselineAlignment.html" title="class in broad.pda.seq.segmentation"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../broad/pda/seq/segmentation/DefineSourceAndSinkFromPairedEnds.html" title="class in broad.pda.seq.segmentation"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?broad/pda/seq/segmentation/ContinuousDataAlignmentModel.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ContinuousDataAlignmentModel.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
